// @pulumi/aws/odb/network.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Network = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing odb Network resource in AWS for Oracle Database@AWS.
 *
 * ## Example Usage
 *
 * ## Import
 *
 * Using `pulumi import`, import Odb Network using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:odb/network:Network example example
 * ```
 */
class Network extends pulumi.CustomResource {
    /**
     * Get an existing Network resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Network(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:odb/network:Network';
    /**
     * Returns true if the given object is an instance of Network.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Network.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["availabilityZone"] = state?.availabilityZone;
            resourceInputs["availabilityZoneId"] = state?.availabilityZoneId;
            resourceInputs["backupSubnetCidr"] = state?.backupSubnetCidr;
            resourceInputs["clientSubnetCidr"] = state?.clientSubnetCidr;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["crossRegionS3RestoreSourcesAccesses"] = state?.crossRegionS3RestoreSourcesAccesses;
            resourceInputs["customDomainName"] = state?.customDomainName;
            resourceInputs["defaultDnsPrefix"] = state?.defaultDnsPrefix;
            resourceInputs["deleteAssociatedResources"] = state?.deleteAssociatedResources;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["kmsAccess"] = state?.kmsAccess;
            resourceInputs["kmsPolicyDocument"] = state?.kmsPolicyDocument;
            resourceInputs["managedServices"] = state?.managedServices;
            resourceInputs["ociDnsForwardingConfigs"] = state?.ociDnsForwardingConfigs;
            resourceInputs["ociNetworkAnchorId"] = state?.ociNetworkAnchorId;
            resourceInputs["ociNetworkAnchorUrl"] = state?.ociNetworkAnchorUrl;
            resourceInputs["ociResourceAnchorName"] = state?.ociResourceAnchorName;
            resourceInputs["ociVcnId"] = state?.ociVcnId;
            resourceInputs["ociVcnUrl"] = state?.ociVcnUrl;
            resourceInputs["peeredCidrs"] = state?.peeredCidrs;
            resourceInputs["percentProgress"] = state?.percentProgress;
            resourceInputs["region"] = state?.region;
            resourceInputs["s3Access"] = state?.s3Access;
            resourceInputs["s3PolicyDocument"] = state?.s3PolicyDocument;
            resourceInputs["status"] = state?.status;
            resourceInputs["statusReason"] = state?.statusReason;
            resourceInputs["stsAccess"] = state?.stsAccess;
            resourceInputs["stsPolicyDocument"] = state?.stsPolicyDocument;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["zeroEtlAccess"] = state?.zeroEtlAccess;
        }
        else {
            const args = argsOrState;
            if (args?.availabilityZoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'availabilityZoneId'");
            }
            if (args?.backupSubnetCidr === undefined && !opts.urn) {
                throw new Error("Missing required property 'backupSubnetCidr'");
            }
            if (args?.clientSubnetCidr === undefined && !opts.urn) {
                throw new Error("Missing required property 'clientSubnetCidr'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.s3Access === undefined && !opts.urn) {
                throw new Error("Missing required property 's3Access'");
            }
            if (args?.zeroEtlAccess === undefined && !opts.urn) {
                throw new Error("Missing required property 'zeroEtlAccess'");
            }
            resourceInputs["availabilityZone"] = args?.availabilityZone;
            resourceInputs["availabilityZoneId"] = args?.availabilityZoneId;
            resourceInputs["backupSubnetCidr"] = args?.backupSubnetCidr;
            resourceInputs["clientSubnetCidr"] = args?.clientSubnetCidr;
            resourceInputs["crossRegionS3RestoreSourcesAccesses"] = args?.crossRegionS3RestoreSourcesAccesses;
            resourceInputs["customDomainName"] = args?.customDomainName;
            resourceInputs["defaultDnsPrefix"] = args?.defaultDnsPrefix;
            resourceInputs["deleteAssociatedResources"] = args?.deleteAssociatedResources;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["kmsAccess"] = args?.kmsAccess;
            resourceInputs["kmsPolicyDocument"] = args?.kmsPolicyDocument;
            resourceInputs["region"] = args?.region;
            resourceInputs["s3Access"] = args?.s3Access;
            resourceInputs["s3PolicyDocument"] = args?.s3PolicyDocument;
            resourceInputs["stsAccess"] = args?.stsAccess;
            resourceInputs["stsPolicyDocument"] = args?.stsPolicyDocument;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["zeroEtlAccess"] = args?.zeroEtlAccess;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["managedServices"] = undefined /*out*/;
            resourceInputs["ociDnsForwardingConfigs"] = undefined /*out*/;
            resourceInputs["ociNetworkAnchorId"] = undefined /*out*/;
            resourceInputs["ociNetworkAnchorUrl"] = undefined /*out*/;
            resourceInputs["ociResourceAnchorName"] = undefined /*out*/;
            resourceInputs["ociVcnId"] = undefined /*out*/;
            resourceInputs["ociVcnUrl"] = undefined /*out*/;
            resourceInputs["peeredCidrs"] = undefined /*out*/;
            resourceInputs["percentProgress"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusReason"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Network.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Network = Network;
