// @pulumi/azure-native/types/enums/mission/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.TransitOptionType = exports.TransitHubState = exports.ServiceIdentifier = exports.PolicyOverride = exports.PolicyAction = exports.Option = exports.Mode = exports.ManagedServiceIdentityType = exports.Justification = exports.FirewallSKU = exports.Enforcement = exports.EnclaveEndpointProtocol = exports.DiagnosticDestination = exports.DestinationType = exports.CommunityEndpointProtocol = exports.ApprovalStatus = exports.ApprovalPolicy = exports.ActionPerformed = void 0;
exports.ActionPerformed = {
    /**
     * Action was Approved
     */
    Approved: "Approved",
    /**
     * Action was Rejected
     */
    Rejected: "Rejected",
};
exports.ApprovalPolicy = {
    /**
     * Approval will be required for the specified action.
     */
    Required: "Required",
    /**
     * Approval will not be required for the specified action.
     */
    NotRequired: "NotRequired",
};
exports.ApprovalStatus = {
    /**
     * ApprovalStatus Type Approved
     */
    Approved: "Approved",
    /**
     * ApprovalStatus Type Rejected
     */
    Rejected: "Rejected",
    /**
     * ApprovalStatus Type Pending
     */
    Pending: "Pending",
    /**
     * ApprovalStatus Type Deleted
     */
    Deleted: "Deleted",
    /**
     * ApprovalStatus Type Expired
     */
    Expired: "Expired",
};
exports.CommunityEndpointProtocol = {
    /**
     * CommunityEndpointProtocol Type ANY
     */
    ANY: "ANY",
    /**
     * CommunityEndpointProtocol Type TCP
     */
    TCP: "TCP",
    /**
     * CommunityEndpointProtocol Type UDP
     */
    UDP: "UDP",
    /**
     * CommunityEndpointProtocol Type ICMP
     */
    ICMP: "ICMP",
    /**
     * CommunityEndpointProtocol Type ESP
     */
    ESP: "ESP",
    /**
     * CommunityEndpointProtocol Type AH
     */
    AH: "AH",
    /**
     * CommunityEndpointProtocol Type HTTPS
     */
    HTTPS: "HTTPS",
    /**
     * CommunityEndpointProtocol Type HTTP
     */
    HTTP: "HTTP",
};
exports.DestinationType = {
    /**
     * DestinationType Type FQDN
     */
    FQDN: "FQDN",
    /**
     * DestinationType Type FQDNTag
     */
    FQDNTag: "FQDNTag",
    /**
     * DestinationType Type IPAddress
     */
    IPAddress: "IPAddress",
    /**
     * DestinationType Type PrivateNetwork
     */
    PrivateNetwork: "PrivateNetwork",
};
exports.DiagnosticDestination = {
    /**
     * DiagnosticDestination Type CommunityOnly
     */
    CommunityOnly: "CommunityOnly",
    /**
     * DiagnosticDestination Type EnclaveOnly
     */
    EnclaveOnly: "EnclaveOnly",
    /**
     * DiagnosticDestination Type Both
     */
    Both: "Both",
};
exports.EnclaveEndpointProtocol = {
    /**
     * EndpointProtocol Type ANY
     */
    ANY: "ANY",
    /**
     * EndpointProtocol Type TCP
     */
    TCP: "TCP",
    /**
     * EndpointProtocol Type UDP
     */
    UDP: "UDP",
    /**
     * EndpointProtocol Type ICMP
     */
    ICMP: "ICMP",
    /**
     * EndpointProtocol Type ESP
     */
    ESP: "ESP",
    /**
     * EndpointProtocol Type AH
     */
    AH: "AH",
};
exports.Enforcement = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FirewallSKU = {
    /**
     * FirewallSKU Basic
     */
    Basic: "Basic",
    /**
     * FirwallSKU Standard
     */
    Standard: "Standard",
    /**
     * FirewallSKU Premium
     */
    Premium: "Premium",
};
exports.Justification = {
    Networking: "Networking",
    Governance: "Governance",
    Off: "Off",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.Mode = {
    On: "On",
    CanNotDelete: "CanNotDelete",
    Off: "Off",
    General: "General",
    Advanced: "Advanced",
};
exports.Option = {
    Allow: "Allow",
    Deny: "Deny",
    ExceptionOnly: "ExceptionOnly",
    NotApplicable: "NotApplicable",
};
exports.PolicyAction = {
    AuditOnly: "AuditOnly",
    Enforce: "Enforce",
    None: "None",
};
exports.PolicyOverride = {
    Enclave: "Enclave",
    None: "None",
};
exports.ServiceIdentifier = {
    /**
     * Service identifier for AKS
     */
    AKS: "AKS",
    /**
     * Service identifier for App Service
     */
    AppService: "AppService",
    /**
     * Service identifier for Azure Firewalls
     */
    AzureFirewalls: "AzureFirewalls",
    /**
     * Service identifier for Container Registry
     */
    ContainerRegistry: "ContainerRegistry",
    /**
     * Service identifier for CosmosDB
     */
    CosmosDB: "CosmosDB",
    /**
     * Service identifier for Data Connectors
     */
    DataConnectors: "DataConnectors",
    /**
     * Service identifier for Insights
     */
    Insights: "Insights",
    /**
     * Service identifier for Key Vault
     */
    KeyVault: "KeyVault",
    /**
     * Service identifier for Logic
     */
    Logic: "Logic",
    /**
     * Service identifier for Microsoft SQL
     */
    MicrosoftSql: "MicrosoftSQL",
    /**
     * Service identifier for Monitoring
     */
    Monitoring: "Monitoring",
    /**
     * Service identifier for PostgreSQL
     */
    PostgreSql: "PostgreSQL",
    /**
     * Service identifier for Private DNS Zones
     */
    PrivateDNSZones: "PrivateDNSZones",
    /**
     * Service identifier for Service Bus
     */
    ServiceBus: "ServiceBus",
    /**
     * Service identifier for Storage
     */
    Storage: "Storage",
};
exports.TransitHubState = {
    /**
     * TransitHubState Type PendingApproval
     */
    PendingApproval: "PendingApproval",
    /**
     * TransitHubState Type Approved
     */
    Approved: "Approved",
    /**
     * TransitHubState Type PendingUpdate
     */
    PendingUpdate: "PendingUpdate",
    /**
     * TransitHubState Type Active
     */
    Active: "Active",
    /**
     * TransitHubState Type Failed
     */
    Failed: "Failed",
};
exports.TransitOptionType = {
    /**
     * TransitOptionType ExpressRoute
     */
    ExpressRoute: "ExpressRoute",
    /**
     * TransitOptionType Gateway
     */
    Gateway: "Gateway",
    /**
     * TransitOptionType Peering
     */
    Peering: "Peering",
};
exports.Type = {
    User: "User",
    Group: "Group",
    ServicePrincipal: "ServicePrincipal",
};
