// @pulumi/kubernetes/flowcontrol/v1beta1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriorityLevelConfigurationPatch = exports.PriorityLevelConfigurationList = exports.PriorityLevelConfiguration = exports.FlowSchemaPatch = exports.FlowSchemaList = exports.FlowSchema = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.FlowSchema = null;
utilities.lazyLoad(exports, ["FlowSchema"], () => require("./flowSchema"));
exports.FlowSchemaList = null;
utilities.lazyLoad(exports, ["FlowSchemaList"], () => require("./flowSchemaList"));
exports.FlowSchemaPatch = null;
utilities.lazyLoad(exports, ["FlowSchemaPatch"], () => require("./flowSchemaPatch"));
exports.PriorityLevelConfiguration = null;
utilities.lazyLoad(exports, ["PriorityLevelConfiguration"], () => require("./priorityLevelConfiguration"));
exports.PriorityLevelConfigurationList = null;
utilities.lazyLoad(exports, ["PriorityLevelConfigurationList"], () => require("./priorityLevelConfigurationList"));
exports.PriorityLevelConfigurationPatch = null;
utilities.lazyLoad(exports, ["PriorityLevelConfigurationPatch"], () => require("./priorityLevelConfigurationPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:FlowSchema":
                return new exports.FlowSchema(name, undefined, { urn });
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:FlowSchemaList":
                return new exports.FlowSchemaList(name, undefined, { urn });
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:FlowSchemaPatch":
                return new exports.FlowSchemaPatch(name, undefined, { urn });
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:PriorityLevelConfiguration":
                return new exports.PriorityLevelConfiguration(name, undefined, { urn });
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:PriorityLevelConfigurationList":
                return new exports.PriorityLevelConfigurationList(name, undefined, { urn });
            case "kubernetes:flowcontrol.apiserver.k8s.io/v1beta1:PriorityLevelConfigurationPatch":
                return new exports.PriorityLevelConfigurationPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "flowcontrol.apiserver.k8s.io/v1beta1", _module);
