// @pulumi/aws/securityhub/aggregatorV2.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatorV2 = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Security Hub V2 Aggregator, which enables cross-region finding aggregation.
 *
 * > **NOTE:** Security Hub V2 must be enabled (`aws.securityhub.AccountV2`) before creating an aggregator.
 *
 * ## Example Usage
 *
 * ### All Regions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.securityhub.AccountV2("example", {});
 * const exampleAggregatorV2 = new aws.securityhub.AggregatorV2("example", {regionLinkingMode: "ALL_REGIONS"}, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ### Specified Regions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.securityhub.AccountV2("example", {});
 * const exampleAggregatorV2 = new aws.securityhub.AggregatorV2("example", {
 *     regionLinkingMode: "SPECIFIED_REGIONS",
 *     linkedRegions: [
 *         "us-west-2",
 *         "eu-west-1",
 *     ],
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `arn` (String) Amazon Resource Name (ARN) of the Security Hub V2 aggregator.
 *
 * Using `pulumi import`, import Security Hub V2 aggregators using `arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:securityhub/aggregatorV2:AggregatorV2 example arn:aws:securityhub:us-east-1:123456789012:aggregator/v2/example
 * ```
 */
class AggregatorV2 extends pulumi.CustomResource {
    /**
     * Get an existing AggregatorV2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AggregatorV2(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:securityhub/aggregatorV2:AggregatorV2';
    /**
     * Returns true if the given object is an instance of AggregatorV2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AggregatorV2.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aggregationRegion"] = state?.aggregationRegion;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["linkedRegions"] = state?.linkedRegions;
            resourceInputs["region"] = state?.region;
            resourceInputs["regionLinkingMode"] = state?.regionLinkingMode;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.regionLinkingMode === undefined && !opts.urn) {
                throw new Error("Missing required property 'regionLinkingMode'");
            }
            resourceInputs["linkedRegions"] = args?.linkedRegions;
            resourceInputs["region"] = args?.region;
            resourceInputs["regionLinkingMode"] = args?.regionLinkingMode;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["aggregationRegion"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AggregatorV2.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AggregatorV2 = AggregatorV2;
