// @pulumi/aws/securityhub/connectorV2.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorV2 = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Security Hub V2 connector.
 *
 * > **NOTE:** Connectors must be created in the aggregation (home) region. A Security Hub V2 Aggregator (`aws.securityhub.AggregatorV2`) must exist before creating connectors.
 *
 * > **NOTE:** After creation, the connector will be in `PENDING_AUTHORIZATION` status. Use the `authUrl` output to complete the OAuth authorization flow.
 *
 * ## Example Usage
 *
 * ### Jira Cloud
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.securityhub.AccountV2("example", {});
 * const exampleAggregatorV2 = new aws.securityhub.AggregatorV2("example", {regionLinkingMode: "ALL_REGIONS"}, {
 *     dependsOn: [example],
 * });
 * const exampleConnectorV2 = new aws.securityhub.ConnectorV2("example", {
 *     name: "jira-connector",
 *     connectorProvider: {
 *         jiraCloud: {
 *             projectKey: "SEC",
 *         },
 *     },
 * }, {
 *     dependsOn: [exampleAggregatorV2],
 * });
 * export const authUrl = exampleConnectorV2.authUrl;
 * ```
 *
 * ### With Description and KMS Key
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.securityhub.ConnectorV2("example", {
 *     name: "jira-connector",
 *     description: "Jira Cloud integration for security findings",
 *     kmsKeyArn: exampleAwsKmsKey.arn,
 *     connectorProvider: {
 *         jiraCloud: {
 *             projectKey: "SEC",
 *         },
 *     },
 * }, {
 *     dependsOn: [exampleAwsSecurityhubAggregatorV2],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `connectorId` (String) ID of the Security Hub V2 connector.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Security Hub V2 connectors using `connectorId`. For example:
 *
 * ```sh
 * $ pulumi import aws:securityhub/connectorV2:ConnectorV2 example 8ecf045f-5a95-c24d-6769-5d52f6929563
 * ```
 */
class ConnectorV2 extends pulumi.CustomResource {
    /**
     * Get an existing ConnectorV2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ConnectorV2(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:securityhub/connectorV2:ConnectorV2';
    /**
     * Returns true if the given object is an instance of ConnectorV2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectorV2.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["connectorId"] = state?.connectorId;
            resourceInputs["connectorProvider"] = state?.connectorProvider;
            resourceInputs["description"] = state?.description;
            resourceInputs["healths"] = state?.healths;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.connectorProvider === undefined && !opts.urn) {
                throw new Error("Missing required property 'connectorProvider'");
            }
            resourceInputs["connectorProvider"] = args?.connectorProvider;
            resourceInputs["description"] = args?.description;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["connectorId"] = undefined /*out*/;
            resourceInputs["healths"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ConnectorV2.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConnectorV2 = ConnectorV2;
