// @pulumi/aws/iot/thingGroupMembership.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThingGroupMembership = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Adds an IoT Thing to an IoT Thing Group.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.iot.ThingGroupMembership("example", {
 *     thingName: "example-thing",
 *     thingGroupName: "example-group",
 *     overrideDynamicGroup: true,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import IoT Thing Group Membership using the thing group name and thing name. For example:
 *
 * ```sh
 * $ pulumi import aws:iot/thingGroupMembership:ThingGroupMembership example thing_group_name/thing_name
 * ```
 */
class ThingGroupMembership extends pulumi.CustomResource {
    /**
     * Get an existing ThingGroupMembership resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ThingGroupMembership(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:iot/thingGroupMembership:ThingGroupMembership';
    /**
     * Returns true if the given object is an instance of ThingGroupMembership.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ThingGroupMembership.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["overrideDynamicGroup"] = state?.overrideDynamicGroup;
            resourceInputs["region"] = state?.region;
            resourceInputs["thingGroupName"] = state?.thingGroupName;
            resourceInputs["thingName"] = state?.thingName;
        }
        else {
            const args = argsOrState;
            if (args?.thingGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'thingGroupName'");
            }
            if (args?.thingName === undefined && !opts.urn) {
                throw new Error("Missing required property 'thingName'");
            }
            resourceInputs["overrideDynamicGroup"] = args?.overrideDynamicGroup;
            resourceInputs["region"] = args?.region;
            resourceInputs["thingGroupName"] = args?.thingGroupName;
            resourceInputs["thingName"] = args?.thingName;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ThingGroupMembership.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ThingGroupMembership = ThingGroupMembership;
