// @pulumi/gcp/apphub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workload = exports.ServiceProjectAttachment = exports.Service = exports.getDiscoveredWorkloadOutput = exports.getDiscoveredWorkload = exports.getDiscoveredServiceOutput = exports.getDiscoveredService = exports.getApplicationOutput = exports.getApplication = exports.Boundary = exports.Application = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.Boundary = null;
utilities.lazyLoad(exports, ["Boundary"], () => require("./boundary"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getDiscoveredService = null;
exports.getDiscoveredServiceOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredService", "getDiscoveredServiceOutput"], () => require("./getDiscoveredService"));
exports.getDiscoveredWorkload = null;
exports.getDiscoveredWorkloadOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredWorkload", "getDiscoveredWorkloadOutput"], () => require("./getDiscoveredWorkload"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceProjectAttachment = null;
utilities.lazyLoad(exports, ["ServiceProjectAttachment"], () => require("./serviceProjectAttachment"));
exports.Workload = null;
utilities.lazyLoad(exports, ["Workload"], () => require("./workload"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:apphub/application:Application":
                return new exports.Application(name, undefined, { urn });
            case "gcp:apphub/boundary:Boundary":
                return new exports.Boundary(name, undefined, { urn });
            case "gcp:apphub/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "gcp:apphub/serviceProjectAttachment:ServiceProjectAttachment":
                return new exports.ServiceProjectAttachment(name, undefined, { urn });
            case "gcp:apphub/workload:Workload":
                return new exports.Workload(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "apphub/application", _module);
pulumi.runtime.registerResourceModule("gcp", "apphub/boundary", _module);
pulumi.runtime.registerResourceModule("gcp", "apphub/service", _module);
pulumi.runtime.registerResourceModule("gcp", "apphub/serviceProjectAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "apphub/workload", _module);
