// @pulumi/aws/cloudwatch/alarmMuteRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlarmMuteRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS CloudWatch Alarm Mute Rule.
 *
 * ## Example Usage
 *
 * ### Basic Usage with Cron Expression
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudwatch.AlarmMuteRule("example", {
 *     name: "example",
 *     rule: {
 *         schedule: {
 *             duration: "PT4H",
 *             expression: "cron(0 2 * * *)",
 *         },
 *     },
 * });
 * ```
 *
 * ### With Start/Expire Dates Option
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.index.CloudwatchAlarm("example", {
 *     alarmName: "example",
 *     comparisonOperator: "GreaterThanThreshold",
 *     evaluationPeriods: 2,
 *     metricName: "CPUUtilization",
 *     namespace: "AWS/EC2",
 *     period: 120,
 *     statistic: "Average",
 *     threshold: 80,
 * });
 * const exampleAlarmMuteRule = new aws.cloudwatch.AlarmMuteRule("example", {
 *     name: "example",
 *     description: "Mute alarms during maintenance window",
 *     startDate: "2026-01-01T00:00:00Z",
 *     expireDate: "2026-12-31T23:59:00Z",
 *     rule: {
 *         schedule: {
 *             duration: "PT4H",
 *             expression: "cron(0 2 * * *)",
 *             timezone: "Asia/Tokyo",
 *         },
 *     },
 *     muteTargets: {
 *         alarmNames: [example.alarmName],
 *     },
 *     tags: {
 *         Environment: "production",
 *     },
 * });
 * ```
 *
 * ### With At Expression
 *
 * > **NOTE:** When using `at()` expressions, do not set `startDate` or `expireDate`. The CloudWatch API returns the error `Can not set start or expire dates for At expressions`.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudwatch.AlarmMuteRule("example", {
 *     name: "example",
 *     rule: {
 *         schedule: {
 *             duration: "PT4H",
 *             expression: "at(2026-12-31T23:59:59)",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `name` (String) Name of the CloudWatch Alarm Mute Rule.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import CloudWatch Alarm Mute Rule using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:cloudwatch/alarmMuteRule:AlarmMuteRule example example
 * ```
 */
class AlarmMuteRule extends pulumi.CustomResource {
    /**
     * Get an existing AlarmMuteRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AlarmMuteRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:cloudwatch/alarmMuteRule:AlarmMuteRule';
    /**
     * Returns true if the given object is an instance of AlarmMuteRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AlarmMuteRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["expireDate"] = state?.expireDate;
            resourceInputs["lastUpdatedTimestamp"] = state?.lastUpdatedTimestamp;
            resourceInputs["muteTargets"] = state?.muteTargets;
            resourceInputs["muteType"] = state?.muteType;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["rule"] = state?.rule;
            resourceInputs["startDate"] = state?.startDate;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            resourceInputs["description"] = args?.description;
            resourceInputs["expireDate"] = args?.expireDate;
            resourceInputs["muteTargets"] = args?.muteTargets;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["rule"] = args?.rule;
            resourceInputs["startDate"] = args?.startDate;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["lastUpdatedTimestamp"] = undefined /*out*/;
            resourceInputs["muteType"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AlarmMuteRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AlarmMuteRule = AlarmMuteRule;
