// @pulumi/aws/vpclattice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetGroupAttachment = exports.TargetGroup = exports.ServiceNetworkVpcAssociation = exports.ServiceNetworkServiceAssociation = exports.ServiceNetworkResourceAssociation = exports.ServiceNetwork = exports.Service = exports.ResourcePolicy = exports.ResourceGateway = exports.ResourceConfiguration = exports.ListenerRule = exports.Listener = exports.getServiceNetworkOutput = exports.getServiceNetwork = exports.getServiceOutput = exports.getService = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getListenerOutput = exports.getListener = exports.getAuthPolicyOutput = exports.getAuthPolicy = exports.DomainVerification = exports.AuthPolicy = exports.AccessLogSubscription = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccessLogSubscription = null;
utilities.lazyLoad(exports, ["AccessLogSubscription"], () => require("./accessLogSubscription"));
exports.AuthPolicy = null;
utilities.lazyLoad(exports, ["AuthPolicy"], () => require("./authPolicy"));
exports.DomainVerification = null;
utilities.lazyLoad(exports, ["DomainVerification"], () => require("./domainVerification"));
exports.getAuthPolicy = null;
exports.getAuthPolicyOutput = null;
utilities.lazyLoad(exports, ["getAuthPolicy", "getAuthPolicyOutput"], () => require("./getAuthPolicy"));
exports.getListener = null;
exports.getListenerOutput = null;
utilities.lazyLoad(exports, ["getListener", "getListenerOutput"], () => require("./getListener"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceNetwork = null;
exports.getServiceNetworkOutput = null;
utilities.lazyLoad(exports, ["getServiceNetwork", "getServiceNetworkOutput"], () => require("./getServiceNetwork"));
exports.Listener = null;
utilities.lazyLoad(exports, ["Listener"], () => require("./listener"));
exports.ListenerRule = null;
utilities.lazyLoad(exports, ["ListenerRule"], () => require("./listenerRule"));
exports.ResourceConfiguration = null;
utilities.lazyLoad(exports, ["ResourceConfiguration"], () => require("./resourceConfiguration"));
exports.ResourceGateway = null;
utilities.lazyLoad(exports, ["ResourceGateway"], () => require("./resourceGateway"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceNetwork = null;
utilities.lazyLoad(exports, ["ServiceNetwork"], () => require("./serviceNetwork"));
exports.ServiceNetworkResourceAssociation = null;
utilities.lazyLoad(exports, ["ServiceNetworkResourceAssociation"], () => require("./serviceNetworkResourceAssociation"));
exports.ServiceNetworkServiceAssociation = null;
utilities.lazyLoad(exports, ["ServiceNetworkServiceAssociation"], () => require("./serviceNetworkServiceAssociation"));
exports.ServiceNetworkVpcAssociation = null;
utilities.lazyLoad(exports, ["ServiceNetworkVpcAssociation"], () => require("./serviceNetworkVpcAssociation"));
exports.TargetGroup = null;
utilities.lazyLoad(exports, ["TargetGroup"], () => require("./targetGroup"));
exports.TargetGroupAttachment = null;
utilities.lazyLoad(exports, ["TargetGroupAttachment"], () => require("./targetGroupAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:vpclattice/accessLogSubscription:AccessLogSubscription":
                return new exports.AccessLogSubscription(name, undefined, { urn });
            case "aws:vpclattice/authPolicy:AuthPolicy":
                return new exports.AuthPolicy(name, undefined, { urn });
            case "aws:vpclattice/domainVerification:DomainVerification":
                return new exports.DomainVerification(name, undefined, { urn });
            case "aws:vpclattice/listener:Listener":
                return new exports.Listener(name, undefined, { urn });
            case "aws:vpclattice/listenerRule:ListenerRule":
                return new exports.ListenerRule(name, undefined, { urn });
            case "aws:vpclattice/resourceConfiguration:ResourceConfiguration":
                return new exports.ResourceConfiguration(name, undefined, { urn });
            case "aws:vpclattice/resourceGateway:ResourceGateway":
                return new exports.ResourceGateway(name, undefined, { urn });
            case "aws:vpclattice/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:vpclattice/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "aws:vpclattice/serviceNetwork:ServiceNetwork":
                return new exports.ServiceNetwork(name, undefined, { urn });
            case "aws:vpclattice/serviceNetworkResourceAssociation:ServiceNetworkResourceAssociation":
                return new exports.ServiceNetworkResourceAssociation(name, undefined, { urn });
            case "aws:vpclattice/serviceNetworkServiceAssociation:ServiceNetworkServiceAssociation":
                return new exports.ServiceNetworkServiceAssociation(name, undefined, { urn });
            case "aws:vpclattice/serviceNetworkVpcAssociation:ServiceNetworkVpcAssociation":
                return new exports.ServiceNetworkVpcAssociation(name, undefined, { urn });
            case "aws:vpclattice/targetGroup:TargetGroup":
                return new exports.TargetGroup(name, undefined, { urn });
            case "aws:vpclattice/targetGroupAttachment:TargetGroupAttachment":
                return new exports.TargetGroupAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "vpclattice/accessLogSubscription", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/authPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/domainVerification", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/listener", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/listenerRule", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/resourceConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/resourceGateway", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/service", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetwork", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetworkResourceAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetworkServiceAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/serviceNetworkVpcAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/targetGroup", _module);
pulumi.runtime.registerResourceModule("aws", "vpclattice/targetGroupAttachment", _module);
