// @pulumi/aws/vpclattice/resourceConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceConfiguration = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS VPC Lattice Resource Configuration.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.ResourceConfiguration("example", {
 *     name: "Example",
 *     resourceGatewayIdentifier: exampleAwsVpclatticeResourceGateway.id,
 *     portRanges: ["80"],
 *     protocol: "TCP",
 *     resourceConfigurationDefinition: {
 *         dnsResource: {
 *             domainName: "example.com",
 *             ipAddressType: "IPV4",
 *         },
 *     },
 *     tags: {
 *         Environment: "Example",
 *     },
 * });
 * ```
 *
 * ### IP Address Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.ResourceConfiguration("example", {
 *     name: "Example",
 *     resourceGatewayIdentifier: exampleAwsVpclatticeResourceGateway.id,
 *     portRanges: ["80"],
 *     protocol: "TCP",
 *     resourceConfigurationDefinition: {
 *         ipResource: {
 *             ipAddress: "10.0.0.1",
 *         },
 *     },
 *     tags: {
 *         Environment: "Example",
 *     },
 * });
 * ```
 *
 * ### With custom domain
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.DomainVerification("example", {domainName: "example.com"});
 * const exampleResourceConfiguration = new aws.vpclattice.ResourceConfiguration("example", {
 *     name: "Example",
 *     resourceGatewayIdentifier: exampleAwsVpclatticeResourceGateway.id,
 *     customDomainName: "custom.example.com",
 *     domainVerificationId: example.id,
 *     portRanges: ["443"],
 *     protocol: "TCP",
 *     resourceConfigurationDefinition: {
 *         dnsResource: {
 *             domainName: "test.example.com",
 *             ipAddressType: "IPV4",
 *         },
 *     },
 *     tags: {
 *         Environment: "Example",
 *     },
 * });
 * ```
 *
 * ### ARN Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.vpclattice.ResourceConfiguration("test", {
 *     name: "Example",
 *     resourceGatewayIdentifier: testAwsVpclatticeResourceGateway.id,
 *     type: "ARN",
 *     resourceConfigurationDefinition: {
 *         arnResource: {
 *             arn: example.arn,
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import VPC Lattice Resource Configuration using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:vpclattice/resourceConfiguration:ResourceConfiguration example rcfg-1234567890abcdef1
 * ```
 */
class ResourceConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing ResourceConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceConfiguration(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:vpclattice/resourceConfiguration:ResourceConfiguration';
    /**
     * Returns true if the given object is an instance of ResourceConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceConfiguration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowAssociationToShareableServiceNetwork"] = state?.allowAssociationToShareableServiceNetwork;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["customDomainName"] = state?.customDomainName;
            resourceInputs["domainVerificationArn"] = state?.domainVerificationArn;
            resourceInputs["domainVerificationId"] = state?.domainVerificationId;
            resourceInputs["domainVerificationStatus"] = state?.domainVerificationStatus;
            resourceInputs["name"] = state?.name;
            resourceInputs["portRanges"] = state?.portRanges;
            resourceInputs["protocol"] = state?.protocol;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceConfigurationDefinition"] = state?.resourceConfigurationDefinition;
            resourceInputs["resourceConfigurationGroupId"] = state?.resourceConfigurationGroupId;
            resourceInputs["resourceGatewayIdentifier"] = state?.resourceGatewayIdentifier;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            resourceInputs["allowAssociationToShareableServiceNetwork"] = args?.allowAssociationToShareableServiceNetwork;
            resourceInputs["customDomainName"] = args?.customDomainName;
            resourceInputs["domainVerificationId"] = args?.domainVerificationId;
            resourceInputs["name"] = args?.name;
            resourceInputs["portRanges"] = args?.portRanges;
            resourceInputs["protocol"] = args?.protocol;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceConfigurationDefinition"] = args?.resourceConfigurationDefinition;
            resourceInputs["resourceConfigurationGroupId"] = args?.resourceConfigurationGroupId;
            resourceInputs["resourceGatewayIdentifier"] = args?.resourceGatewayIdentifier;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["type"] = args?.type;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["domainVerificationArn"] = undefined /*out*/;
            resourceInputs["domainVerificationStatus"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResourceConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResourceConfiguration = ResourceConfiguration;
