// @pulumi/aws/vpclattice/serviceNetworkResourceAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceNetworkResourceAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS VPC Lattice Service Network Resource Association.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.ServiceNetworkResourceAssociation("example", {
 *     resourceConfigurationIdentifier: exampleAwsVpclatticeResourceConfiguration.id,
 *     serviceNetworkIdentifier: exampleAwsVpclatticeServiceNetwork.id,
 *     tags: {
 *         Name: "Example",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import VPC Lattice Service Network Resource Association using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:vpclattice/serviceNetworkResourceAssociation:ServiceNetworkResourceAssociation example snra-1234567890abcef12
 * ```
 */
class ServiceNetworkResourceAssociation extends pulumi.CustomResource {
    /**
     * Get an existing ServiceNetworkResourceAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceNetworkResourceAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:vpclattice/serviceNetworkResourceAssociation:ServiceNetworkResourceAssociation';
    /**
     * Returns true if the given object is an instance of ServiceNetworkResourceAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceNetworkResourceAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["dnsEntries"] = state?.dnsEntries;
            resourceInputs["privateDnsEnabled"] = state?.privateDnsEnabled;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceConfigurationIdentifier"] = state?.resourceConfigurationIdentifier;
            resourceInputs["serviceNetworkIdentifier"] = state?.serviceNetworkIdentifier;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.resourceConfigurationIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceConfigurationIdentifier'");
            }
            if (args?.serviceNetworkIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceNetworkIdentifier'");
            }
            resourceInputs["privateDnsEnabled"] = args?.privateDnsEnabled;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceConfigurationIdentifier"] = args?.resourceConfigurationIdentifier;
            resourceInputs["serviceNetworkIdentifier"] = args?.serviceNetworkIdentifier;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["dnsEntries"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServiceNetworkResourceAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceNetworkResourceAssociation = ServiceNetworkResourceAssociation;
