// @pulumi/azure-native/cosmosdb/databaseAccountGremlinGraph.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountGremlinGraph = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * An Azure Cosmos DB Gremlin graph.
 *
 * Uses Azure REST API version 2016-03-31.
 *
 * Other available API versions: 2015-04-01, 2015-04-08, 2015-11-06, 2016-03-19. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DatabaseAccountGremlinGraph extends pulumi.CustomResource {
    /**
     * Get an existing DatabaseAccountGremlinGraph resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseAccountGremlinGraph(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cosmosdb:DatabaseAccountGremlinGraph';
    /**
     * Returns true if the given object is an instance of DatabaseAccountGremlinGraph.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountGremlinGraph.__pulumiType;
    }
    /**
     * Create a DatabaseAccountGremlinGraph resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.databaseName === undefined && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if (args?.options === undefined && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if (args?.resource === undefined && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["databaseName"] = args?.databaseName;
            resourceInputs["graphName"] = args?.graphName;
            resourceInputs["options"] = args?.options;
            resourceInputs["resource"] = args ? (args.resource ? pulumi.output(args.resource).apply(inputs.cosmosdb.gremlinGraphResourceArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["conflictResolutionPolicy"] = undefined /*out*/;
            resourceInputs["defaultTtl"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["indexingPolicy"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partitionKey"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["conflictResolutionPolicy"] = undefined /*out*/;
            resourceInputs["defaultTtl"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["indexingPolicy"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partitionKey"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20150401:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20150408:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20151106:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20160319:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20160331:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20190801:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20191212:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20200301:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20200401:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20200601preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20200901:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210115:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210301preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210315:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210401preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210415:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210515:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210615:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20210701preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20211015:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20211015preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20211115preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20220215preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20220515:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20220515preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20220815:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20220815preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20221115:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20221115preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230301preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230315:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230315preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230415:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230915:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20230915preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20231115:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20231115preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20240215preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20240515:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20240515preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20240815:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20240901preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20241115:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20241201preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20250415:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20250501preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20251015:DatabaseAccountGremlinGraph" }, { type: "azure-native:cosmosdb/v20251101preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20230315preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20230415:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20230915:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20230915preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20231115:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20231115preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20240215preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20240515:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20240515preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20240815:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20240901preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20241115:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20241201preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb:GremlinResourceGremlinGraph" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountGremlinGraph.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DatabaseAccountGremlinGraph = DatabaseAccountGremlinGraph;
