// @pulumi/azure-native/cosmosdb/databaseAccountGremlinDatabase.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountGremlinDatabase = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An Azure Cosmos DB Gremlin database.
 *
 * Uses Azure REST API version 2016-03-31.
 *
 * Other available API versions: 2015-04-01, 2015-04-08, 2015-11-06, 2016-03-19. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DatabaseAccountGremlinDatabase extends pulumi.CustomResource {
    /**
     * Get an existing DatabaseAccountGremlinDatabase resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseAccountGremlinDatabase(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cosmosdb:DatabaseAccountGremlinDatabase';
    /**
     * Returns true if the given object is an instance of DatabaseAccountGremlinDatabase.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountGremlinDatabase.__pulumiType;
    }
    /**
     * Create a DatabaseAccountGremlinDatabase resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.options === undefined && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if (args?.resource === undefined && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["databaseName"] = args?.databaseName;
            resourceInputs["options"] = args?.options;
            resourceInputs["resource"] = args?.resource;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20150401:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20150408:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20151106:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20160319:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20160331:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20190801:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20191212:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20200301:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20200401:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20200601preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20200901:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210115:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210301preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210315:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210401preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210415:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210515:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210615:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20210701preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20211015:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20211015preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20211115preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20220215preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20220515:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20220515preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20220815:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20220815preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20221115:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20221115preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230301preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230315:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230315preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230415:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230915:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20230915preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20231115:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20231115preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20240215preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20240515:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20240515preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20240815:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20240901preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20241115:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20241201preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20250415:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20250501preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20251015:DatabaseAccountGremlinDatabase" }, { type: "azure-native:cosmosdb/v20251101preview:DatabaseAccountGremlinDatabase" }, { type: "azure-native:documentdb/v20230315preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20230415:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20230915:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20230915preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20231115:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20231115preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20240215preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20240515:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20240515preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20240815:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20240901preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20241115:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb/v20241201preview:GremlinResourceGremlinDatabase" }, { type: "azure-native:documentdb:GremlinResourceGremlinDatabase" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountGremlinDatabase.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DatabaseAccountGremlinDatabase = DatabaseAccountGremlinDatabase;
