// @pulumi/azure-native/cosmosdb/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDatabaseAccountMongoDBDatabaseOutput = exports.getDatabaseAccountMongoDBDatabase = exports.getDatabaseAccountMongoDBCollectionOutput = exports.getDatabaseAccountMongoDBCollection = exports.getDatabaseAccountGremlinGraphOutput = exports.getDatabaseAccountGremlinGraph = exports.getDatabaseAccountGremlinDatabaseOutput = exports.getDatabaseAccountGremlinDatabase = exports.getDatabaseAccountCassandraTableOutput = exports.getDatabaseAccountCassandraTable = exports.getDatabaseAccountCassandraKeyspaceOutput = exports.getDatabaseAccountCassandraKeyspace = exports.getDatabaseAccountOutput = exports.getDatabaseAccount = exports.getCassandraResourceCassandraViewOutput = exports.getCassandraResourceCassandraView = exports.getCassandraResourceCassandraTableOutput = exports.getCassandraResourceCassandraTable = exports.getCassandraResourceCassandraRoleDefinitionOutput = exports.getCassandraResourceCassandraRoleDefinition = exports.getCassandraResourceCassandraRoleAssignmentOutput = exports.getCassandraResourceCassandraRoleAssignment = exports.getCassandraResourceCassandraKeyspaceOutput = exports.getCassandraResourceCassandraKeyspace = exports.getCassandraDataCenterOutput = exports.getCassandraDataCenter = exports.getCassandraClusterOutput = exports.getCassandraCluster = exports.GarnetCluster = exports.FleetspaceAccount = exports.Fleetspace = exports.FleetAnalytic = exports.Fleet = exports.DatabaseAccountTable = exports.DatabaseAccountSqlDatabase = exports.DatabaseAccountSqlContainer = exports.DatabaseAccountMongoDBDatabase = exports.DatabaseAccountMongoDBCollection = exports.DatabaseAccountGremlinGraph = exports.DatabaseAccountGremlinDatabase = exports.DatabaseAccountCassandraTable = exports.DatabaseAccountCassandraKeyspace = exports.DatabaseAccount = exports.CassandraResourceCassandraView = exports.CassandraResourceCassandraTable = exports.CassandraResourceCassandraRoleDefinition = exports.CassandraResourceCassandraRoleAssignment = exports.CassandraResourceCassandraKeyspace = exports.CassandraDataCenter = exports.CassandraCluster = void 0;
exports.getSqlResourceSqlContainerOutput = exports.getSqlResourceSqlContainer = exports.getServiceOutput = exports.getService = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNotebookWorkspaceOutput = exports.getNotebookWorkspace = exports.getMongoMIResourceMongoMIRoleDefinitionOutput = exports.getMongoMIResourceMongoMIRoleDefinition = exports.getMongoMIResourceMongoMIRoleAssignmentOutput = exports.getMongoMIResourceMongoMIRoleAssignment = exports.getMongoDBResourceMongoUserDefinitionOutput = exports.getMongoDBResourceMongoUserDefinition = exports.getMongoDBResourceMongoRoleDefinitionOutput = exports.getMongoDBResourceMongoRoleDefinition = exports.getMongoDBResourceMongoDBDatabaseOutput = exports.getMongoDBResourceMongoDBDatabase = exports.getMongoDBResourceMongoDBCollectionOutput = exports.getMongoDBResourceMongoDBCollection = exports.getMongoClusterFirewallRuleOutput = exports.getMongoClusterFirewallRule = exports.getMongoClusterOutput = exports.getMongoCluster = exports.getGremlinResourceGremlinRoleDefinitionOutput = exports.getGremlinResourceGremlinRoleDefinition = exports.getGremlinResourceGremlinRoleAssignmentOutput = exports.getGremlinResourceGremlinRoleAssignment = exports.getGremlinResourceGremlinGraphOutput = exports.getGremlinResourceGremlinGraph = exports.getGremlinResourceGremlinDatabaseOutput = exports.getGremlinResourceGremlinDatabase = exports.getGraphResourceGraphOutput = exports.getGraphResourceGraph = exports.getGarnetClusterOutput = exports.getGarnetCluster = exports.getFleetspaceAccountOutput = exports.getFleetspaceAccount = exports.getFleetspaceOutput = exports.getFleetspace = exports.getFleetAnalyticOutput = exports.getFleetAnalytic = exports.getFleetOutput = exports.getFleet = exports.getDatabaseAccountTableOutput = exports.getDatabaseAccountTable = exports.getDatabaseAccountSqlDatabaseOutput = exports.getDatabaseAccountSqlDatabase = exports.getDatabaseAccountSqlContainerOutput = exports.getDatabaseAccountSqlContainer = void 0;
exports.SqlResourceSqlRoleDefinition = exports.SqlResourceSqlRoleAssignment = exports.SqlResourceSqlDatabase = exports.SqlResourceSqlContainer = exports.Service = exports.PrivateEndpointConnection = exports.NotebookWorkspace = exports.MongoMIResourceMongoMIRoleDefinition = exports.MongoMIResourceMongoMIRoleAssignment = exports.MongoDBResourceMongoUserDefinition = exports.MongoDBResourceMongoRoleDefinition = exports.MongoDBResourceMongoDBDatabase = exports.MongoDBResourceMongoDBCollection = exports.MongoClusterFirewallRule = exports.MongoCluster = exports.listNotebookWorkspaceConnectionInfoOutput = exports.listNotebookWorkspaceConnectionInfo = exports.listMongoClusterConnectionStringsOutput = exports.listMongoClusterConnectionStrings = exports.listDatabaseAccountKeysOutput = exports.listDatabaseAccountKeys = exports.listDatabaseAccountConnectionStringsOutput = exports.listDatabaseAccountConnectionStrings = exports.GremlinResourceGremlinRoleDefinition = exports.GremlinResourceGremlinRoleAssignment = exports.GremlinResourceGremlinGraph = exports.GremlinResourceGremlinDatabase = exports.GraphResourceGraph = exports.getThroughputPoolAccountOutput = exports.getThroughputPoolAccount = exports.getThroughputPoolOutput = exports.getThroughputPool = exports.getTableResourceTableRoleDefinitionOutput = exports.getTableResourceTableRoleDefinition = exports.getTableResourceTableRoleAssignmentOutput = exports.getTableResourceTableRoleAssignment = exports.getTableResourceTableOutput = exports.getTableResourceTable = exports.getSqlResourceSqlUserDefinedFunctionOutput = exports.getSqlResourceSqlUserDefinedFunction = exports.getSqlResourceSqlTriggerOutput = exports.getSqlResourceSqlTrigger = exports.getSqlResourceSqlStoredProcedureOutput = exports.getSqlResourceSqlStoredProcedure = exports.getSqlResourceSqlRoleDefinitionOutput = exports.getSqlResourceSqlRoleDefinition = exports.getSqlResourceSqlRoleAssignmentOutput = exports.getSqlResourceSqlRoleAssignment = exports.getSqlResourceSqlDatabaseOutput = exports.getSqlResourceSqlDatabase = void 0;
exports.ThroughputPoolAccount = exports.ThroughputPool = exports.TableResourceTableRoleDefinition = exports.TableResourceTableRoleAssignment = exports.TableResourceTable = exports.SqlResourceSqlUserDefinedFunction = exports.SqlResourceSqlTrigger = exports.SqlResourceSqlStoredProcedure = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CassandraCluster = null;
utilities.lazyLoad(exports, ["CassandraCluster"], () => require("./cassandraCluster"));
exports.CassandraDataCenter = null;
utilities.lazyLoad(exports, ["CassandraDataCenter"], () => require("./cassandraDataCenter"));
exports.CassandraResourceCassandraKeyspace = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraKeyspace"], () => require("./cassandraResourceCassandraKeyspace"));
exports.CassandraResourceCassandraRoleAssignment = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraRoleAssignment"], () => require("./cassandraResourceCassandraRoleAssignment"));
exports.CassandraResourceCassandraRoleDefinition = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraRoleDefinition"], () => require("./cassandraResourceCassandraRoleDefinition"));
exports.CassandraResourceCassandraTable = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraTable"], () => require("./cassandraResourceCassandraTable"));
exports.CassandraResourceCassandraView = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraView"], () => require("./cassandraResourceCassandraView"));
exports.DatabaseAccount = null;
utilities.lazyLoad(exports, ["DatabaseAccount"], () => require("./databaseAccount"));
exports.DatabaseAccountCassandraKeyspace = null;
utilities.lazyLoad(exports, ["DatabaseAccountCassandraKeyspace"], () => require("./databaseAccountCassandraKeyspace"));
exports.DatabaseAccountCassandraTable = null;
utilities.lazyLoad(exports, ["DatabaseAccountCassandraTable"], () => require("./databaseAccountCassandraTable"));
exports.DatabaseAccountGremlinDatabase = null;
utilities.lazyLoad(exports, ["DatabaseAccountGremlinDatabase"], () => require("./databaseAccountGremlinDatabase"));
exports.DatabaseAccountGremlinGraph = null;
utilities.lazyLoad(exports, ["DatabaseAccountGremlinGraph"], () => require("./databaseAccountGremlinGraph"));
exports.DatabaseAccountMongoDBCollection = null;
utilities.lazyLoad(exports, ["DatabaseAccountMongoDBCollection"], () => require("./databaseAccountMongoDBCollection"));
exports.DatabaseAccountMongoDBDatabase = null;
utilities.lazyLoad(exports, ["DatabaseAccountMongoDBDatabase"], () => require("./databaseAccountMongoDBDatabase"));
exports.DatabaseAccountSqlContainer = null;
utilities.lazyLoad(exports, ["DatabaseAccountSqlContainer"], () => require("./databaseAccountSqlContainer"));
exports.DatabaseAccountSqlDatabase = null;
utilities.lazyLoad(exports, ["DatabaseAccountSqlDatabase"], () => require("./databaseAccountSqlDatabase"));
exports.DatabaseAccountTable = null;
utilities.lazyLoad(exports, ["DatabaseAccountTable"], () => require("./databaseAccountTable"));
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.FleetAnalytic = null;
utilities.lazyLoad(exports, ["FleetAnalytic"], () => require("./fleetAnalytic"));
exports.Fleetspace = null;
utilities.lazyLoad(exports, ["Fleetspace"], () => require("./fleetspace"));
exports.FleetspaceAccount = null;
utilities.lazyLoad(exports, ["FleetspaceAccount"], () => require("./fleetspaceAccount"));
exports.GarnetCluster = null;
utilities.lazyLoad(exports, ["GarnetCluster"], () => require("./garnetCluster"));
exports.getCassandraCluster = null;
exports.getCassandraClusterOutput = null;
utilities.lazyLoad(exports, ["getCassandraCluster", "getCassandraClusterOutput"], () => require("./getCassandraCluster"));
exports.getCassandraDataCenter = null;
exports.getCassandraDataCenterOutput = null;
utilities.lazyLoad(exports, ["getCassandraDataCenter", "getCassandraDataCenterOutput"], () => require("./getCassandraDataCenter"));
exports.getCassandraResourceCassandraKeyspace = null;
exports.getCassandraResourceCassandraKeyspaceOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraKeyspace", "getCassandraResourceCassandraKeyspaceOutput"], () => require("./getCassandraResourceCassandraKeyspace"));
exports.getCassandraResourceCassandraRoleAssignment = null;
exports.getCassandraResourceCassandraRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraRoleAssignment", "getCassandraResourceCassandraRoleAssignmentOutput"], () => require("./getCassandraResourceCassandraRoleAssignment"));
exports.getCassandraResourceCassandraRoleDefinition = null;
exports.getCassandraResourceCassandraRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraRoleDefinition", "getCassandraResourceCassandraRoleDefinitionOutput"], () => require("./getCassandraResourceCassandraRoleDefinition"));
exports.getCassandraResourceCassandraTable = null;
exports.getCassandraResourceCassandraTableOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraTable", "getCassandraResourceCassandraTableOutput"], () => require("./getCassandraResourceCassandraTable"));
exports.getCassandraResourceCassandraView = null;
exports.getCassandraResourceCassandraViewOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraView", "getCassandraResourceCassandraViewOutput"], () => require("./getCassandraResourceCassandraView"));
exports.getDatabaseAccount = null;
exports.getDatabaseAccountOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccount", "getDatabaseAccountOutput"], () => require("./getDatabaseAccount"));
exports.getDatabaseAccountCassandraKeyspace = null;
exports.getDatabaseAccountCassandraKeyspaceOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountCassandraKeyspace", "getDatabaseAccountCassandraKeyspaceOutput"], () => require("./getDatabaseAccountCassandraKeyspace"));
exports.getDatabaseAccountCassandraTable = null;
exports.getDatabaseAccountCassandraTableOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountCassandraTable", "getDatabaseAccountCassandraTableOutput"], () => require("./getDatabaseAccountCassandraTable"));
exports.getDatabaseAccountGremlinDatabase = null;
exports.getDatabaseAccountGremlinDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountGremlinDatabase", "getDatabaseAccountGremlinDatabaseOutput"], () => require("./getDatabaseAccountGremlinDatabase"));
exports.getDatabaseAccountGremlinGraph = null;
exports.getDatabaseAccountGremlinGraphOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountGremlinGraph", "getDatabaseAccountGremlinGraphOutput"], () => require("./getDatabaseAccountGremlinGraph"));
exports.getDatabaseAccountMongoDBCollection = null;
exports.getDatabaseAccountMongoDBCollectionOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountMongoDBCollection", "getDatabaseAccountMongoDBCollectionOutput"], () => require("./getDatabaseAccountMongoDBCollection"));
exports.getDatabaseAccountMongoDBDatabase = null;
exports.getDatabaseAccountMongoDBDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountMongoDBDatabase", "getDatabaseAccountMongoDBDatabaseOutput"], () => require("./getDatabaseAccountMongoDBDatabase"));
exports.getDatabaseAccountSqlContainer = null;
exports.getDatabaseAccountSqlContainerOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountSqlContainer", "getDatabaseAccountSqlContainerOutput"], () => require("./getDatabaseAccountSqlContainer"));
exports.getDatabaseAccountSqlDatabase = null;
exports.getDatabaseAccountSqlDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountSqlDatabase", "getDatabaseAccountSqlDatabaseOutput"], () => require("./getDatabaseAccountSqlDatabase"));
exports.getDatabaseAccountTable = null;
exports.getDatabaseAccountTableOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccountTable", "getDatabaseAccountTableOutput"], () => require("./getDatabaseAccountTable"));
exports.getFleet = null;
exports.getFleetOutput = null;
utilities.lazyLoad(exports, ["getFleet", "getFleetOutput"], () => require("./getFleet"));
exports.getFleetAnalytic = null;
exports.getFleetAnalyticOutput = null;
utilities.lazyLoad(exports, ["getFleetAnalytic", "getFleetAnalyticOutput"], () => require("./getFleetAnalytic"));
exports.getFleetspace = null;
exports.getFleetspaceOutput = null;
utilities.lazyLoad(exports, ["getFleetspace", "getFleetspaceOutput"], () => require("./getFleetspace"));
exports.getFleetspaceAccount = null;
exports.getFleetspaceAccountOutput = null;
utilities.lazyLoad(exports, ["getFleetspaceAccount", "getFleetspaceAccountOutput"], () => require("./getFleetspaceAccount"));
exports.getGarnetCluster = null;
exports.getGarnetClusterOutput = null;
utilities.lazyLoad(exports, ["getGarnetCluster", "getGarnetClusterOutput"], () => require("./getGarnetCluster"));
exports.getGraphResourceGraph = null;
exports.getGraphResourceGraphOutput = null;
utilities.lazyLoad(exports, ["getGraphResourceGraph", "getGraphResourceGraphOutput"], () => require("./getGraphResourceGraph"));
exports.getGremlinResourceGremlinDatabase = null;
exports.getGremlinResourceGremlinDatabaseOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinDatabase", "getGremlinResourceGremlinDatabaseOutput"], () => require("./getGremlinResourceGremlinDatabase"));
exports.getGremlinResourceGremlinGraph = null;
exports.getGremlinResourceGremlinGraphOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinGraph", "getGremlinResourceGremlinGraphOutput"], () => require("./getGremlinResourceGremlinGraph"));
exports.getGremlinResourceGremlinRoleAssignment = null;
exports.getGremlinResourceGremlinRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinRoleAssignment", "getGremlinResourceGremlinRoleAssignmentOutput"], () => require("./getGremlinResourceGremlinRoleAssignment"));
exports.getGremlinResourceGremlinRoleDefinition = null;
exports.getGremlinResourceGremlinRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinRoleDefinition", "getGremlinResourceGremlinRoleDefinitionOutput"], () => require("./getGremlinResourceGremlinRoleDefinition"));
exports.getMongoCluster = null;
exports.getMongoClusterOutput = null;
utilities.lazyLoad(exports, ["getMongoCluster", "getMongoClusterOutput"], () => require("./getMongoCluster"));
exports.getMongoClusterFirewallRule = null;
exports.getMongoClusterFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getMongoClusterFirewallRule", "getMongoClusterFirewallRuleOutput"], () => require("./getMongoClusterFirewallRule"));
exports.getMongoDBResourceMongoDBCollection = null;
exports.getMongoDBResourceMongoDBCollectionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBCollection", "getMongoDBResourceMongoDBCollectionOutput"], () => require("./getMongoDBResourceMongoDBCollection"));
exports.getMongoDBResourceMongoDBDatabase = null;
exports.getMongoDBResourceMongoDBDatabaseOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBDatabase", "getMongoDBResourceMongoDBDatabaseOutput"], () => require("./getMongoDBResourceMongoDBDatabase"));
exports.getMongoDBResourceMongoRoleDefinition = null;
exports.getMongoDBResourceMongoRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoRoleDefinition", "getMongoDBResourceMongoRoleDefinitionOutput"], () => require("./getMongoDBResourceMongoRoleDefinition"));
exports.getMongoDBResourceMongoUserDefinition = null;
exports.getMongoDBResourceMongoUserDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoUserDefinition", "getMongoDBResourceMongoUserDefinitionOutput"], () => require("./getMongoDBResourceMongoUserDefinition"));
exports.getMongoMIResourceMongoMIRoleAssignment = null;
exports.getMongoMIResourceMongoMIRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getMongoMIResourceMongoMIRoleAssignment", "getMongoMIResourceMongoMIRoleAssignmentOutput"], () => require("./getMongoMIResourceMongoMIRoleAssignment"));
exports.getMongoMIResourceMongoMIRoleDefinition = null;
exports.getMongoMIResourceMongoMIRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoMIResourceMongoMIRoleDefinition", "getMongoMIResourceMongoMIRoleDefinitionOutput"], () => require("./getMongoMIResourceMongoMIRoleDefinition"));
exports.getNotebookWorkspace = null;
exports.getNotebookWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getNotebookWorkspace", "getNotebookWorkspaceOutput"], () => require("./getNotebookWorkspace"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSqlResourceSqlContainer = null;
exports.getSqlResourceSqlContainerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlContainer", "getSqlResourceSqlContainerOutput"], () => require("./getSqlResourceSqlContainer"));
exports.getSqlResourceSqlDatabase = null;
exports.getSqlResourceSqlDatabaseOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlDatabase", "getSqlResourceSqlDatabaseOutput"], () => require("./getSqlResourceSqlDatabase"));
exports.getSqlResourceSqlRoleAssignment = null;
exports.getSqlResourceSqlRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleAssignment", "getSqlResourceSqlRoleAssignmentOutput"], () => require("./getSqlResourceSqlRoleAssignment"));
exports.getSqlResourceSqlRoleDefinition = null;
exports.getSqlResourceSqlRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleDefinition", "getSqlResourceSqlRoleDefinitionOutput"], () => require("./getSqlResourceSqlRoleDefinition"));
exports.getSqlResourceSqlStoredProcedure = null;
exports.getSqlResourceSqlStoredProcedureOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlStoredProcedure", "getSqlResourceSqlStoredProcedureOutput"], () => require("./getSqlResourceSqlStoredProcedure"));
exports.getSqlResourceSqlTrigger = null;
exports.getSqlResourceSqlTriggerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlTrigger", "getSqlResourceSqlTriggerOutput"], () => require("./getSqlResourceSqlTrigger"));
exports.getSqlResourceSqlUserDefinedFunction = null;
exports.getSqlResourceSqlUserDefinedFunctionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlUserDefinedFunction", "getSqlResourceSqlUserDefinedFunctionOutput"], () => require("./getSqlResourceSqlUserDefinedFunction"));
exports.getTableResourceTable = null;
exports.getTableResourceTableOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTable", "getTableResourceTableOutput"], () => require("./getTableResourceTable"));
exports.getTableResourceTableRoleAssignment = null;
exports.getTableResourceTableRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTableRoleAssignment", "getTableResourceTableRoleAssignmentOutput"], () => require("./getTableResourceTableRoleAssignment"));
exports.getTableResourceTableRoleDefinition = null;
exports.getTableResourceTableRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTableRoleDefinition", "getTableResourceTableRoleDefinitionOutput"], () => require("./getTableResourceTableRoleDefinition"));
exports.getThroughputPool = null;
exports.getThroughputPoolOutput = null;
utilities.lazyLoad(exports, ["getThroughputPool", "getThroughputPoolOutput"], () => require("./getThroughputPool"));
exports.getThroughputPoolAccount = null;
exports.getThroughputPoolAccountOutput = null;
utilities.lazyLoad(exports, ["getThroughputPoolAccount", "getThroughputPoolAccountOutput"], () => require("./getThroughputPoolAccount"));
exports.GraphResourceGraph = null;
utilities.lazyLoad(exports, ["GraphResourceGraph"], () => require("./graphResourceGraph"));
exports.GremlinResourceGremlinDatabase = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinDatabase"], () => require("./gremlinResourceGremlinDatabase"));
exports.GremlinResourceGremlinGraph = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinGraph"], () => require("./gremlinResourceGremlinGraph"));
exports.GremlinResourceGremlinRoleAssignment = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinRoleAssignment"], () => require("./gremlinResourceGremlinRoleAssignment"));
exports.GremlinResourceGremlinRoleDefinition = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinRoleDefinition"], () => require("./gremlinResourceGremlinRoleDefinition"));
exports.listDatabaseAccountConnectionStrings = null;
exports.listDatabaseAccountConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountConnectionStrings", "listDatabaseAccountConnectionStringsOutput"], () => require("./listDatabaseAccountConnectionStrings"));
exports.listDatabaseAccountKeys = null;
exports.listDatabaseAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountKeys", "listDatabaseAccountKeysOutput"], () => require("./listDatabaseAccountKeys"));
exports.listMongoClusterConnectionStrings = null;
exports.listMongoClusterConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listMongoClusterConnectionStrings", "listMongoClusterConnectionStringsOutput"], () => require("./listMongoClusterConnectionStrings"));
exports.listNotebookWorkspaceConnectionInfo = null;
exports.listNotebookWorkspaceConnectionInfoOutput = null;
utilities.lazyLoad(exports, ["listNotebookWorkspaceConnectionInfo", "listNotebookWorkspaceConnectionInfoOutput"], () => require("./listNotebookWorkspaceConnectionInfo"));
exports.MongoCluster = null;
utilities.lazyLoad(exports, ["MongoCluster"], () => require("./mongoCluster"));
exports.MongoClusterFirewallRule = null;
utilities.lazyLoad(exports, ["MongoClusterFirewallRule"], () => require("./mongoClusterFirewallRule"));
exports.MongoDBResourceMongoDBCollection = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBCollection"], () => require("./mongoDBResourceMongoDBCollection"));
exports.MongoDBResourceMongoDBDatabase = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBDatabase"], () => require("./mongoDBResourceMongoDBDatabase"));
exports.MongoDBResourceMongoRoleDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoRoleDefinition"], () => require("./mongoDBResourceMongoRoleDefinition"));
exports.MongoDBResourceMongoUserDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoUserDefinition"], () => require("./mongoDBResourceMongoUserDefinition"));
exports.MongoMIResourceMongoMIRoleAssignment = null;
utilities.lazyLoad(exports, ["MongoMIResourceMongoMIRoleAssignment"], () => require("./mongoMIResourceMongoMIRoleAssignment"));
exports.MongoMIResourceMongoMIRoleDefinition = null;
utilities.lazyLoad(exports, ["MongoMIResourceMongoMIRoleDefinition"], () => require("./mongoMIResourceMongoMIRoleDefinition"));
exports.NotebookWorkspace = null;
utilities.lazyLoad(exports, ["NotebookWorkspace"], () => require("./notebookWorkspace"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.SqlResourceSqlContainer = null;
utilities.lazyLoad(exports, ["SqlResourceSqlContainer"], () => require("./sqlResourceSqlContainer"));
exports.SqlResourceSqlDatabase = null;
utilities.lazyLoad(exports, ["SqlResourceSqlDatabase"], () => require("./sqlResourceSqlDatabase"));
exports.SqlResourceSqlRoleAssignment = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleAssignment"], () => require("./sqlResourceSqlRoleAssignment"));
exports.SqlResourceSqlRoleDefinition = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleDefinition"], () => require("./sqlResourceSqlRoleDefinition"));
exports.SqlResourceSqlStoredProcedure = null;
utilities.lazyLoad(exports, ["SqlResourceSqlStoredProcedure"], () => require("./sqlResourceSqlStoredProcedure"));
exports.SqlResourceSqlTrigger = null;
utilities.lazyLoad(exports, ["SqlResourceSqlTrigger"], () => require("./sqlResourceSqlTrigger"));
exports.SqlResourceSqlUserDefinedFunction = null;
utilities.lazyLoad(exports, ["SqlResourceSqlUserDefinedFunction"], () => require("./sqlResourceSqlUserDefinedFunction"));
exports.TableResourceTable = null;
utilities.lazyLoad(exports, ["TableResourceTable"], () => require("./tableResourceTable"));
exports.TableResourceTableRoleAssignment = null;
utilities.lazyLoad(exports, ["TableResourceTableRoleAssignment"], () => require("./tableResourceTableRoleAssignment"));
exports.TableResourceTableRoleDefinition = null;
utilities.lazyLoad(exports, ["TableResourceTableRoleDefinition"], () => require("./tableResourceTableRoleDefinition"));
exports.ThroughputPool = null;
utilities.lazyLoad(exports, ["ThroughputPool"], () => require("./throughputPool"));
exports.ThroughputPoolAccount = null;
utilities.lazyLoad(exports, ["ThroughputPoolAccount"], () => require("./throughputPoolAccount"));
// Export enums:
__exportStar(require("../types/enums/cosmosdb"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cosmosdb:CassandraCluster":
                return new exports.CassandraCluster(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraDataCenter":
                return new exports.CassandraDataCenter(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraResourceCassandraKeyspace":
                return new exports.CassandraResourceCassandraKeyspace(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraResourceCassandraRoleAssignment":
                return new exports.CassandraResourceCassandraRoleAssignment(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraResourceCassandraRoleDefinition":
                return new exports.CassandraResourceCassandraRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraResourceCassandraTable":
                return new exports.CassandraResourceCassandraTable(name, undefined, { urn });
            case "azure-native:cosmosdb:CassandraResourceCassandraView":
                return new exports.CassandraResourceCassandraView(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccount":
                return new exports.DatabaseAccount(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountCassandraKeyspace":
                return new exports.DatabaseAccountCassandraKeyspace(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountCassandraTable":
                return new exports.DatabaseAccountCassandraTable(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountGremlinDatabase":
                return new exports.DatabaseAccountGremlinDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountGremlinGraph":
                return new exports.DatabaseAccountGremlinGraph(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountMongoDBCollection":
                return new exports.DatabaseAccountMongoDBCollection(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountMongoDBDatabase":
                return new exports.DatabaseAccountMongoDBDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountSqlContainer":
                return new exports.DatabaseAccountSqlContainer(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountSqlDatabase":
                return new exports.DatabaseAccountSqlDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:DatabaseAccountTable":
                return new exports.DatabaseAccountTable(name, undefined, { urn });
            case "azure-native:cosmosdb:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "azure-native:cosmosdb:FleetAnalytic":
                return new exports.FleetAnalytic(name, undefined, { urn });
            case "azure-native:cosmosdb:Fleetspace":
                return new exports.Fleetspace(name, undefined, { urn });
            case "azure-native:cosmosdb:FleetspaceAccount":
                return new exports.FleetspaceAccount(name, undefined, { urn });
            case "azure-native:cosmosdb:GarnetCluster":
                return new exports.GarnetCluster(name, undefined, { urn });
            case "azure-native:cosmosdb:GraphResourceGraph":
                return new exports.GraphResourceGraph(name, undefined, { urn });
            case "azure-native:cosmosdb:GremlinResourceGremlinDatabase":
                return new exports.GremlinResourceGremlinDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:GremlinResourceGremlinGraph":
                return new exports.GremlinResourceGremlinGraph(name, undefined, { urn });
            case "azure-native:cosmosdb:GremlinResourceGremlinRoleAssignment":
                return new exports.GremlinResourceGremlinRoleAssignment(name, undefined, { urn });
            case "azure-native:cosmosdb:GremlinResourceGremlinRoleDefinition":
                return new exports.GremlinResourceGremlinRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoCluster":
                return new exports.MongoCluster(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoClusterFirewallRule":
                return new exports.MongoClusterFirewallRule(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoDBResourceMongoDBCollection":
                return new exports.MongoDBResourceMongoDBCollection(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoDBResourceMongoDBDatabase":
                return new exports.MongoDBResourceMongoDBDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoDBResourceMongoRoleDefinition":
                return new exports.MongoDBResourceMongoRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoDBResourceMongoUserDefinition":
                return new exports.MongoDBResourceMongoUserDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoMIResourceMongoMIRoleAssignment":
                return new exports.MongoMIResourceMongoMIRoleAssignment(name, undefined, { urn });
            case "azure-native:cosmosdb:MongoMIResourceMongoMIRoleDefinition":
                return new exports.MongoMIResourceMongoMIRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:NotebookWorkspace":
                return new exports.NotebookWorkspace(name, undefined, { urn });
            case "azure-native:cosmosdb:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:cosmosdb:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlContainer":
                return new exports.SqlResourceSqlContainer(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlDatabase":
                return new exports.SqlResourceSqlDatabase(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlRoleAssignment":
                return new exports.SqlResourceSqlRoleAssignment(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlRoleDefinition":
                return new exports.SqlResourceSqlRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlStoredProcedure":
                return new exports.SqlResourceSqlStoredProcedure(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlTrigger":
                return new exports.SqlResourceSqlTrigger(name, undefined, { urn });
            case "azure-native:cosmosdb:SqlResourceSqlUserDefinedFunction":
                return new exports.SqlResourceSqlUserDefinedFunction(name, undefined, { urn });
            case "azure-native:cosmosdb:TableResourceTable":
                return new exports.TableResourceTable(name, undefined, { urn });
            case "azure-native:cosmosdb:TableResourceTableRoleAssignment":
                return new exports.TableResourceTableRoleAssignment(name, undefined, { urn });
            case "azure-native:cosmosdb:TableResourceTableRoleDefinition":
                return new exports.TableResourceTableRoleDefinition(name, undefined, { urn });
            case "azure-native:cosmosdb:ThroughputPool":
                return new exports.ThroughputPool(name, undefined, { urn });
            case "azure-native:cosmosdb:ThroughputPoolAccount":
                return new exports.ThroughputPoolAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cosmosdb", _module);
