// @pulumi/azure-native/cosmosdb/mongoClusterFirewallRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoClusterFirewallRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a mongo cluster firewall rule.
 *
 * Uses Azure REST API version 2024-02-15-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-03-15-preview, 2023-09-15-preview, 2023-11-15-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class MongoClusterFirewallRule extends pulumi.CustomResource {
    /**
     * Get an existing MongoClusterFirewallRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MongoClusterFirewallRule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cosmosdb:MongoClusterFirewallRule';
    /**
     * Returns true if the given object is an instance of MongoClusterFirewallRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MongoClusterFirewallRule.__pulumiType;
    }
    /**
     * Create a MongoClusterFirewallRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.endIpAddress === undefined && !opts.urn) {
                throw new Error("Missing required property 'endIpAddress'");
            }
            if (args?.mongoClusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'mongoClusterName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.startIpAddress === undefined && !opts.urn) {
                throw new Error("Missing required property 'startIpAddress'");
            }
            resourceInputs["endIpAddress"] = args?.endIpAddress;
            resourceInputs["firewallRuleName"] = args?.firewallRuleName;
            resourceInputs["mongoClusterName"] = args?.mongoClusterName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["startIpAddress"] = args?.startIpAddress;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["endIpAddress"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["startIpAddress"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20230301preview:MongoClusterFirewallRule" }, { type: "azure-native:cosmosdb/v20230315preview:MongoClusterFirewallRule" }, { type: "azure-native:cosmosdb/v20230915preview:MongoClusterFirewallRule" }, { type: "azure-native:cosmosdb/v20231115preview:MongoClusterFirewallRule" }, { type: "azure-native:cosmosdb/v20240215preview:MongoClusterFirewallRule" }, { type: "azure-native:documentdb/v20230315preview:MongoClusterFirewallRule" }, { type: "azure-native:documentdb/v20230915preview:MongoClusterFirewallRule" }, { type: "azure-native:documentdb/v20231115preview:MongoClusterFirewallRule" }, { type: "azure-native:documentdb/v20240215preview:MongoClusterFirewallRule" }, { type: "azure-native:documentdb/v20240301preview:FirewallRule" }, { type: "azure-native:documentdb/v20240601preview:FirewallRule" }, { type: "azure-native:documentdb/v20240701:FirewallRule" }, { type: "azure-native:documentdb/v20241001preview:FirewallRule" }, { type: "azure-native:documentdb:FirewallRule" }, { type: "azure-native:documentdb:MongoClusterFirewallRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MongoClusterFirewallRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MongoClusterFirewallRule = MongoClusterFirewallRule;
