// @pulumi/azure-native/cosmosdb/notebookWorkspace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebookWorkspace = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A notebook workspace resource
 *
 * Uses Azure REST API version 2025-10-15.
 *
 * Other available API versions: 2019-08-01, 2019-12-12, 2020-03-01, 2020-04-01, 2020-06-01-preview, 2020-09-01, 2021-01-15, 2021-03-01-preview, 2021-03-15, 2021-04-01-preview, 2021-04-15, 2021-05-15, 2021-06-15, 2021-07-01-preview, 2021-10-15, 2021-10-15-preview, 2021-11-15-preview, 2022-02-15-preview, 2022-05-15, 2022-05-15-preview, 2022-08-15, 2022-08-15-preview, 2022-11-15, 2022-11-15-preview, 2023-03-01-preview, 2023-03-15, 2023-03-15-preview, 2023-04-15, 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview, 2025-04-15, 2025-05-01-preview, 2025-11-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class NotebookWorkspace extends pulumi.CustomResource {
    /**
     * Get an existing NotebookWorkspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NotebookWorkspace(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cosmosdb:NotebookWorkspace';
    /**
     * Returns true if the given object is an instance of NotebookWorkspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NotebookWorkspace.__pulumiType;
    }
    /**
     * Create a NotebookWorkspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["notebookWorkspaceName"] = args?.notebookWorkspaceName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notebookServerEndpoint"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notebookServerEndpoint"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20190801:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20191212:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20200301:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20200401:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20200601preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20200901:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210115:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210301preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210315:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210401preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210415:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210515:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210615:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20210701preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20211015:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20211015preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20211115preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20220215preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20220515:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20220515preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20220815:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20220815preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20221115:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20221115preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230301preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230315:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230315preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230415:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230915:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20230915preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20231115:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20231115preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20240215preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20240515:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20240515preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20240815:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20240901preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20241115:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20241201preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20250415:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20250501preview:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20251015:NotebookWorkspace" }, { type: "azure-native:cosmosdb/v20251101preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230415:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230915:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230915preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20231115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20231115preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240215preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240515:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240515preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240815:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240901preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20241115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20241201preview:NotebookWorkspace" }, { type: "azure-native:documentdb:NotebookWorkspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NotebookWorkspace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NotebookWorkspace = NotebookWorkspace;
