// @pulumi/azure-native/desktopvirtualization/application.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Schema for Application properties.
 *
 * Uses Azure REST API version 2024-04-03. In version 2.x of the Azure Native provider, it used API version 2022-09-09.
 *
 * Other available API versions: 2022-09-09, 2022-10-14-preview, 2023-09-05, 2023-10-04-preview, 2023-11-01-preview, 2024-01-16-preview, 2024-03-06-preview, 2024-04-08-preview, 2024-08-08-preview, 2024-11-01-preview, 2025-03-01-preview, 2025-04-01-preview, 2025-08-01-preview, 2025-09-01-preview, 2025-11-01-preview, 2026-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native desktopvirtualization [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Application extends pulumi.CustomResource {
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Application(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:desktopvirtualization:Application';
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.applicationGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'applicationGroupName'");
            }
            if (args?.commandLineSetting === undefined && !opts.urn) {
                throw new Error("Missing required property 'commandLineSetting'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationGroupName"] = args?.applicationGroupName;
            resourceInputs["applicationName"] = args?.applicationName;
            resourceInputs["applicationType"] = args?.applicationType;
            resourceInputs["commandLineArguments"] = args?.commandLineArguments;
            resourceInputs["commandLineSetting"] = args?.commandLineSetting;
            resourceInputs["description"] = args?.description;
            resourceInputs["filePath"] = args?.filePath;
            resourceInputs["friendlyName"] = args?.friendlyName;
            resourceInputs["iconIndex"] = args?.iconIndex;
            resourceInputs["iconPath"] = args?.iconPath;
            resourceInputs["msixPackageApplicationId"] = args?.msixPackageApplicationId;
            resourceInputs["msixPackageFamilyName"] = args?.msixPackageFamilyName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["showInPortal"] = args?.showInPortal;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["iconContent"] = undefined /*out*/;
            resourceInputs["iconHash"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationType"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["commandLineArguments"] = undefined /*out*/;
            resourceInputs["commandLineSetting"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["filePath"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["iconContent"] = undefined /*out*/;
            resourceInputs["iconHash"] = undefined /*out*/;
            resourceInputs["iconIndex"] = undefined /*out*/;
            resourceInputs["iconPath"] = undefined /*out*/;
            resourceInputs["msixPackageApplicationId"] = undefined /*out*/;
            resourceInputs["msixPackageFamilyName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["showInPortal"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:desktopvirtualization/v20190123preview:Application" }, { type: "azure-native:desktopvirtualization/v20190924preview:Application" }, { type: "azure-native:desktopvirtualization/v20191210preview:Application" }, { type: "azure-native:desktopvirtualization/v20200921preview:Application" }, { type: "azure-native:desktopvirtualization/v20201019preview:Application" }, { type: "azure-native:desktopvirtualization/v20201102preview:Application" }, { type: "azure-native:desktopvirtualization/v20201110preview:Application" }, { type: "azure-native:desktopvirtualization/v20210114preview:Application" }, { type: "azure-native:desktopvirtualization/v20210201preview:Application" }, { type: "azure-native:desktopvirtualization/v20210309preview:Application" }, { type: "azure-native:desktopvirtualization/v20210401preview:Application" }, { type: "azure-native:desktopvirtualization/v20210712:Application" }, { type: "azure-native:desktopvirtualization/v20210903preview:Application" }, { type: "azure-native:desktopvirtualization/v20220210preview:Application" }, { type: "azure-native:desktopvirtualization/v20220401preview:Application" }, { type: "azure-native:desktopvirtualization/v20220909:Application" }, { type: "azure-native:desktopvirtualization/v20221014preview:Application" }, { type: "azure-native:desktopvirtualization/v20230707preview:Application" }, { type: "azure-native:desktopvirtualization/v20230905:Application" }, { type: "azure-native:desktopvirtualization/v20231004preview:Application" }, { type: "azure-native:desktopvirtualization/v20231101preview:Application" }, { type: "azure-native:desktopvirtualization/v20240116preview:Application" }, { type: "azure-native:desktopvirtualization/v20240306preview:Application" }, { type: "azure-native:desktopvirtualization/v20240403:Application" }, { type: "azure-native:desktopvirtualization/v20240408preview:Application" }, { type: "azure-native:desktopvirtualization/v20240808preview:Application" }, { type: "azure-native:desktopvirtualization/v20241101preview:Application" }, { type: "azure-native:desktopvirtualization/v20250301preview:Application" }, { type: "azure-native:desktopvirtualization/v20250401preview:Application" }, { type: "azure-native:desktopvirtualization/v20250801preview:Application" }, { type: "azure-native:desktopvirtualization/v20250901preview:Application" }, { type: "azure-native:desktopvirtualization/v20251101preview:Application" }, { type: "azure-native:desktopvirtualization/v20260101preview:Application" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Application = Application;
