// @pulumi/gcp/tpu/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Vm = exports.V2QueuedResource = exports.getV2RuntimeVersionsOutput = exports.getV2RuntimeVersions = exports.getV2AcceleratorTypesOutput = exports.getV2AcceleratorTypes = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getV2AcceleratorTypes = null;
exports.getV2AcceleratorTypesOutput = null;
utilities.lazyLoad(exports, ["getV2AcceleratorTypes", "getV2AcceleratorTypesOutput"], () => require("./getV2AcceleratorTypes"));
exports.getV2RuntimeVersions = null;
exports.getV2RuntimeVersionsOutput = null;
utilities.lazyLoad(exports, ["getV2RuntimeVersions", "getV2RuntimeVersionsOutput"], () => require("./getV2RuntimeVersions"));
exports.V2QueuedResource = null;
utilities.lazyLoad(exports, ["V2QueuedResource"], () => require("./v2queuedResource"));
exports.V2Vm = null;
utilities.lazyLoad(exports, ["V2Vm"], () => require("./v2vm"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:tpu/v2QueuedResource:V2QueuedResource":
                return new exports.V2QueuedResource(name, undefined, { urn });
            case "gcp:tpu/v2Vm:V2Vm":
                return new exports.V2Vm(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "tpu/v2QueuedResource", _module);
pulumi.runtime.registerResourceModule("gcp", "tpu/v2Vm", _module);
