// @pulumi/aws/vpc/routeServerPropagation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteServerPropagation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a resource for managing propagation between VPC (Virtual Private Cloud) route server and a route table.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpc.RouteServerPropagation("example", {
 *     routeServerId: exampleAwsVpcRouteServer.routeServerId,
 *     routeTableId: exampleAwsRouteTable.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, to  to import VPC (Virtual Private Cloud) Route Server Propagation using the associated resource ID and route table ID separated by a comma (,). For example:
 *
 * ```sh
 * $ pulumi import aws:vpc/routeServerPropagation:RouteServerPropagation example rs-12345678,rtb-656c65616e6f72
 * ```
 */
class RouteServerPropagation extends pulumi.CustomResource {
    /**
     * Get an existing RouteServerPropagation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RouteServerPropagation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:vpc/routeServerPropagation:RouteServerPropagation';
    /**
     * Returns true if the given object is an instance of RouteServerPropagation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RouteServerPropagation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["routeServerId"] = state?.routeServerId;
            resourceInputs["routeTableId"] = state?.routeTableId;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.routeServerId === undefined && !opts.urn) {
                throw new Error("Missing required property 'routeServerId'");
            }
            if (args?.routeTableId === undefined && !opts.urn) {
                throw new Error("Missing required property 'routeTableId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["routeServerId"] = args?.routeServerId;
            resourceInputs["routeTableId"] = args?.routeTableId;
            resourceInputs["timeouts"] = args?.timeouts;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RouteServerPropagation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RouteServerPropagation = RouteServerPropagation;
