// @pulumi/gcp/cloudrunv2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkerPoolIamPolicy = exports.WorkerPoolIamMember = exports.WorkerPoolIamBinding = exports.WorkerPool = exports.ServiceIamPolicy = exports.ServiceIamMember = exports.ServiceIamBinding = exports.Service = exports.JobIamPolicy = exports.JobIamMember = exports.JobIamBinding = exports.Job = exports.getWorkerPoolIamPolicyOutput = exports.getWorkerPoolIamPolicy = exports.getWorkerPoolOutput = exports.getWorkerPool = exports.getServiceIamPolicyOutput = exports.getServiceIamPolicy = exports.getServiceOutput = exports.getService = exports.getJobIamPolicyOutput = exports.getJobIamPolicy = exports.getJobOutput = exports.getJob = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getJobIamPolicy = null;
exports.getJobIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getJobIamPolicy", "getJobIamPolicyOutput"], () => require("./getJobIamPolicy"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceIamPolicy = null;
exports.getServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getServiceIamPolicy", "getServiceIamPolicyOutput"], () => require("./getServiceIamPolicy"));
exports.getWorkerPool = null;
exports.getWorkerPoolOutput = null;
utilities.lazyLoad(exports, ["getWorkerPool", "getWorkerPoolOutput"], () => require("./getWorkerPool"));
exports.getWorkerPoolIamPolicy = null;
exports.getWorkerPoolIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWorkerPoolIamPolicy", "getWorkerPoolIamPolicyOutput"], () => require("./getWorkerPoolIamPolicy"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.JobIamBinding = null;
utilities.lazyLoad(exports, ["JobIamBinding"], () => require("./jobIamBinding"));
exports.JobIamMember = null;
utilities.lazyLoad(exports, ["JobIamMember"], () => require("./jobIamMember"));
exports.JobIamPolicy = null;
utilities.lazyLoad(exports, ["JobIamPolicy"], () => require("./jobIamPolicy"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceIamBinding = null;
utilities.lazyLoad(exports, ["ServiceIamBinding"], () => require("./serviceIamBinding"));
exports.ServiceIamMember = null;
utilities.lazyLoad(exports, ["ServiceIamMember"], () => require("./serviceIamMember"));
exports.ServiceIamPolicy = null;
utilities.lazyLoad(exports, ["ServiceIamPolicy"], () => require("./serviceIamPolicy"));
exports.WorkerPool = null;
utilities.lazyLoad(exports, ["WorkerPool"], () => require("./workerPool"));
exports.WorkerPoolIamBinding = null;
utilities.lazyLoad(exports, ["WorkerPoolIamBinding"], () => require("./workerPoolIamBinding"));
exports.WorkerPoolIamMember = null;
utilities.lazyLoad(exports, ["WorkerPoolIamMember"], () => require("./workerPoolIamMember"));
exports.WorkerPoolIamPolicy = null;
utilities.lazyLoad(exports, ["WorkerPoolIamPolicy"], () => require("./workerPoolIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudrunv2/job:Job":
                return new exports.Job(name, undefined, { urn });
            case "gcp:cloudrunv2/jobIamBinding:JobIamBinding":
                return new exports.JobIamBinding(name, undefined, { urn });
            case "gcp:cloudrunv2/jobIamMember:JobIamMember":
                return new exports.JobIamMember(name, undefined, { urn });
            case "gcp:cloudrunv2/jobIamPolicy:JobIamPolicy":
                return new exports.JobIamPolicy(name, undefined, { urn });
            case "gcp:cloudrunv2/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "gcp:cloudrunv2/serviceIamBinding:ServiceIamBinding":
                return new exports.ServiceIamBinding(name, undefined, { urn });
            case "gcp:cloudrunv2/serviceIamMember:ServiceIamMember":
                return new exports.ServiceIamMember(name, undefined, { urn });
            case "gcp:cloudrunv2/serviceIamPolicy:ServiceIamPolicy":
                return new exports.ServiceIamPolicy(name, undefined, { urn });
            case "gcp:cloudrunv2/workerPool:WorkerPool":
                return new exports.WorkerPool(name, undefined, { urn });
            case "gcp:cloudrunv2/workerPoolIamBinding:WorkerPoolIamBinding":
                return new exports.WorkerPoolIamBinding(name, undefined, { urn });
            case "gcp:cloudrunv2/workerPoolIamMember:WorkerPoolIamMember":
                return new exports.WorkerPoolIamMember(name, undefined, { urn });
            case "gcp:cloudrunv2/workerPoolIamPolicy:WorkerPoolIamPolicy":
                return new exports.WorkerPoolIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/job", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/jobIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/jobIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/jobIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/service", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/serviceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/serviceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/serviceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/workerPool", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/workerPoolIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/workerPoolIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudrunv2/workerPoolIamPolicy", _module);
