// @pulumi/azure-native/operationsmanagement/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Solution = exports.ManagementConfiguration = exports.ManagementAssociation = exports.getSolutionOutput = exports.getSolution = exports.getManagementConfigurationOutput = exports.getManagementConfiguration = exports.getManagementAssociationOutput = exports.getManagementAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getManagementAssociation = null;
exports.getManagementAssociationOutput = null;
utilities.lazyLoad(exports, ["getManagementAssociation", "getManagementAssociationOutput"], () => require("./getManagementAssociation"));
exports.getManagementConfiguration = null;
exports.getManagementConfigurationOutput = null;
utilities.lazyLoad(exports, ["getManagementConfiguration", "getManagementConfigurationOutput"], () => require("./getManagementConfiguration"));
exports.getSolution = null;
exports.getSolutionOutput = null;
utilities.lazyLoad(exports, ["getSolution", "getSolutionOutput"], () => require("./getSolution"));
exports.ManagementAssociation = null;
utilities.lazyLoad(exports, ["ManagementAssociation"], () => require("./managementAssociation"));
exports.ManagementConfiguration = null;
utilities.lazyLoad(exports, ["ManagementConfiguration"], () => require("./managementConfiguration"));
exports.Solution = null;
utilities.lazyLoad(exports, ["Solution"], () => require("./solution"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:operationsmanagement:ManagementAssociation":
                return new exports.ManagementAssociation(name, undefined, { urn });
            case "azure-native:operationsmanagement:ManagementConfiguration":
                return new exports.ManagementConfiguration(name, undefined, { urn });
            case "azure-native:operationsmanagement:Solution":
                return new exports.Solution(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "operationsmanagement", _module);
