// @pulumi/azure-native/types/enums/billingbenefits/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.PricingPolicy = exports.ManagedServiceIdentityType = exports.MaccStatus = exports.MaccMilestoneStatus = exports.MaccEntityType = exports.FreeServicesStatus = exports.EnablementMode = exports.DiscountType = exports.DiscountRuleType = exports.DiscountEntityType = exports.DiscountCombinationRule = exports.DiscountAppliedScopeType = exports.CreditStatus = exports.CreditRedemptionPolicy = exports.CreditExpirationPolicy = exports.ConditionalCreditStatus = exports.ConditionalCreditEntityType = exports.CommitmentGrain = exports.ApplyDiscountOn = void 0;
exports.ApplyDiscountOn = {
    Purchase: "Purchase",
    Consume: "Consume",
    Renew: "Renew",
};
exports.CommitmentGrain = {
    Unknown: "Unknown",
    Hourly: "Hourly",
    FullTerm: "FullTerm",
};
exports.ConditionalCreditEntityType = {
    Primary: "Primary",
    Contributor: "Contributor",
};
exports.ConditionalCreditStatus = {
    Unknown: "Unknown",
    Scheduled: "Scheduled",
    Active: "Active",
    Pending: "Pending",
    Failed: "Failed",
    Canceled: "Canceled",
    Completed: "Completed",
    Stopped: "Stopped",
    PendingSettlement: "PendingSettlement",
};
exports.CreditExpirationPolicy = {
    /**
     * No policy applied to the expiration of this credit
     */
    None: "None",
    /**
     * Billing profile is suspended when the credit expires
     */
    SuspendBillingProfile: "SuspendBillingProfile",
};
exports.CreditRedemptionPolicy = {
    /**
     * Redemption policy is not applicable for this credit
     */
    NotApplicable: "NotApplicable",
    /**
     * Credit is automatically redeemed
     */
    AutoRedeem: "AutoRedeem",
    /**
     * Credit is manually redeemed
     */
    ManualRedeem: "ManualRedeem",
};
exports.CreditStatus = {
    /**
     * Credit status is unknown
     */
    Unknown: "Unknown",
    /**
     * Credit is pending activation
     */
    Pending: "Pending",
    /**
     * Credit is active and available for use
     */
    Active: "Active",
    /**
     * Credit operation succeeded
     */
    Succeeded: "Succeeded",
    /**
     * Credit has been cancelled
     */
    Canceled: "Canceled",
    /**
     * Credit operation failed
     */
    Failed: "Failed",
    /**
     * Credit has expired
     */
    Expired: "Expired",
    /**
     * Credit has been fully consumed
     */
    Exhausted: "Exhausted",
    /**
     * Credit application has not started yet
     */
    NotStarted: "NotStarted",
};
exports.DiscountAppliedScopeType = {
    BillingAccount: "BillingAccount",
    BillingProfile: "BillingProfile",
    Customer: "Customer",
};
exports.DiscountCombinationRule = {
    BestOf: "BestOf",
    Stackable: "Stackable",
};
exports.DiscountEntityType = {
    Primary: "Primary",
    Affiliate: "Affiliate",
};
exports.DiscountRuleType = {
    FixedPriceLock: "FixedPriceLock",
    FixedListPrice: "FixedListPrice",
    PriceCeiling: "PriceCeiling",
};
exports.DiscountType = {
    ProductFamily: "ProductFamily",
    Product: "Product",
    Sku: "Sku",
    CustomPrice: "CustomPrice",
    CustomPriceMultiCurrency: "CustomPriceMultiCurrency",
};
exports.EnablementMode = {
    Unknown: "Unknown",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FreeServicesStatus = {
    /**
     * Free services status is unknown
     */
    Unknown: "Unknown",
    /**
     * Free services are pending activation
     */
    Pending: "Pending",
    /**
     * Free services are active
     */
    Active: "Active",
    /**
     * Free services have been canceled
     */
    Canceled: "Canceled",
    /**
     * Free services have been fully consumed or completed
     */
    Completed: "Completed",
};
exports.MaccEntityType = {
    Primary: "Primary",
    Contributor: "Contributor",
};
exports.MaccMilestoneStatus = {
    Unknown: "Unknown",
    Scheduled: "Scheduled",
    Active: "Active",
    Pending: "Pending",
    Failed: "Failed",
    Completed: "Completed",
    Canceled: "Canceled",
    Removed: "Removed",
    PendingSettlement: "PendingSettlement",
    ShortfallCharged: "ShortfallCharged",
    ShortfallWaived: "ShortfallWaived",
};
exports.MaccStatus = {
    Unknown: "Unknown",
    Scheduled: "Scheduled",
    Active: "Active",
    Pending: "Pending",
    Failed: "Failed",
    Canceled: "Canceled",
    Completed: "Completed",
    Stopped: "Stopped",
    PendingSettlement: "PendingSettlement",
    ShortfallCharged: "ShortfallCharged",
    ShortfallWaived: "ShortfallWaived",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PricingPolicy = {
    Protected: "Protected",
    Locked: "Locked",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
