// @pulumi/azure-native/types/enums/managednetworkfabric/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WellKnownCommunities = exports.TapRuleActionType = exports.StationConnectionMode = exports.StationConfigurationState = exports.SourceDestinationType = exports.RoutePolicyConditionType = exports.RoutePolicyActionType = exports.RedistributeStaticRoutes = exports.RedistributeConnectedSubnets = exports.PrefixType = exports.PortType = exports.PollingType = exports.PeeringOption = exports.NniType = exports.NfcSku = exports.NetworkRackType = exports.Layer4Protocol = exports.IsWorkloadManagementNetworkEnabled = exports.IsMonitoringEnabled = exports.IsManagementType = exports.IPAddressType = exports.GatewayType = exports.Extension = exports.EncapsulationType = exports.Encapsulation = exports.DestinationType = exports.ConfigurationType = exports.Condition = exports.CommunityActionTypes = exports.BooleanEnumProperty = exports.BmpMonitoredAddressFamily = exports.BmpExportPolicy = exports.AllowASOverride = exports.AddressFamilyType = exports.Action = exports.AclActionType = void 0;
exports.AclActionType = {
    Drop: "Drop",
    Count: "Count",
    Log: "Log",
};
exports.Action = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.AddressFamilyType = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.AllowASOverride = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.BmpExportPolicy = {
    /**
     * BMP ExportPolicy Pre-Policy
     */
    PrePolicy: "Pre-Policy",
    /**
     * BMP ExportPolicy Post-Policy
     */
    PostPolicy: "Post-Policy",
    /**
     * BMP ExportPolicy All
     */
    All: "All",
};
exports.BmpMonitoredAddressFamily = {
    /**
     * IPv4 Unicast
     */
    Ipv4Unicast: "ipv4Unicast",
    /**
     * IPv6 Unicast
     */
    Ipv6Unicast: "ipv6Unicast",
    /**
     * VPN IPv4
     */
    VpnIpv4: "vpnIpv4",
    /**
     * VPN IPv6
     */
    VpnIpv6: "vpnIpv6",
    /**
     * All Address Families
     */
    All: "All",
};
exports.BooleanEnumProperty = {
    True: "True",
    False: "False",
};
exports.CommunityActionTypes = {
    Permit: "Permit",
    Deny: "Deny",
};
exports.Condition = {
    EqualTo: "EqualTo",
    GreaterThanOrEqualTo: "GreaterThanOrEqualTo",
    LesserThanOrEqualTo: "LesserThanOrEqualTo",
    Range: "Range",
};
exports.ConfigurationType = {
    File: "File",
    Inline: "Inline",
};
exports.DestinationType = {
    IsolationDomain: "IsolationDomain",
    Direct: "Direct",
};
exports.Encapsulation = {
    None: "None",
    GRE: "GRE",
};
exports.EncapsulationType = {
    None: "None",
    GTPv1: "GTPv1",
};
exports.Extension = {
    NoExtension: "NoExtension",
    NPB: "NPB",
};
exports.GatewayType = {
    Infrastructure: "Infrastructure",
    Workload: "Workload",
};
exports.IPAddressType = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.IsManagementType = {
    True: "True",
    False: "False",
};
exports.IsMonitoringEnabled = {
    True: "True",
    False: "False",
};
exports.IsWorkloadManagementNetworkEnabled = {
    True: "True",
    False: "False",
};
exports.Layer4Protocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.NetworkRackType = {
    Aggregate: "Aggregate",
    Compute: "Compute",
    Combined: "Combined",
};
exports.NfcSku = {
    Basic: "Basic",
    Standard: "Standard",
    HighPerformance: "HighPerformance",
};
exports.NniType = {
    CE: "CE",
    NPB: "NPB",
};
exports.PeeringOption = {
    OptionA: "OptionA",
    OptionB: "OptionB",
};
exports.PollingType = {
    Pull: "Pull",
    Push: "Push",
};
exports.PortType = {
    SourcePort: "SourcePort",
    DestinationPort: "DestinationPort",
};
exports.PrefixType = {
    Prefix: "Prefix",
    LongestPrefix: "LongestPrefix",
};
exports.RedistributeConnectedSubnets = {
    True: "True",
    False: "False",
};
exports.RedistributeStaticRoutes = {
    True: "True",
    False: "False",
};
exports.RoutePolicyActionType = {
    Permit: "Permit",
    Deny: "Deny",
    Continue: "Continue",
};
exports.RoutePolicyConditionType = {
    Or: "Or",
    And: "And",
};
exports.SourceDestinationType = {
    SourceIP: "SourceIP",
    DestinationIP: "DestinationIP",
};
exports.StationConfigurationState = {
    /**
     * StationConfigurationState Enabled
     */
    Enabled: "Enabled",
    /**
     * StationConfigurationState Disabled
     */
    Disabled: "Disabled",
};
exports.StationConnectionMode = {
    /**
     * StationConnectionMode Active
     */
    Active: "Active",
    /**
     * StationConnectionMode Passive
     */
    Passive: "Passive",
};
exports.TapRuleActionType = {
    Drop: "Drop",
    Count: "Count",
    Log: "Log",
    Replicate: "Replicate",
    Goto: "Goto",
    Redirect: "Redirect",
    Mirror: "Mirror",
};
exports.WellKnownCommunities = {
    Internet: "Internet",
    LocalAS: "LocalAS",
    NoAdvertise: "NoAdvertise",
    NoExport: "NoExport",
    GShut: "GShut",
};
