// @pulumi/azure-native/types/enums/cognitiveservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficRoutingProtocol = exports.SkuTier = exports.ScenarioType = exports.RuleType = exports.RuleStatus = exports.RuleCategory = exports.RoutingMethods = exports.ResourceIdentityType = exports.RaiPolicyMode = exports.RaiPolicyContentSource = exports.RaiActionType = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NetworkRuleAction = exports.ManagedPEStatus = exports.ManagedPERequirement = exports.KeySource = exports.IdentityManagementType = exports.IdentityKind = exports.HostingModel = exports.EncryptionScopeState = exports.DeploymentScaleType = exports.DeploymentModelVersionUpgradeOption = exports.ContentLevel = exports.ConnectionCategory = exports.ConnectionAuthType = exports.CapabilityHostKind = exports.ByPassSelection = exports.BuiltInAuthorizationScheme = exports.AgentProtocol = exports.AgentDeploymentType = exports.AgentDeploymentState = void 0;
exports.AgentDeploymentState = {
    /**
     * The deployment is starting.
     */
    Starting: "Starting",
    /**
     * The deployment started/is operational.
     */
    Running: "Running",
    /**
     * The deployment is being stopped.
     */
    Stopping: "Stopping",
    /**
     * The deployment was stopped.
     */
    Stopped: "Stopped",
    /**
     * The deployment failed.
     */
    Failed: "Failed",
    /**
     * The deployment is being deleted.
     */
    Deleting: "Deleting",
    /**
     * The deployment was deleted.
     */
    Deleted: "Deleted",
    /**
     * The deployment is being updated.
     */
    Updating: "Updating",
};
exports.AgentDeploymentType = {
    /**
     * The underlying infra is managed by the platform in the deployer's subscription
     */
    Managed: "Managed",
    /**
     * The underlying infra is owned by the platform
     */
    Hosted: "Hosted",
    /**
     * The underlying infra is provisioned by the deployer (BYO)
     */
    Custom: "Custom",
};
exports.AgentProtocol = {
    /**
     * Agent protocol (aka Active)
     */
    Agent: "Agent",
    /**
     * Agent2Agent standard
     */
    A2A: "A2A",
    /**
     * OpenAI-compatible
     */
    Responses: "Responses",
};
exports.BuiltInAuthorizationScheme = {
    /**
     * Standard AzureML RBAC
     */
    Default: "Default",
    /**
     * Claim-based, requires membership in the tenant
     */
    OrganizationScope: "OrganizationScope",
    /**
     * Channels-specific (AzureBotService) authorization
     */
    Channels: "Channels",
    /**
     * Custom scheme defined by the application author
     */
    Custom: "Custom",
};
exports.ByPassSelection = {
    None: "None",
    AzureServices: "AzureServices",
};
exports.CapabilityHostKind = {
    Agents: "Agents",
};
exports.ConnectionAuthType = {
    PAT: "PAT",
    ManagedIdentity: "ManagedIdentity",
    UsernamePassword: "UsernamePassword",
    None: "None",
    SAS: "SAS",
    AccountKey: "AccountKey",
    ServicePrincipal: "ServicePrincipal",
    AccessKey: "AccessKey",
    ApiKey: "ApiKey",
    CustomKeys: "CustomKeys",
    OAuth2: "OAuth2",
    AAD: "AAD",
};
exports.ConnectionCategory = {
    PythonFeed: "PythonFeed",
    ContainerRegistry: "ContainerRegistry",
    Git: "Git",
    S3: "S3",
    Snowflake: "Snowflake",
    AzureSqlDb: "AzureSqlDb",
    AzureSynapseAnalytics: "AzureSynapseAnalytics",
    AzureMySqlDb: "AzureMySqlDb",
    AzurePostgresDb: "AzurePostgresDb",
    ADLSGen2: "ADLSGen2",
    Redis: "Redis",
    ApiKey: "ApiKey",
    AzureOpenAI: "AzureOpenAI",
    AIServices: "AIServices",
    CognitiveSearch: "CognitiveSearch",
    CognitiveService: "CognitiveService",
    CustomKeys: "CustomKeys",
    AzureBlob: "AzureBlob",
    AzureOneLake: "AzureOneLake",
    CosmosDb: "CosmosDb",
    CosmosDbMongoDbApi: "CosmosDbMongoDbApi",
    AzureDataExplorer: "AzureDataExplorer",
    AzureMariaDb: "AzureMariaDb",
    AzureDatabricksDeltaLake: "AzureDatabricksDeltaLake",
    AzureSqlMi: "AzureSqlMi",
    AzureTableStorage: "AzureTableStorage",
    AmazonRdsForOracle: "AmazonRdsForOracle",
    AmazonRdsForSqlServer: "AmazonRdsForSqlServer",
    AmazonRedshift: "AmazonRedshift",
    Db2: "Db2",
    Drill: "Drill",
    GoogleBigQuery: "GoogleBigQuery",
    Greenplum: "Greenplum",
    Hbase: "Hbase",
    Hive: "Hive",
    Impala: "Impala",
    Informix: "Informix",
    MariaDb: "MariaDb",
    MicrosoftAccess: "MicrosoftAccess",
    MySql: "MySql",
    Netezza: "Netezza",
    Oracle: "Oracle",
    Phoenix: "Phoenix",
    PostgreSql: "PostgreSql",
    Presto: "Presto",
    SapOpenHub: "SapOpenHub",
    SapBw: "SapBw",
    SapHana: "SapHana",
    SapTable: "SapTable",
    Spark: "Spark",
    SqlServer: "SqlServer",
    Sybase: "Sybase",
    Teradata: "Teradata",
    Vertica: "Vertica",
    Pinecone: "Pinecone",
    Cassandra: "Cassandra",
    Couchbase: "Couchbase",
    MongoDbV2: "MongoDbV2",
    MongoDbAtlas: "MongoDbAtlas",
    AmazonS3Compatible: "AmazonS3Compatible",
    FileServer: "FileServer",
    FtpServer: "FtpServer",
    GoogleCloudStorage: "GoogleCloudStorage",
    Hdfs: "Hdfs",
    OracleCloudStorage: "OracleCloudStorage",
    Sftp: "Sftp",
    GenericHttp: "GenericHttp",
    ODataRest: "ODataRest",
    Odbc: "Odbc",
    GenericRest: "GenericRest",
    AmazonMws: "AmazonMws",
    Concur: "Concur",
    Dynamics: "Dynamics",
    DynamicsAx: "DynamicsAx",
    DynamicsCrm: "DynamicsCrm",
    GoogleAdWords: "GoogleAdWords",
    Hubspot: "Hubspot",
    Jira: "Jira",
    Magento: "Magento",
    Marketo: "Marketo",
    Office365: "Office365",
    Eloqua: "Eloqua",
    Responsys: "Responsys",
    OracleServiceCloud: "OracleServiceCloud",
    PayPal: "PayPal",
    QuickBooks: "QuickBooks",
    Salesforce: "Salesforce",
    SalesforceServiceCloud: "SalesforceServiceCloud",
    SalesforceMarketingCloud: "SalesforceMarketingCloud",
    SapCloudForCustomer: "SapCloudForCustomer",
    SapEcc: "SapEcc",
    ServiceNow: "ServiceNow",
    SharePointOnlineList: "SharePointOnlineList",
    Shopify: "Shopify",
    Square: "Square",
    WebTable: "WebTable",
    Xero: "Xero",
    Zoho: "Zoho",
    GenericContainerRegistry: "GenericContainerRegistry",
    Elasticsearch: "Elasticsearch",
    OpenAI: "OpenAI",
    Serp: "Serp",
    BingLLMSearch: "BingLLMSearch",
    Serverless: "Serverless",
    ManagedOnlineEndpoint: "ManagedOnlineEndpoint",
};
exports.ContentLevel = {
    Low: "Low",
    Medium: "Medium",
    High: "High",
};
exports.DeploymentModelVersionUpgradeOption = {
    OnceNewDefaultVersionAvailable: "OnceNewDefaultVersionAvailable",
    OnceCurrentVersionExpired: "OnceCurrentVersionExpired",
    NoAutoUpgrade: "NoAutoUpgrade",
};
exports.DeploymentScaleType = {
    Standard: "Standard",
    Manual: "Manual",
};
exports.EncryptionScopeState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.HostingModel = {
    Web: "Web",
    ConnectedContainer: "ConnectedContainer",
    DisconnectedContainer: "DisconnectedContainer",
    ProvisionedWeb: "ProvisionedWeb",
};
exports.IdentityKind = {
    /**
     * Represents a class identity, used for agentic applications.
     */
    AgentBlueprint: "AgentBlueprint",
    /**
     * Represents an instance identity.
     */
    AgentInstance: "AgentInstance",
    /**
     * Represents an agentic instance identity with user-like traits.
     */
    AgenticUser: "AgenticUser",
    /**
     * Represents a classic managed identity.
     */
    Managed: "Managed",
    /**
     * No identity.
     */
    None: "None",
};
exports.IdentityManagementType = {
    /**
     * Platform-managed identity.
     */
    System: "System",
    /**
     * User-managed identity.
     */
    User: "User",
    /**
     * No identity.
     */
    None: "None",
};
exports.KeySource = {
    Microsoft_CognitiveServices: "Microsoft.CognitiveServices",
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.ManagedPERequirement = {
    Required: "Required",
    NotRequired: "NotRequired",
    NotApplicable: "NotApplicable",
};
exports.ManagedPEStatus = {
    Inactive: "Inactive",
    Active: "Active",
    NotApplicable: "NotApplicable",
};
exports.NetworkRuleAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RaiActionType = {
    None: "None",
    BLOCKING: "BLOCKING",
    ANNOTATING: "ANNOTATING",
    HITL: "HITL",
    RETRY: "RETRY",
};
exports.RaiPolicyContentSource = {
    Prompt: "Prompt",
    Completion: "Completion",
    PreToolCall: "PreToolCall",
    PostToolCall: "PostToolCall",
    PreRun: "PreRun",
    PostRun: "PostRun",
};
exports.RaiPolicyMode = {
    Default: "Default",
    Deferred: "Deferred",
    Blocking: "Blocking",
    Asynchronous_filter: "Asynchronous_filter",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.RoutingMethods = {
    Priority: "Priority",
    Weighted: "Weighted",
    Performance: "Performance",
};
exports.RuleCategory = {
    Required: "Required",
    Recommended: "Recommended",
    UserDefined: "UserDefined",
    Dependency: "Dependency",
};
exports.RuleStatus = {
    Inactive: "Inactive",
    Active: "Active",
    Provisioning: "Provisioning",
    Deleting: "Deleting",
    Failed: "Failed",
};
exports.RuleType = {
    FQDN: "FQDN",
    PrivateEndpoint: "PrivateEndpoint",
    ServiceTag: "ServiceTag",
};
exports.ScenarioType = {
    None: "none",
    Agent: "agent",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    Enterprise: "Enterprise",
};
exports.TrafficRoutingProtocol = {
    /**
     * Percentage based routing
     */
    FixedRatio: "FixedRatio",
};
