// @pulumi/azure-native/mission/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workload = exports.VirtualEnclave = exports.TransitHub = exports.getWorkloadOutput = exports.getWorkload = exports.getVirtualEnclaveOutput = exports.getVirtualEnclave = exports.getTransitHubOutput = exports.getTransitHub = exports.getEnclaveEndpointOutput = exports.getEnclaveEndpoint = exports.getEnclaveConnectionOutput = exports.getEnclaveConnection = exports.getCommunityEndpointOutput = exports.getCommunityEndpoint = exports.getCommunityOutput = exports.getCommunity = exports.getApprovalOutput = exports.getApproval = exports.EnclaveEndpoint = exports.EnclaveConnection = exports.CommunityEndpoint = exports.Community = exports.Approval = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Approval = null;
utilities.lazyLoad(exports, ["Approval"], () => require("./approval"));
exports.Community = null;
utilities.lazyLoad(exports, ["Community"], () => require("./community"));
exports.CommunityEndpoint = null;
utilities.lazyLoad(exports, ["CommunityEndpoint"], () => require("./communityEndpoint"));
exports.EnclaveConnection = null;
utilities.lazyLoad(exports, ["EnclaveConnection"], () => require("./enclaveConnection"));
exports.EnclaveEndpoint = null;
utilities.lazyLoad(exports, ["EnclaveEndpoint"], () => require("./enclaveEndpoint"));
exports.getApproval = null;
exports.getApprovalOutput = null;
utilities.lazyLoad(exports, ["getApproval", "getApprovalOutput"], () => require("./getApproval"));
exports.getCommunity = null;
exports.getCommunityOutput = null;
utilities.lazyLoad(exports, ["getCommunity", "getCommunityOutput"], () => require("./getCommunity"));
exports.getCommunityEndpoint = null;
exports.getCommunityEndpointOutput = null;
utilities.lazyLoad(exports, ["getCommunityEndpoint", "getCommunityEndpointOutput"], () => require("./getCommunityEndpoint"));
exports.getEnclaveConnection = null;
exports.getEnclaveConnectionOutput = null;
utilities.lazyLoad(exports, ["getEnclaveConnection", "getEnclaveConnectionOutput"], () => require("./getEnclaveConnection"));
exports.getEnclaveEndpoint = null;
exports.getEnclaveEndpointOutput = null;
utilities.lazyLoad(exports, ["getEnclaveEndpoint", "getEnclaveEndpointOutput"], () => require("./getEnclaveEndpoint"));
exports.getTransitHub = null;
exports.getTransitHubOutput = null;
utilities.lazyLoad(exports, ["getTransitHub", "getTransitHubOutput"], () => require("./getTransitHub"));
exports.getVirtualEnclave = null;
exports.getVirtualEnclaveOutput = null;
utilities.lazyLoad(exports, ["getVirtualEnclave", "getVirtualEnclaveOutput"], () => require("./getVirtualEnclave"));
exports.getWorkload = null;
exports.getWorkloadOutput = null;
utilities.lazyLoad(exports, ["getWorkload", "getWorkloadOutput"], () => require("./getWorkload"));
exports.TransitHub = null;
utilities.lazyLoad(exports, ["TransitHub"], () => require("./transitHub"));
exports.VirtualEnclave = null;
utilities.lazyLoad(exports, ["VirtualEnclave"], () => require("./virtualEnclave"));
exports.Workload = null;
utilities.lazyLoad(exports, ["Workload"], () => require("./workload"));
// Export enums:
__exportStar(require("../types/enums/mission"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mission:Approval":
                return new exports.Approval(name, undefined, { urn });
            case "azure-native:mission:Community":
                return new exports.Community(name, undefined, { urn });
            case "azure-native:mission:CommunityEndpoint":
                return new exports.CommunityEndpoint(name, undefined, { urn });
            case "azure-native:mission:EnclaveConnection":
                return new exports.EnclaveConnection(name, undefined, { urn });
            case "azure-native:mission:EnclaveEndpoint":
                return new exports.EnclaveEndpoint(name, undefined, { urn });
            case "azure-native:mission:TransitHub":
                return new exports.TransitHub(name, undefined, { urn });
            case "azure-native:mission:VirtualEnclave":
                return new exports.VirtualEnclave(name, undefined, { urn });
            case "azure-native:mission:Workload":
                return new exports.Workload(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mission", _module);
