// @pulumi/aws/quicksight/template.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Template = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing a QuickSight Template.
 *
 * ## Example Usage
 *
 * ### From Source Template
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.quicksight.Template("example", {
 *     templateId: "example-id",
 *     name: "example-name",
 *     versionDescription: "version",
 *     sourceEntity: {
 *         sourceTemplate: {
 *             arn: source.arn,
 *         },
 *     },
 * });
 * ```
 *
 * ### With Definition
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.quicksight.Template("example", {
 *     templateId: "example-id",
 *     name: "example-name",
 *     versionDescription: "version",
 *     definition: {
 *         dataSetConfigurations: [{
 *             dataSetSchema: {
 *                 columnSchemaLists: [
 *                     {
 *                         name: "Column1",
 *                         dataType: "STRING",
 *                     },
 *                     {
 *                         name: "Column2",
 *                         dataType: "INTEGER",
 *                     },
 *                 ],
 *             },
 *             placeholder: "1",
 *         }],
 *         sheets: [{
 *             title: "Test",
 *             sheetId: "Test1",
 *             visuals: [{
 *                 barChartVisual: {
 *                     visualId: "BarChart",
 *                     chartConfiguration: {
 *                         fieldWells: {
 *                             barChartAggregatedFieldWells: {
 *                                 categories: [{
 *                                     categoricalDimensionField: {
 *                                         fieldId: "1",
 *                                         column: {
 *                                             columnName: "Column1",
 *                                             dataSetIdentifier: "1",
 *                                         },
 *                                     },
 *                                 }],
 *                                 values: [{
 *                                     numericalMeasureField: {
 *                                         fieldId: "2",
 *                                         column: {
 *                                             columnName: "Column2",
 *                                             dataSetIdentifier: "1",
 *                                         },
 *                                         aggregationFunction: {
 *                                             simpleNumericalAggregation: "SUM",
 *                                         },
 *                                     },
 *                                 }],
 *                             },
 *                         },
 *                     },
 *                 },
 *             }],
 *         }],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a QuickSight Template using the AWS account ID and template ID separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:quicksight/template:Template example 123456789012,example-id
 * ```
 */
class Template extends pulumi.CustomResource {
    /**
     * Get an existing Template resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Template(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:quicksight/template:Template';
    /**
     * Returns true if the given object is an instance of Template.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Template.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsAccountId"] = state?.awsAccountId;
            resourceInputs["createdTime"] = state?.createdTime;
            resourceInputs["lastUpdatedTime"] = state?.lastUpdatedTime;
            resourceInputs["name"] = state?.name;
            resourceInputs["permissions"] = state?.permissions;
            resourceInputs["region"] = state?.region;
            resourceInputs["sourceEntity"] = state?.sourceEntity;
            resourceInputs["sourceEntityArn"] = state?.sourceEntityArn;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["templateId"] = state?.templateId;
            resourceInputs["versionDescription"] = state?.versionDescription;
            resourceInputs["versionNumber"] = state?.versionNumber;
        }
        else {
            const args = argsOrState;
            if (args?.templateId === undefined && !opts.urn) {
                throw new Error("Missing required property 'templateId'");
            }
            if (args?.versionDescription === undefined && !opts.urn) {
                throw new Error("Missing required property 'versionDescription'");
            }
            resourceInputs["awsAccountId"] = args?.awsAccountId;
            resourceInputs["name"] = args?.name;
            resourceInputs["permissions"] = args?.permissions;
            resourceInputs["region"] = args?.region;
            resourceInputs["sourceEntity"] = args?.sourceEntity;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["templateId"] = args?.templateId;
            resourceInputs["versionDescription"] = args?.versionDescription;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["sourceEntityArn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["versionNumber"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Template.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Template = Template;
