// @pulumi/gcp/vertex/aiReasoningEngine.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiReasoningEngine = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * ReasoningEngine provides a customizable runtime for models to determine which actions to take and in which order.
 *
 * To get more information about ReasoningEngine, see:
 *
 * * [API documentation](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.reasoningEngines/)
 * * How-to Guides
 *     * [Develop and deploy agents on Vertex AI Agent Engine](https://cloud.google.com/vertex-ai/generative-ai/docs/agent-engine/quickstart)
 *
 * ## Example Usage
 *
 * ### Vertex Ai Reasoning Engine Source Based Deployment
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "A basic reasoning engine",
 *     region: "us-central1",
 *     spec: {
 *         sourceCodeSpec: {
 *             inlineSource: {
 *                 sourceArchive: std.filebase64({
 *                     input: "./test-fixtures/source.tar.gz",
 *                 }).then(invoke => invoke.result),
 *             },
 *             pythonSpec: {
 *                 entrypointModule: "simple_agent",
 *                 entrypointObject: "fixed_name_generator",
 *                 version: "3.14",
 *             },
 *         },
 *     },
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Developer Connect Source
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "A basic reasoning engine",
 *     region: "us-central1",
 *     spec: {
 *         sourceCodeSpec: {
 *             developerConnectSource: {
 *                 config: {
 *                     gitRepositoryLink: project.then(project => `projects/${project.projectId}/locations/us-central1/connections/tpg-test-bot-github/gitRepositoryLinks/tpg-test-vertex-reasoning`),
 *                     dir: "source",
 *                     revision: "main",
 *                 },
 *             },
 *             pythonSpec: {
 *                 version: "3.14",
 *                 entrypointModule: "simple_agent",
 *                 entrypointObject: "fixed_name_generator",
 *             },
 *         },
 *     },
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Image Spec
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "Deployed with BYOC Dockerfile through Terraform",
 *     region: "us-central1",
 *     spec: {
 *         sourceCodeSpec: {
 *             inlineSource: {
 *                 sourceArchive: std.filebase64({
 *                     input: "./test-fixtures/agent_src.tar.gz",
 *                 }).then(invoke => invoke.result),
 *             },
 *             imageSpec: {
 *                 buildArgs: {},
 *             },
 *         },
 *     },
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Byoc
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const project = gcp.organizations.getProject({});
 * const vertexArReader = new gcp.projects.IAMMember("vertex_ar_reader", {
 *     project: project.then(project => project.projectId),
 *     role: "roles/artifactregistry.reader",
 *     member: project.then(project => `serviceAccount:service-${project.number}@gcp-sa-aiplatform-re.iam.gserviceaccount.com`),
 * });
 * // Provision and retrieve the tenant service agent through another agent
 * const tenantMdsAiReasoningEngine = new gcp.vertex.AiReasoningEngine("tenant_mds", {
 *     displayName: "reasoning-engine-mds",
 *     region: "us-central1",
 *     spec: {
 *         sourceCodeSpec: {
 *             inlineSource: {
 *                 sourceArchive: std.filebase64({
 *                     input: "./test-fixtures/mds_agent_src.tar.gz",
 *                 }).then(invoke => invoke.result),
 *             },
 *             pythonSpec: {
 *                 entrypointModule: "metadata_agent",
 *                 entrypointObject: "root_agent",
 *             },
 *         },
 *     },
 * });
 * const tenantMds = gcp.vertex.getAiReasoningEngineQueryOutput({
 *     region: "us-central1",
 *     reasoningEngineId: tenantMdsAiReasoningEngine.name,
 * });
 * const tenantArReader = new gcp.projects.IAMMember("tenant_ar_reader", {
 *     project: project.then(project => project.projectId),
 *     role: "roles/artifactregistry.reader",
 *     member: std.jsondecodeOutput({
 *         input: tenantMds.apply(tenantMds => tenantMds.output),
 *     }).apply(invoke => `serviceAccount:${invoke.result?.output}`),
 * });
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "Deployed with BYOC through Terraform",
 *     region: "us-central1",
 *     spec: {
 *         containerSpec: {
 *             imageUri: project.then(project => `us-central1-docker.pkg.dev/${project.projectId}/vertex-byoc/byoc-agent:latest`),
 *         },
 *     },
 * }, {
 *     dependsOn: [
 *         vertexArReader,
 *         tenantArReader,
 *     ],
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Psc Interface
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumiverse/time";
 *
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "reasoning-engine",
 *     location: "us-central1",
 *     uniformBucketLevelAccess: true,
 *     forceDestroy: true,
 * });
 * const bucketObjRequirementsTxt = new gcp.storage.BucketObject("bucket_obj_requirements_txt", {
 *     name: "requirements.txt",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/requirements_adk.txt"),
 * });
 * const bucketObjPickle = new gcp.storage.BucketObject("bucket_obj_pickle", {
 *     name: "code.pkl",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/pickle_adk.pkl"),
 * });
 * const bucketObjDependenciesTarGz = new gcp.storage.BucketObject("bucket_obj_dependencies_tar_gz", {
 *     name: "dependencies.tar.gz",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/dependencies_adk.tar.gz"),
 * });
 * const network = new gcp.compute.Network("network", {
 *     name: "network",
 *     autoCreateSubnetworks: false,
 * });
 * const subnetwork = new gcp.compute.Subnetwork("subnetwork", {
 *     name: "subnetwork",
 *     region: "us-central1",
 *     ipCidrRange: "10.0.0.0/16",
 *     network: network.id,
 * });
 * const networkAttachment = new gcp.compute.NetworkAttachment("network_attachment", {
 *     name: "network-attachment",
 *     region: "us-central1",
 *     connectionPreference: "ACCEPT_MANUAL",
 *     subnetworks: [subnetwork.id],
 * });
 * // Destroy network attachment 35 minutes after reasoning engine is deleted.
 * // It guarantees that the network attachment has no more active PSC interfaces.
 * const wait35Minutes = new time.Sleep("wait_35_minutes", {destroyDuration: "35m"}, {
 *     dependsOn: [networkAttachment],
 * });
 * const project = gcp.organizations.getProject({});
 * // When PSC-I is configured, Agent deletion will fail,
 * // although the agent will be deleted.
 * // Bug at https://github.com/hashicorp/terraform-provider-google/issues/25637
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "A basic reasoning engine",
 *     region: "us-central1",
 *     spec: {
 *         agentFramework: "google-adk",
 *         packageSpec: {
 *             pythonVersion: "3.11",
 *             dependencyFilesGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjDependenciesTarGz.name}`,
 *             pickleObjectGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjPickle.name}`,
 *             requirementsGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjRequirementsTxt.name}`,
 *         },
 *         deploymentSpec: {
 *             pscInterfaceConfig: {
 *                 networkAttachment: networkAttachment.id,
 *                 dnsPeeringConfigs: [{
 *                     domain: "example.com.",
 *                     targetProject: project.then(project => project.projectId),
 *                     targetNetwork: network.name,
 *                 }],
 *             },
 *         },
 *     },
 * }, {
 *     dependsOn: [wait35Minutes],
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumiverse/time";
 *
 * const classMethods = [{
 *     apiMode: "async",
 *     description: null,
 *     name: "async_query",
 *     parameters: {
 *         type: "object",
 *         required: [],
 *         properties: {},
 *     },
 * }];
 * const secret = new gcp.secretmanager.Secret("secret", {
 *     secretId: "secret",
 *     replication: {
 *         auto: {},
 *     },
 * });
 * const secretVersion = new gcp.secretmanager.SecretVersion("secret_version", {
 *     secret: secret.id,
 *     secretData: "test",
 * });
 * const serviceAccount = new gcp.serviceaccount.Account("service_account", {accountId: "sa"});
 * const secretAccess = new gcp.secretmanager.SecretIamMember("secret_access", {
 *     secretId: secret.id,
 *     role: "roles/secretmanager.secretAccessor",
 *     member: serviceAccount.member,
 * });
 * const project = gcp.organizations.getProject({});
 * const saIamObjectViewer = new gcp.projects.IAMMember("sa_iam_object_viewer", {
 *     role: "roles/storage.objectViewer",
 *     project: project.then(project => project.id),
 *     member: serviceAccount.member,
 * });
 * const saIamAiPlatformUser = new gcp.projects.IAMMember("sa_iam_ai_platform_user", {
 *     role: "roles/aiplatform.user",
 *     project: project.then(project => project.id),
 *     member: serviceAccount.member,
 * });
 * const saIamViewer = new gcp.projects.IAMMember("sa_iam_viewer", {
 *     role: "roles/viewer",
 *     project: project.then(project => project.id),
 *     member: serviceAccount.member,
 * });
 * // Ensure we wait enough time for IAM permissions to be propagated
 * const wait5Minutes = new time.Sleep("wait_5_minutes", {createDuration: "5m"}, {
 *     dependsOn: [
 *         saIamAiPlatformUser,
 *         saIamObjectViewer,
 *         saIamViewer,
 *         secretAccess,
 *         secretVersion,
 *     ],
 * });
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "reasoning-engine",
 *     location: "us-central1",
 *     uniformBucketLevelAccess: true,
 *     forceDestroy: true,
 * });
 * const bucketObjRequirementsTxt = new gcp.storage.BucketObject("bucket_obj_requirements_txt", {
 *     name: "requirements.txt",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/requirements_adk.txt"),
 * });
 * const bucketObjPickle = new gcp.storage.BucketObject("bucket_obj_pickle", {
 *     name: "code.pkl",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/pickle_adk.pkl"),
 * });
 * const bucketObjDependenciesTarGz = new gcp.storage.BucketObject("bucket_obj_dependencies_tar_gz", {
 *     name: "dependencies.tar.gz",
 *     bucket: bucket.id,
 *     source: new pulumi.asset.FileAsset("./test-fixtures/dependencies_adk.tar.gz"),
 * });
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "reasoning-engine",
 *     description: "A basic reasoning engine",
 *     region: "us-central1",
 *     encryptionSpec: {
 *         kmsKeyName: "example-key",
 *     },
 *     spec: {
 *         agentFramework: "google-adk",
 *         classMethods: JSON.stringify(classMethods),
 *         serviceAccount: serviceAccount.email,
 *         deploymentSpec: {
 *             minInstances: 1,
 *             maxInstances: 3,
 *             containerConcurrency: 5,
 *             resourceLimits: {
 *                 cpu: "4",
 *                 memory: "4Gi",
 *             },
 *             envs: [
 *                 {
 *                     name: "var_1",
 *                     value: "value_2",
 *                 },
 *                 {
 *                     name: "var_2",
 *                     value: "value_2",
 *                 },
 *             ],
 *             secretEnvs: [
 *                 {
 *                     name: "secret_var_1",
 *                     secretRef: {
 *                         secret: secret.secretId,
 *                         version: "latest",
 *                     },
 *                 },
 *                 {
 *                     name: "secret_var_2",
 *                     secretRef: {
 *                         secret: secret.secretId,
 *                         version: "latest",
 *                     },
 *                 },
 *             ],
 *         },
 *         packageSpec: {
 *             dependencyFilesGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjDependenciesTarGz.name}`,
 *             pickleObjectGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjPickle.name}`,
 *             pythonVersion: "3.11",
 *             requirementsGcsUri: pulumi.interpolate`${bucket.url}/${bucketObjRequirementsTxt.name}`,
 *         },
 *     },
 * }, {
 *     dependsOn: [wait5Minutes],
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Context Spec
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "re-ctx-spec",
 *     description: "Reasoning engine with context spec",
 *     region: "us-central1",
 *     contextSpec: {
 *         memoryBankConfig: {
 *             generationConfig: {
 *                 model: project.then(project => `projects/${project.projectId}/locations/us-central1/publishers/google/models/gemini-2.5-flash`),
 *             },
 *             similaritySearchConfig: {
 *                 embeddingModel: project.then(project => `projects/${project.projectId}/locations/us-central1/publishers/google/models/text-embedding-005`),
 *             },
 *             disableMemoryRevisions: false,
 *             ttlConfig: {
 *                 defaultTtl: "86400s",
 *             },
 *         },
 *     },
 * });
 * ```
 * ### Vertex Ai Reasoning Engine Granular Ttl
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const reasoningEngine = new gcp.vertex.AiReasoningEngine("reasoning_engine", {
 *     displayName: "re-gran-ttl",
 *     description: "Reasoning engine with granular ttl",
 *     region: "us-central1",
 *     contextSpec: {
 *         memoryBankConfig: {
 *             generationConfig: {
 *                 model: project.then(project => `projects/${project.projectId}/locations/us-central1/publishers/google/models/gemini-2.5-flash`),
 *             },
 *             similaritySearchConfig: {
 *                 embeddingModel: project.then(project => `projects/${project.projectId}/locations/us-central1/publishers/google/models/text-embedding-005`),
 *             },
 *             disableMemoryRevisions: false,
 *             ttlConfig: {
 *                 memoryRevisionDefaultTtl: "86400s",
 *                 granularTtlConfig: {
 *                     createTtl: "86400s",
 *                     generateCreatedTtl: "86400s",
 *                     generateUpdatedTtl: "86400s",
 *                 },
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ReasoningEngine can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{region}}/reasoningEngines/{{name}}`
 * * `{{project}}/{{region}}/{{name}}`
 * * `{{region}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, ReasoningEngine can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:vertex/aiReasoningEngine:AiReasoningEngine default projects/{{project}}/locations/{{region}}/reasoningEngines/{{name}}
 * $ pulumi import gcp:vertex/aiReasoningEngine:AiReasoningEngine default {{project}}/{{region}}/{{name}}
 * $ pulumi import gcp:vertex/aiReasoningEngine:AiReasoningEngine default {{region}}/{{name}}
 * $ pulumi import gcp:vertex/aiReasoningEngine:AiReasoningEngine default {{name}}
 * ```
 */
class AiReasoningEngine extends pulumi.CustomResource {
    /**
     * Get an existing AiReasoningEngine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AiReasoningEngine(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:vertex/aiReasoningEngine:AiReasoningEngine';
    /**
     * Returns true if the given object is an instance of AiReasoningEngine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AiReasoningEngine.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["contextSpec"] = state?.contextSpec;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["encryptionSpec"] = state?.encryptionSpec;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["region"] = state?.region;
            resourceInputs["spec"] = state?.spec;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            resourceInputs["contextSpec"] = args?.contextSpec;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["encryptionSpec"] = args?.encryptionSpec;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["project"] = args?.project;
            resourceInputs["region"] = args?.region;
            resourceInputs["spec"] = args?.spec;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(AiReasoningEngine.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AiReasoningEngine = AiReasoningEngine;
