// @pulumi/gcp/vertex/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAiFeaturestoreEntitytypeIamPolicyOutput = exports.getAiFeaturestoreEntitytypeIamPolicy = exports.getAiFeatureOnlineStoreIamPolicyOutput = exports.getAiFeatureOnlineStoreIamPolicy = exports.getAiFeatureOnlineStoreFeatureviewIamPolicyOutput = exports.getAiFeatureOnlineStoreFeatureviewIamPolicy = exports.getAiFeatureGroupIamPolicyOutput = exports.getAiFeatureGroupIamPolicy = exports.getAiEndpointIamPolicyOutput = exports.getAiEndpointIamPolicy = exports.AiTensorboard = exports.AiReasoningEngineIamPolicy = exports.AiReasoningEngineIamMember = exports.AiReasoningEngineIamBinding = exports.AiReasoningEngine = exports.AiRagEngineConfig = exports.AiMetadataStore = exports.AiIndexEndpointDeployedIndex = exports.AiIndexEndpoint = exports.AiIndex = exports.AiFeatureStoreIamPolicy = exports.AiFeatureStoreIamMember = exports.AiFeatureStoreIamBinding = exports.AiFeatureStoreEntityTypeIamPolicy = exports.AiFeatureStoreEntityTypeIamMember = exports.AiFeatureStoreEntityTypeIamBinding = exports.AiFeatureStoreEntityTypeFeature = exports.AiFeatureStoreEntityType = exports.AiFeatureStore = exports.AiFeatureOnlineStoreIamPolicy = exports.AiFeatureOnlineStoreIamMember = exports.AiFeatureOnlineStoreIamBinding = exports.AiFeatureOnlineStoreFeatureviewIamPolicy = exports.AiFeatureOnlineStoreFeatureviewIamMember = exports.AiFeatureOnlineStoreFeatureviewIamBinding = exports.AiFeatureOnlineStoreFeatureview = exports.AiFeatureOnlineStore = exports.AiFeatureGroupIamPolicy = exports.AiFeatureGroupIamMember = exports.AiFeatureGroupIamBinding = exports.AiFeatureGroupFeature = exports.AiFeatureGroup = exports.AiEndpointWithModelGardenDeployment = exports.AiEndpointIamPolicy = exports.AiEndpointIamMember = exports.AiEndpointIamBinding = exports.AiEndpoint = exports.AiDeploymentResourcePool = exports.AiDataset = exports.AiCacheConfig = void 0;
exports.getAiReasoningEngineQueryOutput = exports.getAiReasoningEngineQuery = exports.getAiReasoningEngineIamPolicyOutput = exports.getAiReasoningEngineIamPolicy = exports.getAiIndexOutput = exports.getAiIndex = exports.getAiFeaturestoreIamPolicyOutput = exports.getAiFeaturestoreIamPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AiCacheConfig = null;
utilities.lazyLoad(exports, ["AiCacheConfig"], () => require("./aiCacheConfig"));
exports.AiDataset = null;
utilities.lazyLoad(exports, ["AiDataset"], () => require("./aiDataset"));
exports.AiDeploymentResourcePool = null;
utilities.lazyLoad(exports, ["AiDeploymentResourcePool"], () => require("./aiDeploymentResourcePool"));
exports.AiEndpoint = null;
utilities.lazyLoad(exports, ["AiEndpoint"], () => require("./aiEndpoint"));
exports.AiEndpointIamBinding = null;
utilities.lazyLoad(exports, ["AiEndpointIamBinding"], () => require("./aiEndpointIamBinding"));
exports.AiEndpointIamMember = null;
utilities.lazyLoad(exports, ["AiEndpointIamMember"], () => require("./aiEndpointIamMember"));
exports.AiEndpointIamPolicy = null;
utilities.lazyLoad(exports, ["AiEndpointIamPolicy"], () => require("./aiEndpointIamPolicy"));
exports.AiEndpointWithModelGardenDeployment = null;
utilities.lazyLoad(exports, ["AiEndpointWithModelGardenDeployment"], () => require("./aiEndpointWithModelGardenDeployment"));
exports.AiFeatureGroup = null;
utilities.lazyLoad(exports, ["AiFeatureGroup"], () => require("./aiFeatureGroup"));
exports.AiFeatureGroupFeature = null;
utilities.lazyLoad(exports, ["AiFeatureGroupFeature"], () => require("./aiFeatureGroupFeature"));
exports.AiFeatureGroupIamBinding = null;
utilities.lazyLoad(exports, ["AiFeatureGroupIamBinding"], () => require("./aiFeatureGroupIamBinding"));
exports.AiFeatureGroupIamMember = null;
utilities.lazyLoad(exports, ["AiFeatureGroupIamMember"], () => require("./aiFeatureGroupIamMember"));
exports.AiFeatureGroupIamPolicy = null;
utilities.lazyLoad(exports, ["AiFeatureGroupIamPolicy"], () => require("./aiFeatureGroupIamPolicy"));
exports.AiFeatureOnlineStore = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStore"], () => require("./aiFeatureOnlineStore"));
exports.AiFeatureOnlineStoreFeatureview = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreFeatureview"], () => require("./aiFeatureOnlineStoreFeatureview"));
exports.AiFeatureOnlineStoreFeatureviewIamBinding = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreFeatureviewIamBinding"], () => require("./aiFeatureOnlineStoreFeatureviewIamBinding"));
exports.AiFeatureOnlineStoreFeatureviewIamMember = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreFeatureviewIamMember"], () => require("./aiFeatureOnlineStoreFeatureviewIamMember"));
exports.AiFeatureOnlineStoreFeatureviewIamPolicy = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreFeatureviewIamPolicy"], () => require("./aiFeatureOnlineStoreFeatureviewIamPolicy"));
exports.AiFeatureOnlineStoreIamBinding = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreIamBinding"], () => require("./aiFeatureOnlineStoreIamBinding"));
exports.AiFeatureOnlineStoreIamMember = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreIamMember"], () => require("./aiFeatureOnlineStoreIamMember"));
exports.AiFeatureOnlineStoreIamPolicy = null;
utilities.lazyLoad(exports, ["AiFeatureOnlineStoreIamPolicy"], () => require("./aiFeatureOnlineStoreIamPolicy"));
exports.AiFeatureStore = null;
utilities.lazyLoad(exports, ["AiFeatureStore"], () => require("./aiFeatureStore"));
exports.AiFeatureStoreEntityType = null;
utilities.lazyLoad(exports, ["AiFeatureStoreEntityType"], () => require("./aiFeatureStoreEntityType"));
exports.AiFeatureStoreEntityTypeFeature = null;
utilities.lazyLoad(exports, ["AiFeatureStoreEntityTypeFeature"], () => require("./aiFeatureStoreEntityTypeFeature"));
exports.AiFeatureStoreEntityTypeIamBinding = null;
utilities.lazyLoad(exports, ["AiFeatureStoreEntityTypeIamBinding"], () => require("./aiFeatureStoreEntityTypeIamBinding"));
exports.AiFeatureStoreEntityTypeIamMember = null;
utilities.lazyLoad(exports, ["AiFeatureStoreEntityTypeIamMember"], () => require("./aiFeatureStoreEntityTypeIamMember"));
exports.AiFeatureStoreEntityTypeIamPolicy = null;
utilities.lazyLoad(exports, ["AiFeatureStoreEntityTypeIamPolicy"], () => require("./aiFeatureStoreEntityTypeIamPolicy"));
exports.AiFeatureStoreIamBinding = null;
utilities.lazyLoad(exports, ["AiFeatureStoreIamBinding"], () => require("./aiFeatureStoreIamBinding"));
exports.AiFeatureStoreIamMember = null;
utilities.lazyLoad(exports, ["AiFeatureStoreIamMember"], () => require("./aiFeatureStoreIamMember"));
exports.AiFeatureStoreIamPolicy = null;
utilities.lazyLoad(exports, ["AiFeatureStoreIamPolicy"], () => require("./aiFeatureStoreIamPolicy"));
exports.AiIndex = null;
utilities.lazyLoad(exports, ["AiIndex"], () => require("./aiIndex"));
exports.AiIndexEndpoint = null;
utilities.lazyLoad(exports, ["AiIndexEndpoint"], () => require("./aiIndexEndpoint"));
exports.AiIndexEndpointDeployedIndex = null;
utilities.lazyLoad(exports, ["AiIndexEndpointDeployedIndex"], () => require("./aiIndexEndpointDeployedIndex"));
exports.AiMetadataStore = null;
utilities.lazyLoad(exports, ["AiMetadataStore"], () => require("./aiMetadataStore"));
exports.AiRagEngineConfig = null;
utilities.lazyLoad(exports, ["AiRagEngineConfig"], () => require("./aiRagEngineConfig"));
exports.AiReasoningEngine = null;
utilities.lazyLoad(exports, ["AiReasoningEngine"], () => require("./aiReasoningEngine"));
exports.AiReasoningEngineIamBinding = null;
utilities.lazyLoad(exports, ["AiReasoningEngineIamBinding"], () => require("./aiReasoningEngineIamBinding"));
exports.AiReasoningEngineIamMember = null;
utilities.lazyLoad(exports, ["AiReasoningEngineIamMember"], () => require("./aiReasoningEngineIamMember"));
exports.AiReasoningEngineIamPolicy = null;
utilities.lazyLoad(exports, ["AiReasoningEngineIamPolicy"], () => require("./aiReasoningEngineIamPolicy"));
exports.AiTensorboard = null;
utilities.lazyLoad(exports, ["AiTensorboard"], () => require("./aiTensorboard"));
exports.getAiEndpointIamPolicy = null;
exports.getAiEndpointIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiEndpointIamPolicy", "getAiEndpointIamPolicyOutput"], () => require("./getAiEndpointIamPolicy"));
exports.getAiFeatureGroupIamPolicy = null;
exports.getAiFeatureGroupIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiFeatureGroupIamPolicy", "getAiFeatureGroupIamPolicyOutput"], () => require("./getAiFeatureGroupIamPolicy"));
exports.getAiFeatureOnlineStoreFeatureviewIamPolicy = null;
exports.getAiFeatureOnlineStoreFeatureviewIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiFeatureOnlineStoreFeatureviewIamPolicy", "getAiFeatureOnlineStoreFeatureviewIamPolicyOutput"], () => require("./getAiFeatureOnlineStoreFeatureviewIamPolicy"));
exports.getAiFeatureOnlineStoreIamPolicy = null;
exports.getAiFeatureOnlineStoreIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiFeatureOnlineStoreIamPolicy", "getAiFeatureOnlineStoreIamPolicyOutput"], () => require("./getAiFeatureOnlineStoreIamPolicy"));
exports.getAiFeaturestoreEntitytypeIamPolicy = null;
exports.getAiFeaturestoreEntitytypeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiFeaturestoreEntitytypeIamPolicy", "getAiFeaturestoreEntitytypeIamPolicyOutput"], () => require("./getAiFeaturestoreEntitytypeIamPolicy"));
exports.getAiFeaturestoreIamPolicy = null;
exports.getAiFeaturestoreIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiFeaturestoreIamPolicy", "getAiFeaturestoreIamPolicyOutput"], () => require("./getAiFeaturestoreIamPolicy"));
exports.getAiIndex = null;
exports.getAiIndexOutput = null;
utilities.lazyLoad(exports, ["getAiIndex", "getAiIndexOutput"], () => require("./getAiIndex"));
exports.getAiReasoningEngineIamPolicy = null;
exports.getAiReasoningEngineIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAiReasoningEngineIamPolicy", "getAiReasoningEngineIamPolicyOutput"], () => require("./getAiReasoningEngineIamPolicy"));
exports.getAiReasoningEngineQuery = null;
exports.getAiReasoningEngineQueryOutput = null;
utilities.lazyLoad(exports, ["getAiReasoningEngineQuery", "getAiReasoningEngineQueryOutput"], () => require("./getAiReasoningEngineQuery"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:vertex/aiCacheConfig:AiCacheConfig":
                return new exports.AiCacheConfig(name, undefined, { urn });
            case "gcp:vertex/aiDataset:AiDataset":
                return new exports.AiDataset(name, undefined, { urn });
            case "gcp:vertex/aiDeploymentResourcePool:AiDeploymentResourcePool":
                return new exports.AiDeploymentResourcePool(name, undefined, { urn });
            case "gcp:vertex/aiEndpoint:AiEndpoint":
                return new exports.AiEndpoint(name, undefined, { urn });
            case "gcp:vertex/aiEndpointIamBinding:AiEndpointIamBinding":
                return new exports.AiEndpointIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiEndpointIamMember:AiEndpointIamMember":
                return new exports.AiEndpointIamMember(name, undefined, { urn });
            case "gcp:vertex/aiEndpointIamPolicy:AiEndpointIamPolicy":
                return new exports.AiEndpointIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiEndpointWithModelGardenDeployment:AiEndpointWithModelGardenDeployment":
                return new exports.AiEndpointWithModelGardenDeployment(name, undefined, { urn });
            case "gcp:vertex/aiFeatureGroup:AiFeatureGroup":
                return new exports.AiFeatureGroup(name, undefined, { urn });
            case "gcp:vertex/aiFeatureGroupFeature:AiFeatureGroupFeature":
                return new exports.AiFeatureGroupFeature(name, undefined, { urn });
            case "gcp:vertex/aiFeatureGroupIamBinding:AiFeatureGroupIamBinding":
                return new exports.AiFeatureGroupIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiFeatureGroupIamMember:AiFeatureGroupIamMember":
                return new exports.AiFeatureGroupIamMember(name, undefined, { urn });
            case "gcp:vertex/aiFeatureGroupIamPolicy:AiFeatureGroupIamPolicy":
                return new exports.AiFeatureGroupIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStore:AiFeatureOnlineStore":
                return new exports.AiFeatureOnlineStore(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreFeatureview:AiFeatureOnlineStoreFeatureview":
                return new exports.AiFeatureOnlineStoreFeatureview(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreFeatureviewIamBinding:AiFeatureOnlineStoreFeatureviewIamBinding":
                return new exports.AiFeatureOnlineStoreFeatureviewIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreFeatureviewIamMember:AiFeatureOnlineStoreFeatureviewIamMember":
                return new exports.AiFeatureOnlineStoreFeatureviewIamMember(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreFeatureviewIamPolicy:AiFeatureOnlineStoreFeatureviewIamPolicy":
                return new exports.AiFeatureOnlineStoreFeatureviewIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreIamBinding:AiFeatureOnlineStoreIamBinding":
                return new exports.AiFeatureOnlineStoreIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreIamMember:AiFeatureOnlineStoreIamMember":
                return new exports.AiFeatureOnlineStoreIamMember(name, undefined, { urn });
            case "gcp:vertex/aiFeatureOnlineStoreIamPolicy:AiFeatureOnlineStoreIamPolicy":
                return new exports.AiFeatureOnlineStoreIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStore:AiFeatureStore":
                return new exports.AiFeatureStore(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreEntityType:AiFeatureStoreEntityType":
                return new exports.AiFeatureStoreEntityType(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreEntityTypeFeature:AiFeatureStoreEntityTypeFeature":
                return new exports.AiFeatureStoreEntityTypeFeature(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreEntityTypeIamBinding:AiFeatureStoreEntityTypeIamBinding":
                return new exports.AiFeatureStoreEntityTypeIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreEntityTypeIamMember:AiFeatureStoreEntityTypeIamMember":
                return new exports.AiFeatureStoreEntityTypeIamMember(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreEntityTypeIamPolicy:AiFeatureStoreEntityTypeIamPolicy":
                return new exports.AiFeatureStoreEntityTypeIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreIamBinding:AiFeatureStoreIamBinding":
                return new exports.AiFeatureStoreIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreIamMember:AiFeatureStoreIamMember":
                return new exports.AiFeatureStoreIamMember(name, undefined, { urn });
            case "gcp:vertex/aiFeatureStoreIamPolicy:AiFeatureStoreIamPolicy":
                return new exports.AiFeatureStoreIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiIndex:AiIndex":
                return new exports.AiIndex(name, undefined, { urn });
            case "gcp:vertex/aiIndexEndpoint:AiIndexEndpoint":
                return new exports.AiIndexEndpoint(name, undefined, { urn });
            case "gcp:vertex/aiIndexEndpointDeployedIndex:AiIndexEndpointDeployedIndex":
                return new exports.AiIndexEndpointDeployedIndex(name, undefined, { urn });
            case "gcp:vertex/aiMetadataStore:AiMetadataStore":
                return new exports.AiMetadataStore(name, undefined, { urn });
            case "gcp:vertex/aiRagEngineConfig:AiRagEngineConfig":
                return new exports.AiRagEngineConfig(name, undefined, { urn });
            case "gcp:vertex/aiReasoningEngine:AiReasoningEngine":
                return new exports.AiReasoningEngine(name, undefined, { urn });
            case "gcp:vertex/aiReasoningEngineIamBinding:AiReasoningEngineIamBinding":
                return new exports.AiReasoningEngineIamBinding(name, undefined, { urn });
            case "gcp:vertex/aiReasoningEngineIamMember:AiReasoningEngineIamMember":
                return new exports.AiReasoningEngineIamMember(name, undefined, { urn });
            case "gcp:vertex/aiReasoningEngineIamPolicy:AiReasoningEngineIamPolicy":
                return new exports.AiReasoningEngineIamPolicy(name, undefined, { urn });
            case "gcp:vertex/aiTensorboard:AiTensorboard":
                return new exports.AiTensorboard(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "vertex/aiCacheConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiDataset", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiDeploymentResourcePool", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiEndpointIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiEndpointIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiEndpointIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiEndpointWithModelGardenDeployment", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureGroupFeature", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureGroupIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureGroupIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureGroupIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStore", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreFeatureview", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreFeatureviewIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreFeatureviewIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreFeatureviewIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureOnlineStoreIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStore", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreEntityType", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreEntityTypeFeature", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreEntityTypeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreEntityTypeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreEntityTypeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiFeatureStoreIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiIndex", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiIndexEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiIndexEndpointDeployedIndex", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiMetadataStore", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiRagEngineConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiReasoningEngine", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiReasoningEngineIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiReasoningEngineIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiReasoningEngineIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "vertex/aiTensorboard", _module);
