// @pulumi/gcp/notebooks/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuntimeIamPolicy = exports.RuntimeIamMember = exports.RuntimeIamBinding = exports.Runtime = exports.InstanceIamPolicy = exports.InstanceIamMember = exports.InstanceIamBinding = exports.Instance = exports.getRuntimeIamPolicyOutput = exports.getRuntimeIamPolicy = exports.getInstanceIamPolicyOutput = exports.getInstanceIamPolicy = exports.Environment = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getInstanceIamPolicy = null;
exports.getInstanceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getInstanceIamPolicy", "getInstanceIamPolicyOutput"], () => require("./getInstanceIamPolicy"));
exports.getRuntimeIamPolicy = null;
exports.getRuntimeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRuntimeIamPolicy", "getRuntimeIamPolicyOutput"], () => require("./getRuntimeIamPolicy"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceIamBinding = null;
utilities.lazyLoad(exports, ["InstanceIamBinding"], () => require("./instanceIamBinding"));
exports.InstanceIamMember = null;
utilities.lazyLoad(exports, ["InstanceIamMember"], () => require("./instanceIamMember"));
exports.InstanceIamPolicy = null;
utilities.lazyLoad(exports, ["InstanceIamPolicy"], () => require("./instanceIamPolicy"));
exports.Runtime = null;
utilities.lazyLoad(exports, ["Runtime"], () => require("./runtime"));
exports.RuntimeIamBinding = null;
utilities.lazyLoad(exports, ["RuntimeIamBinding"], () => require("./runtimeIamBinding"));
exports.RuntimeIamMember = null;
utilities.lazyLoad(exports, ["RuntimeIamMember"], () => require("./runtimeIamMember"));
exports.RuntimeIamPolicy = null;
utilities.lazyLoad(exports, ["RuntimeIamPolicy"], () => require("./runtimeIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:notebooks/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "gcp:notebooks/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "gcp:notebooks/instanceIamBinding:InstanceIamBinding":
                return new exports.InstanceIamBinding(name, undefined, { urn });
            case "gcp:notebooks/instanceIamMember:InstanceIamMember":
                return new exports.InstanceIamMember(name, undefined, { urn });
            case "gcp:notebooks/instanceIamPolicy:InstanceIamPolicy":
                return new exports.InstanceIamPolicy(name, undefined, { urn });
            case "gcp:notebooks/runtime:Runtime":
                return new exports.Runtime(name, undefined, { urn });
            case "gcp:notebooks/runtimeIamBinding:RuntimeIamBinding":
                return new exports.RuntimeIamBinding(name, undefined, { urn });
            case "gcp:notebooks/runtimeIamMember:RuntimeIamMember":
                return new exports.RuntimeIamMember(name, undefined, { urn });
            case "gcp:notebooks/runtimeIamPolicy:RuntimeIamPolicy":
                return new exports.RuntimeIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "notebooks/environment", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/instance", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/instanceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/instanceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/instanceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/runtime", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/runtimeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/runtimeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "notebooks/runtimeIamPolicy", _module);
