// @pulumi/kubernetes/types/input.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.kubeClientSettingsProvideDefaults = exports.helmReleaseSettingsProvideDefaults = void 0;
const utilities = __importStar(require("../utilities"));
/**
 * helmReleaseSettingsProvideDefaults sets the appropriate defaults for HelmReleaseSettings
 */
function helmReleaseSettingsProvideDefaults(val) {
    return {
        ...val,
        driver: (val.driver) ?? utilities.getEnv("PULUMI_K8S_HELM_DRIVER"),
        pluginsPath: (val.pluginsPath) ?? utilities.getEnv("PULUMI_K8S_HELM_PLUGINS_PATH"),
        registryConfigPath: (val.registryConfigPath) ?? utilities.getEnv("PULUMI_K8S_HELM_REGISTRY_CONFIG_PATH"),
        repositoryCache: (val.repositoryCache) ?? utilities.getEnv("PULUMI_K8S_HELM_REPOSITORY_CACHE"),
        repositoryConfigPath: (val.repositoryConfigPath) ?? utilities.getEnv("PULUMI_K8S_HELM_REPOSITORY_CONFIG_PATH"),
    };
}
exports.helmReleaseSettingsProvideDefaults = helmReleaseSettingsProvideDefaults;
/**
 * kubeClientSettingsProvideDefaults sets the appropriate defaults for KubeClientSettings
 */
function kubeClientSettingsProvideDefaults(val) {
    return {
        ...val,
        burst: (val.burst) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_BURST"),
        qps: (val.qps) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_QPS"),
        timeout: (val.timeout) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_TIMEOUT"),
    };
}
exports.kubeClientSettingsProvideDefaults = kubeClientSettingsProvideDefaults;
