// @pulumi/gcp/cloudasset/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectFeed = exports.OrganizationFeed = exports.getSearchAllResourcesOutput = exports.getSearchAllResources = exports.getResourcesSearchAllOutput = exports.getResourcesSearchAll = exports.FolderFeed = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.FolderFeed = null;
utilities.lazyLoad(exports, ["FolderFeed"], () => require("./folderFeed"));
exports.getResourcesSearchAll = null;
exports.getResourcesSearchAllOutput = null;
utilities.lazyLoad(exports, ["getResourcesSearchAll", "getResourcesSearchAllOutput"], () => require("./getResourcesSearchAll"));
exports.getSearchAllResources = null;
exports.getSearchAllResourcesOutput = null;
utilities.lazyLoad(exports, ["getSearchAllResources", "getSearchAllResourcesOutput"], () => require("./getSearchAllResources"));
exports.OrganizationFeed = null;
utilities.lazyLoad(exports, ["OrganizationFeed"], () => require("./organizationFeed"));
exports.ProjectFeed = null;
utilities.lazyLoad(exports, ["ProjectFeed"], () => require("./projectFeed"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudasset/folderFeed:FolderFeed":
                return new exports.FolderFeed(name, undefined, { urn });
            case "gcp:cloudasset/organizationFeed:OrganizationFeed":
                return new exports.OrganizationFeed(name, undefined, { urn });
            case "gcp:cloudasset/projectFeed:ProjectFeed":
                return new exports.ProjectFeed(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudasset/folderFeed", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudasset/organizationFeed", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudasset/projectFeed", _module);
