// @pulumi/gcp/accesscontextmanager/ingressPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IngressPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * This resource has been deprecated, please refer to ServicePerimeterIngressPolicy.
 *
 * To get more information about IngressPolicy, see:
 *
 * * [API documentation](https://cloud.google.com/access-context-manager/docs/reference/rest/v1/accessPolicies.servicePerimeters#ingresspolicy)
 *
 * ## Import
 *
 * IngressPolicy can be imported using any of these accepted formats:
 *
 * * `{{ingress_policy_name}}/{{resource}}`
 *
 * When using the `pulumi import` command, IngressPolicy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:accesscontextmanager/ingressPolicy:IngressPolicy default {{ingress_policy_name}}/{{resource}}
 * ```
 */
class IngressPolicy extends pulumi.CustomResource {
    /**
     * Get an existing IngressPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IngressPolicy(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:accesscontextmanager/ingressPolicy:IngressPolicy';
    /**
     * Returns true if the given object is an instance of IngressPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IngressPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessPolicyId"] = state?.accessPolicyId;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["ingressPolicyName"] = state?.ingressPolicyName;
            resourceInputs["resource"] = state?.resource;
        }
        else {
            const args = argsOrState;
            if (args?.ingressPolicyName === undefined && !opts.urn) {
                throw new Error("Missing required property 'ingressPolicyName'");
            }
            if (args?.resource === undefined && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["ingressPolicyName"] = args?.ingressPolicyName;
            resourceInputs["resource"] = args?.resource;
            resourceInputs["accessPolicyId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IngressPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IngressPolicy = IngressPolicy;
