// @pulumi/gcp/firestore/backupSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupSchedule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A backup schedule for a Cloud Firestore Database.
 * This resource is owned by the database it is backing up, and is deleted along with the database.
 * The actual backups are not though.
 *
 * To get more information about BackupSchedule, see:
 *
 * * [API documentation](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/firestore/docs/backups)
 *
 * > **Warning:** This resource creates a Firestore Backup Schedule on a project that already has
 * a Firestore database.
 * This resource is owned by the database it is backing up, and is deleted along
 * with the database. The actual backups are not though.
 *
 * ## Example Usage
 *
 * ### Firestore Backup Schedule Daily
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const database = new gcp.firestore.Database("database", {
 *     project: "my-project-name",
 *     name: "database-id",
 *     locationId: "nam5",
 *     type: "FIRESTORE_NATIVE",
 *     deleteProtectionState: "DELETE_PROTECTION_ENABLED",
 *     deletionPolicy: "DELETE",
 * });
 * const daily_backup = new gcp.firestore.BackupSchedule("daily-backup", {
 *     project: "my-project-name",
 *     database: database.name,
 *     retention: "8467200s",
 *     dailyRecurrence: {},
 * });
 * ```
 * ### Firestore Backup Schedule Weekly
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const database = new gcp.firestore.Database("database", {
 *     project: "my-project-name",
 *     name: "database-id",
 *     locationId: "nam5",
 *     type: "FIRESTORE_NATIVE",
 *     deleteProtectionState: "DELETE_PROTECTION_ENABLED",
 *     deletionPolicy: "DELETE",
 * });
 * const weekly_backup = new gcp.firestore.BackupSchedule("weekly-backup", {
 *     project: "my-project-name",
 *     database: database.name,
 *     retention: "8467200s",
 *     weeklyRecurrence: {
 *         day: "SUNDAY",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * BackupSchedule can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}`
 * * `{{project}}/{{database}}/{{name}}`
 * * `{{database}}/{{name}}`
 *
 * When using the `pulumi import` command, BackupSchedule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:firestore/backupSchedule:BackupSchedule default projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}
 * $ pulumi import gcp:firestore/backupSchedule:BackupSchedule default {{project}}/{{database}}/{{name}}
 * $ pulumi import gcp:firestore/backupSchedule:BackupSchedule default {{database}}/{{name}}
 * ```
 */
class BackupSchedule extends pulumi.CustomResource {
    /**
     * Get an existing BackupSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BackupSchedule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:firestore/backupSchedule:BackupSchedule';
    /**
     * Returns true if the given object is an instance of BackupSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupSchedule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["dailyRecurrence"] = state?.dailyRecurrence;
            resourceInputs["database"] = state?.database;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["retention"] = state?.retention;
            resourceInputs["weeklyRecurrence"] = state?.weeklyRecurrence;
        }
        else {
            const args = argsOrState;
            if (args?.retention === undefined && !opts.urn) {
                throw new Error("Missing required property 'retention'");
            }
            resourceInputs["dailyRecurrence"] = args?.dailyRecurrence;
            resourceInputs["database"] = args?.database;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["project"] = args?.project;
            resourceInputs["retention"] = args?.retention;
            resourceInputs["weeklyRecurrence"] = args?.weeklyRecurrence;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BackupSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BackupSchedule = BackupSchedule;
