// @pulumi/azure-native/monitor/managementGroupDiagnosticSetting.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementGroupDiagnosticSetting = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The management group diagnostic setting resource.
 *
 * Uses Azure REST API version 2021-05-01-preview.
 */
class ManagementGroupDiagnosticSetting extends pulumi.CustomResource {
    /**
     * Get an existing ManagementGroupDiagnosticSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagementGroupDiagnosticSetting(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:monitor:ManagementGroupDiagnosticSetting';
    /**
     * Returns true if the given object is an instance of ManagementGroupDiagnosticSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementGroupDiagnosticSetting.__pulumiType;
    }
    /**
     * Create a ManagementGroupDiagnosticSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.managementGroupId === undefined && !opts.urn) {
                throw new Error("Missing required property 'managementGroupId'");
            }
            resourceInputs["eventHubAuthorizationRuleId"] = args?.eventHubAuthorizationRuleId;
            resourceInputs["eventHubName"] = args?.eventHubName;
            resourceInputs["logs"] = args?.logs;
            resourceInputs["managementGroupId"] = args?.managementGroupId;
            resourceInputs["marketplacePartnerId"] = args?.marketplacePartnerId;
            resourceInputs["name"] = args?.name;
            resourceInputs["serviceBusRuleId"] = args?.serviceBusRuleId;
            resourceInputs["storageAccountId"] = args?.storageAccountId;
            resourceInputs["workspaceId"] = args?.workspaceId;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["eventHubAuthorizationRuleId"] = undefined /*out*/;
            resourceInputs["eventHubName"] = undefined /*out*/;
            resourceInputs["logs"] = undefined /*out*/;
            resourceInputs["marketplacePartnerId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["serviceBusRuleId"] = undefined /*out*/;
            resourceInputs["storageAccountId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workspaceId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:insights/v20200101preview:ManagementGroupDiagnosticSetting" }, { type: "azure-native:insights/v20210501preview:ManagementGroupDiagnosticSetting" }, { type: "azure-native:insights:ManagementGroupDiagnosticSetting" }, { type: "azure-native:monitor/v20200101preview:ManagementGroupDiagnosticSetting" }, { type: "azure-native:monitor/v20210501preview:ManagementGroupDiagnosticSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagementGroupDiagnosticSetting.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagementGroupDiagnosticSetting = ManagementGroupDiagnosticSetting;
