// @pulumi/azure-native/monitor/tenantActionGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantActionGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A tenant action group resource.
 *
 * Uses Azure REST API version 2023-05-01-preview.
 */
class TenantActionGroup extends pulumi.CustomResource {
    /**
     * Get an existing TenantActionGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TenantActionGroup(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:monitor:TenantActionGroup';
    /**
     * Returns true if the given object is an instance of TenantActionGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TenantActionGroup.__pulumiType;
    }
    /**
     * Create a TenantActionGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.groupShortName === undefined && !opts.urn) {
                throw new Error("Missing required property 'groupShortName'");
            }
            if (args?.managementGroupId === undefined && !opts.urn) {
                throw new Error("Missing required property 'managementGroupId'");
            }
            resourceInputs["azureAppPushReceivers"] = args?.azureAppPushReceivers;
            resourceInputs["emailReceivers"] = args?.emailReceivers;
            resourceInputs["enabled"] = (args?.enabled) ?? true;
            resourceInputs["groupShortName"] = args?.groupShortName;
            resourceInputs["location"] = args?.location;
            resourceInputs["managementGroupId"] = args?.managementGroupId;
            resourceInputs["smsReceivers"] = args?.smsReceivers;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tenantActionGroupName"] = args?.tenantActionGroupName;
            resourceInputs["voiceReceivers"] = args?.voiceReceivers;
            resourceInputs["webhookReceivers"] = args?.webhookReceivers;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureAppPushReceivers"] = undefined /*out*/;
            resourceInputs["emailReceivers"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["groupShortName"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["smsReceivers"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["voiceReceivers"] = undefined /*out*/;
            resourceInputs["webhookReceivers"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:insights/v20230501preview:TenantActionGroup" }, { type: "azure-native:insights:TenantActionGroup" }, { type: "azure-native:monitor/v20230301preview:TenantActionGroup" }, { type: "azure-native:monitor/v20230501preview:TenantActionGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TenantActionGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TenantActionGroup = TenantActionGroup;
