// @pulumi/aws/networkmanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcAttachment = exports.TransitGatewayRouteTableAttachment = exports.TransitGatewayRegistration = exports.TransitGatewayPeering = exports.TransitGatewayConnectPeerAssociation = exports.SiteToSiteVpnAttachment = exports.Site = exports.PrefixListAssociation = exports.LinkAssociation = exports.Link = exports.GlobalNetwork = exports.getSitesOutput = exports.getSites = exports.getSiteOutput = exports.getSite = exports.getLinksOutput = exports.getLinks = exports.getLinkOutput = exports.getLink = exports.getGlobalNetworksOutput = exports.getGlobalNetworks = exports.getGlobalNetworkOutput = exports.getGlobalNetwork = exports.getDevicesOutput = exports.getDevices = exports.getDeviceOutput = exports.getDevice = exports.getCoreNetworkPolicyDocumentOutput = exports.getCoreNetworkPolicyDocument = exports.getCoreNetworkOutput = exports.getCoreNetwork = exports.getConnectionsOutput = exports.getConnections = exports.getConnectionOutput = exports.getConnection = exports.DxGatewayAttachment = exports.Device = exports.CustomerGatewayAssociation = exports.CoreNetworkPolicyAttachment = exports.CoreNetwork = exports.Connection = exports.ConnectPeer = exports.ConnectAttachment = exports.AttachmentRoutingPolicyLabel = exports.AttachmentAccepter = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AttachmentAccepter = null;
utilities.lazyLoad(exports, ["AttachmentAccepter"], () => require("./attachmentAccepter"));
exports.AttachmentRoutingPolicyLabel = null;
utilities.lazyLoad(exports, ["AttachmentRoutingPolicyLabel"], () => require("./attachmentRoutingPolicyLabel"));
exports.ConnectAttachment = null;
utilities.lazyLoad(exports, ["ConnectAttachment"], () => require("./connectAttachment"));
exports.ConnectPeer = null;
utilities.lazyLoad(exports, ["ConnectPeer"], () => require("./connectPeer"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.CoreNetwork = null;
utilities.lazyLoad(exports, ["CoreNetwork"], () => require("./coreNetwork"));
exports.CoreNetworkPolicyAttachment = null;
utilities.lazyLoad(exports, ["CoreNetworkPolicyAttachment"], () => require("./coreNetworkPolicyAttachment"));
exports.CustomerGatewayAssociation = null;
utilities.lazyLoad(exports, ["CustomerGatewayAssociation"], () => require("./customerGatewayAssociation"));
exports.Device = null;
utilities.lazyLoad(exports, ["Device"], () => require("./device"));
exports.DxGatewayAttachment = null;
utilities.lazyLoad(exports, ["DxGatewayAttachment"], () => require("./dxGatewayAttachment"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.getConnections = null;
exports.getConnectionsOutput = null;
utilities.lazyLoad(exports, ["getConnections", "getConnectionsOutput"], () => require("./getConnections"));
exports.getCoreNetwork = null;
exports.getCoreNetworkOutput = null;
utilities.lazyLoad(exports, ["getCoreNetwork", "getCoreNetworkOutput"], () => require("./getCoreNetwork"));
exports.getCoreNetworkPolicyDocument = null;
exports.getCoreNetworkPolicyDocumentOutput = null;
utilities.lazyLoad(exports, ["getCoreNetworkPolicyDocument", "getCoreNetworkPolicyDocumentOutput"], () => require("./getCoreNetworkPolicyDocument"));
exports.getDevice = null;
exports.getDeviceOutput = null;
utilities.lazyLoad(exports, ["getDevice", "getDeviceOutput"], () => require("./getDevice"));
exports.getDevices = null;
exports.getDevicesOutput = null;
utilities.lazyLoad(exports, ["getDevices", "getDevicesOutput"], () => require("./getDevices"));
exports.getGlobalNetwork = null;
exports.getGlobalNetworkOutput = null;
utilities.lazyLoad(exports, ["getGlobalNetwork", "getGlobalNetworkOutput"], () => require("./getGlobalNetwork"));
exports.getGlobalNetworks = null;
exports.getGlobalNetworksOutput = null;
utilities.lazyLoad(exports, ["getGlobalNetworks", "getGlobalNetworksOutput"], () => require("./getGlobalNetworks"));
exports.getLink = null;
exports.getLinkOutput = null;
utilities.lazyLoad(exports, ["getLink", "getLinkOutput"], () => require("./getLink"));
exports.getLinks = null;
exports.getLinksOutput = null;
utilities.lazyLoad(exports, ["getLinks", "getLinksOutput"], () => require("./getLinks"));
exports.getSite = null;
exports.getSiteOutput = null;
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
exports.getSites = null;
exports.getSitesOutput = null;
utilities.lazyLoad(exports, ["getSites", "getSitesOutput"], () => require("./getSites"));
exports.GlobalNetwork = null;
utilities.lazyLoad(exports, ["GlobalNetwork"], () => require("./globalNetwork"));
exports.Link = null;
utilities.lazyLoad(exports, ["Link"], () => require("./link"));
exports.LinkAssociation = null;
utilities.lazyLoad(exports, ["LinkAssociation"], () => require("./linkAssociation"));
exports.PrefixListAssociation = null;
utilities.lazyLoad(exports, ["PrefixListAssociation"], () => require("./prefixListAssociation"));
exports.Site = null;
utilities.lazyLoad(exports, ["Site"], () => require("./site"));
exports.SiteToSiteVpnAttachment = null;
utilities.lazyLoad(exports, ["SiteToSiteVpnAttachment"], () => require("./siteToSiteVpnAttachment"));
exports.TransitGatewayConnectPeerAssociation = null;
utilities.lazyLoad(exports, ["TransitGatewayConnectPeerAssociation"], () => require("./transitGatewayConnectPeerAssociation"));
exports.TransitGatewayPeering = null;
utilities.lazyLoad(exports, ["TransitGatewayPeering"], () => require("./transitGatewayPeering"));
exports.TransitGatewayRegistration = null;
utilities.lazyLoad(exports, ["TransitGatewayRegistration"], () => require("./transitGatewayRegistration"));
exports.TransitGatewayRouteTableAttachment = null;
utilities.lazyLoad(exports, ["TransitGatewayRouteTableAttachment"], () => require("./transitGatewayRouteTableAttachment"));
exports.VpcAttachment = null;
utilities.lazyLoad(exports, ["VpcAttachment"], () => require("./vpcAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:networkmanager/attachmentAccepter:AttachmentAccepter":
                return new exports.AttachmentAccepter(name, undefined, { urn });
            case "aws:networkmanager/attachmentRoutingPolicyLabel:AttachmentRoutingPolicyLabel":
                return new exports.AttachmentRoutingPolicyLabel(name, undefined, { urn });
            case "aws:networkmanager/connectAttachment:ConnectAttachment":
                return new exports.ConnectAttachment(name, undefined, { urn });
            case "aws:networkmanager/connectPeer:ConnectPeer":
                return new exports.ConnectPeer(name, undefined, { urn });
            case "aws:networkmanager/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "aws:networkmanager/coreNetwork:CoreNetwork":
                return new exports.CoreNetwork(name, undefined, { urn });
            case "aws:networkmanager/coreNetworkPolicyAttachment:CoreNetworkPolicyAttachment":
                return new exports.CoreNetworkPolicyAttachment(name, undefined, { urn });
            case "aws:networkmanager/customerGatewayAssociation:CustomerGatewayAssociation":
                return new exports.CustomerGatewayAssociation(name, undefined, { urn });
            case "aws:networkmanager/device:Device":
                return new exports.Device(name, undefined, { urn });
            case "aws:networkmanager/dxGatewayAttachment:DxGatewayAttachment":
                return new exports.DxGatewayAttachment(name, undefined, { urn });
            case "aws:networkmanager/globalNetwork:GlobalNetwork":
                return new exports.GlobalNetwork(name, undefined, { urn });
            case "aws:networkmanager/link:Link":
                return new exports.Link(name, undefined, { urn });
            case "aws:networkmanager/linkAssociation:LinkAssociation":
                return new exports.LinkAssociation(name, undefined, { urn });
            case "aws:networkmanager/prefixListAssociation:PrefixListAssociation":
                return new exports.PrefixListAssociation(name, undefined, { urn });
            case "aws:networkmanager/site:Site":
                return new exports.Site(name, undefined, { urn });
            case "aws:networkmanager/siteToSiteVpnAttachment:SiteToSiteVpnAttachment":
                return new exports.SiteToSiteVpnAttachment(name, undefined, { urn });
            case "aws:networkmanager/transitGatewayConnectPeerAssociation:TransitGatewayConnectPeerAssociation":
                return new exports.TransitGatewayConnectPeerAssociation(name, undefined, { urn });
            case "aws:networkmanager/transitGatewayPeering:TransitGatewayPeering":
                return new exports.TransitGatewayPeering(name, undefined, { urn });
            case "aws:networkmanager/transitGatewayRegistration:TransitGatewayRegistration":
                return new exports.TransitGatewayRegistration(name, undefined, { urn });
            case "aws:networkmanager/transitGatewayRouteTableAttachment:TransitGatewayRouteTableAttachment":
                return new exports.TransitGatewayRouteTableAttachment(name, undefined, { urn });
            case "aws:networkmanager/vpcAttachment:VpcAttachment":
                return new exports.VpcAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "networkmanager/attachmentAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/attachmentRoutingPolicyLabel", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/connectAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/connectPeer", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/connection", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/coreNetwork", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/coreNetworkPolicyAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/customerGatewayAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/device", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/dxGatewayAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/globalNetwork", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/link", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/linkAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/prefixListAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/site", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/siteToSiteVpnAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/transitGatewayConnectPeerAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/transitGatewayPeering", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/transitGatewayRegistration", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/transitGatewayRouteTableAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "networkmanager/vpcAttachment", _module);
