// @pulumi/aws/dynamodb/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = exports.TableReplica = exports.TableItem = exports.TableExport = exports.Table = exports.ResourcePolicy = exports.KinesisStreamingDestination = exports.GlobalTable = exports.GlobalSecondaryIndex = exports.getTablesOutput = exports.getTables = exports.getTableItemOutput = exports.getTableItem = exports.getTableOutput = exports.getTable = exports.getBackupsOutput = exports.getBackups = exports.ContributorInsights = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ContributorInsights = null;
utilities.lazyLoad(exports, ["ContributorInsights"], () => require("./contributorInsights"));
__exportStar(require("./dynamodbMixins"), exports);
exports.getBackups = null;
exports.getBackupsOutput = null;
utilities.lazyLoad(exports, ["getBackups", "getBackupsOutput"], () => require("./getBackups"));
exports.getTable = null;
exports.getTableOutput = null;
utilities.lazyLoad(exports, ["getTable", "getTableOutput"], () => require("./getTable"));
exports.getTableItem = null;
exports.getTableItemOutput = null;
utilities.lazyLoad(exports, ["getTableItem", "getTableItemOutput"], () => require("./getTableItem"));
exports.getTables = null;
exports.getTablesOutput = null;
utilities.lazyLoad(exports, ["getTables", "getTablesOutput"], () => require("./getTables"));
exports.GlobalSecondaryIndex = null;
utilities.lazyLoad(exports, ["GlobalSecondaryIndex"], () => require("./globalSecondaryIndex"));
exports.GlobalTable = null;
utilities.lazyLoad(exports, ["GlobalTable"], () => require("./globalTable"));
exports.KinesisStreamingDestination = null;
utilities.lazyLoad(exports, ["KinesisStreamingDestination"], () => require("./kinesisStreamingDestination"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.TableExport = null;
utilities.lazyLoad(exports, ["TableExport"], () => require("./tableExport"));
exports.TableItem = null;
utilities.lazyLoad(exports, ["TableItem"], () => require("./tableItem"));
exports.TableReplica = null;
utilities.lazyLoad(exports, ["TableReplica"], () => require("./tableReplica"));
exports.Tag = null;
utilities.lazyLoad(exports, ["Tag"], () => require("./tag"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:dynamodb/contributorInsights:ContributorInsights":
                return new exports.ContributorInsights(name, undefined, { urn });
            case "aws:dynamodb/globalSecondaryIndex:GlobalSecondaryIndex":
                return new exports.GlobalSecondaryIndex(name, undefined, { urn });
            case "aws:dynamodb/globalTable:GlobalTable":
                return new exports.GlobalTable(name, undefined, { urn });
            case "aws:dynamodb/kinesisStreamingDestination:KinesisStreamingDestination":
                return new exports.KinesisStreamingDestination(name, undefined, { urn });
            case "aws:dynamodb/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:dynamodb/table:Table":
                return new exports.Table(name, undefined, { urn });
            case "aws:dynamodb/tableExport:TableExport":
                return new exports.TableExport(name, undefined, { urn });
            case "aws:dynamodb/tableItem:TableItem":
                return new exports.TableItem(name, undefined, { urn });
            case "aws:dynamodb/tableReplica:TableReplica":
                return new exports.TableReplica(name, undefined, { urn });
            case "aws:dynamodb/tag:Tag":
                return new exports.Tag(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "dynamodb/contributorInsights", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/globalSecondaryIndex", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/globalTable", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/kinesisStreamingDestination", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/table", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tableExport", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tableItem", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tableReplica", _module);
pulumi.runtime.registerResourceModule("aws", "dynamodb/tag", _module);
