// @pulumi/aws/dynamodb/tableReplica.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableReplica = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a DynamoDB table replica resource for [DynamoDB Global Tables V2 (version 2019.11.21)](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.V2.html).
 *
 * > **Note:** Use `lifecycle` `ignoreChanges` for `replica` in the associated aws.dynamodb.Table configuration.
 *
 * > **Note:** Do not use the `replica` configuration block of aws.dynamodb.Table together with this resource as the two configuration options are mutually exclusive.
 *
 * ## Example Usage
 *
 * ### Basic Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.dynamodb.Table("example", {
 *     name: "TestTable",
 *     hashKey: "BrodoBaggins",
 *     billingMode: "PAY_PER_REQUEST",
 *     streamEnabled: true,
 *     streamViewType: "NEW_AND_OLD_IMAGES",
 *     attributes: [{
 *         name: "BrodoBaggins",
 *         type: "S",
 *     }],
 * });
 * const exampleTableReplica = new aws.dynamodb.TableReplica("example", {
 *     globalTableArn: example.arn,
 *     tags: {
 *         Name: "IZPAWS",
 *         Pozo: "Amargo",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import DynamoDB table replicas using the `table-name:main-region`. For example:
 *
 * > **Note:** When importing, use the region where the initial or _main_ global table resides, _not_ the region of the replica.
 *
 * ```sh
 * $ pulumi import aws:dynamodb/tableReplica:TableReplica example TestTable:us-west-2
 * ```
 */
class TableReplica extends pulumi.CustomResource {
    /**
     * Get an existing TableReplica resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TableReplica(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:dynamodb/tableReplica:TableReplica';
    /**
     * Returns true if the given object is an instance of TableReplica.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TableReplica.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["deletionProtectionEnabled"] = state?.deletionProtectionEnabled;
            resourceInputs["globalTableArn"] = state?.globalTableArn;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["pointInTimeRecovery"] = state?.pointInTimeRecovery;
            resourceInputs["region"] = state?.region;
            resourceInputs["tableClassOverride"] = state?.tableClassOverride;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.globalTableArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'globalTableArn'");
            }
            resourceInputs["deletionProtectionEnabled"] = args?.deletionProtectionEnabled;
            resourceInputs["globalTableArn"] = args?.globalTableArn;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["pointInTimeRecovery"] = args?.pointInTimeRecovery;
            resourceInputs["region"] = args?.region;
            resourceInputs["tableClassOverride"] = args?.tableClassOverride;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TableReplica.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TableReplica = TableReplica;
