// @pulumi/azure-native/community/communityTraining.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunityTraining = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * A CommunityProviderHub resource
 *
 * Uses Azure REST API version 2023-11-01. In version 2.x of the Azure Native provider, it used API version 2023-11-01.
 */
class CommunityTraining extends pulumi.CustomResource {
    /**
     * Get an existing CommunityTraining resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CommunityTraining(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:community:CommunityTraining';
    /**
     * Returns true if the given object is an instance of CommunityTraining.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CommunityTraining.__pulumiType;
    }
    /**
     * Create a CommunityTraining resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.disasterRecoveryEnabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'disasterRecoveryEnabled'");
            }
            if (args?.identityConfiguration === undefined && !opts.urn) {
                throw new Error("Missing required property 'identityConfiguration'");
            }
            if (args?.portalAdminEmailAddress === undefined && !opts.urn) {
                throw new Error("Missing required property 'portalAdminEmailAddress'");
            }
            if (args?.portalName === undefined && !opts.urn) {
                throw new Error("Missing required property 'portalName'");
            }
            if (args?.portalOwnerEmailAddress === undefined && !opts.urn) {
                throw new Error("Missing required property 'portalOwnerEmailAddress'");
            }
            if (args?.portalOwnerOrganizationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'portalOwnerOrganizationName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.zoneRedundancyEnabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneRedundancyEnabled'");
            }
            resourceInputs["communityTrainingName"] = args?.communityTrainingName;
            resourceInputs["disasterRecoveryEnabled"] = args?.disasterRecoveryEnabled;
            resourceInputs["identityConfiguration"] = args ? (args.identityConfiguration ? pulumi.output(args.identityConfiguration).apply(inputs.community.identityConfigurationPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["location"] = args?.location;
            resourceInputs["portalAdminEmailAddress"] = args?.portalAdminEmailAddress;
            resourceInputs["portalName"] = args?.portalName;
            resourceInputs["portalOwnerEmailAddress"] = args?.portalOwnerEmailAddress;
            resourceInputs["portalOwnerOrganizationName"] = args?.portalOwnerOrganizationName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sku"] = args?.sku;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["zoneRedundancyEnabled"] = args?.zoneRedundancyEnabled;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["disasterRecoveryEnabled"] = undefined /*out*/;
            resourceInputs["identityConfiguration"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["portalAdminEmailAddress"] = undefined /*out*/;
            resourceInputs["portalName"] = undefined /*out*/;
            resourceInputs["portalOwnerEmailAddress"] = undefined /*out*/;
            resourceInputs["portalOwnerOrganizationName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["zoneRedundancyEnabled"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:community/v20231101:CommunityTraining" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CommunityTraining.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CommunityTraining = CommunityTraining;
