// @pulumi/azure-native/cdn/afdoriginGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AFDOriginGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * AFDOrigin group comprising of origins is used for load balancing to origins when the content cannot be served from Azure Front Door.
 *
 * Uses Azure REST API version 2025-06-01. In version 2.x of the Azure Native provider, it used API version 2023-05-01.
 *
 * Other available API versions: 2023-05-01, 2023-07-01-preview, 2024-02-01, 2024-05-01-preview, 2024-06-01-preview, 2024-09-01, 2025-01-01-preview, 2025-04-15, 2025-07-01-preview, 2025-09-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cdn [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AFDOriginGroup extends pulumi.CustomResource {
    /**
     * Get an existing AFDOriginGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AFDOriginGroup(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cdn:AFDOriginGroup';
    /**
     * Returns true if the given object is an instance of AFDOriginGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AFDOriginGroup.__pulumiType;
    }
    /**
     * Create a AFDOriginGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.profileName === undefined && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authentication"] = args?.authentication;
            resourceInputs["healthProbeSettings"] = args?.healthProbeSettings;
            resourceInputs["loadBalancingSettings"] = args?.loadBalancingSettings;
            resourceInputs["originGroupName"] = args?.originGroupName;
            resourceInputs["profileName"] = args?.profileName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sessionAffinityState"] = args?.sessionAffinityState;
            resourceInputs["trafficRestorationTimeToHealedOrNewEndpointsInMinutes"] = args?.trafficRestorationTimeToHealedOrNewEndpointsInMinutes;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["authentication"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["healthProbeSettings"] = undefined /*out*/;
            resourceInputs["loadBalancingSettings"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["profileName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sessionAffinityState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["trafficRestorationTimeToHealedOrNewEndpointsInMinutes"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cdn/v20200901:AFDOriginGroup" }, { type: "azure-native:cdn/v20210601:AFDOriginGroup" }, { type: "azure-native:cdn/v20220501preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20221101preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20230501:AFDOriginGroup" }, { type: "azure-native:cdn/v20230701preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20240201:AFDOriginGroup" }, { type: "azure-native:cdn/v20240501preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20240601preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20240901:AFDOriginGroup" }, { type: "azure-native:cdn/v20250101preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20250415:AFDOriginGroup" }, { type: "azure-native:cdn/v20250601:AFDOriginGroup" }, { type: "azure-native:cdn/v20250701preview:AFDOriginGroup" }, { type: "azure-native:cdn/v20250901preview:AFDOriginGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AFDOriginGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AFDOriginGroup = AFDOriginGroup;
