// @pulumi/azure-native/cdn/edgeActionExecutionFilter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EdgeActionExecutionFilter = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Concrete tracked resource types can be created by aliasing this type using a specific property type.
 *
 * Uses Azure REST API version 2025-09-01-preview.
 *
 * Other available API versions: 2024-07-22-preview, 2025-12-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cdn [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class EdgeActionExecutionFilter extends pulumi.CustomResource {
    /**
     * Get an existing EdgeActionExecutionFilter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EdgeActionExecutionFilter(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cdn:EdgeActionExecutionFilter';
    /**
     * Returns true if the given object is an instance of EdgeActionExecutionFilter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EdgeActionExecutionFilter.__pulumiType;
    }
    /**
     * Create a EdgeActionExecutionFilter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.edgeActionName === undefined && !opts.urn) {
                throw new Error("Missing required property 'edgeActionName'");
            }
            if (args?.executionFilterIdentifierHeaderName === undefined && !opts.urn) {
                throw new Error("Missing required property 'executionFilterIdentifierHeaderName'");
            }
            if (args?.executionFilterIdentifierHeaderValue === undefined && !opts.urn) {
                throw new Error("Missing required property 'executionFilterIdentifierHeaderValue'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.versionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'versionId'");
            }
            resourceInputs["edgeActionName"] = args?.edgeActionName;
            resourceInputs["executionFilter"] = args?.executionFilter;
            resourceInputs["executionFilterIdentifierHeaderName"] = args?.executionFilterIdentifierHeaderName;
            resourceInputs["executionFilterIdentifierHeaderValue"] = args?.executionFilterIdentifierHeaderValue;
            resourceInputs["location"] = args?.location;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["versionId"] = args?.versionId;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["lastUpdateTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["executionFilterIdentifierHeaderName"] = undefined /*out*/;
            resourceInputs["executionFilterIdentifierHeaderValue"] = undefined /*out*/;
            resourceInputs["lastUpdateTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["versionId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cdn/v20240722preview:EdgeActionExecutionFilter" }, { type: "azure-native:cdn/v20250901preview:EdgeActionExecutionFilter" }, { type: "azure-native:cdn/v20251201preview:EdgeActionExecutionFilter" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EdgeActionExecutionFilter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EdgeActionExecutionFilter = EdgeActionExecutionFilter;
