// @pulumi/azure-native/cdn/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRouteOutput = exports.getRoute = exports.getProfileSupportedOptimizationTypesOutput = exports.getProfileSupportedOptimizationTypes = exports.getProfileAgentOutput = exports.getProfileAgent = exports.getProfileOutput = exports.getProfile = exports.getPolicyOutput = exports.getPolicy = exports.getOriginGroupOutput = exports.getOriginGroup = exports.getOriginOutput = exports.getOrigin = exports.getKnowledgeSourceOutput = exports.getKnowledgeSource = exports.getKeyGroupOutput = exports.getKeyGroup = exports.getEndpointOutput = exports.getEndpoint = exports.getEdgeActionVersionCodeOutput = exports.getEdgeActionVersionCode = exports.getEdgeActionVersionOutput = exports.getEdgeActionVersion = exports.getEdgeActionExecutionFilterOutput = exports.getEdgeActionExecutionFilter = exports.getEdgeActionOutput = exports.getEdgeAction = exports.getCustomDomainOutput = exports.getCustomDomain = exports.getAFDTargetGroupOutput = exports.getAFDTargetGroup = exports.getAFDOriginGroupOutput = exports.getAFDOriginGroup = exports.getAFDOriginOutput = exports.getAFDOrigin = exports.getAFDEndpointOutput = exports.getAFDEndpoint = exports.getAFDCustomDomainOutput = exports.getAFDCustomDomain = exports.Endpoint = exports.EdgeActionVersion = exports.EdgeActionExecutionFilter = exports.EdgeAction = exports.CustomDomain = exports.AFDTargetGroup = exports.AFDOriginGroup = exports.AFDOrigin = exports.AFDEndpoint = exports.AFDCustomDomain = void 0;
exports.WebAgent = exports.TunnelPolicy = exports.SecurityPolicy = exports.Secret = exports.RuleSet = exports.Rule = exports.Route = exports.ProfileAgent = exports.Profile = exports.Policy = exports.OriginGroup = exports.Origin = exports.KnowledgeSource = exports.KeyGroup = exports.getWebAgentOutput = exports.getWebAgent = exports.getTunnelPolicyOutput = exports.getTunnelPolicy = exports.getSecurityPolicyOutput = exports.getSecurityPolicy = exports.getSecretOutput = exports.getSecret = exports.getRuleSetOutput = exports.getRuleSet = exports.getRuleOutput = exports.getRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AFDCustomDomain = null;
utilities.lazyLoad(exports, ["AFDCustomDomain"], () => require("./afdcustomDomain"));
exports.AFDEndpoint = null;
utilities.lazyLoad(exports, ["AFDEndpoint"], () => require("./afdendpoint"));
exports.AFDOrigin = null;
utilities.lazyLoad(exports, ["AFDOrigin"], () => require("./afdorigin"));
exports.AFDOriginGroup = null;
utilities.lazyLoad(exports, ["AFDOriginGroup"], () => require("./afdoriginGroup"));
exports.AFDTargetGroup = null;
utilities.lazyLoad(exports, ["AFDTargetGroup"], () => require("./afdtargetGroup"));
exports.CustomDomain = null;
utilities.lazyLoad(exports, ["CustomDomain"], () => require("./customDomain"));
exports.EdgeAction = null;
utilities.lazyLoad(exports, ["EdgeAction"], () => require("./edgeAction"));
exports.EdgeActionExecutionFilter = null;
utilities.lazyLoad(exports, ["EdgeActionExecutionFilter"], () => require("./edgeActionExecutionFilter"));
exports.EdgeActionVersion = null;
utilities.lazyLoad(exports, ["EdgeActionVersion"], () => require("./edgeActionVersion"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getAFDCustomDomain = null;
exports.getAFDCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getAFDCustomDomain", "getAFDCustomDomainOutput"], () => require("./getAFDCustomDomain"));
exports.getAFDEndpoint = null;
exports.getAFDEndpointOutput = null;
utilities.lazyLoad(exports, ["getAFDEndpoint", "getAFDEndpointOutput"], () => require("./getAFDEndpoint"));
exports.getAFDOrigin = null;
exports.getAFDOriginOutput = null;
utilities.lazyLoad(exports, ["getAFDOrigin", "getAFDOriginOutput"], () => require("./getAFDOrigin"));
exports.getAFDOriginGroup = null;
exports.getAFDOriginGroupOutput = null;
utilities.lazyLoad(exports, ["getAFDOriginGroup", "getAFDOriginGroupOutput"], () => require("./getAFDOriginGroup"));
exports.getAFDTargetGroup = null;
exports.getAFDTargetGroupOutput = null;
utilities.lazyLoad(exports, ["getAFDTargetGroup", "getAFDTargetGroupOutput"], () => require("./getAFDTargetGroup"));
exports.getCustomDomain = null;
exports.getCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getCustomDomain", "getCustomDomainOutput"], () => require("./getCustomDomain"));
exports.getEdgeAction = null;
exports.getEdgeActionOutput = null;
utilities.lazyLoad(exports, ["getEdgeAction", "getEdgeActionOutput"], () => require("./getEdgeAction"));
exports.getEdgeActionExecutionFilter = null;
exports.getEdgeActionExecutionFilterOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionExecutionFilter", "getEdgeActionExecutionFilterOutput"], () => require("./getEdgeActionExecutionFilter"));
exports.getEdgeActionVersion = null;
exports.getEdgeActionVersionOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionVersion", "getEdgeActionVersionOutput"], () => require("./getEdgeActionVersion"));
exports.getEdgeActionVersionCode = null;
exports.getEdgeActionVersionCodeOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionVersionCode", "getEdgeActionVersionCodeOutput"], () => require("./getEdgeActionVersionCode"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getKeyGroup = null;
exports.getKeyGroupOutput = null;
utilities.lazyLoad(exports, ["getKeyGroup", "getKeyGroupOutput"], () => require("./getKeyGroup"));
exports.getKnowledgeSource = null;
exports.getKnowledgeSourceOutput = null;
utilities.lazyLoad(exports, ["getKnowledgeSource", "getKnowledgeSourceOutput"], () => require("./getKnowledgeSource"));
exports.getOrigin = null;
exports.getOriginOutput = null;
utilities.lazyLoad(exports, ["getOrigin", "getOriginOutput"], () => require("./getOrigin"));
exports.getOriginGroup = null;
exports.getOriginGroupOutput = null;
utilities.lazyLoad(exports, ["getOriginGroup", "getOriginGroupOutput"], () => require("./getOriginGroup"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getProfile = null;
exports.getProfileOutput = null;
utilities.lazyLoad(exports, ["getProfile", "getProfileOutput"], () => require("./getProfile"));
exports.getProfileAgent = null;
exports.getProfileAgentOutput = null;
utilities.lazyLoad(exports, ["getProfileAgent", "getProfileAgentOutput"], () => require("./getProfileAgent"));
exports.getProfileSupportedOptimizationTypes = null;
exports.getProfileSupportedOptimizationTypesOutput = null;
utilities.lazyLoad(exports, ["getProfileSupportedOptimizationTypes", "getProfileSupportedOptimizationTypesOutput"], () => require("./getProfileSupportedOptimizationTypes"));
exports.getRoute = null;
exports.getRouteOutput = null;
utilities.lazyLoad(exports, ["getRoute", "getRouteOutput"], () => require("./getRoute"));
exports.getRule = null;
exports.getRuleOutput = null;
utilities.lazyLoad(exports, ["getRule", "getRuleOutput"], () => require("./getRule"));
exports.getRuleSet = null;
exports.getRuleSetOutput = null;
utilities.lazyLoad(exports, ["getRuleSet", "getRuleSetOutput"], () => require("./getRuleSet"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSecurityPolicy = null;
exports.getSecurityPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecurityPolicy", "getSecurityPolicyOutput"], () => require("./getSecurityPolicy"));
exports.getTunnelPolicy = null;
exports.getTunnelPolicyOutput = null;
utilities.lazyLoad(exports, ["getTunnelPolicy", "getTunnelPolicyOutput"], () => require("./getTunnelPolicy"));
exports.getWebAgent = null;
exports.getWebAgentOutput = null;
utilities.lazyLoad(exports, ["getWebAgent", "getWebAgentOutput"], () => require("./getWebAgent"));
exports.KeyGroup = null;
utilities.lazyLoad(exports, ["KeyGroup"], () => require("./keyGroup"));
exports.KnowledgeSource = null;
utilities.lazyLoad(exports, ["KnowledgeSource"], () => require("./knowledgeSource"));
exports.Origin = null;
utilities.lazyLoad(exports, ["Origin"], () => require("./origin"));
exports.OriginGroup = null;
utilities.lazyLoad(exports, ["OriginGroup"], () => require("./originGroup"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.Profile = null;
utilities.lazyLoad(exports, ["Profile"], () => require("./profile"));
exports.ProfileAgent = null;
utilities.lazyLoad(exports, ["ProfileAgent"], () => require("./profileAgent"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
exports.RuleSet = null;
utilities.lazyLoad(exports, ["RuleSet"], () => require("./ruleSet"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SecurityPolicy = null;
utilities.lazyLoad(exports, ["SecurityPolicy"], () => require("./securityPolicy"));
exports.TunnelPolicy = null;
utilities.lazyLoad(exports, ["TunnelPolicy"], () => require("./tunnelPolicy"));
exports.WebAgent = null;
utilities.lazyLoad(exports, ["WebAgent"], () => require("./webAgent"));
// Export enums:
__exportStar(require("../types/enums/cdn"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cdn:AFDCustomDomain":
                return new exports.AFDCustomDomain(name, undefined, { urn });
            case "azure-native:cdn:AFDEndpoint":
                return new exports.AFDEndpoint(name, undefined, { urn });
            case "azure-native:cdn:AFDOrigin":
                return new exports.AFDOrigin(name, undefined, { urn });
            case "azure-native:cdn:AFDOriginGroup":
                return new exports.AFDOriginGroup(name, undefined, { urn });
            case "azure-native:cdn:AFDTargetGroup":
                return new exports.AFDTargetGroup(name, undefined, { urn });
            case "azure-native:cdn:CustomDomain":
                return new exports.CustomDomain(name, undefined, { urn });
            case "azure-native:cdn:EdgeAction":
                return new exports.EdgeAction(name, undefined, { urn });
            case "azure-native:cdn:EdgeActionExecutionFilter":
                return new exports.EdgeActionExecutionFilter(name, undefined, { urn });
            case "azure-native:cdn:EdgeActionVersion":
                return new exports.EdgeActionVersion(name, undefined, { urn });
            case "azure-native:cdn:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "azure-native:cdn:KeyGroup":
                return new exports.KeyGroup(name, undefined, { urn });
            case "azure-native:cdn:KnowledgeSource":
                return new exports.KnowledgeSource(name, undefined, { urn });
            case "azure-native:cdn:Origin":
                return new exports.Origin(name, undefined, { urn });
            case "azure-native:cdn:OriginGroup":
                return new exports.OriginGroup(name, undefined, { urn });
            case "azure-native:cdn:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:cdn:Profile":
                return new exports.Profile(name, undefined, { urn });
            case "azure-native:cdn:ProfileAgent":
                return new exports.ProfileAgent(name, undefined, { urn });
            case "azure-native:cdn:Route":
                return new exports.Route(name, undefined, { urn });
            case "azure-native:cdn:Rule":
                return new exports.Rule(name, undefined, { urn });
            case "azure-native:cdn:RuleSet":
                return new exports.RuleSet(name, undefined, { urn });
            case "azure-native:cdn:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "azure-native:cdn:SecurityPolicy":
                return new exports.SecurityPolicy(name, undefined, { urn });
            case "azure-native:cdn:TunnelPolicy":
                return new exports.TunnelPolicy(name, undefined, { urn });
            case "azure-native:cdn:WebAgent":
                return new exports.WebAgent(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cdn", _module);
