// @pulumi/azure-native/cdn/origin.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Origin = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * CDN origin is the source of the content being delivered via CDN. When the edge nodes represented by an endpoint do not have the requested content cached, they attempt to fetch it from one or more of the configured origins.
 *
 * Uses Azure REST API version 2025-06-01. In version 2.x of the Azure Native provider, it used API version 2023-05-01.
 *
 * Other available API versions: 2023-05-01, 2023-07-01-preview, 2024-02-01, 2024-05-01-preview, 2024-06-01-preview, 2024-09-01, 2025-01-01-preview, 2025-04-15, 2025-07-01-preview, 2025-09-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cdn [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Origin extends pulumi.CustomResource {
    /**
     * Get an existing Origin resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Origin(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cdn:Origin';
    /**
     * Returns true if the given object is an instance of Origin.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Origin.__pulumiType;
    }
    /**
     * Create a Origin resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.endpointName === undefined && !opts.urn) {
                throw new Error("Missing required property 'endpointName'");
            }
            if (args?.hostName === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostName'");
            }
            if (args?.profileName === undefined && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["endpointName"] = args?.endpointName;
            resourceInputs["hostName"] = args?.hostName;
            resourceInputs["httpPort"] = args?.httpPort;
            resourceInputs["httpsPort"] = args?.httpsPort;
            resourceInputs["originHostHeader"] = args?.originHostHeader;
            resourceInputs["originName"] = args?.originName;
            resourceInputs["priority"] = args?.priority;
            resourceInputs["privateLinkAlias"] = args?.privateLinkAlias;
            resourceInputs["privateLinkApprovalMessage"] = args?.privateLinkApprovalMessage;
            resourceInputs["privateLinkLocation"] = args?.privateLinkLocation;
            resourceInputs["privateLinkResourceId"] = args?.privateLinkResourceId;
            resourceInputs["profileName"] = args?.profileName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["weight"] = args?.weight;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointStatus"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["httpPort"] = undefined /*out*/;
            resourceInputs["httpsPort"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["originHostHeader"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["privateEndpointStatus"] = undefined /*out*/;
            resourceInputs["privateLinkAlias"] = undefined /*out*/;
            resourceInputs["privateLinkApprovalMessage"] = undefined /*out*/;
            resourceInputs["privateLinkLocation"] = undefined /*out*/;
            resourceInputs["privateLinkResourceId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["weight"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cdn/v20150601:Origin" }, { type: "azure-native:cdn/v20160402:Origin" }, { type: "azure-native:cdn/v20191231:Origin" }, { type: "azure-native:cdn/v20200331:Origin" }, { type: "azure-native:cdn/v20200415:Origin" }, { type: "azure-native:cdn/v20200901:Origin" }, { type: "azure-native:cdn/v20210601:Origin" }, { type: "azure-native:cdn/v20220501preview:Origin" }, { type: "azure-native:cdn/v20221101preview:Origin" }, { type: "azure-native:cdn/v20230501:Origin" }, { type: "azure-native:cdn/v20230701preview:Origin" }, { type: "azure-native:cdn/v20240201:Origin" }, { type: "azure-native:cdn/v20240501preview:Origin" }, { type: "azure-native:cdn/v20240601preview:Origin" }, { type: "azure-native:cdn/v20240901:Origin" }, { type: "azure-native:cdn/v20250101preview:Origin" }, { type: "azure-native:cdn/v20250415:Origin" }, { type: "azure-native:cdn/v20250601:Origin" }, { type: "azure-native:cdn/v20250701preview:Origin" }, { type: "azure-native:cdn/v20250901preview:Origin" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Origin.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Origin = Origin;
