// @pulumi/gcp/contactcenterinsights/assessmentRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssessmentRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The CCAI Insights project wide assessment rule. This assessment rule will be
 * applied to all conversations from the previous sampling cycle that match the
 * sample rule defined in the assessment rule. One project can have multiple
 * assessment rules.
 *
 * To get more information about AssessmentRule, see:
 *
 * * [API documentation](https://cloud.google.com/contact-center/insights/docs/reference/rest/v1/projects.locations.assessmentRules)
 * * How-to Guides
 *     * [Configure assessment rules using the API](https://cloud.google.com/contact-center/insights/docs/assessment-rule)
 *
 * ## Example Usage
 *
 * ## Import
 *
 * AssessmentRule can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/assessmentRules/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, AssessmentRule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:contactcenterinsights/assessmentRule:AssessmentRule default projects/{{project}}/locations/{{location}}/assessmentRules/{{name}}
 * $ pulumi import gcp:contactcenterinsights/assessmentRule:AssessmentRule default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:contactcenterinsights/assessmentRule:AssessmentRule default {{location}}/{{name}}
 * ```
 */
class AssessmentRule extends pulumi.CustomResource {
    /**
     * Get an existing AssessmentRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AssessmentRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:contactcenterinsights/assessmentRule:AssessmentRule';
    /**
     * Returns true if the given object is an instance of AssessmentRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AssessmentRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["active"] = state?.active;
            resourceInputs["assessmentRuleId"] = state?.assessmentRuleId;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["sampleRule"] = state?.sampleRule;
            resourceInputs["scheduleInfo"] = state?.scheduleInfo;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["active"] = args?.active;
            resourceInputs["assessmentRuleId"] = args?.assessmentRuleId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["sampleRule"] = args?.sampleRule;
            resourceInputs["scheduleInfo"] = args?.scheduleInfo;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AssessmentRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AssessmentRule = AssessmentRule;
