// @pulumi/gcp/contactcenterinsights/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.View = exports.QaScorecardRevision = exports.QaScorecard = exports.QaQuestion = exports.EncryptionSpec = exports.AutoLabelingRule = exports.AssessmentRule = exports.AnalysisRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AnalysisRule = null;
utilities.lazyLoad(exports, ["AnalysisRule"], () => require("./analysisRule"));
exports.AssessmentRule = null;
utilities.lazyLoad(exports, ["AssessmentRule"], () => require("./assessmentRule"));
exports.AutoLabelingRule = null;
utilities.lazyLoad(exports, ["AutoLabelingRule"], () => require("./autoLabelingRule"));
exports.EncryptionSpec = null;
utilities.lazyLoad(exports, ["EncryptionSpec"], () => require("./encryptionSpec"));
exports.QaQuestion = null;
utilities.lazyLoad(exports, ["QaQuestion"], () => require("./qaQuestion"));
exports.QaScorecard = null;
utilities.lazyLoad(exports, ["QaScorecard"], () => require("./qaScorecard"));
exports.QaScorecardRevision = null;
utilities.lazyLoad(exports, ["QaScorecardRevision"], () => require("./qaScorecardRevision"));
exports.View = null;
utilities.lazyLoad(exports, ["View"], () => require("./view"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:contactcenterinsights/analysisRule:AnalysisRule":
                return new exports.AnalysisRule(name, undefined, { urn });
            case "gcp:contactcenterinsights/assessmentRule:AssessmentRule":
                return new exports.AssessmentRule(name, undefined, { urn });
            case "gcp:contactcenterinsights/autoLabelingRule:AutoLabelingRule":
                return new exports.AutoLabelingRule(name, undefined, { urn });
            case "gcp:contactcenterinsights/encryptionSpec:EncryptionSpec":
                return new exports.EncryptionSpec(name, undefined, { urn });
            case "gcp:contactcenterinsights/qaQuestion:QaQuestion":
                return new exports.QaQuestion(name, undefined, { urn });
            case "gcp:contactcenterinsights/qaScorecard:QaScorecard":
                return new exports.QaScorecard(name, undefined, { urn });
            case "gcp:contactcenterinsights/qaScorecardRevision:QaScorecardRevision":
                return new exports.QaScorecardRevision(name, undefined, { urn });
            case "gcp:contactcenterinsights/view:View":
                return new exports.View(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/analysisRule", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/assessmentRule", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/autoLabelingRule", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/encryptionSpec", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/qaQuestion", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/qaScorecard", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/qaScorecardRevision", _module);
pulumi.runtime.registerResourceModule("gcp", "contactcenterinsights/view", _module);
