// @pulumi/gcp/contactcenterinsights/qaQuestion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QaQuestion = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A single question to be scored by the Insights QA feature.
 *
 * ## Example Usage
 *
 * ## Import
 *
 * QaQuestion can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/qaScorecards/{{qa_scorecard}}/revisions/{{revision}}/qaQuestions/{{name}}`
 * * `{{project}}/{{location}}/{{qa_scorecard}}/{{revision}}/{{name}}`
 * * `{{location}}/{{qa_scorecard}}/{{revision}}/{{name}}`
 *
 * When using the `pulumi import` command, QaQuestion can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:contactcenterinsights/qaQuestion:QaQuestion default projects/{{project}}/locations/{{location}}/qaScorecards/{{qa_scorecard}}/revisions/{{revision}}/qaQuestions/{{name}}
 * $ pulumi import gcp:contactcenterinsights/qaQuestion:QaQuestion default {{project}}/{{location}}/{{qa_scorecard}}/{{revision}}/{{name}}
 * $ pulumi import gcp:contactcenterinsights/qaQuestion:QaQuestion default {{location}}/{{qa_scorecard}}/{{revision}}/{{name}}
 * ```
 */
class QaQuestion extends pulumi.CustomResource {
    /**
     * Get an existing QaQuestion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new QaQuestion(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:contactcenterinsights/qaQuestion:QaQuestion';
    /**
     * Returns true if the given object is an instance of QaQuestion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QaQuestion.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["abbreviation"] = state?.abbreviation;
            resourceInputs["answerChoices"] = state?.answerChoices;
            resourceInputs["answerInstructions"] = state?.answerInstructions;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["location"] = state?.location;
            resourceInputs["metrics"] = state?.metrics;
            resourceInputs["name"] = state?.name;
            resourceInputs["order"] = state?.order;
            resourceInputs["predefinedQuestionConfig"] = state?.predefinedQuestionConfig;
            resourceInputs["project"] = state?.project;
            resourceInputs["qaQuestionDataOptions"] = state?.qaQuestionDataOptions;
            resourceInputs["qaScorecard"] = state?.qaScorecard;
            resourceInputs["questionBody"] = state?.questionBody;
            resourceInputs["questionType"] = state?.questionType;
            resourceInputs["revision"] = state?.revision;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tuningMetadata"] = state?.tuningMetadata;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.qaScorecard === undefined && !opts.urn) {
                throw new Error("Missing required property 'qaScorecard'");
            }
            if (args?.revision === undefined && !opts.urn) {
                throw new Error("Missing required property 'revision'");
            }
            resourceInputs["abbreviation"] = args?.abbreviation;
            resourceInputs["answerChoices"] = args?.answerChoices;
            resourceInputs["answerInstructions"] = args?.answerInstructions;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["location"] = args?.location;
            resourceInputs["metrics"] = args?.metrics;
            resourceInputs["order"] = args?.order;
            resourceInputs["predefinedQuestionConfig"] = args?.predefinedQuestionConfig;
            resourceInputs["project"] = args?.project;
            resourceInputs["qaQuestionDataOptions"] = args?.qaQuestionDataOptions;
            resourceInputs["qaScorecard"] = args?.qaScorecard;
            resourceInputs["questionBody"] = args?.questionBody;
            resourceInputs["questionType"] = args?.questionType;
            resourceInputs["revision"] = args?.revision;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tuningMetadata"] = args?.tuningMetadata;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(QaQuestion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.QaQuestion = QaQuestion;
