// @pulumi/aws/amp/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceConfiguration = exports.Workspace = exports.Scraper = exports.RuleGroupNamespace = exports.ResourcePolicy = exports.QueryLoggingConfiguration = exports.getWorkspacesOutput = exports.getWorkspaces = exports.getWorkspaceOutput = exports.getWorkspace = exports.getDefaultScraperConfigurationOutput = exports.getDefaultScraperConfiguration = exports.AlertManagerDefinition = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AlertManagerDefinition = null;
utilities.lazyLoad(exports, ["AlertManagerDefinition"], () => require("./alertManagerDefinition"));
exports.getDefaultScraperConfiguration = null;
exports.getDefaultScraperConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDefaultScraperConfiguration", "getDefaultScraperConfigurationOutput"], () => require("./getDefaultScraperConfiguration"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspaces = null;
exports.getWorkspacesOutput = null;
utilities.lazyLoad(exports, ["getWorkspaces", "getWorkspacesOutput"], () => require("./getWorkspaces"));
exports.QueryLoggingConfiguration = null;
utilities.lazyLoad(exports, ["QueryLoggingConfiguration"], () => require("./queryLoggingConfiguration"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.RuleGroupNamespace = null;
utilities.lazyLoad(exports, ["RuleGroupNamespace"], () => require("./ruleGroupNamespace"));
exports.Scraper = null;
utilities.lazyLoad(exports, ["Scraper"], () => require("./scraper"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspaceConfiguration = null;
utilities.lazyLoad(exports, ["WorkspaceConfiguration"], () => require("./workspaceConfiguration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:amp/alertManagerDefinition:AlertManagerDefinition":
                return new exports.AlertManagerDefinition(name, undefined, { urn });
            case "aws:amp/queryLoggingConfiguration:QueryLoggingConfiguration":
                return new exports.QueryLoggingConfiguration(name, undefined, { urn });
            case "aws:amp/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:amp/ruleGroupNamespace:RuleGroupNamespace":
                return new exports.RuleGroupNamespace(name, undefined, { urn });
            case "aws:amp/scraper:Scraper":
                return new exports.Scraper(name, undefined, { urn });
            case "aws:amp/workspace:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "aws:amp/workspaceConfiguration:WorkspaceConfiguration":
                return new exports.WorkspaceConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "amp/alertManagerDefinition", _module);
pulumi.runtime.registerResourceModule("aws", "amp/queryLoggingConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "amp/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "amp/ruleGroupNamespace", _module);
pulumi.runtime.registerResourceModule("aws", "amp/scraper", _module);
pulumi.runtime.registerResourceModule("aws", "amp/workspace", _module);
pulumi.runtime.registerResourceModule("aws", "amp/workspaceConfiguration", _module);
