// @pulumi/aws/amp/scraper.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scraper = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * > **Note:** If you change a Scraper's source (EKS cluster or VPC configuration), Terraform
 * will delete the current Scraper and create a new one.
 *
 * Provides an Amazon Managed Service for Prometheus fully managed collector
 * (scraper).
 *
 * Read more in the [Amazon Managed Service for Prometheus user guide](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-collector.html).
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.amp.Scraper("example", {
 *     source: {
 *         eks: {
 *             clusterArn: exampleAwsEksCluster.arn,
 *             subnetIds: exampleAwsEksCluster.vpcConfig[0].subnetIds,
 *         },
 *     },
 *     destination: {
 *         amp: {
 *             workspaceArn: exampleAwsPrometheusWorkspace.arn,
 *         },
 *     },
 *     scrapeConfiguration: `global:
 *   scrape_interval: 30s
 * scrape_configs:
 *   # pod metrics
 *   - job_name: pod_exporter
 *     kubernetes_sd_configs:
 *       - role: pod
 *   # container metrics
 *   - job_name: cadvisor
 *     scheme: https
 *     authorization:
 *       credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
 *     kubernetes_sd_configs:
 *       - role: node
 *     relabel_configs:
 *       - action: labelmap
 *         regex: __meta_kubernetes_node_label_(.+)
 *       - replacement: kubernetes.default.svc:443
 *         target_label: __address__
 *       - source_labels: [__meta_kubernetes_node_name]
 *         regex: (.+)
 *         target_label: __metrics_path__
 *         replacement: /api/v1/nodes/1/proxy/metrics/cadvisor
 *   # apiserver metrics
 *   - bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
 *     job_name: kubernetes-apiservers
 *     kubernetes_sd_configs:
 *     - role: endpoints
 *     relabel_configs:
 *     - action: keep
 *       regex: default;kubernetes;https
 *       source_labels:
 *       - __meta_kubernetes_namespace
 *       - __meta_kubernetes_service_name
 *       - __meta_kubernetes_endpoint_port_name
 *     scheme: https
 *   # kube proxy metrics
 *   - job_name: kube-proxy
 *     honor_labels: true
 *     kubernetes_sd_configs:
 *     - role: pod
 *     relabel_configs:
 *     - action: keep
 *       source_labels:
 *       - __meta_kubernetes_namespace
 *       - __meta_kubernetes_pod_name
 *       separator: '/'
 *       regex: 'kube-system/kube-proxy.+'
 *     - source_labels:
 *       - __address__
 *       action: replace
 *       target_label: __address__
 *       regex: (.+?)(\\\\\\\\:\\\\\\\\d+)?
 *       replacement: 1:10249
 * `,
 * });
 * ```
 *
 * ### VPC Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.amp.Scraper("example", {
 *     source: {
 *         vpc: {
 *             securityGroupIds: [exampleAwsSecurityGroup.id],
 *             subnetIds: [
 *                 example1.id,
 *                 example2.id,
 *             ],
 *         },
 *     },
 *     destination: {
 *         amp: {
 *             workspaceArn: exampleAwsPrometheusWorkspace.arn,
 *         },
 *     },
 *     scrapeConfiguration: `global:
 *   scrape_interval: 30s
 * scrape_configs:
 *   - job_name: 'my-service'
 *     dns_sd_configs:
 *       - names: ['my-service.my-namespace']
 *         type: A
 *         port: 8080
 *     metrics_path: '/metrics'
 *     relabel_configs:
 *       - target_label: service_name
 *         replacement: 'my-service'
 *       - target_label: discovery_method
 *         replacement: 'cloudmap-dns'
 * `,
 * });
 * ```
 *
 * ### Use default EKS scraper configuration
 *
 * You can use the data source `awsPrometheusScraperConfiguration` to use a
 * service managed scrape configuration.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = aws.amp.getDefaultScraperConfiguration({});
 * const exampleScraper = new aws.amp.Scraper("example", {
 *     destination: {
 *         amp: {
 *             workspaceArn: exampleAwsPrometheusWorkspace.arn,
 *         },
 *     },
 *     scrapeConfiguration: exampleAwsPrometheusScraperConfiguration.configuration,
 *     source: {
 *         eks: {
 *             clusterArn: exampleAwsEksCluster.arn,
 *             subnetIds: exampleAwsEksCluster.vpcConfig[0].subnetIds,
 *         },
 *     },
 * });
 * ```
 *
 * ### Ignoring changes to Prometheus Workspace destination
 *
 * A managed scraper will add a `AMPAgentlessScraper` tag to its Prometheus workspace
 * destination. To avoid Terraform state forcing removing the tag from the workspace,
 * you can add this tag to the destination workspace (preferred) or ignore tags
 * changes with `lifecycle`. See example below.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const _this = aws.eks.getCluster({
 *     name: "example",
 * });
 * const example = new aws.amp.Workspace("example", {tags: {
 *     AMPAgentlessScraper: "",
 * }});
 * const exampleScraper = new aws.amp.Scraper("example", {
 *     source: {
 *         eks: {
 *             clusterArn: exampleAwsEksCluster.arn,
 *             subnetIds: exampleAwsEksCluster.vpcConfig[0].subnetIds,
 *         },
 *     },
 *     scrapeConfiguration: "...",
 *     destination: {
 *         amp: {
 *             workspaceArn: example.arn,
 *         },
 *     },
 * });
 * ```
 *
 * ### Configure aws-auth
 *
 * Your source Amazon EKS cluster must be configured to allow the scraper to access
 * metrics. Follow the [user guide](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-collector-how-to.html#AMP-collector-eks-setup)
 * to setup the appropriate Kubernetes permissions.
 *
 * ### Cross-Account Configuration
 *
 * This setup allows the scraper, running in a source account, to remote write its collected metrics to a workspace in a target account. Note that:
 *
 * - The target Role and target Workspace must be in the same account
 * - The source Scraper and target Workspace must be in the same Region
 *
 * Follow [the AWS Best Practices guide](https://aws-observability.github.io/observability-best-practices/patterns/ampxa) to learn about the IAM roles configuration and overall setup.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.amp.Scraper("example", {
 *     source: {
 *         eks: {
 *             clusterArn: exampleAwsEksCluster.arn,
 *             subnetIds: exampleAwsEksCluster.vpcConfig[0].subnetIds,
 *         },
 *     },
 *     destination: {
 *         amp: {
 *             workspaceArn: "<target_account_workspace_arn>",
 *         },
 *     },
 *     roleConfiguration: {
 *         sourceRoleArn: source.arn,
 *         targetRoleArn: "arn:aws:iam::ACCOUNT-ID:role/target-role-name",
 *     },
 *     scrapeConfiguration: "...",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import the Managed Scraper using its identifier.
 * For example:
 *
 * ```sh
 * $ pulumi import aws:amp/scraper:Scraper example s-0123abc-0000-0123-a000-000000000000
 * ```
 */
class Scraper extends pulumi.CustomResource {
    /**
     * Get an existing Scraper resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Scraper(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:amp/scraper:Scraper';
    /**
     * Returns true if the given object is an instance of Scraper.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Scraper.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["alias"] = state?.alias;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["destination"] = state?.destination;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["roleConfiguration"] = state?.roleConfiguration;
            resourceInputs["scrapeConfiguration"] = state?.scrapeConfiguration;
            resourceInputs["source"] = state?.source;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.destination === undefined && !opts.urn) {
                throw new Error("Missing required property 'destination'");
            }
            if (args?.scrapeConfiguration === undefined && !opts.urn) {
                throw new Error("Missing required property 'scrapeConfiguration'");
            }
            resourceInputs["alias"] = args?.alias;
            resourceInputs["destination"] = args?.destination;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleConfiguration"] = args?.roleConfiguration;
            resourceInputs["scrapeConfiguration"] = args?.scrapeConfiguration;
            resourceInputs["source"] = args?.source;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Scraper.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Scraper = Scraper;
