// @pulumi/gcp/biglake/icebergTable.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IcebergTable = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * IcebergTables are the primary objects in an IcebergCatalog.
 *
 * ## Example Usage
 *
 * ### Biglake Iceberg Table Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "my-bucket",
 *     location: "us-central1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 * });
 * const catalog = new gcp.biglake.IcebergCatalog("catalog", {
 *     name: bucket.name,
 *     catalogType: "CATALOG_TYPE_GCS_BUCKET",
 * });
 * const namespace = new gcp.biglake.IcebergNamespace("namespace", {
 *     catalog: catalog.name,
 *     namespaceId: "my_namespace",
 * });
 * const myIcebergTable = new gcp.biglake.IcebergTable("my_iceberg_table", {
 *     catalog: catalog.name,
 *     namespace: namespace.namespaceId,
 *     name: "my_table",
 *     location: pulumi.interpolate`gs://${bucket.name}/${namespace.namespaceId}/my_table`,
 *     schema: {
 *         type: "struct",
 *         fields: [
 *             {
 *                 id: 1,
 *                 name: "id",
 *                 type: "long",
 *                 required: true,
 *                 doc: "The ID of the record",
 *             },
 *             {
 *                 id: 2,
 *                 name: "name",
 *                 type: "string",
 *                 required: false,
 *             },
 *         ],
 *         identifierFieldIds: [1],
 *     },
 *     partitionSpec: {
 *         fields: [{
 *             name: "id_partition",
 *             sourceId: 1,
 *             transform: "identity",
 *         }],
 *     },
 * });
 * ```
 * ### Biglake Iceberg Table Update
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "my-bucket",
 *     location: "us-central1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 * });
 * const catalog = new gcp.biglake.IcebergCatalog("catalog", {
 *     name: bucket.name,
 *     catalogType: "CATALOG_TYPE_GCS_BUCKET",
 * });
 * const namespace = new gcp.biglake.IcebergNamespace("namespace", {
 *     catalog: catalog.name,
 *     namespaceId: "my_namespace",
 * });
 * const myIcebergTable = new gcp.biglake.IcebergTable("my_iceberg_table", {
 *     catalog: catalog.name,
 *     namespace: namespace.namespaceId,
 *     name: "my_table",
 *     schema: {
 *         type: "struct",
 *         fields: [{
 *             id: 1,
 *             name: "id",
 *             type: "long",
 *             required: true,
 *         }],
 *     },
 *     properties: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * IcebergTable can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/catalogs/{{catalog}}/namespaces/{{namespace}}/tables/{{name}}`
 * * `{{project}}/{{catalog}}/{{namespace}}/{{name}}`
 * * `{{catalog}}/{{namespace}}/{{name}}`
 *
 * When using the `pulumi import` command, IcebergTable can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:biglake/icebergTable:IcebergTable default projects/{{project}}/catalogs/{{catalog}}/namespaces/{{namespace}}/tables/{{name}}
 * $ pulumi import gcp:biglake/icebergTable:IcebergTable default {{project}}/{{catalog}}/{{namespace}}/{{name}}
 * $ pulumi import gcp:biglake/icebergTable:IcebergTable default {{catalog}}/{{namespace}}/{{name}}
 * ```
 */
class IcebergTable extends pulumi.CustomResource {
    /**
     * Get an existing IcebergTable resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IcebergTable(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:biglake/icebergTable:IcebergTable';
    /**
     * Returns true if the given object is an instance of IcebergTable.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IcebergTable.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["catalog"] = state?.catalog;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["namespace"] = state?.namespace;
            resourceInputs["partitionSpec"] = state?.partitionSpec;
            resourceInputs["project"] = state?.project;
            resourceInputs["properties"] = state?.properties;
            resourceInputs["schema"] = state?.schema;
        }
        else {
            const args = argsOrState;
            if (args?.catalog === undefined && !opts.urn) {
                throw new Error("Missing required property 'catalog'");
            }
            if (args?.namespace === undefined && !opts.urn) {
                throw new Error("Missing required property 'namespace'");
            }
            if (args?.schema === undefined && !opts.urn) {
                throw new Error("Missing required property 'schema'");
            }
            resourceInputs["catalog"] = args?.catalog;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["namespace"] = args?.namespace;
            resourceInputs["partitionSpec"] = args?.partitionSpec;
            resourceInputs["project"] = args?.project;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["schema"] = args?.schema;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IcebergTable.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IcebergTable = IcebergTable;
