// @pulumi/gcp/biglake/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = exports.IcebergTableIamPolicy = exports.IcebergTableIamMember = exports.IcebergTableIamBinding = exports.IcebergTable = exports.IcebergNamespaceIamPolicy = exports.IcebergNamespaceIamMember = exports.IcebergNamespaceIamBinding = exports.IcebergNamespace = exports.IcebergCatalogIamPolicy = exports.IcebergCatalogIamMember = exports.IcebergCatalogIamBinding = exports.IcebergCatalog = exports.getIcebergTableIamPolicyOutput = exports.getIcebergTableIamPolicy = exports.getIcebergNamespaceIamPolicyOutput = exports.getIcebergNamespaceIamPolicy = exports.getIcebergCatalogIamPolicyOutput = exports.getIcebergCatalogIamPolicy = exports.Database = exports.Catalog = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Catalog = null;
utilities.lazyLoad(exports, ["Catalog"], () => require("./catalog"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.getIcebergCatalogIamPolicy = null;
exports.getIcebergCatalogIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getIcebergCatalogIamPolicy", "getIcebergCatalogIamPolicyOutput"], () => require("./getIcebergCatalogIamPolicy"));
exports.getIcebergNamespaceIamPolicy = null;
exports.getIcebergNamespaceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getIcebergNamespaceIamPolicy", "getIcebergNamespaceIamPolicyOutput"], () => require("./getIcebergNamespaceIamPolicy"));
exports.getIcebergTableIamPolicy = null;
exports.getIcebergTableIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getIcebergTableIamPolicy", "getIcebergTableIamPolicyOutput"], () => require("./getIcebergTableIamPolicy"));
exports.IcebergCatalog = null;
utilities.lazyLoad(exports, ["IcebergCatalog"], () => require("./icebergCatalog"));
exports.IcebergCatalogIamBinding = null;
utilities.lazyLoad(exports, ["IcebergCatalogIamBinding"], () => require("./icebergCatalogIamBinding"));
exports.IcebergCatalogIamMember = null;
utilities.lazyLoad(exports, ["IcebergCatalogIamMember"], () => require("./icebergCatalogIamMember"));
exports.IcebergCatalogIamPolicy = null;
utilities.lazyLoad(exports, ["IcebergCatalogIamPolicy"], () => require("./icebergCatalogIamPolicy"));
exports.IcebergNamespace = null;
utilities.lazyLoad(exports, ["IcebergNamespace"], () => require("./icebergNamespace"));
exports.IcebergNamespaceIamBinding = null;
utilities.lazyLoad(exports, ["IcebergNamespaceIamBinding"], () => require("./icebergNamespaceIamBinding"));
exports.IcebergNamespaceIamMember = null;
utilities.lazyLoad(exports, ["IcebergNamespaceIamMember"], () => require("./icebergNamespaceIamMember"));
exports.IcebergNamespaceIamPolicy = null;
utilities.lazyLoad(exports, ["IcebergNamespaceIamPolicy"], () => require("./icebergNamespaceIamPolicy"));
exports.IcebergTable = null;
utilities.lazyLoad(exports, ["IcebergTable"], () => require("./icebergTable"));
exports.IcebergTableIamBinding = null;
utilities.lazyLoad(exports, ["IcebergTableIamBinding"], () => require("./icebergTableIamBinding"));
exports.IcebergTableIamMember = null;
utilities.lazyLoad(exports, ["IcebergTableIamMember"], () => require("./icebergTableIamMember"));
exports.IcebergTableIamPolicy = null;
utilities.lazyLoad(exports, ["IcebergTableIamPolicy"], () => require("./icebergTableIamPolicy"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:biglake/catalog:Catalog":
                return new exports.Catalog(name, undefined, { urn });
            case "gcp:biglake/database:Database":
                return new exports.Database(name, undefined, { urn });
            case "gcp:biglake/icebergCatalog:IcebergCatalog":
                return new exports.IcebergCatalog(name, undefined, { urn });
            case "gcp:biglake/icebergCatalogIamBinding:IcebergCatalogIamBinding":
                return new exports.IcebergCatalogIamBinding(name, undefined, { urn });
            case "gcp:biglake/icebergCatalogIamMember:IcebergCatalogIamMember":
                return new exports.IcebergCatalogIamMember(name, undefined, { urn });
            case "gcp:biglake/icebergCatalogIamPolicy:IcebergCatalogIamPolicy":
                return new exports.IcebergCatalogIamPolicy(name, undefined, { urn });
            case "gcp:biglake/icebergNamespace:IcebergNamespace":
                return new exports.IcebergNamespace(name, undefined, { urn });
            case "gcp:biglake/icebergNamespaceIamBinding:IcebergNamespaceIamBinding":
                return new exports.IcebergNamespaceIamBinding(name, undefined, { urn });
            case "gcp:biglake/icebergNamespaceIamMember:IcebergNamespaceIamMember":
                return new exports.IcebergNamespaceIamMember(name, undefined, { urn });
            case "gcp:biglake/icebergNamespaceIamPolicy:IcebergNamespaceIamPolicy":
                return new exports.IcebergNamespaceIamPolicy(name, undefined, { urn });
            case "gcp:biglake/icebergTable:IcebergTable":
                return new exports.IcebergTable(name, undefined, { urn });
            case "gcp:biglake/icebergTableIamBinding:IcebergTableIamBinding":
                return new exports.IcebergTableIamBinding(name, undefined, { urn });
            case "gcp:biglake/icebergTableIamMember:IcebergTableIamMember":
                return new exports.IcebergTableIamMember(name, undefined, { urn });
            case "gcp:biglake/icebergTableIamPolicy:IcebergTableIamPolicy":
                return new exports.IcebergTableIamPolicy(name, undefined, { urn });
            case "gcp:biglake/table:Table":
                return new exports.Table(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "biglake/catalog", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/database", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergCatalog", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergCatalogIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergCatalogIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergCatalogIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergNamespace", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergNamespaceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergNamespaceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergNamespaceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergTable", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergTableIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergTableIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/icebergTableIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "biglake/table", _module);
