// @pulumi/gcp/biglake/table.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a table.
 *
 * To get more information about Table, see:
 *
 * * [API documentation](https://cloud.google.com/bigquery/docs/reference/biglake/rest/v1/projects.locations.catalogs.databases.tables)
 * * How-to Guides
 *     * [Manage open source metadata with BigLake Metastore](https://cloud.google.com/bigquery/docs/manage-open-source-metadata#create_tables)
 *
 * ## Example Usage
 *
 * ### Biglake Table
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const catalog = new gcp.biglake.Catalog("catalog", {
 *     name: "my_catalog",
 *     location: "US",
 * });
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     name: "my_bucket",
 *     location: "US",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 * });
 * const metadataFolder = new gcp.storage.BucketObject("metadata_folder", {
 *     name: "metadata/",
 *     content: " ",
 *     bucket: bucket.name,
 * });
 * const dataFolder = new gcp.storage.BucketObject("data_folder", {
 *     name: "data/",
 *     content: " ",
 *     bucket: bucket.name,
 * });
 * const database = new gcp.biglake.Database("database", {
 *     name: "my_database",
 *     catalog: catalog.id,
 *     type: "HIVE",
 *     hiveOptions: {
 *         locationUri: pulumi.interpolate`gs://${bucket.name}/${metadataFolder.name}`,
 *         parameters: {
 *             owner: "Alex",
 *         },
 *     },
 * });
 * const table = new gcp.biglake.Table("table", {
 *     name: "my_table",
 *     database: database.id,
 *     type: "HIVE",
 *     hiveOptions: {
 *         tableType: "MANAGED_TABLE",
 *         storageDescriptor: {
 *             locationUri: pulumi.interpolate`gs://${bucket.name}/${dataFolder.name}`,
 *             inputFormat: "org.apache.hadoop.mapred.SequenceFileInputFormat",
 *             outputFormat: "org.apache.hadoop.hive.ql.io.HiveSequenceFileOutputFormat",
 *         },
 *         parameters: {
 *             "spark.sql.create.version": "3.1.3",
 *             "spark.sql.sources.schema.numParts": "1",
 *             transient_lastDdlTime: "1680894197",
 *             "spark.sql.partitionProvider": "catalog",
 *             owner: "John Doe",
 *             "spark.sql.sources.schema.part.0": "{\"type\":\"struct\",\"fields\":[{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"name\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"age\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}",
 *             "spark.sql.sources.provider": "iceberg",
 *             provider: "iceberg",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Table can be imported using any of these accepted formats:
 *
 * * `{{database}}/tables/{{name}}`
 *
 * When using the `pulumi import` command, Table can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:biglake/table:Table default {{database}}/tables/{{name}}
 * ```
 */
class Table extends pulumi.CustomResource {
    /**
     * Get an existing Table resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Table(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:biglake/table:Table';
    /**
     * Returns true if the given object is an instance of Table.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Table.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["database"] = state?.database;
            resourceInputs["deleteTime"] = state?.deleteTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["expireTime"] = state?.expireTime;
            resourceInputs["hiveOptions"] = state?.hiveOptions;
            resourceInputs["name"] = state?.name;
            resourceInputs["type"] = state?.type;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            resourceInputs["database"] = args?.database;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["hiveOptions"] = args?.hiveOptions;
            resourceInputs["name"] = args?.name;
            resourceInputs["type"] = args?.type;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["deleteTime"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["expireTime"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Table.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Table = Table;
