// @pulumi/gcp/projects/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageExportBucket = exports.ServiceIdentity = exports.Service = exports.OrganizationPolicy = exports.IAMPolicy = exports.IAMMember = exports.IAMCustomRole = exports.IAMBinding = exports.IAMAuditConfig = exports.IamMemberRemove = exports.getProjectServiceOutput = exports.getProjectService = exports.getProjectOutput = exports.getProject = exports.getOrganizationPolicyOutput = exports.getOrganizationPolicy = exports.getIamPolicyOutput = exports.getIamPolicy = exports.getIamCustomRolesOutput = exports.getIamCustomRoles = exports.getIamCustomRoleOutput = exports.getIamCustomRole = exports.getAncestryOutput = exports.getAncestry = exports.DefaultServiceAccounts = exports.ApiKey = exports.AccessApprovalSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccessApprovalSettings = null;
utilities.lazyLoad(exports, ["AccessApprovalSettings"], () => require("./accessApprovalSettings"));
exports.ApiKey = null;
utilities.lazyLoad(exports, ["ApiKey"], () => require("./apiKey"));
exports.DefaultServiceAccounts = null;
utilities.lazyLoad(exports, ["DefaultServiceAccounts"], () => require("./defaultServiceAccounts"));
exports.getAncestry = null;
exports.getAncestryOutput = null;
utilities.lazyLoad(exports, ["getAncestry", "getAncestryOutput"], () => require("./getAncestry"));
exports.getIamCustomRole = null;
exports.getIamCustomRoleOutput = null;
utilities.lazyLoad(exports, ["getIamCustomRole", "getIamCustomRoleOutput"], () => require("./getIamCustomRole"));
exports.getIamCustomRoles = null;
exports.getIamCustomRolesOutput = null;
utilities.lazyLoad(exports, ["getIamCustomRoles", "getIamCustomRolesOutput"], () => require("./getIamCustomRoles"));
exports.getIamPolicy = null;
exports.getIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getIamPolicy", "getIamPolicyOutput"], () => require("./getIamPolicy"));
exports.getOrganizationPolicy = null;
exports.getOrganizationPolicyOutput = null;
utilities.lazyLoad(exports, ["getOrganizationPolicy", "getOrganizationPolicyOutput"], () => require("./getOrganizationPolicy"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getProjectService = null;
exports.getProjectServiceOutput = null;
utilities.lazyLoad(exports, ["getProjectService", "getProjectServiceOutput"], () => require("./getProjectService"));
exports.IamMemberRemove = null;
utilities.lazyLoad(exports, ["IamMemberRemove"], () => require("./iamMemberRemove"));
exports.IAMAuditConfig = null;
utilities.lazyLoad(exports, ["IAMAuditConfig"], () => require("./iamauditConfig"));
exports.IAMBinding = null;
utilities.lazyLoad(exports, ["IAMBinding"], () => require("./iambinding"));
exports.IAMCustomRole = null;
utilities.lazyLoad(exports, ["IAMCustomRole"], () => require("./iamcustomRole"));
exports.IAMMember = null;
utilities.lazyLoad(exports, ["IAMMember"], () => require("./iammember"));
exports.IAMPolicy = null;
utilities.lazyLoad(exports, ["IAMPolicy"], () => require("./iampolicy"));
exports.OrganizationPolicy = null;
utilities.lazyLoad(exports, ["OrganizationPolicy"], () => require("./organizationPolicy"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceIdentity = null;
utilities.lazyLoad(exports, ["ServiceIdentity"], () => require("./serviceIdentity"));
exports.UsageExportBucket = null;
utilities.lazyLoad(exports, ["UsageExportBucket"], () => require("./usageExportBucket"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:projects/accessApprovalSettings:AccessApprovalSettings":
                return new exports.AccessApprovalSettings(name, undefined, { urn });
            case "gcp:projects/apiKey:ApiKey":
                return new exports.ApiKey(name, undefined, { urn });
            case "gcp:projects/defaultServiceAccounts:DefaultServiceAccounts":
                return new exports.DefaultServiceAccounts(name, undefined, { urn });
            case "gcp:projects/iAMAuditConfig:IAMAuditConfig":
                return new exports.IAMAuditConfig(name, undefined, { urn });
            case "gcp:projects/iAMBinding:IAMBinding":
                return new exports.IAMBinding(name, undefined, { urn });
            case "gcp:projects/iAMCustomRole:IAMCustomRole":
                return new exports.IAMCustomRole(name, undefined, { urn });
            case "gcp:projects/iAMMember:IAMMember":
                return new exports.IAMMember(name, undefined, { urn });
            case "gcp:projects/iAMPolicy:IAMPolicy":
                return new exports.IAMPolicy(name, undefined, { urn });
            case "gcp:projects/iamMemberRemove:IamMemberRemove":
                return new exports.IamMemberRemove(name, undefined, { urn });
            case "gcp:projects/organizationPolicy:OrganizationPolicy":
                return new exports.OrganizationPolicy(name, undefined, { urn });
            case "gcp:projects/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "gcp:projects/serviceIdentity:ServiceIdentity":
                return new exports.ServiceIdentity(name, undefined, { urn });
            case "gcp:projects/usageExportBucket:UsageExportBucket":
                return new exports.UsageExportBucket(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "projects/accessApprovalSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/apiKey", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/defaultServiceAccounts", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iAMAuditConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iAMCustomRole", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/iamMemberRemove", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/organizationPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/service", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/serviceIdentity", _module);
pulumi.runtime.registerResourceModule("gcp", "projects/usageExportBucket", _module);
