// @pulumi/azure-native/costmanagement/viewByScope.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewByScope = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * States and configurations of Cost Analysis.
 *
 * Uses Azure REST API version 2024-08-01. In version 2.x of the Azure Native provider, it used API version 2023-03-01.
 *
 * Other available API versions: 2019-04-01-preview, 2019-11-01, 2020-06-01, 2021-10-01, 2022-08-01-preview, 2022-10-01, 2022-10-01-preview, 2022-10-05-preview, 2023-03-01, 2023-04-01-preview, 2023-07-01-preview, 2023-08-01, 2023-09-01, 2023-11-01, 2024-10-01-preview, 2025-03-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native costmanagement [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ViewByScope extends pulumi.CustomResource {
    /**
     * Get an existing ViewByScope resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ViewByScope(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:costmanagement:ViewByScope';
    /**
     * Returns true if the given object is an instance of ViewByScope.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ViewByScope.__pulumiType;
    }
    /**
     * Create a ViewByScope resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.scope === undefined && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            if (args?.timeframe === undefined && !opts.urn) {
                throw new Error("Missing required property 'timeframe'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["accumulated"] = args?.accumulated;
            resourceInputs["chart"] = args?.chart;
            resourceInputs["dataSet"] = args?.dataSet;
            resourceInputs["dateRange"] = args?.dateRange;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["eTag"] = args?.eTag;
            resourceInputs["includeMonetaryCommitment"] = args?.includeMonetaryCommitment;
            resourceInputs["kpis"] = args?.kpis;
            resourceInputs["metric"] = args?.metric;
            resourceInputs["modifiedOn"] = args?.modifiedOn;
            resourceInputs["pivots"] = args?.pivots;
            resourceInputs["scope"] = args?.scope;
            resourceInputs["timePeriod"] = args?.timePeriod;
            resourceInputs["timeframe"] = args?.timeframe;
            resourceInputs["type"] = args?.type;
            resourceInputs["viewName"] = args?.viewName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        else {
            resourceInputs["accumulated"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["chart"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["dataSet"] = undefined /*out*/;
            resourceInputs["dateRange"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["includeMonetaryCommitment"] = undefined /*out*/;
            resourceInputs["kpis"] = undefined /*out*/;
            resourceInputs["metric"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pivots"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["timePeriod"] = undefined /*out*/;
            resourceInputs["timeframe"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:costmanagement/v20190401preview:ViewByScope" }, { type: "azure-native:costmanagement/v20191101:ViewByScope" }, { type: "azure-native:costmanagement/v20200601:ViewByScope" }, { type: "azure-native:costmanagement/v20211001:ViewByScope" }, { type: "azure-native:costmanagement/v20220801preview:ViewByScope" }, { type: "azure-native:costmanagement/v20221001:ViewByScope" }, { type: "azure-native:costmanagement/v20221001preview:ViewByScope" }, { type: "azure-native:costmanagement/v20221005preview:ViewByScope" }, { type: "azure-native:costmanagement/v20230301:ViewByScope" }, { type: "azure-native:costmanagement/v20230401preview:ViewByScope" }, { type: "azure-native:costmanagement/v20230701preview:ViewByScope" }, { type: "azure-native:costmanagement/v20230801:ViewByScope" }, { type: "azure-native:costmanagement/v20230901:ViewByScope" }, { type: "azure-native:costmanagement/v20231101:ViewByScope" }, { type: "azure-native:costmanagement/v20240801:ViewByScope" }, { type: "azure-native:costmanagement/v20241001preview:ViewByScope" }, { type: "azure-native:costmanagement/v20250301:ViewByScope" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ViewByScope.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ViewByScope = ViewByScope;
