// @pulumi/azure-native/resources/azurePowerShellScript.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzurePowerShellScript = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Object model for the Azure PowerShell script.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2020-10-01.
 */
class AzurePowerShellScript extends pulumi.CustomResource {
    /**
     * Get an existing AzurePowerShellScript resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AzurePowerShellScript(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:resources:AzurePowerShellScript';
    /**
     * Returns true if the given object is an instance of AzurePowerShellScript.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AzurePowerShellScript.__pulumiType;
    }
    /**
     * Create a AzurePowerShellScript resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.azPowerShellVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'azPowerShellVersion'");
            }
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.retentionInterval === undefined && !opts.urn) {
                throw new Error("Missing required property 'retentionInterval'");
            }
            resourceInputs["arguments"] = args?.arguments;
            resourceInputs["azPowerShellVersion"] = args?.azPowerShellVersion;
            resourceInputs["cleanupPreference"] = (args?.cleanupPreference) ?? "Always";
            resourceInputs["containerSettings"] = args?.containerSettings;
            resourceInputs["environmentVariables"] = args?.environmentVariables;
            resourceInputs["forceUpdateTag"] = args?.forceUpdateTag;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["kind"] = "AzurePowerShell";
            resourceInputs["location"] = args?.location;
            resourceInputs["primaryScriptUri"] = args?.primaryScriptUri;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["retentionInterval"] = args?.retentionInterval;
            resourceInputs["scriptContent"] = args?.scriptContent;
            resourceInputs["scriptName"] = args?.scriptName;
            resourceInputs["storageAccountSettings"] = args?.storageAccountSettings;
            resourceInputs["supportingScriptUris"] = args?.supportingScriptUris;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeout"] = (args?.timeout) ?? "P1D";
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["outputs"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["arguments"] = undefined /*out*/;
            resourceInputs["azPowerShellVersion"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cleanupPreference"] = undefined /*out*/;
            resourceInputs["containerSettings"] = undefined /*out*/;
            resourceInputs["environmentVariables"] = undefined /*out*/;
            resourceInputs["forceUpdateTag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["outputs"] = undefined /*out*/;
            resourceInputs["primaryScriptUri"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["retentionInterval"] = undefined /*out*/;
            resourceInputs["scriptContent"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["storageAccountSettings"] = undefined /*out*/;
            resourceInputs["supportingScriptUris"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["timeout"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:resources/v20191001preview:AzureCliScript" }, { type: "azure-native:resources/v20191001preview:AzurePowerShellScript" }, { type: "azure-native:resources/v20201001:AzureCliScript" }, { type: "azure-native:resources/v20201001:AzurePowerShellScript" }, { type: "azure-native:resources/v20230801:AzureCliScript" }, { type: "azure-native:resources/v20230801:AzurePowerShellScript" }, { type: "azure-native:resources:AzureCliScript" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AzurePowerShellScript.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AzurePowerShellScript = AzurePowerShellScript;
