// @pulumi/azure-native/types/enums/avs/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemAssignedServiceIdentityType = exports.SslEnum = exports.SkuTier = exports.ScriptExecutionParameterType = exports.PortMirroringDirectionEnum = exports.PlacementPolicyType = exports.PlacementPolicyState = exports.MountOptionEnum = exports.LicenseKind = exports.InternetEnum = exports.EncryptionState = exports.DnsZoneType = exports.DnsServiceLogLevelEnum = exports.DhcpTypeEnum = exports.AzureHybridBenefitType = exports.AvailabilityStrategy = exports.AffinityType = exports.AffinityStrength = exports.AddonType = void 0;
exports.AddonType = {
    SRM: "SRM",
    VR: "VR",
    HCX: "HCX",
    Arc: "Arc",
};
exports.AffinityStrength = {
    /**
     * is should
     */
    Should: "Should",
    /**
     * is must
     */
    Must: "Must",
};
exports.AffinityType = {
    /**
     * is affinity
     */
    Affinity: "Affinity",
    /**
     * is anti-affinity
     */
    AntiAffinity: "AntiAffinity",
};
exports.AvailabilityStrategy = {
    /**
     * in single zone
     */
    SingleZone: "SingleZone",
    /**
     * in two zones
     */
    DualZone: "DualZone",
};
exports.AzureHybridBenefitType = {
    /**
     * is SqlHost
     */
    SqlHost: "SqlHost",
    /**
     * is None
     */
    None: "None",
};
exports.DhcpTypeEnum = {
    SERVER: "SERVER",
    RELAY: "RELAY",
};
exports.DnsServiceLogLevelEnum = {
    /**
     * is debug
     */
    DEBUG: "DEBUG",
    /**
     * is info
     */
    INFO: "INFO",
    /**
     * is warning
     */
    WARNING: "WARNING",
    /**
     * is error
     */
    ERROR: "ERROR",
    /**
     * is fatal
     */
    FATAL: "FATAL",
};
exports.DnsZoneType = {
    /**
     * Primary DNS zone.
     */
    Public: "Public",
    /**
     * Private DNS zone.
     */
    Private: "Private",
};
exports.EncryptionState = {
    /**
     * is enabled
     */
    Enabled: "Enabled",
    /**
     * is disabled
     */
    Disabled: "Disabled",
};
exports.InternetEnum = {
    /**
     * is enabled
     */
    Enabled: "Enabled",
    /**
     * is disabled
     */
    Disabled: "Disabled",
};
exports.LicenseKind = {
    VmwareFirewall: "VmwareFirewall",
};
exports.MountOptionEnum = {
    /**
     * is mount
     */
    MOUNT: "MOUNT",
    /**
     * is attach
     */
    ATTACH: "ATTACH",
};
exports.PlacementPolicyState = {
    /**
     * is enabled
     */
    Enabled: "Enabled",
    /**
     * is disabled
     */
    Disabled: "Disabled",
};
exports.PlacementPolicyType = {
    VmVm: "VmVm",
    VmHost: "VmHost",
};
exports.PortMirroringDirectionEnum = {
    /**
     * is ingress
     */
    INGRESS: "INGRESS",
    /**
     * is egress
     */
    EGRESS: "EGRESS",
    /**
     * is bidirectional
     */
    BIDIRECTIONAL: "BIDIRECTIONAL",
};
exports.ScriptExecutionParameterType = {
    Value: "Value",
    SecureValue: "SecureValue",
    Credential: "Credential",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SslEnum = {
    /**
     * is enabled
     */
    Enabled: "Enabled",
    /**
     * is disabled
     */
    Disabled: "Disabled",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
