// @pulumi/gcp/osconfig/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2PolicyOrchestratorForOrganization = exports.V2PolicyOrchestratorForFolder = exports.V2PolicyOrchestrator = exports.PatchDeployment = exports.OsPolicyAssignment = exports.GuestPolicies = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.GuestPolicies = null;
utilities.lazyLoad(exports, ["GuestPolicies"], () => require("./guestPolicies"));
exports.OsPolicyAssignment = null;
utilities.lazyLoad(exports, ["OsPolicyAssignment"], () => require("./osPolicyAssignment"));
exports.PatchDeployment = null;
utilities.lazyLoad(exports, ["PatchDeployment"], () => require("./patchDeployment"));
exports.V2PolicyOrchestrator = null;
utilities.lazyLoad(exports, ["V2PolicyOrchestrator"], () => require("./v2policyOrchestrator"));
exports.V2PolicyOrchestratorForFolder = null;
utilities.lazyLoad(exports, ["V2PolicyOrchestratorForFolder"], () => require("./v2policyOrchestratorForFolder"));
exports.V2PolicyOrchestratorForOrganization = null;
utilities.lazyLoad(exports, ["V2PolicyOrchestratorForOrganization"], () => require("./v2policyOrchestratorForOrganization"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:osconfig/guestPolicies:GuestPolicies":
                return new exports.GuestPolicies(name, undefined, { urn });
            case "gcp:osconfig/osPolicyAssignment:OsPolicyAssignment":
                return new exports.OsPolicyAssignment(name, undefined, { urn });
            case "gcp:osconfig/patchDeployment:PatchDeployment":
                return new exports.PatchDeployment(name, undefined, { urn });
            case "gcp:osconfig/v2PolicyOrchestrator:V2PolicyOrchestrator":
                return new exports.V2PolicyOrchestrator(name, undefined, { urn });
            case "gcp:osconfig/v2PolicyOrchestratorForFolder:V2PolicyOrchestratorForFolder":
                return new exports.V2PolicyOrchestratorForFolder(name, undefined, { urn });
            case "gcp:osconfig/v2PolicyOrchestratorForOrganization:V2PolicyOrchestratorForOrganization":
                return new exports.V2PolicyOrchestratorForOrganization(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "osconfig/guestPolicies", _module);
pulumi.runtime.registerResourceModule("gcp", "osconfig/osPolicyAssignment", _module);
pulumi.runtime.registerResourceModule("gcp", "osconfig/patchDeployment", _module);
pulumi.runtime.registerResourceModule("gcp", "osconfig/v2PolicyOrchestrator", _module);
pulumi.runtime.registerResourceModule("gcp", "osconfig/v2PolicyOrchestratorForFolder", _module);
pulumi.runtime.registerResourceModule("gcp", "osconfig/v2PolicyOrchestratorForOrganization", _module);
