// @pulumi/aws/servicequotas/autoManagement.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoManagement = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages AWS Service Quotas Automatic Management.
 *
 * > **Note:** Due to AWS API limitations, the `notificationArn` attribute cannot be removed once set without recreating the resource. Removing this value from your configuration will trigger resource replacement.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleNotificationConfiguration = new aws.notifications.NotificationConfiguration("example", {
 *     name: "example",
 *     description: "example configuration",
 * });
 * const example = new aws.servicequotas.AutoManagement("example", {
 *     optInLevel: "ACCOUNT",
 *     optInType: "NotifyOnly",
 *     exclusionList: {
 *         dynamodb: ["L-F98FE922"],
 *     },
 *     notificationArn: exampleNotificationConfiguration.arn,
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Service Quotas Auto Management using the region name. For example:
 *
 * ```sh
 * $ pulumi import aws:servicequotas/autoManagement:AutoManagement example us-west-2
 * ```
 */
class AutoManagement extends pulumi.CustomResource {
    /**
     * Get an existing AutoManagement resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AutoManagement(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:servicequotas/autoManagement:AutoManagement';
    /**
     * Returns true if the given object is an instance of AutoManagement.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AutoManagement.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["exclusionList"] = state?.exclusionList;
            resourceInputs["notificationArn"] = state?.notificationArn;
            resourceInputs["optInLevel"] = state?.optInLevel;
            resourceInputs["optInType"] = state?.optInType;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.optInLevel === undefined && !opts.urn) {
                throw new Error("Missing required property 'optInLevel'");
            }
            if (args?.optInType === undefined && !opts.urn) {
                throw new Error("Missing required property 'optInType'");
            }
            resourceInputs["exclusionList"] = args?.exclusionList;
            resourceInputs["notificationArn"] = args?.notificationArn;
            resourceInputs["optInLevel"] = args?.optInLevel;
            resourceInputs["optInType"] = args?.optInType;
            resourceInputs["region"] = args?.region;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AutoManagement.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AutoManagement = AutoManagement;
