// @pulumi/aws/servicequotas/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateAssociation = exports.Template = exports.ServiceQuota = exports.getTemplatesOutput = exports.getTemplates = exports.getServiceQuotaOutput = exports.getServiceQuota = exports.getServiceOutput = exports.getService = exports.AutoManagement = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AutoManagement = null;
utilities.lazyLoad(exports, ["AutoManagement"], () => require("./autoManagement"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceQuota = null;
exports.getServiceQuotaOutput = null;
utilities.lazyLoad(exports, ["getServiceQuota", "getServiceQuotaOutput"], () => require("./getServiceQuota"));
exports.getTemplates = null;
exports.getTemplatesOutput = null;
utilities.lazyLoad(exports, ["getTemplates", "getTemplatesOutput"], () => require("./getTemplates"));
exports.ServiceQuota = null;
utilities.lazyLoad(exports, ["ServiceQuota"], () => require("./serviceQuota"));
exports.Template = null;
utilities.lazyLoad(exports, ["Template"], () => require("./template"));
exports.TemplateAssociation = null;
utilities.lazyLoad(exports, ["TemplateAssociation"], () => require("./templateAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:servicequotas/autoManagement:AutoManagement":
                return new exports.AutoManagement(name, undefined, { urn });
            case "aws:servicequotas/serviceQuota:ServiceQuota":
                return new exports.ServiceQuota(name, undefined, { urn });
            case "aws:servicequotas/template:Template":
                return new exports.Template(name, undefined, { urn });
            case "aws:servicequotas/templateAssociation:TemplateAssociation":
                return new exports.TemplateAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "servicequotas/autoManagement", _module);
pulumi.runtime.registerResourceModule("aws", "servicequotas/serviceQuota", _module);
pulumi.runtime.registerResourceModule("aws", "servicequotas/template", _module);
pulumi.runtime.registerResourceModule("aws", "servicequotas/templateAssociation", _module);
