// @pulumi/azure-native/sqlvirtualmachine/sqlVirtualMachine.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlVirtualMachine = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * A SQL virtual machine.
 *
 * Uses Azure REST API version 2023-10-01. In version 2.x of the Azure Native provider, it used API version 2022-02-01.
 *
 * Other available API versions: 2022-02-01, 2022-07-01-preview, 2022-08-01-preview, 2023-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sqlvirtualmachine [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SqlVirtualMachine extends pulumi.CustomResource {
    /**
     * Get an existing SqlVirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlVirtualMachine(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:sqlvirtualmachine:SqlVirtualMachine';
    /**
     * Returns true if the given object is an instance of SqlVirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlVirtualMachine.__pulumiType;
    }
    /**
     * Create a SqlVirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assessmentSettings"] = args?.assessmentSettings;
            resourceInputs["autoBackupSettings"] = args?.autoBackupSettings;
            resourceInputs["autoPatchingSettings"] = args ? (args.autoPatchingSettings ? pulumi.output(args.autoPatchingSettings).apply(inputs.sqlvirtualmachine.autoPatchingSettingsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["enableAutomaticUpgrade"] = (args?.enableAutomaticUpgrade) ?? false;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["keyVaultCredentialSettings"] = args?.keyVaultCredentialSettings;
            resourceInputs["leastPrivilegeMode"] = (args?.leastPrivilegeMode) ?? "NotSet";
            resourceInputs["location"] = args?.location;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["serverConfigurationsManagementSettings"] = args?.serverConfigurationsManagementSettings;
            resourceInputs["sqlImageOffer"] = args?.sqlImageOffer;
            resourceInputs["sqlImageSku"] = args?.sqlImageSku;
            resourceInputs["sqlManagement"] = args?.sqlManagement;
            resourceInputs["sqlServerLicenseType"] = args?.sqlServerLicenseType;
            resourceInputs["sqlVirtualMachineGroupResourceId"] = args?.sqlVirtualMachineGroupResourceId;
            resourceInputs["sqlVirtualMachineName"] = args?.sqlVirtualMachineName;
            resourceInputs["storageConfigurationSettings"] = args ? (args.storageConfigurationSettings ? pulumi.output(args.storageConfigurationSettings).apply(inputs.sqlvirtualmachine.storageConfigurationSettingsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["virtualMachineIdentitySettings"] = args?.virtualMachineIdentitySettings;
            resourceInputs["virtualMachineResourceId"] = args?.virtualMachineResourceId;
            resourceInputs["wsfcDomainCredentials"] = args?.wsfcDomainCredentials;
            resourceInputs["wsfcStaticIp"] = args?.wsfcStaticIp;
            resourceInputs["additionalVmPatch"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["troubleshootingStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalVmPatch"] = undefined /*out*/;
            resourceInputs["assessmentSettings"] = undefined /*out*/;
            resourceInputs["autoBackupSettings"] = undefined /*out*/;
            resourceInputs["autoPatchingSettings"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enableAutomaticUpgrade"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["keyVaultCredentialSettings"] = undefined /*out*/;
            resourceInputs["leastPrivilegeMode"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serverConfigurationsManagementSettings"] = undefined /*out*/;
            resourceInputs["sqlImageOffer"] = undefined /*out*/;
            resourceInputs["sqlImageSku"] = undefined /*out*/;
            resourceInputs["sqlManagement"] = undefined /*out*/;
            resourceInputs["sqlServerLicenseType"] = undefined /*out*/;
            resourceInputs["sqlVirtualMachineGroupResourceId"] = undefined /*out*/;
            resourceInputs["storageConfigurationSettings"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["troubleshootingStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualMachineIdentitySettings"] = undefined /*out*/;
            resourceInputs["virtualMachineResourceId"] = undefined /*out*/;
            resourceInputs["wsfcDomainCredentials"] = undefined /*out*/;
            resourceInputs["wsfcStaticIp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sqlvirtualmachine/v20170301preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20211101preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20220201:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20220201preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20220701preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20220801preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20230101preview:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20231001:SqlVirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlVirtualMachine.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlVirtualMachine = SqlVirtualMachine;
