// @pulumi/aws/dax/cluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a DAX Cluster resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const bar = new aws.dax.Cluster("bar", {
 *     clusterName: "cluster-example",
 *     iamRoleArn: example.arn,
 *     nodeType: "dax.r4.large",
 *     replicationFactor: 1,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import DAX Clusters using the `clusterName`. For example:
 *
 * ```sh
 * $ pulumi import aws:dax/cluster:Cluster my_cluster my_cluster
 * ```
 *
 * [1]: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DAX.concepts.cluster.html#DAX.concepts.nodes
 */
class Cluster extends pulumi.CustomResource {
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Cluster(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:dax/cluster:Cluster';
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["availabilityZones"] = state?.availabilityZones;
            resourceInputs["clusterAddress"] = state?.clusterAddress;
            resourceInputs["clusterEndpointEncryptionType"] = state?.clusterEndpointEncryptionType;
            resourceInputs["clusterName"] = state?.clusterName;
            resourceInputs["configurationEndpoint"] = state?.configurationEndpoint;
            resourceInputs["description"] = state?.description;
            resourceInputs["iamRoleArn"] = state?.iamRoleArn;
            resourceInputs["maintenanceWindow"] = state?.maintenanceWindow;
            resourceInputs["nodeType"] = state?.nodeType;
            resourceInputs["nodes"] = state?.nodes;
            resourceInputs["notificationTopicArn"] = state?.notificationTopicArn;
            resourceInputs["parameterGroupName"] = state?.parameterGroupName;
            resourceInputs["port"] = state?.port;
            resourceInputs["region"] = state?.region;
            resourceInputs["replicationFactor"] = state?.replicationFactor;
            resourceInputs["securityGroupIds"] = state?.securityGroupIds;
            resourceInputs["serverSideEncryption"] = state?.serverSideEncryption;
            resourceInputs["subnetGroupName"] = state?.subnetGroupName;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.iamRoleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'iamRoleArn'");
            }
            if (args?.nodeType === undefined && !opts.urn) {
                throw new Error("Missing required property 'nodeType'");
            }
            if (args?.replicationFactor === undefined && !opts.urn) {
                throw new Error("Missing required property 'replicationFactor'");
            }
            resourceInputs["availabilityZones"] = args?.availabilityZones;
            resourceInputs["clusterEndpointEncryptionType"] = args?.clusterEndpointEncryptionType;
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["description"] = args?.description;
            resourceInputs["iamRoleArn"] = args?.iamRoleArn;
            resourceInputs["maintenanceWindow"] = args?.maintenanceWindow;
            resourceInputs["nodeType"] = args?.nodeType;
            resourceInputs["notificationTopicArn"] = args?.notificationTopicArn;
            resourceInputs["parameterGroupName"] = args?.parameterGroupName;
            resourceInputs["region"] = args?.region;
            resourceInputs["replicationFactor"] = args?.replicationFactor;
            resourceInputs["securityGroupIds"] = args?.securityGroupIds;
            resourceInputs["serverSideEncryption"] = args?.serverSideEncryption;
            resourceInputs["subnetGroupName"] = args?.subnetGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["clusterAddress"] = undefined /*out*/;
            resourceInputs["configurationEndpoint"] = undefined /*out*/;
            resourceInputs["nodes"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Cluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Cluster = Cluster;
