// @pulumi/gcp/bigqueryanalyticshub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingSubscription = exports.ListingIamPolicy = exports.ListingIamMember = exports.ListingIamBinding = exports.Listing = exports.getListingIamPolicyOutput = exports.getListingIamPolicy = exports.getDataExchangeIamPolicyOutput = exports.getDataExchangeIamPolicy = exports.DataExchangeSubscription = exports.DataExchangeIamPolicy = exports.DataExchangeIamMember = exports.DataExchangeIamBinding = exports.DataExchange = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.DataExchange = null;
utilities.lazyLoad(exports, ["DataExchange"], () => require("./dataExchange"));
exports.DataExchangeIamBinding = null;
utilities.lazyLoad(exports, ["DataExchangeIamBinding"], () => require("./dataExchangeIamBinding"));
exports.DataExchangeIamMember = null;
utilities.lazyLoad(exports, ["DataExchangeIamMember"], () => require("./dataExchangeIamMember"));
exports.DataExchangeIamPolicy = null;
utilities.lazyLoad(exports, ["DataExchangeIamPolicy"], () => require("./dataExchangeIamPolicy"));
exports.DataExchangeSubscription = null;
utilities.lazyLoad(exports, ["DataExchangeSubscription"], () => require("./dataExchangeSubscription"));
exports.getDataExchangeIamPolicy = null;
exports.getDataExchangeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getDataExchangeIamPolicy", "getDataExchangeIamPolicyOutput"], () => require("./getDataExchangeIamPolicy"));
exports.getListingIamPolicy = null;
exports.getListingIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getListingIamPolicy", "getListingIamPolicyOutput"], () => require("./getListingIamPolicy"));
exports.Listing = null;
utilities.lazyLoad(exports, ["Listing"], () => require("./listing"));
exports.ListingIamBinding = null;
utilities.lazyLoad(exports, ["ListingIamBinding"], () => require("./listingIamBinding"));
exports.ListingIamMember = null;
utilities.lazyLoad(exports, ["ListingIamMember"], () => require("./listingIamMember"));
exports.ListingIamPolicy = null;
utilities.lazyLoad(exports, ["ListingIamPolicy"], () => require("./listingIamPolicy"));
exports.ListingSubscription = null;
utilities.lazyLoad(exports, ["ListingSubscription"], () => require("./listingSubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigqueryanalyticshub/dataExchange:DataExchange":
                return new exports.DataExchange(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamBinding:DataExchangeIamBinding":
                return new exports.DataExchangeIamBinding(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamMember:DataExchangeIamMember":
                return new exports.DataExchangeIamMember(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamPolicy:DataExchangeIamPolicy":
                return new exports.DataExchangeIamPolicy(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeSubscription:DataExchangeSubscription":
                return new exports.DataExchangeSubscription(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listing:Listing":
                return new exports.Listing(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamBinding:ListingIamBinding":
                return new exports.ListingIamBinding(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamMember:ListingIamMember":
                return new exports.ListingIamMember(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamPolicy:ListingIamPolicy":
                return new exports.ListingIamPolicy(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingSubscription:ListingSubscription":
                return new exports.ListingSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchange", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeSubscription", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listing", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingSubscription", _module);
