// @pulumi/gcp/cloudidentity/getPolicies.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPoliciesOutput = exports.getPolicies = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Use this data source to list Cloud Identity policies.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const all = gcp.cloudidentity.getPolicies({});
 * export const firstPolicyName = all.then(all => all.policies?.[0]?.name);
 * export const firstPolicyCustomer = all.then(all => all.policies?.[0]?.customer);
 * export const policyQueryQuery = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.query);
 * export const policyQueryOrgUnit = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.orgUnit);
 * export const policyQueryGroup = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.group);
 * export const policyQuerySortOrder = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.sortOrder);
 * export const policySetting = all.then(all => all.policies?.[0]?.setting);
 * export const policyType = all.then(all => all.policies?.[0]?.type);
 * ```
 */
function getPolicies(args, opts) {
    args = args || {};
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invoke("gcp:cloudidentity/getPolicies:getPolicies", {
        "filter": args.filter,
    }, opts);
}
exports.getPolicies = getPolicies;
/**
 * Use this data source to list Cloud Identity policies.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const all = gcp.cloudidentity.getPolicies({});
 * export const firstPolicyName = all.then(all => all.policies?.[0]?.name);
 * export const firstPolicyCustomer = all.then(all => all.policies?.[0]?.customer);
 * export const policyQueryQuery = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.query);
 * export const policyQueryOrgUnit = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.orgUnit);
 * export const policyQueryGroup = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.group);
 * export const policyQuerySortOrder = all.then(all => all.policies?.[0]?.policyQueries?.[0]?.sortOrder);
 * export const policySetting = all.then(all => all.policies?.[0]?.setting);
 * export const policyType = all.then(all => all.policies?.[0]?.type);
 * ```
 */
function getPoliciesOutput(args, opts) {
    args = args || {};
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invokeOutput("gcp:cloudidentity/getPolicies:getPolicies", {
        "filter": args.filter,
    }, opts);
}
exports.getPoliciesOutput = getPoliciesOutput;
