// @pulumi/gcp/dataform/teamFolder.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamFolder = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A resource represents a Dataform TeamFolder
 *
 * To get more information about TeamFolder, see:
 *
 * * [API documentation](https://cloud.google.com/dataform/reference/rest/v1/projects.locations.teamFolders)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/dataform/docs/)
 *
 * ## Example Usage
 *
 * ### Dataform Team Folder Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const dataformTeamFolderBasic = new gcp.dataform.TeamFolder("dataform_team_folder_basic", {
 *     region: "us-central1",
 *     displayName: "Basic TeamFolder-_34535",
 * });
 * ```
 *
 * ## Import
 *
 * TeamFolder can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{region}}/teamFolders/{{teamfolder_id}}`
 * * `{{project}}/{{region}}/{{teamfolder_id}}`
 * * `{{region}}/{{teamfolder_id}}`
 * * `{{teamfolder_id}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, TeamFolder can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataform/teamFolder:TeamFolder default projects/{{project}}/locations/{{region}}/teamFolders/{{teamfolder_id}}
 * $ pulumi import gcp:dataform/teamFolder:TeamFolder default {{project}}/{{region}}/{{teamfolder_id}}
 * $ pulumi import gcp:dataform/teamFolder:TeamFolder default {{region}}/{{teamfolder_id}}
 * $ pulumi import gcp:dataform/teamFolder:TeamFolder default {{teamfolder_id}}
 * $ pulumi import gcp:dataform/teamFolder:TeamFolder default {{name}}
 * ```
 */
class TeamFolder extends pulumi.CustomResource {
    /**
     * Get an existing TeamFolder resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TeamFolder(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:dataform/teamFolder:TeamFolder';
    /**
     * Returns true if the given object is an instance of TeamFolder.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TeamFolder.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["region"] = state?.region;
            resourceInputs["teamfolderId"] = state?.teamfolderId;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.region === undefined && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["project"] = args?.project;
            resourceInputs["region"] = args?.region;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["teamfolderId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TeamFolder.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TeamFolder = TeamFolder;
