// @pulumi/aws/opensearch/application.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an AWS OpenSearch Application resource. OpenSearch Applications provide a user interface for interacting with OpenSearch data and managing OpenSearch resources.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.opensearch.Application("example", {name: "my-opensearch-app"});
 * ```
 *
 * ### Application with Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.opensearch.Application("example", {
 *     name: "my-opensearch-app",
 *     appConfigs: [
 *         {
 *             key: "opensearchDashboards.dashboardAdmin.users",
 *             value: "admin-user",
 *         },
 *         {
 *             key: "opensearchDashboards.dashboardAdmin.groups",
 *             value: "admin-group",
 *         },
 *     ],
 *     tags: {
 *         Environment: "production",
 *         Team: "data-platform",
 *     },
 * });
 * ```
 *
 * ### Application with Data Sources
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.opensearch.Domain("example", {
 *     domainName: "example-domain",
 *     engineVersion: "OpenSearch_2.3",
 *     clusterConfig: {
 *         instanceType: "t3.small.search",
 *     },
 *     ebsOptions: {
 *         ebsEnabled: true,
 *         volumeSize: 20,
 *     },
 * });
 * const exampleApplication = new aws.opensearch.Application("example", {
 *     name: "my-opensearch-app",
 *     dataSources: [{
 *         dataSourceArn: example.arn,
 *         dataSourceDescription: "Primary OpenSearch domain for analytics",
 *     }],
 *     tags: {
 *         Environment: "production",
 *     },
 * });
 * ```
 *
 * ### Application with IAM Identity Center Integration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * // Data sources for account and region information
 * const example = aws.ssoadmin.getInstances({});
 * const current = aws.getCallerIdentity({});
 * const currentGetRegion = aws.getRegion({});
 * // IAM Policy for OpenSearch Application Identity Center Integration
 * const opensearchIdentityCenter = new aws.iam.Policy("opensearch_identity_center", {
 *     name: "opensearch-identity-center-policy",
 *     description: "Policy for OpenSearch Application Identity Center integration",
 *     policy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [
 *             {
 *                 Sid: "IdentityStoreOpenSearchDomainConnectivity",
 *                 Effect: "Allow",
 *                 Action: [
 *                     "identitystore:DescribeUser",
 *                     "identitystore:ListGroupMembershipsForMember",
 *                     "identitystore:DescribeGroup",
 *                 ],
 *                 Resource: "*",
 *                 Condition: {
 *                     "ForAnyValue:StringEquals": {
 *                         "aws:CalledViaLast": "es.amazonaws.com",
 *                     },
 *                 },
 *             },
 *             {
 *                 Sid: "OpenSearchDomain",
 *                 Effect: "Allow",
 *                 Action: ["es:ESHttp*"],
 *                 Resource: "*",
 *             },
 *             {
 *                 Sid: "OpenSearchServerless",
 *                 Effect: "Allow",
 *                 Action: ["aoss:APIAccessAll"],
 *                 Resource: "*",
 *             },
 *         ],
 *     }),
 * });
 * // IAM Role for OpenSearch Application
 * const opensearchApplication = new aws.iam.Role("opensearch_application", {
 *     name: "opensearch-application-role",
 *     assumeRolePolicy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [
 *             {
 *                 Effect: "Allow",
 *                 Principal: {
 *                     Service: "application.opensearchservice.amazonaws.com",
 *                 },
 *                 Action: "sts:AssumeRole",
 *             },
 *             {
 *                 Effect: "Allow",
 *                 Principal: {
 *                     Service: "application.opensearchservice.amazonaws.com",
 *                 },
 *                 Action: "sts:SetContext",
 *                 Condition: {
 *                     "ForAllValues:ArnEquals": {
 *                         "sts:RequestContextProviders": Promise.all([current, currentGetRegion]).then(([current, currentGetRegion]) => `arn:aws:iam::${current.accountId}:oidc-provider/portal.sso.${currentGetRegion.id}.amazonaws.com/apl/*`),
 *                     },
 *                 },
 *             },
 *         ],
 *     }),
 * });
 * // Attach policy to role
 * const opensearchIdentityCenterRolePolicyAttachment = new aws.iam.RolePolicyAttachment("opensearch_identity_center", {
 *     role: opensearchApplication.name,
 *     policyArn: opensearchIdentityCenter.arn,
 * });
 * const exampleApplication = new aws.opensearch.Application("example", {
 *     name: "my-opensearch-app",
 *     iamIdentityCenterOptions: {
 *         enabled: true,
 *         iamIdentityCenterInstanceArn: example.then(example => example.arns?.[0]),
 *         iamRoleForIdentityCenterApplicationArn: opensearchApplication.arn,
 *     },
 *     tags: {
 *         Environment: "production",
 *     },
 * });
 * ```
 *
 * ## Additional Information
 *
 * For more information about OpenSearch Applications, see the [AWS OpenSearch Service Developer Guide](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/application.html).
 *
 * For information about configuring IAM Identity Center with OpenSearch Applications, see [Using AWS IAM Identity Center authentication](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/application-getting-started.html#create-application).
 *
 * ## Import
 *
 * Using `pulumi import`, import OpenSearch applications using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:opensearch/application:Application example app-1234567890abcdef0
 * ```
 */
class Application extends pulumi.CustomResource {
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Application(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:opensearch/application:Application';
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appConfigs"] = state?.appConfigs;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["dataSources"] = state?.dataSources;
            resourceInputs["endpoint"] = state?.endpoint;
            resourceInputs["iamIdentityCenterOptions"] = state?.iamIdentityCenterOptions;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            resourceInputs["appConfigs"] = args?.appConfigs;
            resourceInputs["dataSources"] = args?.dataSources;
            resourceInputs["iamIdentityCenterOptions"] = args?.iamIdentityCenterOptions;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Application = Application;
