// @pulumi/aws/opensearch/authorizeVpcEndpointAccess.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizeVpcEndpointAccess = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS OpenSearch Authorize Vpc Endpoint Access.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const current = aws.getCallerIdentity({});
 * const test = new aws.opensearch.AuthorizeVpcEndpointAccess("test", {
 *     domainName: testAwsOpensearchDomain.domainName,
 *     account: current.then(current => current.accountId),
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import OpenSearch Authorize Vpc Endpoint Access using the `domainName` and `account` separated by a comma (,). For example:
 *
 * ```sh
 * $ pulumi import aws:opensearch/authorizeVpcEndpointAccess:AuthorizeVpcEndpointAccess example authorize_vpc_endpoint_access-id-12345678,123456789012
 * ```
 */
class AuthorizeVpcEndpointAccess extends pulumi.CustomResource {
    /**
     * Get an existing AuthorizeVpcEndpointAccess resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AuthorizeVpcEndpointAccess(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:opensearch/authorizeVpcEndpointAccess:AuthorizeVpcEndpointAccess';
    /**
     * Returns true if the given object is an instance of AuthorizeVpcEndpointAccess.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AuthorizeVpcEndpointAccess.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["account"] = state?.account;
            resourceInputs["authorizedPrincipals"] = state?.authorizedPrincipals;
            resourceInputs["domainName"] = state?.domainName;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.account === undefined && !opts.urn) {
                throw new Error("Missing required property 'account'");
            }
            if (args?.domainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            resourceInputs["account"] = args?.account;
            resourceInputs["domainName"] = args?.domainName;
            resourceInputs["region"] = args?.region;
            resourceInputs["authorizedPrincipals"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AuthorizeVpcEndpointAccess.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AuthorizeVpcEndpointAccess = AuthorizeVpcEndpointAccess;
