// @pulumi/aws/opensearch/package.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Package = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS Opensearch Package.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as std from "@pulumi/std";
 *
 * const myOpensearchPackages = new aws.s3.Bucket("my_opensearch_packages", {bucket: "my-opensearch-packages"});
 * const example = new aws.s3.BucketObjectv2("example", {
 *     bucket: myOpensearchPackages.bucket,
 *     key: "example.txt",
 *     source: new pulumi.asset.FileAsset("./example.txt"),
 *     etag: std.filemd5({
 *         input: "./example.txt",
 *     }).then(invoke => invoke.result),
 * });
 * const examplePackage = new aws.opensearch.Package("example", {
 *     packageName: "example-txt",
 *     packageSource: {
 *         s3BucketName: myOpensearchPackages.bucket,
 *         s3Key: example.key,
 *     },
 *     packageType: "TXT-DICTIONARY",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import AWS Opensearch Packages using the Package ID. For example:
 *
 * ```sh
 * $ pulumi import aws:opensearch/package:Package example package-id
 * ```
 */
class Package extends pulumi.CustomResource {
    /**
     * Get an existing Package resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Package(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:opensearch/package:Package';
    /**
     * Returns true if the given object is an instance of Package.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Package.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["availablePackageVersion"] = state?.availablePackageVersion;
            resourceInputs["engineVersion"] = state?.engineVersion;
            resourceInputs["packageDescription"] = state?.packageDescription;
            resourceInputs["packageId"] = state?.packageId;
            resourceInputs["packageName"] = state?.packageName;
            resourceInputs["packageSource"] = state?.packageSource;
            resourceInputs["packageType"] = state?.packageType;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.packageName === undefined && !opts.urn) {
                throw new Error("Missing required property 'packageName'");
            }
            if (args?.packageSource === undefined && !opts.urn) {
                throw new Error("Missing required property 'packageSource'");
            }
            if (args?.packageType === undefined && !opts.urn) {
                throw new Error("Missing required property 'packageType'");
            }
            resourceInputs["engineVersion"] = args?.engineVersion;
            resourceInputs["packageDescription"] = args?.packageDescription;
            resourceInputs["packageName"] = args?.packageName;
            resourceInputs["packageSource"] = args?.packageSource;
            resourceInputs["packageType"] = args?.packageType;
            resourceInputs["region"] = args?.region;
            resourceInputs["availablePackageVersion"] = undefined /*out*/;
            resourceInputs["packageId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Package.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Package = Package;
