// @pulumi/aws/opensearch/serverlessCollectionGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerlessCollectionGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS OpenSearch Serverless Collection Group.
 *
 * Collection groups let multiple OpenSearch Serverless collections share compute resources while keeping encryption and access controls independent.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.opensearch.ServerlessCollectionGroup("example", {
 *     name: "example-group",
 *     description: "Shared compute for production collections",
 *     standbyReplicas: "ENABLED",
 *     capacityLimits: [{
 *         minIndexingCapacityInOcu: 2,
 *         maxIndexingCapacityInOcu: 16,
 *         minSearchCapacityInOcu: 2,
 *         maxSearchCapacityInOcu: 16,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `id` (String) Unique identifier for the collection group.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import OpenSearch Serverless Collection Group using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:opensearch/serverlessCollectionGroup:ServerlessCollectionGroup example example-group-id
 * ```
 */
class ServerlessCollectionGroup extends pulumi.CustomResource {
    /**
     * Get an existing ServerlessCollectionGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServerlessCollectionGroup(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:opensearch/serverlessCollectionGroup:ServerlessCollectionGroup';
    /**
     * Returns true if the given object is an instance of ServerlessCollectionGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerlessCollectionGroup.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["capacityLimits"] = state?.capacityLimits;
            resourceInputs["createdDate"] = state?.createdDate;
            resourceInputs["description"] = state?.description;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["standbyReplicas"] = state?.standbyReplicas;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.standbyReplicas === undefined && !opts.urn) {
                throw new Error("Missing required property 'standbyReplicas'");
            }
            resourceInputs["capacityLimits"] = args?.capacityLimits;
            resourceInputs["description"] = args?.description;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["standbyReplicas"] = args?.standbyReplicas;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServerlessCollectionGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServerlessCollectionGroup = ServerlessCollectionGroup;
