// @pulumi/gcp/iap/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TunnelDestGroupIamMember = exports.TunnelDestGroupIamBinding = exports.TunnelDestGroup = exports.Settings = exports.LocationWebIamPolicy = exports.LocationWebIamMember = exports.LocationWebIamBinding = exports.getWebTypeComputeIamPolicyOutput = exports.getWebTypeComputeIamPolicy = exports.getWebTypeAppEngineIamPolicyOutput = exports.getWebTypeAppEngineIamPolicy = exports.getWebRegionForwardingRuleServiceIamPolicyOutput = exports.getWebRegionForwardingRuleServiceIamPolicy = exports.getWebRegionBackendServiceIamPolicyOutput = exports.getWebRegionBackendServiceIamPolicy = exports.getWebIamPolicyOutput = exports.getWebIamPolicy = exports.getWebForwardingRuleServiceIamPolicyOutput = exports.getWebForwardingRuleServiceIamPolicy = exports.getWebCloudRunServiceIamPolicyOutput = exports.getWebCloudRunServiceIamPolicy = exports.getWebBackendServiceIamPolicyOutput = exports.getWebBackendServiceIamPolicy = exports.getTunnelInstanceIamPolicyOutput = exports.getTunnelInstanceIamPolicy = exports.getTunnelIamPolicyOutput = exports.getTunnelIamPolicy = exports.getTunnelDestGroupIamPolicyOutput = exports.getTunnelDestGroupIamPolicy = exports.getLocationWebIamPolicyOutput = exports.getLocationWebIamPolicy = exports.getClientOutput = exports.getClient = exports.getAppEngineVersionIamPolicyOutput = exports.getAppEngineVersionIamPolicy = exports.getAppEngineServiceIamPolicyOutput = exports.getAppEngineServiceIamPolicy = exports.getAgentRegistryIamPolicyOutput = exports.getAgentRegistryIamPolicy = exports.Client = exports.Brand = exports.AppEngineVersionIamPolicy = exports.AppEngineVersionIamMember = exports.AppEngineVersionIamBinding = exports.AppEngineServiceIamPolicy = exports.AppEngineServiceIamMember = exports.AppEngineServiceIamBinding = exports.AgentRegistryIamPolicy = exports.AgentRegistryIamMember = exports.AgentRegistryIamBinding = void 0;
exports.WebTypeComputeIamPolicy = exports.WebTypeComputeIamMember = exports.WebTypeComputeIamBinding = exports.WebTypeAppEngingIamPolicy = exports.WebTypeAppEngingIamMember = exports.WebTypeAppEngingIamBinding = exports.WebRegionForwardingRuleServiceIamPolicy = exports.WebRegionForwardingRuleServiceIamMember = exports.WebRegionForwardingRuleServiceIamBinding = exports.WebRegionBackendServiceIamPolicy = exports.WebRegionBackendServiceIamMember = exports.WebRegionBackendServiceIamBinding = exports.WebIamPolicy = exports.WebIamMember = exports.WebIamBinding = exports.WebForwardingRuleServiceIamPolicy = exports.WebForwardingRuleServiceIamMember = exports.WebForwardingRuleServiceIamBinding = exports.WebCloudRunServiceIamPolicy = exports.WebCloudRunServiceIamMember = exports.WebCloudRunServiceIamBinding = exports.WebBackendServiceIamPolicy = exports.WebBackendServiceIamMember = exports.WebBackendServiceIamBinding = exports.TunnelInstanceIAMPolicy = exports.TunnelInstanceIAMMember = exports.TunnelInstanceIAMBinding = exports.TunnelIamPolicy = exports.TunnelIamMember = exports.TunnelIamBinding = exports.TunnelDestGroupIamPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AgentRegistryIamBinding = null;
utilities.lazyLoad(exports, ["AgentRegistryIamBinding"], () => require("./agentRegistryIamBinding"));
exports.AgentRegistryIamMember = null;
utilities.lazyLoad(exports, ["AgentRegistryIamMember"], () => require("./agentRegistryIamMember"));
exports.AgentRegistryIamPolicy = null;
utilities.lazyLoad(exports, ["AgentRegistryIamPolicy"], () => require("./agentRegistryIamPolicy"));
exports.AppEngineServiceIamBinding = null;
utilities.lazyLoad(exports, ["AppEngineServiceIamBinding"], () => require("./appEngineServiceIamBinding"));
exports.AppEngineServiceIamMember = null;
utilities.lazyLoad(exports, ["AppEngineServiceIamMember"], () => require("./appEngineServiceIamMember"));
exports.AppEngineServiceIamPolicy = null;
utilities.lazyLoad(exports, ["AppEngineServiceIamPolicy"], () => require("./appEngineServiceIamPolicy"));
exports.AppEngineVersionIamBinding = null;
utilities.lazyLoad(exports, ["AppEngineVersionIamBinding"], () => require("./appEngineVersionIamBinding"));
exports.AppEngineVersionIamMember = null;
utilities.lazyLoad(exports, ["AppEngineVersionIamMember"], () => require("./appEngineVersionIamMember"));
exports.AppEngineVersionIamPolicy = null;
utilities.lazyLoad(exports, ["AppEngineVersionIamPolicy"], () => require("./appEngineVersionIamPolicy"));
exports.Brand = null;
utilities.lazyLoad(exports, ["Brand"], () => require("./brand"));
exports.Client = null;
utilities.lazyLoad(exports, ["Client"], () => require("./client"));
exports.getAgentRegistryIamPolicy = null;
exports.getAgentRegistryIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAgentRegistryIamPolicy", "getAgentRegistryIamPolicyOutput"], () => require("./getAgentRegistryIamPolicy"));
exports.getAppEngineServiceIamPolicy = null;
exports.getAppEngineServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAppEngineServiceIamPolicy", "getAppEngineServiceIamPolicyOutput"], () => require("./getAppEngineServiceIamPolicy"));
exports.getAppEngineVersionIamPolicy = null;
exports.getAppEngineVersionIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAppEngineVersionIamPolicy", "getAppEngineVersionIamPolicyOutput"], () => require("./getAppEngineVersionIamPolicy"));
exports.getClient = null;
exports.getClientOutput = null;
utilities.lazyLoad(exports, ["getClient", "getClientOutput"], () => require("./getClient"));
exports.getLocationWebIamPolicy = null;
exports.getLocationWebIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getLocationWebIamPolicy", "getLocationWebIamPolicyOutput"], () => require("./getLocationWebIamPolicy"));
exports.getTunnelDestGroupIamPolicy = null;
exports.getTunnelDestGroupIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTunnelDestGroupIamPolicy", "getTunnelDestGroupIamPolicyOutput"], () => require("./getTunnelDestGroupIamPolicy"));
exports.getTunnelIamPolicy = null;
exports.getTunnelIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTunnelIamPolicy", "getTunnelIamPolicyOutput"], () => require("./getTunnelIamPolicy"));
exports.getTunnelInstanceIamPolicy = null;
exports.getTunnelInstanceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTunnelInstanceIamPolicy", "getTunnelInstanceIamPolicyOutput"], () => require("./getTunnelInstanceIamPolicy"));
exports.getWebBackendServiceIamPolicy = null;
exports.getWebBackendServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebBackendServiceIamPolicy", "getWebBackendServiceIamPolicyOutput"], () => require("./getWebBackendServiceIamPolicy"));
exports.getWebCloudRunServiceIamPolicy = null;
exports.getWebCloudRunServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebCloudRunServiceIamPolicy", "getWebCloudRunServiceIamPolicyOutput"], () => require("./getWebCloudRunServiceIamPolicy"));
exports.getWebForwardingRuleServiceIamPolicy = null;
exports.getWebForwardingRuleServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebForwardingRuleServiceIamPolicy", "getWebForwardingRuleServiceIamPolicyOutput"], () => require("./getWebForwardingRuleServiceIamPolicy"));
exports.getWebIamPolicy = null;
exports.getWebIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebIamPolicy", "getWebIamPolicyOutput"], () => require("./getWebIamPolicy"));
exports.getWebRegionBackendServiceIamPolicy = null;
exports.getWebRegionBackendServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebRegionBackendServiceIamPolicy", "getWebRegionBackendServiceIamPolicyOutput"], () => require("./getWebRegionBackendServiceIamPolicy"));
exports.getWebRegionForwardingRuleServiceIamPolicy = null;
exports.getWebRegionForwardingRuleServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebRegionForwardingRuleServiceIamPolicy", "getWebRegionForwardingRuleServiceIamPolicyOutput"], () => require("./getWebRegionForwardingRuleServiceIamPolicy"));
exports.getWebTypeAppEngineIamPolicy = null;
exports.getWebTypeAppEngineIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebTypeAppEngineIamPolicy", "getWebTypeAppEngineIamPolicyOutput"], () => require("./getWebTypeAppEngineIamPolicy"));
exports.getWebTypeComputeIamPolicy = null;
exports.getWebTypeComputeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWebTypeComputeIamPolicy", "getWebTypeComputeIamPolicyOutput"], () => require("./getWebTypeComputeIamPolicy"));
exports.LocationWebIamBinding = null;
utilities.lazyLoad(exports, ["LocationWebIamBinding"], () => require("./locationWebIamBinding"));
exports.LocationWebIamMember = null;
utilities.lazyLoad(exports, ["LocationWebIamMember"], () => require("./locationWebIamMember"));
exports.LocationWebIamPolicy = null;
utilities.lazyLoad(exports, ["LocationWebIamPolicy"], () => require("./locationWebIamPolicy"));
exports.Settings = null;
utilities.lazyLoad(exports, ["Settings"], () => require("./settings"));
exports.TunnelDestGroup = null;
utilities.lazyLoad(exports, ["TunnelDestGroup"], () => require("./tunnelDestGroup"));
exports.TunnelDestGroupIamBinding = null;
utilities.lazyLoad(exports, ["TunnelDestGroupIamBinding"], () => require("./tunnelDestGroupIamBinding"));
exports.TunnelDestGroupIamMember = null;
utilities.lazyLoad(exports, ["TunnelDestGroupIamMember"], () => require("./tunnelDestGroupIamMember"));
exports.TunnelDestGroupIamPolicy = null;
utilities.lazyLoad(exports, ["TunnelDestGroupIamPolicy"], () => require("./tunnelDestGroupIamPolicy"));
exports.TunnelIamBinding = null;
utilities.lazyLoad(exports, ["TunnelIamBinding"], () => require("./tunnelIamBinding"));
exports.TunnelIamMember = null;
utilities.lazyLoad(exports, ["TunnelIamMember"], () => require("./tunnelIamMember"));
exports.TunnelIamPolicy = null;
utilities.lazyLoad(exports, ["TunnelIamPolicy"], () => require("./tunnelIamPolicy"));
exports.TunnelInstanceIAMBinding = null;
utilities.lazyLoad(exports, ["TunnelInstanceIAMBinding"], () => require("./tunnelInstanceIAMBinding"));
exports.TunnelInstanceIAMMember = null;
utilities.lazyLoad(exports, ["TunnelInstanceIAMMember"], () => require("./tunnelInstanceIAMMember"));
exports.TunnelInstanceIAMPolicy = null;
utilities.lazyLoad(exports, ["TunnelInstanceIAMPolicy"], () => require("./tunnelInstanceIAMPolicy"));
exports.WebBackendServiceIamBinding = null;
utilities.lazyLoad(exports, ["WebBackendServiceIamBinding"], () => require("./webBackendServiceIamBinding"));
exports.WebBackendServiceIamMember = null;
utilities.lazyLoad(exports, ["WebBackendServiceIamMember"], () => require("./webBackendServiceIamMember"));
exports.WebBackendServiceIamPolicy = null;
utilities.lazyLoad(exports, ["WebBackendServiceIamPolicy"], () => require("./webBackendServiceIamPolicy"));
exports.WebCloudRunServiceIamBinding = null;
utilities.lazyLoad(exports, ["WebCloudRunServiceIamBinding"], () => require("./webCloudRunServiceIamBinding"));
exports.WebCloudRunServiceIamMember = null;
utilities.lazyLoad(exports, ["WebCloudRunServiceIamMember"], () => require("./webCloudRunServiceIamMember"));
exports.WebCloudRunServiceIamPolicy = null;
utilities.lazyLoad(exports, ["WebCloudRunServiceIamPolicy"], () => require("./webCloudRunServiceIamPolicy"));
exports.WebForwardingRuleServiceIamBinding = null;
utilities.lazyLoad(exports, ["WebForwardingRuleServiceIamBinding"], () => require("./webForwardingRuleServiceIamBinding"));
exports.WebForwardingRuleServiceIamMember = null;
utilities.lazyLoad(exports, ["WebForwardingRuleServiceIamMember"], () => require("./webForwardingRuleServiceIamMember"));
exports.WebForwardingRuleServiceIamPolicy = null;
utilities.lazyLoad(exports, ["WebForwardingRuleServiceIamPolicy"], () => require("./webForwardingRuleServiceIamPolicy"));
exports.WebIamBinding = null;
utilities.lazyLoad(exports, ["WebIamBinding"], () => require("./webIamBinding"));
exports.WebIamMember = null;
utilities.lazyLoad(exports, ["WebIamMember"], () => require("./webIamMember"));
exports.WebIamPolicy = null;
utilities.lazyLoad(exports, ["WebIamPolicy"], () => require("./webIamPolicy"));
exports.WebRegionBackendServiceIamBinding = null;
utilities.lazyLoad(exports, ["WebRegionBackendServiceIamBinding"], () => require("./webRegionBackendServiceIamBinding"));
exports.WebRegionBackendServiceIamMember = null;
utilities.lazyLoad(exports, ["WebRegionBackendServiceIamMember"], () => require("./webRegionBackendServiceIamMember"));
exports.WebRegionBackendServiceIamPolicy = null;
utilities.lazyLoad(exports, ["WebRegionBackendServiceIamPolicy"], () => require("./webRegionBackendServiceIamPolicy"));
exports.WebRegionForwardingRuleServiceIamBinding = null;
utilities.lazyLoad(exports, ["WebRegionForwardingRuleServiceIamBinding"], () => require("./webRegionForwardingRuleServiceIamBinding"));
exports.WebRegionForwardingRuleServiceIamMember = null;
utilities.lazyLoad(exports, ["WebRegionForwardingRuleServiceIamMember"], () => require("./webRegionForwardingRuleServiceIamMember"));
exports.WebRegionForwardingRuleServiceIamPolicy = null;
utilities.lazyLoad(exports, ["WebRegionForwardingRuleServiceIamPolicy"], () => require("./webRegionForwardingRuleServiceIamPolicy"));
exports.WebTypeAppEngingIamBinding = null;
utilities.lazyLoad(exports, ["WebTypeAppEngingIamBinding"], () => require("./webTypeAppEngingIamBinding"));
exports.WebTypeAppEngingIamMember = null;
utilities.lazyLoad(exports, ["WebTypeAppEngingIamMember"], () => require("./webTypeAppEngingIamMember"));
exports.WebTypeAppEngingIamPolicy = null;
utilities.lazyLoad(exports, ["WebTypeAppEngingIamPolicy"], () => require("./webTypeAppEngingIamPolicy"));
exports.WebTypeComputeIamBinding = null;
utilities.lazyLoad(exports, ["WebTypeComputeIamBinding"], () => require("./webTypeComputeIamBinding"));
exports.WebTypeComputeIamMember = null;
utilities.lazyLoad(exports, ["WebTypeComputeIamMember"], () => require("./webTypeComputeIamMember"));
exports.WebTypeComputeIamPolicy = null;
utilities.lazyLoad(exports, ["WebTypeComputeIamPolicy"], () => require("./webTypeComputeIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:iap/agentRegistryIamBinding:AgentRegistryIamBinding":
                return new exports.AgentRegistryIamBinding(name, undefined, { urn });
            case "gcp:iap/agentRegistryIamMember:AgentRegistryIamMember":
                return new exports.AgentRegistryIamMember(name, undefined, { urn });
            case "gcp:iap/agentRegistryIamPolicy:AgentRegistryIamPolicy":
                return new exports.AgentRegistryIamPolicy(name, undefined, { urn });
            case "gcp:iap/appEngineServiceIamBinding:AppEngineServiceIamBinding":
                return new exports.AppEngineServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/appEngineServiceIamMember:AppEngineServiceIamMember":
                return new exports.AppEngineServiceIamMember(name, undefined, { urn });
            case "gcp:iap/appEngineServiceIamPolicy:AppEngineServiceIamPolicy":
                return new exports.AppEngineServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/appEngineVersionIamBinding:AppEngineVersionIamBinding":
                return new exports.AppEngineVersionIamBinding(name, undefined, { urn });
            case "gcp:iap/appEngineVersionIamMember:AppEngineVersionIamMember":
                return new exports.AppEngineVersionIamMember(name, undefined, { urn });
            case "gcp:iap/appEngineVersionIamPolicy:AppEngineVersionIamPolicy":
                return new exports.AppEngineVersionIamPolicy(name, undefined, { urn });
            case "gcp:iap/brand:Brand":
                return new exports.Brand(name, undefined, { urn });
            case "gcp:iap/client:Client":
                return new exports.Client(name, undefined, { urn });
            case "gcp:iap/locationWebIamBinding:LocationWebIamBinding":
                return new exports.LocationWebIamBinding(name, undefined, { urn });
            case "gcp:iap/locationWebIamMember:LocationWebIamMember":
                return new exports.LocationWebIamMember(name, undefined, { urn });
            case "gcp:iap/locationWebIamPolicy:LocationWebIamPolicy":
                return new exports.LocationWebIamPolicy(name, undefined, { urn });
            case "gcp:iap/settings:Settings":
                return new exports.Settings(name, undefined, { urn });
            case "gcp:iap/tunnelDestGroup:TunnelDestGroup":
                return new exports.TunnelDestGroup(name, undefined, { urn });
            case "gcp:iap/tunnelDestGroupIamBinding:TunnelDestGroupIamBinding":
                return new exports.TunnelDestGroupIamBinding(name, undefined, { urn });
            case "gcp:iap/tunnelDestGroupIamMember:TunnelDestGroupIamMember":
                return new exports.TunnelDestGroupIamMember(name, undefined, { urn });
            case "gcp:iap/tunnelDestGroupIamPolicy:TunnelDestGroupIamPolicy":
                return new exports.TunnelDestGroupIamPolicy(name, undefined, { urn });
            case "gcp:iap/tunnelIamBinding:TunnelIamBinding":
                return new exports.TunnelIamBinding(name, undefined, { urn });
            case "gcp:iap/tunnelIamMember:TunnelIamMember":
                return new exports.TunnelIamMember(name, undefined, { urn });
            case "gcp:iap/tunnelIamPolicy:TunnelIamPolicy":
                return new exports.TunnelIamPolicy(name, undefined, { urn });
            case "gcp:iap/tunnelInstanceIAMBinding:TunnelInstanceIAMBinding":
                return new exports.TunnelInstanceIAMBinding(name, undefined, { urn });
            case "gcp:iap/tunnelInstanceIAMMember:TunnelInstanceIAMMember":
                return new exports.TunnelInstanceIAMMember(name, undefined, { urn });
            case "gcp:iap/tunnelInstanceIAMPolicy:TunnelInstanceIAMPolicy":
                return new exports.TunnelInstanceIAMPolicy(name, undefined, { urn });
            case "gcp:iap/webBackendServiceIamBinding:WebBackendServiceIamBinding":
                return new exports.WebBackendServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/webBackendServiceIamMember:WebBackendServiceIamMember":
                return new exports.WebBackendServiceIamMember(name, undefined, { urn });
            case "gcp:iap/webBackendServiceIamPolicy:WebBackendServiceIamPolicy":
                return new exports.WebBackendServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/webCloudRunServiceIamBinding:WebCloudRunServiceIamBinding":
                return new exports.WebCloudRunServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/webCloudRunServiceIamMember:WebCloudRunServiceIamMember":
                return new exports.WebCloudRunServiceIamMember(name, undefined, { urn });
            case "gcp:iap/webCloudRunServiceIamPolicy:WebCloudRunServiceIamPolicy":
                return new exports.WebCloudRunServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/webForwardingRuleServiceIamBinding:WebForwardingRuleServiceIamBinding":
                return new exports.WebForwardingRuleServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/webForwardingRuleServiceIamMember:WebForwardingRuleServiceIamMember":
                return new exports.WebForwardingRuleServiceIamMember(name, undefined, { urn });
            case "gcp:iap/webForwardingRuleServiceIamPolicy:WebForwardingRuleServiceIamPolicy":
                return new exports.WebForwardingRuleServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/webIamBinding:WebIamBinding":
                return new exports.WebIamBinding(name, undefined, { urn });
            case "gcp:iap/webIamMember:WebIamMember":
                return new exports.WebIamMember(name, undefined, { urn });
            case "gcp:iap/webIamPolicy:WebIamPolicy":
                return new exports.WebIamPolicy(name, undefined, { urn });
            case "gcp:iap/webRegionBackendServiceIamBinding:WebRegionBackendServiceIamBinding":
                return new exports.WebRegionBackendServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/webRegionBackendServiceIamMember:WebRegionBackendServiceIamMember":
                return new exports.WebRegionBackendServiceIamMember(name, undefined, { urn });
            case "gcp:iap/webRegionBackendServiceIamPolicy:WebRegionBackendServiceIamPolicy":
                return new exports.WebRegionBackendServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/webRegionForwardingRuleServiceIamBinding:WebRegionForwardingRuleServiceIamBinding":
                return new exports.WebRegionForwardingRuleServiceIamBinding(name, undefined, { urn });
            case "gcp:iap/webRegionForwardingRuleServiceIamMember:WebRegionForwardingRuleServiceIamMember":
                return new exports.WebRegionForwardingRuleServiceIamMember(name, undefined, { urn });
            case "gcp:iap/webRegionForwardingRuleServiceIamPolicy:WebRegionForwardingRuleServiceIamPolicy":
                return new exports.WebRegionForwardingRuleServiceIamPolicy(name, undefined, { urn });
            case "gcp:iap/webTypeAppEngingIamBinding:WebTypeAppEngingIamBinding":
                return new exports.WebTypeAppEngingIamBinding(name, undefined, { urn });
            case "gcp:iap/webTypeAppEngingIamMember:WebTypeAppEngingIamMember":
                return new exports.WebTypeAppEngingIamMember(name, undefined, { urn });
            case "gcp:iap/webTypeAppEngingIamPolicy:WebTypeAppEngingIamPolicy":
                return new exports.WebTypeAppEngingIamPolicy(name, undefined, { urn });
            case "gcp:iap/webTypeComputeIamBinding:WebTypeComputeIamBinding":
                return new exports.WebTypeComputeIamBinding(name, undefined, { urn });
            case "gcp:iap/webTypeComputeIamMember:WebTypeComputeIamMember":
                return new exports.WebTypeComputeIamMember(name, undefined, { urn });
            case "gcp:iap/webTypeComputeIamPolicy:WebTypeComputeIamPolicy":
                return new exports.WebTypeComputeIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "iap/agentRegistryIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/agentRegistryIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/agentRegistryIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineVersionIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineVersionIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/appEngineVersionIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/brand", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/client", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/locationWebIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/locationWebIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/locationWebIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/settings", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelDestGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelDestGroupIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelDestGroupIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelDestGroupIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelInstanceIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelInstanceIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/tunnelInstanceIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webBackendServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webBackendServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webBackendServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webCloudRunServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webCloudRunServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webCloudRunServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webForwardingRuleServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webForwardingRuleServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webForwardingRuleServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionBackendServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionBackendServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionBackendServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionForwardingRuleServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionForwardingRuleServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webRegionForwardingRuleServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeAppEngingIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeAppEngingIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeAppEngingIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeComputeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeComputeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iap/webTypeComputeIamPolicy", _module);
