// @pulumi/gcp/dataplex/dataProduct.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataProduct = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A data product is a curated collection of data assets, packaged to address
 * specific use cases.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about DataProduct, see:
 *
 * * [API documentation](https://cloud.google.com/dataplex/docs/reference/rest/v1/projects.locations.dataProducts)
 * * How-to Guides
 *     * [Introduction to Data Products](https://cloud.google.com/dataplex/docs/data-products-overview)
 *
 * ## Example Usage
 *
 * ### Dataplex Data Product Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.dataplex.DataProduct("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: "data-product-basic",
 *     displayName: "terraform data product",
 *     ownerEmails: ["gterraformtestuser@gmail.com"],
 *     accessGroups: [{
 *         id: "analyst",
 *         groupId: "analyst",
 *         displayName: "Data Analyst",
 *         principal: {
 *             googleGroup: "tf-test-analysts-_79169@example.com",
 *         },
 *     }],
 * });
 * ```
 * ### Dataplex Data Product Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.dataplex.DataProduct("example", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     dataProductId: "data-product-full",
 *     displayName: "DP Full Test: Special Chars !@#$",
 *     description: "Updated with emojis 🚀 and brackets {test}",
 *     ownerEmails: ["gterraformtestuser@gmail.com"],
 *     labels: {
 *         env: "manual-test",
 *     },
 *     accessGroups: [
 *         {
 *             id: "analyst",
 *             groupId: "analyst",
 *             displayName: "Data Analyst - Updated",
 *             description: "In-place update verified",
 *             principal: {
 *                 googleGroup: "tf-test-analysts-_56529@example.com",
 *             },
 *         },
 *         {
 *             id: "scientist",
 *             groupId: "scientist",
 *             displayName: "Data Scientist",
 *             principal: {
 *                 googleGroup: "tf-test-scientists-_75413@example.com",
 *             },
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * DataProduct can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/dataProducts/{{data_product_id}}`
 * * `{{project}}/{{location}}/{{data_product_id}}`
 * * `{{location}}/{{data_product_id}}`
 *
 * When using the `pulumi import` command, DataProduct can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataplex/dataProduct:DataProduct default projects/{{project}}/locations/{{location}}/dataProducts/{{data_product_id}}
 * $ pulumi import gcp:dataplex/dataProduct:DataProduct default {{project}}/{{location}}/{{data_product_id}}
 * $ pulumi import gcp:dataplex/dataProduct:DataProduct default {{location}}/{{data_product_id}}
 * ```
 */
class DataProduct extends pulumi.CustomResource {
    /**
     * Get an existing DataProduct resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DataProduct(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:dataplex/dataProduct:DataProduct';
    /**
     * Returns true if the given object is an instance of DataProduct.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataProduct.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessGroups"] = state?.accessGroups;
            resourceInputs["assetCount"] = state?.assetCount;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["dataProductId"] = state?.dataProductId;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["ownerEmails"] = state?.ownerEmails;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.dataProductId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataProductId'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.ownerEmails === undefined && !opts.urn) {
                throw new Error("Missing required property 'ownerEmails'");
            }
            resourceInputs["accessGroups"] = args?.accessGroups;
            resourceInputs["dataProductId"] = args?.dataProductId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["ownerEmails"] = args?.ownerEmails;
            resourceInputs["project"] = args?.project;
            resourceInputs["assetCount"] = undefined /*out*/;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(DataProduct.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataProduct = DataProduct;
