// @pulumi/gcp/dataplex/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlossaryIamPolicy = exports.GlossaryIamMember = exports.GlossaryIamBinding = exports.GlossaryCategory = exports.Glossary = exports.getZoneIamPolicyOutput = exports.getZoneIamPolicy = exports.getTaskIamPolicyOutput = exports.getTaskIamPolicy = exports.getLakeIamPolicyOutput = exports.getLakeIamPolicy = exports.getGlossaryIamPolicyOutput = exports.getGlossaryIamPolicy = exports.getEntryTypeIamPolicyOutput = exports.getEntryTypeIamPolicy = exports.getEntryGroupIamPolicyOutput = exports.getEntryGroupIamPolicy = exports.getDatascanIamPolicyOutput = exports.getDatascanIamPolicy = exports.getDataQualityRulesOutput = exports.getDataQualityRules = exports.getAssetIamPolicyOutput = exports.getAssetIamPolicy = exports.getAspectTypeIamPolicyOutput = exports.getAspectTypeIamPolicy = exports.EntryTypeIamPolicy = exports.EntryTypeIamMember = exports.EntryTypeIamBinding = exports.EntryType = exports.EntryLink = exports.EntryGroupIamPolicy = exports.EntryGroupIamMember = exports.EntryGroupIamBinding = exports.EntryGroup = exports.Entry = exports.DatascanIamPolicy = exports.DatascanIamMember = exports.DatascanIamBinding = exports.Datascan = exports.DataProductDataAsset = exports.DataProduct = exports.DataAsset = exports.AssetIamPolicy = exports.AssetIamMember = exports.AssetIamBinding = exports.Asset = exports.AspectTypeIamPolicy = exports.AspectTypeIamMember = exports.AspectTypeIamBinding = exports.AspectType = void 0;
exports.ZoneIamPolicy = exports.ZoneIamMember = exports.ZoneIamBinding = exports.Zone = exports.TaskIamPolicy = exports.TaskIamMember = exports.TaskIamBinding = exports.Task = exports.LakeIamPolicy = exports.LakeIamMember = exports.LakeIamBinding = exports.Lake = exports.GlossaryTerm = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AspectType = null;
utilities.lazyLoad(exports, ["AspectType"], () => require("./aspectType"));
exports.AspectTypeIamBinding = null;
utilities.lazyLoad(exports, ["AspectTypeIamBinding"], () => require("./aspectTypeIamBinding"));
exports.AspectTypeIamMember = null;
utilities.lazyLoad(exports, ["AspectTypeIamMember"], () => require("./aspectTypeIamMember"));
exports.AspectTypeIamPolicy = null;
utilities.lazyLoad(exports, ["AspectTypeIamPolicy"], () => require("./aspectTypeIamPolicy"));
exports.Asset = null;
utilities.lazyLoad(exports, ["Asset"], () => require("./asset"));
exports.AssetIamBinding = null;
utilities.lazyLoad(exports, ["AssetIamBinding"], () => require("./assetIamBinding"));
exports.AssetIamMember = null;
utilities.lazyLoad(exports, ["AssetIamMember"], () => require("./assetIamMember"));
exports.AssetIamPolicy = null;
utilities.lazyLoad(exports, ["AssetIamPolicy"], () => require("./assetIamPolicy"));
exports.DataAsset = null;
utilities.lazyLoad(exports, ["DataAsset"], () => require("./dataAsset"));
exports.DataProduct = null;
utilities.lazyLoad(exports, ["DataProduct"], () => require("./dataProduct"));
exports.DataProductDataAsset = null;
utilities.lazyLoad(exports, ["DataProductDataAsset"], () => require("./dataProductDataAsset"));
exports.Datascan = null;
utilities.lazyLoad(exports, ["Datascan"], () => require("./datascan"));
exports.DatascanIamBinding = null;
utilities.lazyLoad(exports, ["DatascanIamBinding"], () => require("./datascanIamBinding"));
exports.DatascanIamMember = null;
utilities.lazyLoad(exports, ["DatascanIamMember"], () => require("./datascanIamMember"));
exports.DatascanIamPolicy = null;
utilities.lazyLoad(exports, ["DatascanIamPolicy"], () => require("./datascanIamPolicy"));
exports.Entry = null;
utilities.lazyLoad(exports, ["Entry"], () => require("./entry"));
exports.EntryGroup = null;
utilities.lazyLoad(exports, ["EntryGroup"], () => require("./entryGroup"));
exports.EntryGroupIamBinding = null;
utilities.lazyLoad(exports, ["EntryGroupIamBinding"], () => require("./entryGroupIamBinding"));
exports.EntryGroupIamMember = null;
utilities.lazyLoad(exports, ["EntryGroupIamMember"], () => require("./entryGroupIamMember"));
exports.EntryGroupIamPolicy = null;
utilities.lazyLoad(exports, ["EntryGroupIamPolicy"], () => require("./entryGroupIamPolicy"));
exports.EntryLink = null;
utilities.lazyLoad(exports, ["EntryLink"], () => require("./entryLink"));
exports.EntryType = null;
utilities.lazyLoad(exports, ["EntryType"], () => require("./entryType"));
exports.EntryTypeIamBinding = null;
utilities.lazyLoad(exports, ["EntryTypeIamBinding"], () => require("./entryTypeIamBinding"));
exports.EntryTypeIamMember = null;
utilities.lazyLoad(exports, ["EntryTypeIamMember"], () => require("./entryTypeIamMember"));
exports.EntryTypeIamPolicy = null;
utilities.lazyLoad(exports, ["EntryTypeIamPolicy"], () => require("./entryTypeIamPolicy"));
exports.getAspectTypeIamPolicy = null;
exports.getAspectTypeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAspectTypeIamPolicy", "getAspectTypeIamPolicyOutput"], () => require("./getAspectTypeIamPolicy"));
exports.getAssetIamPolicy = null;
exports.getAssetIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAssetIamPolicy", "getAssetIamPolicyOutput"], () => require("./getAssetIamPolicy"));
exports.getDataQualityRules = null;
exports.getDataQualityRulesOutput = null;
utilities.lazyLoad(exports, ["getDataQualityRules", "getDataQualityRulesOutput"], () => require("./getDataQualityRules"));
exports.getDatascanIamPolicy = null;
exports.getDatascanIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getDatascanIamPolicy", "getDatascanIamPolicyOutput"], () => require("./getDatascanIamPolicy"));
exports.getEntryGroupIamPolicy = null;
exports.getEntryGroupIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getEntryGroupIamPolicy", "getEntryGroupIamPolicyOutput"], () => require("./getEntryGroupIamPolicy"));
exports.getEntryTypeIamPolicy = null;
exports.getEntryTypeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getEntryTypeIamPolicy", "getEntryTypeIamPolicyOutput"], () => require("./getEntryTypeIamPolicy"));
exports.getGlossaryIamPolicy = null;
exports.getGlossaryIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getGlossaryIamPolicy", "getGlossaryIamPolicyOutput"], () => require("./getGlossaryIamPolicy"));
exports.getLakeIamPolicy = null;
exports.getLakeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getLakeIamPolicy", "getLakeIamPolicyOutput"], () => require("./getLakeIamPolicy"));
exports.getTaskIamPolicy = null;
exports.getTaskIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTaskIamPolicy", "getTaskIamPolicyOutput"], () => require("./getTaskIamPolicy"));
exports.getZoneIamPolicy = null;
exports.getZoneIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getZoneIamPolicy", "getZoneIamPolicyOutput"], () => require("./getZoneIamPolicy"));
exports.Glossary = null;
utilities.lazyLoad(exports, ["Glossary"], () => require("./glossary"));
exports.GlossaryCategory = null;
utilities.lazyLoad(exports, ["GlossaryCategory"], () => require("./glossaryCategory"));
exports.GlossaryIamBinding = null;
utilities.lazyLoad(exports, ["GlossaryIamBinding"], () => require("./glossaryIamBinding"));
exports.GlossaryIamMember = null;
utilities.lazyLoad(exports, ["GlossaryIamMember"], () => require("./glossaryIamMember"));
exports.GlossaryIamPolicy = null;
utilities.lazyLoad(exports, ["GlossaryIamPolicy"], () => require("./glossaryIamPolicy"));
exports.GlossaryTerm = null;
utilities.lazyLoad(exports, ["GlossaryTerm"], () => require("./glossaryTerm"));
exports.Lake = null;
utilities.lazyLoad(exports, ["Lake"], () => require("./lake"));
exports.LakeIamBinding = null;
utilities.lazyLoad(exports, ["LakeIamBinding"], () => require("./lakeIamBinding"));
exports.LakeIamMember = null;
utilities.lazyLoad(exports, ["LakeIamMember"], () => require("./lakeIamMember"));
exports.LakeIamPolicy = null;
utilities.lazyLoad(exports, ["LakeIamPolicy"], () => require("./lakeIamPolicy"));
exports.Task = null;
utilities.lazyLoad(exports, ["Task"], () => require("./task"));
exports.TaskIamBinding = null;
utilities.lazyLoad(exports, ["TaskIamBinding"], () => require("./taskIamBinding"));
exports.TaskIamMember = null;
utilities.lazyLoad(exports, ["TaskIamMember"], () => require("./taskIamMember"));
exports.TaskIamPolicy = null;
utilities.lazyLoad(exports, ["TaskIamPolicy"], () => require("./taskIamPolicy"));
exports.Zone = null;
utilities.lazyLoad(exports, ["Zone"], () => require("./zone"));
exports.ZoneIamBinding = null;
utilities.lazyLoad(exports, ["ZoneIamBinding"], () => require("./zoneIamBinding"));
exports.ZoneIamMember = null;
utilities.lazyLoad(exports, ["ZoneIamMember"], () => require("./zoneIamMember"));
exports.ZoneIamPolicy = null;
utilities.lazyLoad(exports, ["ZoneIamPolicy"], () => require("./zoneIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:dataplex/aspectType:AspectType":
                return new exports.AspectType(name, undefined, { urn });
            case "gcp:dataplex/aspectTypeIamBinding:AspectTypeIamBinding":
                return new exports.AspectTypeIamBinding(name, undefined, { urn });
            case "gcp:dataplex/aspectTypeIamMember:AspectTypeIamMember":
                return new exports.AspectTypeIamMember(name, undefined, { urn });
            case "gcp:dataplex/aspectTypeIamPolicy:AspectTypeIamPolicy":
                return new exports.AspectTypeIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/asset:Asset":
                return new exports.Asset(name, undefined, { urn });
            case "gcp:dataplex/assetIamBinding:AssetIamBinding":
                return new exports.AssetIamBinding(name, undefined, { urn });
            case "gcp:dataplex/assetIamMember:AssetIamMember":
                return new exports.AssetIamMember(name, undefined, { urn });
            case "gcp:dataplex/assetIamPolicy:AssetIamPolicy":
                return new exports.AssetIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/dataAsset:DataAsset":
                return new exports.DataAsset(name, undefined, { urn });
            case "gcp:dataplex/dataProduct:DataProduct":
                return new exports.DataProduct(name, undefined, { urn });
            case "gcp:dataplex/dataProductDataAsset:DataProductDataAsset":
                return new exports.DataProductDataAsset(name, undefined, { urn });
            case "gcp:dataplex/datascan:Datascan":
                return new exports.Datascan(name, undefined, { urn });
            case "gcp:dataplex/datascanIamBinding:DatascanIamBinding":
                return new exports.DatascanIamBinding(name, undefined, { urn });
            case "gcp:dataplex/datascanIamMember:DatascanIamMember":
                return new exports.DatascanIamMember(name, undefined, { urn });
            case "gcp:dataplex/datascanIamPolicy:DatascanIamPolicy":
                return new exports.DatascanIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/entry:Entry":
                return new exports.Entry(name, undefined, { urn });
            case "gcp:dataplex/entryGroup:EntryGroup":
                return new exports.EntryGroup(name, undefined, { urn });
            case "gcp:dataplex/entryGroupIamBinding:EntryGroupIamBinding":
                return new exports.EntryGroupIamBinding(name, undefined, { urn });
            case "gcp:dataplex/entryGroupIamMember:EntryGroupIamMember":
                return new exports.EntryGroupIamMember(name, undefined, { urn });
            case "gcp:dataplex/entryGroupIamPolicy:EntryGroupIamPolicy":
                return new exports.EntryGroupIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/entryLink:EntryLink":
                return new exports.EntryLink(name, undefined, { urn });
            case "gcp:dataplex/entryType:EntryType":
                return new exports.EntryType(name, undefined, { urn });
            case "gcp:dataplex/entryTypeIamBinding:EntryTypeIamBinding":
                return new exports.EntryTypeIamBinding(name, undefined, { urn });
            case "gcp:dataplex/entryTypeIamMember:EntryTypeIamMember":
                return new exports.EntryTypeIamMember(name, undefined, { urn });
            case "gcp:dataplex/entryTypeIamPolicy:EntryTypeIamPolicy":
                return new exports.EntryTypeIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/glossary:Glossary":
                return new exports.Glossary(name, undefined, { urn });
            case "gcp:dataplex/glossaryCategory:GlossaryCategory":
                return new exports.GlossaryCategory(name, undefined, { urn });
            case "gcp:dataplex/glossaryIamBinding:GlossaryIamBinding":
                return new exports.GlossaryIamBinding(name, undefined, { urn });
            case "gcp:dataplex/glossaryIamMember:GlossaryIamMember":
                return new exports.GlossaryIamMember(name, undefined, { urn });
            case "gcp:dataplex/glossaryIamPolicy:GlossaryIamPolicy":
                return new exports.GlossaryIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/glossaryTerm:GlossaryTerm":
                return new exports.GlossaryTerm(name, undefined, { urn });
            case "gcp:dataplex/lake:Lake":
                return new exports.Lake(name, undefined, { urn });
            case "gcp:dataplex/lakeIamBinding:LakeIamBinding":
                return new exports.LakeIamBinding(name, undefined, { urn });
            case "gcp:dataplex/lakeIamMember:LakeIamMember":
                return new exports.LakeIamMember(name, undefined, { urn });
            case "gcp:dataplex/lakeIamPolicy:LakeIamPolicy":
                return new exports.LakeIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/task:Task":
                return new exports.Task(name, undefined, { urn });
            case "gcp:dataplex/taskIamBinding:TaskIamBinding":
                return new exports.TaskIamBinding(name, undefined, { urn });
            case "gcp:dataplex/taskIamMember:TaskIamMember":
                return new exports.TaskIamMember(name, undefined, { urn });
            case "gcp:dataplex/taskIamPolicy:TaskIamPolicy":
                return new exports.TaskIamPolicy(name, undefined, { urn });
            case "gcp:dataplex/zone:Zone":
                return new exports.Zone(name, undefined, { urn });
            case "gcp:dataplex/zoneIamBinding:ZoneIamBinding":
                return new exports.ZoneIamBinding(name, undefined, { urn });
            case "gcp:dataplex/zoneIamMember:ZoneIamMember":
                return new exports.ZoneIamMember(name, undefined, { urn });
            case "gcp:dataplex/zoneIamPolicy:ZoneIamPolicy":
                return new exports.ZoneIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "dataplex/aspectType", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/aspectTypeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/aspectTypeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/aspectTypeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/asset", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/assetIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/assetIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/assetIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/dataAsset", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/dataProduct", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/dataProductDataAsset", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/datascan", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/datascanIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/datascanIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/datascanIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entry", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryGroupIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryGroupIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryGroupIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryLink", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryType", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryTypeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryTypeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/entryTypeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossary", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossaryCategory", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossaryIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossaryIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossaryIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/glossaryTerm", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/lake", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/lakeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/lakeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/lakeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/task", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/taskIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/taskIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/taskIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/zone", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/zoneIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/zoneIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataplex/zoneIamPolicy", _module);
