// @pulumi/azure-native/workloads/sapCentralServerInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SapCentralServerInstance = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Define the SAP Central Services Instance resource.
 *
 * Uses Azure REST API version 2024-09-01.
 */
class SapCentralServerInstance extends pulumi.CustomResource {
    /**
     * Get an existing SapCentralServerInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SapCentralServerInstance(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:workloads:SapCentralServerInstance';
    /**
     * Returns true if the given object is an instance of SapCentralServerInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SapCentralServerInstance.__pulumiType;
    }
    /**
     * Create a SapCentralServerInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.sapVirtualInstanceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'sapVirtualInstanceName'");
            }
            resourceInputs["centralInstanceName"] = args?.centralInstanceName;
            resourceInputs["location"] = args?.location;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sapVirtualInstanceName"] = args?.sapVirtualInstanceName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enqueueReplicationServerProperties"] = undefined /*out*/;
            resourceInputs["enqueueServerProperties"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["gatewayServerProperties"] = undefined /*out*/;
            resourceInputs["health"] = undefined /*out*/;
            resourceInputs["instanceNo"] = undefined /*out*/;
            resourceInputs["kernelPatch"] = undefined /*out*/;
            resourceInputs["kernelVersion"] = undefined /*out*/;
            resourceInputs["loadBalancerDetails"] = undefined /*out*/;
            resourceInputs["messageServerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subnet"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmDetails"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enqueueReplicationServerProperties"] = undefined /*out*/;
            resourceInputs["enqueueServerProperties"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["gatewayServerProperties"] = undefined /*out*/;
            resourceInputs["health"] = undefined /*out*/;
            resourceInputs["instanceNo"] = undefined /*out*/;
            resourceInputs["kernelPatch"] = undefined /*out*/;
            resourceInputs["kernelVersion"] = undefined /*out*/;
            resourceInputs["loadBalancerDetails"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["messageServerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subnet"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmDetails"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:workloads/v20211201preview:SapCentralServerInstance" }, { type: "azure-native:workloads/v20221101preview:SapCentralServerInstance" }, { type: "azure-native:workloads/v20230401:SAPCentralInstance" }, { type: "azure-native:workloads/v20230401:SapCentralServerInstance" }, { type: "azure-native:workloads/v20231001preview:SAPCentralInstance" }, { type: "azure-native:workloads/v20231001preview:SapCentralServerInstance" }, { type: "azure-native:workloads/v20240901:SapCentralServerInstance" }, { type: "azure-native:workloads:SAPCentralInstance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SapCentralServerInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SapCentralServerInstance = SapCentralServerInstance;
