// @pulumi/azure-native/workloads/serverInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerInstance = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Define the Server Instance resource.
 *
 * Uses Azure REST API version 2023-10-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-10-01-preview.
 */
class ServerInstance extends pulumi.CustomResource {
    /**
     * Get an existing ServerInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerInstance(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:workloads:ServerInstance';
    /**
     * Returns true if the given object is an instance of ServerInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerInstance.__pulumiType;
    }
    /**
     * Create a ServerInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.sapDiscoverySiteName === undefined && !opts.urn) {
                throw new Error("Missing required property 'sapDiscoverySiteName'");
            }
            if (args?.sapInstanceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'sapInstanceName'");
            }
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sapDiscoverySiteName"] = args?.sapDiscoverySiteName;
            resourceInputs["sapInstanceName"] = args?.sapInstanceName;
            resourceInputs["serverInstanceName"] = args?.serverInstanceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationData"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["instanceSid"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operatingSystem"] = undefined /*out*/;
            resourceInputs["performanceData"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sapInstanceType"] = undefined /*out*/;
            resourceInputs["sapProduct"] = undefined /*out*/;
            resourceInputs["sapProductVersion"] = undefined /*out*/;
            resourceInputs["serverName"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationData"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["instanceSid"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operatingSystem"] = undefined /*out*/;
            resourceInputs["performanceData"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sapInstanceType"] = undefined /*out*/;
            resourceInputs["sapProduct"] = undefined /*out*/;
            resourceInputs["sapProductVersion"] = undefined /*out*/;
            resourceInputs["serverName"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:workloads/v20231001preview:ServerInstance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServerInstance = ServerInstance;
