// @pulumi/aws/sqs/sqsMixins.js
"use strict";
// Copyright 2016-2018, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueEventSubscription = void 0;
const lambda = __importStar(require("../lambda"));
const queue = __importStar(require("./queue"));
const utils = __importStar(require("../utils"));
class QueueEventSubscription extends lambda.EventSubscription {
    queue;
    /**
     * The underlying sns object created for the subscription.
     */
    eventSourceMapping;
    constructor(name, queue, handler, args = {}, opts = {}) {
        // We previously did not parent the subscription to the queue. We now do. Provide an alias
        // so this doesn't cause resources to be destroyed/recreated for existing stacks.
        super("aws:sqs:QueueEventSubscription", name, {
            parent: queue,
            ...utils.withAlias(opts, { parent: opts.parent }),
        });
        const parentOpts = { parent: this };
        this.func = createFunctionFromEventHandler(name, handler, parentOpts);
        this.permission = new lambda.Permission(name, {
            action: "lambda:*",
            function: this.func.name,
            principal: "sqs.amazonaws.com",
            sourceArn: queue.arn,
        }, parentOpts);
        this.eventSourceMapping = new lambda.EventSourceMapping(name, {
            batchSize: args.batchSize,
            maximumBatchingWindowInSeconds: args.maximumBatchingWindowInSeconds,
            enabled: true,
            eventSourceArn: queue.arn,
            functionName: this.func.name,
        }, parentOpts);
        this.queue = queue;
        this.registerOutputs();
    }
}
exports.QueueEventSubscription = QueueEventSubscription;
function createFunctionFromEventHandler(name, handler, opts) {
    if (handler instanceof Function) {
        return new lambda.CallbackFunction(name, {
            callback: handler,
            // as we don't pass policies, we will pick up the default policies of the Callbackfunction
        }, opts);
    }
    else {
        return handler;
    }
}
queue.Queue.prototype.onEvent = function (name, handler, args, opts) {
    return new QueueEventSubscription(name, this, handler, args, opts);
};
