// @pulumi/azure-native/kubernetesconfiguration/sourceControlConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceControlConfiguration = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The SourceControl Configuration object returned in Get & Put response.
 *
 * Uses Azure REST API version 2023-05-01. In version 2.x of the Azure Native provider, it used API version 2023-05-01.
 *
 * Other available API versions: 2022-07-01, 2022-11-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native kubernetesconfiguration [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SourceControlConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing SourceControlConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SourceControlConfiguration(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:kubernetesconfiguration:SourceControlConfiguration';
    /**
     * Returns true if the given object is an instance of SourceControlConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SourceControlConfiguration.__pulumiType;
    }
    /**
     * Create a SourceControlConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.clusterResourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterResourceName'");
            }
            if (args?.clusterRp === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterRp'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["clusterResourceName"] = args?.clusterResourceName;
            resourceInputs["clusterRp"] = args?.clusterRp;
            resourceInputs["configurationProtectedSettings"] = args?.configurationProtectedSettings;
            resourceInputs["enableHelmOperator"] = args?.enableHelmOperator;
            resourceInputs["helmOperatorProperties"] = args?.helmOperatorProperties;
            resourceInputs["operatorInstanceName"] = args?.operatorInstanceName;
            resourceInputs["operatorNamespace"] = (args?.operatorNamespace) ?? "default";
            resourceInputs["operatorParams"] = args?.operatorParams;
            resourceInputs["operatorScope"] = args?.operatorScope;
            resourceInputs["operatorType"] = args?.operatorType;
            resourceInputs["repositoryUrl"] = args?.repositoryUrl;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sourceControlConfigurationName"] = args?.sourceControlConfigurationName;
            resourceInputs["sshKnownHostsContents"] = args?.sshKnownHostsContents;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["complianceStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["repositoryPublicKey"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["complianceStatus"] = undefined /*out*/;
            resourceInputs["configurationProtectedSettings"] = undefined /*out*/;
            resourceInputs["enableHelmOperator"] = undefined /*out*/;
            resourceInputs["helmOperatorProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operatorInstanceName"] = undefined /*out*/;
            resourceInputs["operatorNamespace"] = undefined /*out*/;
            resourceInputs["operatorParams"] = undefined /*out*/;
            resourceInputs["operatorScope"] = undefined /*out*/;
            resourceInputs["operatorType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["repositoryPublicKey"] = undefined /*out*/;
            resourceInputs["repositoryUrl"] = undefined /*out*/;
            resourceInputs["sshKnownHostsContents"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:kubernetesconfiguration/v20191101preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20200701preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20201001preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20210301:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20210501preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20211101preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20220101preview:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20220301:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20220701:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20221101:SourceControlConfiguration" }, { type: "azure-native:kubernetesconfiguration/v20230501:SourceControlConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SourceControlConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SourceControlConfiguration = SourceControlConfiguration;
