// @pulumi/aws/servicediscovery/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = exports.PublicDnsNamespace = exports.PrivateDnsNamespace = exports.Instance = exports.HttpNamespace = exports.getServiceOutput = exports.getService = exports.getHttpNamespaceOutput = exports.getHttpNamespace = exports.getDnsNamespaceOutput = exports.getDnsNamespace = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getDnsNamespace = null;
exports.getDnsNamespaceOutput = null;
utilities.lazyLoad(exports, ["getDnsNamespace", "getDnsNamespaceOutput"], () => require("./getDnsNamespace"));
exports.getHttpNamespace = null;
exports.getHttpNamespaceOutput = null;
utilities.lazyLoad(exports, ["getHttpNamespace", "getHttpNamespaceOutput"], () => require("./getHttpNamespace"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.HttpNamespace = null;
utilities.lazyLoad(exports, ["HttpNamespace"], () => require("./httpNamespace"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.PrivateDnsNamespace = null;
utilities.lazyLoad(exports, ["PrivateDnsNamespace"], () => require("./privateDnsNamespace"));
exports.PublicDnsNamespace = null;
utilities.lazyLoad(exports, ["PublicDnsNamespace"], () => require("./publicDnsNamespace"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:servicediscovery/httpNamespace:HttpNamespace":
                return new exports.HttpNamespace(name, undefined, { urn });
            case "aws:servicediscovery/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws:servicediscovery/privateDnsNamespace:PrivateDnsNamespace":
                return new exports.PrivateDnsNamespace(name, undefined, { urn });
            case "aws:servicediscovery/publicDnsNamespace:PublicDnsNamespace":
                return new exports.PublicDnsNamespace(name, undefined, { urn });
            case "aws:servicediscovery/service:Service":
                return new exports.Service(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "servicediscovery/httpNamespace", _module);
pulumi.runtime.registerResourceModule("aws", "servicediscovery/instance", _module);
pulumi.runtime.registerResourceModule("aws", "servicediscovery/privateDnsNamespace", _module);
pulumi.runtime.registerResourceModule("aws", "servicediscovery/publicDnsNamespace", _module);
pulumi.runtime.registerResourceModule("aws", "servicediscovery/service", _module);
