// @pulumi/kubernetes/admissionregistration/v1alpha1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidatingAdmissionPolicyPatch = exports.ValidatingAdmissionPolicyList = exports.ValidatingAdmissionPolicyBindingPatch = exports.ValidatingAdmissionPolicyBindingList = exports.ValidatingAdmissionPolicyBinding = exports.ValidatingAdmissionPolicy = exports.MutatingAdmissionPolicyPatch = exports.MutatingAdmissionPolicyList = exports.MutatingAdmissionPolicyBindingPatch = exports.MutatingAdmissionPolicyBindingList = exports.MutatingAdmissionPolicyBinding = exports.MutatingAdmissionPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.MutatingAdmissionPolicy = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicy"], () => require("./mutatingAdmissionPolicy"));
exports.MutatingAdmissionPolicyBinding = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicyBinding"], () => require("./mutatingAdmissionPolicyBinding"));
exports.MutatingAdmissionPolicyBindingList = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicyBindingList"], () => require("./mutatingAdmissionPolicyBindingList"));
exports.MutatingAdmissionPolicyBindingPatch = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicyBindingPatch"], () => require("./mutatingAdmissionPolicyBindingPatch"));
exports.MutatingAdmissionPolicyList = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicyList"], () => require("./mutatingAdmissionPolicyList"));
exports.MutatingAdmissionPolicyPatch = null;
utilities.lazyLoad(exports, ["MutatingAdmissionPolicyPatch"], () => require("./mutatingAdmissionPolicyPatch"));
exports.ValidatingAdmissionPolicy = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicy"], () => require("./validatingAdmissionPolicy"));
exports.ValidatingAdmissionPolicyBinding = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicyBinding"], () => require("./validatingAdmissionPolicyBinding"));
exports.ValidatingAdmissionPolicyBindingList = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicyBindingList"], () => require("./validatingAdmissionPolicyBindingList"));
exports.ValidatingAdmissionPolicyBindingPatch = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicyBindingPatch"], () => require("./validatingAdmissionPolicyBindingPatch"));
exports.ValidatingAdmissionPolicyList = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicyList"], () => require("./validatingAdmissionPolicyList"));
exports.ValidatingAdmissionPolicyPatch = null;
utilities.lazyLoad(exports, ["ValidatingAdmissionPolicyPatch"], () => require("./validatingAdmissionPolicyPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicy":
                return new exports.MutatingAdmissionPolicy(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicyBinding":
                return new exports.MutatingAdmissionPolicyBinding(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicyBindingList":
                return new exports.MutatingAdmissionPolicyBindingList(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicyBindingPatch":
                return new exports.MutatingAdmissionPolicyBindingPatch(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicyList":
                return new exports.MutatingAdmissionPolicyList(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:MutatingAdmissionPolicyPatch":
                return new exports.MutatingAdmissionPolicyPatch(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicy":
                return new exports.ValidatingAdmissionPolicy(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicyBinding":
                return new exports.ValidatingAdmissionPolicyBinding(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicyBindingList":
                return new exports.ValidatingAdmissionPolicyBindingList(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicyBindingPatch":
                return new exports.ValidatingAdmissionPolicyBindingPatch(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicyList":
                return new exports.ValidatingAdmissionPolicyList(name, undefined, { urn });
            case "kubernetes:admissionregistration.k8s.io/v1alpha1:ValidatingAdmissionPolicyPatch":
                return new exports.ValidatingAdmissionPolicyPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "admissionregistration.k8s.io/v1alpha1", _module);
