// @pulumi/azure-native/databricks/vnetPeering.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VNetPeering = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Peerings in a VirtualNetwork resource
 *
 * Uses Azure REST API version 2024-05-01.
 *
 * Other available API versions: 2023-02-01, 2023-09-15-preview, 2024-09-01-preview, 2025-03-01-preview, 2025-08-01-preview, 2025-10-01-preview, 2026-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native databricks [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class VNetPeering extends pulumi.CustomResource {
    /**
     * Get an existing VNetPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VNetPeering(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:databricks:VNetPeering';
    /**
     * Returns true if the given object is an instance of VNetPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VNetPeering.__pulumiType;
    }
    /**
     * Create a VNetPeering resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.remoteVirtualNetwork === undefined && !opts.urn) {
                throw new Error("Missing required property 'remoteVirtualNetwork'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["allowForwardedTraffic"] = args?.allowForwardedTraffic;
            resourceInputs["allowGatewayTransit"] = args?.allowGatewayTransit;
            resourceInputs["allowVirtualNetworkAccess"] = args?.allowVirtualNetworkAccess;
            resourceInputs["databricksAddressSpace"] = args?.databricksAddressSpace;
            resourceInputs["databricksVirtualNetwork"] = args?.databricksVirtualNetwork;
            resourceInputs["peeringName"] = args?.peeringName;
            resourceInputs["remoteAddressSpace"] = args?.remoteAddressSpace;
            resourceInputs["remoteVirtualNetwork"] = args?.remoteVirtualNetwork;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["useRemoteGateways"] = args?.useRemoteGateways;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peeringState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowForwardedTraffic"] = undefined /*out*/;
            resourceInputs["allowGatewayTransit"] = undefined /*out*/;
            resourceInputs["allowVirtualNetworkAccess"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["databricksAddressSpace"] = undefined /*out*/;
            resourceInputs["databricksVirtualNetwork"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peeringState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["remoteAddressSpace"] = undefined /*out*/;
            resourceInputs["remoteVirtualNetwork"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useRemoteGateways"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databricks/v20180401:VNetPeering" }, { type: "azure-native:databricks/v20210401preview:VNetPeering" }, { type: "azure-native:databricks/v20220401preview:VNetPeering" }, { type: "azure-native:databricks/v20230201:VNetPeering" }, { type: "azure-native:databricks/v20230915preview:VNetPeering" }, { type: "azure-native:databricks/v20240501:VNetPeering" }, { type: "azure-native:databricks/v20240901preview:VNetPeering" }, { type: "azure-native:databricks/v20250301preview:VNetPeering" }, { type: "azure-native:databricks/v20250801preview:VNetPeering" }, { type: "azure-native:databricks/v20251001preview:VNetPeering" }, { type: "azure-native:databricks/v20260101:VNetPeering" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VNetPeering.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VNetPeering = VNetPeering;
