// @pulumi/azure-native/hybridcontainerservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VMSkus = exports.VirtualNetworkRetrieve = exports.StorageSpaceRetrieve = exports.ProvisionedCluster = exports.listProvisionedClusterInstanceUserKubeconfigOutput = exports.listProvisionedClusterInstanceUserKubeconfig = exports.listProvisionedClusterInstanceAdminKubeconfigOutput = exports.listProvisionedClusterInstanceAdminKubeconfig = exports.KubernetesVersions = exports.HybridIdentityMetadatum = exports.getVirtualNetworkRetrieveOutput = exports.getVirtualNetworkRetrieve = exports.getVMSkusOutput = exports.getVMSkus = exports.getStorageSpaceRetrieveOutput = exports.getStorageSpaceRetrieve = exports.getProvisionedClusterOutput = exports.getProvisionedCluster = exports.getKubernetesVersionsOutput = exports.getKubernetesVersions = exports.getHybridIdentityMetadatumOutput = exports.getHybridIdentityMetadatum = exports.getClusterInstanceHybridIdentityMetadatumOutput = exports.getClusterInstanceHybridIdentityMetadatum = exports.getAgentPoolOutput = exports.getAgentPool = exports.ClusterInstanceHybridIdentityMetadatum = exports.AgentPool = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.ClusterInstanceHybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["ClusterInstanceHybridIdentityMetadatum"], () => require("./clusterInstanceHybridIdentityMetadatum"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getClusterInstanceHybridIdentityMetadatum = null;
exports.getClusterInstanceHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getClusterInstanceHybridIdentityMetadatum", "getClusterInstanceHybridIdentityMetadatumOutput"], () => require("./getClusterInstanceHybridIdentityMetadatum"));
exports.getHybridIdentityMetadatum = null;
exports.getHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getHybridIdentityMetadatum", "getHybridIdentityMetadatumOutput"], () => require("./getHybridIdentityMetadatum"));
exports.getKubernetesVersions = null;
exports.getKubernetesVersionsOutput = null;
utilities.lazyLoad(exports, ["getKubernetesVersions", "getKubernetesVersionsOutput"], () => require("./getKubernetesVersions"));
exports.getProvisionedCluster = null;
exports.getProvisionedClusterOutput = null;
utilities.lazyLoad(exports, ["getProvisionedCluster", "getProvisionedClusterOutput"], () => require("./getProvisionedCluster"));
exports.getStorageSpaceRetrieve = null;
exports.getStorageSpaceRetrieveOutput = null;
utilities.lazyLoad(exports, ["getStorageSpaceRetrieve", "getStorageSpaceRetrieveOutput"], () => require("./getStorageSpaceRetrieve"));
exports.getVMSkus = null;
exports.getVMSkusOutput = null;
utilities.lazyLoad(exports, ["getVMSkus", "getVMSkusOutput"], () => require("./getVMSkus"));
exports.getVirtualNetworkRetrieve = null;
exports.getVirtualNetworkRetrieveOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkRetrieve", "getVirtualNetworkRetrieveOutput"], () => require("./getVirtualNetworkRetrieve"));
exports.HybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["HybridIdentityMetadatum"], () => require("./hybridIdentityMetadatum"));
exports.KubernetesVersions = null;
utilities.lazyLoad(exports, ["KubernetesVersions"], () => require("./kubernetesVersions"));
exports.listProvisionedClusterInstanceAdminKubeconfig = null;
exports.listProvisionedClusterInstanceAdminKubeconfigOutput = null;
utilities.lazyLoad(exports, ["listProvisionedClusterInstanceAdminKubeconfig", "listProvisionedClusterInstanceAdminKubeconfigOutput"], () => require("./listProvisionedClusterInstanceAdminKubeconfig"));
exports.listProvisionedClusterInstanceUserKubeconfig = null;
exports.listProvisionedClusterInstanceUserKubeconfigOutput = null;
utilities.lazyLoad(exports, ["listProvisionedClusterInstanceUserKubeconfig", "listProvisionedClusterInstanceUserKubeconfigOutput"], () => require("./listProvisionedClusterInstanceUserKubeconfig"));
exports.ProvisionedCluster = null;
utilities.lazyLoad(exports, ["ProvisionedCluster"], () => require("./provisionedCluster"));
exports.StorageSpaceRetrieve = null;
utilities.lazyLoad(exports, ["StorageSpaceRetrieve"], () => require("./storageSpaceRetrieve"));
exports.VirtualNetworkRetrieve = null;
utilities.lazyLoad(exports, ["VirtualNetworkRetrieve"], () => require("./virtualNetworkRetrieve"));
exports.VMSkus = null;
utilities.lazyLoad(exports, ["VMSkus"], () => require("./vmskus"));
// Export enums:
__exportStar(require("../types/enums/hybridcontainerservice"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcontainerservice:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:ClusterInstanceHybridIdentityMetadatum":
                return new exports.ClusterInstanceHybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:HybridIdentityMetadatum":
                return new exports.HybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:KubernetesVersions":
                return new exports.KubernetesVersions(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:ProvisionedCluster":
                return new exports.ProvisionedCluster(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:StorageSpaceRetrieve":
                return new exports.StorageSpaceRetrieve(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:VMSkus":
                return new exports.VMSkus(name, undefined, { urn });
            case "azure-native:hybridcontainerservice:VirtualNetworkRetrieve":
                return new exports.VirtualNetworkRetrieve(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcontainerservice", _module);
