// @pulumi/gcp/beyondcorp/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGatewayIamPolicy = exports.SecurityGatewayIamMember = exports.SecurityGatewayIamBinding = exports.SecurityGatewayApplicationIamPolicy = exports.SecurityGatewayApplicationIamMember = exports.SecurityGatewayApplicationIamBinding = exports.SecurityGatewayApplication = exports.SecurityGateway = exports.getSecurityGatewayIamPolicyOutput = exports.getSecurityGatewayIamPolicy = exports.getSecurityGatewayApplicationIamPolicyOutput = exports.getSecurityGatewayApplicationIamPolicy = exports.getSecurityGatewayOutput = exports.getSecurityGateway = exports.getAppGatewayOutput = exports.getAppGateway = exports.getAppConnectorOutput = exports.getAppConnector = exports.getAppConnectionOutput = exports.getAppConnection = exports.AppGateway = exports.AppConnector = exports.AppConnection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AppConnection = null;
utilities.lazyLoad(exports, ["AppConnection"], () => require("./appConnection"));
exports.AppConnector = null;
utilities.lazyLoad(exports, ["AppConnector"], () => require("./appConnector"));
exports.AppGateway = null;
utilities.lazyLoad(exports, ["AppGateway"], () => require("./appGateway"));
exports.getAppConnection = null;
exports.getAppConnectionOutput = null;
utilities.lazyLoad(exports, ["getAppConnection", "getAppConnectionOutput"], () => require("./getAppConnection"));
exports.getAppConnector = null;
exports.getAppConnectorOutput = null;
utilities.lazyLoad(exports, ["getAppConnector", "getAppConnectorOutput"], () => require("./getAppConnector"));
exports.getAppGateway = null;
exports.getAppGatewayOutput = null;
utilities.lazyLoad(exports, ["getAppGateway", "getAppGatewayOutput"], () => require("./getAppGateway"));
exports.getSecurityGateway = null;
exports.getSecurityGatewayOutput = null;
utilities.lazyLoad(exports, ["getSecurityGateway", "getSecurityGatewayOutput"], () => require("./getSecurityGateway"));
exports.getSecurityGatewayApplicationIamPolicy = null;
exports.getSecurityGatewayApplicationIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecurityGatewayApplicationIamPolicy", "getSecurityGatewayApplicationIamPolicyOutput"], () => require("./getSecurityGatewayApplicationIamPolicy"));
exports.getSecurityGatewayIamPolicy = null;
exports.getSecurityGatewayIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecurityGatewayIamPolicy", "getSecurityGatewayIamPolicyOutput"], () => require("./getSecurityGatewayIamPolicy"));
exports.SecurityGateway = null;
utilities.lazyLoad(exports, ["SecurityGateway"], () => require("./securityGateway"));
exports.SecurityGatewayApplication = null;
utilities.lazyLoad(exports, ["SecurityGatewayApplication"], () => require("./securityGatewayApplication"));
exports.SecurityGatewayApplicationIamBinding = null;
utilities.lazyLoad(exports, ["SecurityGatewayApplicationIamBinding"], () => require("./securityGatewayApplicationIamBinding"));
exports.SecurityGatewayApplicationIamMember = null;
utilities.lazyLoad(exports, ["SecurityGatewayApplicationIamMember"], () => require("./securityGatewayApplicationIamMember"));
exports.SecurityGatewayApplicationIamPolicy = null;
utilities.lazyLoad(exports, ["SecurityGatewayApplicationIamPolicy"], () => require("./securityGatewayApplicationIamPolicy"));
exports.SecurityGatewayIamBinding = null;
utilities.lazyLoad(exports, ["SecurityGatewayIamBinding"], () => require("./securityGatewayIamBinding"));
exports.SecurityGatewayIamMember = null;
utilities.lazyLoad(exports, ["SecurityGatewayIamMember"], () => require("./securityGatewayIamMember"));
exports.SecurityGatewayIamPolicy = null;
utilities.lazyLoad(exports, ["SecurityGatewayIamPolicy"], () => require("./securityGatewayIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:beyondcorp/appConnection:AppConnection":
                return new exports.AppConnection(name, undefined, { urn });
            case "gcp:beyondcorp/appConnector:AppConnector":
                return new exports.AppConnector(name, undefined, { urn });
            case "gcp:beyondcorp/appGateway:AppGateway":
                return new exports.AppGateway(name, undefined, { urn });
            case "gcp:beyondcorp/securityGateway:SecurityGateway":
                return new exports.SecurityGateway(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayApplication:SecurityGatewayApplication":
                return new exports.SecurityGatewayApplication(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayApplicationIamBinding:SecurityGatewayApplicationIamBinding":
                return new exports.SecurityGatewayApplicationIamBinding(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayApplicationIamMember:SecurityGatewayApplicationIamMember":
                return new exports.SecurityGatewayApplicationIamMember(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayApplicationIamPolicy:SecurityGatewayApplicationIamPolicy":
                return new exports.SecurityGatewayApplicationIamPolicy(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayIamBinding:SecurityGatewayIamBinding":
                return new exports.SecurityGatewayIamBinding(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayIamMember:SecurityGatewayIamMember":
                return new exports.SecurityGatewayIamMember(name, undefined, { urn });
            case "gcp:beyondcorp/securityGatewayIamPolicy:SecurityGatewayIamPolicy":
                return new exports.SecurityGatewayIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appConnection", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appConnector", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayApplication", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayApplicationIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayApplicationIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayApplicationIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/securityGatewayIamPolicy", _module);
