// @pulumi/gcp/beyondcorp/securityGatewayApplication.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGatewayApplication = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Specifies application endpoint(s) to protect behind a Security Gateway.
 *
 * ## Example Usage
 *
 * ### Beyondcorp Security Gateway Application Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.beyondcorp.SecurityGateway("default", {
 *     securityGatewayId: "default-sg",
 *     displayName: "My Security Gateway resource",
 *     hubs: [{
 *         region: "us-central1",
 *     }],
 * });
 * const example = new gcp.beyondcorp.SecurityGatewayApplication("example", {
 *     securityGatewayId: _default.securityGatewayId,
 *     applicationId: "google-sga",
 *     endpointMatchers: [{
 *         hostname: "google.com",
 *         ports: [
 *             80,
 *             443,
 *         ],
 *     }],
 * });
 * ```
 * ### Beyondcorp Security Gateway Application Vpc
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const _default = new gcp.beyondcorp.SecurityGateway("default", {
 *     securityGatewayId: "default-sg",
 *     displayName: "My Security Gateway resource",
 *     hubs: [{
 *         region: "us-central1",
 *     }],
 * });
 * const example = new gcp.beyondcorp.SecurityGatewayApplication("example", {
 *     securityGatewayId: _default.securityGatewayId,
 *     applicationId: "my-vm-service2",
 *     endpointMatchers: [{
 *         hostname: "my-vm-service.com",
 *         ports: [
 *             80,
 *             443,
 *         ],
 *     }],
 *     upstreams: [{
 *         egressPolicy: {
 *             regions: ["us-central1"],
 *         },
 *         network: {
 *             name: project.then(project => `projects/${project.projectId}/global/networks/default`),
 *         },
 *     }],
 * });
 * ```
 * ### Beyondcorp Security Gateway Application Spa Api
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.beyondcorp.SecurityGateway("default", {
 *     securityGatewayId: "default-sg-spa-api",
 *     displayName: "My SPA Security Gateway resource",
 * });
 * const example_spa = new gcp.beyondcorp.SecurityGatewayApplication("example-spa", {
 *     securityGatewayId: _default.securityGatewayId,
 *     applicationId: "app-discovery",
 *     upstreams: [{
 *         external: {
 *             endpoints: [{
 *                 hostname: "my.discovery.service.com",
 *                 port: 443,
 *             }],
 *         },
 *         proxyProtocol: {
 *             allowedClientHeaders: ["header"],
 *         },
 *     }],
 *     schema: "API_GATEWAY",
 * });
 * ```
 * ### Beyondcorp Security Gateway Application Spa Proxy
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.beyondcorp.SecurityGateway("default", {
 *     securityGatewayId: "default-sg-spa-proxy",
 *     displayName: "My SPA Security Gateway resource",
 * });
 * const example_spa = new gcp.beyondcorp.SecurityGatewayApplication("example-spa", {
 *     securityGatewayId: _default.securityGatewayId,
 *     applicationId: "app-proxy",
 *     endpointMatchers: [{
 *         hostname: "a.site.com",
 *         ports: [443],
 *     }],
 *     upstreams: [{
 *         external: {
 *             endpoints: [{
 *                 hostname: "my.proxy.service.com",
 *                 port: 443,
 *             }],
 *         },
 *         proxyProtocol: {
 *             allowedClientHeaders: [
 *                 "header1",
 *                 "header2",
 *             ],
 *             contextualHeaders: {
 *                 userInfo: {
 *                     outputType: "PROTOBUF",
 *                 },
 *                 groupInfo: {
 *                     outputType: "JSON",
 *                 },
 *                 deviceInfo: {
 *                     outputType: "NONE",
 *                 },
 *                 outputType: "JSON",
 *             },
 *             metadataHeaders: {
 *                 "metadata-header1": "value1",
 *                 "metadata-header2": "value2",
 *             },
 *             gatewayIdentity: "RESOURCE_NAME",
 *             clientIp: true,
 *         },
 *     }],
 *     schema: "PROXY_GATEWAY",
 * });
 * ```
 *
 * ## Import
 *
 * SecurityGatewayApplication can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/global/securityGateways/{{security_gateway_id}}/applications/{{application_id}}`
 * * `{{project}}/{{security_gateway_id}}/{{application_id}}`
 * * `{{security_gateway_id}}/{{application_id}}`
 *
 * When using the `pulumi import` command, SecurityGatewayApplication can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:beyondcorp/securityGatewayApplication:SecurityGatewayApplication default projects/{{project}}/locations/global/securityGateways/{{security_gateway_id}}/applications/{{application_id}}
 * $ pulumi import gcp:beyondcorp/securityGatewayApplication:SecurityGatewayApplication default {{project}}/{{security_gateway_id}}/{{application_id}}
 * $ pulumi import gcp:beyondcorp/securityGatewayApplication:SecurityGatewayApplication default {{security_gateway_id}}/{{application_id}}
 * ```
 */
class SecurityGatewayApplication extends pulumi.CustomResource {
    /**
     * Get an existing SecurityGatewayApplication resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SecurityGatewayApplication(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:beyondcorp/securityGatewayApplication:SecurityGatewayApplication';
    /**
     * Returns true if the given object is an instance of SecurityGatewayApplication.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityGatewayApplication.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applicationId"] = state?.applicationId;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["endpointMatchers"] = state?.endpointMatchers;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["schema"] = state?.schema;
            resourceInputs["securityGatewayId"] = state?.securityGatewayId;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["upstreams"] = state?.upstreams;
        }
        else {
            const args = argsOrState;
            if (args?.applicationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'applicationId'");
            }
            if (args?.securityGatewayId === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGatewayId'");
            }
            resourceInputs["applicationId"] = args?.applicationId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["endpointMatchers"] = args?.endpointMatchers;
            resourceInputs["project"] = args?.project;
            resourceInputs["schema"] = args?.schema;
            resourceInputs["securityGatewayId"] = args?.securityGatewayId;
            resourceInputs["upstreams"] = args?.upstreams;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecurityGatewayApplication.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecurityGatewayApplication = SecurityGatewayApplication;
