// @pulumi/gcp/beyondcorp/securityGateway.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGateway = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Deployment of Security Gateway.
 *
 * ## Example Usage
 *
 * ### Beyondcorp Security Gateway Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.beyondcorp.SecurityGateway("example", {
 *     securityGatewayId: "default",
 *     displayName: "My Security Gateway resource",
 *     hubs: [{
 *         region: "us-central1",
 *     }],
 * });
 * ```
 * ### Beyondcorp Security Gateway Logging
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example_logging = new gcp.beyondcorp.SecurityGateway("example-logging", {
 *     securityGatewayId: "default-logging",
 *     displayName: "My Security Gateway resource with logging enabled",
 *     hubs: [{
 *         region: "us-central1",
 *     }],
 *     logging: {},
 * });
 * ```
 * ### Beyondcorp Security Gateway Spa
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example_spa = new gcp.beyondcorp.SecurityGateway("example-spa", {
 *     securityGatewayId: "default-spa",
 *     displayName: "My SPA Security Gateway resource",
 *     proxyProtocolConfig: {
 *         allowedClientHeaders: [
 *             "header1",
 *             "header2",
 *         ],
 *         contextualHeaders: {
 *             userInfo: {
 *                 outputType: "PROTOBUF",
 *             },
 *             groupInfo: {
 *                 outputType: "JSON",
 *             },
 *             deviceInfo: {
 *                 outputType: "NONE",
 *             },
 *             outputType: "NONE",
 *         },
 *         metadataHeaders: {
 *             "metadata-header1": "value1",
 *             "metadata-header2": "value2",
 *         },
 *         gatewayIdentity: "RESOURCE_NAME",
 *         clientIp: true,
 *     },
 *     serviceDiscovery: {
 *         apiGateway: {
 *             resourceOverride: {
 *                 path: "/api/v1/routes",
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * SecurityGateway can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/securityGateways/{{security_gateway_id}}`
 * * `{{project}}/{{location}}/{{security_gateway_id}}`
 * * `{{location}}/{{security_gateway_id}}`
 *
 * When using the `pulumi import` command, SecurityGateway can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:beyondcorp/securityGateway:SecurityGateway default projects/{{project}}/locations/{{location}}/securityGateways/{{security_gateway_id}}
 * $ pulumi import gcp:beyondcorp/securityGateway:SecurityGateway default {{project}}/{{location}}/{{security_gateway_id}}
 * $ pulumi import gcp:beyondcorp/securityGateway:SecurityGateway default {{location}}/{{security_gateway_id}}
 * ```
 */
class SecurityGateway extends pulumi.CustomResource {
    /**
     * Get an existing SecurityGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SecurityGateway(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:beyondcorp/securityGateway:SecurityGateway';
    /**
     * Returns true if the given object is an instance of SecurityGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityGateway.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["delegatingServiceAccount"] = state?.delegatingServiceAccount;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["externalIps"] = state?.externalIps;
            resourceInputs["hubs"] = state?.hubs;
            resourceInputs["location"] = state?.location;
            resourceInputs["logging"] = state?.logging;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["proxyProtocolConfig"] = state?.proxyProtocolConfig;
            resourceInputs["securityGatewayId"] = state?.securityGatewayId;
            resourceInputs["serviceDiscovery"] = state?.serviceDiscovery;
            resourceInputs["state"] = state?.state;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.securityGatewayId === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGatewayId'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["hubs"] = args?.hubs;
            resourceInputs["location"] = args?.location;
            resourceInputs["logging"] = args?.logging;
            resourceInputs["project"] = args?.project;
            resourceInputs["proxyProtocolConfig"] = args?.proxyProtocolConfig;
            resourceInputs["securityGatewayId"] = args?.securityGatewayId;
            resourceInputs["serviceDiscovery"] = args?.serviceDiscovery;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["delegatingServiceAccount"] = undefined /*out*/;
            resourceInputs["externalIps"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecurityGateway.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecurityGateway = SecurityGateway;
