// @pulumi/azure-native/types/enums/azurestackhci/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.State = exports.SoftwareAssuranceIntent = exports.ServiceName = exports.SecurityTypes = exports.SecurityRuleProtocol = exports.SecurityRuleDirection = exports.SecurityRuleAccess = exports.SecurityEncryptionType = exports.SecretType = exports.SdnIntegrationIntent = exports.ResourceIdentityType = exports.RemoteSupportType = exports.RemoteSupportAccessLevel = exports.PublicIPAddressType = exports.ProvisioningOsType = exports.ProvisioningAction = exports.OwnerKeyType = exports.OsTypeEnum = exports.OperationType = exports.OperatingSystemTypes = exports.OnboardingResourceType = exports.OSOperationType = exports.NetworkTypeEnum = exports.ManagedServiceIdentityType = exports.LoadBalancerRuleTransportProtocol = exports.LoadBalancerRuleSessionPersistenceType = exports.LoadBalancerProbeProtocol = exports.LoadBalancerBackendAddressAdminState = exports.IpAssignmentType = exports.IpAllocationMethodEnum = exports.InboundNATRuleProtocol = exports.IPPoolTypeEnum = exports.HyperVGeneration = exports.HciJobType = exports.HciEdgeDeviceJobType = exports.GpuAssignmentTypeEnum = exports.ExtendedLocationTypes = exports.EdgeMachineKind = exports.EdgeMachineJobType = exports.EdgeDeviceKind = exports.EceSecrets = exports.DiskFileFormat = exports.DiskCreateOption = exports.DiagnosticLevel = exports.DeviceKind = exports.DeploymentMode = exports.ConfidentialVmIntent = exports.ComplianceAssignmentType = exports.CloudInitDataSource = exports.AvailabilityType = void 0;
exports.WindowsServerSubscription = exports.VmSizeEnum = exports.UpdateSummariesPropertiesState = exports.UpdateRunPropertiesState = void 0;
exports.AvailabilityType = {
    Local: "Local",
    Online: "Online",
    Notify: "Notify",
};
exports.CloudInitDataSource = {
    /**
     * NoCloud is used as the datasource
     */
    NoCloud: "NoCloud",
    /**
     * Azure is used as the datasource
     */
    Azure: "Azure",
};
exports.ComplianceAssignmentType = {
    /**
     * Report on the state of the machine, but don't make changes.
     */
    Audit: "Audit",
    /**
     * Applied to the machine. If it drifts, the local service inside the machine makes a correction at the next evaluation.
     */
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
};
exports.ConfidentialVmIntent = {
    /**
     * Indicates that the customer intends to enable CVM support on the cluster.
     */
    Enable: "Enable",
    /**
     * Indicates that the customer intends to disable CVM support on the cluster.
     */
    Disable: "Disable",
};
exports.DeploymentMode = {
    /**
     * Validate ECE action deployment for a cluster.
     */
    Validate: "Validate",
    /**
     * Deploy ECE action deployment for a cluster.
     */
    Deploy: "Deploy",
};
exports.DeviceKind = {
    /**
     * Arc-enabled edge device with HCI OS.
     */
    HCI: "HCI",
};
exports.DiagnosticLevel = {
    Off: "Off",
    Basic: "Basic",
    Enhanced: "Enhanced",
};
exports.DiskCreateOption = {
    /**
     * Create a disk by copying from a source resource
     */
    Copy: "Copy",
    /**
     * Create an empty disk
     */
    Empty: "Empty",
};
exports.DiskFileFormat = {
    /**
     * VHDX file format
     */
    Vhdx: "vhdx",
    /**
     * VHD file format
     */
    Vhd: "vhd",
};
exports.EceSecrets = {
    /**
     * AzureStackLCMUserCredential used for LCM operations for AzureStackHCI cluster.
     */
    AzureStackLCMUserCredential: "AzureStackLCMUserCredential",
    /**
     * DefaultARBApplication used to manage Azure Arc resource bridge (ARB) for AzureStackHCI cluster.
     */
    DefaultARBApplication: "DefaultARBApplication",
    /**
     * LocalAdminCredential used for admin operations for AzureStackHCI cluster.
     */
    LocalAdminCredential: "LocalAdminCredential",
    /**
     * WitnessStorageKey used for setting up a cloud witness for AzureStackHCI cluster.
     */
    WitnessStorageKey: "WitnessStorageKey",
};
exports.EdgeDeviceKind = {
    /**
     * Arc-enabled edge device with HCI OS.
     */
    HCI: "HCI",
};
exports.EdgeMachineJobType = {
    /**
     * Job to collect logs from the device.
     */
    CollectLog: "CollectLog",
    /**
     * Job to provide remote support to the device.
     */
    RemoteSupport: "RemoteSupport",
    /**
     * Job to provision operating system in the device.
     */
    ProvisionOs: "ProvisionOs",
    /**
     * Job to download OS packages on to the device
     */
    DownloadOs: "DownloadOs",
};
exports.EdgeMachineKind = {
    /**
     * EdgeMachine resource created using Zero-touch provisioning.
     */
    Standard: "Standard",
    /**
     * EdgeMachine resource created for brownfield HCI customers without zero touch provisioning.
     */
    Dedicated: "Dedicated",
};
exports.ExtendedLocationTypes = {
    /**
     * Custom extended location type
     */
    CustomLocation: "CustomLocation",
};
exports.GpuAssignmentTypeEnum = {
    /**
     * Attach Graphics Processing Unit (GPU) using Discrete Device Assignment (DDA)
     */
    GpuDDA: "GpuDDA",
    /**
     * Attach Graphics Processing Unit (GPU) using GPU Partitioning
     */
    GpuP: "GpuP",
};
exports.HciEdgeDeviceJobType = {
    /**
     * Job to collect logs from the device.
     */
    CollectLog: "CollectLog",
    /**
     * Job to provide remote support to the device.
     */
    RemoteSupport: "RemoteSupport",
};
exports.HciJobType = {
    /**
     * Job to CVM  intent for the cluster.
     */
    ConfigureCVM: "ConfigureCVM",
    /**
     * Job to configure SDN (Software Defined Networking) integration for the cluster.
     */
    ConfigureSdnIntegration: "ConfigureSdnIntegration",
};
exports.HyperVGeneration = {
    /**
     * Generation 1 (V1) hypervisor
     */
    V1: "V1",
    /**
     * Generation 2 (V2) hypervisor
     */
    V2: "V2",
};
exports.IPPoolTypeEnum = {
    /**
     * Virtual Machine IP Pool
     */
    Vm: "vm",
    /**
     * VIP Pool
     */
    Vippool: "vippool",
};
exports.InboundNATRuleProtocol = {
    /**
     * TCP - load balance only tcp traffic
     */
    TCP: "Tcp",
    /**
     * Udp - load balance only UDP traffic
     */
    UDP: "Udp",
};
exports.IpAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.IpAssignmentType = {
    /**
     * Automatic IP assignment
     */
    Automatic: "Automatic",
    /**
     * Manual IP assignment
     */
    Manual: "Manual",
};
exports.LoadBalancerBackendAddressAdminState = {
    /**
     * Up - forced admin state up
     */
    Up: "Up",
    /**
     * Down - forced admin state down
     */
    Down: "Down",
};
exports.LoadBalancerProbeProtocol = {
    /**
     * TCP - TCP port checking
     */
    TCP: "Tcp",
    /**
     * HTTP - HTTP request
     */
    HTTP: "Http",
};
exports.LoadBalancerRuleSessionPersistenceType = {
    /**
     * Default - 5-tuple hashing
     */
    Default: "Default",
    /**
     * Source IP - 2-tuple hashing looking at src-dst ip
     */
    SourceIP: "SourceIP",
    /**
     * SourceIPProtocol - 3-tuple hashing looking at src-dst ip and ip protocol
     */
    SourceIPProtocol: "SourceIPProtocol",
};
exports.LoadBalancerRuleTransportProtocol = {
    /**
     * TCP - load balance only tcp traffic
     */
    TCP: "Tcp",
    /**
     * Udp - load balance only UDP traffic
     */
    UDP: "Udp",
    /**
     * All - load balance all UDP and TCP traffic
     */
    All: "All",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NetworkTypeEnum = {
    NAT: "NAT",
    Transparent: "Transparent",
    L2Bridge: "L2Bridge",
    L2Tunnel: "L2Tunnel",
    ICS: "ICS",
    Private: "Private",
    Overlay: "Overlay",
    Internal: "Internal",
    Mirrored: "Mirrored",
};
exports.OSOperationType = {
    /**
     * OS Provisioning operation
     */
    Provision: "Provision",
    /**
     * OS Update operation
     */
    Update: "Update",
    /**
     * OS ReImage operation
     */
    ReImage: "ReImage",
};
exports.OnboardingResourceType = {
    /**
     * Hybrid Compute Machine.
     */
    HybridComputeMachine: "HybridComputeMachine",
};
exports.OperatingSystemTypes = {
    /**
     * Windows operating system
     */
    Windows: "Windows",
    /**
     * Linux operating system
     */
    Linux: "Linux",
};
exports.OperationType = {
    /**
     * Cluster provisioning operation.
     */
    ClusterProvisioning: "ClusterProvisioning",
    /**
     * Cluster upgrade operation.
     */
    ClusterUpgrade: "ClusterUpgrade",
};
exports.OsTypeEnum = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OwnerKeyType = {
    /**
     * Owner is Microsoft managed key
     */
    MicrosoftManaged: "MicrosoftManaged",
};
exports.ProvisioningAction = {
    /**
     * Install guest agent
     */
    Install: "install",
    /**
     * Uninstall guest agent
     */
    Uninstall: "uninstall",
    /**
     * Repair guest agent
     */
    Repair: "repair",
};
exports.ProvisioningOsType = {
    /**
     * AzureLinux OS.
     */
    AzureLinux: "AzureLinux",
    /**
     * HCI OS.
     */
    HCI: "HCI",
};
exports.PublicIPAddressType = {
    /**
     * IPv4 IP Address
     */
    IPv4: "IPv4",
    /**
     * IPv6 IP Address
     */
    IPv6: "IPv6",
};
exports.RemoteSupportAccessLevel = {
    /**
     * No remote support access is granted.
     */
    None: "None",
    /**
     * Access is limited to diagnostics information only.
     */
    Diagnostics: "Diagnostics",
    /**
     * Access includes diagnostics information and the ability to perform repairs.
     */
    DiagnosticsAndRepair: "DiagnosticsAndRepair",
};
exports.RemoteSupportType = {
    /**
     * Enable remote support for the cluster.
     */
    Enable: "Enable",
    /**
     * Revoke previously enabled remote support for the cluster.
     */
    Revoke: "Revoke",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SdnIntegrationIntent = {
    /**
     * Enable SDN integration for the deployment.
     */
    Enable: "Enable",
    /**
     * Disable SDN integration for the deployment.
     */
    Disable: "Disable",
};
exports.SecretType = {
    /**
     * Key Vault based authentication
     */
    KeyVault: "KeyVault",
    /**
     * SSH Public Key based authentication
     */
    SshPubKey: "SshPubKey",
};
exports.SecurityEncryptionType = {
    /**
     * Non-persisted TPM encryption type
     */
    NonPersistedTPM: "NonPersistedTPM",
};
exports.SecurityRuleAccess = {
    /**
     * Network traffic is allowed
     */
    Allow: "Allow",
    /**
     * Network traffic is denied
     */
    Deny: "Deny",
};
exports.SecurityRuleDirection = {
    /**
     * Rule is evaluated on incoming traffic
     */
    Inbound: "Inbound",
    /**
     * Rule is evaluated on outgoing traffic
     */
    Outbound: "Outbound",
};
exports.SecurityRuleProtocol = {
    /**
     * Transmission Control Protocol
     */
    Tcp: "Tcp",
    /**
     * User Datagram Protocol
     */
    Udp: "Udp",
    /**
     * Internet Control Message Protocol
     */
    Icmp: "Icmp",
    /**
     * Wildcard rule for all protocols
     */
    Asterisk: "*",
};
exports.SecurityTypes = {
    /**
     * Trusted Launch security type
     */
    TrustedLaunch: "TrustedLaunch",
    /**
     * Confidential VM security type
     */
    ConfidentialVM: "ConfidentialVM",
};
exports.ServiceName = {
    WAC: "WAC",
};
exports.SoftwareAssuranceIntent = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.State = {
    HasPrerequisite: "HasPrerequisite",
    Obsolete: "Obsolete",
    Ready: "Ready",
    NotApplicableBecauseAnotherUpdateIsInProgress: "NotApplicableBecauseAnotherUpdateIsInProgress",
    Preparing: "Preparing",
    Installing: "Installing",
    Installed: "Installed",
    PreparationFailed: "PreparationFailed",
    InstallationFailed: "InstallationFailed",
    Invalid: "Invalid",
    Recalled: "Recalled",
    Downloading: "Downloading",
    DownloadFailed: "DownloadFailed",
    HealthChecking: "HealthChecking",
    HealthCheckFailed: "HealthCheckFailed",
    ReadyToInstall: "ReadyToInstall",
    ScanInProgress: "ScanInProgress",
    ScanFailed: "ScanFailed",
    AdditionalContentRequired: "AdditionalContentRequired",
};
exports.UpdateRunPropertiesState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    InProgress: "InProgress",
    Failed: "Failed",
};
exports.UpdateSummariesPropertiesState = {
    Unknown: "Unknown",
    AppliedSuccessfully: "AppliedSuccessfully",
    UpdateAvailable: "UpdateAvailable",
    UpdateInProgress: "UpdateInProgress",
    UpdateFailed: "UpdateFailed",
    NeedsAttention: "NeedsAttention",
    PreparationInProgress: "PreparationInProgress",
    PreparationFailed: "PreparationFailed",
};
exports.VmSizeEnum = {
    /**
     * Default virtual machine size
     */
    Default: "Default",
    /**
     * Standard A2 v2 virtual machine size
     */
    Standard_A2_v2: "Standard_A2_v2",
    /**
     * Standard A4 v2 virtual machine size
     */
    Standard_A4_v2: "Standard_A4_v2",
    /**
     * Standard D2s v3 virtual machine size
     */
    Standard_D2s_v3: "Standard_D2s_v3",
    /**
     * Standard D4s v3 virtual machine size
     */
    Standard_D4s_v3: "Standard_D4s_v3",
    /**
     * Standard D8s v3 virtual machine size
     */
    Standard_D8s_v3: "Standard_D8s_v3",
    /**
     * Standard D16s v3 virtual machine size
     */
    Standard_D16s_v3: "Standard_D16s_v3",
    /**
     * Standard D32s v3 virtual machine size
     */
    Standard_D32s_v3: "Standard_D32s_v3",
    /**
     * Standard DS2 v2 virtual machine size
     */
    Standard_DS2_v2: "Standard_DS2_v2",
    /**
     * Standard DS3 v2 virtual machine size
     */
    Standard_DS3_v2: "Standard_DS3_v2",
    /**
     * Standard DS4 v2 virtual machine size
     */
    Standard_DS4_v2: "Standard_DS4_v2",
    /**
     * Standard DS5 v2 virtual machine size
     */
    Standard_DS5_v2: "Standard_DS5_v2",
    /**
     * Standard DS13 v2 virtual machine size
     */
    Standard_DS13_v2: "Standard_DS13_v2",
    /**
     * Standard K8S v1 virtual machine size
     */
    Standard_K8S_v1: "Standard_K8S_v1",
    /**
     * Standard K8S2 v1 virtual machine size
     */
    Standard_K8S2_v1: "Standard_K8S2_v1",
    /**
     * Standard K8S3 v1 virtual machine size
     */
    Standard_K8S3_v1: "Standard_K8S3_v1",
    /**
     * Standard K8S4 v1 virtual machine size
     */
    Standard_K8S4_v1: "Standard_K8S4_v1",
    /**
     * Standard NK6 virtual machine size
     */
    Standard_NK6: "Standard_NK6",
    /**
     * Standard NK12 virtual machine size
     */
    Standard_NK12: "Standard_NK12",
    /**
     * Standard NV6 virtual machine size
     */
    Standard_NV6: "Standard_NV6",
    /**
     * Standard NV12 virtual machine size
     */
    Standard_NV12: "Standard_NV12",
    /**
     * Standard K8S5 v1 virtual machine size
     */
    Standard_K8S5_v1: "Standard_K8S5_v1",
    /**
     * Custom virtual machine size
     */
    Custom: "Custom",
};
exports.WindowsServerSubscription = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
