// @pulumi/azure-native/containerservice/agentPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentPool = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * Agent Pool.
 *
 * Uses Azure REST API version 2025-09-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2019-11-01, 2020-01-01, 2020-02-01, 2020-03-01, 2020-04-01, 2020-06-01, 2020-07-01, 2020-09-01, 2020-11-01, 2020-12-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-07-01, 2021-08-01, 2021-09-01, 2021-10-01, 2021-11-01-preview, 2022-01-01, 2022-01-02-preview, 2022-02-01, 2022-02-02-preview, 2022-03-01, 2022-03-02-preview, 2022-04-01, 2022-04-02-preview, 2022-05-02-preview, 2022-06-01, 2022-06-02-preview, 2022-07-01, 2022-07-02-preview, 2022-08-02-preview, 2022-08-03-preview, 2022-09-01, 2022-09-02-preview, 2022-10-02-preview, 2022-11-01, 2022-11-02-preview, 2023-01-01, 2023-01-02-preview, 2023-02-01, 2023-02-02-preview, 2023-03-01, 2023-03-02-preview, 2023-04-01, 2023-04-02-preview, 2023-05-01, 2023-05-02-preview, 2023-06-01, 2023-06-02-preview, 2023-07-01, 2023-07-02-preview, 2023-08-01, 2023-08-02-preview, 2023-09-01, 2023-09-02-preview, 2023-10-01, 2023-10-02-preview, 2023-11-01, 2023-11-02-preview, 2024-01-01, 2024-01-02-preview, 2024-02-01, 2024-02-02-preview, 2024-03-02-preview, 2024-04-02-preview, 2024-05-01, 2024-05-02-preview, 2024-06-02-preview, 2024-07-01, 2024-07-02-preview, 2024-08-01, 2024-09-01, 2024-09-02-preview, 2024-10-01, 2024-10-02-preview, 2025-01-01, 2025-01-02-preview, 2025-02-01, 2025-02-02-preview, 2025-03-01, 2025-03-02-preview, 2025-04-01, 2025-04-02-preview, 2025-05-01, 2025-05-02-preview, 2025-06-02-preview, 2025-07-01, 2025-07-02-preview, 2025-08-01, 2025-08-02-preview, 2025-09-02-preview, 2025-10-01, 2025-10-02-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native containerservice [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AgentPool extends pulumi.CustomResource {
    /**
     * Get an existing AgentPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AgentPool(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:containerservice:AgentPool';
    /**
     * Returns true if the given object is an instance of AgentPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AgentPool.__pulumiType;
    }
    /**
     * Create a AgentPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.resourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["agentPoolName"] = args?.agentPoolName;
            resourceInputs["availabilityZones"] = args?.availabilityZones;
            resourceInputs["capacityReservationGroupID"] = args?.capacityReservationGroupID;
            resourceInputs["count"] = args?.count;
            resourceInputs["creationData"] = args?.creationData;
            resourceInputs["enableAutoScaling"] = args?.enableAutoScaling;
            resourceInputs["enableEncryptionAtHost"] = args?.enableEncryptionAtHost;
            resourceInputs["enableFIPS"] = args?.enableFIPS;
            resourceInputs["enableNodePublicIP"] = args?.enableNodePublicIP;
            resourceInputs["enableUltraSSD"] = args?.enableUltraSSD;
            resourceInputs["gatewayProfile"] = args ? (args.gatewayProfile ? pulumi.output(args.gatewayProfile).apply(inputs.containerservice.agentPoolGatewayProfileArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["gpuInstanceProfile"] = args?.gpuInstanceProfile;
            resourceInputs["gpuProfile"] = args?.gpuProfile;
            resourceInputs["hostGroupID"] = args?.hostGroupID;
            resourceInputs["kubeletConfig"] = args?.kubeletConfig;
            resourceInputs["kubeletDiskType"] = args?.kubeletDiskType;
            resourceInputs["linuxOSConfig"] = args?.linuxOSConfig;
            resourceInputs["localDNSProfile"] = args ? (args.localDNSProfile ? pulumi.output(args.localDNSProfile).apply(inputs.containerservice.localDNSProfileArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["maxCount"] = args?.maxCount;
            resourceInputs["maxPods"] = args?.maxPods;
            resourceInputs["messageOfTheDay"] = args?.messageOfTheDay;
            resourceInputs["minCount"] = args?.minCount;
            resourceInputs["mode"] = args?.mode;
            resourceInputs["networkProfile"] = args?.networkProfile;
            resourceInputs["nodeLabels"] = args?.nodeLabels;
            resourceInputs["nodePublicIPPrefixID"] = args?.nodePublicIPPrefixID;
            resourceInputs["nodeTaints"] = args?.nodeTaints;
            resourceInputs["orchestratorVersion"] = args?.orchestratorVersion;
            resourceInputs["osDiskSizeGB"] = args?.osDiskSizeGB;
            resourceInputs["osDiskType"] = args?.osDiskType;
            resourceInputs["osSKU"] = args?.osSKU;
            resourceInputs["osType"] = args?.osType;
            resourceInputs["podIPAllocationMode"] = args?.podIPAllocationMode;
            resourceInputs["podSubnetID"] = args?.podSubnetID;
            resourceInputs["powerState"] = args?.powerState;
            resourceInputs["proximityPlacementGroupID"] = args?.proximityPlacementGroupID;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceName"] = args?.resourceName;
            resourceInputs["scaleDownMode"] = args?.scaleDownMode;
            resourceInputs["scaleSetEvictionPolicy"] = args?.scaleSetEvictionPolicy;
            resourceInputs["scaleSetPriority"] = args?.scaleSetPriority;
            resourceInputs["securityProfile"] = args?.securityProfile;
            resourceInputs["spotMaxPrice"] = args?.spotMaxPrice;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["type"] = args?.type;
            resourceInputs["upgradeSettings"] = args?.upgradeSettings;
            resourceInputs["virtualMachineNodesStatus"] = args?.virtualMachineNodesStatus;
            resourceInputs["virtualMachinesProfile"] = args?.virtualMachinesProfile;
            resourceInputs["vmSize"] = args?.vmSize;
            resourceInputs["vnetSubnetID"] = args?.vnetSubnetID;
            resourceInputs["windowsProfile"] = args?.windowsProfile;
            resourceInputs["workloadRuntime"] = args?.workloadRuntime;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["currentOrchestratorVersion"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeImageVersion"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["availabilityZones"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["capacityReservationGroupID"] = undefined /*out*/;
            resourceInputs["count"] = undefined /*out*/;
            resourceInputs["creationData"] = undefined /*out*/;
            resourceInputs["currentOrchestratorVersion"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["enableAutoScaling"] = undefined /*out*/;
            resourceInputs["enableEncryptionAtHost"] = undefined /*out*/;
            resourceInputs["enableFIPS"] = undefined /*out*/;
            resourceInputs["enableNodePublicIP"] = undefined /*out*/;
            resourceInputs["enableUltraSSD"] = undefined /*out*/;
            resourceInputs["gatewayProfile"] = undefined /*out*/;
            resourceInputs["gpuInstanceProfile"] = undefined /*out*/;
            resourceInputs["gpuProfile"] = undefined /*out*/;
            resourceInputs["hostGroupID"] = undefined /*out*/;
            resourceInputs["kubeletConfig"] = undefined /*out*/;
            resourceInputs["kubeletDiskType"] = undefined /*out*/;
            resourceInputs["linuxOSConfig"] = undefined /*out*/;
            resourceInputs["localDNSProfile"] = undefined /*out*/;
            resourceInputs["maxCount"] = undefined /*out*/;
            resourceInputs["maxPods"] = undefined /*out*/;
            resourceInputs["messageOfTheDay"] = undefined /*out*/;
            resourceInputs["minCount"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkProfile"] = undefined /*out*/;
            resourceInputs["nodeImageVersion"] = undefined /*out*/;
            resourceInputs["nodeLabels"] = undefined /*out*/;
            resourceInputs["nodePublicIPPrefixID"] = undefined /*out*/;
            resourceInputs["nodeTaints"] = undefined /*out*/;
            resourceInputs["orchestratorVersion"] = undefined /*out*/;
            resourceInputs["osDiskSizeGB"] = undefined /*out*/;
            resourceInputs["osDiskType"] = undefined /*out*/;
            resourceInputs["osSKU"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["podIPAllocationMode"] = undefined /*out*/;
            resourceInputs["podSubnetID"] = undefined /*out*/;
            resourceInputs["powerState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["proximityPlacementGroupID"] = undefined /*out*/;
            resourceInputs["scaleDownMode"] = undefined /*out*/;
            resourceInputs["scaleSetEvictionPolicy"] = undefined /*out*/;
            resourceInputs["scaleSetPriority"] = undefined /*out*/;
            resourceInputs["securityProfile"] = undefined /*out*/;
            resourceInputs["spotMaxPrice"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upgradeSettings"] = undefined /*out*/;
            resourceInputs["virtualMachineNodesStatus"] = undefined /*out*/;
            resourceInputs["virtualMachinesProfile"] = undefined /*out*/;
            resourceInputs["vmSize"] = undefined /*out*/;
            resourceInputs["vnetSubnetID"] = undefined /*out*/;
            resourceInputs["windowsProfile"] = undefined /*out*/;
            resourceInputs["workloadRuntime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerservice/v20190201:AgentPool" }, { type: "azure-native:containerservice/v20190401:AgentPool" }, { type: "azure-native:containerservice/v20190601:AgentPool" }, { type: "azure-native:containerservice/v20190801:AgentPool" }, { type: "azure-native:containerservice/v20191001:AgentPool" }, { type: "azure-native:containerservice/v20191101:AgentPool" }, { type: "azure-native:containerservice/v20200101:AgentPool" }, { type: "azure-native:containerservice/v20200201:AgentPool" }, { type: "azure-native:containerservice/v20200301:AgentPool" }, { type: "azure-native:containerservice/v20200401:AgentPool" }, { type: "azure-native:containerservice/v20200601:AgentPool" }, { type: "azure-native:containerservice/v20200701:AgentPool" }, { type: "azure-native:containerservice/v20200901:AgentPool" }, { type: "azure-native:containerservice/v20201101:AgentPool" }, { type: "azure-native:containerservice/v20201201:AgentPool" }, { type: "azure-native:containerservice/v20210201:AgentPool" }, { type: "azure-native:containerservice/v20210301:AgentPool" }, { type: "azure-native:containerservice/v20210501:AgentPool" }, { type: "azure-native:containerservice/v20210701:AgentPool" }, { type: "azure-native:containerservice/v20210801:AgentPool" }, { type: "azure-native:containerservice/v20210901:AgentPool" }, { type: "azure-native:containerservice/v20211001:AgentPool" }, { type: "azure-native:containerservice/v20211101preview:AgentPool" }, { type: "azure-native:containerservice/v20220101:AgentPool" }, { type: "azure-native:containerservice/v20220102preview:AgentPool" }, { type: "azure-native:containerservice/v20220201:AgentPool" }, { type: "azure-native:containerservice/v20220202preview:AgentPool" }, { type: "azure-native:containerservice/v20220301:AgentPool" }, { type: "azure-native:containerservice/v20220302preview:AgentPool" }, { type: "azure-native:containerservice/v20220401:AgentPool" }, { type: "azure-native:containerservice/v20220402preview:AgentPool" }, { type: "azure-native:containerservice/v20220502preview:AgentPool" }, { type: "azure-native:containerservice/v20220601:AgentPool" }, { type: "azure-native:containerservice/v20220602preview:AgentPool" }, { type: "azure-native:containerservice/v20220701:AgentPool" }, { type: "azure-native:containerservice/v20220702preview:AgentPool" }, { type: "azure-native:containerservice/v20220802preview:AgentPool" }, { type: "azure-native:containerservice/v20220803preview:AgentPool" }, { type: "azure-native:containerservice/v20220901:AgentPool" }, { type: "azure-native:containerservice/v20220902preview:AgentPool" }, { type: "azure-native:containerservice/v20221002preview:AgentPool" }, { type: "azure-native:containerservice/v20221101:AgentPool" }, { type: "azure-native:containerservice/v20221102preview:AgentPool" }, { type: "azure-native:containerservice/v20230101:AgentPool" }, { type: "azure-native:containerservice/v20230102preview:AgentPool" }, { type: "azure-native:containerservice/v20230201:AgentPool" }, { type: "azure-native:containerservice/v20230202preview:AgentPool" }, { type: "azure-native:containerservice/v20230301:AgentPool" }, { type: "azure-native:containerservice/v20230302preview:AgentPool" }, { type: "azure-native:containerservice/v20230401:AgentPool" }, { type: "azure-native:containerservice/v20230402preview:AgentPool" }, { type: "azure-native:containerservice/v20230501:AgentPool" }, { type: "azure-native:containerservice/v20230502preview:AgentPool" }, { type: "azure-native:containerservice/v20230601:AgentPool" }, { type: "azure-native:containerservice/v20230602preview:AgentPool" }, { type: "azure-native:containerservice/v20230701:AgentPool" }, { type: "azure-native:containerservice/v20230702preview:AgentPool" }, { type: "azure-native:containerservice/v20230801:AgentPool" }, { type: "azure-native:containerservice/v20230802preview:AgentPool" }, { type: "azure-native:containerservice/v20230901:AgentPool" }, { type: "azure-native:containerservice/v20230902preview:AgentPool" }, { type: "azure-native:containerservice/v20231001:AgentPool" }, { type: "azure-native:containerservice/v20231002preview:AgentPool" }, { type: "azure-native:containerservice/v20231101:AgentPool" }, { type: "azure-native:containerservice/v20231102preview:AgentPool" }, { type: "azure-native:containerservice/v20240101:AgentPool" }, { type: "azure-native:containerservice/v20240102preview:AgentPool" }, { type: "azure-native:containerservice/v20240201:AgentPool" }, { type: "azure-native:containerservice/v20240202preview:AgentPool" }, { type: "azure-native:containerservice/v20240302preview:AgentPool" }, { type: "azure-native:containerservice/v20240402preview:AgentPool" }, { type: "azure-native:containerservice/v20240501:AgentPool" }, { type: "azure-native:containerservice/v20240502preview:AgentPool" }, { type: "azure-native:containerservice/v20240602preview:AgentPool" }, { type: "azure-native:containerservice/v20240701:AgentPool" }, { type: "azure-native:containerservice/v20240702preview:AgentPool" }, { type: "azure-native:containerservice/v20240801:AgentPool" }, { type: "azure-native:containerservice/v20240901:AgentPool" }, { type: "azure-native:containerservice/v20240902preview:AgentPool" }, { type: "azure-native:containerservice/v20241001:AgentPool" }, { type: "azure-native:containerservice/v20241002preview:AgentPool" }, { type: "azure-native:containerservice/v20250101:AgentPool" }, { type: "azure-native:containerservice/v20250102preview:AgentPool" }, { type: "azure-native:containerservice/v20250201:AgentPool" }, { type: "azure-native:containerservice/v20250202preview:AgentPool" }, { type: "azure-native:containerservice/v20250301:AgentPool" }, { type: "azure-native:containerservice/v20250302preview:AgentPool" }, { type: "azure-native:containerservice/v20250401:AgentPool" }, { type: "azure-native:containerservice/v20250402preview:AgentPool" }, { type: "azure-native:containerservice/v20250501:AgentPool" }, { type: "azure-native:containerservice/v20250502preview:AgentPool" }, { type: "azure-native:containerservice/v20250602preview:AgentPool" }, { type: "azure-native:containerservice/v20250701:AgentPool" }, { type: "azure-native:containerservice/v20250702preview:AgentPool" }, { type: "azure-native:containerservice/v20250801:AgentPool" }, { type: "azure-native:containerservice/v20250802preview:AgentPool" }, { type: "azure-native:containerservice/v20250901:AgentPool" }, { type: "azure-native:containerservice/v20250902preview:AgentPool" }, { type: "azure-native:containerservice/v20251001:AgentPool" }, { type: "azure-native:containerservice/v20251002preview:AgentPool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AgentPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AgentPool = AgentPool;
