// @pulumi/aws/redshiftserverless/customDomainAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomDomainAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS Redshift Serverless Custom Domain Association.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.acm.Certificate("example", {domainName: "example.com"});
 * const exampleNamespace = new aws.redshiftserverless.Namespace("example", {namespaceName: "example-namespace"});
 * const exampleWorkgroup = new aws.redshiftserverless.Workgroup("example", {
 *     workgroupName: "example-workgroup",
 *     namespaceName: exampleNamespace.namespaceName,
 * });
 * const exampleCustomDomainAssociation = new aws.redshiftserverless.CustomDomainAssociation("example", {
 *     workgroupName: exampleWorkgroup.workgroupName,
 *     customDomainName: "example.com",
 *     customDomainCertificateArn: example.arn,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Redshift Serverless Custom Domain Association using the `workgroupName` and `customDomainName`, separated by the coma. For example:
 *
 * ```sh
 * $ pulumi import aws:redshiftserverless/customDomainAssociation:CustomDomainAssociation example example-workgroup,example.com
 * ```
 */
class CustomDomainAssociation extends pulumi.CustomResource {
    /**
     * Get an existing CustomDomainAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomDomainAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:redshiftserverless/customDomainAssociation:CustomDomainAssociation';
    /**
     * Returns true if the given object is an instance of CustomDomainAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomDomainAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customDomainCertificateArn"] = state?.customDomainCertificateArn;
            resourceInputs["customDomainCertificateExpiryTime"] = state?.customDomainCertificateExpiryTime;
            resourceInputs["customDomainName"] = state?.customDomainName;
            resourceInputs["region"] = state?.region;
            resourceInputs["workgroupName"] = state?.workgroupName;
        }
        else {
            const args = argsOrState;
            if (args?.customDomainCertificateArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'customDomainCertificateArn'");
            }
            if (args?.customDomainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'customDomainName'");
            }
            if (args?.workgroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workgroupName'");
            }
            resourceInputs["customDomainCertificateArn"] = args?.customDomainCertificateArn;
            resourceInputs["customDomainName"] = args?.customDomainName;
            resourceInputs["region"] = args?.region;
            resourceInputs["workgroupName"] = args?.workgroupName;
            resourceInputs["customDomainCertificateExpiryTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomDomainAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomDomainAssociation = CustomDomainAssociation;
