// @pulumi/aws/redshiftserverless/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workgroup = exports.UsageLimit = exports.Snapshot = exports.ResourcePolicy = exports.Namespace = exports.getWorkgroupOutput = exports.getWorkgroup = exports.getNamespaceOutput = exports.getNamespace = exports.getCredentialsOutput = exports.getCredentials = exports.EndpointAccess = exports.CustomDomainAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CustomDomainAssociation = null;
utilities.lazyLoad(exports, ["CustomDomainAssociation"], () => require("./customDomainAssociation"));
exports.EndpointAccess = null;
utilities.lazyLoad(exports, ["EndpointAccess"], () => require("./endpointAccess"));
exports.getCredentials = null;
exports.getCredentialsOutput = null;
utilities.lazyLoad(exports, ["getCredentials", "getCredentialsOutput"], () => require("./getCredentials"));
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getWorkgroup = null;
exports.getWorkgroupOutput = null;
utilities.lazyLoad(exports, ["getWorkgroup", "getWorkgroupOutput"], () => require("./getWorkgroup"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.UsageLimit = null;
utilities.lazyLoad(exports, ["UsageLimit"], () => require("./usageLimit"));
exports.Workgroup = null;
utilities.lazyLoad(exports, ["Workgroup"], () => require("./workgroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:redshiftserverless/customDomainAssociation:CustomDomainAssociation":
                return new exports.CustomDomainAssociation(name, undefined, { urn });
            case "aws:redshiftserverless/endpointAccess:EndpointAccess":
                return new exports.EndpointAccess(name, undefined, { urn });
            case "aws:redshiftserverless/namespace:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "aws:redshiftserverless/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:redshiftserverless/snapshot:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "aws:redshiftserverless/usageLimit:UsageLimit":
                return new exports.UsageLimit(name, undefined, { urn });
            case "aws:redshiftserverless/workgroup:Workgroup":
                return new exports.Workgroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/customDomainAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/endpointAccess", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/namespace", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/snapshot", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/usageLimit", _module);
pulumi.runtime.registerResourceModule("aws", "redshiftserverless/workgroup", _module);
