// @pulumi/aws/ec2transitgateway/meteringPolicyEntry.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeteringPolicyEntry = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an EC2 Transit Gateway Metering Policy Entry. Each entry defines a traffic matching rule within a Transit Gateway Metering Policy that determines which account is charged for matching traffic flows.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2transitgateway.MeteringPolicyEntry("example", {
 *     transitGatewayMeteringPolicyId: exampleAwsEc2TransitGatewayMeteringPolicy.transitGatewayMeteringPolicyId,
 *     policyRuleNumber: 100,
 *     meteredAccount: "source-attachment-owner",
 * });
 * ```
 *
 * ### Full Traffic Matching Rule
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2transitgateway.MeteringPolicyEntry("example", {
 *     transitGatewayMeteringPolicyId: exampleAwsEc2TransitGatewayMeteringPolicy.transitGatewayMeteringPolicyId,
 *     policyRuleNumber: 200,
 *     meteredAccount: "destination-attachment-owner",
 *     sourceCidrBlock: "10.0.0.0/8",
 *     destinationCidrBlock: "172.16.0.0/12",
 *     protocol: "6",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws.ec2transitgateway.MeteringPolicyEntry` using the composite identifier. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2transitgateway/meteringPolicyEntry:MeteringPolicyEntry example tgw-policy-12345678,100
 * ```
 */
class MeteringPolicyEntry extends pulumi.CustomResource {
    /**
     * Get an existing MeteringPolicyEntry resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MeteringPolicyEntry(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2transitgateway/meteringPolicyEntry:MeteringPolicyEntry';
    /**
     * Returns true if the given object is an instance of MeteringPolicyEntry.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MeteringPolicyEntry.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["destinationCidrBlock"] = state?.destinationCidrBlock;
            resourceInputs["destinationPortRange"] = state?.destinationPortRange;
            resourceInputs["destinationTransitGatewayAttachmentId"] = state?.destinationTransitGatewayAttachmentId;
            resourceInputs["destinationTransitGatewayAttachmentType"] = state?.destinationTransitGatewayAttachmentType;
            resourceInputs["meteredAccount"] = state?.meteredAccount;
            resourceInputs["policyRuleNumber"] = state?.policyRuleNumber;
            resourceInputs["protocol"] = state?.protocol;
            resourceInputs["region"] = state?.region;
            resourceInputs["sourceCidrBlock"] = state?.sourceCidrBlock;
            resourceInputs["sourcePortRange"] = state?.sourcePortRange;
            resourceInputs["sourceTransitGatewayAttachmentId"] = state?.sourceTransitGatewayAttachmentId;
            resourceInputs["sourceTransitGatewayAttachmentType"] = state?.sourceTransitGatewayAttachmentType;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["transitGatewayMeteringPolicyId"] = state?.transitGatewayMeteringPolicyId;
        }
        else {
            const args = argsOrState;
            if (args?.meteredAccount === undefined && !opts.urn) {
                throw new Error("Missing required property 'meteredAccount'");
            }
            if (args?.policyRuleNumber === undefined && !opts.urn) {
                throw new Error("Missing required property 'policyRuleNumber'");
            }
            if (args?.transitGatewayMeteringPolicyId === undefined && !opts.urn) {
                throw new Error("Missing required property 'transitGatewayMeteringPolicyId'");
            }
            resourceInputs["destinationCidrBlock"] = args?.destinationCidrBlock;
            resourceInputs["destinationPortRange"] = args?.destinationPortRange;
            resourceInputs["destinationTransitGatewayAttachmentId"] = args?.destinationTransitGatewayAttachmentId;
            resourceInputs["destinationTransitGatewayAttachmentType"] = args?.destinationTransitGatewayAttachmentType;
            resourceInputs["meteredAccount"] = args?.meteredAccount;
            resourceInputs["policyRuleNumber"] = args?.policyRuleNumber;
            resourceInputs["protocol"] = args?.protocol;
            resourceInputs["region"] = args?.region;
            resourceInputs["sourceCidrBlock"] = args?.sourceCidrBlock;
            resourceInputs["sourcePortRange"] = args?.sourcePortRange;
            resourceInputs["sourceTransitGatewayAttachmentId"] = args?.sourceTransitGatewayAttachmentId;
            resourceInputs["sourceTransitGatewayAttachmentType"] = args?.sourceTransitGatewayAttachmentType;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["transitGatewayMeteringPolicyId"] = args?.transitGatewayMeteringPolicyId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MeteringPolicyEntry.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MeteringPolicyEntry = MeteringPolicyEntry;
