// @pulumi/azure-native/cognitiveservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSharedCommitmentPlan = exports.getRaiTopicOutput = exports.getRaiTopic = exports.getRaiToolLabelOutput = exports.getRaiToolLabel = exports.getRaiPolicyOutput = exports.getRaiPolicy = exports.getRaiExternalSafetyProviderOutput = exports.getRaiExternalSafetyProvider = exports.getRaiBlocklistItemOutput = exports.getRaiBlocklistItem = exports.getRaiBlocklistOutput = exports.getRaiBlocklist = exports.getProjectConnectionOutput = exports.getProjectConnection = exports.getProjectCapabilityHostOutput = exports.getProjectCapabilityHost = exports.getProjectOutput = exports.getProject = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getOutboundRuleOutput = exports.getOutboundRule = exports.getEncryptionScopeOutput = exports.getEncryptionScope = exports.getDeploymentOutput = exports.getDeployment = exports.getCommitmentPlanAssociationOutput = exports.getCommitmentPlanAssociation = exports.getCommitmentPlanOutput = exports.getCommitmentPlan = exports.getAgentDeploymentOutput = exports.getAgentDeployment = exports.getAgentApplicationOutput = exports.getAgentApplication = exports.getAccountConnectionOutput = exports.getAccountConnection = exports.getAccountCapabilityHostOutput = exports.getAccountCapabilityHost = exports.getAccountOutput = exports.getAccount = exports.EncryptionScope = exports.Deployment = exports.CommitmentPlanAssociation = exports.CommitmentPlan = exports.AgentDeployment = exports.AgentApplication = exports.AccountConnection = exports.AccountCapabilityHost = exports.Account = void 0;
exports.SubscriptionRaiPolicy = exports.SharedCommitmentPlan = exports.RaiTopic = exports.RaiToolLabel = exports.RaiPolicy = exports.RaiExternalSafetyProvider = exports.RaiBlocklistItem = exports.RaiBlocklist = exports.ProjectConnection = exports.ProjectCapabilityHost = exports.Project = exports.PrivateEndpointConnection = exports.OutboundRule = exports.listAgentApplicationAgentsOutput = exports.listAgentApplicationAgents = exports.listAccountKeysOutput = exports.listAccountKeys = exports.getSubscriptionRaiPolicyOutput = exports.getSubscriptionRaiPolicy = exports.getSharedCommitmentPlanOutput = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.AccountCapabilityHost = null;
utilities.lazyLoad(exports, ["AccountCapabilityHost"], () => require("./accountCapabilityHost"));
exports.AccountConnection = null;
utilities.lazyLoad(exports, ["AccountConnection"], () => require("./accountConnection"));
exports.AgentApplication = null;
utilities.lazyLoad(exports, ["AgentApplication"], () => require("./agentApplication"));
exports.AgentDeployment = null;
utilities.lazyLoad(exports, ["AgentDeployment"], () => require("./agentDeployment"));
exports.CommitmentPlan = null;
utilities.lazyLoad(exports, ["CommitmentPlan"], () => require("./commitmentPlan"));
exports.CommitmentPlanAssociation = null;
utilities.lazyLoad(exports, ["CommitmentPlanAssociation"], () => require("./commitmentPlanAssociation"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.EncryptionScope = null;
utilities.lazyLoad(exports, ["EncryptionScope"], () => require("./encryptionScope"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getAccountCapabilityHost = null;
exports.getAccountCapabilityHostOutput = null;
utilities.lazyLoad(exports, ["getAccountCapabilityHost", "getAccountCapabilityHostOutput"], () => require("./getAccountCapabilityHost"));
exports.getAccountConnection = null;
exports.getAccountConnectionOutput = null;
utilities.lazyLoad(exports, ["getAccountConnection", "getAccountConnectionOutput"], () => require("./getAccountConnection"));
exports.getAgentApplication = null;
exports.getAgentApplicationOutput = null;
utilities.lazyLoad(exports, ["getAgentApplication", "getAgentApplicationOutput"], () => require("./getAgentApplication"));
exports.getAgentDeployment = null;
exports.getAgentDeploymentOutput = null;
utilities.lazyLoad(exports, ["getAgentDeployment", "getAgentDeploymentOutput"], () => require("./getAgentDeployment"));
exports.getCommitmentPlan = null;
exports.getCommitmentPlanOutput = null;
utilities.lazyLoad(exports, ["getCommitmentPlan", "getCommitmentPlanOutput"], () => require("./getCommitmentPlan"));
exports.getCommitmentPlanAssociation = null;
exports.getCommitmentPlanAssociationOutput = null;
utilities.lazyLoad(exports, ["getCommitmentPlanAssociation", "getCommitmentPlanAssociationOutput"], () => require("./getCommitmentPlanAssociation"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getEncryptionScope = null;
exports.getEncryptionScopeOutput = null;
utilities.lazyLoad(exports, ["getEncryptionScope", "getEncryptionScopeOutput"], () => require("./getEncryptionScope"));
exports.getOutboundRule = null;
exports.getOutboundRuleOutput = null;
utilities.lazyLoad(exports, ["getOutboundRule", "getOutboundRuleOutput"], () => require("./getOutboundRule"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getProjectCapabilityHost = null;
exports.getProjectCapabilityHostOutput = null;
utilities.lazyLoad(exports, ["getProjectCapabilityHost", "getProjectCapabilityHostOutput"], () => require("./getProjectCapabilityHost"));
exports.getProjectConnection = null;
exports.getProjectConnectionOutput = null;
utilities.lazyLoad(exports, ["getProjectConnection", "getProjectConnectionOutput"], () => require("./getProjectConnection"));
exports.getRaiBlocklist = null;
exports.getRaiBlocklistOutput = null;
utilities.lazyLoad(exports, ["getRaiBlocklist", "getRaiBlocklistOutput"], () => require("./getRaiBlocklist"));
exports.getRaiBlocklistItem = null;
exports.getRaiBlocklistItemOutput = null;
utilities.lazyLoad(exports, ["getRaiBlocklistItem", "getRaiBlocklistItemOutput"], () => require("./getRaiBlocklistItem"));
exports.getRaiExternalSafetyProvider = null;
exports.getRaiExternalSafetyProviderOutput = null;
utilities.lazyLoad(exports, ["getRaiExternalSafetyProvider", "getRaiExternalSafetyProviderOutput"], () => require("./getRaiExternalSafetyProvider"));
exports.getRaiPolicy = null;
exports.getRaiPolicyOutput = null;
utilities.lazyLoad(exports, ["getRaiPolicy", "getRaiPolicyOutput"], () => require("./getRaiPolicy"));
exports.getRaiToolLabel = null;
exports.getRaiToolLabelOutput = null;
utilities.lazyLoad(exports, ["getRaiToolLabel", "getRaiToolLabelOutput"], () => require("./getRaiToolLabel"));
exports.getRaiTopic = null;
exports.getRaiTopicOutput = null;
utilities.lazyLoad(exports, ["getRaiTopic", "getRaiTopicOutput"], () => require("./getRaiTopic"));
exports.getSharedCommitmentPlan = null;
exports.getSharedCommitmentPlanOutput = null;
utilities.lazyLoad(exports, ["getSharedCommitmentPlan", "getSharedCommitmentPlanOutput"], () => require("./getSharedCommitmentPlan"));
exports.getSubscriptionRaiPolicy = null;
exports.getSubscriptionRaiPolicyOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionRaiPolicy", "getSubscriptionRaiPolicyOutput"], () => require("./getSubscriptionRaiPolicy"));
exports.listAccountKeys = null;
exports.listAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listAccountKeys", "listAccountKeysOutput"], () => require("./listAccountKeys"));
exports.listAgentApplicationAgents = null;
exports.listAgentApplicationAgentsOutput = null;
utilities.lazyLoad(exports, ["listAgentApplicationAgents", "listAgentApplicationAgentsOutput"], () => require("./listAgentApplicationAgents"));
exports.OutboundRule = null;
utilities.lazyLoad(exports, ["OutboundRule"], () => require("./outboundRule"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ProjectCapabilityHost = null;
utilities.lazyLoad(exports, ["ProjectCapabilityHost"], () => require("./projectCapabilityHost"));
exports.ProjectConnection = null;
utilities.lazyLoad(exports, ["ProjectConnection"], () => require("./projectConnection"));
exports.RaiBlocklist = null;
utilities.lazyLoad(exports, ["RaiBlocklist"], () => require("./raiBlocklist"));
exports.RaiBlocklistItem = null;
utilities.lazyLoad(exports, ["RaiBlocklistItem"], () => require("./raiBlocklistItem"));
exports.RaiExternalSafetyProvider = null;
utilities.lazyLoad(exports, ["RaiExternalSafetyProvider"], () => require("./raiExternalSafetyProvider"));
exports.RaiPolicy = null;
utilities.lazyLoad(exports, ["RaiPolicy"], () => require("./raiPolicy"));
exports.RaiToolLabel = null;
utilities.lazyLoad(exports, ["RaiToolLabel"], () => require("./raiToolLabel"));
exports.RaiTopic = null;
utilities.lazyLoad(exports, ["RaiTopic"], () => require("./raiTopic"));
exports.SharedCommitmentPlan = null;
utilities.lazyLoad(exports, ["SharedCommitmentPlan"], () => require("./sharedCommitmentPlan"));
exports.SubscriptionRaiPolicy = null;
utilities.lazyLoad(exports, ["SubscriptionRaiPolicy"], () => require("./subscriptionRaiPolicy"));
// Export enums:
__exportStar(require("../types/enums/cognitiveservices"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cognitiveservices:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:cognitiveservices:AccountCapabilityHost":
                return new exports.AccountCapabilityHost(name, undefined, { urn });
            case "azure-native:cognitiveservices:AccountConnection":
                return new exports.AccountConnection(name, undefined, { urn });
            case "azure-native:cognitiveservices:AgentApplication":
                return new exports.AgentApplication(name, undefined, { urn });
            case "azure-native:cognitiveservices:AgentDeployment":
                return new exports.AgentDeployment(name, undefined, { urn });
            case "azure-native:cognitiveservices:CommitmentPlan":
                return new exports.CommitmentPlan(name, undefined, { urn });
            case "azure-native:cognitiveservices:CommitmentPlanAssociation":
                return new exports.CommitmentPlanAssociation(name, undefined, { urn });
            case "azure-native:cognitiveservices:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:cognitiveservices:EncryptionScope":
                return new exports.EncryptionScope(name, undefined, { urn });
            case "azure-native:cognitiveservices:OutboundRule":
                return new exports.OutboundRule(name, undefined, { urn });
            case "azure-native:cognitiveservices:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:cognitiveservices:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:cognitiveservices:ProjectCapabilityHost":
                return new exports.ProjectCapabilityHost(name, undefined, { urn });
            case "azure-native:cognitiveservices:ProjectConnection":
                return new exports.ProjectConnection(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiBlocklist":
                return new exports.RaiBlocklist(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiBlocklistItem":
                return new exports.RaiBlocklistItem(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiExternalSafetyProvider":
                return new exports.RaiExternalSafetyProvider(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiPolicy":
                return new exports.RaiPolicy(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiToolLabel":
                return new exports.RaiToolLabel(name, undefined, { urn });
            case "azure-native:cognitiveservices:RaiTopic":
                return new exports.RaiTopic(name, undefined, { urn });
            case "azure-native:cognitiveservices:SharedCommitmentPlan":
                return new exports.SharedCommitmentPlan(name, undefined, { urn });
            case "azure-native:cognitiveservices:SubscriptionRaiPolicy":
                return new exports.SubscriptionRaiPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cognitiveservices", _module);
