// @pulumi/azure-native/iotoperationsmq/broker.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Broker = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * MQ broker resource
 *
 * Uses Azure REST API version 2023-10-04-preview. In version 2.x of the Azure Native provider, it used API version 2023-10-04-preview.
 */
class Broker extends pulumi.CustomResource {
    /**
     * Get an existing Broker resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Broker(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:iotoperationsmq:Broker';
    /**
     * Returns true if the given object is an instance of Broker.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Broker.__pulumiType;
    }
    /**
     * Create a Broker resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.authImage === undefined && !opts.urn) {
                throw new Error("Missing required property 'authImage'");
            }
            if (args?.brokerImage === undefined && !opts.urn) {
                throw new Error("Missing required property 'brokerImage'");
            }
            if (args?.extendedLocation === undefined && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if (args?.healthManagerImage === undefined && !opts.urn) {
                throw new Error("Missing required property 'healthManagerImage'");
            }
            if (args?.mode === undefined && !opts.urn) {
                throw new Error("Missing required property 'mode'");
            }
            if (args?.mqName === undefined && !opts.urn) {
                throw new Error("Missing required property 'mqName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authImage"] = args?.authImage;
            resourceInputs["brokerImage"] = args?.brokerImage;
            resourceInputs["brokerName"] = args?.brokerName;
            resourceInputs["brokerNodeTolerations"] = args?.brokerNodeTolerations;
            resourceInputs["cardinality"] = args ? (args.cardinality ? pulumi.output(args.cardinality).apply(inputs.iotoperationsmq.cardinalityArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["diagnostics"] = args ? (args.diagnostics ? pulumi.output(args.diagnostics).apply(inputs.iotoperationsmq.brokerDiagnosticsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["diskBackedMessageBufferSettings"] = args?.diskBackedMessageBufferSettings;
            resourceInputs["encryptInternalTraffic"] = (args?.encryptInternalTraffic) ?? true;
            resourceInputs["extendedLocation"] = args?.extendedLocation;
            resourceInputs["healthManagerImage"] = args?.healthManagerImage;
            resourceInputs["healthManagerNodeTolerations"] = args?.healthManagerNodeTolerations;
            resourceInputs["internalCerts"] = args ? (args.internalCerts ? pulumi.output(args.internalCerts).apply(inputs.iotoperationsmq.certManagerCertOptionsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["location"] = args?.location;
            resourceInputs["memoryProfile"] = (args?.memoryProfile) ?? "medium";
            resourceInputs["mode"] = args?.mode;
            resourceInputs["mqName"] = args?.mqName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["authImage"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["brokerImage"] = undefined /*out*/;
            resourceInputs["brokerNodeTolerations"] = undefined /*out*/;
            resourceInputs["cardinality"] = undefined /*out*/;
            resourceInputs["diagnostics"] = undefined /*out*/;
            resourceInputs["diskBackedMessageBufferSettings"] = undefined /*out*/;
            resourceInputs["encryptInternalTraffic"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["healthManagerImage"] = undefined /*out*/;
            resourceInputs["healthManagerNodeTolerations"] = undefined /*out*/;
            resourceInputs["internalCerts"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["memoryProfile"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:iotoperationsmq/v20231004preview:Broker" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Broker.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Broker = Broker;
