// @pulumi/azure-native/datamigration/databaseMigrationsMongoToCosmosDbvCoreMongo.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseMigrationsMongoToCosmosDbvCoreMongo = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Database Migration Resource for Mongo to CosmosDb.
 *
 * Uses Azure REST API version 2023-07-15-preview. In version 2.x of the Azure Native provider, it used API version 2023-07-15-preview.
 *
 * Other available API versions: 2025-03-15-preview, 2025-06-30, 2025-09-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native datamigration [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DatabaseMigrationsMongoToCosmosDbvCoreMongo extends pulumi.CustomResource {
    /**
     * Get an existing DatabaseMigrationsMongoToCosmosDbvCoreMongo resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseMigrationsMongoToCosmosDbvCoreMongo(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:datamigration:DatabaseMigrationsMongoToCosmosDbvCoreMongo';
    /**
     * Returns true if the given object is an instance of DatabaseMigrationsMongoToCosmosDbvCoreMongo.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseMigrationsMongoToCosmosDbvCoreMongo.__pulumiType;
    }
    /**
     * Create a DatabaseMigrationsMongoToCosmosDbvCoreMongo resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.targetResourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetResourceName'");
            }
            resourceInputs["collectionList"] = args?.collectionList;
            resourceInputs["kind"] = "MongoToCosmosDbMongo";
            resourceInputs["migrationName"] = args?.migrationName;
            resourceInputs["migrationOperationId"] = args?.migrationOperationId;
            resourceInputs["migrationService"] = args?.migrationService;
            resourceInputs["provisioningError"] = args?.provisioningError;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["scope"] = args?.scope;
            resourceInputs["sourceMongoConnection"] = args?.sourceMongoConnection;
            resourceInputs["targetMongoConnection"] = args?.targetMongoConnection;
            resourceInputs["targetResourceName"] = args?.targetResourceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["endedOn"] = undefined /*out*/;
            resourceInputs["migrationFailureError"] = undefined /*out*/;
            resourceInputs["migrationStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["startedOn"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["collectionList"] = undefined /*out*/;
            resourceInputs["endedOn"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["migrationFailureError"] = undefined /*out*/;
            resourceInputs["migrationOperationId"] = undefined /*out*/;
            resourceInputs["migrationService"] = undefined /*out*/;
            resourceInputs["migrationStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningError"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["sourceMongoConnection"] = undefined /*out*/;
            resourceInputs["startedOn"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["targetMongoConnection"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:datamigration/v20230715preview:DatabaseMigrationsMongoToCosmosDbvCoreMongo" }, { type: "azure-native:datamigration/v20250315preview:DatabaseMigrationsMongoToCosmosDbvCoreMongo" }, { type: "azure-native:datamigration/v20250630:DatabaseMigrationsMongoToCosmosDbvCoreMongo" }, { type: "azure-native:datamigration/v20250901preview:DatabaseMigrationsMongoToCosmosDbvCoreMongo" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseMigrationsMongoToCosmosDbvCoreMongo.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DatabaseMigrationsMongoToCosmosDbvCoreMongo = DatabaseMigrationsMongoToCosmosDbvCoreMongo;
