// @pulumi/azure-native/datamigration/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Task = exports.SqlMigrationService = exports.ServiceTask = exports.Service = exports.Project = exports.MigrationService = exports.listSqlMigrationServiceMonitoringDataOutput = exports.listSqlMigrationServiceMonitoringData = exports.listSqlMigrationServiceAuthKeysOutput = exports.listSqlMigrationServiceAuthKeys = exports.getTaskOutput = exports.getTask = exports.getSqlMigrationServiceOutput = exports.getSqlMigrationService = exports.getServiceTaskOutput = exports.getServiceTask = exports.getServiceOutput = exports.getService = exports.getProjectOutput = exports.getProject = exports.getMigrationServiceOutput = exports.getMigrationService = exports.getFileOutput = exports.getFile = exports.getDatabaseMigrationsSqlVmOutput = exports.getDatabaseMigrationsSqlVm = exports.getDatabaseMigrationsSqlMiOutput = exports.getDatabaseMigrationsSqlMi = exports.getDatabaseMigrationsSqlDbOutput = exports.getDatabaseMigrationsSqlDb = exports.getDatabaseMigrationsMongoToCosmosDbvCoreMongoOutput = exports.getDatabaseMigrationsMongoToCosmosDbvCoreMongo = exports.getDatabaseMigrationsMongoToCosmosDbRUMongoOutput = exports.getDatabaseMigrationsMongoToCosmosDbRUMongo = exports.File = exports.DatabaseMigrationsSqlVm = exports.DatabaseMigrationsSqlMi = exports.DatabaseMigrationsSqlDb = exports.DatabaseMigrationsMongoToCosmosDbvCoreMongo = exports.DatabaseMigrationsMongoToCosmosDbRUMongo = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.DatabaseMigrationsMongoToCosmosDbRUMongo = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsMongoToCosmosDbRUMongo"], () => require("./databaseMigrationsMongoToCosmosDbRUMongo"));
exports.DatabaseMigrationsMongoToCosmosDbvCoreMongo = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsMongoToCosmosDbvCoreMongo"], () => require("./databaseMigrationsMongoToCosmosDbvCoreMongo"));
exports.DatabaseMigrationsSqlDb = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsSqlDb"], () => require("./databaseMigrationsSqlDb"));
exports.DatabaseMigrationsSqlMi = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsSqlMi"], () => require("./databaseMigrationsSqlMi"));
exports.DatabaseMigrationsSqlVm = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsSqlVm"], () => require("./databaseMigrationsSqlVm"));
exports.File = null;
utilities.lazyLoad(exports, ["File"], () => require("./file"));
exports.getDatabaseMigrationsMongoToCosmosDbRUMongo = null;
exports.getDatabaseMigrationsMongoToCosmosDbRUMongoOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsMongoToCosmosDbRUMongo", "getDatabaseMigrationsMongoToCosmosDbRUMongoOutput"], () => require("./getDatabaseMigrationsMongoToCosmosDbRUMongo"));
exports.getDatabaseMigrationsMongoToCosmosDbvCoreMongo = null;
exports.getDatabaseMigrationsMongoToCosmosDbvCoreMongoOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsMongoToCosmosDbvCoreMongo", "getDatabaseMigrationsMongoToCosmosDbvCoreMongoOutput"], () => require("./getDatabaseMigrationsMongoToCosmosDbvCoreMongo"));
exports.getDatabaseMigrationsSqlDb = null;
exports.getDatabaseMigrationsSqlDbOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsSqlDb", "getDatabaseMigrationsSqlDbOutput"], () => require("./getDatabaseMigrationsSqlDb"));
exports.getDatabaseMigrationsSqlMi = null;
exports.getDatabaseMigrationsSqlMiOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsSqlMi", "getDatabaseMigrationsSqlMiOutput"], () => require("./getDatabaseMigrationsSqlMi"));
exports.getDatabaseMigrationsSqlVm = null;
exports.getDatabaseMigrationsSqlVmOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsSqlVm", "getDatabaseMigrationsSqlVmOutput"], () => require("./getDatabaseMigrationsSqlVm"));
exports.getFile = null;
exports.getFileOutput = null;
utilities.lazyLoad(exports, ["getFile", "getFileOutput"], () => require("./getFile"));
exports.getMigrationService = null;
exports.getMigrationServiceOutput = null;
utilities.lazyLoad(exports, ["getMigrationService", "getMigrationServiceOutput"], () => require("./getMigrationService"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceTask = null;
exports.getServiceTaskOutput = null;
utilities.lazyLoad(exports, ["getServiceTask", "getServiceTaskOutput"], () => require("./getServiceTask"));
exports.getSqlMigrationService = null;
exports.getSqlMigrationServiceOutput = null;
utilities.lazyLoad(exports, ["getSqlMigrationService", "getSqlMigrationServiceOutput"], () => require("./getSqlMigrationService"));
exports.getTask = null;
exports.getTaskOutput = null;
utilities.lazyLoad(exports, ["getTask", "getTaskOutput"], () => require("./getTask"));
exports.listSqlMigrationServiceAuthKeys = null;
exports.listSqlMigrationServiceAuthKeysOutput = null;
utilities.lazyLoad(exports, ["listSqlMigrationServiceAuthKeys", "listSqlMigrationServiceAuthKeysOutput"], () => require("./listSqlMigrationServiceAuthKeys"));
exports.listSqlMigrationServiceMonitoringData = null;
exports.listSqlMigrationServiceMonitoringDataOutput = null;
utilities.lazyLoad(exports, ["listSqlMigrationServiceMonitoringData", "listSqlMigrationServiceMonitoringDataOutput"], () => require("./listSqlMigrationServiceMonitoringData"));
exports.MigrationService = null;
utilities.lazyLoad(exports, ["MigrationService"], () => require("./migrationService"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceTask = null;
utilities.lazyLoad(exports, ["ServiceTask"], () => require("./serviceTask"));
exports.SqlMigrationService = null;
utilities.lazyLoad(exports, ["SqlMigrationService"], () => require("./sqlMigrationService"));
exports.Task = null;
utilities.lazyLoad(exports, ["Task"], () => require("./task"));
// Export enums:
__exportStar(require("../types/enums/datamigration"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datamigration:DatabaseMigrationsMongoToCosmosDbRUMongo":
                return new exports.DatabaseMigrationsMongoToCosmosDbRUMongo(name, undefined, { urn });
            case "azure-native:datamigration:DatabaseMigrationsMongoToCosmosDbvCoreMongo":
                return new exports.DatabaseMigrationsMongoToCosmosDbvCoreMongo(name, undefined, { urn });
            case "azure-native:datamigration:DatabaseMigrationsSqlDb":
                return new exports.DatabaseMigrationsSqlDb(name, undefined, { urn });
            case "azure-native:datamigration:DatabaseMigrationsSqlMi":
                return new exports.DatabaseMigrationsSqlMi(name, undefined, { urn });
            case "azure-native:datamigration:DatabaseMigrationsSqlVm":
                return new exports.DatabaseMigrationsSqlVm(name, undefined, { urn });
            case "azure-native:datamigration:File":
                return new exports.File(name, undefined, { urn });
            case "azure-native:datamigration:MigrationService":
                return new exports.MigrationService(name, undefined, { urn });
            case "azure-native:datamigration:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:datamigration:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:datamigration:ServiceTask":
                return new exports.ServiceTask(name, undefined, { urn });
            case "azure-native:datamigration:SqlMigrationService":
                return new exports.SqlMigrationService(name, undefined, { urn });
            case "azure-native:datamigration:Task":
                return new exports.Task(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datamigration", _module);
