// @pulumi/aws/s3tables/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableReplication = exports.TablePolicy = exports.TableBucketReplication = exports.TableBucketPolicy = exports.TableBucket = exports.Table = exports.Namespace = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.TableBucket = null;
utilities.lazyLoad(exports, ["TableBucket"], () => require("./tableBucket"));
exports.TableBucketPolicy = null;
utilities.lazyLoad(exports, ["TableBucketPolicy"], () => require("./tableBucketPolicy"));
exports.TableBucketReplication = null;
utilities.lazyLoad(exports, ["TableBucketReplication"], () => require("./tableBucketReplication"));
exports.TablePolicy = null;
utilities.lazyLoad(exports, ["TablePolicy"], () => require("./tablePolicy"));
exports.TableReplication = null;
utilities.lazyLoad(exports, ["TableReplication"], () => require("./tableReplication"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:s3tables/namespace:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "aws:s3tables/table:Table":
                return new exports.Table(name, undefined, { urn });
            case "aws:s3tables/tableBucket:TableBucket":
                return new exports.TableBucket(name, undefined, { urn });
            case "aws:s3tables/tableBucketPolicy:TableBucketPolicy":
                return new exports.TableBucketPolicy(name, undefined, { urn });
            case "aws:s3tables/tableBucketReplication:TableBucketReplication":
                return new exports.TableBucketReplication(name, undefined, { urn });
            case "aws:s3tables/tablePolicy:TablePolicy":
                return new exports.TablePolicy(name, undefined, { urn });
            case "aws:s3tables/tableReplication:TableReplication":
                return new exports.TableReplication(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "s3tables/namespace", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/table", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/tableBucket", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/tableBucketPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/tableBucketReplication", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/tablePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "s3tables/tableReplication", _module);
