// @pulumi/azure-native/hybridnetwork/vendorSkus.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorSkus = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Sku sub resource.
 *
 * Uses Azure REST API version 2022-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2022-01-01-preview.
 */
class VendorSkus extends pulumi.CustomResource {
    /**
     * Get an existing VendorSkus resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VendorSkus(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:hybridnetwork:VendorSkus';
    /**
     * Returns true if the given object is an instance of VendorSkus.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VendorSkus.__pulumiType;
    }
    /**
     * Create a VendorSkus resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.vendorName === undefined && !opts.urn) {
                throw new Error("Missing required property 'vendorName'");
            }
            resourceInputs["deploymentMode"] = args?.deploymentMode;
            resourceInputs["managedApplicationParameters"] = args?.managedApplicationParameters;
            resourceInputs["managedApplicationTemplate"] = args?.managedApplicationTemplate;
            resourceInputs["networkFunctionTemplate"] = args?.networkFunctionTemplate;
            resourceInputs["networkFunctionType"] = args?.networkFunctionType;
            resourceInputs["preview"] = args?.preview;
            resourceInputs["skuName"] = args?.skuName;
            resourceInputs["skuType"] = args?.skuType;
            resourceInputs["vendorName"] = args?.vendorName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentMode"] = undefined /*out*/;
            resourceInputs["managedApplicationParameters"] = undefined /*out*/;
            resourceInputs["managedApplicationTemplate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkFunctionTemplate"] = undefined /*out*/;
            resourceInputs["networkFunctionType"] = undefined /*out*/;
            resourceInputs["preview"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["skuType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hybridnetwork/v20200101preview:VendorSkus" }, { type: "azure-native:hybridnetwork/v20210501:VendorSkus" }, { type: "azure-native:hybridnetwork/v20220101preview:VendorSkus" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VendorSkus.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VendorSkus = VendorSkus;
