// @pulumi/aws/outposts/capacityTask.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityTask = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS Outposts Capacity Task.
 *
 * A capacity task redistributes the instance pools available on an Outpost rack or server to match the `instancePool` configuration declared in the resource. Starting a capacity task is a long-running, asynchronous operation — Terraform waits for it to reach a terminal state (`COMPLETED`, `CANCELLED`, or `FAILED`) before finishing the apply.
 *
 * ## Example Usage
 *
 * ### Minimal
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = aws.outposts.getOutposts({});
 * const exampleCapacityTask = new aws.outposts.CapacityTask("example", {
 *     outpostIdentifier: example.then(example => example.arns?.[0]),
 *     instancePools: [{
 *         instanceType: "m5.large",
 *         count: 2,
 *     }],
 * });
 * ```
 *
 * ### Multiple instance pools, excluded instances, and a specified blocking-instance action
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = aws.outposts.getAssets({
 *     arn: "arn:aws:outposts:us-west-2:123456789012:outpost/op-1234567890abcdef",
 * });
 * const exampleCapacityTask = new aws.outposts.CapacityTask("example", {
 *     outpostIdentifier: "op-1234567890abcdef",
 *     taskActionOnBlockingInstances: "WAIT_FOR_EVACUATION",
 *     assetId: example.then(example => example.assetIds?.[0]),
 *     instancePools: [
 *         {
 *             instanceType: "m5.large",
 *             count: 4,
 *         },
 *         {
 *             instanceType: "c5.xlarge",
 *             count: 2,
 *         },
 *     ],
 *     instancesToExclude: {
 *         instances: [
 *             "i-0123456789abcdef0",
 *             "i-0fedcba9876543210",
 *         ],
 *     },
 * });
 * ```
 *
 * ## Lifecycle
 *
 * Because every argument of this resource is marked as forces-new, any change to the configuration results in destroying and re-creating the capacity task. Tasks that are already in a terminal state (`COMPLETED` or `CANCELLED`) are left in place on destroy and only removed from Terraform state; tasks still in flight are cancelled and Terraform waits for them to reach `CANCELLED`. If a task reaches the terminal state `FAILED` during `delete`, the provider tolerates the "already in a terminal state" error returned by `CancelCapacityTask` and considers the resource successfully destroyed.
 *
 * If a create operation produces a `FAILED` task, the resource is not written to Terraform state (the `failureReason` is surfaced in the diagnostic instead), so no follow-up destroy is required.
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `outpostIdentifier` (String) Outpost identifier supplied when the task was created (ID or ARN).
 * * `capacityTaskId` (String) AWS-assigned capacity task ID.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import a Capacity Task using the same composite ID. For example:
 *
 * ```sh
 * $ pulumi import aws:outposts/capacityTask:CapacityTask example op-1234567890abcdef/cap-1a2b3c4d5e6f7g8h9
 * ```
 */
class CapacityTask extends pulumi.CustomResource {
    /**
     * Get an existing CapacityTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CapacityTask(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:outposts/capacityTask:CapacityTask';
    /**
     * Returns true if the given object is an instance of CapacityTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityTask.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["assetId"] = state?.assetId;
            resourceInputs["capacityTaskId"] = state?.capacityTaskId;
            resourceInputs["completionDate"] = state?.completionDate;
            resourceInputs["creationDate"] = state?.creationDate;
            resourceInputs["failureReason"] = state?.failureReason;
            resourceInputs["instancePools"] = state?.instancePools;
            resourceInputs["instancesToExclude"] = state?.instancesToExclude;
            resourceInputs["orderId"] = state?.orderId;
            resourceInputs["outpostIdentifier"] = state?.outpostIdentifier;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["taskActionOnBlockingInstances"] = state?.taskActionOnBlockingInstances;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.outpostIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'outpostIdentifier'");
            }
            resourceInputs["assetId"] = args?.assetId;
            resourceInputs["instancePools"] = args?.instancePools;
            resourceInputs["instancesToExclude"] = args?.instancesToExclude;
            resourceInputs["orderId"] = args?.orderId;
            resourceInputs["outpostIdentifier"] = args?.outpostIdentifier;
            resourceInputs["region"] = args?.region;
            resourceInputs["taskActionOnBlockingInstances"] = args?.taskActionOnBlockingInstances;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["capacityTaskId"] = undefined /*out*/;
            resourceInputs["completionDate"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CapacityTask.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityTask = CapacityTask;
