// @pulumi/aws/outposts/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSitesOutput = exports.getSites = exports.getSiteOutput = exports.getSite = exports.getOutpostsOutput = exports.getOutposts = exports.getOutpostInstanceTypesOutput = exports.getOutpostInstanceTypes = exports.getOutpostInstanceTypeOutput = exports.getOutpostInstanceType = exports.getOutpostOutput = exports.getOutpost = exports.getAssetsOutput = exports.getAssets = exports.getAssetOutput = exports.getAsset = exports.CapacityTask = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CapacityTask = null;
utilities.lazyLoad(exports, ["CapacityTask"], () => require("./capacityTask"));
exports.getAsset = null;
exports.getAssetOutput = null;
utilities.lazyLoad(exports, ["getAsset", "getAssetOutput"], () => require("./getAsset"));
exports.getAssets = null;
exports.getAssetsOutput = null;
utilities.lazyLoad(exports, ["getAssets", "getAssetsOutput"], () => require("./getAssets"));
exports.getOutpost = null;
exports.getOutpostOutput = null;
utilities.lazyLoad(exports, ["getOutpost", "getOutpostOutput"], () => require("./getOutpost"));
exports.getOutpostInstanceType = null;
exports.getOutpostInstanceTypeOutput = null;
utilities.lazyLoad(exports, ["getOutpostInstanceType", "getOutpostInstanceTypeOutput"], () => require("./getOutpostInstanceType"));
exports.getOutpostInstanceTypes = null;
exports.getOutpostInstanceTypesOutput = null;
utilities.lazyLoad(exports, ["getOutpostInstanceTypes", "getOutpostInstanceTypesOutput"], () => require("./getOutpostInstanceTypes"));
exports.getOutposts = null;
exports.getOutpostsOutput = null;
utilities.lazyLoad(exports, ["getOutposts", "getOutpostsOutput"], () => require("./getOutposts"));
exports.getSite = null;
exports.getSiteOutput = null;
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
exports.getSites = null;
exports.getSitesOutput = null;
utilities.lazyLoad(exports, ["getSites", "getSitesOutput"], () => require("./getSites"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:outposts/capacityTask:CapacityTask":
                return new exports.CapacityTask(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "outposts/capacityTask", _module);
