// @pulumi/aws/directoryservice/sharedDirectoryAccepter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDirectoryAccepter = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Accepts a shared directory in a consumer account.
 *
 * > **NOTE:** Destroying this resource removes the shared directory from the consumer account only.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.directoryservice.SharedDirectory("example", {
 *     directoryId: exampleAwsDirectoryServiceDirectory.id,
 *     notes: "example",
 *     target: {
 *         id: receiver.accountId,
 *     },
 * });
 * const exampleSharedDirectoryAccepter = new aws.directoryservice.SharedDirectoryAccepter("example", {sharedDirectoryId: example.sharedDirectoryId});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Directory Service Shared Directories using the shared directory ID. For example:
 *
 * ```sh
 * $ pulumi import aws:directoryservice/sharedDirectoryAccepter:SharedDirectoryAccepter example d-9267633ece
 * ```
 */
class SharedDirectoryAccepter extends pulumi.CustomResource {
    /**
     * Get an existing SharedDirectoryAccepter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SharedDirectoryAccepter(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:directoryservice/sharedDirectoryAccepter:SharedDirectoryAccepter';
    /**
     * Returns true if the given object is an instance of SharedDirectoryAccepter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SharedDirectoryAccepter.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["method"] = state?.method;
            resourceInputs["notes"] = state?.notes;
            resourceInputs["ownerAccountId"] = state?.ownerAccountId;
            resourceInputs["ownerDirectoryId"] = state?.ownerDirectoryId;
            resourceInputs["region"] = state?.region;
            resourceInputs["sharedDirectoryId"] = state?.sharedDirectoryId;
        }
        else {
            const args = argsOrState;
            if (args?.sharedDirectoryId === undefined && !opts.urn) {
                throw new Error("Missing required property 'sharedDirectoryId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["sharedDirectoryId"] = args?.sharedDirectoryId;
            resourceInputs["method"] = undefined /*out*/;
            resourceInputs["notes"] = undefined /*out*/;
            resourceInputs["ownerAccountId"] = undefined /*out*/;
            resourceInputs["ownerDirectoryId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SharedDirectoryAccepter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SharedDirectoryAccepter = SharedDirectoryAccepter;
