// @pulumi/azure-native/deviceregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaVersion = exports.SchemaRegistry = exports.Schema = exports.Policy = exports.NamespaceDiscoveredDevice = exports.NamespaceDiscoveredAsset = exports.NamespaceDevice = exports.NamespaceAsset = exports.Namespace = exports.getSchemaVersionOutput = exports.getSchemaVersion = exports.getSchemaRegistryOutput = exports.getSchemaRegistry = exports.getSchemaOutput = exports.getSchema = exports.getPolicyOutput = exports.getPolicy = exports.getNamespaceDiscoveredDeviceOutput = exports.getNamespaceDiscoveredDevice = exports.getNamespaceDiscoveredAssetOutput = exports.getNamespaceDiscoveredAsset = exports.getNamespaceDeviceOutput = exports.getNamespaceDevice = exports.getNamespaceAssetOutput = exports.getNamespaceAsset = exports.getNamespaceOutput = exports.getNamespace = exports.getDiscoveredAssetEndpointProfileOutput = exports.getDiscoveredAssetEndpointProfile = exports.getDiscoveredAssetOutput = exports.getDiscoveredAsset = exports.getCredentialOutput = exports.getCredential = exports.getAssetEndpointProfileOutput = exports.getAssetEndpointProfile = exports.getAssetOutput = exports.getAsset = exports.DiscoveredAssetEndpointProfile = exports.DiscoveredAsset = exports.Credential = exports.AssetEndpointProfile = exports.Asset = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Asset = null;
utilities.lazyLoad(exports, ["Asset"], () => require("./asset"));
exports.AssetEndpointProfile = null;
utilities.lazyLoad(exports, ["AssetEndpointProfile"], () => require("./assetEndpointProfile"));
exports.Credential = null;
utilities.lazyLoad(exports, ["Credential"], () => require("./credential"));
exports.DiscoveredAsset = null;
utilities.lazyLoad(exports, ["DiscoveredAsset"], () => require("./discoveredAsset"));
exports.DiscoveredAssetEndpointProfile = null;
utilities.lazyLoad(exports, ["DiscoveredAssetEndpointProfile"], () => require("./discoveredAssetEndpointProfile"));
exports.getAsset = null;
exports.getAssetOutput = null;
utilities.lazyLoad(exports, ["getAsset", "getAssetOutput"], () => require("./getAsset"));
exports.getAssetEndpointProfile = null;
exports.getAssetEndpointProfileOutput = null;
utilities.lazyLoad(exports, ["getAssetEndpointProfile", "getAssetEndpointProfileOutput"], () => require("./getAssetEndpointProfile"));
exports.getCredential = null;
exports.getCredentialOutput = null;
utilities.lazyLoad(exports, ["getCredential", "getCredentialOutput"], () => require("./getCredential"));
exports.getDiscoveredAsset = null;
exports.getDiscoveredAssetOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredAsset", "getDiscoveredAssetOutput"], () => require("./getDiscoveredAsset"));
exports.getDiscoveredAssetEndpointProfile = null;
exports.getDiscoveredAssetEndpointProfileOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredAssetEndpointProfile", "getDiscoveredAssetEndpointProfileOutput"], () => require("./getDiscoveredAssetEndpointProfile"));
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getNamespaceAsset = null;
exports.getNamespaceAssetOutput = null;
utilities.lazyLoad(exports, ["getNamespaceAsset", "getNamespaceAssetOutput"], () => require("./getNamespaceAsset"));
exports.getNamespaceDevice = null;
exports.getNamespaceDeviceOutput = null;
utilities.lazyLoad(exports, ["getNamespaceDevice", "getNamespaceDeviceOutput"], () => require("./getNamespaceDevice"));
exports.getNamespaceDiscoveredAsset = null;
exports.getNamespaceDiscoveredAssetOutput = null;
utilities.lazyLoad(exports, ["getNamespaceDiscoveredAsset", "getNamespaceDiscoveredAssetOutput"], () => require("./getNamespaceDiscoveredAsset"));
exports.getNamespaceDiscoveredDevice = null;
exports.getNamespaceDiscoveredDeviceOutput = null;
utilities.lazyLoad(exports, ["getNamespaceDiscoveredDevice", "getNamespaceDiscoveredDeviceOutput"], () => require("./getNamespaceDiscoveredDevice"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getSchema = null;
exports.getSchemaOutput = null;
utilities.lazyLoad(exports, ["getSchema", "getSchemaOutput"], () => require("./getSchema"));
exports.getSchemaRegistry = null;
exports.getSchemaRegistryOutput = null;
utilities.lazyLoad(exports, ["getSchemaRegistry", "getSchemaRegistryOutput"], () => require("./getSchemaRegistry"));
exports.getSchemaVersion = null;
exports.getSchemaVersionOutput = null;
utilities.lazyLoad(exports, ["getSchemaVersion", "getSchemaVersionOutput"], () => require("./getSchemaVersion"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.NamespaceAsset = null;
utilities.lazyLoad(exports, ["NamespaceAsset"], () => require("./namespaceAsset"));
exports.NamespaceDevice = null;
utilities.lazyLoad(exports, ["NamespaceDevice"], () => require("./namespaceDevice"));
exports.NamespaceDiscoveredAsset = null;
utilities.lazyLoad(exports, ["NamespaceDiscoveredAsset"], () => require("./namespaceDiscoveredAsset"));
exports.NamespaceDiscoveredDevice = null;
utilities.lazyLoad(exports, ["NamespaceDiscoveredDevice"], () => require("./namespaceDiscoveredDevice"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
exports.SchemaRegistry = null;
utilities.lazyLoad(exports, ["SchemaRegistry"], () => require("./schemaRegistry"));
exports.SchemaVersion = null;
utilities.lazyLoad(exports, ["SchemaVersion"], () => require("./schemaVersion"));
// Export enums:
__exportStar(require("../types/enums/deviceregistry"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:deviceregistry:Asset":
                return new exports.Asset(name, undefined, { urn });
            case "azure-native:deviceregistry:AssetEndpointProfile":
                return new exports.AssetEndpointProfile(name, undefined, { urn });
            case "azure-native:deviceregistry:Credential":
                return new exports.Credential(name, undefined, { urn });
            case "azure-native:deviceregistry:DiscoveredAsset":
                return new exports.DiscoveredAsset(name, undefined, { urn });
            case "azure-native:deviceregistry:DiscoveredAssetEndpointProfile":
                return new exports.DiscoveredAssetEndpointProfile(name, undefined, { urn });
            case "azure-native:deviceregistry:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure-native:deviceregistry:NamespaceAsset":
                return new exports.NamespaceAsset(name, undefined, { urn });
            case "azure-native:deviceregistry:NamespaceDevice":
                return new exports.NamespaceDevice(name, undefined, { urn });
            case "azure-native:deviceregistry:NamespaceDiscoveredAsset":
                return new exports.NamespaceDiscoveredAsset(name, undefined, { urn });
            case "azure-native:deviceregistry:NamespaceDiscoveredDevice":
                return new exports.NamespaceDiscoveredDevice(name, undefined, { urn });
            case "azure-native:deviceregistry:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:deviceregistry:Schema":
                return new exports.Schema(name, undefined, { urn });
            case "azure-native:deviceregistry:SchemaRegistry":
                return new exports.SchemaRegistry(name, undefined, { urn });
            case "azure-native:deviceregistry:SchemaVersion":
                return new exports.SchemaVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "deviceregistry", _module);
