// @pulumi/aws/eks/capability.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Capability = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an EKS Capability for an EKS cluster.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.eks.Capability("example", {
 *     clusterName: exampleAwsEksCluster.name,
 *     capabilityName: "argocd",
 *     type: "ARGOCD",
 *     roleArn: exampleAwsIamRole.arn,
 *     deletePropagationPolicy: "RETAIN",
 *     configuration: {
 *         argoCd: {
 *             awsIdc: {
 *                 idcInstanceArn: "arn:aws:sso:::instance/ssoins-1234567890abcdef0",
 *             },
 *             namespace: "argocd",
 *         },
 *     },
 *     tags: {
 *         Name: "example-capability",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `clusterName` (String) Name of the EKS Cluster.
 * * `capabilityName` (String) Name of the capability.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Capabilities using `clusterName` and `capabilityName` separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:eks/capability:Capability example example-cluster,example-capability
 * ```
 */
class Capability extends pulumi.CustomResource {
    /**
     * Get an existing Capability resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Capability(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:eks/capability:Capability';
    /**
     * Returns true if the given object is an instance of Capability.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Capability.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["capabilityName"] = state?.capabilityName;
            resourceInputs["clusterName"] = state?.clusterName;
            resourceInputs["configuration"] = state?.configuration;
            resourceInputs["deletePropagationPolicy"] = state?.deletePropagationPolicy;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["type"] = state?.type;
            resourceInputs["version"] = state?.version;
        }
        else {
            const args = argsOrState;
            if (args?.capabilityName === undefined && !opts.urn) {
                throw new Error("Missing required property 'capabilityName'");
            }
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.deletePropagationPolicy === undefined && !opts.urn) {
                throw new Error("Missing required property 'deletePropagationPolicy'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["capabilityName"] = args?.capabilityName;
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["configuration"] = args?.configuration;
            resourceInputs["deletePropagationPolicy"] = args?.deletePropagationPolicy;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["type"] = args?.type;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Capability.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Capability = Capability;
