// @pulumi/gcp/secretmanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretVersion = exports.SecretIamPolicy = exports.SecretIamMember = exports.SecretIamBinding = exports.Secret = exports.RegionalSecretVersion = exports.RegionalSecretIamPolicy = exports.RegionalSecretIamMember = exports.RegionalSecretIamBinding = exports.RegionalSecret = exports.getSecretsOutput = exports.getSecrets = exports.getSecretVersionAccessOutput = exports.getSecretVersionAccess = exports.getSecretVersionOutput = exports.getSecretVersion = exports.getSecretIamPolicyOutput = exports.getSecretIamPolicy = exports.getSecretOutput = exports.getSecret = exports.getRegionalSecretsOutput = exports.getRegionalSecrets = exports.getRegionalSecretVersionAccessOutput = exports.getRegionalSecretVersionAccess = exports.getRegionalSecretVersionOutput = exports.getRegionalSecretVersion = exports.getRegionalSecretIamPolicyOutput = exports.getRegionalSecretIamPolicy = exports.getRegionalSecretOutput = exports.getRegionalSecret = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getRegionalSecret = null;
exports.getRegionalSecretOutput = null;
utilities.lazyLoad(exports, ["getRegionalSecret", "getRegionalSecretOutput"], () => require("./getRegionalSecret"));
exports.getRegionalSecretIamPolicy = null;
exports.getRegionalSecretIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionalSecretIamPolicy", "getRegionalSecretIamPolicyOutput"], () => require("./getRegionalSecretIamPolicy"));
exports.getRegionalSecretVersion = null;
exports.getRegionalSecretVersionOutput = null;
utilities.lazyLoad(exports, ["getRegionalSecretVersion", "getRegionalSecretVersionOutput"], () => require("./getRegionalSecretVersion"));
exports.getRegionalSecretVersionAccess = null;
exports.getRegionalSecretVersionAccessOutput = null;
utilities.lazyLoad(exports, ["getRegionalSecretVersionAccess", "getRegionalSecretVersionAccessOutput"], () => require("./getRegionalSecretVersionAccess"));
exports.getRegionalSecrets = null;
exports.getRegionalSecretsOutput = null;
utilities.lazyLoad(exports, ["getRegionalSecrets", "getRegionalSecretsOutput"], () => require("./getRegionalSecrets"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSecretIamPolicy = null;
exports.getSecretIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecretIamPolicy", "getSecretIamPolicyOutput"], () => require("./getSecretIamPolicy"));
exports.getSecretVersion = null;
exports.getSecretVersionOutput = null;
utilities.lazyLoad(exports, ["getSecretVersion", "getSecretVersionOutput"], () => require("./getSecretVersion"));
exports.getSecretVersionAccess = null;
exports.getSecretVersionAccessOutput = null;
utilities.lazyLoad(exports, ["getSecretVersionAccess", "getSecretVersionAccessOutput"], () => require("./getSecretVersionAccess"));
exports.getSecrets = null;
exports.getSecretsOutput = null;
utilities.lazyLoad(exports, ["getSecrets", "getSecretsOutput"], () => require("./getSecrets"));
exports.RegionalSecret = null;
utilities.lazyLoad(exports, ["RegionalSecret"], () => require("./regionalSecret"));
exports.RegionalSecretIamBinding = null;
utilities.lazyLoad(exports, ["RegionalSecretIamBinding"], () => require("./regionalSecretIamBinding"));
exports.RegionalSecretIamMember = null;
utilities.lazyLoad(exports, ["RegionalSecretIamMember"], () => require("./regionalSecretIamMember"));
exports.RegionalSecretIamPolicy = null;
utilities.lazyLoad(exports, ["RegionalSecretIamPolicy"], () => require("./regionalSecretIamPolicy"));
exports.RegionalSecretVersion = null;
utilities.lazyLoad(exports, ["RegionalSecretVersion"], () => require("./regionalSecretVersion"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SecretIamBinding = null;
utilities.lazyLoad(exports, ["SecretIamBinding"], () => require("./secretIamBinding"));
exports.SecretIamMember = null;
utilities.lazyLoad(exports, ["SecretIamMember"], () => require("./secretIamMember"));
exports.SecretIamPolicy = null;
utilities.lazyLoad(exports, ["SecretIamPolicy"], () => require("./secretIamPolicy"));
exports.SecretVersion = null;
utilities.lazyLoad(exports, ["SecretVersion"], () => require("./secretVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:secretmanager/regionalSecret:RegionalSecret":
                return new exports.RegionalSecret(name, undefined, { urn });
            case "gcp:secretmanager/regionalSecretIamBinding:RegionalSecretIamBinding":
                return new exports.RegionalSecretIamBinding(name, undefined, { urn });
            case "gcp:secretmanager/regionalSecretIamMember:RegionalSecretIamMember":
                return new exports.RegionalSecretIamMember(name, undefined, { urn });
            case "gcp:secretmanager/regionalSecretIamPolicy:RegionalSecretIamPolicy":
                return new exports.RegionalSecretIamPolicy(name, undefined, { urn });
            case "gcp:secretmanager/regionalSecretVersion:RegionalSecretVersion":
                return new exports.RegionalSecretVersion(name, undefined, { urn });
            case "gcp:secretmanager/secret:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "gcp:secretmanager/secretIamBinding:SecretIamBinding":
                return new exports.SecretIamBinding(name, undefined, { urn });
            case "gcp:secretmanager/secretIamMember:SecretIamMember":
                return new exports.SecretIamMember(name, undefined, { urn });
            case "gcp:secretmanager/secretIamPolicy:SecretIamPolicy":
                return new exports.SecretIamPolicy(name, undefined, { urn });
            case "gcp:secretmanager/secretVersion:SecretVersion":
                return new exports.SecretVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "secretmanager/regionalSecret", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/regionalSecretIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/regionalSecretIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/regionalSecretIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/regionalSecretVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/secret", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/secretIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/secretIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/secretIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "secretmanager/secretVersion", _module);
