// @pulumi/azure-native/discovery/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspacePrivateEndpointConnection = exports.Workspace = exports.Tool = exports.Supercomputer = exports.StorageContainer = exports.StorageAsset = exports.Project = exports.NodePool = exports.getWorkspacePrivateEndpointConnectionOutput = exports.getWorkspacePrivateEndpointConnection = exports.getWorkspaceOutput = exports.getWorkspace = exports.getToolOutput = exports.getTool = exports.getSupercomputerOutput = exports.getSupercomputer = exports.getStorageContainerOutput = exports.getStorageContainer = exports.getStorageAssetOutput = exports.getStorageAsset = exports.getProjectOutput = exports.getProject = exports.getNodePoolOutput = exports.getNodePool = exports.getChatModelDeploymentOutput = exports.getChatModelDeployment = exports.getBookshelfPrivateEndpointConnectionOutput = exports.getBookshelfPrivateEndpointConnection = exports.getBookshelfOutput = exports.getBookshelf = exports.ChatModelDeployment = exports.BookshelfPrivateEndpointConnection = exports.Bookshelf = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Bookshelf = null;
utilities.lazyLoad(exports, ["Bookshelf"], () => require("./bookshelf"));
exports.BookshelfPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["BookshelfPrivateEndpointConnection"], () => require("./bookshelfPrivateEndpointConnection"));
exports.ChatModelDeployment = null;
utilities.lazyLoad(exports, ["ChatModelDeployment"], () => require("./chatModelDeployment"));
exports.getBookshelf = null;
exports.getBookshelfOutput = null;
utilities.lazyLoad(exports, ["getBookshelf", "getBookshelfOutput"], () => require("./getBookshelf"));
exports.getBookshelfPrivateEndpointConnection = null;
exports.getBookshelfPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getBookshelfPrivateEndpointConnection", "getBookshelfPrivateEndpointConnectionOutput"], () => require("./getBookshelfPrivateEndpointConnection"));
exports.getChatModelDeployment = null;
exports.getChatModelDeploymentOutput = null;
utilities.lazyLoad(exports, ["getChatModelDeployment", "getChatModelDeploymentOutput"], () => require("./getChatModelDeployment"));
exports.getNodePool = null;
exports.getNodePoolOutput = null;
utilities.lazyLoad(exports, ["getNodePool", "getNodePoolOutput"], () => require("./getNodePool"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getStorageAsset = null;
exports.getStorageAssetOutput = null;
utilities.lazyLoad(exports, ["getStorageAsset", "getStorageAssetOutput"], () => require("./getStorageAsset"));
exports.getStorageContainer = null;
exports.getStorageContainerOutput = null;
utilities.lazyLoad(exports, ["getStorageContainer", "getStorageContainerOutput"], () => require("./getStorageContainer"));
exports.getSupercomputer = null;
exports.getSupercomputerOutput = null;
utilities.lazyLoad(exports, ["getSupercomputer", "getSupercomputerOutput"], () => require("./getSupercomputer"));
exports.getTool = null;
exports.getToolOutput = null;
utilities.lazyLoad(exports, ["getTool", "getToolOutput"], () => require("./getTool"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspacePrivateEndpointConnection = null;
exports.getWorkspacePrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspacePrivateEndpointConnection", "getWorkspacePrivateEndpointConnectionOutput"], () => require("./getWorkspacePrivateEndpointConnection"));
exports.NodePool = null;
utilities.lazyLoad(exports, ["NodePool"], () => require("./nodePool"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.StorageAsset = null;
utilities.lazyLoad(exports, ["StorageAsset"], () => require("./storageAsset"));
exports.StorageContainer = null;
utilities.lazyLoad(exports, ["StorageContainer"], () => require("./storageContainer"));
exports.Supercomputer = null;
utilities.lazyLoad(exports, ["Supercomputer"], () => require("./supercomputer"));
exports.Tool = null;
utilities.lazyLoad(exports, ["Tool"], () => require("./tool"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspacePrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["WorkspacePrivateEndpointConnection"], () => require("./workspacePrivateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/discovery"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:discovery:Bookshelf":
                return new exports.Bookshelf(name, undefined, { urn });
            case "azure-native:discovery:BookshelfPrivateEndpointConnection":
                return new exports.BookshelfPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:discovery:ChatModelDeployment":
                return new exports.ChatModelDeployment(name, undefined, { urn });
            case "azure-native:discovery:NodePool":
                return new exports.NodePool(name, undefined, { urn });
            case "azure-native:discovery:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:discovery:StorageAsset":
                return new exports.StorageAsset(name, undefined, { urn });
            case "azure-native:discovery:StorageContainer":
                return new exports.StorageContainer(name, undefined, { urn });
            case "azure-native:discovery:Supercomputer":
                return new exports.Supercomputer(name, undefined, { urn });
            case "azure-native:discovery:Tool":
                return new exports.Tool(name, undefined, { urn });
            case "azure-native:discovery:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "azure-native:discovery:WorkspacePrivateEndpointConnection":
                return new exports.WorkspacePrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "discovery", _module);
