// @pulumi/azure-native/types/enums/discovery/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmSize = exports.SystemSku = exports.StorageStoreType = exports.ScaleSetPriority = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NetworkEgressType = exports.CustomerManagedKeys = void 0;
exports.CustomerManagedKeys = {
    /**
     * Customer managed keys are enabled.
     */
    Enabled: "Enabled",
    /**
     * Customer managed keys are disabled.
     */
    Disabled: "Disabled",
};
exports.NetworkEgressType = {
    /**
     * Public outbound network via load balancer (Default)
     */
    LoadBalancer: "LoadBalancer",
    /**
     * No default outbound
     */
    None: "None",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    /**
     * Public network access is enabled.
     */
    Enabled: "Enabled",
    /**
     * Public network access is disabled.
     */
    Disabled: "Disabled",
};
exports.ScaleSetPriority = {
    /**
     * Regular priority Virtual Machine Scale Set.
     */
    Regular: "Regular",
    /**
     * Spot priority Virtual Machine Scale Set.
     */
    Spot: "Spot",
};
exports.StorageStoreType = {
    /**
     * The Azure storage blob kind.
     */
    AzureStorageBlob: "AzureStorageBlob",
    /**
     * The Azure NetApp Files kind.
     */
    AzureNetAppFiles: "AzureNetAppFiles",
};
exports.SystemSku = {
    /**
     * Standard_D4s_v6 basic compute VM (default)
     */
    Standard_D4s_v6: "Standard_D4s_v6",
    /**
     * Standard_D4s_v5 SKU
     */
    Standard_D4s_v5: "Standard_D4s_v5",
    /**
     * Standard_D4s_v4 SKU
     */
    Standard_D4s_v4: "Standard_D4s_v4",
};
exports.VmSize = {
    /**
     * Standard_NC24ads_A100_v4 GPU-optimized Azure VM Size
     */
    Standard_NC24ads_A100_v4: "Standard_NC24ads_A100_v4",
    /**
     * Standard_NC48ads_A100_v4 GPU-optimized Azure VM Size
     */
    Standard_NC48ads_A100_v4: "Standard_NC48ads_A100_v4",
    /**
     * Standard_NC96ads_A100_v4 GPU-optimized Azure VM Size
     */
    Standard_NC96ads_A100_v4: "Standard_NC96ads_A100_v4",
    /**
     * Standard_NC4as_T4_v3 GPU-optimized Azure VM Size
     */
    Standard_NC4as_T4_v3: "Standard_NC4as_T4_v3",
    /**
     * Standard_NC8as_T4_v3 GPU-optimized Azure VM Size
     */
    Standard_NC8as_T4_v3: "Standard_NC8as_T4_v3",
    /**
     * Standard_NC16as_T4_v3 GPU-optimized Azure VM Size
     */
    Standard_NC16as_T4_v3: "Standard_NC16as_T4_v3",
    /**
     * Standard_NC64as_T4_v3 GPU-optimized Azure VM Size
     */
    Standard_NC64as_T4_v3: "Standard_NC64as_T4_v3",
    /**
     * Standard_NV6ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV6ads_A10_v5: "Standard_NV6ads_A10_v5",
    /**
     * Standard_NV12ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV12ads_A10_v5: "Standard_NV12ads_A10_v5",
    /**
     * Standard_NV24ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV24ads_A10_v5: "Standard_NV24ads_A10_v5",
    /**
     * Standard_NV36ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV36ads_A10_v5: "Standard_NV36ads_A10_v5",
    /**
     * Standard_NV36ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV36adms_A10_v5: "Standard_NV36adms_A10_v5",
    /**
     * Standard_NV36ads_A10_v5 GPU-optimized Azure VM Size
     */
    Standard_NV72ads_A10_v5: "Standard_NV72ads_A10_v5",
    /**
     * Standard_ND40rs_v2 GPU-optimized Azure VM Size
     */
    Standard_ND40rs_v2: "Standard_ND40rs_v2",
};
