// @pulumi/azure-native/types/enums/netapp/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.SnapshotUsage = exports.SnapshotDirectoryVisibility = exports.SmbNonBrowsable = exports.SmbEncryptionState = exports.SmbAccessBasedEnumeration = exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.ProtocolTypes = exports.PolicyStatus = exports.NetworkFeatures = exports.ManagedServiceIdentityType = exports.LdapState = exports.LdapServerType = exports.KeySource = exports.KerberosState = exports.GlobalFileLockingState = exports.EndpointType = exports.EncryptionType = exports.EncryptionKeySource = exports.EnableWriteBackState = exports.EnableSubvolumes = exports.ElasticVolumePolicyEnforcement = exports.ElasticUnixAccessRule = exports.ElasticSmbEncryption = exports.ElasticServiceLevel = exports.ElasticRootAccess = exports.ElasticProtocolType = exports.ElasticPoolEncryptionKeySource = exports.ElasticNfsv4Access = exports.ElasticNfsv3Access = exports.ElasticBackupPolicyState = exports.DayOfWeek = exports.CoolAccessTieringPolicy = exports.CoolAccessRetrievalPolicy = exports.CifsChangeNotifyState = exports.ChownMode = exports.AvsDataStore = exports.ApplicationType = void 0;
exports.ApplicationType = {
    SAP_HANA: "SAP-HANA",
    ORACLE: "ORACLE",
};
exports.AvsDataStore = {
    /**
     * avsDataStore is enabled
     */
    Enabled: "Enabled",
    /**
     * avsDataStore is disabled
     */
    Disabled: "Disabled",
};
exports.ChownMode = {
    Restricted: "Restricted",
    Unrestricted: "Unrestricted",
};
exports.CifsChangeNotifyState = {
    /**
     * CIFS change notification is disabled
     */
    Disabled: "Disabled",
    /**
     * CIFS change notification is enabled
     */
    Enabled: "Enabled",
};
exports.CoolAccessRetrievalPolicy = {
    Default: "Default",
    OnRead: "OnRead",
    Never: "Never",
};
exports.CoolAccessTieringPolicy = {
    Auto: "Auto",
    SnapshotOnly: "SnapshotOnly",
};
exports.DayOfWeek = {
    /**
     * Take a snapshot each Sunday
     */
    Sunday: "Sunday",
    /**
     * Take a snapshot each Monday
     */
    Monday: "Monday",
    /**
     * Take a snapshot each Tuesday
     */
    Tuesday: "Tuesday",
    /**
     * Take a snapshot each Wednesday
     */
    Wednesday: "Wednesday",
    /**
     * Take a snapshot each Thursday
     */
    Thursday: "Thursday",
    /**
     * Take a snapshot each Friday
     */
    Friday: "Friday",
    /**
     * Take a snapshot each Saturday
     */
    Saturday: "Saturday",
};
exports.ElasticBackupPolicyState = {
    /**
     * Value indicating the policy is enabled
     */
    Enabled: "Enabled",
    /**
     * Value indicating the policy is disabled
     */
    Disabled: "Disabled",
};
exports.ElasticNfsv3Access = {
    /**
     * Clients can connect to the volume using the NFSv3 protocol.
     */
    Enabled: "Enabled",
    /**
     * Clients can't connect to the volume using the NFSv3 protocol.
     */
    Disabled: "Disabled",
};
exports.ElasticNfsv4Access = {
    /**
     * Clients can connect to the volume using the NFSv4 protocol.
     */
    Enabled: "Enabled",
    /**
     * Clients can't connect to the volume using the NFSv4 protocol.
     */
    Disabled: "Disabled",
};
exports.ElasticPoolEncryptionKeySource = {
    /**
     * Represents the encryption key source of Elastic pool is Microsoft.NetApp
     */
    NetApp: "NetApp",
    /**
     * Represents the encryption key source of Elastic pool is Microsoft.KeyVault
     */
    KeyVault: "KeyVault",
};
exports.ElasticProtocolType = {
    /**
     * NFSv3 protocol type
     */
    NFSv3: "NFSv3",
    /**
     * NFSv4 protocol type
     */
    NFSv4: "NFSv4",
    /**
     * SMB/CIFS protocol type
     */
    SMB: "SMB",
};
exports.ElasticRootAccess = {
    /**
     * Root user access is enabled for clients affected by this rule
     */
    Enabled: "Enabled",
    /**
     * Root user access is disabled for clients affected by this rule
     */
    Disabled: "Disabled",
};
exports.ElasticServiceLevel = {
    /**
     * Zone redundant storage service level.
     */
    ZoneRedundant: "ZoneRedundant",
};
exports.ElasticSmbEncryption = {
    /**
     * Value indicating the SMB encryption is enabled
     */
    Enabled: "Enabled",
    /**
     * Value indicating the SMB encryption is disabled
     */
    Disabled: "Disabled",
};
exports.ElasticUnixAccessRule = {
    /**
     * Clients connecting with this rule will only have read access to the volume
     */
    ReadOnly: "ReadOnly",
    /**
     * Clients connecting with this rule will have full read and write access to the volume
     */
    ReadWrite: "ReadWrite",
    /**
     * Clients connecting with this rule will have no access to the volume
     */
    NoAccess: "NoAccess",
};
exports.ElasticVolumePolicyEnforcement = {
    /**
     * Value indicating the policy is enforced on the volume.
     */
    Enforced: "Enforced",
    /**
     * Value indicating the policy is not enforced on the volume.
     */
    NotEnforced: "NotEnforced",
};
exports.EnableSubvolumes = {
    /**
     * subvolumes are enabled
     */
    Enabled: "Enabled",
    /**
     * subvolumes are not enabled
     */
    Disabled: "Disabled",
};
exports.EnableWriteBackState = {
    /**
     * Writeback cache is disabled
     */
    Disabled: "Disabled",
    /**
     * Writeback cache is enabled
     */
    Enabled: "Enabled",
};
exports.EncryptionKeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.EncryptionType = {
    /**
     * EncryptionType Single, volumes will use single encryption at rest
     */
    Single: "Single",
    /**
     * EncryptionType Double, volumes will use double encryption at rest
     */
    Double: "Double",
};
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.GlobalFileLockingState = {
    /**
     * Global file locking is disabled
     */
    Disabled: "Disabled",
    /**
     * Global file locking is enabled
     */
    Enabled: "Enabled",
};
exports.KerberosState = {
    /**
     * Kerberos is disabled
     */
    Disabled: "Disabled",
    /**
     * Kerberos is enabled
     */
    Enabled: "Enabled",
};
exports.KeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.LdapServerType = {
    /**
     * The volume should use Active Directory for LDAP connections.
     */
    ActiveDirectory: "ActiveDirectory",
    /**
     * The volume should use OpenLDAP for LDAP connections.
     */
    OpenLDAP: "OpenLDAP",
};
exports.LdapState = {
    /**
     * ldap is disabled.
     */
    Disabled: "Disabled",
    /**
     * ldap is enabled
     */
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NetworkFeatures = {
    /**
     * Basic network features.
     */
    Basic: "Basic",
    /**
     * Standard network features.
     */
    Standard: "Standard",
    /**
     * Updating from Basic to Standard network features.
     */
    Basic_Standard: "Basic_Standard",
    /**
     * Updating from Standard to Basic network features.
     */
    Standard_Basic: "Standard_Basic",
};
exports.PolicyStatus = {
    /**
     * Value indicating the policy is enabled
     */
    Enabled: "Enabled",
    /**
     * Value indicating the policy is disabled
     */
    Disabled: "Disabled",
};
exports.ProtocolTypes = {
    /**
     * NFSv3 protocol type
     */
    NFSv3: "NFSv3",
    /**
     * NFSv4 protocol type
     */
    NFSv4: "NFSv4",
    /**
     * SMB protocol type
     */
    SMB: "SMB",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
    /**
     * Zone redundant storage service level
     */
    StandardZRS: "StandardZRS",
};
exports.SmbAccessBasedEnumeration = {
    /**
     * smbAccessBasedEnumeration share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbAccessBasedEnumeration share setting is enabled
     */
    Enabled: "Enabled",
};
exports.SmbEncryptionState = {
    /**
     * SMB encryption is disabled
     */
    Disabled: "Disabled",
    /**
     * SMB encryption is enabled
     */
    Enabled: "Enabled",
};
exports.SmbNonBrowsable = {
    /**
     * smbNonBrowsable share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbNonBrowsable share setting is enabled
     */
    Enabled: "Enabled",
};
exports.SnapshotDirectoryVisibility = {
    /**
     * Value indicating the read-only snapshot directory is not visible
     */
    Hidden: "Hidden",
    /**
     * Value indicating the read-only snapshot directory is visible
     */
    Visible: "Visible",
};
exports.SnapshotUsage = {
    /**
     * Value indicating an existing snapshot is used
     */
    UseExistingSnapshot: "UseExistingSnapshot",
    /**
     * Value indicating a new snapshot is created
     */
    CreateNewSnapshot: "CreateNewSnapshot",
};
exports.Type = {
    /**
     * Default user quota
     */
    DefaultUserQuota: "DefaultUserQuota",
    /**
     * Default group quota
     */
    DefaultGroupQuota: "DefaultGroupQuota",
    /**
     * Individual user quota
     */
    IndividualUserQuota: "IndividualUserQuota",
    /**
     * Individual group quota
     */
    IndividualGroupQuota: "IndividualGroupQuota",
};
