// @pulumi/aws/elasticsearch/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcEndpoint = exports.getDomainOutput = exports.getDomain = exports.DomainSamlOptions = exports.DomainPolicy = exports.Domain = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.DomainPolicy = null;
utilities.lazyLoad(exports, ["DomainPolicy"], () => require("./domainPolicy"));
exports.DomainSamlOptions = null;
utilities.lazyLoad(exports, ["DomainSamlOptions"], () => require("./domainSamlOptions"));
exports.getDomain = null;
exports.getDomainOutput = null;
utilities.lazyLoad(exports, ["getDomain", "getDomainOutput"], () => require("./getDomain"));
exports.VpcEndpoint = null;
utilities.lazyLoad(exports, ["VpcEndpoint"], () => require("./vpcEndpoint"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:elasticsearch/domain:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "aws:elasticsearch/domainPolicy:DomainPolicy":
                return new exports.DomainPolicy(name, undefined, { urn });
            case "aws:elasticsearch/domainSamlOptions:DomainSamlOptions":
                return new exports.DomainSamlOptions(name, undefined, { urn });
            case "aws:elasticsearch/vpcEndpoint:VpcEndpoint":
                return new exports.VpcEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "elasticsearch/domain", _module);
pulumi.runtime.registerResourceModule("aws", "elasticsearch/domainPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "elasticsearch/domainSamlOptions", _module);
pulumi.runtime.registerResourceModule("aws", "elasticsearch/vpcEndpoint", _module);
