// @pulumi/azure-native/storagecache/autoExportJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoExportJob = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An auto export job instance. Follows Azure Resource Manager standards: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md
 *
 * Uses Azure REST API version 2024-07-01.
 *
 * Other available API versions: 2025-07-01, 2026-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storagecache [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AutoExportJob extends pulumi.CustomResource {
    /**
     * Get an existing AutoExportJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AutoExportJob(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:storagecache:AutoExportJob';
    /**
     * Returns true if the given object is an instance of AutoExportJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AutoExportJob.__pulumiType;
    }
    /**
     * Create a AutoExportJob resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.amlFilesystemName === undefined && !opts.urn) {
                throw new Error("Missing required property 'amlFilesystemName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["adminStatus"] = (args?.adminStatus) ?? "Enable";
            resourceInputs["amlFilesystemName"] = args?.amlFilesystemName;
            resourceInputs["autoExportJobName"] = args?.autoExportJobName;
            resourceInputs["autoExportPrefixes"] = args?.autoExportPrefixes;
            resourceInputs["location"] = args?.location;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["state"] = args?.state;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["currentIterationFilesDiscovered"] = undefined /*out*/;
            resourceInputs["currentIterationFilesExported"] = undefined /*out*/;
            resourceInputs["currentIterationFilesFailed"] = undefined /*out*/;
            resourceInputs["currentIterationMiBDiscovered"] = undefined /*out*/;
            resourceInputs["currentIterationMiBExported"] = undefined /*out*/;
            resourceInputs["exportIterationCount"] = undefined /*out*/;
            resourceInputs["lastCompletionTimeUTC"] = undefined /*out*/;
            resourceInputs["lastStartedTimeUTC"] = undefined /*out*/;
            resourceInputs["lastSuccessfulIterationCompletionTimeUTC"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["statusCode"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["totalFilesExported"] = undefined /*out*/;
            resourceInputs["totalFilesFailed"] = undefined /*out*/;
            resourceInputs["totalMiBExported"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["adminStatus"] = undefined /*out*/;
            resourceInputs["autoExportPrefixes"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["currentIterationFilesDiscovered"] = undefined /*out*/;
            resourceInputs["currentIterationFilesExported"] = undefined /*out*/;
            resourceInputs["currentIterationFilesFailed"] = undefined /*out*/;
            resourceInputs["currentIterationMiBDiscovered"] = undefined /*out*/;
            resourceInputs["currentIterationMiBExported"] = undefined /*out*/;
            resourceInputs["exportIterationCount"] = undefined /*out*/;
            resourceInputs["lastCompletionTimeUTC"] = undefined /*out*/;
            resourceInputs["lastStartedTimeUTC"] = undefined /*out*/;
            resourceInputs["lastSuccessfulIterationCompletionTimeUTC"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["statusCode"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["totalFilesExported"] = undefined /*out*/;
            resourceInputs["totalFilesFailed"] = undefined /*out*/;
            resourceInputs["totalMiBExported"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storagecache/v20240701:AutoExportJob" }, { type: "azure-native:storagecache/v20250701:AutoExportJob" }, { type: "azure-native:storagecache/v20260101:AutoExportJob" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AutoExportJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AutoExportJob = AutoExportJob;
