// @pulumi/azure-native/storagecache/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageTarget = exports.ImportJob = exports.getStorageTargetOutput = exports.getStorageTarget = exports.getRequiredAmlFSSubnetsSizeOutput = exports.getRequiredAmlFSSubnetsSize = exports.getImportJobOutput = exports.getImportJob = exports.getCacheOutput = exports.getCache = exports.getAutoImportJobOutput = exports.getAutoImportJob = exports.getAutoExportJobOutput = exports.getAutoExportJob = exports.getAmlFilesystemOutput = exports.getAmlFilesystem = exports.Cache = exports.AutoImportJob = exports.AutoExportJob = exports.AmlFilesystem = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AmlFilesystem = null;
utilities.lazyLoad(exports, ["AmlFilesystem"], () => require("./amlFilesystem"));
exports.AutoExportJob = null;
utilities.lazyLoad(exports, ["AutoExportJob"], () => require("./autoExportJob"));
exports.AutoImportJob = null;
utilities.lazyLoad(exports, ["AutoImportJob"], () => require("./autoImportJob"));
exports.Cache = null;
utilities.lazyLoad(exports, ["Cache"], () => require("./cache"));
exports.getAmlFilesystem = null;
exports.getAmlFilesystemOutput = null;
utilities.lazyLoad(exports, ["getAmlFilesystem", "getAmlFilesystemOutput"], () => require("./getAmlFilesystem"));
exports.getAutoExportJob = null;
exports.getAutoExportJobOutput = null;
utilities.lazyLoad(exports, ["getAutoExportJob", "getAutoExportJobOutput"], () => require("./getAutoExportJob"));
exports.getAutoImportJob = null;
exports.getAutoImportJobOutput = null;
utilities.lazyLoad(exports, ["getAutoImportJob", "getAutoImportJobOutput"], () => require("./getAutoImportJob"));
exports.getCache = null;
exports.getCacheOutput = null;
utilities.lazyLoad(exports, ["getCache", "getCacheOutput"], () => require("./getCache"));
exports.getImportJob = null;
exports.getImportJobOutput = null;
utilities.lazyLoad(exports, ["getImportJob", "getImportJobOutput"], () => require("./getImportJob"));
exports.getRequiredAmlFSSubnetsSize = null;
exports.getRequiredAmlFSSubnetsSizeOutput = null;
utilities.lazyLoad(exports, ["getRequiredAmlFSSubnetsSize", "getRequiredAmlFSSubnetsSizeOutput"], () => require("./getRequiredAmlFSSubnetsSize"));
exports.getStorageTarget = null;
exports.getStorageTargetOutput = null;
utilities.lazyLoad(exports, ["getStorageTarget", "getStorageTargetOutput"], () => require("./getStorageTarget"));
exports.ImportJob = null;
utilities.lazyLoad(exports, ["ImportJob"], () => require("./importJob"));
exports.StorageTarget = null;
utilities.lazyLoad(exports, ["StorageTarget"], () => require("./storageTarget"));
// Export enums:
__exportStar(require("../types/enums/storagecache"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storagecache:AmlFilesystem":
                return new exports.AmlFilesystem(name, undefined, { urn });
            case "azure-native:storagecache:AutoExportJob":
                return new exports.AutoExportJob(name, undefined, { urn });
            case "azure-native:storagecache:AutoImportJob":
                return new exports.AutoImportJob(name, undefined, { urn });
            case "azure-native:storagecache:Cache":
                return new exports.Cache(name, undefined, { urn });
            case "azure-native:storagecache:ImportJob":
                return new exports.ImportJob(name, undefined, { urn });
            case "azure-native:storagecache:StorageTarget":
                return new exports.StorageTarget(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storagecache", _module);
