// @pulumi/azure-native/types/enums/edge/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceState = exports.ReconciliationState = exports.ExtendedLocationType = exports.ErrorActionMode = exports.ActiveState = void 0;
exports.ActiveState = {
    /**
     * Instance is active
     */
    Active: "active",
    /**
     * Instance is inactive
     */
    Inactive: "inactive",
};
exports.ErrorActionMode = {
    /**
     * Stop on any failure
     */
    StopOnAnyFailure: "stopOnAnyFailure",
    /**
     * Stop after N cumulative failures
     */
    StopOnNFailures: "stopOnNFailures",
    /**
     * Continue silently despite errors
     */
    SilentlyContinue: "silentlyContinue",
};
exports.ExtendedLocationType = {
    /**
     * Azure Edge Zones location type
     */
    EdgeZone: "EdgeZone",
    /**
     * Azure Custom Locations type
     */
    CustomLocation: "CustomLocation",
};
exports.ReconciliationState = {
    /**
     * Reconciliation is inactive
     */
    Inactive: "inactive",
    /**
     * Reconciliation is active
     */
    Active: "active",
};
exports.ResourceState = {
    /**
     * Resource is active
     */
    Active: "active",
    /**
     * Resource is inactive
     */
    Inactive: "inactive",
};
