// @pulumi/aws/chime/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceConnectorTerminationCredentials = exports.VoiceConnectorTermination = exports.VoiceConnectorStreaming = exports.VoiceConnectorOrigination = exports.VoiceConnectorLogging = exports.VoiceConnectorGroup = exports.VoiceConnector = exports.SdkvoiceVoiceProfileDomain = exports.SdkvoiceSipRule = exports.SdkvoiceSipMediaApplication = exports.SdkvoiceGlobalSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.SdkvoiceGlobalSettings = null;
utilities.lazyLoad(exports, ["SdkvoiceGlobalSettings"], () => require("./sdkvoiceGlobalSettings"));
exports.SdkvoiceSipMediaApplication = null;
utilities.lazyLoad(exports, ["SdkvoiceSipMediaApplication"], () => require("./sdkvoiceSipMediaApplication"));
exports.SdkvoiceSipRule = null;
utilities.lazyLoad(exports, ["SdkvoiceSipRule"], () => require("./sdkvoiceSipRule"));
exports.SdkvoiceVoiceProfileDomain = null;
utilities.lazyLoad(exports, ["SdkvoiceVoiceProfileDomain"], () => require("./sdkvoiceVoiceProfileDomain"));
exports.VoiceConnector = null;
utilities.lazyLoad(exports, ["VoiceConnector"], () => require("./voiceConnector"));
exports.VoiceConnectorGroup = null;
utilities.lazyLoad(exports, ["VoiceConnectorGroup"], () => require("./voiceConnectorGroup"));
exports.VoiceConnectorLogging = null;
utilities.lazyLoad(exports, ["VoiceConnectorLogging"], () => require("./voiceConnectorLogging"));
exports.VoiceConnectorOrigination = null;
utilities.lazyLoad(exports, ["VoiceConnectorOrigination"], () => require("./voiceConnectorOrigination"));
exports.VoiceConnectorStreaming = null;
utilities.lazyLoad(exports, ["VoiceConnectorStreaming"], () => require("./voiceConnectorStreaming"));
exports.VoiceConnectorTermination = null;
utilities.lazyLoad(exports, ["VoiceConnectorTermination"], () => require("./voiceConnectorTermination"));
exports.VoiceConnectorTerminationCredentials = null;
utilities.lazyLoad(exports, ["VoiceConnectorTerminationCredentials"], () => require("./voiceConnectorTerminationCredentials"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:chime/sdkvoiceGlobalSettings:SdkvoiceGlobalSettings":
                return new exports.SdkvoiceGlobalSettings(name, undefined, { urn });
            case "aws:chime/sdkvoiceSipMediaApplication:SdkvoiceSipMediaApplication":
                return new exports.SdkvoiceSipMediaApplication(name, undefined, { urn });
            case "aws:chime/sdkvoiceSipRule:SdkvoiceSipRule":
                return new exports.SdkvoiceSipRule(name, undefined, { urn });
            case "aws:chime/sdkvoiceVoiceProfileDomain:SdkvoiceVoiceProfileDomain":
                return new exports.SdkvoiceVoiceProfileDomain(name, undefined, { urn });
            case "aws:chime/voiceConnector:VoiceConnector":
                return new exports.VoiceConnector(name, undefined, { urn });
            case "aws:chime/voiceConnectorGroup:VoiceConnectorGroup":
                return new exports.VoiceConnectorGroup(name, undefined, { urn });
            case "aws:chime/voiceConnectorLogging:VoiceConnectorLogging":
                return new exports.VoiceConnectorLogging(name, undefined, { urn });
            case "aws:chime/voiceConnectorOrigination:VoiceConnectorOrigination":
                return new exports.VoiceConnectorOrigination(name, undefined, { urn });
            case "aws:chime/voiceConnectorStreaming:VoiceConnectorStreaming":
                return new exports.VoiceConnectorStreaming(name, undefined, { urn });
            case "aws:chime/voiceConnectorTermination:VoiceConnectorTermination":
                return new exports.VoiceConnectorTermination(name, undefined, { urn });
            case "aws:chime/voiceConnectorTerminationCredentials:VoiceConnectorTerminationCredentials":
                return new exports.VoiceConnectorTerminationCredentials(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "chime/sdkvoiceGlobalSettings", _module);
pulumi.runtime.registerResourceModule("aws", "chime/sdkvoiceSipMediaApplication", _module);
pulumi.runtime.registerResourceModule("aws", "chime/sdkvoiceSipRule", _module);
pulumi.runtime.registerResourceModule("aws", "chime/sdkvoiceVoiceProfileDomain", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnector", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorGroup", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorLogging", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorOrigination", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorStreaming", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorTermination", _module);
pulumi.runtime.registerResourceModule("aws", "chime/voiceConnectorTerminationCredentials", _module);
