// @pulumi/aws/ec2/eipDomainName.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EipDomainName = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Assigns a static reverse DNS record to an Elastic IP addresses. See [Using reverse DNS for email applications](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#Using_Elastic_Addressing_Reverse_DNS).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.Eip("example", {domain: "vpc"});
 * const exampleRecord = new aws.route53.Record("example", {
 *     zoneId: main.zoneId,
 *     name: "reverse",
 *     type: aws.route53.RecordType.A,
 *     records: [example.publicIp],
 * });
 * const exampleEipDomainName = new aws.ec2.EipDomainName("example", {
 *     allocationId: example.allocationId,
 *     domainName: exampleRecord.fqdn,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a static reverse DNS record to an Elastic IP addresses using their association IDs. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/eipDomainName:EipDomainName test eipassoc-ab12c345
 * ```
 */
class EipDomainName extends pulumi.CustomResource {
    /**
     * Get an existing EipDomainName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EipDomainName(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/eipDomainName:EipDomainName';
    /**
     * Returns true if the given object is an instance of EipDomainName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EipDomainName.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allocationId"] = state?.allocationId;
            resourceInputs["domainName"] = state?.domainName;
            resourceInputs["ptrRecord"] = state?.ptrRecord;
            resourceInputs["region"] = state?.region;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.allocationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'allocationId'");
            }
            if (args?.domainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            resourceInputs["allocationId"] = args?.allocationId;
            resourceInputs["domainName"] = args?.domainName;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["ptrRecord"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EipDomainName.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EipDomainName = EipDomainName;
