// @pulumi/aws/ec2/networkInterfacePermission.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkInterfacePermission = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Grant cross-account access to an Elastic network interface (ENI).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.NetworkInterface("example", {
 *     subnetId: exampleAwsSubnet.id,
 *     privateIps: ["10.0.0.50"],
 *     securityGroups: [exampleAwsSecurityGroup.id],
 *     attachments: [{
 *         instance: exampleAwsInstance.id,
 *         deviceIndex: 1,
 *     }],
 * });
 * const exampleNetworkInterfacePermission = new aws.ec2.NetworkInterfacePermission("example", {
 *     networkInterfaceId: example.id,
 *     awsAccountId: "123456789012",
 *     permission: "INSTANCE-ATTACH",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Network Interface Permissions using the `networkInterfacePermissionId`. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/networkInterfacePermission:NetworkInterfacePermission example eni-perm-056ad97ce2ac377ed
 * ```
 */
class NetworkInterfacePermission extends pulumi.CustomResource {
    /**
     * Get an existing NetworkInterfacePermission resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkInterfacePermission(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/networkInterfacePermission:NetworkInterfacePermission';
    /**
     * Returns true if the given object is an instance of NetworkInterfacePermission.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkInterfacePermission.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["awsAccountId"] = state?.awsAccountId;
            resourceInputs["networkInterfaceId"] = state?.networkInterfaceId;
            resourceInputs["networkInterfacePermissionId"] = state?.networkInterfacePermissionId;
            resourceInputs["permission"] = state?.permission;
            resourceInputs["region"] = state?.region;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.awsAccountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'awsAccountId'");
            }
            if (args?.networkInterfaceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkInterfaceId'");
            }
            if (args?.permission === undefined && !opts.urn) {
                throw new Error("Missing required property 'permission'");
            }
            resourceInputs["awsAccountId"] = args?.awsAccountId;
            resourceInputs["networkInterfaceId"] = args?.networkInterfaceId;
            resourceInputs["permission"] = args?.permission;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["networkInterfacePermissionId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkInterfacePermission.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkInterfacePermission = NetworkInterfacePermission;
