// @pulumi/aws/ec2/secondarySubnet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecondarySubnet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an EC2 Secondary Subnet resource.
 *
 * A secondary subnet is a subnet within a secondary network that provides high-performance networking capabilities for specialized workloads such as RDMA (Remote Direct Memory Access) applications.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.SecondaryNetwork("example", {
 *     ipv4CidrBlock: "10.0.0.0/16",
 *     networkType: "rdma",
 *     tags: {
 *         Name: "example-secondary-network",
 *     },
 * });
 * const exampleSecondarySubnet = new aws.ec2.SecondarySubnet("example", {
 *     secondaryNetworkId: example.id,
 *     ipv4CidrBlock: "10.0.1.0/24",
 *     availabilityZone: "us-west-2a",
 *     tags: {
 *         Name: "example-secondary-subnet",
 *     },
 * });
 * ```
 *
 * ### Using Availability Zone ID
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const available = aws.getAvailabilityZones({
 *     state: "available",
 *     filters: [{
 *         name: "opt-in-status",
 *         values: ["opt-in-not-required"],
 *     }],
 * });
 * const example = new aws.ec2.SecondaryNetwork("example", {
 *     ipv4CidrBlock: "10.0.0.0/16",
 *     networkType: "rdma",
 *     tags: {
 *         Name: "example-secondary-network",
 *     },
 * });
 * const exampleSecondarySubnet = new aws.ec2.SecondarySubnet("example", {
 *     secondaryNetworkId: example.id,
 *     ipv4CidrBlock: "10.0.1.0/24",
 *     availabilityZoneId: available.then(available => available.zoneIds?.[0]),
 *     tags: {
 *         Name: "example-secondary-subnet",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `id` - (String) ID of the secondary subnet.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import EC2 Secondary Subnets using the secondary subnet ID. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/secondarySubnet:SecondarySubnet example ss-0123456789abcdef0
 * ```
 */
class SecondarySubnet extends pulumi.CustomResource {
    /**
     * Get an existing SecondarySubnet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SecondarySubnet(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/secondarySubnet:SecondarySubnet';
    /**
     * Returns true if the given object is an instance of SecondarySubnet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecondarySubnet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["availabilityZone"] = state?.availabilityZone;
            resourceInputs["availabilityZoneId"] = state?.availabilityZoneId;
            resourceInputs["ipv4CidrBlock"] = state?.ipv4CidrBlock;
            resourceInputs["ipv4CidrBlockAssociations"] = state?.ipv4CidrBlockAssociations;
            resourceInputs["ownerId"] = state?.ownerId;
            resourceInputs["region"] = state?.region;
            resourceInputs["secondaryNetworkId"] = state?.secondaryNetworkId;
            resourceInputs["secondaryNetworkType"] = state?.secondaryNetworkType;
            resourceInputs["secondarySubnetId"] = state?.secondarySubnetId;
            resourceInputs["state"] = state?.state;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.ipv4CidrBlock === undefined && !opts.urn) {
                throw new Error("Missing required property 'ipv4CidrBlock'");
            }
            if (args?.secondaryNetworkId === undefined && !opts.urn) {
                throw new Error("Missing required property 'secondaryNetworkId'");
            }
            resourceInputs["availabilityZone"] = args?.availabilityZone;
            resourceInputs["availabilityZoneId"] = args?.availabilityZoneId;
            resourceInputs["ipv4CidrBlock"] = args?.ipv4CidrBlock;
            resourceInputs["region"] = args?.region;
            resourceInputs["secondaryNetworkId"] = args?.secondaryNetworkId;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["ipv4CidrBlockAssociations"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["secondaryNetworkType"] = undefined /*out*/;
            resourceInputs["secondarySubnetId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecondarySubnet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecondarySubnet = SecondarySubnet;
