// @pulumi/aws/ec2/trafficMirrorFilterRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficMirrorFilterRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Traffic mirror filter rule.\
 * Read [limits and considerations](https://docs.aws.amazon.com/vpc/latest/mirroring/traffic-mirroring-considerations.html) for traffic mirroring
 *
 * ## Example Usage
 *
 * To create a basic traffic mirror session
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const filter = new aws.ec2.TrafficMirrorFilter("filter", {
 *     description: "traffic mirror filter - example",
 *     networkServices: ["amazon-dns"],
 * });
 * const ruleout = new aws.ec2.TrafficMirrorFilterRule("ruleout", {
 *     description: "test rule",
 *     trafficMirrorFilterId: filter.id,
 *     destinationCidrBlock: "10.0.0.0/8",
 *     sourceCidrBlock: "10.0.0.0/8",
 *     ruleNumber: 1,
 *     ruleAction: "accept",
 *     trafficDirection: "egress",
 * });
 * const rulein = new aws.ec2.TrafficMirrorFilterRule("rulein", {
 *     description: "test rule",
 *     trafficMirrorFilterId: filter.id,
 *     destinationCidrBlock: "10.0.0.0/8",
 *     sourceCidrBlock: "10.0.0.0/8",
 *     ruleNumber: 1,
 *     ruleAction: "accept",
 *     trafficDirection: "ingress",
 *     protocol: 6,
 *     destinationPortRange: {
 *         fromPort: 22,
 *         toPort: 53,
 *     },
 *     sourcePortRange: {
 *         fromPort: 0,
 *         toPort: 10,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import traffic mirror rules using the `trafficMirrorFilterId` and `id` separated by `:`. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/trafficMirrorFilterRule:TrafficMirrorFilterRule rule tmf-0fbb93ddf38198f64:tmfr-05a458f06445d0aee
 * ```
 */
class TrafficMirrorFilterRule extends pulumi.CustomResource {
    /**
     * Get an existing TrafficMirrorFilterRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TrafficMirrorFilterRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/trafficMirrorFilterRule:TrafficMirrorFilterRule';
    /**
     * Returns true if the given object is an instance of TrafficMirrorFilterRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrafficMirrorFilterRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["destinationCidrBlock"] = state?.destinationCidrBlock;
            resourceInputs["destinationPortRange"] = state?.destinationPortRange;
            resourceInputs["protocol"] = state?.protocol;
            resourceInputs["region"] = state?.region;
            resourceInputs["ruleAction"] = state?.ruleAction;
            resourceInputs["ruleNumber"] = state?.ruleNumber;
            resourceInputs["sourceCidrBlock"] = state?.sourceCidrBlock;
            resourceInputs["sourcePortRange"] = state?.sourcePortRange;
            resourceInputs["trafficDirection"] = state?.trafficDirection;
            resourceInputs["trafficMirrorFilterId"] = state?.trafficMirrorFilterId;
        }
        else {
            const args = argsOrState;
            if (args?.destinationCidrBlock === undefined && !opts.urn) {
                throw new Error("Missing required property 'destinationCidrBlock'");
            }
            if (args?.ruleAction === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleAction'");
            }
            if (args?.ruleNumber === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleNumber'");
            }
            if (args?.sourceCidrBlock === undefined && !opts.urn) {
                throw new Error("Missing required property 'sourceCidrBlock'");
            }
            if (args?.trafficDirection === undefined && !opts.urn) {
                throw new Error("Missing required property 'trafficDirection'");
            }
            if (args?.trafficMirrorFilterId === undefined && !opts.urn) {
                throw new Error("Missing required property 'trafficMirrorFilterId'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["destinationCidrBlock"] = args?.destinationCidrBlock;
            resourceInputs["destinationPortRange"] = args?.destinationPortRange;
            resourceInputs["protocol"] = args?.protocol;
            resourceInputs["region"] = args?.region;
            resourceInputs["ruleAction"] = args?.ruleAction;
            resourceInputs["ruleNumber"] = args?.ruleNumber;
            resourceInputs["sourceCidrBlock"] = args?.sourceCidrBlock;
            resourceInputs["sourcePortRange"] = args?.sourcePortRange;
            resourceInputs["trafficDirection"] = args?.trafficDirection;
            resourceInputs["trafficMirrorFilterId"] = args?.trafficMirrorFilterId;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TrafficMirrorFilterRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrafficMirrorFilterRule = TrafficMirrorFilterRule;
