// @pulumi/aws/ec2/vpcIpamOrganizationAdminAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcIpamOrganizationAdminAccount = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Enables the IPAM Service and promotes a delegated administrator.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const delegated = aws.getCallerIdentity({});
 * const example = new aws.ec2.VpcIpamOrganizationAdminAccount("example", {delegatedAdminAccountId: delegated.then(delegated => delegated.accountId)});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import IPAMs using the delegate account `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/vpcIpamOrganizationAdminAccount:VpcIpamOrganizationAdminAccount example 12345678901
 * ```
 */
class VpcIpamOrganizationAdminAccount extends pulumi.CustomResource {
    /**
     * Get an existing VpcIpamOrganizationAdminAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcIpamOrganizationAdminAccount(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/vpcIpamOrganizationAdminAccount:VpcIpamOrganizationAdminAccount';
    /**
     * Returns true if the given object is an instance of VpcIpamOrganizationAdminAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcIpamOrganizationAdminAccount.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["delegatedAdminAccountId"] = state?.delegatedAdminAccountId;
            resourceInputs["email"] = state?.email;
            resourceInputs["name"] = state?.name;
            resourceInputs["servicePrincipal"] = state?.servicePrincipal;
        }
        else {
            const args = argsOrState;
            if (args?.delegatedAdminAccountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'delegatedAdminAccountId'");
            }
            resourceInputs["delegatedAdminAccountId"] = args?.delegatedAdminAccountId;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["email"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["servicePrincipal"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcIpamOrganizationAdminAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcIpamOrganizationAdminAccount = VpcIpamOrganizationAdminAccount;
