// @pulumi/aws/ec2/vpcSecurityGroupRulesExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcSecurityGroupRulesExclusive = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an exclusive set of AWS VPC (Virtual Private Cloud) Security Group Rules.
 *
 * This resource manages the complete set of ingress and egress rules assigned to a security group. It provides exclusive control by removing any rules not explicitly defined in the configuration.
 *
 * !> This resource takes exclusive ownership over ingress and egress rules assigned to a security group. This includes removal of rules which are not explicitly configured. To prevent persistent drift, ensure any `aws.vpc.SecurityGroupIngressRule` and `aws.vpc.SecurityGroupEgressRule` resources managed alongside this resource are included in the `ingressRuleIds` and `egressRuleIds` arguments.
 *
 * > Destruction of this resource means Terraform will no longer manage reconciliation of the configured security group rules. It **will not** revoke the configured rules from the security group.
 *
 * > When this resource detects a configured rule ID which must be created, a warning diagnostic is emitted. This is due to a limitation in the [`AuthorizeSecurityGroupEgress`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AuthorizeSecurityGroupEgress.html) and [`AuthorizeSecurityGroupIngress`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AuthorizeSecurityGroupIngress.html) APIs, which require the full rule definition to be provided rather than a reference to an existing rule ID.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.Vpc("example", {cidrBlock: "10.0.0.0/16"});
 * const exampleSecurityGroup = new aws.ec2.SecurityGroup("example", {
 *     name: "example",
 *     vpcId: example.id,
 * });
 * const exampleSecurityGroupIngressRule = new aws.vpc.SecurityGroupIngressRule("example", {
 *     securityGroupId: exampleSecurityGroup.id,
 *     cidrIpv4: "10.0.0.0/8",
 *     fromPort: 80,
 *     toPort: 80,
 *     ipProtocol: "tcp",
 * });
 * const exampleSecurityGroupEgressRule = new aws.vpc.SecurityGroupEgressRule("example", {
 *     securityGroupId: exampleSecurityGroup.id,
 *     cidrIpv4: "0.0.0.0/0",
 *     ipProtocol: "-1",
 * });
 * const exampleVpcSecurityGroupRulesExclusive = new aws.ec2.VpcSecurityGroupRulesExclusive("example", {
 *     securityGroupId: exampleSecurityGroup.id,
 *     ingressRuleIds: [exampleSecurityGroupIngressRule.id],
 *     egressRuleIds: [exampleSecurityGroupEgressRule.id],
 * });
 * ```
 *
 * ### Disallow All Rules
 *
 * To automatically remove any configured security group rules, set both `ingressRuleIds` and `egressRuleIds` to empty lists.
 *
 * > This will not __prevent__ rules from being assigned to a security group via Terraform (or any other interface). This resource enables bringing security group rule assignments into a configured state, however, this reconciliation happens only when `apply` is proactively run.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.VpcSecurityGroupRulesExclusive("example", {
 *     securityGroupId: exampleAwsSecurityGroup.id,
 *     ingressRuleIds: [],
 *     egressRuleIds: [],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import exclusive management of security group rules using the `securityGroupId`. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/vpcSecurityGroupRulesExclusive:VpcSecurityGroupRulesExclusive example sg-1234567890abcdef0
 * ```
 */
class VpcSecurityGroupRulesExclusive extends pulumi.CustomResource {
    /**
     * Get an existing VpcSecurityGroupRulesExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcSecurityGroupRulesExclusive(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ec2/vpcSecurityGroupRulesExclusive:VpcSecurityGroupRulesExclusive';
    /**
     * Returns true if the given object is an instance of VpcSecurityGroupRulesExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcSecurityGroupRulesExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["egressRuleIds"] = state?.egressRuleIds;
            resourceInputs["ingressRuleIds"] = state?.ingressRuleIds;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroupId"] = state?.securityGroupId;
        }
        else {
            const args = argsOrState;
            if (args?.egressRuleIds === undefined && !opts.urn) {
                throw new Error("Missing required property 'egressRuleIds'");
            }
            if (args?.ingressRuleIds === undefined && !opts.urn) {
                throw new Error("Missing required property 'ingressRuleIds'");
            }
            if (args?.securityGroupId === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGroupId'");
            }
            resourceInputs["egressRuleIds"] = args?.egressRuleIds;
            resourceInputs["ingressRuleIds"] = args?.ingressRuleIds;
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroupId"] = args?.securityGroupId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcSecurityGroupRulesExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcSecurityGroupRulesExclusive = VpcSecurityGroupRulesExclusive;
