// @pulumi/azure-native/offazure/webAppDiscoverySiteDataSourcesController.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppDiscoverySiteDataSourcesController = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Web app data source web model.
 *
 * Uses Azure REST API version 2023-10-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-06-06.
 *
 * Other available API versions: 2023-06-06, 2024-05-01-preview, 2024-07-01-preview, 2024-12-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native offazure [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppDiscoverySiteDataSourcesController extends pulumi.CustomResource {
    /**
     * Get an existing WebAppDiscoverySiteDataSourcesController resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppDiscoverySiteDataSourcesController(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:offazure:WebAppDiscoverySiteDataSourcesController';
    /**
     * Returns true if the given object is an instance of WebAppDiscoverySiteDataSourcesController.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppDiscoverySiteDataSourcesController.__pulumiType;
    }
    /**
     * Create a WebAppDiscoverySiteDataSourcesController resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.siteName === undefined && !opts.urn) {
                throw new Error("Missing required property 'siteName'");
            }
            if (args?.webAppSiteName === undefined && !opts.urn) {
                throw new Error("Missing required property 'webAppSiteName'");
            }
            resourceInputs["discoverySiteDataSourceName"] = args?.discoverySiteDataSourceName;
            resourceInputs["discoverySiteId"] = args?.discoverySiteId;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["siteName"] = args?.siteName;
            resourceInputs["webAppSiteName"] = args?.webAppSiteName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["discoverySiteId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:offazure/v20230606:WebAppDiscoverySiteDataSourcesController" }, { type: "azure-native:offazure/v20231001preview:WebAppDiscoverySiteDataSourcesController" }, { type: "azure-native:offazure/v20240501preview:WebAppDiscoverySiteDataSourcesController" }, { type: "azure-native:offazure/v20240701preview:WebAppDiscoverySiteDataSourcesController" }, { type: "azure-native:offazure/v20241201preview:WebAppDiscoverySiteDataSourcesController" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppDiscoverySiteDataSourcesController.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppDiscoverySiteDataSourcesController = WebAppDiscoverySiteDataSourcesController;
