// @pulumi/gcp/storage/controlOrganizationIntelligenceConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ControlOrganizationIntelligenceConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The Organization Storage Intelligence Config resource represents GCS Storage Intelligence operating on individual GCP organization. Storage Intelligence Config is a singleton resource and individual instance exists on each GCP organization.
 *
 * Storage Intelligence is for Storage Admins to manage GCP storage assets at scale for performance, cost, security & compliance.
 *
 * > **Warning:** Storage Intelligence is a singleton resource which cannot be created or deleted. A single instance of Storage Intelligence exist for each GCP Organization. Terraform does not create or destroy this resource.
 * Terraform resource creation for this resource is simply an update operation on existing resource with specified properties, absence of any optional field in the create operation will result in clearance of that field. Terraform deletion won't have any effect on this resource rather it will only remove it from the state file.
 *
 * ## Example Usage
 *
 * ### Storage Control Organization Intelligence Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.storage.ControlOrganizationIntelligenceConfig("example", {
 *     name: "12345678",
 *     editionConfig: "STANDARD",
 *     filter: {
 *         excludedCloudStorageBuckets: {
 *             bucketIdRegexes: [
 *                 "test-id-1*",
 *                 "test-id-2*",
 *             ],
 *         },
 *         includedCloudStorageLocations: {
 *             locations: [
 *                 "test-location-1*",
 *                 "test-locations-2*",
 *             ],
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * OrganizationIntelligenceConfig can be imported using any of these accepted formats:
 *
 * * `organizations/{{name}}/locations/global/intelligenceConfig`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, OrganizationIntelligenceConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:storage/controlOrganizationIntelligenceConfig:ControlOrganizationIntelligenceConfig default organizations/{{name}}/locations/global/intelligenceConfig
 * $ pulumi import gcp:storage/controlOrganizationIntelligenceConfig:ControlOrganizationIntelligenceConfig default {{name}}
 * ```
 */
class ControlOrganizationIntelligenceConfig extends pulumi.CustomResource {
    /**
     * Get an existing ControlOrganizationIntelligenceConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ControlOrganizationIntelligenceConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:storage/controlOrganizationIntelligenceConfig:ControlOrganizationIntelligenceConfig';
    /**
     * Returns true if the given object is an instance of ControlOrganizationIntelligenceConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ControlOrganizationIntelligenceConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["editionConfig"] = state?.editionConfig;
            resourceInputs["effectiveIntelligenceConfigs"] = state?.effectiveIntelligenceConfigs;
            resourceInputs["filter"] = state?.filter;
            resourceInputs["name"] = state?.name;
            resourceInputs["trialConfigs"] = state?.trialConfigs;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            resourceInputs["editionConfig"] = args?.editionConfig;
            resourceInputs["filter"] = args?.filter;
            resourceInputs["name"] = args?.name;
            resourceInputs["effectiveIntelligenceConfigs"] = undefined /*out*/;
            resourceInputs["trialConfigs"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ControlOrganizationIntelligenceConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ControlOrganizationIntelligenceConfig = ControlOrganizationIntelligenceConfig;
