// @pulumi/gcp/storage/transferAgentPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferAgentPool = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents an On-Premises Agent pool.
 *
 * To get more information about AgentPool, see:
 *
 * * [API documentation](https://cloud.google.com/storage-transfer/docs/reference/rest/v1/projects.agentPools)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/storage-transfer/docs/on-prem-agent-pools)
 *
 * ## Example Usage
 *
 * ### Agent Pool Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = gcp.storage.getTransferProjectServiceAccount({
 *     project: "my-project-name",
 * });
 * const pubsubEditorRole = new gcp.projects.IAMMember("pubsub_editor_role", {
 *     project: "my-project-name",
 *     role: "roles/pubsub.editor",
 *     member: _default.then(_default => `serviceAccount:${_default.email}`),
 * });
 * const example = new gcp.storage.TransferAgentPool("example", {
 *     name: "agent-pool-example",
 *     displayName: "Source A to destination Z",
 *     bandwidthLimit: {
 *         limitMbps: "120",
 *     },
 * }, {
 *     dependsOn: [pubsubEditorRole],
 * });
 * ```
 *
 * ## Import
 *
 * AgentPool can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/agentPools/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, AgentPool can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:storage/transferAgentPool:TransferAgentPool default projects/{{project}}/agentPools/{{name}}
 * $ pulumi import gcp:storage/transferAgentPool:TransferAgentPool default {{project}}/{{name}}
 * $ pulumi import gcp:storage/transferAgentPool:TransferAgentPool default {{name}}
 * ```
 */
class TransferAgentPool extends pulumi.CustomResource {
    /**
     * Get an existing TransferAgentPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TransferAgentPool(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:storage/transferAgentPool:TransferAgentPool';
    /**
     * Returns true if the given object is an instance of TransferAgentPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TransferAgentPool.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bandwidthLimit"] = state?.bandwidthLimit;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["state"] = state?.state;
        }
        else {
            const args = argsOrState;
            resourceInputs["bandwidthLimit"] = args?.bandwidthLimit;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TransferAgentPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TransferAgentPool = TransferAgentPool;
