// @pulumi/aws/lakeformation/optIn.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptIn = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS Lake Formation Opt In.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.lakeformation.OptIn("example", {});
 * ```
 */
class OptIn extends pulumi.CustomResource {
    /**
     * Get an existing OptIn resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OptIn(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:lakeformation/optIn:OptIn';
    /**
     * Returns true if the given object is an instance of OptIn.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OptIn.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["conditions"] = state?.conditions;
            resourceInputs["lastModified"] = state?.lastModified;
            resourceInputs["lastUpdatedBy"] = state?.lastUpdatedBy;
            resourceInputs["principals"] = state?.principals;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceDatas"] = state?.resourceDatas;
        }
        else {
            const args = argsOrState;
            resourceInputs["conditions"] = args?.conditions;
            resourceInputs["principals"] = args?.principals;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceDatas"] = args?.resourceDatas;
            resourceInputs["lastModified"] = undefined /*out*/;
            resourceInputs["lastUpdatedBy"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(OptIn.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OptIn = OptIn;
