// @pulumi/aws/lakeformation/resourceLfTags.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceLfTags = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an attachment between one or more existing LF-tags and an existing Lake Formation resource.
 *
 * ## Example Usage
 *
 * ### Database Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.lakeformation.LfTag("example", {
 *     key: "right",
 *     values: [
 *         "abbey",
 *         "village",
 *         "luffield",
 *         "woodcote",
 *         "copse",
 *         "chapel",
 *         "stowe",
 *         "club",
 *     ],
 * });
 * const exampleResourceLfTags = new aws.lakeformation.ResourceLfTags("example", {
 *     database: {
 *         name: exampleAwsGlueCatalogDatabase.name,
 *     },
 *     lfTags: [{
 *         key: example.key,
 *         value: "stowe",
 *     }],
 * });
 * ```
 *
 * ### Multiple Tags Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.lakeformation.LfTag("example", {
 *     key: "right",
 *     values: [
 *         "abbey",
 *         "village",
 *         "luffield",
 *         "woodcote",
 *         "copse",
 *         "chapel",
 *         "stowe",
 *         "club",
 *     ],
 * });
 * const example2 = new aws.lakeformation.LfTag("example2", {
 *     key: "left",
 *     values: [
 *         "farm",
 *         "theloop",
 *         "aintree",
 *         "brooklands",
 *         "maggotts",
 *         "becketts",
 *         "vale",
 *     ],
 * });
 * const exampleResourceLfTags = new aws.lakeformation.ResourceLfTags("example", {
 *     database: {
 *         name: exampleAwsGlueCatalogDatabase.name,
 *     },
 *     lfTags: [
 *         {
 *             key: "right",
 *             value: "luffield",
 *         },
 *         {
 *             key: "left",
 *             value: "aintree",
 *         },
 *     ],
 * });
 * ```
 */
class ResourceLfTags extends pulumi.CustomResource {
    /**
     * Get an existing ResourceLfTags resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceLfTags(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:lakeformation/resourceLfTags:ResourceLfTags';
    /**
     * Returns true if the given object is an instance of ResourceLfTags.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceLfTags.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["catalogId"] = state?.catalogId;
            resourceInputs["database"] = state?.database;
            resourceInputs["lfTags"] = state?.lfTags;
            resourceInputs["region"] = state?.region;
            resourceInputs["table"] = state?.table;
            resourceInputs["tableWithColumns"] = state?.tableWithColumns;
        }
        else {
            const args = argsOrState;
            if (args?.lfTags === undefined && !opts.urn) {
                throw new Error("Missing required property 'lfTags'");
            }
            resourceInputs["catalogId"] = args?.catalogId;
            resourceInputs["database"] = args?.database;
            resourceInputs["lfTags"] = args?.lfTags;
            resourceInputs["region"] = args?.region;
            resourceInputs["table"] = args?.table;
            resourceInputs["tableWithColumns"] = args?.tableWithColumns;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResourceLfTags.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResourceLfTags = ResourceLfTags;
