// @pulumi/gcp/cloudtasks/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueIamPolicy = exports.QueueIamMember = exports.QueueIamBinding = exports.Queue = exports.getQueueIamPolicyOutput = exports.getQueueIamPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getQueueIamPolicy = null;
exports.getQueueIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getQueueIamPolicy", "getQueueIamPolicyOutput"], () => require("./getQueueIamPolicy"));
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueIamBinding = null;
utilities.lazyLoad(exports, ["QueueIamBinding"], () => require("./queueIamBinding"));
exports.QueueIamMember = null;
utilities.lazyLoad(exports, ["QueueIamMember"], () => require("./queueIamMember"));
exports.QueueIamPolicy = null;
utilities.lazyLoad(exports, ["QueueIamPolicy"], () => require("./queueIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudtasks/queue:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamBinding:QueueIamBinding":
                return new exports.QueueIamBinding(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamMember:QueueIamMember":
                return new exports.QueueIamMember(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamPolicy:QueueIamPolicy":
                return new exports.QueueIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queue", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamPolicy", _module);
