// @pulumi/azure-native/recoveryservices/privateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Private Endpoint Connection Response Properties
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-02-01, 2023-04-01, 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2024-04-30-preview, 2024-07-30-preview, 2024-11-01-preview, 2025-01-01, 2025-02-01, 2025-02-28-preview, 2025-08-01, 2026-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native recoveryservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing PrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateEndpointConnection(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:recoveryservices:PrivateEndpointConnection';
    /**
     * Returns true if the given object is an instance of PrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a PrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.vaultName === undefined && !opts.urn) {
                throw new Error("Missing required property 'vaultName'");
            }
            resourceInputs["eTag"] = args?.eTag;
            resourceInputs["location"] = args?.location;
            resourceInputs["privateEndpointConnectionName"] = args?.privateEndpointConnectionName;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["vaultName"] = args?.vaultName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20200202:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20201001:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20201201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210101:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210201preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210210:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210301:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210401:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210601:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210701:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20210801:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20211001:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20211201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220101:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220301:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220401:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220601preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220901preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20220930preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20221001:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20230101:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20230201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20230401:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20230601:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20230801:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20240101:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20240201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20240401:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20240430preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20240730preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20241001:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20241101preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20250101:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20250201:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20250228preview:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20250801:PrivateEndpointConnection" }, { type: "azure-native:recoveryservices/v20260101preview:PrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateEndpointConnection = PrivateEndpointConnection;
