// @pulumi/azure-native/recoveryservices/replicationProtectedItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationProtectedItem = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Replication protected item.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-02-01, 2023-04-01, 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2025-01-01, 2025-02-01, 2025-08-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native recoveryservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ReplicationProtectedItem extends pulumi.CustomResource {
    /**
     * Get an existing ReplicationProtectedItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationProtectedItem(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:recoveryservices:ReplicationProtectedItem';
    /**
     * Returns true if the given object is an instance of ReplicationProtectedItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationProtectedItem.__pulumiType;
    }
    /**
     * Create a ReplicationProtectedItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.fabricName === undefined && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if (args?.protectionContainerName === undefined && !opts.urn) {
                throw new Error("Missing required property 'protectionContainerName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.resourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args?.fabricName;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["protectionContainerName"] = args?.protectionContainerName;
            resourceInputs["replicatedProtectedItemName"] = args?.replicatedProtectedItemName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceName"] = args?.resourceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20160810:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20180110:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20180710:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210210:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210301:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210401:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210601:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210701:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210801:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211001:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220301:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220401:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220501:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220801:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220910:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20221001:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20230101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20230201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20230401:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20230601:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20230801:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20240101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20240201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20240401:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20241001:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20250101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20250201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20250801:ReplicationProtectedItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationProtectedItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicationProtectedItem = ReplicationProtectedItem;
