// @pulumi/azure-native/providerhub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkusNestedResourceTypeThird = exports.SkusNestedResourceTypeSecond = exports.SkusNestedResourceTypeFirst = exports.Skus = exports.ResourceTypeRegistration = exports.ProviderRegistration = exports.ProviderMonitorSetting = exports.OperationByProviderRegistration = exports.NotificationRegistration = exports.getSkusNestedResourceTypeThirdOutput = exports.getSkusNestedResourceTypeThird = exports.getSkusNestedResourceTypeSecondOutput = exports.getSkusNestedResourceTypeSecond = exports.getSkusNestedResourceTypeFirstOutput = exports.getSkusNestedResourceTypeFirst = exports.getSkusOutput = exports.getSkus = exports.getResourceTypeRegistrationOutput = exports.getResourceTypeRegistration = exports.getProviderRegistrationOutput = exports.getProviderRegistration = exports.getProviderMonitorSettingOutput = exports.getProviderMonitorSetting = exports.getNotificationRegistrationOutput = exports.getNotificationRegistration = exports.getDefaultRolloutOutput = exports.getDefaultRollout = exports.getCustomRolloutOutput = exports.getCustomRollout = exports.getAuthorizedApplicationOutput = exports.getAuthorizedApplication = exports.DefaultRollout = exports.CustomRollout = exports.AuthorizedApplication = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AuthorizedApplication = null;
utilities.lazyLoad(exports, ["AuthorizedApplication"], () => require("./authorizedApplication"));
exports.CustomRollout = null;
utilities.lazyLoad(exports, ["CustomRollout"], () => require("./customRollout"));
exports.DefaultRollout = null;
utilities.lazyLoad(exports, ["DefaultRollout"], () => require("./defaultRollout"));
exports.getAuthorizedApplication = null;
exports.getAuthorizedApplicationOutput = null;
utilities.lazyLoad(exports, ["getAuthorizedApplication", "getAuthorizedApplicationOutput"], () => require("./getAuthorizedApplication"));
exports.getCustomRollout = null;
exports.getCustomRolloutOutput = null;
utilities.lazyLoad(exports, ["getCustomRollout", "getCustomRolloutOutput"], () => require("./getCustomRollout"));
exports.getDefaultRollout = null;
exports.getDefaultRolloutOutput = null;
utilities.lazyLoad(exports, ["getDefaultRollout", "getDefaultRolloutOutput"], () => require("./getDefaultRollout"));
exports.getNotificationRegistration = null;
exports.getNotificationRegistrationOutput = null;
utilities.lazyLoad(exports, ["getNotificationRegistration", "getNotificationRegistrationOutput"], () => require("./getNotificationRegistration"));
exports.getProviderMonitorSetting = null;
exports.getProviderMonitorSettingOutput = null;
utilities.lazyLoad(exports, ["getProviderMonitorSetting", "getProviderMonitorSettingOutput"], () => require("./getProviderMonitorSetting"));
exports.getProviderRegistration = null;
exports.getProviderRegistrationOutput = null;
utilities.lazyLoad(exports, ["getProviderRegistration", "getProviderRegistrationOutput"], () => require("./getProviderRegistration"));
exports.getResourceTypeRegistration = null;
exports.getResourceTypeRegistrationOutput = null;
utilities.lazyLoad(exports, ["getResourceTypeRegistration", "getResourceTypeRegistrationOutput"], () => require("./getResourceTypeRegistration"));
exports.getSkus = null;
exports.getSkusOutput = null;
utilities.lazyLoad(exports, ["getSkus", "getSkusOutput"], () => require("./getSkus"));
exports.getSkusNestedResourceTypeFirst = null;
exports.getSkusNestedResourceTypeFirstOutput = null;
utilities.lazyLoad(exports, ["getSkusNestedResourceTypeFirst", "getSkusNestedResourceTypeFirstOutput"], () => require("./getSkusNestedResourceTypeFirst"));
exports.getSkusNestedResourceTypeSecond = null;
exports.getSkusNestedResourceTypeSecondOutput = null;
utilities.lazyLoad(exports, ["getSkusNestedResourceTypeSecond", "getSkusNestedResourceTypeSecondOutput"], () => require("./getSkusNestedResourceTypeSecond"));
exports.getSkusNestedResourceTypeThird = null;
exports.getSkusNestedResourceTypeThirdOutput = null;
utilities.lazyLoad(exports, ["getSkusNestedResourceTypeThird", "getSkusNestedResourceTypeThirdOutput"], () => require("./getSkusNestedResourceTypeThird"));
exports.NotificationRegistration = null;
utilities.lazyLoad(exports, ["NotificationRegistration"], () => require("./notificationRegistration"));
exports.OperationByProviderRegistration = null;
utilities.lazyLoad(exports, ["OperationByProviderRegistration"], () => require("./operationByProviderRegistration"));
exports.ProviderMonitorSetting = null;
utilities.lazyLoad(exports, ["ProviderMonitorSetting"], () => require("./providerMonitorSetting"));
exports.ProviderRegistration = null;
utilities.lazyLoad(exports, ["ProviderRegistration"], () => require("./providerRegistration"));
exports.ResourceTypeRegistration = null;
utilities.lazyLoad(exports, ["ResourceTypeRegistration"], () => require("./resourceTypeRegistration"));
exports.Skus = null;
utilities.lazyLoad(exports, ["Skus"], () => require("./skus"));
exports.SkusNestedResourceTypeFirst = null;
utilities.lazyLoad(exports, ["SkusNestedResourceTypeFirst"], () => require("./skusNestedResourceTypeFirst"));
exports.SkusNestedResourceTypeSecond = null;
utilities.lazyLoad(exports, ["SkusNestedResourceTypeSecond"], () => require("./skusNestedResourceTypeSecond"));
exports.SkusNestedResourceTypeThird = null;
utilities.lazyLoad(exports, ["SkusNestedResourceTypeThird"], () => require("./skusNestedResourceTypeThird"));
// Export enums:
__exportStar(require("../types/enums/providerhub"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:providerhub:AuthorizedApplication":
                return new exports.AuthorizedApplication(name, undefined, { urn });
            case "azure-native:providerhub:CustomRollout":
                return new exports.CustomRollout(name, undefined, { urn });
            case "azure-native:providerhub:DefaultRollout":
                return new exports.DefaultRollout(name, undefined, { urn });
            case "azure-native:providerhub:NotificationRegistration":
                return new exports.NotificationRegistration(name, undefined, { urn });
            case "azure-native:providerhub:OperationByProviderRegistration":
                return new exports.OperationByProviderRegistration(name, undefined, { urn });
            case "azure-native:providerhub:ProviderMonitorSetting":
                return new exports.ProviderMonitorSetting(name, undefined, { urn });
            case "azure-native:providerhub:ProviderRegistration":
                return new exports.ProviderRegistration(name, undefined, { urn });
            case "azure-native:providerhub:ResourceTypeRegistration":
                return new exports.ResourceTypeRegistration(name, undefined, { urn });
            case "azure-native:providerhub:Skus":
                return new exports.Skus(name, undefined, { urn });
            case "azure-native:providerhub:SkusNestedResourceTypeFirst":
                return new exports.SkusNestedResourceTypeFirst(name, undefined, { urn });
            case "azure-native:providerhub:SkusNestedResourceTypeSecond":
                return new exports.SkusNestedResourceTypeSecond(name, undefined, { urn });
            case "azure-native:providerhub:SkusNestedResourceTypeThird":
                return new exports.SkusNestedResourceTypeThird(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "providerhub", _module);
