// @pulumi/azure-native/testbase/draftPackage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DraftPackage = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The Test Base Draft Package resource.
 *
 * Uses Azure REST API version 2023-11-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-11-01-preview.
 */
class DraftPackage extends pulumi.CustomResource {
    /**
     * Get an existing DraftPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DraftPackage(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:testbase:DraftPackage';
    /**
     * Returns true if the given object is an instance of DraftPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DraftPackage.__pulumiType;
    }
    /**
     * Create a DraftPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.testBaseAccountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'testBaseAccountName'");
            }
            resourceInputs["appFileName"] = args?.appFileName;
            resourceInputs["applicationName"] = args?.applicationName;
            resourceInputs["comments"] = args?.comments;
            resourceInputs["draftPackageName"] = args?.draftPackageName;
            resourceInputs["editPackage"] = (args?.editPackage) ?? false;
            resourceInputs["executableLaunchCommand"] = args?.executableLaunchCommand;
            resourceInputs["firstPartyApps"] = args?.firstPartyApps;
            resourceInputs["flightingRing"] = args?.flightingRing;
            resourceInputs["galleryApps"] = args?.galleryApps;
            resourceInputs["highlightedFiles"] = args?.highlightedFiles;
            resourceInputs["inplaceUpgradeOSPair"] = args?.inplaceUpgradeOSPair;
            resourceInputs["intuneEnrollmentMetadata"] = args?.intuneEnrollmentMetadata;
            resourceInputs["intuneMetadata"] = args?.intuneMetadata;
            resourceInputs["packageId"] = args?.packageId;
            resourceInputs["packageTags"] = args?.packageTags;
            resourceInputs["processName"] = args?.processName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sourceType"] = (args?.sourceType) ?? "Native";
            resourceInputs["tabState"] = args?.tabState;
            resourceInputs["targetOSList"] = args?.targetOSList;
            resourceInputs["testBaseAccountName"] = args?.testBaseAccountName;
            resourceInputs["testTypes"] = args?.testTypes;
            resourceInputs["tests"] = args?.tests;
            resourceInputs["useAutofill"] = args?.useAutofill;
            resourceInputs["useSample"] = args?.useSample;
            resourceInputs["version"] = args?.version;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["draftPackagePath"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workingPath"] = undefined /*out*/;
        }
        else {
            resourceInputs["appFileName"] = undefined /*out*/;
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["comments"] = undefined /*out*/;
            resourceInputs["draftPackagePath"] = undefined /*out*/;
            resourceInputs["editPackage"] = undefined /*out*/;
            resourceInputs["executableLaunchCommand"] = undefined /*out*/;
            resourceInputs["firstPartyApps"] = undefined /*out*/;
            resourceInputs["flightingRing"] = undefined /*out*/;
            resourceInputs["galleryApps"] = undefined /*out*/;
            resourceInputs["highlightedFiles"] = undefined /*out*/;
            resourceInputs["inplaceUpgradeOSPair"] = undefined /*out*/;
            resourceInputs["intuneEnrollmentMetadata"] = undefined /*out*/;
            resourceInputs["intuneMetadata"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["packageId"] = undefined /*out*/;
            resourceInputs["packageTags"] = undefined /*out*/;
            resourceInputs["processName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sourceType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tabState"] = undefined /*out*/;
            resourceInputs["targetOSList"] = undefined /*out*/;
            resourceInputs["testTypes"] = undefined /*out*/;
            resourceInputs["tests"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useAutofill"] = undefined /*out*/;
            resourceInputs["useSample"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
            resourceInputs["workingPath"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:testbase/v20231101preview:DraftPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DraftPackage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DraftPackage = DraftPackage;
