// @pulumi/azure-native/hdinsight/extensionAzureMonitorStatus.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionAzureMonitorStatus = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The azure monitor status response.
 *
 * Uses Azure REST API version 2024-08-01-preview. In version 2.x of the Azure Native provider, it used API version 2021-06-01.
 *
 * Other available API versions: 2021-06-01, 2023-04-15-preview, 2023-08-15-preview, 2025-01-15-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native hdinsight [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ExtensionAzureMonitorStatus extends pulumi.CustomResource {
    /**
     * Get an existing ExtensionAzureMonitorStatus resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExtensionAzureMonitorStatus(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:hdinsight:ExtensionAzureMonitorStatus';
    /**
     * Returns true if the given object is an instance of ExtensionAzureMonitorStatus.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExtensionAzureMonitorStatus.__pulumiType;
    }
    /**
     * Create a ExtensionAzureMonitorStatus resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["primaryKey"] = args?.primaryKey;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["selectedConfigurations"] = args?.selectedConfigurations;
            resourceInputs["workspaceId"] = args?.workspaceId;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clusterMonitoringEnabled"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clusterMonitoringEnabled"] = undefined /*out*/;
            resourceInputs["selectedConfigurations"] = undefined /*out*/;
            resourceInputs["workspaceId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hdinsight/v20150301preview:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20180601preview:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20210601:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20230415preview:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20230815preview:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20240801preview:ExtensionAzureMonitorStatus" }, { type: "azure-native:hdinsight/v20250115preview:ExtensionAzureMonitorStatus" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExtensionAzureMonitorStatus.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExtensionAzureMonitorStatus = ExtensionAzureMonitorStatus;
