// @pulumi/azure-native/hdinsight/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getExtensionMonitoringStatusOutput = exports.getExtensionMonitoringStatus = exports.getExtensionAzureMonitorStatusOutput = exports.getExtensionAzureMonitorStatus = exports.getExtensionAzureMonitorAgentStatusOutput = exports.getExtensionAzureMonitorAgentStatus = exports.getExtensionOutput = exports.getExtension = exports.getClusterGatewaySettingsOutput = exports.getClusterGatewaySettings = exports.getClusterOutput = exports.getCluster = exports.getApplicationOutput = exports.getApplication = exports.ExtensionMonitoringStatus = exports.ExtensionAzureMonitorStatus = exports.ExtensionAzureMonitorAgentStatus = exports.Extension = exports.Cluster = exports.Application = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.ExtensionAzureMonitorAgentStatus = null;
utilities.lazyLoad(exports, ["ExtensionAzureMonitorAgentStatus"], () => require("./extensionAzureMonitorAgentStatus"));
exports.ExtensionAzureMonitorStatus = null;
utilities.lazyLoad(exports, ["ExtensionAzureMonitorStatus"], () => require("./extensionAzureMonitorStatus"));
exports.ExtensionMonitoringStatus = null;
utilities.lazyLoad(exports, ["ExtensionMonitoringStatus"], () => require("./extensionMonitoringStatus"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterGatewaySettings = null;
exports.getClusterGatewaySettingsOutput = null;
utilities.lazyLoad(exports, ["getClusterGatewaySettings", "getClusterGatewaySettingsOutput"], () => require("./getClusterGatewaySettings"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getExtensionAzureMonitorAgentStatus = null;
exports.getExtensionAzureMonitorAgentStatusOutput = null;
utilities.lazyLoad(exports, ["getExtensionAzureMonitorAgentStatus", "getExtensionAzureMonitorAgentStatusOutput"], () => require("./getExtensionAzureMonitorAgentStatus"));
exports.getExtensionAzureMonitorStatus = null;
exports.getExtensionAzureMonitorStatusOutput = null;
utilities.lazyLoad(exports, ["getExtensionAzureMonitorStatus", "getExtensionAzureMonitorStatusOutput"], () => require("./getExtensionAzureMonitorStatus"));
exports.getExtensionMonitoringStatus = null;
exports.getExtensionMonitoringStatusOutput = null;
utilities.lazyLoad(exports, ["getExtensionMonitoringStatus", "getExtensionMonitoringStatusOutput"], () => require("./getExtensionMonitoringStatus"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/hdinsight"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hdinsight:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:hdinsight:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:hdinsight:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:hdinsight:ExtensionAzureMonitorAgentStatus":
                return new exports.ExtensionAzureMonitorAgentStatus(name, undefined, { urn });
            case "azure-native:hdinsight:ExtensionAzureMonitorStatus":
                return new exports.ExtensionAzureMonitorStatus(name, undefined, { urn });
            case "azure-native:hdinsight:ExtensionMonitoringStatus":
                return new exports.ExtensionMonitoringStatus(name, undefined, { urn });
            case "azure-native:hdinsight:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hdinsight", _module);
