// @pulumi/azure-native/networkcloud/cluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * Uses Azure REST API version 2025-02-01. In version 2.x of the Azure Native provider, it used API version 2023-10-01-preview.
 *
 * Other available API versions: 2024-07-01, 2024-10-01-preview, 2025-07-01-preview, 2025-09-01, 2026-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native networkcloud [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Cluster extends pulumi.CustomResource {
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Cluster(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:networkcloud:Cluster';
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
    /**
     * Create a Cluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.aggregatorOrSingleRackDefinition === undefined && !opts.urn) {
                throw new Error("Missing required property 'aggregatorOrSingleRackDefinition'");
            }
            if (args?.clusterType === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterType'");
            }
            if (args?.clusterVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterVersion'");
            }
            if (args?.extendedLocation === undefined && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if (args?.networkFabricId === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkFabricId'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["aggregatorOrSingleRackDefinition"] = args?.aggregatorOrSingleRackDefinition;
            resourceInputs["analyticsOutputSettings"] = args?.analyticsOutputSettings;
            resourceInputs["analyticsWorkspaceId"] = args?.analyticsWorkspaceId;
            resourceInputs["clusterLocation"] = args?.clusterLocation;
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["clusterServicePrincipal"] = args?.clusterServicePrincipal;
            resourceInputs["clusterType"] = args?.clusterType;
            resourceInputs["clusterVersion"] = args?.clusterVersion;
            resourceInputs["commandOutputSettings"] = args?.commandOutputSettings;
            resourceInputs["computeDeploymentThreshold"] = args?.computeDeploymentThreshold;
            resourceInputs["computeRackDefinitions"] = args?.computeRackDefinitions;
            resourceInputs["extendedLocation"] = args?.extendedLocation;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["location"] = args?.location;
            resourceInputs["managedResourceGroupConfiguration"] = args?.managedResourceGroupConfiguration;
            resourceInputs["networkFabricId"] = args?.networkFabricId;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["runtimeProtectionConfiguration"] = args ? (args.runtimeProtectionConfiguration ? pulumi.output(args.runtimeProtectionConfiguration).apply(inputs.networkcloud.runtimeProtectionConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["secretArchive"] = args ? (args.secretArchive ? pulumi.output(args.secretArchive).apply(inputs.networkcloud.clusterSecretArchiveArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["secretArchiveSettings"] = args?.secretArchiveSettings;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["updateStrategy"] = args ? (args.updateStrategy ? pulumi.output(args.updateStrategy).apply(inputs.networkcloud.clusterUpdateStrategyArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["vulnerabilityScanningSettings"] = args ? (args.vulnerabilityScanningSettings ? pulumi.output(args.vulnerabilityScanningSettings).apply(inputs.networkcloud.vulnerabilityScanningSettingsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["availableUpgradeVersions"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clusterCapacity"] = undefined /*out*/;
            resourceInputs["clusterConnectionStatus"] = undefined /*out*/;
            resourceInputs["clusterExtendedLocation"] = undefined /*out*/;
            resourceInputs["clusterManagerConnectionStatus"] = undefined /*out*/;
            resourceInputs["clusterManagerId"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["hybridAksExtendedLocation"] = undefined /*out*/;
            resourceInputs["manualActionCount"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["supportExpiryDate"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workloadResourceIds"] = undefined /*out*/;
        }
        else {
            resourceInputs["aggregatorOrSingleRackDefinition"] = undefined /*out*/;
            resourceInputs["analyticsOutputSettings"] = undefined /*out*/;
            resourceInputs["analyticsWorkspaceId"] = undefined /*out*/;
            resourceInputs["availableUpgradeVersions"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clusterCapacity"] = undefined /*out*/;
            resourceInputs["clusterConnectionStatus"] = undefined /*out*/;
            resourceInputs["clusterExtendedLocation"] = undefined /*out*/;
            resourceInputs["clusterLocation"] = undefined /*out*/;
            resourceInputs["clusterManagerConnectionStatus"] = undefined /*out*/;
            resourceInputs["clusterManagerId"] = undefined /*out*/;
            resourceInputs["clusterServicePrincipal"] = undefined /*out*/;
            resourceInputs["clusterType"] = undefined /*out*/;
            resourceInputs["clusterVersion"] = undefined /*out*/;
            resourceInputs["commandOutputSettings"] = undefined /*out*/;
            resourceInputs["computeDeploymentThreshold"] = undefined /*out*/;
            resourceInputs["computeRackDefinitions"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["hybridAksExtendedLocation"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedResourceGroupConfiguration"] = undefined /*out*/;
            resourceInputs["manualActionCount"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkFabricId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["runtimeProtectionConfiguration"] = undefined /*out*/;
            resourceInputs["secretArchive"] = undefined /*out*/;
            resourceInputs["secretArchiveSettings"] = undefined /*out*/;
            resourceInputs["supportExpiryDate"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updateStrategy"] = undefined /*out*/;
            resourceInputs["vulnerabilityScanningSettings"] = undefined /*out*/;
            resourceInputs["workloadResourceIds"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:networkcloud/v20230701:Cluster" }, { type: "azure-native:networkcloud/v20231001preview:Cluster" }, { type: "azure-native:networkcloud/v20240601preview:Cluster" }, { type: "azure-native:networkcloud/v20240701:Cluster" }, { type: "azure-native:networkcloud/v20241001preview:Cluster" }, { type: "azure-native:networkcloud/v20250201:Cluster" }, { type: "azure-native:networkcloud/v20250701preview:Cluster" }, { type: "azure-native:networkcloud/v20250901:Cluster" }, { type: "azure-native:networkcloud/v20260101preview:Cluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Cluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Cluster = Cluster;
