// @pulumi/azure-native/networkcloud/virtualMachine.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachine = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * Uses Azure REST API version 2025-02-01. In version 2.x of the Azure Native provider, it used API version 2023-10-01-preview.
 *
 * Other available API versions: 2024-07-01, 2024-10-01-preview, 2025-07-01-preview, 2025-09-01, 2026-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native networkcloud [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class VirtualMachine extends pulumi.CustomResource {
    /**
     * Get an existing VirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualMachine(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:networkcloud:VirtualMachine';
    /**
     * Returns true if the given object is an instance of VirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMachine.__pulumiType;
    }
    /**
     * Create a VirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.adminUsername === undefined && !opts.urn) {
                throw new Error("Missing required property 'adminUsername'");
            }
            if (args?.cloudServicesNetworkAttachment === undefined && !opts.urn) {
                throw new Error("Missing required property 'cloudServicesNetworkAttachment'");
            }
            if (args?.cpuCores === undefined && !opts.urn) {
                throw new Error("Missing required property 'cpuCores'");
            }
            if (args?.extendedLocation === undefined && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if (args?.memorySizeGB === undefined && !opts.urn) {
                throw new Error("Missing required property 'memorySizeGB'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.storageProfile === undefined && !opts.urn) {
                throw new Error("Missing required property 'storageProfile'");
            }
            if (args?.vmImage === undefined && !opts.urn) {
                throw new Error("Missing required property 'vmImage'");
            }
            resourceInputs["adminUsername"] = args?.adminUsername;
            resourceInputs["bootMethod"] = (args?.bootMethod) ?? "UEFI";
            resourceInputs["cloudServicesNetworkAttachment"] = args?.cloudServicesNetworkAttachment;
            resourceInputs["consoleExtendedLocation"] = args?.consoleExtendedLocation;
            resourceInputs["cpuCores"] = args?.cpuCores;
            resourceInputs["extendedLocation"] = args?.extendedLocation;
            resourceInputs["isolateEmulatorThread"] = (args?.isolateEmulatorThread) ?? "True";
            resourceInputs["location"] = args?.location;
            resourceInputs["memorySizeGB"] = args?.memorySizeGB;
            resourceInputs["networkAttachments"] = args?.networkAttachments;
            resourceInputs["networkData"] = args?.networkData;
            resourceInputs["placementHints"] = args?.placementHints;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sshPublicKeys"] = args?.sshPublicKeys;
            resourceInputs["storageProfile"] = args ? (args.storageProfile ? pulumi.output(args.storageProfile).apply(inputs.networkcloud.storageProfileArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["userData"] = args?.userData;
            resourceInputs["virtioInterface"] = (args?.virtioInterface) ?? "Modern";
            resourceInputs["virtualMachineName"] = args?.virtualMachineName;
            resourceInputs["vmDeviceModel"] = (args?.vmDeviceModel) ?? "T2";
            resourceInputs["vmImage"] = args?.vmImage;
            resourceInputs["vmImageRepositoryCredentials"] = args?.vmImageRepositoryCredentials;
            resourceInputs["availabilityZone"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bareMetalMachineId"] = undefined /*out*/;
            resourceInputs["clusterId"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["powerState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["volumes"] = undefined /*out*/;
        }
        else {
            resourceInputs["adminUsername"] = undefined /*out*/;
            resourceInputs["availabilityZone"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bareMetalMachineId"] = undefined /*out*/;
            resourceInputs["bootMethod"] = undefined /*out*/;
            resourceInputs["cloudServicesNetworkAttachment"] = undefined /*out*/;
            resourceInputs["clusterId"] = undefined /*out*/;
            resourceInputs["consoleExtendedLocation"] = undefined /*out*/;
            resourceInputs["cpuCores"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["isolateEmulatorThread"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["memorySizeGB"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkAttachments"] = undefined /*out*/;
            resourceInputs["networkData"] = undefined /*out*/;
            resourceInputs["placementHints"] = undefined /*out*/;
            resourceInputs["powerState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sshPublicKeys"] = undefined /*out*/;
            resourceInputs["storageProfile"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userData"] = undefined /*out*/;
            resourceInputs["virtioInterface"] = undefined /*out*/;
            resourceInputs["vmDeviceModel"] = undefined /*out*/;
            resourceInputs["vmImage"] = undefined /*out*/;
            resourceInputs["vmImageRepositoryCredentials"] = undefined /*out*/;
            resourceInputs["volumes"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:networkcloud/v20230701:VirtualMachine" }, { type: "azure-native:networkcloud/v20231001preview:VirtualMachine" }, { type: "azure-native:networkcloud/v20240601preview:VirtualMachine" }, { type: "azure-native:networkcloud/v20240701:VirtualMachine" }, { type: "azure-native:networkcloud/v20241001preview:VirtualMachine" }, { type: "azure-native:networkcloud/v20250201:VirtualMachine" }, { type: "azure-native:networkcloud/v20250701preview:VirtualMachine" }, { type: "azure-native:networkcloud/v20250901:VirtualMachine" }, { type: "azure-native:networkcloud/v20260101preview:VirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualMachine.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualMachine = VirtualMachine;
