// @pulumi/azure-native/fileshares/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getFileShareUsageDataOutput = exports.getFileShareUsageData = exports.getFileShareSnapshotOutput = exports.getFileShareSnapshot = exports.getFileShareProvisioningRecommendationOutput = exports.getFileShareProvisioningRecommendation = exports.getFileShareLimitsOutput = exports.getFileShareLimits = exports.getFileShareOutput = exports.getFileShare = exports.FileShareSnapshot = exports.FileShare = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.FileShare = null;
utilities.lazyLoad(exports, ["FileShare"], () => require("./fileShare"));
exports.FileShareSnapshot = null;
utilities.lazyLoad(exports, ["FileShareSnapshot"], () => require("./fileShareSnapshot"));
exports.getFileShare = null;
exports.getFileShareOutput = null;
utilities.lazyLoad(exports, ["getFileShare", "getFileShareOutput"], () => require("./getFileShare"));
exports.getFileShareLimits = null;
exports.getFileShareLimitsOutput = null;
utilities.lazyLoad(exports, ["getFileShareLimits", "getFileShareLimitsOutput"], () => require("./getFileShareLimits"));
exports.getFileShareProvisioningRecommendation = null;
exports.getFileShareProvisioningRecommendationOutput = null;
utilities.lazyLoad(exports, ["getFileShareProvisioningRecommendation", "getFileShareProvisioningRecommendationOutput"], () => require("./getFileShareProvisioningRecommendation"));
exports.getFileShareSnapshot = null;
exports.getFileShareSnapshotOutput = null;
utilities.lazyLoad(exports, ["getFileShareSnapshot", "getFileShareSnapshotOutput"], () => require("./getFileShareSnapshot"));
exports.getFileShareUsageData = null;
exports.getFileShareUsageDataOutput = null;
utilities.lazyLoad(exports, ["getFileShareUsageData", "getFileShareUsageDataOutput"], () => require("./getFileShareUsageData"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/fileshares"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:fileshares:FileShare":
                return new exports.FileShare(name, undefined, { urn });
            case "azure-native:fileshares:FileShareSnapshot":
                return new exports.FileShareSnapshot(name, undefined, { urn });
            case "azure-native:fileshares:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "fileshares", _module);
