// @pulumi/aws/redshift/namespaceRegistration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceRegistration = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an Amazon Redshift namespace registration to the AWS Glue Data Catalog. Use this resource to enable access to a Redshift data warehouse using the Apache Iceberg REST API.
 *
 * > **NOTE:** This resource has limited drift detection capabilities. AWS does not provide a reliable API to verify registration status after creation. The resource verifies that the underlying cluster or namespace exists and that an internal data share was created, but cannot detect if the registration was removed outside of Terraform.
 *
 * ## Example Usage
 *
 * ### Serverless Namespace
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as std from "@pulumi/std";
 *
 * const current = aws.getCallerIdentity({});
 * const example = new aws.redshiftserverless.Namespace("example", {
 *     namespaceName: "example",
 *     dbName: "example",
 * });
 * const exampleWorkgroup = new aws.redshiftserverless.Workgroup("example", {
 *     namespaceName: example.namespaceName,
 *     workgroupName: "example",
 * });
 * const exampleNamespaceRegistration = new aws.redshift.NamespaceRegistration("example", {
 *     consumerIdentifier: std.format({
 *         input: "DataCatalog/%s",
 *         args: [current.then(current => current.accountId)],
 *     }).then(invoke => invoke.result),
 *     namespaceType: "serverless",
 *     serverlessNamespaceIdentifier: example.namespaceName,
 *     serverlessWorkgroupIdentifier: exampleWorkgroup.workgroupName,
 * });
 * ```
 *
 * ### Provisioned Cluster
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as std from "@pulumi/std";
 *
 * const current = aws.getCallerIdentity({});
 * const example = new aws.redshift.Cluster("example", {
 *     clusterIdentifier: "example",
 *     databaseName: "example",
 *     masterUsername: "exampleuser",
 *     masterPassword: "ExamplePassword123!",
 *     nodeType: "dc2.large",
 *     clusterType: "single-node",
 * });
 * const exampleNamespaceRegistration = new aws.redshift.NamespaceRegistration("example", {
 *     consumerIdentifier: std.format({
 *         input: "DataCatalog/%s",
 *         args: [current.then(current => current.accountId)],
 *     }).then(invoke => invoke.result),
 *     namespaceType: "provisioned",
 *     provisionedClusterIdentifier: example.clusterIdentifier,
 * });
 * ```
 *
 * ## Import
 *
 * For provisioned clusters:
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `consumerIdentifier` (String) Consumer identifier for the registration.
 * - `namespaceType` (String) Type of namespace being registered. Valid values: `serverless`, `provisioned`.
 *
 * #### Optional
 *
 * - `provisionedClusterIdentifier` (String) Identifier of the provisioned cluster. Required when `namespaceType` is `provisioned`.
 * - `serverlessNamespaceIdentifier` (String) Identifier of the serverless namespace. Required when `namespaceType` is `serverless`.
 * - `serverlessWorkgroupIdentifier` (String) Identifier of the serverless workgroup. Required when `namespaceType` is `serverless`.
 *
 * For provisioned clusters:
 *
 * Using `pulumi import`, import Redshift Namespace Registration using the composite ID. For example:
 *
 * For serverless namespaces:
 *
 * ```sh
 * $ pulumi import aws:redshift/namespaceRegistration:NamespaceRegistration example DataCatalog/123456789012,example-namespace,example-workgroup
 * ```
 *
 * For provisioned clusters:
 *
 * ```sh
 * $ pulumi import aws:redshift/namespaceRegistration:NamespaceRegistration example DataCatalog/123456789012,example-cluster
 * ```
 */
class NamespaceRegistration extends pulumi.CustomResource {
    /**
     * Get an existing NamespaceRegistration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NamespaceRegistration(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:redshift/namespaceRegistration:NamespaceRegistration';
    /**
     * Returns true if the given object is an instance of NamespaceRegistration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NamespaceRegistration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["consumerIdentifier"] = state?.consumerIdentifier;
            resourceInputs["namespaceType"] = state?.namespaceType;
            resourceInputs["provisionedClusterIdentifier"] = state?.provisionedClusterIdentifier;
            resourceInputs["region"] = state?.region;
            resourceInputs["serverlessNamespaceIdentifier"] = state?.serverlessNamespaceIdentifier;
            resourceInputs["serverlessWorkgroupIdentifier"] = state?.serverlessWorkgroupIdentifier;
        }
        else {
            const args = argsOrState;
            if (args?.consumerIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'consumerIdentifier'");
            }
            if (args?.namespaceType === undefined && !opts.urn) {
                throw new Error("Missing required property 'namespaceType'");
            }
            resourceInputs["consumerIdentifier"] = args?.consumerIdentifier;
            resourceInputs["namespaceType"] = args?.namespaceType;
            resourceInputs["provisionedClusterIdentifier"] = args?.provisionedClusterIdentifier;
            resourceInputs["region"] = args?.region;
            resourceInputs["serverlessNamespaceIdentifier"] = args?.serverlessNamespaceIdentifier;
            resourceInputs["serverlessWorkgroupIdentifier"] = args?.serverlessWorkgroupIdentifier;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NamespaceRegistration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NamespaceRegistration = NamespaceRegistration;
