// @pulumi/gcp/gkebackup/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestorePlanIamPolicy = exports.RestorePlanIamMember = exports.RestorePlanIamBinding = exports.RestorePlan = exports.RestoreChannel = exports.getRestorePlanIamPolicyOutput = exports.getRestorePlanIamPolicy = exports.getBackupPlanIamPolicyOutput = exports.getBackupPlanIamPolicy = exports.BackupPlanIamPolicy = exports.BackupPlanIamMember = exports.BackupPlanIamBinding = exports.BackupPlan = exports.BackupChannel = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.BackupChannel = null;
utilities.lazyLoad(exports, ["BackupChannel"], () => require("./backupChannel"));
exports.BackupPlan = null;
utilities.lazyLoad(exports, ["BackupPlan"], () => require("./backupPlan"));
exports.BackupPlanIamBinding = null;
utilities.lazyLoad(exports, ["BackupPlanIamBinding"], () => require("./backupPlanIamBinding"));
exports.BackupPlanIamMember = null;
utilities.lazyLoad(exports, ["BackupPlanIamMember"], () => require("./backupPlanIamMember"));
exports.BackupPlanIamPolicy = null;
utilities.lazyLoad(exports, ["BackupPlanIamPolicy"], () => require("./backupPlanIamPolicy"));
exports.getBackupPlanIamPolicy = null;
exports.getBackupPlanIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackupPlanIamPolicy", "getBackupPlanIamPolicyOutput"], () => require("./getBackupPlanIamPolicy"));
exports.getRestorePlanIamPolicy = null;
exports.getRestorePlanIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRestorePlanIamPolicy", "getRestorePlanIamPolicyOutput"], () => require("./getRestorePlanIamPolicy"));
exports.RestoreChannel = null;
utilities.lazyLoad(exports, ["RestoreChannel"], () => require("./restoreChannel"));
exports.RestorePlan = null;
utilities.lazyLoad(exports, ["RestorePlan"], () => require("./restorePlan"));
exports.RestorePlanIamBinding = null;
utilities.lazyLoad(exports, ["RestorePlanIamBinding"], () => require("./restorePlanIamBinding"));
exports.RestorePlanIamMember = null;
utilities.lazyLoad(exports, ["RestorePlanIamMember"], () => require("./restorePlanIamMember"));
exports.RestorePlanIamPolicy = null;
utilities.lazyLoad(exports, ["RestorePlanIamPolicy"], () => require("./restorePlanIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:gkebackup/backupChannel:BackupChannel":
                return new exports.BackupChannel(name, undefined, { urn });
            case "gcp:gkebackup/backupPlan:BackupPlan":
                return new exports.BackupPlan(name, undefined, { urn });
            case "gcp:gkebackup/backupPlanIamBinding:BackupPlanIamBinding":
                return new exports.BackupPlanIamBinding(name, undefined, { urn });
            case "gcp:gkebackup/backupPlanIamMember:BackupPlanIamMember":
                return new exports.BackupPlanIamMember(name, undefined, { urn });
            case "gcp:gkebackup/backupPlanIamPolicy:BackupPlanIamPolicy":
                return new exports.BackupPlanIamPolicy(name, undefined, { urn });
            case "gcp:gkebackup/restoreChannel:RestoreChannel":
                return new exports.RestoreChannel(name, undefined, { urn });
            case "gcp:gkebackup/restorePlan:RestorePlan":
                return new exports.RestorePlan(name, undefined, { urn });
            case "gcp:gkebackup/restorePlanIamBinding:RestorePlanIamBinding":
                return new exports.RestorePlanIamBinding(name, undefined, { urn });
            case "gcp:gkebackup/restorePlanIamMember:RestorePlanIamMember":
                return new exports.RestorePlanIamMember(name, undefined, { urn });
            case "gcp:gkebackup/restorePlanIamPolicy:RestorePlanIamPolicy":
                return new exports.RestorePlanIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "gkebackup/backupChannel", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/backupPlan", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/backupPlanIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/backupPlanIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/backupPlanIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/restoreChannel", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/restorePlan", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/restorePlanIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/restorePlanIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "gkebackup/restorePlanIamPolicy", _module);
