// @pulumi/aws/mq/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInstanceTypeOfferingsOutput = exports.getInstanceTypeOfferings = exports.getBrokerEngineTypesOutput = exports.getBrokerEngineTypes = exports.getBrokerOutput = exports.getBroker = exports.Configuration = exports.Broker = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Broker = null;
utilities.lazyLoad(exports, ["Broker"], () => require("./broker"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.getBroker = null;
exports.getBrokerOutput = null;
utilities.lazyLoad(exports, ["getBroker", "getBrokerOutput"], () => require("./getBroker"));
exports.getBrokerEngineTypes = null;
exports.getBrokerEngineTypesOutput = null;
utilities.lazyLoad(exports, ["getBrokerEngineTypes", "getBrokerEngineTypesOutput"], () => require("./getBrokerEngineTypes"));
exports.getInstanceTypeOfferings = null;
exports.getInstanceTypeOfferingsOutput = null;
utilities.lazyLoad(exports, ["getInstanceTypeOfferings", "getInstanceTypeOfferingsOutput"], () => require("./getInstanceTypeOfferings"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:mq/broker:Broker":
                return new exports.Broker(name, undefined, { urn });
            case "aws:mq/configuration:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "mq/broker", _module);
pulumi.runtime.registerResourceModule("aws", "mq/configuration", _module);
