// @pulumi/azure-native/types/enums/fileshares/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShareRootSquash = exports.Redundancy = exports.PublicNetworkAccess = exports.Protocol = exports.PrivateEndpointServiceConnectionStatus = exports.MediaTier = void 0;
exports.MediaTier = {
    /**
     * SSD media tier.
     */
    SSD: "SSD",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.Protocol = {
    /**
     * NFS protocol.
     */
    NFS: "NFS",
};
exports.PublicNetworkAccess = {
    /**
     * The public network access is enabled
     */
    Enabled: "Enabled",
    /**
     * The public network access is disabled
     */
    Disabled: "Disabled",
};
exports.Redundancy = {
    /**
     * Local redundancy.
     */
    Local: "Local",
    /**
     * Zone redundancy.
     */
    Zone: "Zone",
};
exports.ShareRootSquash = {
    /**
     * No root squash.
     */
    NoRootSquash: "NoRootSquash",
    /**
     * Root squash.
     */
    RootSquash: "RootSquash",
    /**
     * All squash.
     */
    AllSquash: "AllSquash",
};
