// @pulumi/aws/controltower/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LandingZone = exports.getControlsOutput = exports.getControls = exports.ControlTowerControl = exports.Baseline = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Baseline = null;
utilities.lazyLoad(exports, ["Baseline"], () => require("./baseline"));
exports.ControlTowerControl = null;
utilities.lazyLoad(exports, ["ControlTowerControl"], () => require("./controlTowerControl"));
exports.getControls = null;
exports.getControlsOutput = null;
utilities.lazyLoad(exports, ["getControls", "getControlsOutput"], () => require("./getControls"));
exports.LandingZone = null;
utilities.lazyLoad(exports, ["LandingZone"], () => require("./landingZone"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:controltower/baseline:Baseline":
                return new exports.Baseline(name, undefined, { urn });
            case "aws:controltower/controlTowerControl:ControlTowerControl":
                return new exports.ControlTowerControl(name, undefined, { urn });
            case "aws:controltower/landingZone:LandingZone":
                return new exports.LandingZone(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "controltower/baseline", _module);
pulumi.runtime.registerResourceModule("aws", "controltower/controlTowerControl", _module);
pulumi.runtime.registerResourceModule("aws", "controltower/landingZone", _module);
