// @pulumi/azure-native/securityinsights/anomalySecurityMLAnalyticsSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnomalySecurityMLAnalyticsSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents Anomaly Security ML Analytics Settings
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 */
class AnomalySecurityMLAnalyticsSettings extends pulumi.CustomResource {
    /**
     * Get an existing AnomalySecurityMLAnalyticsSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AnomalySecurityMLAnalyticsSettings(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:AnomalySecurityMLAnalyticsSettings';
    /**
     * Returns true if the given object is an instance of AnomalySecurityMLAnalyticsSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnomalySecurityMLAnalyticsSettings.__pulumiType;
    }
    /**
     * Create a AnomalySecurityMLAnalyticsSettings resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.anomalyVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'anomalyVersion'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.frequency === undefined && !opts.urn) {
                throw new Error("Missing required property 'frequency'");
            }
            if (args?.isDefaultSettings === undefined && !opts.urn) {
                throw new Error("Missing required property 'isDefaultSettings'");
            }
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.settingsStatus === undefined && !opts.urn) {
                throw new Error("Missing required property 'settingsStatus'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["anomalySettingsVersion"] = args?.anomalySettingsVersion;
            resourceInputs["anomalyVersion"] = args?.anomalyVersion;
            resourceInputs["customizableObservations"] = args?.customizableObservations;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["frequency"] = args?.frequency;
            resourceInputs["isDefaultSettings"] = args?.isDefaultSettings;
            resourceInputs["kind"] = "Anomaly";
            resourceInputs["requiredDataConnectors"] = args?.requiredDataConnectors;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["settingsDefinitionId"] = args?.settingsDefinitionId;
            resourceInputs["settingsResourceName"] = args?.settingsResourceName;
            resourceInputs["settingsStatus"] = args?.settingsStatus;
            resourceInputs["tactics"] = args?.tactics;
            resourceInputs["techniques"] = args?.techniques;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["anomalySettingsVersion"] = undefined /*out*/;
            resourceInputs["anomalyVersion"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["customizableObservations"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["frequency"] = undefined /*out*/;
            resourceInputs["isDefaultSettings"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastModifiedUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["requiredDataConnectors"] = undefined /*out*/;
            resourceInputs["settingsDefinitionId"] = undefined /*out*/;
            resourceInputs["settingsStatus"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tactics"] = undefined /*out*/;
            resourceInputs["techniques"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20220501preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20220601preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20220701preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20220801preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20220901preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20221001preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20221101:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20221101preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20221201preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230201:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230201preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230301preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230401preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230501preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230601preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230701preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230801preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20230901preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20231001preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20231101:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20231201preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20240101preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20240301:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20240401preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20240901:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20241001preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250101preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250301:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250401preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250601:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250701preview:AnomalySecurityMLAnalyticsSettings" }, { type: "azure-native:securityinsights/v20250901:AnomalySecurityMLAnalyticsSettings" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AnomalySecurityMLAnalyticsSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AnomalySecurityMLAnalyticsSettings = AnomalySecurityMLAnalyticsSettings;
