// @pulumi/azure-native/securityinsights/automationRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview, 2025-06-01, 2025-07-01-preview, 2025-09-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AutomationRule extends pulumi.CustomResource {
    /**
     * Get an existing AutomationRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AutomationRule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:AutomationRule';
    /**
     * Returns true if the given object is an instance of AutomationRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AutomationRule.__pulumiType;
    }
    /**
     * Create a AutomationRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.actions === undefined && !opts.urn) {
                throw new Error("Missing required property 'actions'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.order === undefined && !opts.urn) {
                throw new Error("Missing required property 'order'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.triggeringLogic === undefined && !opts.urn) {
                throw new Error("Missing required property 'triggeringLogic'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["actions"] = args?.actions;
            resourceInputs["automationRuleId"] = args?.automationRuleId;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["order"] = args?.order;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["triggeringLogic"] = args?.triggeringLogic;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedBy"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["actions"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedBy"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["order"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["triggeringLogic"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:AutomationRule" }, { type: "azure-native:securityinsights/v20210901preview:AutomationRule" }, { type: "azure-native:securityinsights/v20211001:AutomationRule" }, { type: "azure-native:securityinsights/v20211001preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220101preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220401preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220501preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220601preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220701preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220801:AutomationRule" }, { type: "azure-native:securityinsights/v20220801preview:AutomationRule" }, { type: "azure-native:securityinsights/v20220901preview:AutomationRule" }, { type: "azure-native:securityinsights/v20221001preview:AutomationRule" }, { type: "azure-native:securityinsights/v20221101:AutomationRule" }, { type: "azure-native:securityinsights/v20221101preview:AutomationRule" }, { type: "azure-native:securityinsights/v20221201preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230201:AutomationRule" }, { type: "azure-native:securityinsights/v20230201preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230301preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230401preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230501preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230601preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230701preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230801preview:AutomationRule" }, { type: "azure-native:securityinsights/v20230901preview:AutomationRule" }, { type: "azure-native:securityinsights/v20231001preview:AutomationRule" }, { type: "azure-native:securityinsights/v20231101:AutomationRule" }, { type: "azure-native:securityinsights/v20231201preview:AutomationRule" }, { type: "azure-native:securityinsights/v20240101preview:AutomationRule" }, { type: "azure-native:securityinsights/v20240301:AutomationRule" }, { type: "azure-native:securityinsights/v20240401preview:AutomationRule" }, { type: "azure-native:securityinsights/v20240901:AutomationRule" }, { type: "azure-native:securityinsights/v20241001preview:AutomationRule" }, { type: "azure-native:securityinsights/v20250101preview:AutomationRule" }, { type: "azure-native:securityinsights/v20250301:AutomationRule" }, { type: "azure-native:securityinsights/v20250401preview:AutomationRule" }, { type: "azure-native:securityinsights/v20250601:AutomationRule" }, { type: "azure-native:securityinsights/v20250701preview:AutomationRule" }, { type: "azure-native:securityinsights/v20250901:AutomationRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AutomationRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AutomationRule = AutomationRule;
