// @pulumi/azure-native/securityinsights/contentPackage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentPackage = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a Package in Azure Security Insights.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview, 2025-06-01, 2025-07-01-preview, 2025-09-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ContentPackage extends pulumi.CustomResource {
    /**
     * Get an existing ContentPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ContentPackage(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:ContentPackage';
    /**
     * Returns true if the given object is an instance of ContentPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContentPackage.__pulumiType;
    }
    /**
     * Create a ContentPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.contentId === undefined && !opts.urn) {
                throw new Error("Missing required property 'contentId'");
            }
            if (args?.contentKind === undefined && !opts.urn) {
                throw new Error("Missing required property 'contentKind'");
            }
            if (args?.contentProductId === undefined && !opts.urn) {
                throw new Error("Missing required property 'contentProductId'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.version === undefined && !opts.urn) {
                throw new Error("Missing required property 'version'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["author"] = args?.author;
            resourceInputs["categories"] = args?.categories;
            resourceInputs["contentId"] = args?.contentId;
            resourceInputs["contentKind"] = args?.contentKind;
            resourceInputs["contentProductId"] = args?.contentProductId;
            resourceInputs["contentSchemaVersion"] = args?.contentSchemaVersion;
            resourceInputs["dependencies"] = args?.dependencies;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["firstPublishDate"] = args?.firstPublishDate;
            resourceInputs["icon"] = args?.icon;
            resourceInputs["isDeprecated"] = args?.isDeprecated;
            resourceInputs["isFeatured"] = args?.isFeatured;
            resourceInputs["isNew"] = args?.isNew;
            resourceInputs["isPreview"] = args?.isPreview;
            resourceInputs["lastPublishDate"] = args?.lastPublishDate;
            resourceInputs["packageId"] = args?.packageId;
            resourceInputs["providers"] = args?.providers;
            resourceInputs["publisherDisplayName"] = args?.publisherDisplayName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["source"] = args?.source;
            resourceInputs["support"] = args?.support;
            resourceInputs["threatAnalysisTactics"] = args?.threatAnalysisTactics;
            resourceInputs["threatAnalysisTechniques"] = args?.threatAnalysisTechniques;
            resourceInputs["version"] = args?.version;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["categories"] = undefined /*out*/;
            resourceInputs["contentId"] = undefined /*out*/;
            resourceInputs["contentKind"] = undefined /*out*/;
            resourceInputs["contentProductId"] = undefined /*out*/;
            resourceInputs["contentSchemaVersion"] = undefined /*out*/;
            resourceInputs["dependencies"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["firstPublishDate"] = undefined /*out*/;
            resourceInputs["icon"] = undefined /*out*/;
            resourceInputs["isDeprecated"] = undefined /*out*/;
            resourceInputs["isFeatured"] = undefined /*out*/;
            resourceInputs["isNew"] = undefined /*out*/;
            resourceInputs["isPreview"] = undefined /*out*/;
            resourceInputs["lastPublishDate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["providers"] = undefined /*out*/;
            resourceInputs["publisherDisplayName"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
            resourceInputs["support"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["threatAnalysisTactics"] = undefined /*out*/;
            resourceInputs["threatAnalysisTechniques"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230501preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230601preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230701preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230801preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230901preview:ContentPackage" }, { type: "azure-native:securityinsights/v20231001preview:ContentPackage" }, { type: "azure-native:securityinsights/v20231101:ContentPackage" }, { type: "azure-native:securityinsights/v20231201preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240101preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240301:ContentPackage" }, { type: "azure-native:securityinsights/v20240401preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240901:ContentPackage" }, { type: "azure-native:securityinsights/v20241001preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250101preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250301:ContentPackage" }, { type: "azure-native:securityinsights/v20250401preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250601:ContentPackage" }, { type: "azure-native:securityinsights/v20250701preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250901:ContentPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ContentPackage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ContentPackage = ContentPackage;
