// @pulumi/azure-native/securityinsights/incidentTask.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentTask = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Describes incident task properties
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview, 2025-06-01, 2025-07-01-preview, 2025-09-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class IncidentTask extends pulumi.CustomResource {
    /**
     * Get an existing IncidentTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IncidentTask(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:IncidentTask';
    /**
     * Returns true if the given object is an instance of IncidentTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IncidentTask.__pulumiType;
    }
    /**
     * Create a IncidentTask resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.incidentId === undefined && !opts.urn) {
                throw new Error("Missing required property 'incidentId'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.status === undefined && !opts.urn) {
                throw new Error("Missing required property 'status'");
            }
            if (args?.title === undefined && !opts.urn) {
                throw new Error("Missing required property 'title'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["createdBy"] = args?.createdBy;
            resourceInputs["description"] = args?.description;
            resourceInputs["incidentId"] = args?.incidentId;
            resourceInputs["incidentTaskId"] = args?.incidentTaskId;
            resourceInputs["lastModifiedBy"] = args?.lastModifiedBy;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["status"] = args?.status;
            resourceInputs["title"] = args?.title;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedBy"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["title"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20221201preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230201preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230301preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230401preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230501preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230601preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230701preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230801preview:IncidentTask" }, { type: "azure-native:securityinsights/v20230901preview:IncidentTask" }, { type: "azure-native:securityinsights/v20231001preview:IncidentTask" }, { type: "azure-native:securityinsights/v20231201preview:IncidentTask" }, { type: "azure-native:securityinsights/v20240101preview:IncidentTask" }, { type: "azure-native:securityinsights/v20240301:IncidentTask" }, { type: "azure-native:securityinsights/v20240401preview:IncidentTask" }, { type: "azure-native:securityinsights/v20240901:IncidentTask" }, { type: "azure-native:securityinsights/v20241001preview:IncidentTask" }, { type: "azure-native:securityinsights/v20250101preview:IncidentTask" }, { type: "azure-native:securityinsights/v20250301:IncidentTask" }, { type: "azure-native:securityinsights/v20250401preview:IncidentTask" }, { type: "azure-native:securityinsights/v20250601:IncidentTask" }, { type: "azure-native:securityinsights/v20250701preview:IncidentTask" }, { type: "azure-native:securityinsights/v20250901:IncidentTask" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IncidentTask.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IncidentTask = IncidentTask;
