// @pulumi/azure-native/securityinsights/watchlistItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatchlistItem = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a Watchlist Item in Azure Security Insights.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview, 2025-06-01, 2025-07-01-preview, 2025-09-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WatchlistItem extends pulumi.CustomResource {
    /**
     * Get an existing WatchlistItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WatchlistItem(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:WatchlistItem';
    /**
     * Returns true if the given object is an instance of WatchlistItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WatchlistItem.__pulumiType;
    }
    /**
     * Create a WatchlistItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.itemsKeyValue === undefined && !opts.urn) {
                throw new Error("Missing required property 'itemsKeyValue'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.watchlistAlias === undefined && !opts.urn) {
                throw new Error("Missing required property 'watchlistAlias'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["created"] = args?.created;
            resourceInputs["createdBy"] = args?.createdBy;
            resourceInputs["entityMapping"] = args?.entityMapping;
            resourceInputs["isDeleted"] = args?.isDeleted;
            resourceInputs["itemsKeyValue"] = args?.itemsKeyValue;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tenantId"] = args?.tenantId;
            resourceInputs["updated"] = args?.updated;
            resourceInputs["updatedBy"] = args?.updatedBy;
            resourceInputs["watchlistAlias"] = args?.watchlistAlias;
            resourceInputs["watchlistItemId"] = args?.watchlistItemId;
            resourceInputs["watchlistItemType"] = args?.watchlistItemType;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["entityMapping"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isDeleted"] = undefined /*out*/;
            resourceInputs["itemsKeyValue"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updated"] = undefined /*out*/;
            resourceInputs["updatedBy"] = undefined /*out*/;
            resourceInputs["watchlistItemId"] = undefined /*out*/;
            resourceInputs["watchlistItemType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20210301preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20210401:WatchlistItem" }, { type: "azure-native:securityinsights/v20210901preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20211001:WatchlistItem" }, { type: "azure-native:securityinsights/v20211001preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220101preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220401preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220501preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220601preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220701preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220801:WatchlistItem" }, { type: "azure-native:securityinsights/v20220801preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20220901preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20221001preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20221101:WatchlistItem" }, { type: "azure-native:securityinsights/v20221101preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20221201preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230201:WatchlistItem" }, { type: "azure-native:securityinsights/v20230201preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230301preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230401preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230501preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230601preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230701preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230801preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20230901preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20231001preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20231101:WatchlistItem" }, { type: "azure-native:securityinsights/v20231201preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20240101preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20240301:WatchlistItem" }, { type: "azure-native:securityinsights/v20240401preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20240901:WatchlistItem" }, { type: "azure-native:securityinsights/v20241001preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20250101preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20250301:WatchlistItem" }, { type: "azure-native:securityinsights/v20250401preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20250601:WatchlistItem" }, { type: "azure-native:securityinsights/v20250701preview:WatchlistItem" }, { type: "azure-native:securityinsights/v20250901:WatchlistItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WatchlistItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WatchlistItem = WatchlistItem;
