// @pulumi/azure-native/securityinsights/workspaceManagerAssignment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceManagerAssignment = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The workspace manager assignment
 *
 * Uses Azure REST API version 2025-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-12-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-10-01-preview, 2025-04-01-preview, 2025-07-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WorkspaceManagerAssignment extends pulumi.CustomResource {
    /**
     * Get an existing WorkspaceManagerAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspaceManagerAssignment(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:securityinsights:WorkspaceManagerAssignment';
    /**
     * Returns true if the given object is an instance of WorkspaceManagerAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceManagerAssignment.__pulumiType;
    }
    /**
     * Create a WorkspaceManagerAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.items === undefined && !opts.urn) {
                throw new Error("Missing required property 'items'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.targetResourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetResourceName'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["items"] = args?.items;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["targetResourceName"] = args?.targetResourceName;
            resourceInputs["workspaceManagerAssignmentName"] = args?.workspaceManagerAssignmentName;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastJobEndTime"] = undefined /*out*/;
            resourceInputs["lastJobProvisioningState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["items"] = undefined /*out*/;
            resourceInputs["lastJobEndTime"] = undefined /*out*/;
            resourceInputs["lastJobProvisioningState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["targetResourceName"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20230501preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20230601preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20230701preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20230801preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20230901preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20231001preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20231201preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20240101preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20240401preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20241001preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20250101preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20250401preview:WorkspaceManagerAssignment" }, { type: "azure-native:securityinsights/v20250701preview:WorkspaceManagerAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkspaceManagerAssignment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkspaceManagerAssignment = WorkspaceManagerAssignment;
