// @pulumi/aws/autoscaling/lifecycleHook.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleHook = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an AutoScaling Lifecycle Hook resource.
 *
 * > **NOTE:** This provider has two types of ways you can add lifecycle hooks - via
 * the `initialLifecycleHook` attribute from the
 * `aws.autoscaling.Group`
 * resource, or via this one. Hooks added via this resource will not be added
 * until the autoscaling group has been created, and depending on your
 * capacity
 * settings, after the initial instances have been launched, creating unintended
 * behavior. If you need hooks to run on all instances, add them with
 * `initialLifecycleHook` in
 * `aws.autoscaling.Group`,
 * but take care to not duplicate those hooks with this resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const foobar = new aws.autoscaling.Group("foobar", {
 *     availabilityZones: ["us-west-2a"],
 *     name: "test-foobar5",
 *     healthCheckType: "EC2",
 *     terminationPolicies: ["OldestInstance"],
 *     tags: [{
 *         key: "Foo",
 *         value: "foo-bar",
 *         propagateAtLaunch: true,
 *     }],
 * });
 * const foobarLifecycleHook = new aws.autoscaling.LifecycleHook("foobar", {
 *     name: "foobar",
 *     autoscalingGroupName: foobar.name,
 *     defaultResult: "CONTINUE",
 *     heartbeatTimeout: 2000,
 *     lifecycleTransition: "autoscaling:EC2_INSTANCE_LAUNCHING",
 *     notificationMetadata: JSON.stringify({
 *         foo: "bar",
 *     }),
 *     notificationTargetArn: "arn:aws:sqs:us-east-1:444455556666:queue1*",
 *     roleArn: "arn:aws:iam::123456789012:role/S3Access",
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `autoscalingGroupName` (String) Name of the Auto Scaling group.
 * * `name` (String) Name of the lifecycle hook.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import AutoScaling Lifecycle Hooks using `autoscalingGroupName` and `name` separated by a forward slash (`/`). For example:
 *
 * ```sh
 * $ pulumi import aws:autoscaling/lifecycleHook:LifecycleHook example example-asg/example-hook
 * ```
 */
class LifecycleHook extends pulumi.CustomResource {
    /**
     * Get an existing LifecycleHook resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LifecycleHook(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:autoscaling/lifecycleHook:LifecycleHook';
    /**
     * Returns true if the given object is an instance of LifecycleHook.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LifecycleHook.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["autoscalingGroupName"] = state?.autoscalingGroupName;
            resourceInputs["defaultResult"] = state?.defaultResult;
            resourceInputs["heartbeatTimeout"] = state?.heartbeatTimeout;
            resourceInputs["lifecycleTransition"] = state?.lifecycleTransition;
            resourceInputs["name"] = state?.name;
            resourceInputs["notificationMetadata"] = state?.notificationMetadata;
            resourceInputs["notificationTargetArn"] = state?.notificationTargetArn;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
        }
        else {
            const args = argsOrState;
            if (args?.autoscalingGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'autoscalingGroupName'");
            }
            if (args?.lifecycleTransition === undefined && !opts.urn) {
                throw new Error("Missing required property 'lifecycleTransition'");
            }
            resourceInputs["autoscalingGroupName"] = args?.autoscalingGroupName;
            resourceInputs["defaultResult"] = args?.defaultResult;
            resourceInputs["heartbeatTimeout"] = args?.heartbeatTimeout;
            resourceInputs["lifecycleTransition"] = args?.lifecycleTransition;
            resourceInputs["name"] = args?.name;
            resourceInputs["notificationMetadata"] = args?.notificationMetadata;
            resourceInputs["notificationTargetArn"] = args?.notificationTargetArn;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LifecycleHook.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LifecycleHook = LifecycleHook;
