// @pulumi/aws/lb/trustStoreRevocation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustStoreRevocation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a ELBv2 Trust Store Revocation for use with Application Load Balancer Listener resources.
 *
 * ## Example Usage
 *
 * ### Trust Store With Revocations
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.lb.TrustStore("test", {
 *     name: "tf-example-lb-ts",
 *     caCertificatesBundleS3Bucket: "...",
 *     caCertificatesBundleS3Key: "...",
 * });
 * const testTrustStoreRevocation = new aws.lb.TrustStoreRevocation("test", {
 *     trustStoreArn: test.arn,
 *     revocationsS3Bucket: "...",
 *     revocationsS3Key: "...",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Trust Store Revocations using their ARN. For example:
 *
 * ```sh
 * $ pulumi import aws:lb/trustStoreRevocation:TrustStoreRevocation example arn:aws:elasticloadbalancing:us-west-2:187416307283:truststore/my-trust-store/20cfe21448b66314,6
 * ```
 */
class TrustStoreRevocation extends pulumi.CustomResource {
    /**
     * Get an existing TrustStoreRevocation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TrustStoreRevocation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:lb/trustStoreRevocation:TrustStoreRevocation';
    /**
     * Returns true if the given object is an instance of TrustStoreRevocation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrustStoreRevocation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["revocationId"] = state?.revocationId;
            resourceInputs["revocationsS3Bucket"] = state?.revocationsS3Bucket;
            resourceInputs["revocationsS3Key"] = state?.revocationsS3Key;
            resourceInputs["revocationsS3ObjectVersion"] = state?.revocationsS3ObjectVersion;
            resourceInputs["trustStoreArn"] = state?.trustStoreArn;
        }
        else {
            const args = argsOrState;
            if (args?.revocationsS3Bucket === undefined && !opts.urn) {
                throw new Error("Missing required property 'revocationsS3Bucket'");
            }
            if (args?.revocationsS3Key === undefined && !opts.urn) {
                throw new Error("Missing required property 'revocationsS3Key'");
            }
            if (args?.trustStoreArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'trustStoreArn'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["revocationsS3Bucket"] = args?.revocationsS3Bucket;
            resourceInputs["revocationsS3Key"] = args?.revocationsS3Key;
            resourceInputs["revocationsS3ObjectVersion"] = args?.revocationsS3ObjectVersion;
            resourceInputs["trustStoreArn"] = args?.trustStoreArn;
            resourceInputs["revocationId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TrustStoreRevocation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrustStoreRevocation = TrustStoreRevocation;
