// @pulumi/aws/gamelift/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Script = exports.GameSessionQueue = exports.GameServerGroup = exports.Fleet = exports.Build = exports.Alias = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Alias = null;
utilities.lazyLoad(exports, ["Alias"], () => require("./alias"));
exports.Build = null;
utilities.lazyLoad(exports, ["Build"], () => require("./build"));
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.GameServerGroup = null;
utilities.lazyLoad(exports, ["GameServerGroup"], () => require("./gameServerGroup"));
exports.GameSessionQueue = null;
utilities.lazyLoad(exports, ["GameSessionQueue"], () => require("./gameSessionQueue"));
exports.Script = null;
utilities.lazyLoad(exports, ["Script"], () => require("./script"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:gamelift/alias:Alias":
                return new exports.Alias(name, undefined, { urn });
            case "aws:gamelift/build:Build":
                return new exports.Build(name, undefined, { urn });
            case "aws:gamelift/fleet:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "aws:gamelift/gameServerGroup:GameServerGroup":
                return new exports.GameServerGroup(name, undefined, { urn });
            case "aws:gamelift/gameSessionQueue:GameSessionQueue":
                return new exports.GameSessionQueue(name, undefined, { urn });
            case "aws:gamelift/script:Script":
                return new exports.Script(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "gamelift/alias", _module);
pulumi.runtime.registerResourceModule("aws", "gamelift/build", _module);
pulumi.runtime.registerResourceModule("aws", "gamelift/fleet", _module);
pulumi.runtime.registerResourceModule("aws", "gamelift/gameServerGroup", _module);
pulumi.runtime.registerResourceModule("aws", "gamelift/gameSessionQueue", _module);
pulumi.runtime.registerResourceModule("aws", "gamelift/script", _module);
