// @pulumi/azure-native/avs/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalReachConnection = exports.getWorkloadNetworkVMGroupOutput = exports.getWorkloadNetworkVMGroup = exports.getWorkloadNetworkSegmentOutput = exports.getWorkloadNetworkSegment = exports.getWorkloadNetworkPublicIPOutput = exports.getWorkloadNetworkPublicIP = exports.getWorkloadNetworkPortMirroringOutput = exports.getWorkloadNetworkPortMirroring = exports.getWorkloadNetworkDnsZoneOutput = exports.getWorkloadNetworkDnsZone = exports.getWorkloadNetworkDnsServiceOutput = exports.getWorkloadNetworkDnsService = exports.getWorkloadNetworkDhcpOutput = exports.getWorkloadNetworkDhcp = exports.getScriptExecutionLogsOutput = exports.getScriptExecutionLogs = exports.getScriptExecutionOutput = exports.getScriptExecution = exports.getPureStoragePolicyOutput = exports.getPureStoragePolicy = exports.getPrivateCloudVcfLicenseOutput = exports.getPrivateCloudVcfLicense = exports.getPrivateCloudOutput = exports.getPrivateCloud = exports.getPlacementPolicyOutput = exports.getPlacementPolicy = exports.getLicenseOutput = exports.getLicense = exports.getIscsiPathOutput = exports.getIscsiPath = exports.getHcxEnterpriseSiteOutput = exports.getHcxEnterpriseSite = exports.getGlobalReachConnectionOutput = exports.getGlobalReachConnection = exports.getDatastoreOutput = exports.getDatastore = exports.getClusterOutput = exports.getCluster = exports.getCloudLinkOutput = exports.getCloudLink = exports.getAuthorizationOutput = exports.getAuthorization = exports.getAddonOutput = exports.getAddon = exports.Datastore = exports.Cluster = exports.CloudLink = exports.Authorization = exports.Addon = void 0;
exports.WorkloadNetworkVMGroup = exports.WorkloadNetworkSegment = exports.WorkloadNetworkPublicIP = exports.WorkloadNetworkPortMirroring = exports.WorkloadNetworkDnsZone = exports.WorkloadNetworkDnsService = exports.WorkloadNetworkDhcp = exports.ScriptExecution = exports.PureStoragePolicy = exports.PrivateCloud = exports.PlacementPolicy = exports.listPrivateCloudAdminCredentialsOutput = exports.listPrivateCloudAdminCredentials = exports.listClusterZonesOutput = exports.listClusterZones = exports.License = exports.IscsiPath = exports.HcxEnterpriseSite = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Addon = null;
utilities.lazyLoad(exports, ["Addon"], () => require("./addon"));
exports.Authorization = null;
utilities.lazyLoad(exports, ["Authorization"], () => require("./authorization"));
exports.CloudLink = null;
utilities.lazyLoad(exports, ["CloudLink"], () => require("./cloudLink"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.Datastore = null;
utilities.lazyLoad(exports, ["Datastore"], () => require("./datastore"));
exports.getAddon = null;
exports.getAddonOutput = null;
utilities.lazyLoad(exports, ["getAddon", "getAddonOutput"], () => require("./getAddon"));
exports.getAuthorization = null;
exports.getAuthorizationOutput = null;
utilities.lazyLoad(exports, ["getAuthorization", "getAuthorizationOutput"], () => require("./getAuthorization"));
exports.getCloudLink = null;
exports.getCloudLinkOutput = null;
utilities.lazyLoad(exports, ["getCloudLink", "getCloudLinkOutput"], () => require("./getCloudLink"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getDatastore = null;
exports.getDatastoreOutput = null;
utilities.lazyLoad(exports, ["getDatastore", "getDatastoreOutput"], () => require("./getDatastore"));
exports.getGlobalReachConnection = null;
exports.getGlobalReachConnectionOutput = null;
utilities.lazyLoad(exports, ["getGlobalReachConnection", "getGlobalReachConnectionOutput"], () => require("./getGlobalReachConnection"));
exports.getHcxEnterpriseSite = null;
exports.getHcxEnterpriseSiteOutput = null;
utilities.lazyLoad(exports, ["getHcxEnterpriseSite", "getHcxEnterpriseSiteOutput"], () => require("./getHcxEnterpriseSite"));
exports.getIscsiPath = null;
exports.getIscsiPathOutput = null;
utilities.lazyLoad(exports, ["getIscsiPath", "getIscsiPathOutput"], () => require("./getIscsiPath"));
exports.getLicense = null;
exports.getLicenseOutput = null;
utilities.lazyLoad(exports, ["getLicense", "getLicenseOutput"], () => require("./getLicense"));
exports.getPlacementPolicy = null;
exports.getPlacementPolicyOutput = null;
utilities.lazyLoad(exports, ["getPlacementPolicy", "getPlacementPolicyOutput"], () => require("./getPlacementPolicy"));
exports.getPrivateCloud = null;
exports.getPrivateCloudOutput = null;
utilities.lazyLoad(exports, ["getPrivateCloud", "getPrivateCloudOutput"], () => require("./getPrivateCloud"));
exports.getPrivateCloudVcfLicense = null;
exports.getPrivateCloudVcfLicenseOutput = null;
utilities.lazyLoad(exports, ["getPrivateCloudVcfLicense", "getPrivateCloudVcfLicenseOutput"], () => require("./getPrivateCloudVcfLicense"));
exports.getPureStoragePolicy = null;
exports.getPureStoragePolicyOutput = null;
utilities.lazyLoad(exports, ["getPureStoragePolicy", "getPureStoragePolicyOutput"], () => require("./getPureStoragePolicy"));
exports.getScriptExecution = null;
exports.getScriptExecutionOutput = null;
utilities.lazyLoad(exports, ["getScriptExecution", "getScriptExecutionOutput"], () => require("./getScriptExecution"));
exports.getScriptExecutionLogs = null;
exports.getScriptExecutionLogsOutput = null;
utilities.lazyLoad(exports, ["getScriptExecutionLogs", "getScriptExecutionLogsOutput"], () => require("./getScriptExecutionLogs"));
exports.getWorkloadNetworkDhcp = null;
exports.getWorkloadNetworkDhcpOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkDhcp", "getWorkloadNetworkDhcpOutput"], () => require("./getWorkloadNetworkDhcp"));
exports.getWorkloadNetworkDnsService = null;
exports.getWorkloadNetworkDnsServiceOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkDnsService", "getWorkloadNetworkDnsServiceOutput"], () => require("./getWorkloadNetworkDnsService"));
exports.getWorkloadNetworkDnsZone = null;
exports.getWorkloadNetworkDnsZoneOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkDnsZone", "getWorkloadNetworkDnsZoneOutput"], () => require("./getWorkloadNetworkDnsZone"));
exports.getWorkloadNetworkPortMirroring = null;
exports.getWorkloadNetworkPortMirroringOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkPortMirroring", "getWorkloadNetworkPortMirroringOutput"], () => require("./getWorkloadNetworkPortMirroring"));
exports.getWorkloadNetworkPublicIP = null;
exports.getWorkloadNetworkPublicIPOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkPublicIP", "getWorkloadNetworkPublicIPOutput"], () => require("./getWorkloadNetworkPublicIP"));
exports.getWorkloadNetworkSegment = null;
exports.getWorkloadNetworkSegmentOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkSegment", "getWorkloadNetworkSegmentOutput"], () => require("./getWorkloadNetworkSegment"));
exports.getWorkloadNetworkVMGroup = null;
exports.getWorkloadNetworkVMGroupOutput = null;
utilities.lazyLoad(exports, ["getWorkloadNetworkVMGroup", "getWorkloadNetworkVMGroupOutput"], () => require("./getWorkloadNetworkVMGroup"));
exports.GlobalReachConnection = null;
utilities.lazyLoad(exports, ["GlobalReachConnection"], () => require("./globalReachConnection"));
exports.HcxEnterpriseSite = null;
utilities.lazyLoad(exports, ["HcxEnterpriseSite"], () => require("./hcxEnterpriseSite"));
exports.IscsiPath = null;
utilities.lazyLoad(exports, ["IscsiPath"], () => require("./iscsiPath"));
exports.License = null;
utilities.lazyLoad(exports, ["License"], () => require("./license"));
exports.listClusterZones = null;
exports.listClusterZonesOutput = null;
utilities.lazyLoad(exports, ["listClusterZones", "listClusterZonesOutput"], () => require("./listClusterZones"));
exports.listPrivateCloudAdminCredentials = null;
exports.listPrivateCloudAdminCredentialsOutput = null;
utilities.lazyLoad(exports, ["listPrivateCloudAdminCredentials", "listPrivateCloudAdminCredentialsOutput"], () => require("./listPrivateCloudAdminCredentials"));
exports.PlacementPolicy = null;
utilities.lazyLoad(exports, ["PlacementPolicy"], () => require("./placementPolicy"));
exports.PrivateCloud = null;
utilities.lazyLoad(exports, ["PrivateCloud"], () => require("./privateCloud"));
exports.PureStoragePolicy = null;
utilities.lazyLoad(exports, ["PureStoragePolicy"], () => require("./pureStoragePolicy"));
exports.ScriptExecution = null;
utilities.lazyLoad(exports, ["ScriptExecution"], () => require("./scriptExecution"));
exports.WorkloadNetworkDhcp = null;
utilities.lazyLoad(exports, ["WorkloadNetworkDhcp"], () => require("./workloadNetworkDhcp"));
exports.WorkloadNetworkDnsService = null;
utilities.lazyLoad(exports, ["WorkloadNetworkDnsService"], () => require("./workloadNetworkDnsService"));
exports.WorkloadNetworkDnsZone = null;
utilities.lazyLoad(exports, ["WorkloadNetworkDnsZone"], () => require("./workloadNetworkDnsZone"));
exports.WorkloadNetworkPortMirroring = null;
utilities.lazyLoad(exports, ["WorkloadNetworkPortMirroring"], () => require("./workloadNetworkPortMirroring"));
exports.WorkloadNetworkPublicIP = null;
utilities.lazyLoad(exports, ["WorkloadNetworkPublicIP"], () => require("./workloadNetworkPublicIP"));
exports.WorkloadNetworkSegment = null;
utilities.lazyLoad(exports, ["WorkloadNetworkSegment"], () => require("./workloadNetworkSegment"));
exports.WorkloadNetworkVMGroup = null;
utilities.lazyLoad(exports, ["WorkloadNetworkVMGroup"], () => require("./workloadNetworkVMGroup"));
// Export enums:
__exportStar(require("../types/enums/avs"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:avs:Addon":
                return new exports.Addon(name, undefined, { urn });
            case "azure-native:avs:Authorization":
                return new exports.Authorization(name, undefined, { urn });
            case "azure-native:avs:CloudLink":
                return new exports.CloudLink(name, undefined, { urn });
            case "azure-native:avs:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:avs:Datastore":
                return new exports.Datastore(name, undefined, { urn });
            case "azure-native:avs:GlobalReachConnection":
                return new exports.GlobalReachConnection(name, undefined, { urn });
            case "azure-native:avs:HcxEnterpriseSite":
                return new exports.HcxEnterpriseSite(name, undefined, { urn });
            case "azure-native:avs:IscsiPath":
                return new exports.IscsiPath(name, undefined, { urn });
            case "azure-native:avs:License":
                return new exports.License(name, undefined, { urn });
            case "azure-native:avs:PlacementPolicy":
                return new exports.PlacementPolicy(name, undefined, { urn });
            case "azure-native:avs:PrivateCloud":
                return new exports.PrivateCloud(name, undefined, { urn });
            case "azure-native:avs:PureStoragePolicy":
                return new exports.PureStoragePolicy(name, undefined, { urn });
            case "azure-native:avs:ScriptExecution":
                return new exports.ScriptExecution(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkDhcp":
                return new exports.WorkloadNetworkDhcp(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkDnsService":
                return new exports.WorkloadNetworkDnsService(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkDnsZone":
                return new exports.WorkloadNetworkDnsZone(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkPortMirroring":
                return new exports.WorkloadNetworkPortMirroring(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkPublicIP":
                return new exports.WorkloadNetworkPublicIP(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkSegment":
                return new exports.WorkloadNetworkSegment(name, undefined, { urn });
            case "azure-native:avs:WorkloadNetworkVMGroup":
                return new exports.WorkloadNetworkVMGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "avs", _module);
