// @pulumi/aws/ram/resourceShareAssociationsExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceShareAssociationsExclusive = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for maintaining exclusive management of principal and resource associations for an AWS RAM (Resource Access Manager) Resource Share.
 *
 * !> This resource takes exclusive ownership over principal and resource associations for a resource share. This includes removal of principals and resources which are not explicitly configured.
 *
 * > Destruction of this resource will disassociate all configured principals and resources from the resource share.
 *
 * > **NOTE:** This resource cannot be used in conjunction with `aws.ram.PrincipalAssociation` or `aws.ram.ResourceAssociation` for the same resource share. Using them together will cause persistent drift and conflicts.
 *
 * ## Example Usage
 *
 * ### Basic Usage with Principals
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ram.ResourceShare("example", {
 *     name: "example",
 *     allowExternalPrincipals: true,
 * });
 * const exampleVpc = new aws.ec2.Vpc("example", {cidrBlock: "10.0.0.0/16"});
 * const exampleSubnet = new aws.ec2.Subnet("example", {
 *     vpcId: exampleVpc.id,
 *     cidrBlock: "10.0.1.0/24",
 * });
 * const exampleResourceShareAssociationsExclusive = new aws.ram.ResourceShareAssociationsExclusive("example", {
 *     resourceShareArn: example.arn,
 *     principals: [
 *         "111111111111",
 *         "222222222222",
 *     ],
 *     resourceArns: [exampleSubnet.arn],
 * });
 * ```
 *
 * ### With Organization Principal
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as std from "@pulumi/std";
 *
 * const example = new aws.ram.ResourceShare("example", {name: "example"});
 * const exampleVpc = new aws.ec2.Vpc("example", {cidrBlock: "10.0.0.0/16"});
 * const exampleSubnet: aws.ec2.Subnet[] = [];
 * for (const range = {value: 0}; range.value < 2; range.value++) {
 *     exampleSubnet.push(new aws.ec2.Subnet(`example-${range.value}`, {
 *         vpcId: exampleVpc.id,
 *         cidrBlock: std.cidrsubnetOutput({
 *             input: exampleVpc.cidrBlock,
 *             newbits: 8,
 *             netnum: range.value,
 *         }).apply(invoke => invoke.result),
 *     }));
 * }
 * const exampleResourceShareAssociationsExclusive = new aws.ram.ResourceShareAssociationsExclusive("example", {
 *     resourceShareArn: example.arn,
 *     principals: [exampleAwsOrganizationsOrganization.arn],
 *     resourceArns: exampleSubnet.map(__item => __item.arn),
 * });
 * ```
 *
 * ### With Service Principals
 *
 * When sharing resources with AWS services, use service principals. Service principals follow the pattern `service-id.amazonaws.com` (e.g., `pca-connector-ad.amazonaws.com`, `elasticmapreduce.amazonaws.com`). The `sources` argument can be used to restrict which AWS accounts the service can access the shared resources from.
 *
 * > **NOTE:** Service principals cannot be mixed with other principal types (AWS account IDs, organization ARNs, OU ARNs, IAM role ARNs, or IAM user ARNs) in the same resource.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ram.ResourceShare("example", {
 *     name: "example-service-share",
 *     allowExternalPrincipals: true,
 * });
 * const exampleCertificateAuthority = new aws.acmpca.CertificateAuthority("example", {
 *     type: "ROOT",
 *     certificateAuthorityConfiguration: {
 *         keyAlgorithm: "RSA_4096",
 *         signingAlgorithm: "SHA512WITHRSA",
 *         subject: {
 *             commonName: "example.com",
 *         },
 *     },
 * });
 * const exampleResourceShareAssociationsExclusive = new aws.ram.ResourceShareAssociationsExclusive("example", {
 *     resourceShareArn: example.arn,
 *     principals: ["pca-connector-ad.amazonaws.com"],
 *     resourceArns: [exampleCertificateAuthority.arn],
 *     sources: [
 *         "111111111111",
 *         "222222222222",
 *     ],
 * });
 * ```
 *
 * ### Disallow All Associations
 *
 * To automatically remove any configured associations, omit the `principals` and `resourceArns` arguments or set them to empty lists.
 *
 * > This will not **prevent** associations from being created via Terraform (or any other interface). This resource enables bringing associations into a configured state, however, this reconciliation happens only when `apply` is proactively run.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ram.ResourceShareAssociationsExclusive("example", {resourceShareArn: exampleAwsRamResourceShare.arn});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import RAM Resource Share Association Exclusive using the `resourceShareArn`. For example:
 *
 * ```sh
 * $ pulumi import aws:ram/resourceShareAssociationsExclusive:ResourceShareAssociationsExclusive example arn:aws:ram:eu-west-1:123456789012:resource-share/73da1ab9-b94a-4ba3-8eb4-45917f7f4b12
 * ```
 */
class ResourceShareAssociationsExclusive extends pulumi.CustomResource {
    /**
     * Get an existing ResourceShareAssociationsExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceShareAssociationsExclusive(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ram/resourceShareAssociationsExclusive:ResourceShareAssociationsExclusive';
    /**
     * Returns true if the given object is an instance of ResourceShareAssociationsExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceShareAssociationsExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["principals"] = state?.principals;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceArns"] = state?.resourceArns;
            resourceInputs["resourceShareArn"] = state?.resourceShareArn;
            resourceInputs["sources"] = state?.sources;
        }
        else {
            const args = argsOrState;
            if (args?.resourceShareArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceShareArn'");
            }
            resourceInputs["principals"] = args?.principals;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceArns"] = args?.resourceArns;
            resourceInputs["resourceShareArn"] = args?.resourceShareArn;
            resourceInputs["sources"] = args?.sources;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResourceShareAssociationsExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResourceShareAssociationsExclusive = ResourceShareAssociationsExclusive;
