// @pulumi/aws/shield/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscription = exports.ProtectionHealthCheckAssociation = exports.ProtectionGroup = exports.Protection = exports.ProactiveEngagement = exports.getProtectionOutput = exports.getProtection = exports.DrtAccessRoleArnAssociation = exports.DrtAccessLogBucketAssociation = exports.ApplicationLayerAutomaticResponse = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ApplicationLayerAutomaticResponse = null;
utilities.lazyLoad(exports, ["ApplicationLayerAutomaticResponse"], () => require("./applicationLayerAutomaticResponse"));
exports.DrtAccessLogBucketAssociation = null;
utilities.lazyLoad(exports, ["DrtAccessLogBucketAssociation"], () => require("./drtAccessLogBucketAssociation"));
exports.DrtAccessRoleArnAssociation = null;
utilities.lazyLoad(exports, ["DrtAccessRoleArnAssociation"], () => require("./drtAccessRoleArnAssociation"));
exports.getProtection = null;
exports.getProtectionOutput = null;
utilities.lazyLoad(exports, ["getProtection", "getProtectionOutput"], () => require("./getProtection"));
exports.ProactiveEngagement = null;
utilities.lazyLoad(exports, ["ProactiveEngagement"], () => require("./proactiveEngagement"));
exports.Protection = null;
utilities.lazyLoad(exports, ["Protection"], () => require("./protection"));
exports.ProtectionGroup = null;
utilities.lazyLoad(exports, ["ProtectionGroup"], () => require("./protectionGroup"));
exports.ProtectionHealthCheckAssociation = null;
utilities.lazyLoad(exports, ["ProtectionHealthCheckAssociation"], () => require("./protectionHealthCheckAssociation"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:shield/applicationLayerAutomaticResponse:ApplicationLayerAutomaticResponse":
                return new exports.ApplicationLayerAutomaticResponse(name, undefined, { urn });
            case "aws:shield/drtAccessLogBucketAssociation:DrtAccessLogBucketAssociation":
                return new exports.DrtAccessLogBucketAssociation(name, undefined, { urn });
            case "aws:shield/drtAccessRoleArnAssociation:DrtAccessRoleArnAssociation":
                return new exports.DrtAccessRoleArnAssociation(name, undefined, { urn });
            case "aws:shield/proactiveEngagement:ProactiveEngagement":
                return new exports.ProactiveEngagement(name, undefined, { urn });
            case "aws:shield/protection:Protection":
                return new exports.Protection(name, undefined, { urn });
            case "aws:shield/protectionGroup:ProtectionGroup":
                return new exports.ProtectionGroup(name, undefined, { urn });
            case "aws:shield/protectionHealthCheckAssociation:ProtectionHealthCheckAssociation":
                return new exports.ProtectionHealthCheckAssociation(name, undefined, { urn });
            case "aws:shield/subscription:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "shield/applicationLayerAutomaticResponse", _module);
pulumi.runtime.registerResourceModule("aws", "shield/drtAccessLogBucketAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "shield/drtAccessRoleArnAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "shield/proactiveEngagement", _module);
pulumi.runtime.registerResourceModule("aws", "shield/protection", _module);
pulumi.runtime.registerResourceModule("aws", "shield/protectionGroup", _module);
pulumi.runtime.registerResourceModule("aws", "shield/protectionHealthCheckAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "shield/subscription", _module);
