// @pulumi/azure-native/servicefabric/managedClusterApplicationType.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedClusterApplicationType = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The application type name resource
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2023-03-01-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01-preview, 2024-02-01-preview, 2024-06-01-preview, 2024-09-01-preview, 2024-11-01-preview, 2025-03-01-preview, 2025-06-01-preview, 2025-10-01-preview, 2026-02-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native servicefabric [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ManagedClusterApplicationType extends pulumi.CustomResource {
    /**
     * Get an existing ManagedClusterApplicationType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedClusterApplicationType(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:servicefabric:ManagedClusterApplicationType';
    /**
     * Returns true if the given object is an instance of ManagedClusterApplicationType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedClusterApplicationType.__pulumiType;
    }
    /**
     * Create a ManagedClusterApplicationType resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationTypeName"] = args?.applicationTypeName;
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["location"] = args?.location;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabric/v20210101preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20210501:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20210701preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20210901privatepreview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20211101preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20220101:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20220201preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20220601preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20220801preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20221001preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20230201preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20230301preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20230701preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20230901preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20231101preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20231201preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20240201preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20240401:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20240601preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20240901preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20241101preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20250301preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20250601preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20251001preview:ManagedClusterApplicationType" }, { type: "azure-native:servicefabric/v20260201:ManagedClusterApplicationType" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedClusterApplicationType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagedClusterApplicationType = ManagedClusterApplicationType;
