// @pulumi/kubernetes/resource/v1alpha3/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceSlicePatch = exports.ResourceSlice = exports.ResourcePoolStatusRequestPatch = exports.ResourcePoolStatusRequestList = exports.ResourcePoolStatusRequest = exports.ResourceClaimTemplatePatch = exports.ResourceClaimTemplateList = exports.ResourceClaimTemplate = exports.ResourceClaimPatch = exports.ResourceClaimList = exports.ResourceClaim = exports.PodSchedulingContextPatch = exports.PodSchedulingContextList = exports.PodSchedulingContext = exports.DeviceTaintRulePatch = exports.DeviceTaintRuleList = exports.DeviceTaintRule = exports.DeviceClassPatch = exports.DeviceClassList = exports.DeviceClass = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.DeviceClass = null;
utilities.lazyLoad(exports, ["DeviceClass"], () => require("./deviceClass"));
exports.DeviceClassList = null;
utilities.lazyLoad(exports, ["DeviceClassList"], () => require("./deviceClassList"));
exports.DeviceClassPatch = null;
utilities.lazyLoad(exports, ["DeviceClassPatch"], () => require("./deviceClassPatch"));
exports.DeviceTaintRule = null;
utilities.lazyLoad(exports, ["DeviceTaintRule"], () => require("./deviceTaintRule"));
exports.DeviceTaintRuleList = null;
utilities.lazyLoad(exports, ["DeviceTaintRuleList"], () => require("./deviceTaintRuleList"));
exports.DeviceTaintRulePatch = null;
utilities.lazyLoad(exports, ["DeviceTaintRulePatch"], () => require("./deviceTaintRulePatch"));
exports.PodSchedulingContext = null;
utilities.lazyLoad(exports, ["PodSchedulingContext"], () => require("./podSchedulingContext"));
exports.PodSchedulingContextList = null;
utilities.lazyLoad(exports, ["PodSchedulingContextList"], () => require("./podSchedulingContextList"));
exports.PodSchedulingContextPatch = null;
utilities.lazyLoad(exports, ["PodSchedulingContextPatch"], () => require("./podSchedulingContextPatch"));
exports.ResourceClaim = null;
utilities.lazyLoad(exports, ["ResourceClaim"], () => require("./resourceClaim"));
exports.ResourceClaimList = null;
utilities.lazyLoad(exports, ["ResourceClaimList"], () => require("./resourceClaimList"));
exports.ResourceClaimPatch = null;
utilities.lazyLoad(exports, ["ResourceClaimPatch"], () => require("./resourceClaimPatch"));
exports.ResourceClaimTemplate = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplate"], () => require("./resourceClaimTemplate"));
exports.ResourceClaimTemplateList = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplateList"], () => require("./resourceClaimTemplateList"));
exports.ResourceClaimTemplatePatch = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplatePatch"], () => require("./resourceClaimTemplatePatch"));
exports.ResourcePoolStatusRequest = null;
utilities.lazyLoad(exports, ["ResourcePoolStatusRequest"], () => require("./resourcePoolStatusRequest"));
exports.ResourcePoolStatusRequestList = null;
utilities.lazyLoad(exports, ["ResourcePoolStatusRequestList"], () => require("./resourcePoolStatusRequestList"));
exports.ResourcePoolStatusRequestPatch = null;
utilities.lazyLoad(exports, ["ResourcePoolStatusRequestPatch"], () => require("./resourcePoolStatusRequestPatch"));
exports.ResourceSlice = null;
utilities.lazyLoad(exports, ["ResourceSlice"], () => require("./resourceSlice"));
exports.ResourceSlicePatch = null;
utilities.lazyLoad(exports, ["ResourceSlicePatch"], () => require("./resourceSlicePatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceClass":
                return new exports.DeviceClass(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceClassList":
                return new exports.DeviceClassList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceClassPatch":
                return new exports.DeviceClassPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceTaintRule":
                return new exports.DeviceTaintRule(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceTaintRuleList":
                return new exports.DeviceTaintRuleList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:DeviceTaintRulePatch":
                return new exports.DeviceTaintRulePatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:PodSchedulingContext":
                return new exports.PodSchedulingContext(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:PodSchedulingContextList":
                return new exports.PodSchedulingContextList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:PodSchedulingContextPatch":
                return new exports.PodSchedulingContextPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaim":
                return new exports.ResourceClaim(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaimList":
                return new exports.ResourceClaimList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaimPatch":
                return new exports.ResourceClaimPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaimTemplate":
                return new exports.ResourceClaimTemplate(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaimTemplateList":
                return new exports.ResourceClaimTemplateList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceClaimTemplatePatch":
                return new exports.ResourceClaimTemplatePatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourcePoolStatusRequest":
                return new exports.ResourcePoolStatusRequest(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourcePoolStatusRequestList":
                return new exports.ResourcePoolStatusRequestList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourcePoolStatusRequestPatch":
                return new exports.ResourcePoolStatusRequestPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceSlice":
                return new exports.ResourceSlice(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha3:ResourceSlicePatch":
                return new exports.ResourceSlicePatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "resource.k8s.io/v1alpha3", _module);
