// @pulumi/aws/route53/trafficPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Route53 Traffic Policy.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.route53.TrafficPolicy("example", {
 *     name: "example",
 *     comment: "example comment",
 *     document: `{
 *   \\"AWSPolicyFormatVersion\\": \\"2015-10-01\\",
 *   \\"RecordType\\": \\"A\\",
 *   \\"Endpoints\\": {
 *     \\"endpoint-start-NkPh\\": {
 *       \\"Type\\": \\"value\\",
 *       \\"Value\\": \\"10.0.0.2\\"
 *     }
 *   },
 *   \\"StartEndpoint\\": \\"endpoint-start-NkPh\\"
 * }
 * `,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Route53 Traffic Policy using the `id` and `version`. For example:
 *
 * ```sh
 * $ pulumi import aws:route53/trafficPolicy:TrafficPolicy example 01a52019-d16f-422a-ae72-c306d2b6df7e/1
 * ```
 */
class TrafficPolicy extends pulumi.CustomResource {
    /**
     * Get an existing TrafficPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TrafficPolicy(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:route53/trafficPolicy:TrafficPolicy';
    /**
     * Returns true if the given object is an instance of TrafficPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrafficPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["comment"] = state?.comment;
            resourceInputs["document"] = state?.document;
            resourceInputs["name"] = state?.name;
            resourceInputs["type"] = state?.type;
            resourceInputs["version"] = state?.version;
        }
        else {
            const args = argsOrState;
            if (args?.document === undefined && !opts.urn) {
                throw new Error("Missing required property 'document'");
            }
            resourceInputs["comment"] = args?.comment;
            resourceInputs["document"] = args?.document;
            resourceInputs["name"] = args?.name;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TrafficPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrafficPolicy = TrafficPolicy;
