// @pulumi/aws/route53/trafficPolicyInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficPolicyInstance = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a Route53 traffic policy instance resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.route53.TrafficPolicyInstance("test", {
 *     name: "test.example.com",
 *     trafficPolicyId: "b3gb108f-ea6f-45a5-baab-9d112d8b4037",
 *     trafficPolicyVersion: 1,
 *     hostedZoneId: "Z033120931TAQO548OGJC",
 *     ttl: 360,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Route53 traffic policy instance using its id. For example:
 *
 * ```sh
 * $ pulumi import aws:route53/trafficPolicyInstance:TrafficPolicyInstance test df579d9a-6396-410e-ac22-e7ad60cf9e7e
 * ```
 */
class TrafficPolicyInstance extends pulumi.CustomResource {
    /**
     * Get an existing TrafficPolicyInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TrafficPolicyInstance(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:route53/trafficPolicyInstance:TrafficPolicyInstance';
    /**
     * Returns true if the given object is an instance of TrafficPolicyInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrafficPolicyInstance.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["hostedZoneId"] = state?.hostedZoneId;
            resourceInputs["name"] = state?.name;
            resourceInputs["trafficPolicyId"] = state?.trafficPolicyId;
            resourceInputs["trafficPolicyVersion"] = state?.trafficPolicyVersion;
            resourceInputs["ttl"] = state?.ttl;
        }
        else {
            const args = argsOrState;
            if (args?.hostedZoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostedZoneId'");
            }
            if (args?.trafficPolicyId === undefined && !opts.urn) {
                throw new Error("Missing required property 'trafficPolicyId'");
            }
            if (args?.trafficPolicyVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'trafficPolicyVersion'");
            }
            if (args?.ttl === undefined && !opts.urn) {
                throw new Error("Missing required property 'ttl'");
            }
            resourceInputs["hostedZoneId"] = args?.hostedZoneId;
            resourceInputs["name"] = args?.name;
            resourceInputs["trafficPolicyId"] = args?.trafficPolicyId;
            resourceInputs["trafficPolicyVersion"] = args?.trafficPolicyVersion;
            resourceInputs["ttl"] = args?.ttl;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TrafficPolicyInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrafficPolicyInstance = TrafficPolicyInstance;
