// @pulumi/azure-native/redis/linkedServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkedServer = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Response to put/get linked server (with properties) for Redis cache.
 *
 * Uses Azure REST API version 2024-11-01.
 *
 * Other available API versions: 2017-10-01, 2018-03-01, 2019-07-01, 2020-06-01, 2020-12-01, 2021-06-01, 2022-05-01, 2022-06-01, 2023-04-01, 2023-05-01-preview, 2023-08-01, 2024-03-01, 2024-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native redis [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class LinkedServer extends pulumi.CustomResource {
    /**
     * Get an existing LinkedServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LinkedServer(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:redis:LinkedServer';
    /**
     * Returns true if the given object is an instance of LinkedServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LinkedServer.__pulumiType;
    }
    /**
     * Create a LinkedServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.linkedRedisCacheId === undefined && !opts.urn) {
                throw new Error("Missing required property 'linkedRedisCacheId'");
            }
            if (args?.linkedRedisCacheLocation === undefined && !opts.urn) {
                throw new Error("Missing required property 'linkedRedisCacheLocation'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.serverRole === undefined && !opts.urn) {
                throw new Error("Missing required property 'serverRole'");
            }
            resourceInputs["linkedRedisCacheId"] = args?.linkedRedisCacheId;
            resourceInputs["linkedRedisCacheLocation"] = args?.linkedRedisCacheLocation;
            resourceInputs["linkedServerName"] = args?.linkedServerName;
            resourceInputs["name"] = args?.name;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["serverRole"] = args?.serverRole;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["geoReplicatedPrimaryHostName"] = undefined /*out*/;
            resourceInputs["primaryHostName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["geoReplicatedPrimaryHostName"] = undefined /*out*/;
            resourceInputs["linkedRedisCacheId"] = undefined /*out*/;
            resourceInputs["linkedRedisCacheLocation"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["primaryHostName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serverRole"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cache/v20230401:LinkedServer" }, { type: "azure-native:cache/v20230501preview:LinkedServer" }, { type: "azure-native:cache/v20230801:LinkedServer" }, { type: "azure-native:cache/v20240301:LinkedServer" }, { type: "azure-native:cache/v20240401preview:LinkedServer" }, { type: "azure-native:cache/v20241101:LinkedServer" }, { type: "azure-native:cache:LinkedServer" }, { type: "azure-native:redis/v20170201:LinkedServer" }, { type: "azure-native:redis/v20171001:LinkedServer" }, { type: "azure-native:redis/v20180301:LinkedServer" }, { type: "azure-native:redis/v20190701:LinkedServer" }, { type: "azure-native:redis/v20200601:LinkedServer" }, { type: "azure-native:redis/v20201201:LinkedServer" }, { type: "azure-native:redis/v20210601:LinkedServer" }, { type: "azure-native:redis/v20220501:LinkedServer" }, { type: "azure-native:redis/v20220601:LinkedServer" }, { type: "azure-native:redis/v20230401:LinkedServer" }, { type: "azure-native:redis/v20230501preview:LinkedServer" }, { type: "azure-native:redis/v20230801:LinkedServer" }, { type: "azure-native:redis/v20240301:LinkedServer" }, { type: "azure-native:redis/v20240401preview:LinkedServer" }, { type: "azure-native:redis/v20241101:LinkedServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LinkedServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LinkedServer = LinkedServer;
