// @pulumi/azure-native/fluidrelay/fluidRelayServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FluidRelayServer = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A FluidRelay Server.
 *
 * Uses Azure REST API version 2022-06-01. In version 2.x of the Azure Native provider, it used API version 2022-06-01.
 *
 * Other available API versions: 2025-06-20-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native fluidrelay [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class FluidRelayServer extends pulumi.CustomResource {
    /**
     * Get an existing FluidRelayServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FluidRelayServer(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:fluidrelay:FluidRelayServer';
    /**
     * Returns true if the given object is an instance of FluidRelayServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FluidRelayServer.__pulumiType;
    }
    /**
     * Create a FluidRelayServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroup === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            resourceInputs["encryption"] = args?.encryption;
            resourceInputs["fluidRelayServerName"] = args?.fluidRelayServerName;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["location"] = args?.location;
            resourceInputs["provisioningState"] = args?.provisioningState;
            resourceInputs["resourceGroup"] = args?.resourceGroup;
            resourceInputs["storagesku"] = args?.storagesku;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["fluidRelayEndpoints"] = undefined /*out*/;
            resourceInputs["frsTenantId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["encryption"] = undefined /*out*/;
            resourceInputs["fluidRelayEndpoints"] = undefined /*out*/;
            resourceInputs["frsTenantId"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["storagesku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:fluidrelay/v20210312preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210615preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210830preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210910preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220215:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220421:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220511:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220526:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220601:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20250620preview:FluidRelayServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FluidRelayServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FluidRelayServer = FluidRelayServer;
