// @pulumi/azure-native/fluidrelay/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listFluidRelayServerKeysOutput = exports.listFluidRelayServerKeys = exports.getFluidRelayServerOutput = exports.getFluidRelayServer = exports.getFluidRelayPrivateEndpointConnectionOutput = exports.getFluidRelayPrivateEndpointConnection = exports.FluidRelayServer = exports.FluidRelayPrivateEndpointConnection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.FluidRelayPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["FluidRelayPrivateEndpointConnection"], () => require("./fluidRelayPrivateEndpointConnection"));
exports.FluidRelayServer = null;
utilities.lazyLoad(exports, ["FluidRelayServer"], () => require("./fluidRelayServer"));
exports.getFluidRelayPrivateEndpointConnection = null;
exports.getFluidRelayPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getFluidRelayPrivateEndpointConnection", "getFluidRelayPrivateEndpointConnectionOutput"], () => require("./getFluidRelayPrivateEndpointConnection"));
exports.getFluidRelayServer = null;
exports.getFluidRelayServerOutput = null;
utilities.lazyLoad(exports, ["getFluidRelayServer", "getFluidRelayServerOutput"], () => require("./getFluidRelayServer"));
exports.listFluidRelayServerKeys = null;
exports.listFluidRelayServerKeysOutput = null;
utilities.lazyLoad(exports, ["listFluidRelayServerKeys", "listFluidRelayServerKeysOutput"], () => require("./listFluidRelayServerKeys"));
// Export enums:
__exportStar(require("../types/enums/fluidrelay"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:fluidrelay:FluidRelayPrivateEndpointConnection":
                return new exports.FluidRelayPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:fluidrelay:FluidRelayServer":
                return new exports.FluidRelayServer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "fluidrelay", _module);
