// @pulumi/aws/directconnect/gatewayAssociationProposal.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayAssociationProposal = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Direct Connect Gateway Association Proposal, typically for enabling cross-account associations. For single account associations, see the `aws.directconnect.GatewayAssociation` resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.directconnect.GatewayAssociationProposal("example", {
 *     dxGatewayId: exampleAwsDxGateway.id,
 *     dxGatewayOwnerAccountId: exampleAwsDxGateway.ownerAccountId,
 *     associatedGatewayId: exampleAwsVpnGateway.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using a proposal ID, Direct Connect Gateway ID and associated gateway ID separated by `/`:
 *
 * **With `pulumi import`**, import Direct Connect Gateway Association Proposals using either a proposal ID or proposal ID, Direct Connect Gateway ID and associated gateway ID separated by `/`. For example:
 *
 * Using a proposal ID:
 *
 * ```sh
 * $ pulumi import aws:directconnect/gatewayAssociationProposal:GatewayAssociationProposal example ac90e981-b718-4364-872d-65478c84fafe
 * ```
 *
 * Using a proposal ID, Direct Connect Gateway ID and associated gateway ID separated by `/`:
 *
 * ```sh
 * $ pulumi import aws:directconnect/gatewayAssociationProposal:GatewayAssociationProposal example ac90e981-b718-4364-872d-65478c84fafe/abcd1234-dcba-5678-be23-cdef9876ab45/vgw-12345678
 * ```
 *
 * The latter case is useful when a previous proposal has been accepted and deleted by AWS.
 * The `aws.directconnect.GatewayAssociationProposal` resource will then represent a pseudo-proposal for the same Direct Connect Gateway and associated gateway. If no previous proposal is available, use a tool like [`uuidgen`](http://manpages.ubuntu.com/manpages/bionic/man1/uuidgen.1.html) to generate a new random pseudo-proposal ID.
 */
class GatewayAssociationProposal extends pulumi.CustomResource {
    /**
     * Get an existing GatewayAssociationProposal resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GatewayAssociationProposal(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:directconnect/gatewayAssociationProposal:GatewayAssociationProposal';
    /**
     * Returns true if the given object is an instance of GatewayAssociationProposal.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GatewayAssociationProposal.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowedPrefixes"] = state?.allowedPrefixes;
            resourceInputs["associatedGatewayId"] = state?.associatedGatewayId;
            resourceInputs["associatedGatewayOwnerAccountId"] = state?.associatedGatewayOwnerAccountId;
            resourceInputs["associatedGatewayType"] = state?.associatedGatewayType;
            resourceInputs["dxGatewayId"] = state?.dxGatewayId;
            resourceInputs["dxGatewayOwnerAccountId"] = state?.dxGatewayOwnerAccountId;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.associatedGatewayId === undefined && !opts.urn) {
                throw new Error("Missing required property 'associatedGatewayId'");
            }
            if (args?.dxGatewayId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dxGatewayId'");
            }
            if (args?.dxGatewayOwnerAccountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dxGatewayOwnerAccountId'");
            }
            resourceInputs["allowedPrefixes"] = args?.allowedPrefixes;
            resourceInputs["associatedGatewayId"] = args?.associatedGatewayId;
            resourceInputs["dxGatewayId"] = args?.dxGatewayId;
            resourceInputs["dxGatewayOwnerAccountId"] = args?.dxGatewayOwnerAccountId;
            resourceInputs["region"] = args?.region;
            resourceInputs["associatedGatewayOwnerAccountId"] = undefined /*out*/;
            resourceInputs["associatedGatewayType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GatewayAssociationProposal.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GatewayAssociationProposal = GatewayAssociationProposal;
