// @pulumi/gcp/observability/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TraceScope = exports.ProjectSettings = exports.OrganizationSettings = exports.getProjectSettingsOutput = exports.getProjectSettings = exports.getOrganizationSettingsOutput = exports.getOrganizationSettings = exports.getFolderSettingsOutput = exports.getFolderSettings = exports.FolderSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.FolderSettings = null;
utilities.lazyLoad(exports, ["FolderSettings"], () => require("./folderSettings"));
exports.getFolderSettings = null;
exports.getFolderSettingsOutput = null;
utilities.lazyLoad(exports, ["getFolderSettings", "getFolderSettingsOutput"], () => require("./getFolderSettings"));
exports.getOrganizationSettings = null;
exports.getOrganizationSettingsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationSettings", "getOrganizationSettingsOutput"], () => require("./getOrganizationSettings"));
exports.getProjectSettings = null;
exports.getProjectSettingsOutput = null;
utilities.lazyLoad(exports, ["getProjectSettings", "getProjectSettingsOutput"], () => require("./getProjectSettings"));
exports.OrganizationSettings = null;
utilities.lazyLoad(exports, ["OrganizationSettings"], () => require("./organizationSettings"));
exports.ProjectSettings = null;
utilities.lazyLoad(exports, ["ProjectSettings"], () => require("./projectSettings"));
exports.TraceScope = null;
utilities.lazyLoad(exports, ["TraceScope"], () => require("./traceScope"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:observability/folderSettings:FolderSettings":
                return new exports.FolderSettings(name, undefined, { urn });
            case "gcp:observability/organizationSettings:OrganizationSettings":
                return new exports.OrganizationSettings(name, undefined, { urn });
            case "gcp:observability/projectSettings:ProjectSettings":
                return new exports.ProjectSettings(name, undefined, { urn });
            case "gcp:observability/traceScope:TraceScope":
                return new exports.TraceScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "observability/folderSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "observability/organizationSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "observability/projectSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "observability/traceScope", _module);
