// @pulumi/aws/appfabric/appAuthorizationConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppAuthorizationConnection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for managing an AWS AppFabric App Authorization Connection.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.appfabric.AppAuthorizationConnection("example", {
 *     appAuthorizationArn: test.arn,
 *     appBundleArn: arn,
 * });
 * ```
 */
class AppAuthorizationConnection extends pulumi.CustomResource {
    /**
     * Get an existing AppAuthorizationConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AppAuthorizationConnection(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:appfabric/appAuthorizationConnection:AppAuthorizationConnection';
    /**
     * Returns true if the given object is an instance of AppAuthorizationConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AppAuthorizationConnection.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["app"] = state?.app;
            resourceInputs["appAuthorizationArn"] = state?.appAuthorizationArn;
            resourceInputs["appBundleArn"] = state?.appBundleArn;
            resourceInputs["authRequest"] = state?.authRequest;
            resourceInputs["region"] = state?.region;
            resourceInputs["tenants"] = state?.tenants;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.appAuthorizationArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'appAuthorizationArn'");
            }
            if (args?.appBundleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'appBundleArn'");
            }
            resourceInputs["appAuthorizationArn"] = args?.appAuthorizationArn;
            resourceInputs["appBundleArn"] = args?.appBundleArn;
            resourceInputs["authRequest"] = args?.authRequest;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["app"] = undefined /*out*/;
            resourceInputs["tenants"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AppAuthorizationConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AppAuthorizationConnection = AppAuthorizationConnection;
