// @pulumi/azure-native/web/webAppInstanceFunctionSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppInstanceFunctionSlot = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Function information.
 *
 * Uses Azure REST API version 2024-11-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2016-08-01, 2018-02-01, 2018-11-01, 2019-08-01, 2020-06-01, 2020-09-01, 2020-10-01, 2020-12-01, 2021-01-01, 2021-01-15, 2021-02-01, 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2023-12-01, 2024-04-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native web [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppInstanceFunctionSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppInstanceFunctionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppInstanceFunctionSlot(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:web:WebAppInstanceFunctionSlot';
    /**
     * Returns true if the given object is an instance of WebAppInstanceFunctionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppInstanceFunctionSlot.__pulumiType;
    }
    /**
     * Create a WebAppInstanceFunctionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.slot === undefined && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["config"] = args?.config;
            resourceInputs["configHref"] = args?.configHref;
            resourceInputs["files"] = args?.files;
            resourceInputs["functionAppId"] = args?.functionAppId;
            resourceInputs["functionName"] = args?.functionName;
            resourceInputs["href"] = args?.href;
            resourceInputs["invokeUrlTemplate"] = args?.invokeUrlTemplate;
            resourceInputs["isDisabled"] = args?.isDisabled;
            resourceInputs["kind"] = args?.kind;
            resourceInputs["language"] = args?.language;
            resourceInputs["name"] = args?.name;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["scriptHref"] = args?.scriptHref;
            resourceInputs["scriptRootPathHref"] = args?.scriptRootPathHref;
            resourceInputs["secretsFileHref"] = args?.secretsFileHref;
            resourceInputs["slot"] = args?.slot;
            resourceInputs["testData"] = args?.testData;
            resourceInputs["testDataHref"] = args?.testDataHref;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["config"] = undefined /*out*/;
            resourceInputs["configHref"] = undefined /*out*/;
            resourceInputs["files"] = undefined /*out*/;
            resourceInputs["functionAppId"] = undefined /*out*/;
            resourceInputs["href"] = undefined /*out*/;
            resourceInputs["invokeUrlTemplate"] = undefined /*out*/;
            resourceInputs["isDisabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["language"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scriptHref"] = undefined /*out*/;
            resourceInputs["scriptRootPathHref"] = undefined /*out*/;
            resourceInputs["secretsFileHref"] = undefined /*out*/;
            resourceInputs["testData"] = undefined /*out*/;
            resourceInputs["testDataHref"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20160801:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20180201:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20181101:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20190801:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20200601:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20200901:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20201001:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20201201:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20210101:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20210115:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20210201:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20210301:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20220301:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20220901:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20230101:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20231201:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20240401:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20241101:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20250301:WebAppInstanceFunctionSlot" }, { type: "azure-native:web/v20250501:WebAppInstanceFunctionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppInstanceFunctionSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppInstanceFunctionSlot = WebAppInstanceFunctionSlot;
