// @pulumi/azure-native/web/webAppAuthSettingsV2WithoutSecretsSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAuthSettingsV2WithoutSecretsSlot = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Configuration settings for the Azure App Service Authentication / Authorization V2 feature.
 *
 * Uses Azure REST API version 2024-11-01.
 *
 * Other available API versions: 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2023-12-01, 2024-04-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native web [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppAuthSettingsV2WithoutSecretsSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppAuthSettingsV2WithoutSecretsSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppAuthSettingsV2WithoutSecretsSlot(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:web:WebAppAuthSettingsV2WithoutSecretsSlot';
    /**
     * Returns true if the given object is an instance of WebAppAuthSettingsV2WithoutSecretsSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppAuthSettingsV2WithoutSecretsSlot.__pulumiType;
    }
    /**
     * Create a WebAppAuthSettingsV2WithoutSecretsSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.slot === undefined && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["globalValidation"] = args?.globalValidation;
            resourceInputs["httpSettings"] = args?.httpSettings;
            resourceInputs["identityProviders"] = args?.identityProviders;
            resourceInputs["kind"] = args?.kind;
            resourceInputs["login"] = args?.login;
            resourceInputs["name"] = args?.name;
            resourceInputs["platform"] = args?.platform;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["slot"] = args?.slot;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["globalValidation"] = undefined /*out*/;
            resourceInputs["httpSettings"] = undefined /*out*/;
            resourceInputs["identityProviders"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["login"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20200601:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20200901:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20201001:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20201001:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20201201:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20210101:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20210115:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20210201:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20210201:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20210301:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20220301:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20220901:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20230101:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20231201:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20240401:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20241101:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20250301:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web/v20250501:WebAppAuthSettingsV2WithoutSecretsSlot" }, { type: "azure-native:web:WebAppAuthSettingsV2Slot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppAuthSettingsV2WithoutSecretsSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppAuthSettingsV2WithoutSecretsSlot = WebAppAuthSettingsV2WithoutSecretsSlot;
