// @pulumi/gcp/firebase/aiLogicPromptTemplateLock.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiLogicPromptTemplateLock = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A resource that manages the lock state of a PromptTemplate.
 * When this resource is created, the template is locked.
 * When this resource is deleted, the template is unlocked.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * ## Example Usage
 *
 * ### Firebaseailogic Prompt Template Lock Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const basic = new gcp.firebase.AiLogicPromptTemplate("basic", {
 *     location: "global",
 *     templateId: "lock-template",
 *     templateString: `---
 * model: googleai/gemini-1.5-flash
 * ---
 * Hello World
 * `,
 * });
 * const basicLock = new gcp.firebase.AiLogicPromptTemplateLock("basic_lock", {
 *     location: basic.location,
 *     templateId: basic.templateId,
 * });
 * ```
 *
 * ## Import
 *
 * PromptTemplateLock can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/templates/{{template_id}}`
 * * `{{project}}/{{location}}/{{template_id}}`
 * * `{{location}}/{{template_id}}`
 *
 * When using the `pulumi import` command, PromptTemplateLock can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:firebase/aiLogicPromptTemplateLock:AiLogicPromptTemplateLock default projects/{{project}}/locations/{{location}}/templates/{{template_id}}
 * $ pulumi import gcp:firebase/aiLogicPromptTemplateLock:AiLogicPromptTemplateLock default {{project}}/{{location}}/{{template_id}}
 * $ pulumi import gcp:firebase/aiLogicPromptTemplateLock:AiLogicPromptTemplateLock default {{location}}/{{template_id}}
 * ```
 */
class AiLogicPromptTemplateLock extends pulumi.CustomResource {
    /**
     * Get an existing AiLogicPromptTemplateLock resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AiLogicPromptTemplateLock(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:firebase/aiLogicPromptTemplateLock:AiLogicPromptTemplateLock';
    /**
     * Returns true if the given object is an instance of AiLogicPromptTemplateLock.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AiLogicPromptTemplateLock.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["location"] = state?.location;
            resourceInputs["locked"] = state?.locked;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["templateId"] = state?.templateId;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.templateId === undefined && !opts.urn) {
                throw new Error("Missing required property 'templateId'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["templateId"] = args?.templateId;
            resourceInputs["locked"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AiLogicPromptTemplateLock.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AiLogicPromptTemplateLock = AiLogicPromptTemplateLock;
