// @pulumi/gcp/firebase/hostingSite.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostingSite = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A `Site` represents a Firebase Hosting site.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about Site, see:
 *
 * * [API documentation](https://firebase.google.com/docs/reference/hosting/rest/v1beta1/projects.sites)
 * * How-to Guides
 *     * [Official Documentation](https://firebase.google.com/docs/hosting)
 *
 * ## Example Usage
 *
 * ### Firebasehosting Site Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.firebase.HostingSite("default", {
 *     project: "my-project-name",
 *     siteId: "site-no-app",
 * });
 * ```
 * ### Firebasehosting Site Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.firebase.WebApp("default", {
 *     project: "my-project-name",
 *     displayName: "Test web app for Firebase Hosting",
 * });
 * const full = new gcp.firebase.HostingSite("full", {
 *     project: "my-project-name",
 *     siteId: "site-with-app",
 *     appId: _default.appId,
 * });
 * ```
 * ### Firebasehosting Site Default
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.firebase.HostingSite("default", {
 *     project: "my-project-name",
 *     siteId: "my-project-name",
 * });
 * ```
 *
 * ## Import
 *
 * Site can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/sites/{{site_id}}`
 * * `{{project}}/{{site_id}}`
 * * `sites/{{site_id}}`
 * * `{{site_id}}`
 *
 * When using the `pulumi import` command, Site can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:firebase/hostingSite:HostingSite default projects/{{project}}/sites/{{site_id}}
 * $ pulumi import gcp:firebase/hostingSite:HostingSite default {{project}}/{{site_id}}
 * $ pulumi import gcp:firebase/hostingSite:HostingSite default sites/{{site_id}}
 * $ pulumi import gcp:firebase/hostingSite:HostingSite default {{site_id}}
 * ```
 */
class HostingSite extends pulumi.CustomResource {
    /**
     * Get an existing HostingSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HostingSite(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:firebase/hostingSite:HostingSite';
    /**
     * Returns true if the given object is an instance of HostingSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HostingSite.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appId"] = state?.appId;
            resourceInputs["defaultUrl"] = state?.defaultUrl;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["siteId"] = state?.siteId;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            resourceInputs["appId"] = args?.appId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["project"] = args?.project;
            resourceInputs["siteId"] = args?.siteId;
            resourceInputs["defaultUrl"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HostingSite.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HostingSite = HostingSite;
