// @pulumi/gcp/firebase/storageDefaultBucket.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageDefaultBucket = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A resource that manages the creation of the default Google Cloud Storage bucket
 * for a Firebase project.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about DefaultBucket, see:
 *
 * * [API documentation](https://firebase.google.com/docs/reference/rest/storage/rest/v1alpha/projects.defaultBucket)
 * * How-to Guides
 *     * [Official Documentation](https://firebase.google.com/docs/storage/)
 *
 * ## Example Usage
 *
 * ### Firebasestorage Default Bucket Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.firebase.StorageDefaultBucket("default", {
 *     project: "my-project-name",
 *     location: "us-central1",
 * });
 * ```
 *
 * ## Import
 *
 * DefaultBucket can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/defaultBucket`
 * * `{{project}}`
 *
 * When using the `pulumi import` command, DefaultBucket can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:firebase/storageDefaultBucket:StorageDefaultBucket default projects/{{project}}/defaultBucket
 * $ pulumi import gcp:firebase/storageDefaultBucket:StorageDefaultBucket default {{project}}
 * ```
 */
class StorageDefaultBucket extends pulumi.CustomResource {
    /**
     * Get an existing StorageDefaultBucket resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StorageDefaultBucket(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:firebase/storageDefaultBucket:StorageDefaultBucket';
    /**
     * Returns true if the given object is an instance of StorageDefaultBucket.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageDefaultBucket.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["buckets"] = state?.buckets;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["buckets"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StorageDefaultBucket.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StorageDefaultBucket = StorageDefaultBucket;
