// @pulumi/gcp/oracledatabase/cloudExadataInfrastructure.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudExadataInfrastructure = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A CloudExadataInfrastructure resource.
 *
 * To get more information about CloudExadataInfrastructure, see:
 *
 * * [API documentation](https://cloud.google.com/oracle/database/docs/reference/rest/v1/projects.locations.cloudExadataInfrastructures)
 * * How-to Guides
 *     * [Create Exadata Infrastructure instances](https://cloud.google.com/oracle/database/docs/create-instances)
 *
 * ## Example Usage
 *
 * ### Oracledatabase Cloud Exadata Infrastructure Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const my_cloud_exadata = new gcp.oracledatabase.CloudExadataInfrastructure("my-cloud-exadata", {
 *     cloudExadataInfrastructureId: "my-instance",
 *     displayName: "my-instance displayname",
 *     location: "us-east4",
 *     project: "my-project",
 *     properties: {
 *         shape: "Exadata.X9M",
 *         computeCount: 2,
 *         storageCount: 3,
 *     },
 *     deletionProtection: true,
 * });
 * ```
 * ### Oracledatabase Cloud Exadata Infrastructure Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const my_cloud_exadata = new gcp.oracledatabase.CloudExadataInfrastructure("my-cloud-exadata", {
 *     cloudExadataInfrastructureId: "my-instance",
 *     displayName: "my-instance displayname",
 *     location: "us-east4",
 *     project: "my-project",
 *     gcpOracleZone: "us-east4-b-r1",
 *     properties: {
 *         shape: "Exadata.X9M",
 *         computeCount: 2,
 *         storageCount: 3,
 *         customerContacts: [{
 *             email: "xyz@example.com",
 *         }],
 *         maintenanceWindow: {
 *             customActionTimeoutMins: 20,
 *             daysOfWeeks: ["SUNDAY"],
 *             hoursOfDays: [4],
 *             isCustomActionTimeoutEnabled: false,
 *             leadTimeWeek: 1,
 *             months: [
 *                 "JANUARY",
 *                 "APRIL",
 *                 "MAY",
 *                 "OCTOBER",
 *             ],
 *             patchingMode: "ROLLING",
 *             preference: "CUSTOM_PREFERENCE",
 *             weeksOfMonths: [4],
 *         },
 *         totalStorageSizeGb: 196608,
 *     },
 *     labels: {
 *         "label-one": "value-one",
 *     },
 *     deletionProtection: true,
 * });
 * ```
 *
 * ## Import
 *
 * CloudExadataInfrastructure can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/cloudExadataInfrastructures/{{cloud_exadata_infrastructure_id}}`
 * * `{{project}}/{{location}}/{{cloud_exadata_infrastructure_id}}`
 * * `{{location}}/{{cloud_exadata_infrastructure_id}}`
 *
 * When using the `pulumi import` command, CloudExadataInfrastructure can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:oracledatabase/cloudExadataInfrastructure:CloudExadataInfrastructure default projects/{{project}}/locations/{{location}}/cloudExadataInfrastructures/{{cloud_exadata_infrastructure_id}}
 * $ pulumi import gcp:oracledatabase/cloudExadataInfrastructure:CloudExadataInfrastructure default {{project}}/{{location}}/{{cloud_exadata_infrastructure_id}}
 * $ pulumi import gcp:oracledatabase/cloudExadataInfrastructure:CloudExadataInfrastructure default {{location}}/{{cloud_exadata_infrastructure_id}}
 * ```
 */
class CloudExadataInfrastructure extends pulumi.CustomResource {
    /**
     * Get an existing CloudExadataInfrastructure resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CloudExadataInfrastructure(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:oracledatabase/cloudExadataInfrastructure:CloudExadataInfrastructure';
    /**
     * Returns true if the given object is an instance of CloudExadataInfrastructure.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudExadataInfrastructure.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cloudExadataInfrastructureId"] = state?.cloudExadataInfrastructureId;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["deletionProtection"] = state?.deletionProtection;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["entitlementId"] = state?.entitlementId;
            resourceInputs["gcpOracleZone"] = state?.gcpOracleZone;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["properties"] = state?.properties;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
        }
        else {
            const args = argsOrState;
            if (args?.cloudExadataInfrastructureId === undefined && !opts.urn) {
                throw new Error("Missing required property 'cloudExadataInfrastructureId'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["cloudExadataInfrastructureId"] = args?.cloudExadataInfrastructureId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["deletionProtection"] = args?.deletionProtection;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["gcpOracleZone"] = args?.gcpOracleZone;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["entitlementId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(CloudExadataInfrastructure.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CloudExadataInfrastructure = CloudExadataInfrastructure;
