// @pulumi/gcp/oracledatabase/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdbSubnet = exports.OdbNetwork = exports.getOdbSubnetOutput = exports.getOdbSubnet = exports.getOdbNetworkOutput = exports.getOdbNetwork = exports.getDbServersOutput = exports.getDbServers = exports.getDbNodesOutput = exports.getDbNodes = exports.getCloudVmClustersOutput = exports.getCloudVmClusters = exports.getCloudVmClusterOutput = exports.getCloudVmCluster = exports.getCloudExadataInfrastructuresOutput = exports.getCloudExadataInfrastructures = exports.getCloudExadataInfrastructureOutput = exports.getCloudExadataInfrastructure = exports.getAutonomousDatabasesOutput = exports.getAutonomousDatabases = exports.getAutonomousDatabaseOutput = exports.getAutonomousDatabase = exports.ExascaleDbStorageVault = exports.ExadbVmCluster = exports.DbSystem = exports.CloudVmCluster = exports.CloudExadataInfrastructure = exports.AutonomousDatabase = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AutonomousDatabase = null;
utilities.lazyLoad(exports, ["AutonomousDatabase"], () => require("./autonomousDatabase"));
exports.CloudExadataInfrastructure = null;
utilities.lazyLoad(exports, ["CloudExadataInfrastructure"], () => require("./cloudExadataInfrastructure"));
exports.CloudVmCluster = null;
utilities.lazyLoad(exports, ["CloudVmCluster"], () => require("./cloudVmCluster"));
exports.DbSystem = null;
utilities.lazyLoad(exports, ["DbSystem"], () => require("./dbSystem"));
exports.ExadbVmCluster = null;
utilities.lazyLoad(exports, ["ExadbVmCluster"], () => require("./exadbVmCluster"));
exports.ExascaleDbStorageVault = null;
utilities.lazyLoad(exports, ["ExascaleDbStorageVault"], () => require("./exascaleDbStorageVault"));
exports.getAutonomousDatabase = null;
exports.getAutonomousDatabaseOutput = null;
utilities.lazyLoad(exports, ["getAutonomousDatabase", "getAutonomousDatabaseOutput"], () => require("./getAutonomousDatabase"));
exports.getAutonomousDatabases = null;
exports.getAutonomousDatabasesOutput = null;
utilities.lazyLoad(exports, ["getAutonomousDatabases", "getAutonomousDatabasesOutput"], () => require("./getAutonomousDatabases"));
exports.getCloudExadataInfrastructure = null;
exports.getCloudExadataInfrastructureOutput = null;
utilities.lazyLoad(exports, ["getCloudExadataInfrastructure", "getCloudExadataInfrastructureOutput"], () => require("./getCloudExadataInfrastructure"));
exports.getCloudExadataInfrastructures = null;
exports.getCloudExadataInfrastructuresOutput = null;
utilities.lazyLoad(exports, ["getCloudExadataInfrastructures", "getCloudExadataInfrastructuresOutput"], () => require("./getCloudExadataInfrastructures"));
exports.getCloudVmCluster = null;
exports.getCloudVmClusterOutput = null;
utilities.lazyLoad(exports, ["getCloudVmCluster", "getCloudVmClusterOutput"], () => require("./getCloudVmCluster"));
exports.getCloudVmClusters = null;
exports.getCloudVmClustersOutput = null;
utilities.lazyLoad(exports, ["getCloudVmClusters", "getCloudVmClustersOutput"], () => require("./getCloudVmClusters"));
exports.getDbNodes = null;
exports.getDbNodesOutput = null;
utilities.lazyLoad(exports, ["getDbNodes", "getDbNodesOutput"], () => require("./getDbNodes"));
exports.getDbServers = null;
exports.getDbServersOutput = null;
utilities.lazyLoad(exports, ["getDbServers", "getDbServersOutput"], () => require("./getDbServers"));
exports.getOdbNetwork = null;
exports.getOdbNetworkOutput = null;
utilities.lazyLoad(exports, ["getOdbNetwork", "getOdbNetworkOutput"], () => require("./getOdbNetwork"));
exports.getOdbSubnet = null;
exports.getOdbSubnetOutput = null;
utilities.lazyLoad(exports, ["getOdbSubnet", "getOdbSubnetOutput"], () => require("./getOdbSubnet"));
exports.OdbNetwork = null;
utilities.lazyLoad(exports, ["OdbNetwork"], () => require("./odbNetwork"));
exports.OdbSubnet = null;
utilities.lazyLoad(exports, ["OdbSubnet"], () => require("./odbSubnet"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:oracledatabase/autonomousDatabase:AutonomousDatabase":
                return new exports.AutonomousDatabase(name, undefined, { urn });
            case "gcp:oracledatabase/cloudExadataInfrastructure:CloudExadataInfrastructure":
                return new exports.CloudExadataInfrastructure(name, undefined, { urn });
            case "gcp:oracledatabase/cloudVmCluster:CloudVmCluster":
                return new exports.CloudVmCluster(name, undefined, { urn });
            case "gcp:oracledatabase/dbSystem:DbSystem":
                return new exports.DbSystem(name, undefined, { urn });
            case "gcp:oracledatabase/exadbVmCluster:ExadbVmCluster":
                return new exports.ExadbVmCluster(name, undefined, { urn });
            case "gcp:oracledatabase/exascaleDbStorageVault:ExascaleDbStorageVault":
                return new exports.ExascaleDbStorageVault(name, undefined, { urn });
            case "gcp:oracledatabase/odbNetwork:OdbNetwork":
                return new exports.OdbNetwork(name, undefined, { urn });
            case "gcp:oracledatabase/odbSubnet:OdbSubnet":
                return new exports.OdbSubnet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/autonomousDatabase", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/cloudExadataInfrastructure", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/cloudVmCluster", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/dbSystem", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/exadbVmCluster", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/exascaleDbStorageVault", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/odbNetwork", _module);
pulumi.runtime.registerResourceModule("gcp", "oracledatabase/odbSubnet", _module);
