// @pulumi/gcp/oracledatabase/odbSubnet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdbSubnet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An OdbSubnet resource which represents a subnet under an OdbNetwork.
 *
 * To get more information about OdbSubnet, see:
 * * How-to Guides
 *     * [OracleDatabase@Google Cloud](https://cloud.google.com/oracle/database/docs/overview')
 *
 * ## Example Usage
 *
 * ### Oracledatabase Odbsubnet
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const my_odbsubnet = new gcp.oracledatabase.OdbSubnet("my-odbsubnet", {
 *     odbSubnetId: "my-odbsubnet",
 *     location: "europe-west2",
 *     project: "my-project",
 *     odbnetwork: "my-odbnetwork",
 *     cidrRange: "10.1.1.0/24",
 *     purpose: "CLIENT_SUBNET",
 *     labels: {
 *         terraform_created: "true",
 *     },
 *     deletionProtection: true,
 * });
 * ```
 *
 * ## Import
 *
 * OdbSubnet can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/odbNetworks/{{odbnetwork}}/odbSubnets/{{odb_subnet_id}}`
 * * `{{project}}/{{location}}/{{odbnetwork}}/{{odb_subnet_id}}`
 * * `{{location}}/{{odbnetwork}}/{{odb_subnet_id}}`
 *
 * When using the `pulumi import` command, OdbSubnet can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:oracledatabase/odbSubnet:OdbSubnet default projects/{{project}}/locations/{{location}}/odbNetworks/{{odbnetwork}}/odbSubnets/{{odb_subnet_id}}
 * $ pulumi import gcp:oracledatabase/odbSubnet:OdbSubnet default {{project}}/{{location}}/{{odbnetwork}}/{{odb_subnet_id}}
 * $ pulumi import gcp:oracledatabase/odbSubnet:OdbSubnet default {{location}}/{{odbnetwork}}/{{odb_subnet_id}}
 * ```
 */
class OdbSubnet extends pulumi.CustomResource {
    /**
     * Get an existing OdbSubnet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OdbSubnet(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:oracledatabase/odbSubnet:OdbSubnet';
    /**
     * Returns true if the given object is an instance of OdbSubnet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OdbSubnet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cidrRange"] = state?.cidrRange;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["deletionProtection"] = state?.deletionProtection;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["odbSubnetId"] = state?.odbSubnetId;
            resourceInputs["odbnetwork"] = state?.odbnetwork;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["purpose"] = state?.purpose;
            resourceInputs["state"] = state?.state;
        }
        else {
            const args = argsOrState;
            if (args?.cidrRange === undefined && !opts.urn) {
                throw new Error("Missing required property 'cidrRange'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.odbSubnetId === undefined && !opts.urn) {
                throw new Error("Missing required property 'odbSubnetId'");
            }
            if (args?.odbnetwork === undefined && !opts.urn) {
                throw new Error("Missing required property 'odbnetwork'");
            }
            if (args?.purpose === undefined && !opts.urn) {
                throw new Error("Missing required property 'purpose'");
            }
            resourceInputs["cidrRange"] = args?.cidrRange;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["deletionProtection"] = args?.deletionProtection;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["odbSubnetId"] = args?.odbSubnetId;
            resourceInputs["odbnetwork"] = args?.odbnetwork;
            resourceInputs["project"] = args?.project;
            resourceInputs["purpose"] = args?.purpose;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(OdbSubnet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OdbSubnet = OdbSubnet;
