// @pulumi/aws/notifications/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationsAccess = exports.OrganizationalUnitAssociation = exports.NotificationHub = exports.NotificationConfiguration = exports.ManagedNotificationAdditionalChannelAssociation = exports.ManagedNotificationAccountContactAssociation = exports.EventRule = exports.ContactsEmailContact = exports.ChannelAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ChannelAssociation = null;
utilities.lazyLoad(exports, ["ChannelAssociation"], () => require("./channelAssociation"));
exports.ContactsEmailContact = null;
utilities.lazyLoad(exports, ["ContactsEmailContact"], () => require("./contactsEmailContact"));
exports.EventRule = null;
utilities.lazyLoad(exports, ["EventRule"], () => require("./eventRule"));
exports.ManagedNotificationAccountContactAssociation = null;
utilities.lazyLoad(exports, ["ManagedNotificationAccountContactAssociation"], () => require("./managedNotificationAccountContactAssociation"));
exports.ManagedNotificationAdditionalChannelAssociation = null;
utilities.lazyLoad(exports, ["ManagedNotificationAdditionalChannelAssociation"], () => require("./managedNotificationAdditionalChannelAssociation"));
exports.NotificationConfiguration = null;
utilities.lazyLoad(exports, ["NotificationConfiguration"], () => require("./notificationConfiguration"));
exports.NotificationHub = null;
utilities.lazyLoad(exports, ["NotificationHub"], () => require("./notificationHub"));
exports.OrganizationalUnitAssociation = null;
utilities.lazyLoad(exports, ["OrganizationalUnitAssociation"], () => require("./organizationalUnitAssociation"));
exports.OrganizationsAccess = null;
utilities.lazyLoad(exports, ["OrganizationsAccess"], () => require("./organizationsAccess"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:notifications/channelAssociation:ChannelAssociation":
                return new exports.ChannelAssociation(name, undefined, { urn });
            case "aws:notifications/contactsEmailContact:ContactsEmailContact":
                return new exports.ContactsEmailContact(name, undefined, { urn });
            case "aws:notifications/eventRule:EventRule":
                return new exports.EventRule(name, undefined, { urn });
            case "aws:notifications/managedNotificationAccountContactAssociation:ManagedNotificationAccountContactAssociation":
                return new exports.ManagedNotificationAccountContactAssociation(name, undefined, { urn });
            case "aws:notifications/managedNotificationAdditionalChannelAssociation:ManagedNotificationAdditionalChannelAssociation":
                return new exports.ManagedNotificationAdditionalChannelAssociation(name, undefined, { urn });
            case "aws:notifications/notificationConfiguration:NotificationConfiguration":
                return new exports.NotificationConfiguration(name, undefined, { urn });
            case "aws:notifications/notificationHub:NotificationHub":
                return new exports.NotificationHub(name, undefined, { urn });
            case "aws:notifications/organizationalUnitAssociation:OrganizationalUnitAssociation":
                return new exports.OrganizationalUnitAssociation(name, undefined, { urn });
            case "aws:notifications/organizationsAccess:OrganizationsAccess":
                return new exports.OrganizationsAccess(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "notifications/channelAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/contactsEmailContact", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/eventRule", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/managedNotificationAccountContactAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/managedNotificationAdditionalChannelAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/notificationConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/notificationHub", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/organizationalUnitAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "notifications/organizationsAccess", _module);
