// @pulumi/aws/verifiedaccess/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustProvider = exports.InstanceTrustProviderAttachment = exports.InstanceLoggingConfiguration = exports.Instance = exports.Group = exports.Endpoint = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceLoggingConfiguration = null;
utilities.lazyLoad(exports, ["InstanceLoggingConfiguration"], () => require("./instanceLoggingConfiguration"));
exports.InstanceTrustProviderAttachment = null;
utilities.lazyLoad(exports, ["InstanceTrustProviderAttachment"], () => require("./instanceTrustProviderAttachment"));
exports.TrustProvider = null;
utilities.lazyLoad(exports, ["TrustProvider"], () => require("./trustProvider"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:verifiedaccess/endpoint:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "aws:verifiedaccess/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws:verifiedaccess/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws:verifiedaccess/instanceLoggingConfiguration:InstanceLoggingConfiguration":
                return new exports.InstanceLoggingConfiguration(name, undefined, { urn });
            case "aws:verifiedaccess/instanceTrustProviderAttachment:InstanceTrustProviderAttachment":
                return new exports.InstanceTrustProviderAttachment(name, undefined, { urn });
            case "aws:verifiedaccess/trustProvider:TrustProvider":
                return new exports.TrustProvider(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/endpoint", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/group", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/instance", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/instanceLoggingConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/instanceTrustProviderAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedaccess/trustProvider", _module);
