// @pulumi/gcp/cloudbuildv2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = exports.getConnectionIamPolicyOutput = exports.getConnectionIamPolicy = exports.ConnectionIAMPolicy = exports.ConnectionIAMMember = exports.ConnectionIAMBinding = exports.Connection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionIAMBinding = null;
utilities.lazyLoad(exports, ["ConnectionIAMBinding"], () => require("./connectionIAMBinding"));
exports.ConnectionIAMMember = null;
utilities.lazyLoad(exports, ["ConnectionIAMMember"], () => require("./connectionIAMMember"));
exports.ConnectionIAMPolicy = null;
utilities.lazyLoad(exports, ["ConnectionIAMPolicy"], () => require("./connectionIAMPolicy"));
exports.getConnectionIamPolicy = null;
exports.getConnectionIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getConnectionIamPolicy", "getConnectionIamPolicyOutput"], () => require("./getConnectionIamPolicy"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudbuildv2/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMBinding:ConnectionIAMBinding":
                return new exports.ConnectionIAMBinding(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMMember:ConnectionIAMMember":
                return new exports.ConnectionIAMMember(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMPolicy:ConnectionIAMPolicy":
                return new exports.ConnectionIAMPolicy(name, undefined, { urn });
            case "gcp:cloudbuildv2/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connection", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/repository", _module);
