// @pulumi/aws/finspace/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KxVolume = exports.KxUser = exports.KxScalingGroup = exports.KxEnvironment = exports.KxDataview = exports.KxDatabase = exports.KxCluster = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.KxCluster = null;
utilities.lazyLoad(exports, ["KxCluster"], () => require("./kxCluster"));
exports.KxDatabase = null;
utilities.lazyLoad(exports, ["KxDatabase"], () => require("./kxDatabase"));
exports.KxDataview = null;
utilities.lazyLoad(exports, ["KxDataview"], () => require("./kxDataview"));
exports.KxEnvironment = null;
utilities.lazyLoad(exports, ["KxEnvironment"], () => require("./kxEnvironment"));
exports.KxScalingGroup = null;
utilities.lazyLoad(exports, ["KxScalingGroup"], () => require("./kxScalingGroup"));
exports.KxUser = null;
utilities.lazyLoad(exports, ["KxUser"], () => require("./kxUser"));
exports.KxVolume = null;
utilities.lazyLoad(exports, ["KxVolume"], () => require("./kxVolume"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:finspace/kxCluster:KxCluster":
                return new exports.KxCluster(name, undefined, { urn });
            case "aws:finspace/kxDatabase:KxDatabase":
                return new exports.KxDatabase(name, undefined, { urn });
            case "aws:finspace/kxDataview:KxDataview":
                return new exports.KxDataview(name, undefined, { urn });
            case "aws:finspace/kxEnvironment:KxEnvironment":
                return new exports.KxEnvironment(name, undefined, { urn });
            case "aws:finspace/kxScalingGroup:KxScalingGroup":
                return new exports.KxScalingGroup(name, undefined, { urn });
            case "aws:finspace/kxUser:KxUser":
                return new exports.KxUser(name, undefined, { urn });
            case "aws:finspace/kxVolume:KxVolume":
                return new exports.KxVolume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "finspace/kxCluster", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxDatabase", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxDataview", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxEnvironment", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxScalingGroup", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxUser", _module);
pulumi.runtime.registerResourceModule("aws", "finspace/kxVolume", _module);
