// @pulumi/aws/servicecatalog/tagOptionResourceAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagOptionResourceAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Service Catalog Tag Option Resource Association.
 *
 * > **Tip:** A "resource" is either a Service Catalog portfolio or product.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.servicecatalog.TagOptionResourceAssociation("example", {
 *     resourceId: "prod-dnigbtea24ste",
 *     tagOptionId: "tag-pjtvyakdlyo3m",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws.servicecatalog.TagOptionResourceAssociation` using the tag option ID and resource ID. For example:
 *
 * ```sh
 * $ pulumi import aws:servicecatalog/tagOptionResourceAssociation:TagOptionResourceAssociation example tag-pjtvyakdlyo3m:prod-dnigbtea24ste
 * ```
 */
class TagOptionResourceAssociation extends pulumi.CustomResource {
    /**
     * Get an existing TagOptionResourceAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TagOptionResourceAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:servicecatalog/tagOptionResourceAssociation:TagOptionResourceAssociation';
    /**
     * Returns true if the given object is an instance of TagOptionResourceAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TagOptionResourceAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceArn"] = state?.resourceArn;
            resourceInputs["resourceCreatedTime"] = state?.resourceCreatedTime;
            resourceInputs["resourceDescription"] = state?.resourceDescription;
            resourceInputs["resourceId"] = state?.resourceId;
            resourceInputs["resourceName"] = state?.resourceName;
            resourceInputs["tagOptionId"] = state?.tagOptionId;
        }
        else {
            const args = argsOrState;
            if (args?.resourceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceId'");
            }
            if (args?.tagOptionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'tagOptionId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceId"] = args?.resourceId;
            resourceInputs["tagOptionId"] = args?.tagOptionId;
            resourceInputs["resourceArn"] = undefined /*out*/;
            resourceInputs["resourceCreatedTime"] = undefined /*out*/;
            resourceInputs["resourceDescription"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TagOptionResourceAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TagOptionResourceAssociation = TagOptionResourceAssociation;
