// @pulumi/gcp/folder/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceIdentity = exports.OrganizationPolicy = exports.IAMPolicy = exports.IAMMember = exports.IAMBinding = exports.IamAuditConfig = exports.getOrganizationPolicyOutput = exports.getOrganizationPolicy = exports.getIamPolicyOutput = exports.getIamPolicy = exports.AccessApprovalSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccessApprovalSettings = null;
utilities.lazyLoad(exports, ["AccessApprovalSettings"], () => require("./accessApprovalSettings"));
exports.getIamPolicy = null;
exports.getIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getIamPolicy", "getIamPolicyOutput"], () => require("./getIamPolicy"));
exports.getOrganizationPolicy = null;
exports.getOrganizationPolicyOutput = null;
utilities.lazyLoad(exports, ["getOrganizationPolicy", "getOrganizationPolicyOutput"], () => require("./getOrganizationPolicy"));
exports.IamAuditConfig = null;
utilities.lazyLoad(exports, ["IamAuditConfig"], () => require("./iamAuditConfig"));
exports.IAMBinding = null;
utilities.lazyLoad(exports, ["IAMBinding"], () => require("./iambinding"));
exports.IAMMember = null;
utilities.lazyLoad(exports, ["IAMMember"], () => require("./iammember"));
exports.IAMPolicy = null;
utilities.lazyLoad(exports, ["IAMPolicy"], () => require("./iampolicy"));
exports.OrganizationPolicy = null;
utilities.lazyLoad(exports, ["OrganizationPolicy"], () => require("./organizationPolicy"));
exports.ServiceIdentity = null;
utilities.lazyLoad(exports, ["ServiceIdentity"], () => require("./serviceIdentity"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:folder/accessApprovalSettings:AccessApprovalSettings":
                return new exports.AccessApprovalSettings(name, undefined, { urn });
            case "gcp:folder/iAMBinding:IAMBinding":
                return new exports.IAMBinding(name, undefined, { urn });
            case "gcp:folder/iAMMember:IAMMember":
                return new exports.IAMMember(name, undefined, { urn });
            case "gcp:folder/iAMPolicy:IAMPolicy":
                return new exports.IAMPolicy(name, undefined, { urn });
            case "gcp:folder/iamAuditConfig:IamAuditConfig":
                return new exports.IamAuditConfig(name, undefined, { urn });
            case "gcp:folder/organizationPolicy:OrganizationPolicy":
                return new exports.OrganizationPolicy(name, undefined, { urn });
            case "gcp:folder/serviceIdentity:ServiceIdentity":
                return new exports.ServiceIdentity(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "folder/accessApprovalSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/iAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/iAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/iAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/iamAuditConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/organizationPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "folder/serviceIdentity", _module);
