// @pulumi/gcp/folder/serviceIdentity.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceIdentity = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Generate folder service identity for a service.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * > **Note:** Once created, this resource cannot be updated or destroyed. These
 * actions are a no-op.
 *
 * > **Note:** This resource can be used to retrieve the emails of the [Google-managed folder service accounts](https://cloud.google.com/iam/docs/service-agents)
 * of the APIs that Google has configured with a Service Identity. You can run `gcloud beta services identity create --service SERVICE_NAME.googleapis.com --folder FOLDER` to
 * verify if an API supports this.
 *
 * To get more information about Service Identity, see:
 *
 * * [API documentation](https://cloud.google.com/service-usage/docs/reference/rest/v1beta1/services/generateServiceIdentity)
 *
 * ## Example Usage
 *
 * ### Folder Service Identity Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const myFolder = new gcp.organizations.Folder("my_folder", {
 *     parent: "organizations/1234567",
 *     displayName: "my-folder",
 * });
 * const osconfigSa = new gcp.folder.ServiceIdentity("osconfig_sa", {
 *     folder: myFolder.folderId,
 *     service: "osconfig.googleapis.com",
 * });
 * const admin = new gcp.folder.IAMMember("admin", {
 *     folder: myFolder.name,
 *     role: "roles/osconfig.serviceAgent",
 *     member: osconfigSa.member,
 * });
 * ```
 *
 * ## Import
 *
 * This resource does not support import.
 */
class ServiceIdentity extends pulumi.CustomResource {
    /**
     * Get an existing ServiceIdentity resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceIdentity(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:folder/serviceIdentity:ServiceIdentity';
    /**
     * Returns true if the given object is an instance of ServiceIdentity.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceIdentity.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["email"] = state?.email;
            resourceInputs["folder"] = state?.folder;
            resourceInputs["member"] = state?.member;
            resourceInputs["service"] = state?.service;
        }
        else {
            const args = argsOrState;
            if (args?.folder === undefined && !opts.urn) {
                throw new Error("Missing required property 'folder'");
            }
            if (args?.service === undefined && !opts.urn) {
                throw new Error("Missing required property 'service'");
            }
            resourceInputs["folder"] = args?.folder;
            resourceInputs["service"] = args?.service;
            resourceInputs["email"] = undefined /*out*/;
            resourceInputs["member"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServiceIdentity.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceIdentity = ServiceIdentity;
