// @pulumi/gcp/workstations/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkstationIamPolicy = exports.WorkstationIamMember = exports.WorkstationIamBinding = exports.WorkstationConfigIamPolicy = exports.WorkstationConfigIamMember = exports.WorkstationConfigIamBinding = exports.WorkstationConfig = exports.WorkstationCluster = exports.Workstation = exports.getWorkstationIamPolicyOutput = exports.getWorkstationIamPolicy = exports.getWorkstationConfigIamPolicyOutput = exports.getWorkstationConfigIamPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getWorkstationConfigIamPolicy = null;
exports.getWorkstationConfigIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWorkstationConfigIamPolicy", "getWorkstationConfigIamPolicyOutput"], () => require("./getWorkstationConfigIamPolicy"));
exports.getWorkstationIamPolicy = null;
exports.getWorkstationIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getWorkstationIamPolicy", "getWorkstationIamPolicyOutput"], () => require("./getWorkstationIamPolicy"));
exports.Workstation = null;
utilities.lazyLoad(exports, ["Workstation"], () => require("./workstation"));
exports.WorkstationCluster = null;
utilities.lazyLoad(exports, ["WorkstationCluster"], () => require("./workstationCluster"));
exports.WorkstationConfig = null;
utilities.lazyLoad(exports, ["WorkstationConfig"], () => require("./workstationConfig"));
exports.WorkstationConfigIamBinding = null;
utilities.lazyLoad(exports, ["WorkstationConfigIamBinding"], () => require("./workstationConfigIamBinding"));
exports.WorkstationConfigIamMember = null;
utilities.lazyLoad(exports, ["WorkstationConfigIamMember"], () => require("./workstationConfigIamMember"));
exports.WorkstationConfigIamPolicy = null;
utilities.lazyLoad(exports, ["WorkstationConfigIamPolicy"], () => require("./workstationConfigIamPolicy"));
exports.WorkstationIamBinding = null;
utilities.lazyLoad(exports, ["WorkstationIamBinding"], () => require("./workstationIamBinding"));
exports.WorkstationIamMember = null;
utilities.lazyLoad(exports, ["WorkstationIamMember"], () => require("./workstationIamMember"));
exports.WorkstationIamPolicy = null;
utilities.lazyLoad(exports, ["WorkstationIamPolicy"], () => require("./workstationIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:workstations/workstation:Workstation":
                return new exports.Workstation(name, undefined, { urn });
            case "gcp:workstations/workstationCluster:WorkstationCluster":
                return new exports.WorkstationCluster(name, undefined, { urn });
            case "gcp:workstations/workstationConfig:WorkstationConfig":
                return new exports.WorkstationConfig(name, undefined, { urn });
            case "gcp:workstations/workstationConfigIamBinding:WorkstationConfigIamBinding":
                return new exports.WorkstationConfigIamBinding(name, undefined, { urn });
            case "gcp:workstations/workstationConfigIamMember:WorkstationConfigIamMember":
                return new exports.WorkstationConfigIamMember(name, undefined, { urn });
            case "gcp:workstations/workstationConfigIamPolicy:WorkstationConfigIamPolicy":
                return new exports.WorkstationConfigIamPolicy(name, undefined, { urn });
            case "gcp:workstations/workstationIamBinding:WorkstationIamBinding":
                return new exports.WorkstationIamBinding(name, undefined, { urn });
            case "gcp:workstations/workstationIamMember:WorkstationIamMember":
                return new exports.WorkstationIamMember(name, undefined, { urn });
            case "gcp:workstations/workstationIamPolicy:WorkstationIamPolicy":
                return new exports.WorkstationIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "workstations/workstation", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationCluster", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationConfigIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationConfigIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationConfigIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "workstations/workstationIamPolicy", _module);
