// @pulumi/gcp/logging/billingAccountSink.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingAccountSink = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * * [API documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/billingAccounts.sinks)
 * * How-to Guides
 *     * [Exporting Logs](https://cloud.google.com/logging/docs/export)
 *
 * > **Note** You must have the "Logs Configuration Writer" IAM role (`roles/logging.configWriter`)
 * [granted on the billing account](https://cloud.google.com/billing/reference/rest/v1/billingAccounts/getIamPolicy) to
 * the credentials used with this provider. [IAM roles granted on a billing account](https://cloud.google.com/billing/docs/how-to/billing-access) are separate from the
 * typical IAM roles granted on a project.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const log_bucket = new gcp.storage.Bucket("log-bucket", {
 *     name: "billing-logging-bucket",
 *     location: "US",
 * });
 * const my_sink = new gcp.logging.BillingAccountSink("my-sink", {
 *     name: "my-sink",
 *     description: "some explanation on what this is",
 *     billingAccount: "ABCDEF-012345-GHIJKL",
 *     destination: pulumi.interpolate`storage.googleapis.com/${log_bucket.name}`,
 * });
 * const log_writer = new gcp.projects.IAMBinding("log-writer", {
 *     project: "your-project-id",
 *     role: "roles/storage.objectCreator",
 *     members: [my_sink.writerIdentity],
 * });
 * ```
 *
 * ## Import
 *
 * Billing account logging sinks can be imported using this format:
 *
 * * `billingAccounts/{{billing_account_id}}/sinks/{{sink_id}}`
 *
 * When using the `pulumi import` command, billing account logging sinks can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:logging/billingAccountSink:BillingAccountSink default billingAccounts/{{billing_account_id}}/sinks/{{sink_id}}
 * ```
 */
class BillingAccountSink extends pulumi.CustomResource {
    /**
     * Get an existing BillingAccountSink resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BillingAccountSink(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:logging/billingAccountSink:BillingAccountSink';
    /**
     * Returns true if the given object is an instance of BillingAccountSink.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BillingAccountSink.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bigqueryOptions"] = state?.bigqueryOptions;
            resourceInputs["billingAccount"] = state?.billingAccount;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["destination"] = state?.destination;
            resourceInputs["disabled"] = state?.disabled;
            resourceInputs["exclusions"] = state?.exclusions;
            resourceInputs["filter"] = state?.filter;
            resourceInputs["name"] = state?.name;
            resourceInputs["writerIdentity"] = state?.writerIdentity;
        }
        else {
            const args = argsOrState;
            if (args?.billingAccount === undefined && !opts.urn) {
                throw new Error("Missing required property 'billingAccount'");
            }
            if (args?.destination === undefined && !opts.urn) {
                throw new Error("Missing required property 'destination'");
            }
            resourceInputs["bigqueryOptions"] = args?.bigqueryOptions;
            resourceInputs["billingAccount"] = args?.billingAccount;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["destination"] = args?.destination;
            resourceInputs["disabled"] = args?.disabled;
            resourceInputs["exclusions"] = args?.exclusions;
            resourceInputs["filter"] = args?.filter;
            resourceInputs["name"] = args?.name;
            resourceInputs["writerIdentity"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BillingAccountSink.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BillingAccountSink = BillingAccountSink;
