// @pulumi/gcp/logging/linkedDataset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkedDataset = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Describes a BigQuery linked dataset
 *
 * To get more information about LinkedDataset, see:
 *
 * * [API documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/locations.buckets.links)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/logging/docs/apis)
 *
 * ## Example Usage
 *
 * ### Logging Linked Dataset Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingLinkedDataset = new gcp.logging.ProjectBucketConfig("logging_linked_dataset", {
 *     location: "global",
 *     project: "my-project-name",
 *     enableAnalytics: true,
 *     bucketId: "my-bucket",
 * });
 * const loggingLinkedDatasetLinkedDataset = new gcp.logging.LinkedDataset("logging_linked_dataset", {
 *     linkId: "mylink",
 *     bucket: loggingLinkedDataset.id,
 *     description: "Linked dataset test",
 * });
 * ```
 * ### Logging Linked Dataset All Params
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingLinkedDataset = new gcp.logging.ProjectBucketConfig("logging_linked_dataset", {
 *     location: "global",
 *     project: "my-project-name",
 *     enableAnalytics: true,
 *     bucketId: "my-bucket",
 * });
 * const loggingLinkedDatasetLinkedDataset = new gcp.logging.LinkedDataset("logging_linked_dataset", {
 *     linkId: "mylink",
 *     bucket: "my-bucket",
 *     parent: "projects/my-project-name",
 *     location: "global",
 *     description: "Linked dataset test",
 * }, {
 *     dependsOn: [loggingLinkedDataset],
 * });
 * ```
 *
 * ## Import
 *
 * LinkedDataset can be imported using any of these accepted formats:
 *
 * * `{{parent}}/locations/{{location}}/buckets/{{bucket}}/links/{{link_id}}`
 *
 * When using the `pulumi import` command, LinkedDataset can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:logging/linkedDataset:LinkedDataset default {{parent}}/locations/{{location}}/buckets/{{bucket}}/links/{{link_id}}
 * ```
 */
class LinkedDataset extends pulumi.CustomResource {
    /**
     * Get an existing LinkedDataset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LinkedDataset(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:logging/linkedDataset:LinkedDataset';
    /**
     * Returns true if the given object is an instance of LinkedDataset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LinkedDataset.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bigqueryDatasets"] = state?.bigqueryDatasets;
            resourceInputs["bucket"] = state?.bucket;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["lifecycleState"] = state?.lifecycleState;
            resourceInputs["linkId"] = state?.linkId;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["parent"] = state?.parent;
        }
        else {
            const args = argsOrState;
            if (args?.bucket === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucket'");
            }
            if (args?.linkId === undefined && !opts.urn) {
                throw new Error("Missing required property 'linkId'");
            }
            resourceInputs["bigqueryDatasets"] = args?.bigqueryDatasets;
            resourceInputs["bucket"] = args?.bucket;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["linkId"] = args?.linkId;
            resourceInputs["location"] = args?.location;
            resourceInputs["parent"] = args?.parent;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["lifecycleState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LinkedDataset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LinkedDataset = LinkedDataset;
