// @pulumi/gcp/logging/logScope.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogScope = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Describes a group of resources to read log entries from
 *
 * To get more information about LogScope, see:
 *
 * * [API documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.locations.logScopes)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/logging/docs/apis)
 *
 * ## Example Usage
 *
 * ### Logging Log Scope Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingLogScope = new gcp.logging.LogScope("logging_log_scope", {
 *     parent: "projects/my-project-name",
 *     location: "global",
 *     name: "projects/my-project-name/locations/global/logScopes/my-log-scope",
 *     resourceNames: [
 *         "projects/my-project-name",
 *         "projects/my-project-name/locations/global/buckets/_Default/views/view1",
 *         "projects/my-project-name/locations/global/buckets/_Default/views/view2",
 *     ],
 *     description: "A log scope configured with Terraform",
 * });
 * ```
 *
 * ## Import
 *
 * LogScope can be imported using any of these accepted formats:
 *
 * * `{{parent}}/locations/{{location}}/logScopes/{{name}}`
 *
 * When using the `pulumi import` command, LogScope can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:logging/logScope:LogScope default {{parent}}/locations/{{location}}/logScopes/{{name}}
 * ```
 */
class LogScope extends pulumi.CustomResource {
    /**
     * Get an existing LogScope resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LogScope(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:logging/logScope:LogScope';
    /**
     * Returns true if the given object is an instance of LogScope.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogScope.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["parent"] = state?.parent;
            resourceInputs["resourceNames"] = state?.resourceNames;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.resourceNames === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceNames'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["parent"] = args?.parent;
            resourceInputs["resourceNames"] = args?.resourceNames;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LogScope.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogScope = LogScope;
