// @pulumi/gcp/logging/metric.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Metric = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Logs-based metric can also be used to extract values from logs and create a a distribution
 * of the values. The distribution records the statistics of the extracted values along with
 * an optional histogram of the values as specified by the bucket options.
 *
 * To get more information about Metric, see:
 *
 * * [API documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.metrics/create)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/logging/docs/apis)
 *
 * ## Example Usage
 *
 * ### Logging Metric Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingMetric = new gcp.logging.Metric("logging_metric", {
 *     name: "my-(custom)/metric",
 *     filter: "resource.type=gae_app AND severity>=ERROR",
 *     metricDescriptor: {
 *         metricKind: "DELTA",
 *         valueType: "DISTRIBUTION",
 *         unit: "1",
 *         labels: [
 *             {
 *                 key: "mass",
 *                 valueType: "STRING",
 *                 description: "amount of matter",
 *             },
 *             {
 *                 key: "sku",
 *                 valueType: "INT64",
 *                 description: "Identifying number for item",
 *             },
 *         ],
 *         displayName: "My metric",
 *     },
 *     valueExtractor: "EXTRACT(jsonPayload.request)",
 *     labelExtractors: {
 *         mass: "EXTRACT(jsonPayload.request)",
 *         sku: "EXTRACT(jsonPayload.id)",
 *     },
 *     bucketOptions: {
 *         linearBuckets: {
 *             numFiniteBuckets: 3,
 *             width: 1,
 *             offset: 1,
 *         },
 *     },
 * });
 * ```
 * ### Logging Metric Counter Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingMetric = new gcp.logging.Metric("logging_metric", {
 *     name: "my-(custom)/metric",
 *     filter: "resource.type=gae_app AND severity>=ERROR",
 *     metricDescriptor: {
 *         metricKind: "DELTA",
 *         valueType: "INT64",
 *     },
 * });
 * ```
 * ### Logging Metric Counter Labels
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingMetric = new gcp.logging.Metric("logging_metric", {
 *     name: "my-(custom)/metric",
 *     filter: "resource.type=gae_app AND severity>=ERROR",
 *     metricDescriptor: {
 *         metricKind: "DELTA",
 *         valueType: "INT64",
 *         labels: [{
 *             key: "mass",
 *             valueType: "STRING",
 *             description: "amount of matter",
 *         }],
 *     },
 *     labelExtractors: {
 *         mass: "EXTRACT(jsonPayload.request)",
 *     },
 * });
 * ```
 * ### Logging Metric Logging Bucket
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingMetric = new gcp.logging.ProjectBucketConfig("logging_metric", {
 *     location: "global",
 *     project: "my-project-name",
 *     bucketId: "_Default",
 * });
 * const loggingMetricMetric = new gcp.logging.Metric("logging_metric", {
 *     name: "my-(custom)/metric",
 *     filter: "resource.type=gae_app AND severity>=ERROR",
 *     bucketName: loggingMetric.name,
 * });
 * ```
 * ### Logging Metric Disabled
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const loggingMetric = new gcp.logging.Metric("logging_metric", {
 *     name: "my-(custom)/metric",
 *     filter: "resource.type=gae_app AND severity>=ERROR",
 *     metricDescriptor: {
 *         metricKind: "DELTA",
 *         valueType: "INT64",
 *     },
 *     disabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * Metric can be imported using any of these accepted formats:
 *
 * * `{{project}} {{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, Metric can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ terraform import google_logging_metric.default "{{project}} {{name}}"
 * $ pulumi import gcp:logging/metric:Metric default {{name}}
 * ```
 */
class Metric extends pulumi.CustomResource {
    /**
     * Get an existing Metric resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Metric(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:logging/metric:Metric';
    /**
     * Returns true if the given object is an instance of Metric.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Metric.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bucketName"] = state?.bucketName;
            resourceInputs["bucketOptions"] = state?.bucketOptions;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["disabled"] = state?.disabled;
            resourceInputs["filter"] = state?.filter;
            resourceInputs["labelExtractors"] = state?.labelExtractors;
            resourceInputs["metricDescriptor"] = state?.metricDescriptor;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["valueExtractor"] = state?.valueExtractor;
        }
        else {
            const args = argsOrState;
            if (args?.filter === undefined && !opts.urn) {
                throw new Error("Missing required property 'filter'");
            }
            resourceInputs["bucketName"] = args?.bucketName;
            resourceInputs["bucketOptions"] = args?.bucketOptions;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["disabled"] = args?.disabled;
            resourceInputs["filter"] = args?.filter;
            resourceInputs["labelExtractors"] = args?.labelExtractors;
            resourceInputs["metricDescriptor"] = args?.metricDescriptor;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["valueExtractor"] = args?.valueExtractor;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Metric.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Metric = Metric;
