// @pulumi/gcp/logging/organizationSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Default resource settings control whether CMEK is required for new log buckets. These settings also determine the storage location for the _Default and _Required log buckets, and whether the _Default sink is enabled or disabled.
 *
 * To get more information about OrganizationSettings, see:
 *
 * * [API documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/TopLevel/getSettings)
 * * How-to Guides
 *     * [Configure default settings for organizations and folders](https://cloud.google.com/logging/docs/default-settings)
 *
 * ## Example Usage
 *
 * ### Logging Organization Settings All
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const settings = gcp.logging.getOrganizationSettings({
 *     organization: "123456789",
 * });
 * const iam = new gcp.kms.CryptoKeyIAMMember("iam", {
 *     cryptoKeyId: "kms-key",
 *     role: "roles/cloudkms.cryptoKeyEncrypterDecrypter",
 *     member: settings.then(settings => `serviceAccount:${settings.kmsServiceAccountId}`),
 * });
 * const example = new gcp.logging.OrganizationSettings("example", {
 *     disableDefaultSink: true,
 *     kmsKeyName: "kms-key",
 *     organization: "123456789",
 *     storageLocation: "us-central1",
 * }, {
 *     dependsOn: [iam],
 * });
 * ```
 *
 * ## Import
 *
 * OrganizationSettings can be imported using any of these accepted formats:
 *
 * * `organizations/{{organization}}/settings`
 * * `{{organization}}`
 *
 * When using the `pulumi import` command, OrganizationSettings can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:logging/organizationSettings:OrganizationSettings default organizations/{{organization}}/settings
 * $ pulumi import gcp:logging/organizationSettings:OrganizationSettings default {{organization}}
 * ```
 */
class OrganizationSettings extends pulumi.CustomResource {
    /**
     * Get an existing OrganizationSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OrganizationSettings(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:logging/organizationSettings:OrganizationSettings';
    /**
     * Returns true if the given object is an instance of OrganizationSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OrganizationSettings.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["disableDefaultSink"] = state?.disableDefaultSink;
            resourceInputs["kmsKeyName"] = state?.kmsKeyName;
            resourceInputs["kmsServiceAccountId"] = state?.kmsServiceAccountId;
            resourceInputs["loggingServiceAccountId"] = state?.loggingServiceAccountId;
            resourceInputs["name"] = state?.name;
            resourceInputs["organization"] = state?.organization;
            resourceInputs["storageLocation"] = state?.storageLocation;
        }
        else {
            const args = argsOrState;
            if (args?.organization === undefined && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            resourceInputs["disableDefaultSink"] = args?.disableDefaultSink;
            resourceInputs["kmsKeyName"] = args?.kmsKeyName;
            resourceInputs["organization"] = args?.organization;
            resourceInputs["storageLocation"] = args?.storageLocation;
            resourceInputs["kmsServiceAccountId"] = undefined /*out*/;
            resourceInputs["loggingServiceAccountId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(OrganizationSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OrganizationSettings = OrganizationSettings;
