// @pulumi/aws/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.autoscalingplans = exports.autoscaling = exports.auditmanager = exports.athena = exports.arczonalshift = exports.arcregionswitch = exports.appsync = exports.appstream = exports.apprunner = exports.appmesh = exports.applicationloadbalancing = exports.applicationinsights = exports.appintegrations = exports.appflow = exports.appfabric = exports.appconfig = exports.appautoscaling = exports.apigatewayv2 = exports.apigateway = exports.amplify = exports.amp = exports.alb = exports.acmpca = exports.acm = exports.account = exports.accessanalyzer = exports.getServicePrincipalOutput = exports.getServicePrincipal = exports.getServiceOutput = exports.getService = exports.getRegionsOutput = exports.getRegions = exports.getRegionOutput = exports.getRegion = exports.getPartitionOutput = exports.getPartition = exports.getIpRangesOutput = exports.getIpRanges = exports.getDefaultTagsOutput = exports.getDefaultTags = exports.getCallerIdentityOutput = exports.getCallerIdentity = exports.getBillingServiceAccountOutput = exports.getBillingServiceAccount = exports.getAvailabilityZonesOutput = exports.getAvailabilityZones = exports.getAvailabilityZoneOutput = exports.getAvailabilityZone = exports.getArnOutput = exports.getArn = void 0;
exports.devopsguru = exports.devicefarm = exports.detective = exports.dax = exports.datazone = exports.datasync = exports.datapipeline = exports.dataexchange = exports.customerprofiles = exports.cur = exports.costoptimizationhub = exports.costexplorer = exports.controltower = exports.connect = exports.config = exports.computeoptimizer = exports.comprehend = exports.cognito = exports.codestarnotifications = exports.codestarconnections = exports.codepipeline = exports.codegurureviewer = exports.codeguruprofiler = exports.codedeploy = exports.codeconnections = exports.codecommit = exports.codecatalyst = exports.codebuild = exports.codeartifact = exports.cloudwatch = exports.cloudtrail = exports.cloudsearch = exports.cloudhsmv2 = exports.cloudfront = exports.cloudformation = exports.cloudcontrol = exports.cloud9 = exports.cleanrooms = exports.chimesdkmediapipelines = exports.chime = exports.chatbot = exports.cfg = exports.budgets = exports.billing = exports.bedrockmodel = exports.bedrockfoundation = exports.bedrock = exports.bcmdata = exports.batch = exports.backup = void 0;
exports.kms = exports.kinesisanalyticsv2 = exports.kinesis = exports.keyspaces = exports.kendra = exports.ivschat = exports.ivs = exports.iot = exports.invoicing = exports.inspector2 = exports.inspector = exports.imagebuilder = exports.identitystore = exports.iam = exports.guardduty = exports.grafana = exports.glue = exports.globalaccelerator = exports.glacier = exports.gamelift = exports.fsx = exports.fms = exports.fis = exports.finspace = exports.evidently = exports.emrserverless = exports.emrcontainers = exports.emr = exports.elb = exports.elastictranscoder = exports.elasticsearch = exports.elasticbeanstalk = exports.elasticache = exports.eks = exports.efs = exports.ecs = exports.ecrpublic = exports.ecr = exports.ec2transitgateway = exports.ec2clientvpn = exports.ec2 = exports.ebs = exports.dynamodb = exports.dsql = exports.drs = exports.docdb = exports.dms = exports.dlm = exports.directoryservice = exports.directconnect = void 0;
exports.resiliencehub = exports.rekognition = exports.redshiftserverless = exports.redshiftdata = exports.redshift = exports.rds = exports.rbin = exports.ram = exports.quicksight = exports.qldb = exports.qbusiness = exports.pricing = exports.polly = exports.pipes = exports.pinpoint = exports.paymentcryptography = exports.outposts = exports.organizations = exports.opensearchingest = exports.opensearch = exports.odb = exports.observabilityadmin = exports.oam = exports.notifications = exports.networkmonitor = exports.networkmanager = exports.networkflowmonitor = exports.networkfirewall = exports.neptunegraph = exports.neptune = exports.mwaa = exports.mskconnect = exports.msk = exports.mq = exports.memorydb = exports.mediastore = exports.mediapackagev2 = exports.mediapackage = exports.medialive = exports.mediaconvert = exports.macie2 = exports.macie = exports.m2 = exports.location = exports.lightsail = exports.licensemanager = exports.lex = exports.lb = exports.lambda = exports.lakeformation = void 0;
exports.vpn = exports.vpclattice = exports.vpc = exports.verifiedpermissions = exports.verifiedaccess = exports.uxc = exports.types = exports.transfer = exports.transcribe = exports.timestreamwrite = exports.timestreamquery = exports.timestreaminfluxdb = exports.synthetics = exports.swf = exports.storagegateway = exports.ssoadmin = exports.ssmincidents = exports.ssmcontacts = exports.ssm = exports.sqs = exports.sns = exports.signer = exports.shield = exports.sfn = exports.sesv2 = exports.ses = exports.servicequotas = exports.servicediscovery = exports.servicecatalog = exports.serverlessrepository = exports.securitylake = exports.securityhub = exports.secretsmanager = exports.schemas = exports.scheduler = exports.savingsplans = exports.sagemaker = exports.s3tables = exports.s3outposts = exports.s3control = exports.s3 = exports.rum = exports.route53recoveryreadiness = exports.route53recoverycontrol = exports.route53domains = exports.route53 = exports.rolesanywhere = exports.resourcegroupstaggingapi = exports.resourcegroups = exports.resourceexplorer = void 0;
exports.xray = exports.workspacesweb = exports.workspaces = exports.workmail = exports.wafv2 = exports.wafregional = exports.waf = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("./utilities"));
exports.getArn = null;
exports.getArnOutput = null;
utilities.lazyLoad(exports, ["getArn", "getArnOutput"], () => require("./getArn"));
exports.getAvailabilityZone = null;
exports.getAvailabilityZoneOutput = null;
utilities.lazyLoad(exports, ["getAvailabilityZone", "getAvailabilityZoneOutput"], () => require("./getAvailabilityZone"));
exports.getAvailabilityZones = null;
exports.getAvailabilityZonesOutput = null;
utilities.lazyLoad(exports, ["getAvailabilityZones", "getAvailabilityZonesOutput"], () => require("./getAvailabilityZones"));
exports.getBillingServiceAccount = null;
exports.getBillingServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getBillingServiceAccount", "getBillingServiceAccountOutput"], () => require("./getBillingServiceAccount"));
exports.getCallerIdentity = null;
exports.getCallerIdentityOutput = null;
utilities.lazyLoad(exports, ["getCallerIdentity", "getCallerIdentityOutput"], () => require("./getCallerIdentity"));
exports.getDefaultTags = null;
exports.getDefaultTagsOutput = null;
utilities.lazyLoad(exports, ["getDefaultTags", "getDefaultTagsOutput"], () => require("./getDefaultTags"));
exports.getIpRanges = null;
exports.getIpRangesOutput = null;
utilities.lazyLoad(exports, ["getIpRanges", "getIpRangesOutput"], () => require("./getIpRanges"));
exports.getPartition = null;
exports.getPartitionOutput = null;
utilities.lazyLoad(exports, ["getPartition", "getPartitionOutput"], () => require("./getPartition"));
exports.getRegion = null;
exports.getRegionOutput = null;
utilities.lazyLoad(exports, ["getRegion", "getRegionOutput"], () => require("./getRegion"));
exports.getRegions = null;
exports.getRegionsOutput = null;
utilities.lazyLoad(exports, ["getRegions", "getRegionsOutput"], () => require("./getRegions"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServicePrincipal = null;
exports.getServicePrincipalOutput = null;
utilities.lazyLoad(exports, ["getServicePrincipal", "getServicePrincipalOutput"], () => require("./getServicePrincipal"));
__exportStar(require("./provider"), exports);
const provider_1 = require("./provider");
__exportStar(require("./tags"), exports);
__exportStar(require("./utils"), exports);
// Export enums:
__exportStar(require("./types/enums"), exports);
// Export sub-modules:
const accessanalyzer = __importStar(require("./accessanalyzer"));
exports.accessanalyzer = accessanalyzer;
const account = __importStar(require("./account"));
exports.account = account;
const acm = __importStar(require("./acm"));
exports.acm = acm;
const acmpca = __importStar(require("./acmpca"));
exports.acmpca = acmpca;
const alb = __importStar(require("./alb"));
exports.alb = alb;
const amp = __importStar(require("./amp"));
exports.amp = amp;
const amplify = __importStar(require("./amplify"));
exports.amplify = amplify;
const apigateway = __importStar(require("./apigateway"));
exports.apigateway = apigateway;
const apigatewayv2 = __importStar(require("./apigatewayv2"));
exports.apigatewayv2 = apigatewayv2;
const appautoscaling = __importStar(require("./appautoscaling"));
exports.appautoscaling = appautoscaling;
const appconfig = __importStar(require("./appconfig"));
exports.appconfig = appconfig;
const appfabric = __importStar(require("./appfabric"));
exports.appfabric = appfabric;
const appflow = __importStar(require("./appflow"));
exports.appflow = appflow;
const appintegrations = __importStar(require("./appintegrations"));
exports.appintegrations = appintegrations;
const applicationinsights = __importStar(require("./applicationinsights"));
exports.applicationinsights = applicationinsights;
const applicationloadbalancing = __importStar(require("./applicationloadbalancing"));
exports.applicationloadbalancing = applicationloadbalancing;
const appmesh = __importStar(require("./appmesh"));
exports.appmesh = appmesh;
const apprunner = __importStar(require("./apprunner"));
exports.apprunner = apprunner;
const appstream = __importStar(require("./appstream"));
exports.appstream = appstream;
const appsync = __importStar(require("./appsync"));
exports.appsync = appsync;
const arcregionswitch = __importStar(require("./arcregionswitch"));
exports.arcregionswitch = arcregionswitch;
const arczonalshift = __importStar(require("./arczonalshift"));
exports.arczonalshift = arczonalshift;
const athena = __importStar(require("./athena"));
exports.athena = athena;
const auditmanager = __importStar(require("./auditmanager"));
exports.auditmanager = auditmanager;
const autoscaling = __importStar(require("./autoscaling"));
exports.autoscaling = autoscaling;
const autoscalingplans = __importStar(require("./autoscalingplans"));
exports.autoscalingplans = autoscalingplans;
const backup = __importStar(require("./backup"));
exports.backup = backup;
const batch = __importStar(require("./batch"));
exports.batch = batch;
const bcmdata = __importStar(require("./bcmdata"));
exports.bcmdata = bcmdata;
const bedrock = __importStar(require("./bedrock"));
exports.bedrock = bedrock;
const bedrockfoundation = __importStar(require("./bedrockfoundation"));
exports.bedrockfoundation = bedrockfoundation;
const bedrockmodel = __importStar(require("./bedrockmodel"));
exports.bedrockmodel = bedrockmodel;
const billing = __importStar(require("./billing"));
exports.billing = billing;
const budgets = __importStar(require("./budgets"));
exports.budgets = budgets;
const cfg = __importStar(require("./cfg"));
exports.cfg = cfg;
const chatbot = __importStar(require("./chatbot"));
exports.chatbot = chatbot;
const chime = __importStar(require("./chime"));
exports.chime = chime;
const chimesdkmediapipelines = __importStar(require("./chimesdkmediapipelines"));
exports.chimesdkmediapipelines = chimesdkmediapipelines;
const cleanrooms = __importStar(require("./cleanrooms"));
exports.cleanrooms = cleanrooms;
const cloud9 = __importStar(require("./cloud9"));
exports.cloud9 = cloud9;
const cloudcontrol = __importStar(require("./cloudcontrol"));
exports.cloudcontrol = cloudcontrol;
const cloudformation = __importStar(require("./cloudformation"));
exports.cloudformation = cloudformation;
const cloudfront = __importStar(require("./cloudfront"));
exports.cloudfront = cloudfront;
const cloudhsmv2 = __importStar(require("./cloudhsmv2"));
exports.cloudhsmv2 = cloudhsmv2;
const cloudsearch = __importStar(require("./cloudsearch"));
exports.cloudsearch = cloudsearch;
const cloudtrail = __importStar(require("./cloudtrail"));
exports.cloudtrail = cloudtrail;
const cloudwatch = __importStar(require("./cloudwatch"));
exports.cloudwatch = cloudwatch;
const codeartifact = __importStar(require("./codeartifact"));
exports.codeartifact = codeartifact;
const codebuild = __importStar(require("./codebuild"));
exports.codebuild = codebuild;
const codecatalyst = __importStar(require("./codecatalyst"));
exports.codecatalyst = codecatalyst;
const codecommit = __importStar(require("./codecommit"));
exports.codecommit = codecommit;
const codeconnections = __importStar(require("./codeconnections"));
exports.codeconnections = codeconnections;
const codedeploy = __importStar(require("./codedeploy"));
exports.codedeploy = codedeploy;
const codeguruprofiler = __importStar(require("./codeguruprofiler"));
exports.codeguruprofiler = codeguruprofiler;
const codegurureviewer = __importStar(require("./codegurureviewer"));
exports.codegurureviewer = codegurureviewer;
const codepipeline = __importStar(require("./codepipeline"));
exports.codepipeline = codepipeline;
const codestarconnections = __importStar(require("./codestarconnections"));
exports.codestarconnections = codestarconnections;
const codestarnotifications = __importStar(require("./codestarnotifications"));
exports.codestarnotifications = codestarnotifications;
const cognito = __importStar(require("./cognito"));
exports.cognito = cognito;
const comprehend = __importStar(require("./comprehend"));
exports.comprehend = comprehend;
const computeoptimizer = __importStar(require("./computeoptimizer"));
exports.computeoptimizer = computeoptimizer;
const config = __importStar(require("./config"));
exports.config = config;
const connect = __importStar(require("./connect"));
exports.connect = connect;
const controltower = __importStar(require("./controltower"));
exports.controltower = controltower;
const costexplorer = __importStar(require("./costexplorer"));
exports.costexplorer = costexplorer;
const costoptimizationhub = __importStar(require("./costoptimizationhub"));
exports.costoptimizationhub = costoptimizationhub;
const cur = __importStar(require("./cur"));
exports.cur = cur;
const customerprofiles = __importStar(require("./customerprofiles"));
exports.customerprofiles = customerprofiles;
const dataexchange = __importStar(require("./dataexchange"));
exports.dataexchange = dataexchange;
const datapipeline = __importStar(require("./datapipeline"));
exports.datapipeline = datapipeline;
const datasync = __importStar(require("./datasync"));
exports.datasync = datasync;
const datazone = __importStar(require("./datazone"));
exports.datazone = datazone;
const dax = __importStar(require("./dax"));
exports.dax = dax;
const detective = __importStar(require("./detective"));
exports.detective = detective;
const devicefarm = __importStar(require("./devicefarm"));
exports.devicefarm = devicefarm;
const devopsguru = __importStar(require("./devopsguru"));
exports.devopsguru = devopsguru;
const directconnect = __importStar(require("./directconnect"));
exports.directconnect = directconnect;
const directoryservice = __importStar(require("./directoryservice"));
exports.directoryservice = directoryservice;
const dlm = __importStar(require("./dlm"));
exports.dlm = dlm;
const dms = __importStar(require("./dms"));
exports.dms = dms;
const docdb = __importStar(require("./docdb"));
exports.docdb = docdb;
const drs = __importStar(require("./drs"));
exports.drs = drs;
const dsql = __importStar(require("./dsql"));
exports.dsql = dsql;
const dynamodb = __importStar(require("./dynamodb"));
exports.dynamodb = dynamodb;
const ebs = __importStar(require("./ebs"));
exports.ebs = ebs;
const ec2 = __importStar(require("./ec2"));
exports.ec2 = ec2;
const ec2clientvpn = __importStar(require("./ec2clientvpn"));
exports.ec2clientvpn = ec2clientvpn;
const ec2transitgateway = __importStar(require("./ec2transitgateway"));
exports.ec2transitgateway = ec2transitgateway;
const ecr = __importStar(require("./ecr"));
exports.ecr = ecr;
const ecrpublic = __importStar(require("./ecrpublic"));
exports.ecrpublic = ecrpublic;
const ecs = __importStar(require("./ecs"));
exports.ecs = ecs;
const efs = __importStar(require("./efs"));
exports.efs = efs;
const eks = __importStar(require("./eks"));
exports.eks = eks;
const elasticache = __importStar(require("./elasticache"));
exports.elasticache = elasticache;
const elasticbeanstalk = __importStar(require("./elasticbeanstalk"));
exports.elasticbeanstalk = elasticbeanstalk;
const elasticsearch = __importStar(require("./elasticsearch"));
exports.elasticsearch = elasticsearch;
const elastictranscoder = __importStar(require("./elastictranscoder"));
exports.elastictranscoder = elastictranscoder;
const elb = __importStar(require("./elb"));
exports.elb = elb;
const emr = __importStar(require("./emr"));
exports.emr = emr;
const emrcontainers = __importStar(require("./emrcontainers"));
exports.emrcontainers = emrcontainers;
const emrserverless = __importStar(require("./emrserverless"));
exports.emrserverless = emrserverless;
const evidently = __importStar(require("./evidently"));
exports.evidently = evidently;
const finspace = __importStar(require("./finspace"));
exports.finspace = finspace;
const fis = __importStar(require("./fis"));
exports.fis = fis;
const fms = __importStar(require("./fms"));
exports.fms = fms;
const fsx = __importStar(require("./fsx"));
exports.fsx = fsx;
const gamelift = __importStar(require("./gamelift"));
exports.gamelift = gamelift;
const glacier = __importStar(require("./glacier"));
exports.glacier = glacier;
const globalaccelerator = __importStar(require("./globalaccelerator"));
exports.globalaccelerator = globalaccelerator;
const glue = __importStar(require("./glue"));
exports.glue = glue;
const grafana = __importStar(require("./grafana"));
exports.grafana = grafana;
const guardduty = __importStar(require("./guardduty"));
exports.guardduty = guardduty;
const iam = __importStar(require("./iam"));
exports.iam = iam;
const identitystore = __importStar(require("./identitystore"));
exports.identitystore = identitystore;
const imagebuilder = __importStar(require("./imagebuilder"));
exports.imagebuilder = imagebuilder;
const inspector = __importStar(require("./inspector"));
exports.inspector = inspector;
const inspector2 = __importStar(require("./inspector2"));
exports.inspector2 = inspector2;
const invoicing = __importStar(require("./invoicing"));
exports.invoicing = invoicing;
const iot = __importStar(require("./iot"));
exports.iot = iot;
const ivs = __importStar(require("./ivs"));
exports.ivs = ivs;
const ivschat = __importStar(require("./ivschat"));
exports.ivschat = ivschat;
const kendra = __importStar(require("./kendra"));
exports.kendra = kendra;
const keyspaces = __importStar(require("./keyspaces"));
exports.keyspaces = keyspaces;
const kinesis = __importStar(require("./kinesis"));
exports.kinesis = kinesis;
const kinesisanalyticsv2 = __importStar(require("./kinesisanalyticsv2"));
exports.kinesisanalyticsv2 = kinesisanalyticsv2;
const kms = __importStar(require("./kms"));
exports.kms = kms;
const lakeformation = __importStar(require("./lakeformation"));
exports.lakeformation = lakeformation;
const lambda = __importStar(require("./lambda"));
exports.lambda = lambda;
const lb = __importStar(require("./lb"));
exports.lb = lb;
const lex = __importStar(require("./lex"));
exports.lex = lex;
const licensemanager = __importStar(require("./licensemanager"));
exports.licensemanager = licensemanager;
const lightsail = __importStar(require("./lightsail"));
exports.lightsail = lightsail;
const location = __importStar(require("./location"));
exports.location = location;
const m2 = __importStar(require("./m2"));
exports.m2 = m2;
const macie = __importStar(require("./macie"));
exports.macie = macie;
const macie2 = __importStar(require("./macie2"));
exports.macie2 = macie2;
const mediaconvert = __importStar(require("./mediaconvert"));
exports.mediaconvert = mediaconvert;
const medialive = __importStar(require("./medialive"));
exports.medialive = medialive;
const mediapackage = __importStar(require("./mediapackage"));
exports.mediapackage = mediapackage;
const mediapackagev2 = __importStar(require("./mediapackagev2"));
exports.mediapackagev2 = mediapackagev2;
const mediastore = __importStar(require("./mediastore"));
exports.mediastore = mediastore;
const memorydb = __importStar(require("./memorydb"));
exports.memorydb = memorydb;
const mq = __importStar(require("./mq"));
exports.mq = mq;
const msk = __importStar(require("./msk"));
exports.msk = msk;
const mskconnect = __importStar(require("./mskconnect"));
exports.mskconnect = mskconnect;
const mwaa = __importStar(require("./mwaa"));
exports.mwaa = mwaa;
const neptune = __importStar(require("./neptune"));
exports.neptune = neptune;
const neptunegraph = __importStar(require("./neptunegraph"));
exports.neptunegraph = neptunegraph;
const networkfirewall = __importStar(require("./networkfirewall"));
exports.networkfirewall = networkfirewall;
const networkflowmonitor = __importStar(require("./networkflowmonitor"));
exports.networkflowmonitor = networkflowmonitor;
const networkmanager = __importStar(require("./networkmanager"));
exports.networkmanager = networkmanager;
const networkmonitor = __importStar(require("./networkmonitor"));
exports.networkmonitor = networkmonitor;
const notifications = __importStar(require("./notifications"));
exports.notifications = notifications;
const oam = __importStar(require("./oam"));
exports.oam = oam;
const observabilityadmin = __importStar(require("./observabilityadmin"));
exports.observabilityadmin = observabilityadmin;
const odb = __importStar(require("./odb"));
exports.odb = odb;
const opensearch = __importStar(require("./opensearch"));
exports.opensearch = opensearch;
const opensearchingest = __importStar(require("./opensearchingest"));
exports.opensearchingest = opensearchingest;
const organizations = __importStar(require("./organizations"));
exports.organizations = organizations;
const outposts = __importStar(require("./outposts"));
exports.outposts = outposts;
const paymentcryptography = __importStar(require("./paymentcryptography"));
exports.paymentcryptography = paymentcryptography;
const pinpoint = __importStar(require("./pinpoint"));
exports.pinpoint = pinpoint;
const pipes = __importStar(require("./pipes"));
exports.pipes = pipes;
const polly = __importStar(require("./polly"));
exports.polly = polly;
const pricing = __importStar(require("./pricing"));
exports.pricing = pricing;
const qbusiness = __importStar(require("./qbusiness"));
exports.qbusiness = qbusiness;
const qldb = __importStar(require("./qldb"));
exports.qldb = qldb;
const quicksight = __importStar(require("./quicksight"));
exports.quicksight = quicksight;
const ram = __importStar(require("./ram"));
exports.ram = ram;
const rbin = __importStar(require("./rbin"));
exports.rbin = rbin;
const rds = __importStar(require("./rds"));
exports.rds = rds;
const redshift = __importStar(require("./redshift"));
exports.redshift = redshift;
const redshiftdata = __importStar(require("./redshiftdata"));
exports.redshiftdata = redshiftdata;
const redshiftserverless = __importStar(require("./redshiftserverless"));
exports.redshiftserverless = redshiftserverless;
const rekognition = __importStar(require("./rekognition"));
exports.rekognition = rekognition;
const resiliencehub = __importStar(require("./resiliencehub"));
exports.resiliencehub = resiliencehub;
const resourceexplorer = __importStar(require("./resourceexplorer"));
exports.resourceexplorer = resourceexplorer;
const resourcegroups = __importStar(require("./resourcegroups"));
exports.resourcegroups = resourcegroups;
const resourcegroupstaggingapi = __importStar(require("./resourcegroupstaggingapi"));
exports.resourcegroupstaggingapi = resourcegroupstaggingapi;
const rolesanywhere = __importStar(require("./rolesanywhere"));
exports.rolesanywhere = rolesanywhere;
const route53 = __importStar(require("./route53"));
exports.route53 = route53;
const route53domains = __importStar(require("./route53domains"));
exports.route53domains = route53domains;
const route53recoverycontrol = __importStar(require("./route53recoverycontrol"));
exports.route53recoverycontrol = route53recoverycontrol;
const route53recoveryreadiness = __importStar(require("./route53recoveryreadiness"));
exports.route53recoveryreadiness = route53recoveryreadiness;
const rum = __importStar(require("./rum"));
exports.rum = rum;
const s3 = __importStar(require("./s3"));
exports.s3 = s3;
const s3control = __importStar(require("./s3control"));
exports.s3control = s3control;
const s3outposts = __importStar(require("./s3outposts"));
exports.s3outposts = s3outposts;
const s3tables = __importStar(require("./s3tables"));
exports.s3tables = s3tables;
const sagemaker = __importStar(require("./sagemaker"));
exports.sagemaker = sagemaker;
const savingsplans = __importStar(require("./savingsplans"));
exports.savingsplans = savingsplans;
const scheduler = __importStar(require("./scheduler"));
exports.scheduler = scheduler;
const schemas = __importStar(require("./schemas"));
exports.schemas = schemas;
const secretsmanager = __importStar(require("./secretsmanager"));
exports.secretsmanager = secretsmanager;
const securityhub = __importStar(require("./securityhub"));
exports.securityhub = securityhub;
const securitylake = __importStar(require("./securitylake"));
exports.securitylake = securitylake;
const serverlessrepository = __importStar(require("./serverlessrepository"));
exports.serverlessrepository = serverlessrepository;
const servicecatalog = __importStar(require("./servicecatalog"));
exports.servicecatalog = servicecatalog;
const servicediscovery = __importStar(require("./servicediscovery"));
exports.servicediscovery = servicediscovery;
const servicequotas = __importStar(require("./servicequotas"));
exports.servicequotas = servicequotas;
const ses = __importStar(require("./ses"));
exports.ses = ses;
const sesv2 = __importStar(require("./sesv2"));
exports.sesv2 = sesv2;
const sfn = __importStar(require("./sfn"));
exports.sfn = sfn;
const shield = __importStar(require("./shield"));
exports.shield = shield;
const signer = __importStar(require("./signer"));
exports.signer = signer;
const sns = __importStar(require("./sns"));
exports.sns = sns;
const sqs = __importStar(require("./sqs"));
exports.sqs = sqs;
const ssm = __importStar(require("./ssm"));
exports.ssm = ssm;
const ssmcontacts = __importStar(require("./ssmcontacts"));
exports.ssmcontacts = ssmcontacts;
const ssmincidents = __importStar(require("./ssmincidents"));
exports.ssmincidents = ssmincidents;
const ssoadmin = __importStar(require("./ssoadmin"));
exports.ssoadmin = ssoadmin;
const storagegateway = __importStar(require("./storagegateway"));
exports.storagegateway = storagegateway;
const swf = __importStar(require("./swf"));
exports.swf = swf;
const synthetics = __importStar(require("./synthetics"));
exports.synthetics = synthetics;
const timestreaminfluxdb = __importStar(require("./timestreaminfluxdb"));
exports.timestreaminfluxdb = timestreaminfluxdb;
const timestreamquery = __importStar(require("./timestreamquery"));
exports.timestreamquery = timestreamquery;
const timestreamwrite = __importStar(require("./timestreamwrite"));
exports.timestreamwrite = timestreamwrite;
const transcribe = __importStar(require("./transcribe"));
exports.transcribe = transcribe;
const transfer = __importStar(require("./transfer"));
exports.transfer = transfer;
const types = __importStar(require("./types"));
exports.types = types;
const uxc = __importStar(require("./uxc"));
exports.uxc = uxc;
const verifiedaccess = __importStar(require("./verifiedaccess"));
exports.verifiedaccess = verifiedaccess;
const verifiedpermissions = __importStar(require("./verifiedpermissions"));
exports.verifiedpermissions = verifiedpermissions;
const vpc = __importStar(require("./vpc"));
exports.vpc = vpc;
const vpclattice = __importStar(require("./vpclattice"));
exports.vpclattice = vpclattice;
const vpn = __importStar(require("./vpn"));
exports.vpn = vpn;
const waf = __importStar(require("./waf"));
exports.waf = waf;
const wafregional = __importStar(require("./wafregional"));
exports.wafregional = wafregional;
const wafv2 = __importStar(require("./wafv2"));
exports.wafv2 = wafv2;
const workmail = __importStar(require("./workmail"));
exports.workmail = workmail;
const workspaces = __importStar(require("./workspaces"));
exports.workspaces = workspaces;
const workspacesweb = __importStar(require("./workspacesweb"));
exports.workspacesweb = workspacesweb;
const xray = __importStar(require("./xray"));
exports.xray = xray;
pulumi.runtime.registerResourcePackage("aws", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:aws") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new provider_1.Provider(name, undefined, { urn });
    },
});
