// @pulumi/azure-native/provider.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("./utilities"));
/**
 * The provider type for the native Azure package.
 */
class Provider extends pulumi.ProviderResource {
    /** @internal */
    static __pulumiType = 'azure-native';
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === "pulumi:providers:" + Provider.__pulumiType;
    }
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        {
            resourceInputs["auxiliaryTenantIds"] = pulumi.output(args?.auxiliaryTenantIds).apply(JSON.stringify);
            resourceInputs["clientCertificatePassword"] = args?.clientCertificatePassword ? pulumi.secret(args.clientCertificatePassword) : undefined;
            resourceInputs["clientCertificatePath"] = args?.clientCertificatePath;
            resourceInputs["clientId"] = args?.clientId ? pulumi.secret(args.clientId) : undefined;
            resourceInputs["clientSecret"] = args?.clientSecret ? pulumi.secret(args.clientSecret) : undefined;
            resourceInputs["disableInstanceDiscovery"] = pulumi.output(args?.disableInstanceDiscovery).apply(JSON.stringify);
            resourceInputs["disablePulumiPartnerId"] = pulumi.output(args?.disablePulumiPartnerId).apply(JSON.stringify);
            resourceInputs["environment"] = (args?.environment) ?? "public";
            resourceInputs["location"] = args?.location;
            resourceInputs["metadataHost"] = args?.metadataHost;
            resourceInputs["msiEndpoint"] = args?.msiEndpoint;
            resourceInputs["oidcRequestToken"] = args?.oidcRequestToken;
            resourceInputs["oidcRequestUrl"] = args?.oidcRequestUrl;
            resourceInputs["oidcToken"] = args?.oidcToken;
            resourceInputs["oidcTokenFilePath"] = args?.oidcTokenFilePath;
            resourceInputs["partnerId"] = args?.partnerId;
            resourceInputs["subscriptionId"] = args?.subscriptionId;
            resourceInputs["tenantId"] = args?.tenantId;
            resourceInputs["useDefaultAzureCredential"] = pulumi.output(args?.useDefaultAzureCredential).apply(JSON.stringify);
            resourceInputs["useMsi"] = pulumi.output(args?.useMsi).apply(JSON.stringify);
            resourceInputs["useOidc"] = pulumi.output(args?.useOidc).apply(JSON.stringify);
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Provider = Provider;
