// @pulumi/azure-native/connectedcache/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IspCustomer = exports.IspCacheNodesOperation = exports.getIspCustomerOutput = exports.getIspCustomer = exports.getIspCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput = exports.getIspCacheNodesOperationCacheNodeMccIssueDetailsHistory = exports.getIspCacheNodesOperationCacheNodeInstallDetailsOutput = exports.getIspCacheNodesOperationCacheNodeInstallDetails = exports.getIspCacheNodesOperationCacheNodeAutoUpdateHistoryOutput = exports.getIspCacheNodesOperationCacheNodeAutoUpdateHistory = exports.getIspCacheNodesOperationBgpCidrsOutput = exports.getIspCacheNodesOperationBgpCidrs = exports.getIspCacheNodesOperationOutput = exports.getIspCacheNodesOperation = exports.getEnterpriseMccCustomerOutput = exports.getEnterpriseMccCustomer = exports.getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistoryOutput = exports.getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistory = exports.getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput = exports.getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistory = exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput = exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails = exports.getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistoryOutput = exports.getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistory = exports.getEnterpriseMccCacheNodesOperationOutput = exports.getEnterpriseMccCacheNodesOperation = exports.getEnterpriseCustomerOperationOutput = exports.getEnterpriseCustomerOperation = exports.getCacheNodesOperationOutput = exports.getCacheNodesOperation = exports.EnterpriseMccCustomer = exports.EnterpriseMccCacheNodesOperation = exports.EnterpriseCustomerOperation = exports.CacheNodesOperation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CacheNodesOperation = null;
utilities.lazyLoad(exports, ["CacheNodesOperation"], () => require("./cacheNodesOperation"));
exports.EnterpriseCustomerOperation = null;
utilities.lazyLoad(exports, ["EnterpriseCustomerOperation"], () => require("./enterpriseCustomerOperation"));
exports.EnterpriseMccCacheNodesOperation = null;
utilities.lazyLoad(exports, ["EnterpriseMccCacheNodesOperation"], () => require("./enterpriseMccCacheNodesOperation"));
exports.EnterpriseMccCustomer = null;
utilities.lazyLoad(exports, ["EnterpriseMccCustomer"], () => require("./enterpriseMccCustomer"));
exports.getCacheNodesOperation = null;
exports.getCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getCacheNodesOperation", "getCacheNodesOperationOutput"], () => require("./getCacheNodesOperation"));
exports.getEnterpriseCustomerOperation = null;
exports.getEnterpriseCustomerOperationOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseCustomerOperation", "getEnterpriseCustomerOperationOutput"], () => require("./getEnterpriseCustomerOperation"));
exports.getEnterpriseMccCacheNodesOperation = null;
exports.getEnterpriseMccCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperation", "getEnterpriseMccCacheNodesOperationOutput"], () => require("./getEnterpriseMccCacheNodesOperation"));
exports.getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistory = null;
exports.getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistoryOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistory", "getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistoryOutput"], () => require("./getEnterpriseMccCacheNodesOperationCacheNodeAutoUpdateHistory"));
exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails = null;
exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails", "getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput"], () => require("./getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails"));
exports.getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistory = null;
exports.getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistory", "getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput"], () => require("./getEnterpriseMccCacheNodesOperationCacheNodeMccIssueDetailsHistory"));
exports.getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistory = null;
exports.getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistoryOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistory", "getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistoryOutput"], () => require("./getEnterpriseMccCacheNodesOperationCacheNodeTlsCertificateHistory"));
exports.getEnterpriseMccCustomer = null;
exports.getEnterpriseMccCustomerOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCustomer", "getEnterpriseMccCustomerOutput"], () => require("./getEnterpriseMccCustomer"));
exports.getIspCacheNodesOperation = null;
exports.getIspCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperation", "getIspCacheNodesOperationOutput"], () => require("./getIspCacheNodesOperation"));
exports.getIspCacheNodesOperationBgpCidrs = null;
exports.getIspCacheNodesOperationBgpCidrsOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationBgpCidrs", "getIspCacheNodesOperationBgpCidrsOutput"], () => require("./getIspCacheNodesOperationBgpCidrs"));
exports.getIspCacheNodesOperationCacheNodeAutoUpdateHistory = null;
exports.getIspCacheNodesOperationCacheNodeAutoUpdateHistoryOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationCacheNodeAutoUpdateHistory", "getIspCacheNodesOperationCacheNodeAutoUpdateHistoryOutput"], () => require("./getIspCacheNodesOperationCacheNodeAutoUpdateHistory"));
exports.getIspCacheNodesOperationCacheNodeInstallDetails = null;
exports.getIspCacheNodesOperationCacheNodeInstallDetailsOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationCacheNodeInstallDetails", "getIspCacheNodesOperationCacheNodeInstallDetailsOutput"], () => require("./getIspCacheNodesOperationCacheNodeInstallDetails"));
exports.getIspCacheNodesOperationCacheNodeMccIssueDetailsHistory = null;
exports.getIspCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationCacheNodeMccIssueDetailsHistory", "getIspCacheNodesOperationCacheNodeMccIssueDetailsHistoryOutput"], () => require("./getIspCacheNodesOperationCacheNodeMccIssueDetailsHistory"));
exports.getIspCustomer = null;
exports.getIspCustomerOutput = null;
utilities.lazyLoad(exports, ["getIspCustomer", "getIspCustomerOutput"], () => require("./getIspCustomer"));
exports.IspCacheNodesOperation = null;
utilities.lazyLoad(exports, ["IspCacheNodesOperation"], () => require("./ispCacheNodesOperation"));
exports.IspCustomer = null;
utilities.lazyLoad(exports, ["IspCustomer"], () => require("./ispCustomer"));
// Export enums:
__exportStar(require("../types/enums/connectedcache"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:connectedcache:CacheNodesOperation":
                return new exports.CacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache:EnterpriseCustomerOperation":
                return new exports.EnterpriseCustomerOperation(name, undefined, { urn });
            case "azure-native:connectedcache:EnterpriseMccCacheNodesOperation":
                return new exports.EnterpriseMccCacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache:EnterpriseMccCustomer":
                return new exports.EnterpriseMccCustomer(name, undefined, { urn });
            case "azure-native:connectedcache:IspCacheNodesOperation":
                return new exports.IspCacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache:IspCustomer":
                return new exports.IspCustomer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "connectedcache", _module);
