// @pulumi/azure-native/billingbenefits/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Source = exports.Macc = exports.getSourceOutput = exports.getSource = exports.getMaccOutput = exports.getMacc = exports.getFreeServiceOutput = exports.getFreeService = exports.getDiscountOutput = exports.getDiscount = exports.getCreditOutput = exports.getCredit = exports.getConditionalCreditOutput = exports.getConditionalCredit = exports.FreeService = exports.Discount = exports.Credit = exports.ConditionalCredit = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ConditionalCredit = null;
utilities.lazyLoad(exports, ["ConditionalCredit"], () => require("./conditionalCredit"));
exports.Credit = null;
utilities.lazyLoad(exports, ["Credit"], () => require("./credit"));
exports.Discount = null;
utilities.lazyLoad(exports, ["Discount"], () => require("./discount"));
exports.FreeService = null;
utilities.lazyLoad(exports, ["FreeService"], () => require("./freeService"));
exports.getConditionalCredit = null;
exports.getConditionalCreditOutput = null;
utilities.lazyLoad(exports, ["getConditionalCredit", "getConditionalCreditOutput"], () => require("./getConditionalCredit"));
exports.getCredit = null;
exports.getCreditOutput = null;
utilities.lazyLoad(exports, ["getCredit", "getCreditOutput"], () => require("./getCredit"));
exports.getDiscount = null;
exports.getDiscountOutput = null;
utilities.lazyLoad(exports, ["getDiscount", "getDiscountOutput"], () => require("./getDiscount"));
exports.getFreeService = null;
exports.getFreeServiceOutput = null;
utilities.lazyLoad(exports, ["getFreeService", "getFreeServiceOutput"], () => require("./getFreeService"));
exports.getMacc = null;
exports.getMaccOutput = null;
utilities.lazyLoad(exports, ["getMacc", "getMaccOutput"], () => require("./getMacc"));
exports.getSource = null;
exports.getSourceOutput = null;
utilities.lazyLoad(exports, ["getSource", "getSourceOutput"], () => require("./getSource"));
exports.Macc = null;
utilities.lazyLoad(exports, ["Macc"], () => require("./macc"));
exports.Source = null;
utilities.lazyLoad(exports, ["Source"], () => require("./source"));
// Export enums:
__exportStar(require("../types/enums/billingbenefits"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:billingbenefits:ConditionalCredit":
                return new exports.ConditionalCredit(name, undefined, { urn });
            case "azure-native:billingbenefits:Credit":
                return new exports.Credit(name, undefined, { urn });
            case "azure-native:billingbenefits:Discount":
                return new exports.Discount(name, undefined, { urn });
            case "azure-native:billingbenefits:FreeService":
                return new exports.FreeService(name, undefined, { urn });
            case "azure-native:billingbenefits:Macc":
                return new exports.Macc(name, undefined, { urn });
            case "azure-native:billingbenefits:Source":
                return new exports.Source(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "billingbenefits", _module);
