// @pulumi/aws/bedrock/agentcoreOauth2CredentialProvider.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentcoreOauth2CredentialProvider = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS Bedrock AgentCore OAuth2 Credential Provider. OAuth2 credential providers enable secure authentication with external OAuth2/OpenID Connect identity providers for agent runtimes.
 *
 * > **Note:** Write-Only arguments `clientIdWo` and `clientSecretWo` are available to use in place of `clientId` and `clientSecret`. Write-Only arguments are supported in HashiCorp Terraform 1.11.0 and later. Learn more.
 *
 * ## Example Usage
 *
 * ### GitHub OAuth Provider
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const github = new aws.bedrock.AgentcoreOauth2CredentialProvider("github", {
 *     name: "github-oauth-provider",
 *     credentialProviderVendor: "GithubOauth2",
 *     oauth2ProviderConfig: {
 *         githubOauth2ProviderConfig: {
 *             clientId: "your-github-client-id",
 *             clientSecret: "your-github-client-secret",
 *         },
 *     },
 * });
 * ```
 *
 * ### Custom OAuth Provider with Discovery URL
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const auth0 = new aws.bedrock.AgentcoreOauth2CredentialProvider("auth0", {
 *     name: "auth0-oauth-provider",
 *     credentialProviderVendor: "CustomOauth2",
 *     customOauth2ProviderConfig: [{
 *         custom: [{
 *             clientIdWo: "auth0-client-id",
 *             clientSecretWo: "auth0-client-secret",
 *             clientCredentialsWoVersion: 1,
 *             oauthDiscovery: [{
 *                 discoveryUrl: "https://dev-company.auth0.com/.well-known/openid-configuration",
 *             }],
 *         }],
 *     }],
 * });
 * ```
 *
 * ### Custom OAuth Provider with Authorization Server Metadata
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const keycloak = new aws.bedrock.AgentcoreOauth2CredentialProvider("keycloak", {
 *     name: "keycloak-oauth-provider",
 *     credentialProviderVendor: "CustomOauth2",
 *     oauth2ProviderConfig: {
 *         customOauth2ProviderConfig: {
 *             clientIdWo: "keycloak-client-id",
 *             clientSecretWo: "keycloak-client-secret",
 *             clientCredentialsWoVersion: 1,
 *             oauthDiscovery: {
 *                 authorizationServerMetadata: {
 *                     issuer: "https://auth.company.com/realms/production",
 *                     authorizationEndpoint: "https://auth.company.com/realms/production/protocol/openid-connect/auth",
 *                     tokenEndpoint: "https://auth.company.com/realms/production/protocol/openid-connect/token",
 *                     responseTypes: [
 *                         "code",
 *                         "id_token",
 *                     ],
 *                 },
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Bedrock AgentCore OAuth2 Credential Provider using the provider name. For example:
 *
 * ```sh
 * $ pulumi import aws:bedrock/agentcoreOauth2CredentialProvider:AgentcoreOauth2CredentialProvider example oauth2-provider-name
 * ```
 */
class AgentcoreOauth2CredentialProvider extends pulumi.CustomResource {
    /**
     * Get an existing AgentcoreOauth2CredentialProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AgentcoreOauth2CredentialProvider(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:bedrock/agentcoreOauth2CredentialProvider:AgentcoreOauth2CredentialProvider';
    /**
     * Returns true if the given object is an instance of AgentcoreOauth2CredentialProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AgentcoreOauth2CredentialProvider.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["clientSecretArns"] = state?.clientSecretArns;
            resourceInputs["credentialProviderArn"] = state?.credentialProviderArn;
            resourceInputs["credentialProviderVendor"] = state?.credentialProviderVendor;
            resourceInputs["name"] = state?.name;
            resourceInputs["oauth2ProviderConfig"] = state?.oauth2ProviderConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.credentialProviderVendor === undefined && !opts.urn) {
                throw new Error("Missing required property 'credentialProviderVendor'");
            }
            resourceInputs["credentialProviderVendor"] = args?.credentialProviderVendor;
            resourceInputs["name"] = args?.name;
            resourceInputs["oauth2ProviderConfig"] = args?.oauth2ProviderConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["clientSecretArns"] = undefined /*out*/;
            resourceInputs["credentialProviderArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AgentcoreOauth2CredentialProvider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AgentcoreOauth2CredentialProvider = AgentcoreOauth2CredentialProvider;
