// @pulumi/gcp/gkeonprem/bareMetalAdminCluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BareMetalAdminCluster = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A Google Bare Metal Admin Cluster.
 *
 * To get more information about BareMetalAdminCluster, see:
 *
 * * [API documentation](https://cloud.google.com/kubernetes-engine/distributed-cloud/reference/on-prem-api/rest/v1/projects.locations.bareMetalAdminClusters)
 *
 * ## Example Usage
 *
 * ### Gkeonprem Bare Metal Admin Cluster Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const admin_cluster_basic = new gcp.gkeonprem.BareMetalAdminCluster("admin-cluster-basic", {
 *     name: "my-cluster",
 *     location: "us-west1",
 *     bareMetalVersion: "1.13.4",
 *     networkConfig: {
 *         islandModeCidr: {
 *             serviceAddressCidrBlocks: ["172.26.0.0/16"],
 *             podAddressCidrBlocks: ["10.240.0.0/13"],
 *         },
 *     },
 *     nodeConfig: {
 *         maxPodsPerNode: 250,
 *     },
 *     controlPlane: {
 *         controlPlaneNodePoolConfig: {
 *             nodePoolConfig: {
 *                 labels: {},
 *                 operatingSystem: "LINUX",
 *                 nodeConfigs: [
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.2",
 *                     },
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.3",
 *                     },
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.4",
 *                     },
 *                 ],
 *             },
 *         },
 *     },
 *     loadBalancer: {
 *         portConfig: {
 *             controlPlaneLoadBalancerPort: 443,
 *         },
 *         vipConfig: {
 *             controlPlaneVip: "10.200.0.5",
 *         },
 *     },
 *     storage: {
 *         lvpShareConfig: {
 *             lvpConfig: {
 *                 path: "/mnt/localpv-share",
 *                 storageClass: "local-shared",
 *             },
 *             sharedPathPvCount: 5,
 *         },
 *         lvpNodeMountsConfig: {
 *             path: "/mnt/localpv-disk",
 *             storageClass: "local-disks",
 *         },
 *     },
 *     nodeAccessConfig: {
 *         loginUser: "root",
 *     },
 * });
 * ```
 * ### Gkeonprem Bare Metal Admin Cluster Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const admin_cluster_basic = new gcp.gkeonprem.BareMetalAdminCluster("admin-cluster-basic", {
 *     name: "my-cluster",
 *     location: "us-west1",
 *     description: "test description",
 *     bareMetalVersion: "1.13.4",
 *     annotations: {
 *         env: "test",
 *     },
 *     networkConfig: {
 *         islandModeCidr: {
 *             serviceAddressCidrBlocks: ["172.26.0.0/16"],
 *             podAddressCidrBlocks: ["10.240.0.0/13"],
 *         },
 *         advancedNetworking: true,
 *         multipleNetworkInterfacesConfig: {
 *             enabled: true,
 *         },
 *     },
 *     nodeConfig: {
 *         maxPodsPerNode: 250,
 *     },
 *     controlPlane: {
 *         controlPlaneNodePoolConfig: {
 *             nodePoolConfig: {
 *                 labels: {},
 *                 operatingSystem: "LINUX",
 *                 nodeConfigs: [
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.2",
 *                     },
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.3",
 *                     },
 *                     {
 *                         labels: {},
 *                         nodeIp: "10.200.0.4",
 *                     },
 *                 ],
 *                 taints: [{
 *                     key: "test-key",
 *                     value: "test-value",
 *                     effect: "NO_EXECUTE",
 *                 }],
 *             },
 *         },
 *         apiServerArgs: [{
 *             argument: "test argument",
 *             value: "test value",
 *         }],
 *     },
 *     loadBalancer: {
 *         portConfig: {
 *             controlPlaneLoadBalancerPort: 443,
 *         },
 *         vipConfig: {
 *             controlPlaneVip: "10.200.0.5",
 *         },
 *         bgpLbConfig: {
 *             asn: 123456,
 *             bgpPeerConfigs: [{
 *                 asn: 123457,
 *                 ipAddress: "10.0.0.1",
 *                 controlPlaneNodes: ["test-node"],
 *             }],
 *             addressPools: [{
 *                 pool: "loadBalancerAddressPool-1",
 *                 addresses: [
 *                     "10.200.0.14/32",
 *                     "10.200.0.15/32",
 *                     "10.200.0.16/32",
 *                     "10.200.0.17/32",
 *                     "10.200.0.18/32",
 *                     "fd00:1::f/128",
 *                     "fd00:1::10/128",
 *                     "fd00:1::11/128",
 *                     "fd00:1::12/128",
 *                 ],
 *                 manualAssign: true,
 *                 avoidBuggyIps: true,
 *             }],
 *             loadBalancerNodePoolConfig: {
 *                 nodePoolConfig: {
 *                     labels: {},
 *                     operatingSystem: "LINUX",
 *                     nodeConfigs: [{
 *                         labels: {},
 *                         nodeIp: "10.200.0.9",
 *                     }],
 *                     kubeletConfig: {
 *                         registryBurst: 12,
 *                         registryPullQps: 10,
 *                         serializeImagePullsDisabled: true,
 *                     },
 *                     taints: [{
 *                         key: "test-key",
 *                         value: "test-value",
 *                         effect: "NO_EXECUTE",
 *                     }],
 *                 },
 *             },
 *         },
 *     },
 *     storage: {
 *         lvpShareConfig: {
 *             lvpConfig: {
 *                 path: "/mnt/localpv-share",
 *                 storageClass: "local-shared",
 *             },
 *             sharedPathPvCount: 5,
 *         },
 *         lvpNodeMountsConfig: {
 *             path: "/mnt/localpv-disk",
 *             storageClass: "local-disks",
 *         },
 *     },
 *     nodeAccessConfig: {
 *         loginUser: "root",
 *     },
 *     securityConfig: {
 *         authorization: {
 *             adminUsers: [{
 *                 username: "admin@hashicorptest.com",
 *             }],
 *         },
 *     },
 *     maintenanceConfig: {
 *         maintenanceAddressCidrBlocks: [
 *             "10.0.0.1/32",
 *             "10.0.0.2/32",
 *         ],
 *     },
 *     clusterOperations: {
 *         enableApplicationLogs: true,
 *     },
 *     proxy: {
 *         uri: "test proxy uri",
 *         noProxies: ["127.0.0.1"],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * BareMetalAdminCluster can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/bareMetalAdminClusters/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, BareMetalAdminCluster can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:gkeonprem/bareMetalAdminCluster:BareMetalAdminCluster default projects/{{project}}/locations/{{location}}/bareMetalAdminClusters/{{name}}
 * $ pulumi import gcp:gkeonprem/bareMetalAdminCluster:BareMetalAdminCluster default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:gkeonprem/bareMetalAdminCluster:BareMetalAdminCluster default {{location}}/{{name}}
 * ```
 */
class BareMetalAdminCluster extends pulumi.CustomResource {
    /**
     * Get an existing BareMetalAdminCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BareMetalAdminCluster(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:gkeonprem/bareMetalAdminCluster:BareMetalAdminCluster';
    /**
     * Returns true if the given object is an instance of BareMetalAdminCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BareMetalAdminCluster.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["annotations"] = state?.annotations;
            resourceInputs["bareMetalVersion"] = state?.bareMetalVersion;
            resourceInputs["clusterOperations"] = state?.clusterOperations;
            resourceInputs["controlPlane"] = state?.controlPlane;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deleteTime"] = state?.deleteTime;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveAnnotations"] = state?.effectiveAnnotations;
            resourceInputs["endpoint"] = state?.endpoint;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["fleets"] = state?.fleets;
            resourceInputs["loadBalancer"] = state?.loadBalancer;
            resourceInputs["localName"] = state?.localName;
            resourceInputs["location"] = state?.location;
            resourceInputs["maintenanceConfig"] = state?.maintenanceConfig;
            resourceInputs["name"] = state?.name;
            resourceInputs["networkConfig"] = state?.networkConfig;
            resourceInputs["nodeAccessConfig"] = state?.nodeAccessConfig;
            resourceInputs["nodeConfig"] = state?.nodeConfig;
            resourceInputs["project"] = state?.project;
            resourceInputs["proxy"] = state?.proxy;
            resourceInputs["reconciling"] = state?.reconciling;
            resourceInputs["securityConfig"] = state?.securityConfig;
            resourceInputs["state"] = state?.state;
            resourceInputs["statuses"] = state?.statuses;
            resourceInputs["storage"] = state?.storage;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["validationChecks"] = state?.validationChecks;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["annotations"] = args?.annotations;
            resourceInputs["bareMetalVersion"] = args?.bareMetalVersion;
            resourceInputs["clusterOperations"] = args?.clusterOperations;
            resourceInputs["controlPlane"] = args?.controlPlane;
            resourceInputs["description"] = args?.description;
            resourceInputs["loadBalancer"] = args?.loadBalancer;
            resourceInputs["location"] = args?.location;
            resourceInputs["maintenanceConfig"] = args?.maintenanceConfig;
            resourceInputs["name"] = args?.name;
            resourceInputs["networkConfig"] = args?.networkConfig;
            resourceInputs["nodeAccessConfig"] = args?.nodeAccessConfig;
            resourceInputs["nodeConfig"] = args?.nodeConfig;
            resourceInputs["project"] = args?.project;
            resourceInputs["proxy"] = args?.proxy;
            resourceInputs["securityConfig"] = args?.securityConfig;
            resourceInputs["storage"] = args?.storage;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["deleteTime"] = undefined /*out*/;
            resourceInputs["effectiveAnnotations"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fleets"] = undefined /*out*/;
            resourceInputs["localName"] = undefined /*out*/;
            resourceInputs["reconciling"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["statuses"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
            resourceInputs["validationChecks"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BareMetalAdminCluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BareMetalAdminCluster = BareMetalAdminCluster;
