// @pulumi/aws/ec2clientvpn/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Route = exports.NetworkAssociation = exports.getEndpointOutput = exports.getEndpoint = exports.Endpoint = exports.AuthorizationRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AuthorizationRule = null;
utilities.lazyLoad(exports, ["AuthorizationRule"], () => require("./authorizationRule"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.NetworkAssociation = null;
utilities.lazyLoad(exports, ["NetworkAssociation"], () => require("./networkAssociation"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ec2clientvpn/authorizationRule:AuthorizationRule":
                return new exports.AuthorizationRule(name, undefined, { urn });
            case "aws:ec2clientvpn/endpoint:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "aws:ec2clientvpn/networkAssociation:NetworkAssociation":
                return new exports.NetworkAssociation(name, undefined, { urn });
            case "aws:ec2clientvpn/route:Route":
                return new exports.Route(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/authorizationRule", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/endpoint", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/networkAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "ec2clientvpn/route", _module);
