// @pulumi/aws/apigateway/restApiPut.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestApiPut = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for updating an AWS API Gateway REST API with a new API description.
 *
 * > **NOTE:** When importing an OpenAPI specification using the `body` argument, the `info.title` field updates the `name` of the `aws.apigateway.RestApi`. If the imported `title` differs from the configured `name`, Terraform will report the difference.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apigateway.RestApiPut("example", {
 *     body: JSON.stringify({
 *         swagger: "2.0",
 *         info: {
 *             title: "Example API",
 *             version: "v1",
 *         },
 *         schemes: ["https"],
 *         paths: {
 *             "/example": {
 *                 get: {
 *                     responses: {
 *                         "200": {
 *                             description: "OK",
 *                         },
 *                     },
 *                     "x-amazon-apigateway-integration": {
 *                         httpMethod: "GET",
 *                         type: "HTTP",
 *                         responses: {
 *                             "default": {
 *                                 statusCode: 200,
 *                             },
 *                         },
 *                         uri: "https://api.example.com/",
 *                     },
 *                 },
 *             },
 *         },
 *     }),
 *     failOnWarnings: true,
 *     restApiId: exampleAwsApiGatewayRestApi.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import API Gateway REST API Put using the `restApiId`. For example:
 *
 * ```sh
 * $ pulumi import aws:apigateway/restApiPut:RestApiPut example import-id-12345678
 * ```
 */
class RestApiPut extends pulumi.CustomResource {
    /**
     * Get an existing RestApiPut resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RestApiPut(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:apigateway/restApiPut:RestApiPut';
    /**
     * Returns true if the given object is an instance of RestApiPut.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RestApiPut.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["body"] = state?.body;
            resourceInputs["failOnWarnings"] = state?.failOnWarnings;
            resourceInputs["parameters"] = state?.parameters;
            resourceInputs["region"] = state?.region;
            resourceInputs["restApiId"] = state?.restApiId;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["triggers"] = state?.triggers;
        }
        else {
            const args = argsOrState;
            if (args?.body === undefined && !opts.urn) {
                throw new Error("Missing required property 'body'");
            }
            if (args?.restApiId === undefined && !opts.urn) {
                throw new Error("Missing required property 'restApiId'");
            }
            resourceInputs["body"] = args?.body;
            resourceInputs["failOnWarnings"] = args?.failOnWarnings;
            resourceInputs["parameters"] = args?.parameters;
            resourceInputs["region"] = args?.region;
            resourceInputs["restApiId"] = args?.restApiId;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["triggers"] = args?.triggers;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RestApiPut.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RestApiPut = RestApiPut;
