// @pulumi/aws/globalaccelerator/endpointGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a Global Accelerator endpoint group.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.globalaccelerator.EndpointGroup("example", {
 *     listenerArn: exampleAwsGlobalacceleratorListener.arn,
 *     endpointConfigurations: [{
 *         endpointId: exampleAwsLb.arn,
 *         weight: 100,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `arn` (String) Amazon Resource Name (ARN) of the Global Accelerator endpoint group.
 *
 * Using `pulumi import`, import Global Accelerator endpoint groups using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:globalaccelerator/endpointGroup:EndpointGroup example arn:aws:globalaccelerator::111111111111:accelerator/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/listener/xxxxxxx/endpoint-group/xxxxxxxx
 * ```
 */
class EndpointGroup extends pulumi.CustomResource {
    /**
     * Get an existing EndpointGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointGroup(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:globalaccelerator/endpointGroup:EndpointGroup';
    /**
     * Returns true if the given object is an instance of EndpointGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointGroup.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["endpointConfigurations"] = state?.endpointConfigurations;
            resourceInputs["endpointGroupRegion"] = state?.endpointGroupRegion;
            resourceInputs["healthCheckIntervalSeconds"] = state?.healthCheckIntervalSeconds;
            resourceInputs["healthCheckPath"] = state?.healthCheckPath;
            resourceInputs["healthCheckPort"] = state?.healthCheckPort;
            resourceInputs["healthCheckProtocol"] = state?.healthCheckProtocol;
            resourceInputs["listenerArn"] = state?.listenerArn;
            resourceInputs["portOverrides"] = state?.portOverrides;
            resourceInputs["thresholdCount"] = state?.thresholdCount;
            resourceInputs["trafficDialPercentage"] = state?.trafficDialPercentage;
        }
        else {
            const args = argsOrState;
            if (args?.listenerArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'listenerArn'");
            }
            resourceInputs["endpointConfigurations"] = args?.endpointConfigurations;
            resourceInputs["endpointGroupRegion"] = args?.endpointGroupRegion;
            resourceInputs["healthCheckIntervalSeconds"] = args?.healthCheckIntervalSeconds;
            resourceInputs["healthCheckPath"] = args?.healthCheckPath;
            resourceInputs["healthCheckPort"] = args?.healthCheckPort;
            resourceInputs["healthCheckProtocol"] = args?.healthCheckProtocol;
            resourceInputs["listenerArn"] = args?.listenerArn;
            resourceInputs["portOverrides"] = args?.portOverrides;
            resourceInputs["thresholdCount"] = args?.thresholdCount;
            resourceInputs["trafficDialPercentage"] = args?.trafficDialPercentage;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EndpointGroup = EndpointGroup;
