// @pulumi/gcp/endpoints/service.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * This resource creates and rolls out a Cloud Endpoints service using OpenAPI or gRPC.  View the relevant docs for [OpenAPI](https://cloud.google.com/endpoints/docs/openapi/) and [gRPC](https://cloud.google.com/endpoints/docs/grpc/).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const openapiService = new gcp.endpoints.Service("openapi_service", {
 *     serviceName: "api-name.endpoints.project-id.cloud.goog",
 *     project: "project-id",
 *     openapiConfig: std.file({
 *         input: "openapi_spec.yml",
 *     }).then(invoke => invoke.result),
 * });
 * const grpcService = new gcp.endpoints.Service("grpc_service", {
 *     serviceName: "api-name.endpoints.project-id.cloud.goog",
 *     project: "project-id",
 *     grpcConfig: std.file({
 *         input: "service_spec.yml",
 *     }).then(invoke => invoke.result),
 *     protocOutputBase64: std.filebase64({
 *         input: "compiled_descriptor_file.pb",
 *     }).then(invoke => invoke.result),
 * });
 * ```
 *
 * The example in `examples/endpoints_on_compute_engine` shows the API from the quickstart running on a Compute Engine VM and reachable through Cloud Endpoints, which may also be useful.
 *
 * ## Import
 *
 * This resource does not support import.
 */
class Service extends pulumi.CustomResource {
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Service(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:endpoints/service:Service';
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["apis"] = state?.apis;
            resourceInputs["configId"] = state?.configId;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["dnsAddress"] = state?.dnsAddress;
            resourceInputs["endpoints"] = state?.endpoints;
            resourceInputs["grpcConfig"] = state?.grpcConfig;
            resourceInputs["openapiConfig"] = state?.openapiConfig;
            resourceInputs["project"] = state?.project;
            resourceInputs["protocOutputBase64"] = state?.protocOutputBase64;
            resourceInputs["serviceName"] = state?.serviceName;
        }
        else {
            const args = argsOrState;
            if (args?.serviceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["grpcConfig"] = args?.grpcConfig;
            resourceInputs["openapiConfig"] = args?.openapiConfig;
            resourceInputs["project"] = args?.project;
            resourceInputs["protocOutputBase64"] = args?.protocOutputBase64;
            resourceInputs["serviceName"] = args?.serviceName;
            resourceInputs["apis"] = undefined /*out*/;
            resourceInputs["configId"] = undefined /*out*/;
            resourceInputs["dnsAddress"] = undefined /*out*/;
            resourceInputs["endpoints"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Service.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Service = Service;
