// @pulumi/azure-native/types/enums/monitor/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReceiverType = exports.PublicNetworkAccess = exports.ProcessorType = exports.PrivateEndpointServiceConnectionStatus = exports.PredictiveAutoscalePolicyScaleMode = exports.PipelineType = exports.Operator = exports.OperationType = exports.Odatatype = exports.MetricStatisticType = exports.MetricAggregationType = exports.ManagedServiceIdentityType = exports.KnownWindowsEventLogDataSourceStreams = exports.KnownSyslogDataSourceStreams = exports.KnownSyslogDataSourceLogLevels = exports.KnownSyslogDataSourceFacilityNames = exports.KnownPublicNetworkAccessOptions = exports.KnownPrometheusForwarderDataSourceStreams = exports.KnownPerfCounterDataSourceStreams = exports.KnownLogFilesDataSourceFormat = exports.KnownLogFileTextSettingsRecordStartTimestampFormat = exports.KnownExtensionDataSourceStreams = exports.KnownDataFlowStreams = exports.KnownDataCollectionRuleResourceKind = exports.KnownDataCollectionEndpointResourceKind = exports.KnownColumnDefinitionType = exports.Kind = exports.JsonMapperElement = exports.IncidentManagementService = exports.IdentityType = exports.ExternalNetworkingMode = exports.ExtendedLocationType = exports.ExporterType = exports.EntityImpact = exports.DynamicThresholdSensitivity = exports.DynamicThresholdOperator = exports.DynamicThresholdModel = exports.DynamicThresholdDirection = exports.DiscoveryRuleRelationshipDiscoveryBehavior = exports.DiscoveryRuleRecommendedSignalsBehavior = exports.DiscoveryRuleKind = exports.DimensionOperator = exports.DependenciesAggregationType = exports.CriterionType = exports.ConditionOperator = exports.ComparisonOperationType = exports.AuthenticationKind = exports.AlertSeverity = exports.AggregationTypeEnum = exports.AccessMode = void 0;
exports.TimeAggregationType = exports.TimeAggregation = exports.SyslogProtocol = exports.StreamEncodingType = exports.Status = exports.SignalOperator = exports.SignalKind = exports.ScopedResourceKind = exports.ScaleType = exports.ScaleRuleMetricDimensionOperationType = exports.ScaleDirection = exports.RefreshInterval = exports.RecurrenceFrequency = void 0;
exports.AccessMode = {
    Open: "Open",
    PrivateOnly: "PrivateOnly",
};
exports.AggregationTypeEnum = {
    Average: "Average",
    Count: "Count",
    Minimum: "Minimum",
    Maximum: "Maximum",
    Total: "Total",
};
exports.AlertSeverity = {
    /**
     * Critical
     */
    Sev0: "Sev0",
    /**
     * Error
     */
    Sev1: "Sev1",
    /**
     * Warning
     */
    Sev2: "Sev2",
    /**
     * Informational
     */
    Sev3: "Sev3",
    /**
     * Verbose
     */
    Sev4: "Sev4",
};
exports.AuthenticationKind = {
    ManagedIdentity: "ManagedIdentity",
};
exports.ComparisonOperationType = {
    Equals: "Equals",
    NotEquals: "NotEquals",
    GreaterThan: "GreaterThan",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    LessThan: "LessThan",
    LessThanOrEqual: "LessThanOrEqual",
};
exports.ConditionOperator = {
    Equals: "Equals",
    GreaterThan: "GreaterThan",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    LessThan: "LessThan",
    LessThanOrEqual: "LessThanOrEqual",
    GreaterOrLessThan: "GreaterOrLessThan",
};
exports.CriterionType = {
    StaticThresholdCriterion: "StaticThresholdCriterion",
    DynamicThresholdCriterion: "DynamicThresholdCriterion",
};
exports.DependenciesAggregationType = {
    /**
     * Default behavior: Worst child health state is propagated.
     */
    WorstOf: "WorstOf",
    /**
     * Based on configurable thresholds.
     */
    Thresholds: "Thresholds",
};
exports.DimensionOperator = {
    Include: "Include",
    Exclude: "Exclude",
};
exports.DiscoveryRuleKind = {
    ResourceGraphQuery: "ResourceGraphQuery",
    ApplicationInsightsTopology: "ApplicationInsightsTopology",
};
exports.DiscoveryRuleRecommendedSignalsBehavior = {
    /**
     * Automatically add recommended signals
     */
    Enabled: "Enabled",
    /**
     * Do not automatically add recommended signals
     */
    Disabled: "Disabled",
};
exports.DiscoveryRuleRelationshipDiscoveryBehavior = {
    /**
     * Automatically attempt to discover relationships
     */
    Enabled: "Enabled",
    /**
     * Do not automatically attempt to discover relationships
     */
    Disabled: "Disabled",
};
exports.DynamicThresholdDirection = {
    /**
     * Lower than
     */
    LowerThan: "LowerThan",
    /**
     * Greater than
     */
    GreaterThan: "GreaterThan",
    /**
     * Greater or Lower Than
     */
    GreaterOrLowerThan: "GreaterOrLowerThan",
};
exports.DynamicThresholdModel = {
    /**
     * Anomaly detection model
     */
    AnomalyDetection: "AnomalyDetection",
};
exports.DynamicThresholdOperator = {
    GreaterThan: "GreaterThan",
    LessThan: "LessThan",
    GreaterOrLessThan: "GreaterOrLessThan",
};
exports.DynamicThresholdSensitivity = {
    Low: "Low",
    Medium: "Medium",
    High: "High",
};
exports.EntityImpact = {
    /**
     * Standard impact
     */
    Standard: "Standard",
    /**
     * Limited impact
     */
    Limited: "Limited",
    /**
     * Suppressed impact
     */
    Suppressed: "Suppressed",
};
exports.ExporterType = {
    /**
     * Export logs to Azure Monitor Workspace.
     */
    AzureMonitorWorkspaceLogs: "AzureMonitorWorkspaceLogs",
    /**
     * Export data to another pipeline group instance.
     */
    PipelineGroup: "PipelineGroup",
};
exports.ExtendedLocationType = {
    /**
     * Azure Edge Zones location type
     */
    EdgeZone: "EdgeZone",
    /**
     * Azure Custom Locations type
     */
    CustomLocation: "CustomLocation",
};
exports.ExternalNetworkingMode = {
    /**
     * Load balancer only.
     */
    LoadBalancerOnly: "LoadBalancerOnly",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.IncidentManagementService = {
    Icm: "Icm",
};
exports.JsonMapperElement = {
    /**
     * Read or write the json array from or to the body of the message.
     */
    Body: "body",
    /**
     * Read or write the json array from or to the attributes of the message.
     */
    Attributes: "attributes",
};
exports.Kind = {
    LogAlert: "LogAlert",
    SimpleLogAlert: "SimpleLogAlert",
    LogToMetric: "LogToMetric",
};
exports.KnownColumnDefinitionType = {
    String: "string",
    Int: "int",
    Long: "long",
    Real: "real",
    Boolean: "boolean",
    Datetime: "datetime",
    Dynamic: "dynamic",
};
exports.KnownDataCollectionEndpointResourceKind = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.KnownDataCollectionRuleResourceKind = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.KnownDataFlowStreams = {
    Microsoft_Event: "Microsoft-Event",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_Syslog: "Microsoft-Syslog",
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
};
exports.KnownExtensionDataSourceStreams = {
    Microsoft_Event: "Microsoft-Event",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_Syslog: "Microsoft-Syslog",
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
};
exports.KnownLogFileTextSettingsRecordStartTimestampFormat = {
    ISO_8601: "ISO 8601",
    YYYY_MM_DD_HH_MM_SS: "YYYY-MM-DD HH:MM:SS",
    M_D_YYYY_HH_MM_SS_AM_PM: "M/D/YYYY HH:MM:SS AM/PM",
    Mon_DD_YYYY_HH_MM_SS: "Mon DD, YYYY HH:MM:SS",
    YyMMdd_HH_mm_ss: "yyMMdd HH:mm:ss",
    DdMMyy_HH_mm_ss: "ddMMyy HH:mm:ss",
    MMM_d_hh_mm_ss: "MMM d hh:mm:ss",
    Dd_MMM_yyyy_HH_mm_ss_zzz: "dd/MMM/yyyy:HH:mm:ss zzz",
    Yyyy_MM_ddTHH_mm_ssK: "yyyy-MM-ddTHH:mm:ssK",
};
exports.KnownLogFilesDataSourceFormat = {
    Text: "text",
};
exports.KnownPerfCounterDataSourceStreams = {
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
};
exports.KnownPrometheusForwarderDataSourceStreams = {
    Microsoft_PrometheusMetrics: "Microsoft-PrometheusMetrics",
};
exports.KnownPublicNetworkAccessOptions = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    SecuredByPerimeter: "SecuredByPerimeter",
};
exports.KnownSyslogDataSourceFacilityNames = {
    Alert: "alert",
    Audit: "audit",
    Auth: "auth",
    Authpriv: "authpriv",
    Clock: "clock",
    Cron: "cron",
    Daemon: "daemon",
    Ftp: "ftp",
    Kern: "kern",
    Lpr: "lpr",
    Mail: "mail",
    Mark: "mark",
    News: "news",
    Nopri: "nopri",
    Ntp: "ntp",
    Syslog: "syslog",
    User: "user",
    Uucp: "uucp",
    Local0: "local0",
    Local1: "local1",
    Local2: "local2",
    Local3: "local3",
    Local4: "local4",
    Local5: "local5",
    Local6: "local6",
    Local7: "local7",
    Asterisk: "*",
};
exports.KnownSyslogDataSourceLogLevels = {
    Debug: "Debug",
    Info: "Info",
    Notice: "Notice",
    Warning: "Warning",
    Error: "Error",
    Critical: "Critical",
    Alert: "Alert",
    Emergency: "Emergency",
    Asterisk: "*",
};
exports.KnownSyslogDataSourceStreams = {
    Microsoft_Syslog: "Microsoft-Syslog",
};
exports.KnownWindowsEventLogDataSourceStreams = {
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
    Microsoft_Event: "Microsoft-Event",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.MetricAggregationType = {
    None: "None",
    Average: "Average",
    Count: "Count",
    Minimum: "Minimum",
    Maximum: "Maximum",
    Total: "Total",
};
exports.MetricStatisticType = {
    Average: "Average",
    Min: "Min",
    Max: "Max",
    Sum: "Sum",
    Count: "Count",
};
exports.Odatatype = {
    Microsoft_Azure_Monitor_SingleResourceMultipleMetricCriteria: "Microsoft.Azure.Monitor.SingleResourceMultipleMetricCriteria",
    Microsoft_Azure_Monitor_MultipleResourceMultipleMetricCriteria: "Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria",
    Microsoft_Azure_Monitor_WebtestLocationAvailabilityCriteria: "Microsoft.Azure.Monitor.WebtestLocationAvailabilityCriteria",
};
exports.OperationType = {
    Scale: "Scale",
};
exports.Operator = {
    Equals: "Equals",
    GreaterThan: "GreaterThan",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    LessThan: "LessThan",
    LessThanOrEqual: "LessThanOrEqual",
};
exports.PipelineType = {
    /**
     * Pipeline for logs telemetry.
     */
    Logs: "Logs",
};
exports.PredictiveAutoscalePolicyScaleMode = {
    Disabled: "Disabled",
    ForecastOnly: "ForecastOnly",
    Enabled: "Enabled",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ProcessorType = {
    /**
     * Batch processor.
     */
    Batch: "Batch",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReceiverType = {
    /**
     * Linux syslog.
     */
    Syslog: "Syslog",
    /**
     * Receives data from azure monitor agent receiver.
     */
    Ama: "Ama",
    /**
     * Receives data from another pipeline group.
     */
    PipelineGroup: "PipelineGroup",
    /**
     * Receives data from a OTLP collector.
     */
    OTLP: "OTLP",
    /**
     * Receives data from an UDP collector.
     */
    UDP: "UDP",
};
exports.RecurrenceFrequency = {
    None: "None",
    Second: "Second",
    Minute: "Minute",
    Hour: "Hour",
    Day: "Day",
    Week: "Week",
    Month: "Month",
    Year: "Year",
};
exports.RefreshInterval = {
    /**
     * One Minute
     */
    PT1M: "PT1M",
    /**
     * Five Minutes
     */
    PT5M: "PT5M",
    /**
     * Ten Minutes
     */
    PT10M: "PT10M",
    /**
     * Thirty Minutes
     */
    PT30M: "PT30M",
    /**
     * One Hour
     */
    PT1H: "PT1H",
    /**
     * Two Hours
     */
    PT2H: "PT2H",
};
exports.ScaleDirection = {
    None: "None",
    Increase: "Increase",
    Decrease: "Decrease",
};
exports.ScaleRuleMetricDimensionOperationType = {
    Equals: "Equals",
    NotEquals: "NotEquals",
};
exports.ScaleType = {
    ChangeCount: "ChangeCount",
    PercentChangeCount: "PercentChangeCount",
    ExactCount: "ExactCount",
    ServiceAllowedNextValue: "ServiceAllowedNextValue",
};
exports.ScopedResourceKind = {
    Resource: "Resource",
    Metrics: "Metrics",
};
exports.SignalKind = {
    AzureResourceMetric: "AzureResourceMetric",
    LogAnalyticsQuery: "LogAnalyticsQuery",
    PrometheusMetricsQuery: "PrometheusMetricsQuery",
};
exports.SignalOperator = {
    /**
     * Less than
     */
    LessThan: "LessThan",
    /**
     * Less than or equal to
     */
    LessThanOrEqual: "LessThanOrEqual",
    /**
     * Greater than
     */
    GreaterThan: "GreaterThan",
    /**
     * Greater than or equal to
     */
    GreaterThanOrEqual: "GreaterThanOrEqual",
    /**
     * Equal to
     */
    Equal: "Equal",
};
exports.Status = {
    /**
     * The issue is new
     */
    New: "New",
    /**
     * The issue is in progress
     */
    InProgress: "InProgress",
    /**
     * The issue is mitigated
     */
    Mitigated: "Mitigated",
    /**
     * The issue is closed
     */
    Closed: "Closed",
    /**
     * The issue is canceled
     */
    Canceled: "Canceled",
};
exports.StreamEncodingType = {
    /**
     * No encoding validation. Treats the file as a stream of raw bytes.
     */
    Nop: "nop",
    /**
     * UTF-8 encoding.
     */
    Utf_8: "utf-8",
    /**
     * UTF-16 encoding with little-endian byte order.
     */
    Utf_16le: "utf-16le",
    /**
     * UTF-16 encoding with little-endian byte order.
     */
    Utf_16be: "utf-16be",
    /**
     * ASCII encoding.
     */
    Ascii: "ascii",
    /**
     * The Big5 Chinese character encoding.
     */
    Big5: "big5",
};
exports.SyslogProtocol = {
    /**
     * rfc3164 protocol.
     */
    Rfc3164: "rfc3164",
    /**
     * rfc5424 protocol.
     */
    Rfc5424: "rfc5424",
};
exports.TimeAggregation = {
    Count: "Count",
    Average: "Average",
    Minimum: "Minimum",
    Maximum: "Maximum",
    Total: "Total",
};
exports.TimeAggregationType = {
    Average: "Average",
    Minimum: "Minimum",
    Maximum: "Maximum",
    Total: "Total",
    Count: "Count",
    Last: "Last",
};
