// @pulumi/azure-native/cloudngfw/globalRulestack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalRulestack = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * PaloAltoNetworks GlobalRulestack
 *
 * Uses Azure REST API version 2025-05-23. In version 2.x of the Azure Native provider, it used API version 2023-09-01.
 *
 * Other available API versions: 2023-09-01, 2023-10-10-preview, 2024-01-19-preview, 2024-02-07-preview, 2025-02-06-preview, 2025-07-07-preview, 2025-10-08, 2026-01-26-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cloudngfw [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class GlobalRulestack extends pulumi.CustomResource {
    /**
     * Get an existing GlobalRulestack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GlobalRulestack(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cloudngfw:GlobalRulestack';
    /**
     * Returns true if the given object is an instance of GlobalRulestack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GlobalRulestack.__pulumiType;
    }
    /**
     * Create a GlobalRulestack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["associatedSubscriptions"] = args?.associatedSubscriptions;
            resourceInputs["defaultMode"] = args?.defaultMode;
            resourceInputs["description"] = args?.description;
            resourceInputs["globalRulestackName"] = args?.globalRulestackName;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["location"] = args?.location;
            resourceInputs["minAppIdVersion"] = args?.minAppIdVersion;
            resourceInputs["panEtag"] = args?.panEtag;
            resourceInputs["panLocation"] = args?.panLocation;
            resourceInputs["scope"] = args?.scope;
            resourceInputs["securityServices"] = args?.securityServices;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["associatedSubscriptions"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["defaultMode"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["minAppIdVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["panEtag"] = undefined /*out*/;
            resourceInputs["panLocation"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["securityServices"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cloudngfw/v20220829:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20220829preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20230901:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20230901preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20231010preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20240119preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20240207preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20250206preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20250523:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20250707preview:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20251008:GlobalRulestack" }, { type: "azure-native:cloudngfw/v20260126preview:GlobalRulestack" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GlobalRulestack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GlobalRulestack = GlobalRulestack;
