// @pulumi/azure-native/cloudngfw/prefixListLocalRulestack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrefixListLocalRulestack = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * LocalRulestack prefixList
 *
 * Uses Azure REST API version 2025-05-23. In version 2.x of the Azure Native provider, it used API version 2023-09-01.
 *
 * Other available API versions: 2023-09-01, 2023-10-10-preview, 2024-01-19-preview, 2024-02-07-preview, 2025-02-06-preview, 2025-07-07-preview, 2025-10-08, 2026-01-26-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cloudngfw [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrefixListLocalRulestack extends pulumi.CustomResource {
    /**
     * Get an existing PrefixListLocalRulestack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrefixListLocalRulestack(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:cloudngfw:PrefixListLocalRulestack';
    /**
     * Returns true if the given object is an instance of PrefixListLocalRulestack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrefixListLocalRulestack.__pulumiType;
    }
    /**
     * Create a PrefixListLocalRulestack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.localRulestackName === undefined && !opts.urn) {
                throw new Error("Missing required property 'localRulestackName'");
            }
            if (args?.prefixList === undefined && !opts.urn) {
                throw new Error("Missing required property 'prefixList'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["auditComment"] = args?.auditComment;
            resourceInputs["description"] = args?.description;
            resourceInputs["localRulestackName"] = args?.localRulestackName;
            resourceInputs["name"] = args?.name;
            resourceInputs["prefixList"] = args?.prefixList;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["auditComment"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["prefixList"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cloudngfw/v20220829:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20220829preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20230901:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20230901preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20231010preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20240119preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20240207preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20250206preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20250523:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20250707preview:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20251008:PrefixListLocalRulestack" }, { type: "azure-native:cloudngfw/v20260126preview:PrefixListLocalRulestack" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrefixListLocalRulestack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrefixListLocalRulestack = PrefixListLocalRulestack;
