// @pulumi/azure-native/dns/recordSet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordSet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Describes a DNS record set (a collection of DNS records with the same name and type).
 *
 * Uses Azure REST API version 2023-07-01-preview.
 *
 * Other available API versions: 2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-10-01, 2018-03-01-preview, 2018-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native dns [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class RecordSet extends pulumi.CustomResource {
    /**
     * Get an existing RecordSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RecordSet(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:dns:RecordSet';
    /**
     * Returns true if the given object is an instance of RecordSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RecordSet.__pulumiType;
    }
    /**
     * Create a RecordSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.recordType === undefined && !opts.urn) {
                throw new Error("Missing required property 'recordType'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.zoneName === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneName'");
            }
            resourceInputs["aRecords"] = args?.aRecords;
            resourceInputs["aaaaRecords"] = args?.aaaaRecords;
            resourceInputs["caaRecords"] = args?.caaRecords;
            resourceInputs["cnameRecord"] = args?.cnameRecord;
            resourceInputs["dsRecords"] = args?.dsRecords;
            resourceInputs["metadata"] = args?.metadata;
            resourceInputs["mxRecords"] = args?.mxRecords;
            resourceInputs["naptrRecords"] = args?.naptrRecords;
            resourceInputs["nsRecords"] = args?.nsRecords;
            resourceInputs["ptrRecords"] = args?.ptrRecords;
            resourceInputs["recordType"] = args?.recordType;
            resourceInputs["relativeRecordSetName"] = args?.relativeRecordSetName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["soaRecord"] = args?.soaRecord;
            resourceInputs["srvRecords"] = args?.srvRecords;
            resourceInputs["targetResource"] = args?.targetResource;
            resourceInputs["tlsaRecords"] = args?.tlsaRecords;
            resourceInputs["trafficManagementProfile"] = args?.trafficManagementProfile;
            resourceInputs["ttl"] = args?.ttl;
            resourceInputs["txtRecords"] = args?.txtRecords;
            resourceInputs["zoneName"] = args?.zoneName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["aRecords"] = undefined /*out*/;
            resourceInputs["aaaaRecords"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["caaRecords"] = undefined /*out*/;
            resourceInputs["cnameRecord"] = undefined /*out*/;
            resourceInputs["dsRecords"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["mxRecords"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["naptrRecords"] = undefined /*out*/;
            resourceInputs["nsRecords"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["ptrRecords"] = undefined /*out*/;
            resourceInputs["soaRecord"] = undefined /*out*/;
            resourceInputs["srvRecords"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["targetResource"] = undefined /*out*/;
            resourceInputs["tlsaRecords"] = undefined /*out*/;
            resourceInputs["trafficManagementProfile"] = undefined /*out*/;
            resourceInputs["ttl"] = undefined /*out*/;
            resourceInputs["txtRecords"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:dns/v20150504preview:RecordSet" }, { type: "azure-native:dns/v20160401:RecordSet" }, { type: "azure-native:dns/v20170901:RecordSet" }, { type: "azure-native:dns/v20171001:RecordSet" }, { type: "azure-native:dns/v20180301preview:RecordSet" }, { type: "azure-native:dns/v20180501:RecordSet" }, { type: "azure-native:dns/v20230701preview:RecordSet" }, { type: "azure-native:network/v20180501:RecordSet" }, { type: "azure-native:network/v20230701preview:RecordSet" }, { type: "azure-native:network:RecordSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RecordSet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RecordSet = RecordSet;
