// @pulumi/gcp/networkmanagement/organizationVpcFlowLogsConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationVpcFlowLogsConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * VPC Flow Logs Config is a resource that lets you configure Flow Logs for Organization.
 *
 * ## Example Usage
 *
 * ### Network Management Org Vpc Flow Logs Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const org_test = new gcp.networkmanagement.OrganizationVpcFlowLogsConfig("org-test", {
 *     vpcFlowLogsConfigId: "basic-org-test-id",
 *     location: "global",
 *     organization: "123456789",
 * });
 * ```
 *
 * ## Import
 *
 * OrganizationVpcFlowLogsConfig can be imported using any of these accepted formats:
 *
 * * `organizations/{{organization}}/locations/{{location}}/vpcFlowLogsConfigs/{{vpc_flow_logs_config_id}}`
 * * `{{organization}}/{{location}}/{{vpc_flow_logs_config_id}}`
 *
 * When using the `pulumi import` command, OrganizationVpcFlowLogsConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkmanagement/organizationVpcFlowLogsConfig:OrganizationVpcFlowLogsConfig default organizations/{{organization}}/locations/{{location}}/vpcFlowLogsConfigs/{{vpc_flow_logs_config_id}}
 * $ pulumi import gcp:networkmanagement/organizationVpcFlowLogsConfig:OrganizationVpcFlowLogsConfig default {{organization}}/{{location}}/{{vpc_flow_logs_config_id}}
 * ```
 */
class OrganizationVpcFlowLogsConfig extends pulumi.CustomResource {
    /**
     * Get an existing OrganizationVpcFlowLogsConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OrganizationVpcFlowLogsConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkmanagement/organizationVpcFlowLogsConfig:OrganizationVpcFlowLogsConfig';
    /**
     * Returns true if the given object is an instance of OrganizationVpcFlowLogsConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OrganizationVpcFlowLogsConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aggregationInterval"] = state?.aggregationInterval;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["crossProjectMetadata"] = state?.crossProjectMetadata;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["filterExpr"] = state?.filterExpr;
            resourceInputs["flowSampling"] = state?.flowSampling;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["metadata"] = state?.metadata;
            resourceInputs["metadataFields"] = state?.metadataFields;
            resourceInputs["name"] = state?.name;
            resourceInputs["organization"] = state?.organization;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["state"] = state?.state;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["vpcFlowLogsConfigId"] = state?.vpcFlowLogsConfigId;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.organization === undefined && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if (args?.vpcFlowLogsConfigId === undefined && !opts.urn) {
                throw new Error("Missing required property 'vpcFlowLogsConfigId'");
            }
            resourceInputs["aggregationInterval"] = args?.aggregationInterval;
            resourceInputs["crossProjectMetadata"] = args?.crossProjectMetadata;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["filterExpr"] = args?.filterExpr;
            resourceInputs["flowSampling"] = args?.flowSampling;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["metadata"] = args?.metadata;
            resourceInputs["metadataFields"] = args?.metadataFields;
            resourceInputs["organization"] = args?.organization;
            resourceInputs["state"] = args?.state;
            resourceInputs["vpcFlowLogsConfigId"] = args?.vpcFlowLogsConfigId;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(OrganizationVpcFlowLogsConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OrganizationVpcFlowLogsConfig = OrganizationVpcFlowLogsConfig;
