// @pulumi/azure-native/app/containerAppsSourceControl.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerAppsSourceControl = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Container App SourceControl.
 *
 * Uses Azure REST API version 2025-02-02-preview. In version 2.x of the Azure Native provider, it used API version 2022-10-01.
 *
 * Other available API versions: 2022-10-01, 2022-11-01-preview, 2023-04-01-preview, 2023-05-01, 2023-05-02-preview, 2023-08-01-preview, 2023-11-02-preview, 2024-02-02-preview, 2024-03-01, 2024-08-02-preview, 2024-10-02-preview, 2025-01-01, 2025-07-01, 2025-10-02-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native app [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ContainerAppsSourceControl extends pulumi.CustomResource {
    /**
     * Get an existing ContainerAppsSourceControl resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ContainerAppsSourceControl(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:app:ContainerAppsSourceControl';
    /**
     * Returns true if the given object is an instance of ContainerAppsSourceControl.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContainerAppsSourceControl.__pulumiType;
    }
    /**
     * Create a ContainerAppsSourceControl resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.containerAppName === undefined && !opts.urn) {
                throw new Error("Missing required property 'containerAppName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["branch"] = args?.branch;
            resourceInputs["containerAppName"] = args?.containerAppName;
            resourceInputs["githubActionConfiguration"] = args?.githubActionConfiguration;
            resourceInputs["repoUrl"] = args?.repoUrl;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sourceControlName"] = args?.sourceControlName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["branch"] = undefined /*out*/;
            resourceInputs["githubActionConfiguration"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationState"] = undefined /*out*/;
            resourceInputs["repoUrl"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app/v20220101preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20220301:ContainerAppsSourceControl" }, { type: "azure-native:app/v20220601preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20221001:ContainerAppsSourceControl" }, { type: "azure-native:app/v20221101preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20230401preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20230501:ContainerAppsSourceControl" }, { type: "azure-native:app/v20230502preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20230801preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20231102preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20240202preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20240301:ContainerAppsSourceControl" }, { type: "azure-native:app/v20240802preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20241002preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20250101:ContainerAppsSourceControl" }, { type: "azure-native:app/v20250202preview:ContainerAppsSourceControl" }, { type: "azure-native:app/v20250701:ContainerAppsSourceControl" }, { type: "azure-native:app/v20251002preview:ContainerAppsSourceControl" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ContainerAppsSourceControl.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ContainerAppsSourceControl = ContainerAppsSourceControl;
