// @pulumi/azure-native/types/enums/redis/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZonalAllocationPolicy = exports.UpdateChannel = exports.TlsVersion = exports.SkuName = exports.SkuFamily = exports.ReplicationRole = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.DayOfWeek = void 0;
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
    /**
     * As a convenience, 'Everyday' is also accepted and means the same as specifying all days of the week.
     */
    Everyday: "Everyday",
    /**
     * As a convenience, 'Weekend' is also accepted and means the same as specifying Saturday and Sunday.
     */
    Weekend: "Weekend",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    /**
     * Public internet access to the cache, via its public IP address, is enabled. Connections may use any network path.
     */
    Enabled: "Enabled",
    /**
     * Public internet access to the cache, via its public IP address, is disabled. Connections must use be made via private endpoints.
     */
    Disabled: "Disabled",
};
exports.ReplicationRole = {
    Primary: "Primary",
    Secondary: "Secondary",
};
exports.SkuFamily = {
    /**
     * The SKU family to use - must be 'C' for Basic/Standard SKU redis caches.
     */
    C: "C",
    /**
     * The SKU family to use - must be 'P' for Premium SKU redis caches.
     */
    P: "P",
};
exports.SkuName = {
    /**
     * The well known 'Basic' SKU for Azure Cache for Redis. Basic SKU does not have an availability SLA.
     */
    Basic: "Basic",
    /**
     * The well known 'Standard' SKU for Azure Cache for Redis. Standard SKU has an availability SLA.
     */
    Standard: "Standard",
    /**
     * The well known 'Premium' SKU for Azure Cache for Redis. Premium SKU has an availability SLA, and higher performance tiers and more features compared with Standard SKU.
     */
    Premium: "Premium",
};
exports.TlsVersion = {
    /**
     * TLS protocol version 1.0 -- deprecated for security reasons. Do not use this value for new caches.
     */
    TlsVersion_1_0: "1.0",
    /**
     * TLS protocol version 1.1 -- deprecated for security reasons. Do not use this value for new caches.
     */
    TlsVersion_1_1: "1.1",
    /**
     * TLS protocol version 1.2 -- use this value, or higher, for new caches. Or do not specify, so that your cache uses the recommended default value
     */
    TlsVersion_1_2: "1.2",
};
exports.UpdateChannel = {
    /**
     * Stable channel receives updates, which may include important security and stability updates, later than Preview channel.
     */
    Stable: "Stable",
    /**
     * Preview channel normally receives updates before Stable channel, and is the recommended channel for non-production workloads.
     */
    Preview: "Preview",
};
exports.ZonalAllocationPolicy = {
    /**
     * The zones for the cache will be selected automatically based on availability and capacity.
     */
    Automatic: "Automatic",
    /**
     * UserDefined means the zones for the cache are manually configured using the 'zones' property, and can not be automatically selected.
     */
    UserDefined: "UserDefined",
    /**
     * The cache will not use multiple availability zones.
     */
    NoZones: "NoZones",
};
