// @pulumi/azure-native/operationalinsights/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.Table = exports.SummaryLog = exports.StorageInsightConfig = exports.SavedSearch = exports.QueryPack = exports.Query = exports.MachineGroup = exports.LinkedStorageAccount = exports.LinkedService = exports.getWorkspaceSharedKeysOutput = exports.getWorkspaceSharedKeys = exports.getWorkspaceOutput = exports.getWorkspace = exports.getTableOutput = exports.getTable = exports.getSummaryLogOutput = exports.getSummaryLog = exports.getStorageInsightConfigOutput = exports.getStorageInsightConfig = exports.getSharedKeysOutput = exports.getSharedKeys = exports.getSavedSearchOutput = exports.getSavedSearch = exports.getQueryPackOutput = exports.getQueryPack = exports.getQueryOutput = exports.getQuery = exports.getMachineGroupOutput = exports.getMachineGroup = exports.getLinkedStorageAccountOutput = exports.getLinkedStorageAccount = exports.getLinkedServiceOutput = exports.getLinkedService = exports.getDataSourceOutput = exports.getDataSource = exports.getDataExportOutput = exports.getDataExport = exports.getClusterOutput = exports.getCluster = exports.DataSource = exports.DataExport = exports.Cluster = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.DataExport = null;
utilities.lazyLoad(exports, ["DataExport"], () => require("./dataExport"));
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getDataExport = null;
exports.getDataExportOutput = null;
utilities.lazyLoad(exports, ["getDataExport", "getDataExportOutput"], () => require("./getDataExport"));
exports.getDataSource = null;
exports.getDataSourceOutput = null;
utilities.lazyLoad(exports, ["getDataSource", "getDataSourceOutput"], () => require("./getDataSource"));
exports.getLinkedService = null;
exports.getLinkedServiceOutput = null;
utilities.lazyLoad(exports, ["getLinkedService", "getLinkedServiceOutput"], () => require("./getLinkedService"));
exports.getLinkedStorageAccount = null;
exports.getLinkedStorageAccountOutput = null;
utilities.lazyLoad(exports, ["getLinkedStorageAccount", "getLinkedStorageAccountOutput"], () => require("./getLinkedStorageAccount"));
exports.getMachineGroup = null;
exports.getMachineGroupOutput = null;
utilities.lazyLoad(exports, ["getMachineGroup", "getMachineGroupOutput"], () => require("./getMachineGroup"));
exports.getQuery = null;
exports.getQueryOutput = null;
utilities.lazyLoad(exports, ["getQuery", "getQueryOutput"], () => require("./getQuery"));
exports.getQueryPack = null;
exports.getQueryPackOutput = null;
utilities.lazyLoad(exports, ["getQueryPack", "getQueryPackOutput"], () => require("./getQueryPack"));
exports.getSavedSearch = null;
exports.getSavedSearchOutput = null;
utilities.lazyLoad(exports, ["getSavedSearch", "getSavedSearchOutput"], () => require("./getSavedSearch"));
exports.getSharedKeys = null;
exports.getSharedKeysOutput = null;
utilities.lazyLoad(exports, ["getSharedKeys", "getSharedKeysOutput"], () => require("./getSharedKeys"));
exports.getStorageInsightConfig = null;
exports.getStorageInsightConfigOutput = null;
utilities.lazyLoad(exports, ["getStorageInsightConfig", "getStorageInsightConfigOutput"], () => require("./getStorageInsightConfig"));
exports.getSummaryLog = null;
exports.getSummaryLogOutput = null;
utilities.lazyLoad(exports, ["getSummaryLog", "getSummaryLogOutput"], () => require("./getSummaryLog"));
exports.getTable = null;
exports.getTableOutput = null;
utilities.lazyLoad(exports, ["getTable", "getTableOutput"], () => require("./getTable"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspaceSharedKeys = null;
exports.getWorkspaceSharedKeysOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceSharedKeys", "getWorkspaceSharedKeysOutput"], () => require("./getWorkspaceSharedKeys"));
exports.LinkedService = null;
utilities.lazyLoad(exports, ["LinkedService"], () => require("./linkedService"));
exports.LinkedStorageAccount = null;
utilities.lazyLoad(exports, ["LinkedStorageAccount"], () => require("./linkedStorageAccount"));
exports.MachineGroup = null;
utilities.lazyLoad(exports, ["MachineGroup"], () => require("./machineGroup"));
exports.Query = null;
utilities.lazyLoad(exports, ["Query"], () => require("./query"));
exports.QueryPack = null;
utilities.lazyLoad(exports, ["QueryPack"], () => require("./queryPack"));
exports.SavedSearch = null;
utilities.lazyLoad(exports, ["SavedSearch"], () => require("./savedSearch"));
exports.StorageInsightConfig = null;
utilities.lazyLoad(exports, ["StorageInsightConfig"], () => require("./storageInsightConfig"));
exports.SummaryLog = null;
utilities.lazyLoad(exports, ["SummaryLog"], () => require("./summaryLog"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/operationalinsights"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:operationalinsights:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:operationalinsights:DataExport":
                return new exports.DataExport(name, undefined, { urn });
            case "azure-native:operationalinsights:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "azure-native:operationalinsights:LinkedService":
                return new exports.LinkedService(name, undefined, { urn });
            case "azure-native:operationalinsights:LinkedStorageAccount":
                return new exports.LinkedStorageAccount(name, undefined, { urn });
            case "azure-native:operationalinsights:MachineGroup":
                return new exports.MachineGroup(name, undefined, { urn });
            case "azure-native:operationalinsights:Query":
                return new exports.Query(name, undefined, { urn });
            case "azure-native:operationalinsights:QueryPack":
                return new exports.QueryPack(name, undefined, { urn });
            case "azure-native:operationalinsights:SavedSearch":
                return new exports.SavedSearch(name, undefined, { urn });
            case "azure-native:operationalinsights:StorageInsightConfig":
                return new exports.StorageInsightConfig(name, undefined, { urn });
            case "azure-native:operationalinsights:SummaryLog":
                return new exports.SummaryLog(name, undefined, { urn });
            case "azure-native:operationalinsights:Table":
                return new exports.Table(name, undefined, { urn });
            case "azure-native:operationalinsights:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "operationalinsights", _module);
