// @pulumi/aws/licensemanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseGrantAccepter = exports.LicenseGrant = exports.LicenseConfiguration = exports.getReceivedLicensesOutput = exports.getReceivedLicenses = exports.getReceivedLicenseOutput = exports.getReceivedLicense = exports.getLicenseGrantsOutput = exports.getLicenseGrants = exports.Association = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Association = null;
utilities.lazyLoad(exports, ["Association"], () => require("./association"));
exports.getLicenseGrants = null;
exports.getLicenseGrantsOutput = null;
utilities.lazyLoad(exports, ["getLicenseGrants", "getLicenseGrantsOutput"], () => require("./getLicenseGrants"));
exports.getReceivedLicense = null;
exports.getReceivedLicenseOutput = null;
utilities.lazyLoad(exports, ["getReceivedLicense", "getReceivedLicenseOutput"], () => require("./getReceivedLicense"));
exports.getReceivedLicenses = null;
exports.getReceivedLicensesOutput = null;
utilities.lazyLoad(exports, ["getReceivedLicenses", "getReceivedLicensesOutput"], () => require("./getReceivedLicenses"));
exports.LicenseConfiguration = null;
utilities.lazyLoad(exports, ["LicenseConfiguration"], () => require("./licenseConfiguration"));
exports.LicenseGrant = null;
utilities.lazyLoad(exports, ["LicenseGrant"], () => require("./licenseGrant"));
exports.LicenseGrantAccepter = null;
utilities.lazyLoad(exports, ["LicenseGrantAccepter"], () => require("./licenseGrantAccepter"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:licensemanager/association:Association":
                return new exports.Association(name, undefined, { urn });
            case "aws:licensemanager/licenseConfiguration:LicenseConfiguration":
                return new exports.LicenseConfiguration(name, undefined, { urn });
            case "aws:licensemanager/licenseGrant:LicenseGrant":
                return new exports.LicenseGrant(name, undefined, { urn });
            case "aws:licensemanager/licenseGrantAccepter:LicenseGrantAccepter":
                return new exports.LicenseGrantAccepter(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "licensemanager/association", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseGrant", _module);
pulumi.runtime.registerResourceModule("aws", "licensemanager/licenseGrantAccepter", _module);
