// @pulumi/gcp/parametermanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionalParameterVersion = exports.RegionalParameter = exports.ParameterVersion = exports.Parameter = exports.getRegionalParametersOutput = exports.getRegionalParameters = exports.getRegionalParameterVersionRenderOutput = exports.getRegionalParameterVersionRender = exports.getRegionalParameterVersionOutput = exports.getRegionalParameterVersion = exports.getRegionalParameterOutput = exports.getRegionalParameter = exports.getParametersOutput = exports.getParameters = exports.getParameterVersionRenderOutput = exports.getParameterVersionRender = exports.getParameterVersionOutput = exports.getParameterVersion = exports.getParameterOutput = exports.getParameter = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getParameter = null;
exports.getParameterOutput = null;
utilities.lazyLoad(exports, ["getParameter", "getParameterOutput"], () => require("./getParameter"));
exports.getParameterVersion = null;
exports.getParameterVersionOutput = null;
utilities.lazyLoad(exports, ["getParameterVersion", "getParameterVersionOutput"], () => require("./getParameterVersion"));
exports.getParameterVersionRender = null;
exports.getParameterVersionRenderOutput = null;
utilities.lazyLoad(exports, ["getParameterVersionRender", "getParameterVersionRenderOutput"], () => require("./getParameterVersionRender"));
exports.getParameters = null;
exports.getParametersOutput = null;
utilities.lazyLoad(exports, ["getParameters", "getParametersOutput"], () => require("./getParameters"));
exports.getRegionalParameter = null;
exports.getRegionalParameterOutput = null;
utilities.lazyLoad(exports, ["getRegionalParameter", "getRegionalParameterOutput"], () => require("./getRegionalParameter"));
exports.getRegionalParameterVersion = null;
exports.getRegionalParameterVersionOutput = null;
utilities.lazyLoad(exports, ["getRegionalParameterVersion", "getRegionalParameterVersionOutput"], () => require("./getRegionalParameterVersion"));
exports.getRegionalParameterVersionRender = null;
exports.getRegionalParameterVersionRenderOutput = null;
utilities.lazyLoad(exports, ["getRegionalParameterVersionRender", "getRegionalParameterVersionRenderOutput"], () => require("./getRegionalParameterVersionRender"));
exports.getRegionalParameters = null;
exports.getRegionalParametersOutput = null;
utilities.lazyLoad(exports, ["getRegionalParameters", "getRegionalParametersOutput"], () => require("./getRegionalParameters"));
exports.Parameter = null;
utilities.lazyLoad(exports, ["Parameter"], () => require("./parameter"));
exports.ParameterVersion = null;
utilities.lazyLoad(exports, ["ParameterVersion"], () => require("./parameterVersion"));
exports.RegionalParameter = null;
utilities.lazyLoad(exports, ["RegionalParameter"], () => require("./regionalParameter"));
exports.RegionalParameterVersion = null;
utilities.lazyLoad(exports, ["RegionalParameterVersion"], () => require("./regionalParameterVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:parametermanager/parameter:Parameter":
                return new exports.Parameter(name, undefined, { urn });
            case "gcp:parametermanager/parameterVersion:ParameterVersion":
                return new exports.ParameterVersion(name, undefined, { urn });
            case "gcp:parametermanager/regionalParameter:RegionalParameter":
                return new exports.RegionalParameter(name, undefined, { urn });
            case "gcp:parametermanager/regionalParameterVersion:RegionalParameterVersion":
                return new exports.RegionalParameterVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "parametermanager/parameter", _module);
pulumi.runtime.registerResourceModule("gcp", "parametermanager/parameterVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "parametermanager/regionalParameter", _module);
pulumi.runtime.registerResourceModule("gcp", "parametermanager/regionalParameterVersion", _module);
