// @pulumi/gcp/parametermanager/regionalParameter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionalParameter = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A Regional Parameter is a logical regional parameter.
 *
 * To get more information about RegionalParameter, see:
 *
 * * [API documentation](https://cloud.google.com/secret-manager/parameter-manager/docs/reference/rest/v1/projects.locations.parameters)
 *
 * ## Example Usage
 *
 * ### Regional Parameter Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const regional_parameter_basic = new gcp.parametermanager.RegionalParameter("regional-parameter-basic", {
 *     parameterId: "regional_parameter",
 *     location: "us-central1",
 * });
 * ```
 * ### Regional Parameter With Format
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const regional_parameter_with_format = new gcp.parametermanager.RegionalParameter("regional-parameter-with-format", {
 *     parameterId: "regional_parameter",
 *     location: "us-central1",
 *     format: "JSON",
 * });
 * ```
 * ### Regional Parameter With Labels
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const regional_parameter_with_labels = new gcp.parametermanager.RegionalParameter("regional-parameter-with-labels", {
 *     parameterId: "regional_parameter",
 *     location: "us-central1",
 *     labels: {
 *         key1: "val1",
 *         key2: "val2",
 *         key3: "val3",
 *         key4: "val4",
 *         key5: "val5",
 *     },
 * });
 * ```
 * ### Regional Parameter With Kms Key
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const regional_parameter_with_kms_key = new gcp.parametermanager.RegionalParameter("regional-parameter-with-kms-key", {
 *     parameterId: "regional_parameter",
 *     location: "us-central1",
 *     kmsKey: "kms-key",
 * });
 * ```
 *
 * ## Import
 *
 * RegionalParameter can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/parameters/{{parameter_id}}`
 * * `{{project}}/{{location}}/{{parameter_id}}`
 * * `{{location}}/{{parameter_id}}`
 *
 * When using the `pulumi import` command, RegionalParameter can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:parametermanager/regionalParameter:RegionalParameter default projects/{{project}}/locations/{{location}}/parameters/{{parameter_id}}
 * $ pulumi import gcp:parametermanager/regionalParameter:RegionalParameter default {{project}}/{{location}}/{{parameter_id}}
 * $ pulumi import gcp:parametermanager/regionalParameter:RegionalParameter default {{location}}/{{parameter_id}}
 * ```
 */
class RegionalParameter extends pulumi.CustomResource {
    /**
     * Get an existing RegionalParameter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegionalParameter(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:parametermanager/regionalParameter:RegionalParameter';
    /**
     * Returns true if the given object is an instance of RegionalParameter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegionalParameter.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["format"] = state?.format;
            resourceInputs["kmsKey"] = state?.kmsKey;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["parameterId"] = state?.parameterId;
            resourceInputs["policyMembers"] = state?.policyMembers;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.parameterId === undefined && !opts.urn) {
                throw new Error("Missing required property 'parameterId'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["format"] = args?.format;
            resourceInputs["kmsKey"] = args?.kmsKey;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["parameterId"] = args?.parameterId;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["policyMembers"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(RegionalParameter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegionalParameter = RegionalParameter;
