// @pulumi/gcp/monitoring/monitoredProject.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoredProject = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A [project being monitored](https://cloud.google.com/monitoring/settings/multiple-projects#create-multi) by a Metrics Scope.
 *
 * To get more information about MonitoredProject, see:
 *
 * * [API documentation](https://cloud.google.com/monitoring/api/ref_v3/rest/v1/locations.global.metricsScopes.projects)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/monitoring/settings/manage-api)
 *
 * ## Example Usage
 *
 * ### Monitoring Monitored Project Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const basic = new gcp.organizations.Project("basic", {
 *     projectId: "m-id",
 *     name: "m-id-display",
 *     orgId: "123456789",
 *     deletionPolicy: "DELETE",
 * });
 * const primary = new gcp.monitoring.MonitoredProject("primary", {
 *     metricsScope: "my-project-name",
 *     name: basic.projectId,
 * });
 * ```
 *
 * ## Import
 *
 * MonitoredProject can be imported using any of these accepted formats:
 *
 * * `v1/locations/global/metricsScopes/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, MonitoredProject can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:monitoring/monitoredProject:MonitoredProject default v1/locations/global/metricsScopes/{{name}}
 * $ pulumi import gcp:monitoring/monitoredProject:MonitoredProject default {{name}}
 * ```
 */
class MonitoredProject extends pulumi.CustomResource {
    /**
     * Get an existing MonitoredProject resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MonitoredProject(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:monitoring/monitoredProject:MonitoredProject';
    /**
     * Returns true if the given object is an instance of MonitoredProject.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MonitoredProject.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["metricsScope"] = state?.metricsScope;
            resourceInputs["name"] = state?.name;
        }
        else {
            const args = argsOrState;
            if (args?.metricsScope === undefined && !opts.urn) {
                throw new Error("Missing required property 'metricsScope'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["metricsScope"] = args?.metricsScope;
            resourceInputs["name"] = args?.name;
            resourceInputs["createTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MonitoredProject.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MonitoredProject = MonitoredProject;
