// @pulumi/azure-native/applicationinsights/exportConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportConfiguration = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Properties that define a Continuous Export configuration.
 *
 * Uses Azure REST API version 2015-05-01.
 */
class ExportConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing ExportConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExportConfiguration(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:applicationinsights:ExportConfiguration';
    /**
     * Returns true if the given object is an instance of ExportConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExportConfiguration.__pulumiType;
    }
    /**
     * Create a ExportConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.resourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["destinationAccountId"] = args?.destinationAccountId;
            resourceInputs["destinationAddress"] = args?.destinationAddress;
            resourceInputs["destinationStorageLocationId"] = args?.destinationStorageLocationId;
            resourceInputs["destinationStorageSubscriptionId"] = args?.destinationStorageSubscriptionId;
            resourceInputs["destinationType"] = args?.destinationType;
            resourceInputs["exportId"] = args?.exportId;
            resourceInputs["isEnabled"] = args?.isEnabled;
            resourceInputs["notificationQueueEnabled"] = args?.notificationQueueEnabled;
            resourceInputs["notificationQueueUri"] = args?.notificationQueueUri;
            resourceInputs["recordTypes"] = args?.recordTypes;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceName"] = args?.resourceName;
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["containerName"] = undefined /*out*/;
            resourceInputs["exportStatus"] = undefined /*out*/;
            resourceInputs["instrumentationKey"] = undefined /*out*/;
            resourceInputs["isUserEnabled"] = undefined /*out*/;
            resourceInputs["lastGapTime"] = undefined /*out*/;
            resourceInputs["lastSuccessTime"] = undefined /*out*/;
            resourceInputs["lastUserUpdate"] = undefined /*out*/;
            resourceInputs["permanentErrorReason"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["storageName"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["containerName"] = undefined /*out*/;
            resourceInputs["destinationAccountId"] = undefined /*out*/;
            resourceInputs["destinationStorageLocationId"] = undefined /*out*/;
            resourceInputs["destinationStorageSubscriptionId"] = undefined /*out*/;
            resourceInputs["destinationType"] = undefined /*out*/;
            resourceInputs["exportId"] = undefined /*out*/;
            resourceInputs["exportStatus"] = undefined /*out*/;
            resourceInputs["instrumentationKey"] = undefined /*out*/;
            resourceInputs["isUserEnabled"] = undefined /*out*/;
            resourceInputs["lastGapTime"] = undefined /*out*/;
            resourceInputs["lastSuccessTime"] = undefined /*out*/;
            resourceInputs["lastUserUpdate"] = undefined /*out*/;
            resourceInputs["notificationQueueEnabled"] = undefined /*out*/;
            resourceInputs["permanentErrorReason"] = undefined /*out*/;
            resourceInputs["recordTypes"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["storageName"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:applicationinsights/v20150501:ExportConfiguration" }, { type: "azure-native:insights/v20150501:ExportConfiguration" }, { type: "azure-native:insights:ExportConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExportConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExportConfiguration = ExportConfiguration;
