// @pulumi/azure-native/applicationinsights/webTest.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTest = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An Application Insights WebTest definition.
 *
 * Uses Azure REST API version 2022-06-15.
 *
 * Other available API versions: 2015-05-01, 2018-05-01-preview, 2020-10-05-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native applicationinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebTest extends pulumi.CustomResource {
    /**
     * Get an existing WebTest resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebTest(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:applicationinsights:WebTest';
    /**
     * Returns true if the given object is an instance of WebTest.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebTest.__pulumiType;
    }
    /**
     * Create a WebTest resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.locations === undefined && !opts.urn) {
                throw new Error("Missing required property 'locations'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.syntheticMonitorId === undefined && !opts.urn) {
                throw new Error("Missing required property 'syntheticMonitorId'");
            }
            if (args?.webTestKind === undefined && !opts.urn) {
                throw new Error("Missing required property 'webTestKind'");
            }
            resourceInputs["configuration"] = args?.configuration;
            resourceInputs["description"] = args?.description;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["frequency"] = (args?.frequency) ?? 300;
            resourceInputs["kind"] = (args?.kind) ?? "ping";
            resourceInputs["location"] = args?.location;
            resourceInputs["locations"] = args?.locations;
            resourceInputs["request"] = args?.request;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["retryEnabled"] = args?.retryEnabled;
            resourceInputs["syntheticMonitorId"] = args?.syntheticMonitorId;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeout"] = (args?.timeout) ?? 30;
            resourceInputs["validationRules"] = args?.validationRules;
            resourceInputs["webTestKind"] = (args?.webTestKind) ?? "ping";
            resourceInputs["webTestName"] = args?.webTestName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configuration"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["frequency"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["locations"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["request"] = undefined /*out*/;
            resourceInputs["retryEnabled"] = undefined /*out*/;
            resourceInputs["syntheticMonitorId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["timeout"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["validationRules"] = undefined /*out*/;
            resourceInputs["webTestKind"] = undefined /*out*/;
            resourceInputs["webTestName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:applicationinsights/v20150501:WebTest" }, { type: "azure-native:applicationinsights/v20180501preview:WebTest" }, { type: "azure-native:applicationinsights/v20201005preview:WebTest" }, { type: "azure-native:applicationinsights/v20220615:WebTest" }, { type: "azure-native:insights/v20201005preview:WebTest" }, { type: "azure-native:insights/v20220615:WebTest" }, { type: "azure-native:insights:WebTest" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebTest.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebTest = WebTest;
