// @pulumi/aws/elastictranscoder/preset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Preset = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Elastic Transcoder preset resource.
 *
 * > **Warning:** This resource is deprecated. Use [AWS Elemental MediaConvert](https://aws.amazon.com/blogs/media/migrating-workflows-from-amazon-elastic-transcoder-to-aws-elemental-mediaconvert/) instead. AWS will [discontinue support for Amazon Elastic Transcoder](https://aws.amazon.com/blogs/media/support-for-amazon-elastic-transcoder-ending-soon/), effective November 13, 2025.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const bar = new aws.elastictranscoder.Preset("bar", {
 *     container: "mp4",
 *     description: "Sample Preset",
 *     name: "sample_preset",
 *     audio: {
 *         audioPackingMode: "SingleTrack",
 *         bitRate: "96",
 *         channels: "2",
 *         codec: "AAC",
 *         sampleRate: "44100",
 *     },
 *     audioCodecOptions: {
 *         profile: "AAC-LC",
 *     },
 *     video: {
 *         bitRate: "1600",
 *         codec: "H.264",
 *         displayAspectRatio: "16:9",
 *         fixedGop: "false",
 *         frameRate: "auto",
 *         maxFrameRate: "60",
 *         keyframesMaxDist: "240",
 *         maxHeight: "auto",
 *         maxWidth: "auto",
 *         paddingPolicy: "Pad",
 *         sizingPolicy: "Fit",
 *     },
 *     videoCodecOptions: {
 *         Profile: "main",
 *         Level: "2.2",
 *         MaxReferenceFrames: "3",
 *         InterlacedMode: "Progressive",
 *         ColorSpaceConversionMode: "None",
 *     },
 *     videoWatermarks: [{
 *         id: "Test",
 *         maxWidth: "20%",
 *         maxHeight: "20%",
 *         sizingPolicy: "ShrinkToFit",
 *         horizontalAlign: "Right",
 *         horizontalOffset: "10px",
 *         verticalAlign: "Bottom",
 *         verticalOffset: "10px",
 *         opacity: "55.5",
 *         target: "Content",
 *     }],
 *     thumbnails: {
 *         format: "png",
 *         interval: "120",
 *         maxWidth: "auto",
 *         maxHeight: "auto",
 *         paddingPolicy: "Pad",
 *         sizingPolicy: "Fit",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Elastic Transcoder presets using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:elastictranscoder/preset:Preset basic_preset 1407981661351-cttk8b
 * ```
 */
class Preset extends pulumi.CustomResource {
    /**
     * Get an existing Preset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Preset(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:elastictranscoder/preset:Preset';
    /**
     * Returns true if the given object is an instance of Preset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Preset.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["audio"] = state?.audio;
            resourceInputs["audioCodecOptions"] = state?.audioCodecOptions;
            resourceInputs["container"] = state?.container;
            resourceInputs["description"] = state?.description;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["thumbnails"] = state?.thumbnails;
            resourceInputs["type"] = state?.type;
            resourceInputs["video"] = state?.video;
            resourceInputs["videoCodecOptions"] = state?.videoCodecOptions;
            resourceInputs["videoWatermarks"] = state?.videoWatermarks;
        }
        else {
            const args = argsOrState;
            if (args?.container === undefined && !opts.urn) {
                throw new Error("Missing required property 'container'");
            }
            resourceInputs["audio"] = args?.audio;
            resourceInputs["audioCodecOptions"] = args?.audioCodecOptions;
            resourceInputs["container"] = args?.container;
            resourceInputs["description"] = args?.description;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["thumbnails"] = args?.thumbnails;
            resourceInputs["type"] = args?.type;
            resourceInputs["video"] = args?.video;
            resourceInputs["videoCodecOptions"] = args?.videoCodecOptions;
            resourceInputs["videoWatermarks"] = args?.videoWatermarks;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Preset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Preset = Preset;
