// @pulumi/azure-native/privatedns/privateRecordSet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateRecordSet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Describes a DNS record set (a collection of DNS records with the same name and type) in a Private DNS zone.
 *
 * Uses Azure REST API version 2024-06-01.
 *
 * Other available API versions: 2018-09-01, 2020-01-01, 2020-06-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native privatedns [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateRecordSet extends pulumi.CustomResource {
    /**
     * Get an existing PrivateRecordSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateRecordSet(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:privatedns:PrivateRecordSet';
    /**
     * Returns true if the given object is an instance of PrivateRecordSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateRecordSet.__pulumiType;
    }
    /**
     * Create a PrivateRecordSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.privateZoneName === undefined && !opts.urn) {
                throw new Error("Missing required property 'privateZoneName'");
            }
            if (args?.recordType === undefined && !opts.urn) {
                throw new Error("Missing required property 'recordType'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["aRecords"] = args?.aRecords;
            resourceInputs["aaaaRecords"] = args?.aaaaRecords;
            resourceInputs["cnameRecord"] = args?.cnameRecord;
            resourceInputs["metadata"] = args?.metadata;
            resourceInputs["mxRecords"] = args?.mxRecords;
            resourceInputs["privateZoneName"] = args?.privateZoneName;
            resourceInputs["ptrRecords"] = args?.ptrRecords;
            resourceInputs["recordType"] = args?.recordType;
            resourceInputs["relativeRecordSetName"] = args?.relativeRecordSetName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["soaRecord"] = args?.soaRecord;
            resourceInputs["srvRecords"] = args?.srvRecords;
            resourceInputs["ttl"] = args?.ttl;
            resourceInputs["txtRecords"] = args?.txtRecords;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["isAutoRegistered"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["aRecords"] = undefined /*out*/;
            resourceInputs["aaaaRecords"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cnameRecord"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["isAutoRegistered"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["mxRecords"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["ptrRecords"] = undefined /*out*/;
            resourceInputs["soaRecord"] = undefined /*out*/;
            resourceInputs["srvRecords"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["ttl"] = undefined /*out*/;
            resourceInputs["txtRecords"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20200601:PrivateRecordSet" }, { type: "azure-native:network/v20240601:PrivateRecordSet" }, { type: "azure-native:network:PrivateRecordSet" }, { type: "azure-native:privatedns/v20180901:PrivateRecordSet" }, { type: "azure-native:privatedns/v20200101:PrivateRecordSet" }, { type: "azure-native:privatedns/v20200601:PrivateRecordSet" }, { type: "azure-native:privatedns/v20240601:PrivateRecordSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateRecordSet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateRecordSet = PrivateRecordSet;
