// @pulumi/gcp/bigtable/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableIamPolicy = exports.TableIamMember = exports.TableIamBinding = exports.Table = exports.SchemaBundle = exports.MaterializedView = exports.LogicalView = exports.InstanceIamPolicy = exports.InstanceIamMember = exports.InstanceIamBinding = exports.Instance = exports.getTableIamPolicyOutput = exports.getTableIamPolicy = exports.getInstanceIamPolicyOutput = exports.getInstanceIamPolicy = exports.GCPolicy = exports.AuthorizedView = exports.AppProfile = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AppProfile = null;
utilities.lazyLoad(exports, ["AppProfile"], () => require("./appProfile"));
exports.AuthorizedView = null;
utilities.lazyLoad(exports, ["AuthorizedView"], () => require("./authorizedView"));
exports.GCPolicy = null;
utilities.lazyLoad(exports, ["GCPolicy"], () => require("./gcpolicy"));
exports.getInstanceIamPolicy = null;
exports.getInstanceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getInstanceIamPolicy", "getInstanceIamPolicyOutput"], () => require("./getInstanceIamPolicy"));
exports.getTableIamPolicy = null;
exports.getTableIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTableIamPolicy", "getTableIamPolicyOutput"], () => require("./getTableIamPolicy"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceIamBinding = null;
utilities.lazyLoad(exports, ["InstanceIamBinding"], () => require("./instanceIamBinding"));
exports.InstanceIamMember = null;
utilities.lazyLoad(exports, ["InstanceIamMember"], () => require("./instanceIamMember"));
exports.InstanceIamPolicy = null;
utilities.lazyLoad(exports, ["InstanceIamPolicy"], () => require("./instanceIamPolicy"));
exports.LogicalView = null;
utilities.lazyLoad(exports, ["LogicalView"], () => require("./logicalView"));
exports.MaterializedView = null;
utilities.lazyLoad(exports, ["MaterializedView"], () => require("./materializedView"));
exports.SchemaBundle = null;
utilities.lazyLoad(exports, ["SchemaBundle"], () => require("./schemaBundle"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.TableIamBinding = null;
utilities.lazyLoad(exports, ["TableIamBinding"], () => require("./tableIamBinding"));
exports.TableIamMember = null;
utilities.lazyLoad(exports, ["TableIamMember"], () => require("./tableIamMember"));
exports.TableIamPolicy = null;
utilities.lazyLoad(exports, ["TableIamPolicy"], () => require("./tableIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigtable/appProfile:AppProfile":
                return new exports.AppProfile(name, undefined, { urn });
            case "gcp:bigtable/authorizedView:AuthorizedView":
                return new exports.AuthorizedView(name, undefined, { urn });
            case "gcp:bigtable/gCPolicy:GCPolicy":
                return new exports.GCPolicy(name, undefined, { urn });
            case "gcp:bigtable/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "gcp:bigtable/instanceIamBinding:InstanceIamBinding":
                return new exports.InstanceIamBinding(name, undefined, { urn });
            case "gcp:bigtable/instanceIamMember:InstanceIamMember":
                return new exports.InstanceIamMember(name, undefined, { urn });
            case "gcp:bigtable/instanceIamPolicy:InstanceIamPolicy":
                return new exports.InstanceIamPolicy(name, undefined, { urn });
            case "gcp:bigtable/logicalView:LogicalView":
                return new exports.LogicalView(name, undefined, { urn });
            case "gcp:bigtable/materializedView:MaterializedView":
                return new exports.MaterializedView(name, undefined, { urn });
            case "gcp:bigtable/schemaBundle:SchemaBundle":
                return new exports.SchemaBundle(name, undefined, { urn });
            case "gcp:bigtable/table:Table":
                return new exports.Table(name, undefined, { urn });
            case "gcp:bigtable/tableIamBinding:TableIamBinding":
                return new exports.TableIamBinding(name, undefined, { urn });
            case "gcp:bigtable/tableIamMember:TableIamMember":
                return new exports.TableIamMember(name, undefined, { urn });
            case "gcp:bigtable/tableIamPolicy:TableIamPolicy":
                return new exports.TableIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigtable/appProfile", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/authorizedView", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/gCPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instance", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/logicalView", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/materializedView", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/schemaBundle", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/table", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamPolicy", _module);
