// @pulumi/gcp/bigtable/logicalView.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogicalView = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A logical view object that can be referenced in SQL queries.
 *
 * To get more information about LogicalView, see:
 *
 * * [API documentation](https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/projects.instances.logicalViews)
 *
 * ## Example Usage
 *
 * ### Bigtable Logical View
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const instance = new gcp.bigtable.Instance("instance", {
 *     name: "bt-instance",
 *     clusters: [{
 *         clusterId: "cluster-1",
 *         zone: "us-east1-b",
 *         numNodes: 3,
 *         storageType: "HDD",
 *     }],
 *     deletionProtection: false,
 * });
 * const table = new gcp.bigtable.Table("table", {
 *     name: "bt-table",
 *     instanceName: instance.name,
 *     columnFamilies: [{
 *         family: "CF",
 *     }],
 * });
 * const logicalView = new gcp.bigtable.LogicalView("logical_view", {
 *     logicalViewId: "bt-logical-view",
 *     instance: instance.name,
 *     deletionProtection: false,
 *     query: `SELECT _key, CF
 * FROM \` + \\"\`bt-table\`\\" + \`
 * `,
 * }, {
 *     dependsOn: [table],
 * });
 * ```
 *
 * ## Import
 *
 * LogicalView can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/instances/{{instance}}/logicalViews/{{logical_view_id}}`
 * * `{{project}}/{{instance}}/{{logical_view_id}}`
 * * `{{instance}}/{{logical_view_id}}`
 *
 * When using the `pulumi import` command, LogicalView can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:bigtable/logicalView:LogicalView default projects/{{project}}/instances/{{instance}}/logicalViews/{{logical_view_id}}
 * $ pulumi import gcp:bigtable/logicalView:LogicalView default {{project}}/{{instance}}/{{logical_view_id}}
 * $ pulumi import gcp:bigtable/logicalView:LogicalView default {{instance}}/{{logical_view_id}}
 * ```
 */
class LogicalView extends pulumi.CustomResource {
    /**
     * Get an existing LogicalView resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LogicalView(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:bigtable/logicalView:LogicalView';
    /**
     * Returns true if the given object is an instance of LogicalView.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogicalView.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["deletionProtection"] = state?.deletionProtection;
            resourceInputs["instance"] = state?.instance;
            resourceInputs["logicalViewId"] = state?.logicalViewId;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["query"] = state?.query;
        }
        else {
            const args = argsOrState;
            if (args?.logicalViewId === undefined && !opts.urn) {
                throw new Error("Missing required property 'logicalViewId'");
            }
            if (args?.query === undefined && !opts.urn) {
                throw new Error("Missing required property 'query'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["deletionProtection"] = args?.deletionProtection;
            resourceInputs["instance"] = args?.instance;
            resourceInputs["logicalViewId"] = args?.logicalViewId;
            resourceInputs["project"] = args?.project;
            resourceInputs["query"] = args?.query;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LogicalView.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogicalView = LogicalView;
