// @pulumi/aws/ecr/pullThroughCacheRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PullThroughCacheRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Elastic Container Registry Pull Through Cache Rule.
 *
 * More information about pull through cache rules, including the set of supported
 * upstream repositories, see [Using pull through cache rules](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache.html).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecr.PullThroughCacheRule("example", {
 *     ecrRepositoryPrefix: "ecr-public",
 *     upstreamRegistryUrl: "public.ecr.aws",
 *     credentialArn: "arn:aws:secretsmanager:us-east-1:123456789:secret:ecr-pullthroughcache/ecrpublic",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a pull-through cache rule using the `ecrRepositoryPrefix`. For example:
 *
 * ```sh
 * $ pulumi import aws:ecr/pullThroughCacheRule:PullThroughCacheRule example ecr-public
 * ```
 */
class PullThroughCacheRule extends pulumi.CustomResource {
    /**
     * Get an existing PullThroughCacheRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PullThroughCacheRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ecr/pullThroughCacheRule:PullThroughCacheRule';
    /**
     * Returns true if the given object is an instance of PullThroughCacheRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PullThroughCacheRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["credentialArn"] = state?.credentialArn;
            resourceInputs["customRoleArn"] = state?.customRoleArn;
            resourceInputs["ecrRepositoryPrefix"] = state?.ecrRepositoryPrefix;
            resourceInputs["region"] = state?.region;
            resourceInputs["registryId"] = state?.registryId;
            resourceInputs["upstreamRegistryUrl"] = state?.upstreamRegistryUrl;
            resourceInputs["upstreamRepositoryPrefix"] = state?.upstreamRepositoryPrefix;
        }
        else {
            const args = argsOrState;
            if (args?.ecrRepositoryPrefix === undefined && !opts.urn) {
                throw new Error("Missing required property 'ecrRepositoryPrefix'");
            }
            if (args?.upstreamRegistryUrl === undefined && !opts.urn) {
                throw new Error("Missing required property 'upstreamRegistryUrl'");
            }
            resourceInputs["credentialArn"] = args?.credentialArn;
            resourceInputs["customRoleArn"] = args?.customRoleArn;
            resourceInputs["ecrRepositoryPrefix"] = args?.ecrRepositoryPrefix;
            resourceInputs["region"] = args?.region;
            resourceInputs["upstreamRegistryUrl"] = args?.upstreamRegistryUrl;
            resourceInputs["upstreamRepositoryPrefix"] = args?.upstreamRepositoryPrefix;
            resourceInputs["registryId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PullThroughCacheRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PullThroughCacheRule = PullThroughCacheRule;
