// @pulumi/azure-native/managednetworkfabric/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getNetworkToNetworkInterconnectOutput = exports.getNetworkToNetworkInterconnect = exports.getNetworkTapRuleOutput = exports.getNetworkTapRule = exports.getNetworkTapOutput = exports.getNetworkTap = exports.getNetworkRackOutput = exports.getNetworkRack = exports.getNetworkPacketBrokerOutput = exports.getNetworkPacketBroker = exports.getNetworkMonitorOutput = exports.getNetworkMonitor = exports.getNetworkInterfaceStatusOutput = exports.getNetworkInterfaceStatus = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getNetworkFabricTopologyOutput = exports.getNetworkFabricTopology = exports.getNetworkFabricControllerOutput = exports.getNetworkFabricController = exports.getNetworkFabricOutput = exports.getNetworkFabric = exports.getNetworkDeviceStatusOutput = exports.getNetworkDeviceStatus = exports.getNetworkDeviceOutput = exports.getNetworkDevice = exports.getNeighborGroupOutput = exports.getNeighborGroup = exports.getL3IsolationDomainOutput = exports.getL3IsolationDomain = exports.getL2IsolationDomainOutput = exports.getL2IsolationDomain = exports.getIpPrefixOutput = exports.getIpPrefix = exports.getIpExtendedCommunityOutput = exports.getIpExtendedCommunity = exports.getIpCommunityOutput = exports.getIpCommunity = exports.getInternetGatewayRuleOutput = exports.getInternetGatewayRule = exports.getInternetGatewayOutput = exports.getInternetGateway = exports.getInternalNetworkOutput = exports.getInternalNetwork = exports.getExternalNetworkOutput = exports.getExternalNetwork = exports.getAccessControlListOutput = exports.getAccessControlList = exports.ExternalNetwork = exports.AccessControlList = void 0;
exports.RoutePolicy = exports.NetworkToNetworkInterconnect = exports.NetworkTapRule = exports.NetworkTap = exports.NetworkRack = exports.NetworkPacketBroker = exports.NetworkMonitor = exports.NetworkInterface = exports.NetworkFabricController = exports.NetworkFabric = exports.NetworkDevice = exports.NeighborGroup = exports.L3IsolationDomain = exports.L2IsolationDomain = exports.IpPrefix = exports.IpExtendedCommunity = exports.IpCommunity = exports.InternetGatewayRule = exports.InternetGateway = exports.InternalNetwork = exports.getRoutePolicyOutput = exports.getRoutePolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccessControlList = null;
utilities.lazyLoad(exports, ["AccessControlList"], () => require("./accessControlList"));
exports.ExternalNetwork = null;
utilities.lazyLoad(exports, ["ExternalNetwork"], () => require("./externalNetwork"));
exports.getAccessControlList = null;
exports.getAccessControlListOutput = null;
utilities.lazyLoad(exports, ["getAccessControlList", "getAccessControlListOutput"], () => require("./getAccessControlList"));
exports.getExternalNetwork = null;
exports.getExternalNetworkOutput = null;
utilities.lazyLoad(exports, ["getExternalNetwork", "getExternalNetworkOutput"], () => require("./getExternalNetwork"));
exports.getInternalNetwork = null;
exports.getInternalNetworkOutput = null;
utilities.lazyLoad(exports, ["getInternalNetwork", "getInternalNetworkOutput"], () => require("./getInternalNetwork"));
exports.getInternetGateway = null;
exports.getInternetGatewayOutput = null;
utilities.lazyLoad(exports, ["getInternetGateway", "getInternetGatewayOutput"], () => require("./getInternetGateway"));
exports.getInternetGatewayRule = null;
exports.getInternetGatewayRuleOutput = null;
utilities.lazyLoad(exports, ["getInternetGatewayRule", "getInternetGatewayRuleOutput"], () => require("./getInternetGatewayRule"));
exports.getIpCommunity = null;
exports.getIpCommunityOutput = null;
utilities.lazyLoad(exports, ["getIpCommunity", "getIpCommunityOutput"], () => require("./getIpCommunity"));
exports.getIpExtendedCommunity = null;
exports.getIpExtendedCommunityOutput = null;
utilities.lazyLoad(exports, ["getIpExtendedCommunity", "getIpExtendedCommunityOutput"], () => require("./getIpExtendedCommunity"));
exports.getIpPrefix = null;
exports.getIpPrefixOutput = null;
utilities.lazyLoad(exports, ["getIpPrefix", "getIpPrefixOutput"], () => require("./getIpPrefix"));
exports.getL2IsolationDomain = null;
exports.getL2IsolationDomainOutput = null;
utilities.lazyLoad(exports, ["getL2IsolationDomain", "getL2IsolationDomainOutput"], () => require("./getL2IsolationDomain"));
exports.getL3IsolationDomain = null;
exports.getL3IsolationDomainOutput = null;
utilities.lazyLoad(exports, ["getL3IsolationDomain", "getL3IsolationDomainOutput"], () => require("./getL3IsolationDomain"));
exports.getNeighborGroup = null;
exports.getNeighborGroupOutput = null;
utilities.lazyLoad(exports, ["getNeighborGroup", "getNeighborGroupOutput"], () => require("./getNeighborGroup"));
exports.getNetworkDevice = null;
exports.getNetworkDeviceOutput = null;
utilities.lazyLoad(exports, ["getNetworkDevice", "getNetworkDeviceOutput"], () => require("./getNetworkDevice"));
exports.getNetworkDeviceStatus = null;
exports.getNetworkDeviceStatusOutput = null;
utilities.lazyLoad(exports, ["getNetworkDeviceStatus", "getNetworkDeviceStatusOutput"], () => require("./getNetworkDeviceStatus"));
exports.getNetworkFabric = null;
exports.getNetworkFabricOutput = null;
utilities.lazyLoad(exports, ["getNetworkFabric", "getNetworkFabricOutput"], () => require("./getNetworkFabric"));
exports.getNetworkFabricController = null;
exports.getNetworkFabricControllerOutput = null;
utilities.lazyLoad(exports, ["getNetworkFabricController", "getNetworkFabricControllerOutput"], () => require("./getNetworkFabricController"));
exports.getNetworkFabricTopology = null;
exports.getNetworkFabricTopologyOutput = null;
utilities.lazyLoad(exports, ["getNetworkFabricTopology", "getNetworkFabricTopologyOutput"], () => require("./getNetworkFabricTopology"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getNetworkInterfaceStatus = null;
exports.getNetworkInterfaceStatusOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterfaceStatus", "getNetworkInterfaceStatusOutput"], () => require("./getNetworkInterfaceStatus"));
exports.getNetworkMonitor = null;
exports.getNetworkMonitorOutput = null;
utilities.lazyLoad(exports, ["getNetworkMonitor", "getNetworkMonitorOutput"], () => require("./getNetworkMonitor"));
exports.getNetworkPacketBroker = null;
exports.getNetworkPacketBrokerOutput = null;
utilities.lazyLoad(exports, ["getNetworkPacketBroker", "getNetworkPacketBrokerOutput"], () => require("./getNetworkPacketBroker"));
exports.getNetworkRack = null;
exports.getNetworkRackOutput = null;
utilities.lazyLoad(exports, ["getNetworkRack", "getNetworkRackOutput"], () => require("./getNetworkRack"));
exports.getNetworkTap = null;
exports.getNetworkTapOutput = null;
utilities.lazyLoad(exports, ["getNetworkTap", "getNetworkTapOutput"], () => require("./getNetworkTap"));
exports.getNetworkTapRule = null;
exports.getNetworkTapRuleOutput = null;
utilities.lazyLoad(exports, ["getNetworkTapRule", "getNetworkTapRuleOutput"], () => require("./getNetworkTapRule"));
exports.getNetworkToNetworkInterconnect = null;
exports.getNetworkToNetworkInterconnectOutput = null;
utilities.lazyLoad(exports, ["getNetworkToNetworkInterconnect", "getNetworkToNetworkInterconnectOutput"], () => require("./getNetworkToNetworkInterconnect"));
exports.getRoutePolicy = null;
exports.getRoutePolicyOutput = null;
utilities.lazyLoad(exports, ["getRoutePolicy", "getRoutePolicyOutput"], () => require("./getRoutePolicy"));
exports.InternalNetwork = null;
utilities.lazyLoad(exports, ["InternalNetwork"], () => require("./internalNetwork"));
exports.InternetGateway = null;
utilities.lazyLoad(exports, ["InternetGateway"], () => require("./internetGateway"));
exports.InternetGatewayRule = null;
utilities.lazyLoad(exports, ["InternetGatewayRule"], () => require("./internetGatewayRule"));
exports.IpCommunity = null;
utilities.lazyLoad(exports, ["IpCommunity"], () => require("./ipCommunity"));
exports.IpExtendedCommunity = null;
utilities.lazyLoad(exports, ["IpExtendedCommunity"], () => require("./ipExtendedCommunity"));
exports.IpPrefix = null;
utilities.lazyLoad(exports, ["IpPrefix"], () => require("./ipPrefix"));
exports.L2IsolationDomain = null;
utilities.lazyLoad(exports, ["L2IsolationDomain"], () => require("./l2isolationDomain"));
exports.L3IsolationDomain = null;
utilities.lazyLoad(exports, ["L3IsolationDomain"], () => require("./l3isolationDomain"));
exports.NeighborGroup = null;
utilities.lazyLoad(exports, ["NeighborGroup"], () => require("./neighborGroup"));
exports.NetworkDevice = null;
utilities.lazyLoad(exports, ["NetworkDevice"], () => require("./networkDevice"));
exports.NetworkFabric = null;
utilities.lazyLoad(exports, ["NetworkFabric"], () => require("./networkFabric"));
exports.NetworkFabricController = null;
utilities.lazyLoad(exports, ["NetworkFabricController"], () => require("./networkFabricController"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.NetworkMonitor = null;
utilities.lazyLoad(exports, ["NetworkMonitor"], () => require("./networkMonitor"));
exports.NetworkPacketBroker = null;
utilities.lazyLoad(exports, ["NetworkPacketBroker"], () => require("./networkPacketBroker"));
exports.NetworkRack = null;
utilities.lazyLoad(exports, ["NetworkRack"], () => require("./networkRack"));
exports.NetworkTap = null;
utilities.lazyLoad(exports, ["NetworkTap"], () => require("./networkTap"));
exports.NetworkTapRule = null;
utilities.lazyLoad(exports, ["NetworkTapRule"], () => require("./networkTapRule"));
exports.NetworkToNetworkInterconnect = null;
utilities.lazyLoad(exports, ["NetworkToNetworkInterconnect"], () => require("./networkToNetworkInterconnect"));
exports.RoutePolicy = null;
utilities.lazyLoad(exports, ["RoutePolicy"], () => require("./routePolicy"));
// Export enums:
__exportStar(require("../types/enums/managednetworkfabric"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managednetworkfabric:AccessControlList":
                return new exports.AccessControlList(name, undefined, { urn });
            case "azure-native:managednetworkfabric:ExternalNetwork":
                return new exports.ExternalNetwork(name, undefined, { urn });
            case "azure-native:managednetworkfabric:InternalNetwork":
                return new exports.InternalNetwork(name, undefined, { urn });
            case "azure-native:managednetworkfabric:InternetGateway":
                return new exports.InternetGateway(name, undefined, { urn });
            case "azure-native:managednetworkfabric:InternetGatewayRule":
                return new exports.InternetGatewayRule(name, undefined, { urn });
            case "azure-native:managednetworkfabric:IpCommunity":
                return new exports.IpCommunity(name, undefined, { urn });
            case "azure-native:managednetworkfabric:IpExtendedCommunity":
                return new exports.IpExtendedCommunity(name, undefined, { urn });
            case "azure-native:managednetworkfabric:IpPrefix":
                return new exports.IpPrefix(name, undefined, { urn });
            case "azure-native:managednetworkfabric:L2IsolationDomain":
                return new exports.L2IsolationDomain(name, undefined, { urn });
            case "azure-native:managednetworkfabric:L3IsolationDomain":
                return new exports.L3IsolationDomain(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NeighborGroup":
                return new exports.NeighborGroup(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkDevice":
                return new exports.NetworkDevice(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkFabric":
                return new exports.NetworkFabric(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkFabricController":
                return new exports.NetworkFabricController(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkMonitor":
                return new exports.NetworkMonitor(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkPacketBroker":
                return new exports.NetworkPacketBroker(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkRack":
                return new exports.NetworkRack(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkTap":
                return new exports.NetworkTap(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkTapRule":
                return new exports.NetworkTapRule(name, undefined, { urn });
            case "azure-native:managednetworkfabric:NetworkToNetworkInterconnect":
                return new exports.NetworkToNetworkInterconnect(name, undefined, { urn });
            case "azure-native:managednetworkfabric:RoutePolicy":
                return new exports.RoutePolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managednetworkfabric", _module);
