// @pulumi/azure-native/managednetworkfabric/internalNetwork.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalNetwork = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * Defines the Internal Network resource.
 *
 * Uses Azure REST API version 2023-06-15. In version 2.x of the Azure Native provider, it used API version 2023-02-01-preview.
 *
 * Other available API versions: 2023-02-01-preview, 2024-02-15-preview, 2024-06-15-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native managednetworkfabric [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class InternalNetwork extends pulumi.CustomResource {
    /**
     * Get an existing InternalNetwork resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InternalNetwork(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:managednetworkfabric:InternalNetwork';
    /**
     * Returns true if the given object is an instance of InternalNetwork.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InternalNetwork.__pulumiType;
    }
    /**
     * Create a InternalNetwork resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.l3IsolationDomainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'l3IsolationDomainName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.vlanId === undefined && !opts.urn) {
                throw new Error("Missing required property 'vlanId'");
            }
            resourceInputs["annotation"] = args?.annotation;
            resourceInputs["bgpConfiguration"] = args ? (args.bgpConfiguration ? pulumi.output(args.bgpConfiguration).apply(inputs.managednetworkfabric.internalNetworkPropertiesBgpConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["connectedIPv4Subnets"] = args?.connectedIPv4Subnets;
            resourceInputs["connectedIPv6Subnets"] = args?.connectedIPv6Subnets;
            resourceInputs["egressAclId"] = args?.egressAclId;
            resourceInputs["exportRoutePolicy"] = args?.exportRoutePolicy;
            resourceInputs["exportRoutePolicyId"] = args?.exportRoutePolicyId;
            resourceInputs["extension"] = (args?.extension) ?? "NoExtension";
            resourceInputs["importRoutePolicy"] = args?.importRoutePolicy;
            resourceInputs["importRoutePolicyId"] = args?.importRoutePolicyId;
            resourceInputs["ingressAclId"] = args?.ingressAclId;
            resourceInputs["internalNetworkName"] = args?.internalNetworkName;
            resourceInputs["isMonitoringEnabled"] = (args?.isMonitoringEnabled) ?? "False";
            resourceInputs["l3IsolationDomainName"] = args?.l3IsolationDomainName;
            resourceInputs["mtu"] = (args?.mtu) ?? 1500;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["staticRouteConfiguration"] = args ? (args.staticRouteConfiguration ? pulumi.output(args.staticRouteConfiguration).apply(inputs.managednetworkfabric.internalNetworkPropertiesStaticRouteConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["vlanId"] = args?.vlanId;
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["annotation"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bgpConfiguration"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["connectedIPv4Subnets"] = undefined /*out*/;
            resourceInputs["connectedIPv6Subnets"] = undefined /*out*/;
            resourceInputs["egressAclId"] = undefined /*out*/;
            resourceInputs["exportRoutePolicy"] = undefined /*out*/;
            resourceInputs["exportRoutePolicyId"] = undefined /*out*/;
            resourceInputs["extension"] = undefined /*out*/;
            resourceInputs["importRoutePolicy"] = undefined /*out*/;
            resourceInputs["importRoutePolicyId"] = undefined /*out*/;
            resourceInputs["ingressAclId"] = undefined /*out*/;
            resourceInputs["isMonitoringEnabled"] = undefined /*out*/;
            resourceInputs["mtu"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["staticRouteConfiguration"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vlanId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:managednetworkfabric/v20230201preview:InternalNetwork" }, { type: "azure-native:managednetworkfabric/v20230615:InternalNetwork" }, { type: "azure-native:managednetworkfabric/v20240215preview:InternalNetwork" }, { type: "azure-native:managednetworkfabric/v20240615preview:InternalNetwork" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InternalNetwork.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InternalNetwork = InternalNetwork;
