// @pulumi/aws/schemas/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = exports.RegistryPolicy = exports.Registry = exports.Discoverer = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Discoverer = null;
utilities.lazyLoad(exports, ["Discoverer"], () => require("./discoverer"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.RegistryPolicy = null;
utilities.lazyLoad(exports, ["RegistryPolicy"], () => require("./registryPolicy"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:schemas/discoverer:Discoverer":
                return new exports.Discoverer(name, undefined, { urn });
            case "aws:schemas/registry:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "aws:schemas/registryPolicy:RegistryPolicy":
                return new exports.RegistryPolicy(name, undefined, { urn });
            case "aws:schemas/schema:Schema":
                return new exports.Schema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "schemas/discoverer", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/registry", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/registryPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "schemas/schema", _module);
