// @pulumi/gcp/apihub/pluginInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PluginInstance = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Description
 *
 * ## Example Usage
 *
 * ### Apihub Plugin Instance Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const apihubPluginInstanceBasic = new gcp.apihub.PluginInstance("apihub_plugin_instance_basic", {
 *     location: "us-central1",
 *     plugin: "existing-plugin-id",
 *     pluginInstanceId: "test",
 *     displayName: "Sample Plugin Instance Display Name",
 *     disable: false,
 *     actions: [{
 *         actionId: "existing-action-id",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * PluginInstance can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/plugins/{{plugin}}/instances/{{plugin_instance_id}}`
 * * `{{project}}/{{location}}/{{plugin}}/{{plugin_instance_id}}`
 * * `{{location}}/{{plugin}}/{{plugin_instance_id}}`
 *
 * When using the `pulumi import` command, PluginInstance can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:apihub/pluginInstance:PluginInstance default projects/{{project}}/locations/{{location}}/plugins/{{plugin}}/instances/{{plugin_instance_id}}
 * $ pulumi import gcp:apihub/pluginInstance:PluginInstance default {{project}}/{{location}}/{{plugin}}/{{plugin_instance_id}}
 * $ pulumi import gcp:apihub/pluginInstance:PluginInstance default {{location}}/{{plugin}}/{{plugin_instance_id}}
 * ```
 */
class PluginInstance extends pulumi.CustomResource {
    /**
     * Get an existing PluginInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PluginInstance(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:apihub/pluginInstance:PluginInstance';
    /**
     * Returns true if the given object is an instance of PluginInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PluginInstance.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["actions"] = state?.actions;
            resourceInputs["authConfig"] = state?.authConfig;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["disable"] = state?.disable;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["errorMessage"] = state?.errorMessage;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["plugin"] = state?.plugin;
            resourceInputs["pluginInstanceId"] = state?.pluginInstanceId;
            resourceInputs["project"] = state?.project;
            resourceInputs["state"] = state?.state;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.plugin === undefined && !opts.urn) {
                throw new Error("Missing required property 'plugin'");
            }
            if (args?.pluginInstanceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'pluginInstanceId'");
            }
            resourceInputs["actions"] = args?.actions;
            resourceInputs["authConfig"] = args?.authConfig;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["disable"] = args?.disable;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["location"] = args?.location;
            resourceInputs["plugin"] = args?.plugin;
            resourceInputs["pluginInstanceId"] = args?.pluginInstanceId;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PluginInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PluginInstance = PluginInstance;
