// @pulumi/kubernetes/rbac/v1beta1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RolePatch = exports.RoleList = exports.RoleBindingPatch = exports.RoleBindingList = exports.RoleBinding = exports.Role = exports.ClusterRolePatch = exports.ClusterRoleList = exports.ClusterRoleBindingPatch = exports.ClusterRoleBindingList = exports.ClusterRoleBinding = exports.ClusterRole = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.ClusterRole = null;
utilities.lazyLoad(exports, ["ClusterRole"], () => require("./clusterRole"));
exports.ClusterRoleBinding = null;
utilities.lazyLoad(exports, ["ClusterRoleBinding"], () => require("./clusterRoleBinding"));
exports.ClusterRoleBindingList = null;
utilities.lazyLoad(exports, ["ClusterRoleBindingList"], () => require("./clusterRoleBindingList"));
exports.ClusterRoleBindingPatch = null;
utilities.lazyLoad(exports, ["ClusterRoleBindingPatch"], () => require("./clusterRoleBindingPatch"));
exports.ClusterRoleList = null;
utilities.lazyLoad(exports, ["ClusterRoleList"], () => require("./clusterRoleList"));
exports.ClusterRolePatch = null;
utilities.lazyLoad(exports, ["ClusterRolePatch"], () => require("./clusterRolePatch"));
exports.Role = null;
utilities.lazyLoad(exports, ["Role"], () => require("./role"));
exports.RoleBinding = null;
utilities.lazyLoad(exports, ["RoleBinding"], () => require("./roleBinding"));
exports.RoleBindingList = null;
utilities.lazyLoad(exports, ["RoleBindingList"], () => require("./roleBindingList"));
exports.RoleBindingPatch = null;
utilities.lazyLoad(exports, ["RoleBindingPatch"], () => require("./roleBindingPatch"));
exports.RoleList = null;
utilities.lazyLoad(exports, ["RoleList"], () => require("./roleList"));
exports.RolePatch = null;
utilities.lazyLoad(exports, ["RolePatch"], () => require("./rolePatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRole":
                return new exports.ClusterRole(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRoleBinding":
                return new exports.ClusterRoleBinding(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRoleBindingList":
                return new exports.ClusterRoleBindingList(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRoleBindingPatch":
                return new exports.ClusterRoleBindingPatch(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRoleList":
                return new exports.ClusterRoleList(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:ClusterRolePatch":
                return new exports.ClusterRolePatch(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:Role":
                return new exports.Role(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:RoleBinding":
                return new exports.RoleBinding(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:RoleBindingList":
                return new exports.RoleBindingList(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:RoleBindingPatch":
                return new exports.RoleBindingPatch(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:RoleList":
                return new exports.RoleList(name, undefined, { urn });
            case "kubernetes:rbac.authorization.k8s.io/v1beta1:RolePatch":
                return new exports.RolePatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "rbac.authorization.k8s.io/v1beta1", _module);
