// @pulumi/aws/wafv2/webAclRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAclRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an individual rule within a WAFv2 Web ACL. This resource creates proper Terraform dependencies for safe deletion of referenced resources like IP sets, solving the `WAFAssociatedItemException` error that occurs when deleting IP sets that are still referenced by Web ACL rules.
 *
 * > **NOTE:** When using this resource, you must add `lifecycle { ignoreChanges = [rule] }` to your `aws.wafv2.WebAcl` resource to prevent conflicts. See the `aws.wafv2.WebAcl` documentation for a full description of the limitations of inline rules that this resource addresses.
 *
 * ## Example Usage
 *
 * ### Migrating from Inline Rules
 *
 * This resource supports a "create-or-adopt" pattern that allows seamless migration from inline Web ACL rules to separate `aws.wafv2.WebAclRule` resources without infrastructure changes.
 *
 * When you create an `aws.wafv2.WebAclRule` resource with the same name as an existing inline rule in the Web ACL, the resource will automatically adopt the existing rule instead of creating a duplicate. This enables zero-downtime migration from inline rules to separate resources.
 *
 * Starting with inline rules, update your configuration to use separate rule resources and apply:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.wafv2.WebAcl("example", {
 *     name: "example",
 *     scope: "REGIONAL",
 *     defaultAction: {
 *         allow: {},
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: false,
 *         metricName: "example",
 *         sampledRequestsEnabled: false,
 *     },
 * });
 * // Separate rule resource with identical configuration
 * const blockCountries = new aws.wafv2.WebAclRule("block_countries", {
 *     name: "block-countries",
 *     priority: 1,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         geoMatchStatement: {
 *             countryCodes: [
 *                 "CN",
 *                 "RU",
 *             ],
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: false,
 *         metricName: "block-countries",
 *         sampledRequestsEnabled: false,
 *     },
 * });
 * ```
 *
 * Apply the configuration:
 *
 * The `aws.wafv2.WebAclRule` resource will adopt the existing inline rule without making any changes to the actual Web ACL infrastructure. The rule continues to function identically, but is now managed as a separate Terraform resource.
 *
 * - The rule name in the `aws.wafv2.WebAclRule` resource must exactly match the existing inline rule name
 * - Add `lifecycle { ignoreChanges = [rule] }` to your Web ACL resource to prevent conflicts
 * - The create-or-adopt behavior only applies when a rule with the same name already exists in the Web ACL
 *
 * ### Basic Geo Match Rule
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.wafv2.WebAcl("example", {
 *     name: "example",
 *     scope: "REGIONAL",
 *     defaultAction: {
 *         allow: {},
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: false,
 *         metricName: "example",
 *         sampledRequestsEnabled: false,
 *     },
 * });
 * const blockCountries = new aws.wafv2.WebAclRule("block_countries", {
 *     name: "block-countries",
 *     priority: 1,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         geoMatchStatement: {
 *             countryCodes: [
 *                 "CN",
 *                 "RU",
 *             ],
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: false,
 *         metricName: "block-countries",
 *         sampledRequestsEnabled: false,
 *     },
 * });
 * ```
 *
 * ### IP Set Reference (Solves Deletion Ordering)
 *
 * This example demonstrates the primary use case: referencing an IP set in a way that allows safe deletion.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const blockedIps = new aws.wafv2.IpSet("blocked_ips", {
 *     name: "blocked-ips",
 *     scope: "REGIONAL",
 *     ipAddressVersion: "IPV4",
 *     addresses: [
 *         "1.2.3.4/32",
 *         "5.6.7.8/32",
 *     ],
 * });
 * const example = new aws.wafv2.WebAcl("example", {
 *     name: "example",
 *     scope: "REGIONAL",
 *     defaultAction: {
 *         allow: {},
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "example",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * const blockIps = new aws.wafv2.WebAclRule("block_ips", {
 *     name: "block-bad-ips",
 *     priority: 1,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         ipSetReferenceStatement: {
 *             arn: blockedIps.arn,
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "block-bad-ips",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### Rate-Based Rule
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const rateLimit = new aws.wafv2.WebAclRule("rate_limit", {
 *     name: "rate-limit",
 *     priority: 2,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         rateBasedStatement: {
 *             limit: 2000,
 *             aggregateKeyType: "IP",
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "rate-limit",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### Managed Rule Group with Override Action
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const awsManagedRules = new aws.wafv2.WebAclRule("aws_managed_rules", {
 *     name: "aws-managed-rules",
 *     priority: 3,
 *     webAclArn: example.arn,
 *     overrideAction: {
 *         none: {},
 *     },
 *     statement: {
 *         managedRuleGroupStatement: {
 *             name: "AWSManagedRulesCommonRuleSet",
 *             vendorName: "AWS",
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "aws-managed-rules",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### Custom Request Handling
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const captchaWithHeaders = new aws.wafv2.WebAclRule("captcha_with_headers", {
 *     name: "captcha-with-headers",
 *     priority: 4,
 *     webAclArn: example.arn,
 *     action: {
 *         captcha: {
 *             customRequestHandling: {
 *                 insertHeaders: [{
 *                     name: "x-captcha-rule",
 *                     value: "triggered",
 *                 }],
 *             },
 *         },
 *     },
 *     statement: {
 *         geoMatchStatement: {
 *             countryCodes: ["US"],
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "captcha-with-headers",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### IP Set Reference
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const blockedIps = new aws.wafv2.WebAclRule("blocked_ips", {
 *     name: "blocked-ips",
 *     priority: 1,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         ipSetReferenceStatement: {
 *             arn: blockedIpsAwsWafv2IpSet.arn,
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "block-bad-ips",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * With this configuration, when you remove both the `aws.wafv2.WebAclRule` and `aws.wafv2.IpSet` resources, Terraform will:
 *
 * 1. Delete the rule first (removing the reference from the Web ACL)
 * 2. Delete the IP set second (now safe because it's no longer referenced)
 *
 * This prevents the `WAFAssociatedItemException` error.
 *
 * ### Logical AND Statement
 *
 * Block requests that match multiple conditions (e.g., from a specific country AND containing a specific string):
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const blockSuspicious = new aws.wafv2.WebAclRule("block_suspicious", {
 *     name: "block-suspicious",
 *     priority: 1,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         andStatement: {
 *             statements: [
 *                 {
 *                     geoMatchStatement: {
 *                         countryCodes: ["CN"],
 *                     },
 *                 },
 *                 {
 *                     byteMatchStatement: {
 *                         searchString: "admin",
 *                         positionalConstraint: "CONTAINS",
 *                         fieldToMatch: {
 *                             uriPath: {},
 *                         },
 *                         textTransformations: [{
 *                             priority: 0,
 *                             type: "LOWERCASE",
 *                         }],
 *                     },
 *                 },
 *             ],
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "block-suspicious",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### Logical OR Statement
 *
 * Block requests that match any of multiple conditions:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const blockCountries = new aws.wafv2.WebAclRule("block_countries", {
 *     name: "block-countries",
 *     priority: 2,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         orStatement: {
 *             statements: [
 *                 {
 *                     geoMatchStatement: {
 *                         countryCodes: ["CN"],
 *                     },
 *                 },
 *                 {
 *                     geoMatchStatement: {
 *                         countryCodes: ["RU"],
 *                     },
 *                 },
 *             ],
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "block-countries",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ### Logical NOT Statement
 *
 * Allow requests only from specific countries by negating a geo match:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const allowOnlyUs = new aws.wafv2.WebAclRule("allow_only_us", {
 *     name: "allow-only-us",
 *     priority: 3,
 *     webAclArn: example.arn,
 *     action: {
 *         block: {},
 *     },
 *     statement: {
 *         notStatement: {
 *             statement: {
 *                 geoMatchStatement: {
 *                     countryCodes: [
 *                         "US",
 *                         "CA",
 *                     ],
 *                 },
 *             },
 *         },
 *     },
 *     visibilityConfig: {
 *         cloudwatchMetricsEnabled: true,
 *         metricName: "allow-only-us",
 *         sampledRequestsEnabled: true,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `name` (String) Rule name, unique within the Web ACL.
 * * `webAclArn` (String) ARN of the Web ACL.
 *
 * #### Optional
 *
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import WAFv2 Web ACL Rules using the `webAclArn` and `name` separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:wafv2/webAclRule:WebAclRule example arn:aws:wafv2:us-east-1:123456789012:regional/webacl/example/abc123def456,my-rule
 * ```
 */
class WebAclRule extends pulumi.CustomResource {
    /**
     * Get an existing WebAclRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WebAclRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:wafv2/webAclRule:WebAclRule';
    /**
     * Returns true if the given object is an instance of WebAclRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAclRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["action"] = state?.action;
            resourceInputs["captchaConfig"] = state?.captchaConfig;
            resourceInputs["challengeConfig"] = state?.challengeConfig;
            resourceInputs["name"] = state?.name;
            resourceInputs["overrideAction"] = state?.overrideAction;
            resourceInputs["priority"] = state?.priority;
            resourceInputs["region"] = state?.region;
            resourceInputs["ruleLabels"] = state?.ruleLabels;
            resourceInputs["statement"] = state?.statement;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["visibilityConfig"] = state?.visibilityConfig;
            resourceInputs["webAclArn"] = state?.webAclArn;
        }
        else {
            const args = argsOrState;
            if (args?.priority === undefined && !opts.urn) {
                throw new Error("Missing required property 'priority'");
            }
            if (args?.webAclArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'webAclArn'");
            }
            resourceInputs["action"] = args?.action;
            resourceInputs["captchaConfig"] = args?.captchaConfig;
            resourceInputs["challengeConfig"] = args?.challengeConfig;
            resourceInputs["name"] = args?.name;
            resourceInputs["overrideAction"] = args?.overrideAction;
            resourceInputs["priority"] = args?.priority;
            resourceInputs["region"] = args?.region;
            resourceInputs["ruleLabels"] = args?.ruleLabels;
            resourceInputs["statement"] = args?.statement;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["visibilityConfig"] = args?.visibilityConfig;
            resourceInputs["webAclArn"] = args?.webAclArn;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WebAclRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAclRule = WebAclRule;
