// @pulumi/aws/organizations/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = exports.ResourcePolicy = exports.PolicyAttachment = exports.Policy = exports.OrganizationalUnit = exports.Organization = exports.getResourceTagsOutput = exports.getResourceTags = exports.getPolicyOutput = exports.getPolicy = exports.getPoliciesForTargetOutput = exports.getPoliciesForTarget = exports.getPoliciesOutput = exports.getPolicies = exports.getOrganizationalUnitsOutput = exports.getOrganizationalUnits = exports.getOrganizationalUnitDescendantOrganizationalUnitsOutput = exports.getOrganizationalUnitDescendantOrganizationalUnits = exports.getOrganizationalUnitDescendantAccountsOutput = exports.getOrganizationalUnitDescendantAccounts = exports.getOrganizationalUnitChildAccountsOutput = exports.getOrganizationalUnitChildAccounts = exports.getOrganizationalUnitOutput = exports.getOrganizationalUnit = exports.getOrganizationOutput = exports.getOrganization = exports.getEntityPathOutput = exports.getEntityPath = exports.getDelegatedServicesOutput = exports.getDelegatedServices = exports.getDelegatedAdministratorsOutput = exports.getDelegatedAdministrators = exports.getAccountOutput = exports.getAccount = exports.DelegatedAdministrator = exports.AwsServiceAccess = exports.Account = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.AwsServiceAccess = null;
utilities.lazyLoad(exports, ["AwsServiceAccess"], () => require("./awsServiceAccess"));
exports.DelegatedAdministrator = null;
utilities.lazyLoad(exports, ["DelegatedAdministrator"], () => require("./delegatedAdministrator"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getDelegatedAdministrators = null;
exports.getDelegatedAdministratorsOutput = null;
utilities.lazyLoad(exports, ["getDelegatedAdministrators", "getDelegatedAdministratorsOutput"], () => require("./getDelegatedAdministrators"));
exports.getDelegatedServices = null;
exports.getDelegatedServicesOutput = null;
utilities.lazyLoad(exports, ["getDelegatedServices", "getDelegatedServicesOutput"], () => require("./getDelegatedServices"));
exports.getEntityPath = null;
exports.getEntityPathOutput = null;
utilities.lazyLoad(exports, ["getEntityPath", "getEntityPathOutput"], () => require("./getEntityPath"));
exports.getOrganization = null;
exports.getOrganizationOutput = null;
utilities.lazyLoad(exports, ["getOrganization", "getOrganizationOutput"], () => require("./getOrganization"));
exports.getOrganizationalUnit = null;
exports.getOrganizationalUnitOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnit", "getOrganizationalUnitOutput"], () => require("./getOrganizationalUnit"));
exports.getOrganizationalUnitChildAccounts = null;
exports.getOrganizationalUnitChildAccountsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnitChildAccounts", "getOrganizationalUnitChildAccountsOutput"], () => require("./getOrganizationalUnitChildAccounts"));
exports.getOrganizationalUnitDescendantAccounts = null;
exports.getOrganizationalUnitDescendantAccountsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnitDescendantAccounts", "getOrganizationalUnitDescendantAccountsOutput"], () => require("./getOrganizationalUnitDescendantAccounts"));
exports.getOrganizationalUnitDescendantOrganizationalUnits = null;
exports.getOrganizationalUnitDescendantOrganizationalUnitsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnitDescendantOrganizationalUnits", "getOrganizationalUnitDescendantOrganizationalUnitsOutput"], () => require("./getOrganizationalUnitDescendantOrganizationalUnits"));
exports.getOrganizationalUnits = null;
exports.getOrganizationalUnitsOutput = null;
utilities.lazyLoad(exports, ["getOrganizationalUnits", "getOrganizationalUnitsOutput"], () => require("./getOrganizationalUnits"));
exports.getPolicies = null;
exports.getPoliciesOutput = null;
utilities.lazyLoad(exports, ["getPolicies", "getPoliciesOutput"], () => require("./getPolicies"));
exports.getPoliciesForTarget = null;
exports.getPoliciesForTargetOutput = null;
utilities.lazyLoad(exports, ["getPoliciesForTarget", "getPoliciesForTargetOutput"], () => require("./getPoliciesForTarget"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getResourceTags = null;
exports.getResourceTagsOutput = null;
utilities.lazyLoad(exports, ["getResourceTags", "getResourceTagsOutput"], () => require("./getResourceTags"));
exports.Organization = null;
utilities.lazyLoad(exports, ["Organization"], () => require("./organization"));
exports.OrganizationalUnit = null;
utilities.lazyLoad(exports, ["OrganizationalUnit"], () => require("./organizationalUnit"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.PolicyAttachment = null;
utilities.lazyLoad(exports, ["PolicyAttachment"], () => require("./policyAttachment"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.Tag = null;
utilities.lazyLoad(exports, ["Tag"], () => require("./tag"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:organizations/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "aws:organizations/awsServiceAccess:AwsServiceAccess":
                return new exports.AwsServiceAccess(name, undefined, { urn });
            case "aws:organizations/delegatedAdministrator:DelegatedAdministrator":
                return new exports.DelegatedAdministrator(name, undefined, { urn });
            case "aws:organizations/organization:Organization":
                return new exports.Organization(name, undefined, { urn });
            case "aws:organizations/organizationalUnit:OrganizationalUnit":
                return new exports.OrganizationalUnit(name, undefined, { urn });
            case "aws:organizations/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "aws:organizations/policyAttachment:PolicyAttachment":
                return new exports.PolicyAttachment(name, undefined, { urn });
            case "aws:organizations/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:organizations/tag:Tag":
                return new exports.Tag(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "organizations/account", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/awsServiceAccess", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/delegatedAdministrator", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/organization", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/organizationalUnit", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/policy", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/policyAttachment", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "organizations/tag", _module);
