// @pulumi/azure-native/cloudhealth/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalDefinition = exports.Relationship = exports.HealthModel = exports.getSignalDefinitionOutput = exports.getSignalDefinition = exports.getRelationshipOutput = exports.getRelationship = exports.getHealthModelOutput = exports.getHealthModel = exports.getEntitySignalHistoryOutput = exports.getEntitySignalHistory = exports.getEntityHistoryOutput = exports.getEntityHistory = exports.getEntityOutput = exports.getEntity = exports.getDiscoveryRuleOutput = exports.getDiscoveryRule = exports.getAuthenticationSettingOutput = exports.getAuthenticationSetting = exports.Entity = exports.DiscoveryRule = exports.AuthenticationSetting = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AuthenticationSetting = null;
utilities.lazyLoad(exports, ["AuthenticationSetting"], () => require("./authenticationSetting"));
exports.DiscoveryRule = null;
utilities.lazyLoad(exports, ["DiscoveryRule"], () => require("./discoveryRule"));
exports.Entity = null;
utilities.lazyLoad(exports, ["Entity"], () => require("./entity"));
exports.getAuthenticationSetting = null;
exports.getAuthenticationSettingOutput = null;
utilities.lazyLoad(exports, ["getAuthenticationSetting", "getAuthenticationSettingOutput"], () => require("./getAuthenticationSetting"));
exports.getDiscoveryRule = null;
exports.getDiscoveryRuleOutput = null;
utilities.lazyLoad(exports, ["getDiscoveryRule", "getDiscoveryRuleOutput"], () => require("./getDiscoveryRule"));
exports.getEntity = null;
exports.getEntityOutput = null;
utilities.lazyLoad(exports, ["getEntity", "getEntityOutput"], () => require("./getEntity"));
exports.getEntityHistory = null;
exports.getEntityHistoryOutput = null;
utilities.lazyLoad(exports, ["getEntityHistory", "getEntityHistoryOutput"], () => require("./getEntityHistory"));
exports.getEntitySignalHistory = null;
exports.getEntitySignalHistoryOutput = null;
utilities.lazyLoad(exports, ["getEntitySignalHistory", "getEntitySignalHistoryOutput"], () => require("./getEntitySignalHistory"));
exports.getHealthModel = null;
exports.getHealthModelOutput = null;
utilities.lazyLoad(exports, ["getHealthModel", "getHealthModelOutput"], () => require("./getHealthModel"));
exports.getRelationship = null;
exports.getRelationshipOutput = null;
utilities.lazyLoad(exports, ["getRelationship", "getRelationshipOutput"], () => require("./getRelationship"));
exports.getSignalDefinition = null;
exports.getSignalDefinitionOutput = null;
utilities.lazyLoad(exports, ["getSignalDefinition", "getSignalDefinitionOutput"], () => require("./getSignalDefinition"));
exports.HealthModel = null;
utilities.lazyLoad(exports, ["HealthModel"], () => require("./healthModel"));
exports.Relationship = null;
utilities.lazyLoad(exports, ["Relationship"], () => require("./relationship"));
exports.SignalDefinition = null;
utilities.lazyLoad(exports, ["SignalDefinition"], () => require("./signalDefinition"));
// Export enums:
__exportStar(require("../types/enums/cloudhealth"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cloudhealth:AuthenticationSetting":
                return new exports.AuthenticationSetting(name, undefined, { urn });
            case "azure-native:cloudhealth:DiscoveryRule":
                return new exports.DiscoveryRule(name, undefined, { urn });
            case "azure-native:cloudhealth:Entity":
                return new exports.Entity(name, undefined, { urn });
            case "azure-native:cloudhealth:HealthModel":
                return new exports.HealthModel(name, undefined, { urn });
            case "azure-native:cloudhealth:Relationship":
                return new exports.Relationship(name, undefined, { urn });
            case "azure-native:cloudhealth:SignalDefinition":
                return new exports.SignalDefinition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cloudhealth", _module);
