// @pulumi/azure-native/databasewatcher/alertRuleResource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertRuleResource = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Concrete proxy resource types can be created by aliasing this type using a specific property type.
 *
 * Uses Azure REST API version 2024-10-01-preview. In version 2.x of the Azure Native provider, it used API version 2024-07-19-preview.
 *
 * Other available API versions: 2024-07-19-preview, 2025-01-02. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native databasewatcher [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AlertRuleResource extends pulumi.CustomResource {
    /**
     * Get an existing AlertRuleResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AlertRuleResource(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:databasewatcher:AlertRuleResource';
    /**
     * Returns true if the given object is an instance of AlertRuleResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AlertRuleResource.__pulumiType;
    }
    /**
     * Create a AlertRuleResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.alertRuleResourceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'alertRuleResourceId'");
            }
            if (args?.alertRuleTemplateId === undefined && !opts.urn) {
                throw new Error("Missing required property 'alertRuleTemplateId'");
            }
            if (args?.alertRuleTemplateVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'alertRuleTemplateVersion'");
            }
            if (args?.createdWithProperties === undefined && !opts.urn) {
                throw new Error("Missing required property 'createdWithProperties'");
            }
            if (args?.creationTime === undefined && !opts.urn) {
                throw new Error("Missing required property 'creationTime'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.watcherName === undefined && !opts.urn) {
                throw new Error("Missing required property 'watcherName'");
            }
            resourceInputs["alertRuleResourceId"] = args?.alertRuleResourceId;
            resourceInputs["alertRuleResourceName"] = args?.alertRuleResourceName;
            resourceInputs["alertRuleTemplateId"] = args?.alertRuleTemplateId;
            resourceInputs["alertRuleTemplateVersion"] = args?.alertRuleTemplateVersion;
            resourceInputs["createdWithProperties"] = args?.createdWithProperties;
            resourceInputs["creationTime"] = args?.creationTime;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["watcherName"] = args?.watcherName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["alertRuleResourceId"] = undefined /*out*/;
            resourceInputs["alertRuleTemplateId"] = undefined /*out*/;
            resourceInputs["alertRuleTemplateVersion"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdWithProperties"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databasewatcher/v20240719preview:AlertRuleResource" }, { type: "azure-native:databasewatcher/v20241001preview:AlertRuleResource" }, { type: "azure-native:databasewatcher/v20250102:AlertRuleResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AlertRuleResource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AlertRuleResource = AlertRuleResource;
