// @pulumi/aws/pinpoint/gcmChannel.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GcmChannel = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a Pinpoint GCM Channel resource.
 *
 * > **Note:** Credentials (Service Account JSON and API Key) will be stored in the raw state as plain-text.
 * ## Import
 *
 * Using `pulumi import`, import Pinpoint GCM Channel using the `application-id`. For example:
 *
 * ```sh
 * $ pulumi import aws:pinpoint/gcmChannel:GcmChannel gcm application-id
 * ```
 */
class GcmChannel extends pulumi.CustomResource {
    /**
     * Get an existing GcmChannel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GcmChannel(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:pinpoint/gcmChannel:GcmChannel';
    /**
     * Returns true if the given object is an instance of GcmChannel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GcmChannel.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["apiKey"] = state?.apiKey;
            resourceInputs["applicationId"] = state?.applicationId;
            resourceInputs["defaultAuthenticationMethod"] = state?.defaultAuthenticationMethod;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["region"] = state?.region;
            resourceInputs["serviceJson"] = state?.serviceJson;
        }
        else {
            const args = argsOrState;
            if (args?.applicationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'applicationId'");
            }
            resourceInputs["apiKey"] = args?.apiKey ? pulumi.secret(args.apiKey) : undefined;
            resourceInputs["applicationId"] = args?.applicationId;
            resourceInputs["defaultAuthenticationMethod"] = args?.defaultAuthenticationMethod;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["region"] = args?.region;
            resourceInputs["serviceJson"] = args?.serviceJson ? pulumi.secret(args.serviceJson) : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["apiKey", "serviceJson"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(GcmChannel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GcmChannel = GcmChannel;
