// @pulumi/gcp/kms/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationKajPolicyConfig = exports.KeyRingImportJob = exports.KeyRingIAMPolicy = exports.KeyRingIAMMember = exports.KeyRingIAMBinding = exports.KeyRing = exports.KeyHandle = exports.getKeyRingsOutput = exports.getKeyRings = exports.getKeyRingIamPolicyOutput = exports.getKeyRingIamPolicy = exports.getKeyHandlesOutput = exports.getKeyHandles = exports.getKeyHandleOutput = exports.getKeyHandle = exports.getKMSSecretCiphertextOutput = exports.getKMSSecretCiphertext = exports.getKMSSecretAsymmetricOutput = exports.getKMSSecretAsymmetric = exports.getKMSSecretOutput = exports.getKMSSecret = exports.getKMSKeyRingOutput = exports.getKMSKeyRing = exports.getKMSCryptoKeyVersionOutput = exports.getKMSCryptoKeyVersion = exports.getKMSCryptoKeyOutput = exports.getKMSCryptoKey = exports.getEkmConnectionIamPolicyOutput = exports.getEkmConnectionIamPolicy = exports.getCryptoKeysOutput = exports.getCryptoKeys = exports.getCryptoKeyVersionsOutput = exports.getCryptoKeyVersions = exports.getCryptoKeyLatestVersionOutput = exports.getCryptoKeyLatestVersion = exports.getCryptoKeyIamPolicyOutput = exports.getCryptoKeyIamPolicy = exports.getAutokeyConfigOutput = exports.getAutokeyConfig = exports.FolderKajPolicyConfig = exports.EkmConnectionIamPolicy = exports.EkmConnectionIamMember = exports.EkmConnectionIamBinding = exports.EkmConnection = exports.CryptoKeyVersion = exports.CryptoKeyIAMPolicy = exports.CryptoKeyIAMMember = exports.CryptoKeyIAMBinding = exports.CryptoKey = exports.AutokeyConfig = void 0;
exports.SecretCiphertext = exports.ProjectKajPolicyConfig = exports.ProjectAutokeyConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AutokeyConfig = null;
utilities.lazyLoad(exports, ["AutokeyConfig"], () => require("./autokeyConfig"));
exports.CryptoKey = null;
utilities.lazyLoad(exports, ["CryptoKey"], () => require("./cryptoKey"));
exports.CryptoKeyIAMBinding = null;
utilities.lazyLoad(exports, ["CryptoKeyIAMBinding"], () => require("./cryptoKeyIAMBinding"));
exports.CryptoKeyIAMMember = null;
utilities.lazyLoad(exports, ["CryptoKeyIAMMember"], () => require("./cryptoKeyIAMMember"));
exports.CryptoKeyIAMPolicy = null;
utilities.lazyLoad(exports, ["CryptoKeyIAMPolicy"], () => require("./cryptoKeyIAMPolicy"));
exports.CryptoKeyVersion = null;
utilities.lazyLoad(exports, ["CryptoKeyVersion"], () => require("./cryptoKeyVersion"));
exports.EkmConnection = null;
utilities.lazyLoad(exports, ["EkmConnection"], () => require("./ekmConnection"));
exports.EkmConnectionIamBinding = null;
utilities.lazyLoad(exports, ["EkmConnectionIamBinding"], () => require("./ekmConnectionIamBinding"));
exports.EkmConnectionIamMember = null;
utilities.lazyLoad(exports, ["EkmConnectionIamMember"], () => require("./ekmConnectionIamMember"));
exports.EkmConnectionIamPolicy = null;
utilities.lazyLoad(exports, ["EkmConnectionIamPolicy"], () => require("./ekmConnectionIamPolicy"));
exports.FolderKajPolicyConfig = null;
utilities.lazyLoad(exports, ["FolderKajPolicyConfig"], () => require("./folderKajPolicyConfig"));
exports.getAutokeyConfig = null;
exports.getAutokeyConfigOutput = null;
utilities.lazyLoad(exports, ["getAutokeyConfig", "getAutokeyConfigOutput"], () => require("./getAutokeyConfig"));
exports.getCryptoKeyIamPolicy = null;
exports.getCryptoKeyIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getCryptoKeyIamPolicy", "getCryptoKeyIamPolicyOutput"], () => require("./getCryptoKeyIamPolicy"));
exports.getCryptoKeyLatestVersion = null;
exports.getCryptoKeyLatestVersionOutput = null;
utilities.lazyLoad(exports, ["getCryptoKeyLatestVersion", "getCryptoKeyLatestVersionOutput"], () => require("./getCryptoKeyLatestVersion"));
exports.getCryptoKeyVersions = null;
exports.getCryptoKeyVersionsOutput = null;
utilities.lazyLoad(exports, ["getCryptoKeyVersions", "getCryptoKeyVersionsOutput"], () => require("./getCryptoKeyVersions"));
exports.getCryptoKeys = null;
exports.getCryptoKeysOutput = null;
utilities.lazyLoad(exports, ["getCryptoKeys", "getCryptoKeysOutput"], () => require("./getCryptoKeys"));
exports.getEkmConnectionIamPolicy = null;
exports.getEkmConnectionIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getEkmConnectionIamPolicy", "getEkmConnectionIamPolicyOutput"], () => require("./getEkmConnectionIamPolicy"));
exports.getKMSCryptoKey = null;
exports.getKMSCryptoKeyOutput = null;
utilities.lazyLoad(exports, ["getKMSCryptoKey", "getKMSCryptoKeyOutput"], () => require("./getKMSCryptoKey"));
exports.getKMSCryptoKeyVersion = null;
exports.getKMSCryptoKeyVersionOutput = null;
utilities.lazyLoad(exports, ["getKMSCryptoKeyVersion", "getKMSCryptoKeyVersionOutput"], () => require("./getKMSCryptoKeyVersion"));
exports.getKMSKeyRing = null;
exports.getKMSKeyRingOutput = null;
utilities.lazyLoad(exports, ["getKMSKeyRing", "getKMSKeyRingOutput"], () => require("./getKMSKeyRing"));
exports.getKMSSecret = null;
exports.getKMSSecretOutput = null;
utilities.lazyLoad(exports, ["getKMSSecret", "getKMSSecretOutput"], () => require("./getKMSSecret"));
exports.getKMSSecretAsymmetric = null;
exports.getKMSSecretAsymmetricOutput = null;
utilities.lazyLoad(exports, ["getKMSSecretAsymmetric", "getKMSSecretAsymmetricOutput"], () => require("./getKMSSecretAsymmetric"));
exports.getKMSSecretCiphertext = null;
exports.getKMSSecretCiphertextOutput = null;
utilities.lazyLoad(exports, ["getKMSSecretCiphertext", "getKMSSecretCiphertextOutput"], () => require("./getKMSSecretCiphertext"));
exports.getKeyHandle = null;
exports.getKeyHandleOutput = null;
utilities.lazyLoad(exports, ["getKeyHandle", "getKeyHandleOutput"], () => require("./getKeyHandle"));
exports.getKeyHandles = null;
exports.getKeyHandlesOutput = null;
utilities.lazyLoad(exports, ["getKeyHandles", "getKeyHandlesOutput"], () => require("./getKeyHandles"));
exports.getKeyRingIamPolicy = null;
exports.getKeyRingIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getKeyRingIamPolicy", "getKeyRingIamPolicyOutput"], () => require("./getKeyRingIamPolicy"));
exports.getKeyRings = null;
exports.getKeyRingsOutput = null;
utilities.lazyLoad(exports, ["getKeyRings", "getKeyRingsOutput"], () => require("./getKeyRings"));
exports.KeyHandle = null;
utilities.lazyLoad(exports, ["KeyHandle"], () => require("./keyHandle"));
exports.KeyRing = null;
utilities.lazyLoad(exports, ["KeyRing"], () => require("./keyRing"));
exports.KeyRingIAMBinding = null;
utilities.lazyLoad(exports, ["KeyRingIAMBinding"], () => require("./keyRingIAMBinding"));
exports.KeyRingIAMMember = null;
utilities.lazyLoad(exports, ["KeyRingIAMMember"], () => require("./keyRingIAMMember"));
exports.KeyRingIAMPolicy = null;
utilities.lazyLoad(exports, ["KeyRingIAMPolicy"], () => require("./keyRingIAMPolicy"));
exports.KeyRingImportJob = null;
utilities.lazyLoad(exports, ["KeyRingImportJob"], () => require("./keyRingImportJob"));
exports.OrganizationKajPolicyConfig = null;
utilities.lazyLoad(exports, ["OrganizationKajPolicyConfig"], () => require("./organizationKajPolicyConfig"));
exports.ProjectAutokeyConfig = null;
utilities.lazyLoad(exports, ["ProjectAutokeyConfig"], () => require("./projectAutokeyConfig"));
exports.ProjectKajPolicyConfig = null;
utilities.lazyLoad(exports, ["ProjectKajPolicyConfig"], () => require("./projectKajPolicyConfig"));
exports.SecretCiphertext = null;
utilities.lazyLoad(exports, ["SecretCiphertext"], () => require("./secretCiphertext"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:kms/autokeyConfig:AutokeyConfig":
                return new exports.AutokeyConfig(name, undefined, { urn });
            case "gcp:kms/cryptoKey:CryptoKey":
                return new exports.CryptoKey(name, undefined, { urn });
            case "gcp:kms/cryptoKeyIAMBinding:CryptoKeyIAMBinding":
                return new exports.CryptoKeyIAMBinding(name, undefined, { urn });
            case "gcp:kms/cryptoKeyIAMMember:CryptoKeyIAMMember":
                return new exports.CryptoKeyIAMMember(name, undefined, { urn });
            case "gcp:kms/cryptoKeyIAMPolicy:CryptoKeyIAMPolicy":
                return new exports.CryptoKeyIAMPolicy(name, undefined, { urn });
            case "gcp:kms/cryptoKeyVersion:CryptoKeyVersion":
                return new exports.CryptoKeyVersion(name, undefined, { urn });
            case "gcp:kms/ekmConnection:EkmConnection":
                return new exports.EkmConnection(name, undefined, { urn });
            case "gcp:kms/ekmConnectionIamBinding:EkmConnectionIamBinding":
                return new exports.EkmConnectionIamBinding(name, undefined, { urn });
            case "gcp:kms/ekmConnectionIamMember:EkmConnectionIamMember":
                return new exports.EkmConnectionIamMember(name, undefined, { urn });
            case "gcp:kms/ekmConnectionIamPolicy:EkmConnectionIamPolicy":
                return new exports.EkmConnectionIamPolicy(name, undefined, { urn });
            case "gcp:kms/folderKajPolicyConfig:FolderKajPolicyConfig":
                return new exports.FolderKajPolicyConfig(name, undefined, { urn });
            case "gcp:kms/keyHandle:KeyHandle":
                return new exports.KeyHandle(name, undefined, { urn });
            case "gcp:kms/keyRing:KeyRing":
                return new exports.KeyRing(name, undefined, { urn });
            case "gcp:kms/keyRingIAMBinding:KeyRingIAMBinding":
                return new exports.KeyRingIAMBinding(name, undefined, { urn });
            case "gcp:kms/keyRingIAMMember:KeyRingIAMMember":
                return new exports.KeyRingIAMMember(name, undefined, { urn });
            case "gcp:kms/keyRingIAMPolicy:KeyRingIAMPolicy":
                return new exports.KeyRingIAMPolicy(name, undefined, { urn });
            case "gcp:kms/keyRingImportJob:KeyRingImportJob":
                return new exports.KeyRingImportJob(name, undefined, { urn });
            case "gcp:kms/organizationKajPolicyConfig:OrganizationKajPolicyConfig":
                return new exports.OrganizationKajPolicyConfig(name, undefined, { urn });
            case "gcp:kms/projectAutokeyConfig:ProjectAutokeyConfig":
                return new exports.ProjectAutokeyConfig(name, undefined, { urn });
            case "gcp:kms/projectKajPolicyConfig:ProjectKajPolicyConfig":
                return new exports.ProjectKajPolicyConfig(name, undefined, { urn });
            case "gcp:kms/secretCiphertext:SecretCiphertext":
                return new exports.SecretCiphertext(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "kms/autokeyConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/cryptoKey", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/cryptoKeyIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/cryptoKeyIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/cryptoKeyIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/cryptoKeyVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/ekmConnection", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/ekmConnectionIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/ekmConnectionIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/ekmConnectionIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/folderKajPolicyConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyHandle", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyRing", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyRingIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyRingIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyRingIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/keyRingImportJob", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/organizationKajPolicyConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/projectAutokeyConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/projectKajPolicyConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "kms/secretCiphertext", _module);
