// @pulumi/gcp/dns/responsePolicyRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsePolicyRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A Response Policy Rule is a selector that applies its behavior to queries that match the selector.
 * Selectors are DNS names, which may be wildcards or exact matches.
 * Each DNS query subject to a Response Policy matches at most one ResponsePolicyRule,
 * as identified by the dnsName field with the longest matching suffix.
 *
 * ## Example Usage
 *
 * ### Dns Response Policy Rule Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const network_1 = new gcp.compute.Network("network-1", {
 *     name: "network-1",
 *     autoCreateSubnetworks: false,
 * });
 * const network_2 = new gcp.compute.Network("network-2", {
 *     name: "network-2",
 *     autoCreateSubnetworks: false,
 * });
 * const response_policy = new gcp.dns.ResponsePolicy("response-policy", {
 *     responsePolicyName: "example-response-policy",
 *     networks: [
 *         {
 *             networkUrl: network_1.id,
 *         },
 *         {
 *             networkUrl: network_2.id,
 *         },
 *     ],
 * });
 * const example_response_policy_rule = new gcp.dns.ResponsePolicyRule("example-response-policy-rule", {
 *     responsePolicy: response_policy.responsePolicyName,
 *     ruleName: "example-rule",
 *     dnsName: "dns.example.com.",
 *     localData: {
 *         localDatas: [{
 *             name: "dns.example.com.",
 *             type: "A",
 *             ttl: 300,
 *             rrdatas: ["192.0.2.91"],
 *         }],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ResponsePolicyRule can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/responsePolicies/{{response_policy}}/rules/{{rule_name}}`
 * * `{{project}}/{{response_policy}}/{{rule_name}}`
 * * `{{response_policy}}/{{rule_name}}`
 *
 * When using the `pulumi import` command, ResponsePolicyRule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dns/responsePolicyRule:ResponsePolicyRule default projects/{{project}}/responsePolicies/{{response_policy}}/rules/{{rule_name}}
 * $ pulumi import gcp:dns/responsePolicyRule:ResponsePolicyRule default {{project}}/{{response_policy}}/{{rule_name}}
 * $ pulumi import gcp:dns/responsePolicyRule:ResponsePolicyRule default {{response_policy}}/{{rule_name}}
 * ```
 */
class ResponsePolicyRule extends pulumi.CustomResource {
    /**
     * Get an existing ResponsePolicyRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResponsePolicyRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:dns/responsePolicyRule:ResponsePolicyRule';
    /**
     * Returns true if the given object is an instance of ResponsePolicyRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResponsePolicyRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["behavior"] = state?.behavior;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["dnsName"] = state?.dnsName;
            resourceInputs["localData"] = state?.localData;
            resourceInputs["project"] = state?.project;
            resourceInputs["responsePolicy"] = state?.responsePolicy;
            resourceInputs["ruleName"] = state?.ruleName;
        }
        else {
            const args = argsOrState;
            if (args?.dnsName === undefined && !opts.urn) {
                throw new Error("Missing required property 'dnsName'");
            }
            if (args?.responsePolicy === undefined && !opts.urn) {
                throw new Error("Missing required property 'responsePolicy'");
            }
            if (args?.ruleName === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleName'");
            }
            resourceInputs["behavior"] = args?.behavior;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["dnsName"] = args?.dnsName;
            resourceInputs["localData"] = args?.localData;
            resourceInputs["project"] = args?.project;
            resourceInputs["responsePolicy"] = args?.responsePolicy;
            resourceInputs["ruleName"] = args?.ruleName;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResponsePolicyRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResponsePolicyRule = ResponsePolicyRule;
