// @pulumi/gcp/diagflow/generator.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Generator = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * LLM generator.
 *
 * To get more information about Generator, see:
 *
 * * [API documentation](https://cloud.google.com/dialogflow/es/docs/reference/rest/v2beta1/projects.locations.generators)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/agent-assist/docs)
 *
 * ## Example Usage
 *
 * ### Dialogflow Generator Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const summarizationBasicGenerator = new gcp.diagflow.Generator("summarization_basic_generator", {
 *     location: "global",
 *     description: "A v4.0 summarization generator.",
 *     inferenceParameter: {
 *         maxOutputTokens: 1024,
 *         temperature: 0,
 *         topK: 40,
 *         topP: 0.95,
 *     },
 *     summarizationContext: {
 *         version: "4.0",
 *         outputLanguageCode: "en",
 *     },
 *     triggerEvent: "MANUAL_CALL",
 * });
 * ```
 *
 * ## Import
 *
 * Generator can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/generators/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, Generator can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:diagflow/generator:Generator default projects/{{project}}/locations/{{location}}/generators/{{name}}
 * $ pulumi import gcp:diagflow/generator:Generator default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:diagflow/generator:Generator default {{location}}/{{name}}
 * ```
 */
class Generator extends pulumi.CustomResource {
    /**
     * Get an existing Generator resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Generator(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:diagflow/generator:Generator';
    /**
     * Returns true if the given object is an instance of Generator.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Generator.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["generatorId"] = state?.generatorId;
            resourceInputs["inferenceParameter"] = state?.inferenceParameter;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["publishedModel"] = state?.publishedModel;
            resourceInputs["summarizationContext"] = state?.summarizationContext;
            resourceInputs["triggerEvent"] = state?.triggerEvent;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.summarizationContext === undefined && !opts.urn) {
                throw new Error("Missing required property 'summarizationContext'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["generatorId"] = args?.generatorId;
            resourceInputs["inferenceParameter"] = args?.inferenceParameter;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["publishedModel"] = args?.publishedModel;
            resourceInputs["summarizationContext"] = args?.summarizationContext;
            resourceInputs["triggerEvent"] = args?.triggerEvent;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Generator.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Generator = Generator;
