// @pulumi/gcp/sql/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.SslCert = exports.SourceRepresentationInstance = exports.ProvisionScript = exports.getTiersOutput = exports.getTiers = exports.getDatabasesOutput = exports.getDatabases = exports.getDatabaseInstancesOutput = exports.getDatabaseInstances = exports.getDatabaseInstanceLatestRecoveryTimeOutput = exports.getDatabaseInstanceLatestRecoveryTime = exports.getDatabaseInstanceOutput = exports.getDatabaseInstance = exports.getDatabaseOutput = exports.getDatabase = exports.getCaCertsOutput = exports.getCaCerts = exports.getBackupRunOutput = exports.getBackupRun = exports.DatabaseInstance = exports.Database = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.DatabaseInstance = null;
utilities.lazyLoad(exports, ["DatabaseInstance"], () => require("./databaseInstance"));
exports.getBackupRun = null;
exports.getBackupRunOutput = null;
utilities.lazyLoad(exports, ["getBackupRun", "getBackupRunOutput"], () => require("./getBackupRun"));
exports.getCaCerts = null;
exports.getCaCertsOutput = null;
utilities.lazyLoad(exports, ["getCaCerts", "getCaCertsOutput"], () => require("./getCaCerts"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getDatabaseInstance = null;
exports.getDatabaseInstanceOutput = null;
utilities.lazyLoad(exports, ["getDatabaseInstance", "getDatabaseInstanceOutput"], () => require("./getDatabaseInstance"));
exports.getDatabaseInstanceLatestRecoveryTime = null;
exports.getDatabaseInstanceLatestRecoveryTimeOutput = null;
utilities.lazyLoad(exports, ["getDatabaseInstanceLatestRecoveryTime", "getDatabaseInstanceLatestRecoveryTimeOutput"], () => require("./getDatabaseInstanceLatestRecoveryTime"));
exports.getDatabaseInstances = null;
exports.getDatabaseInstancesOutput = null;
utilities.lazyLoad(exports, ["getDatabaseInstances", "getDatabaseInstancesOutput"], () => require("./getDatabaseInstances"));
exports.getDatabases = null;
exports.getDatabasesOutput = null;
utilities.lazyLoad(exports, ["getDatabases", "getDatabasesOutput"], () => require("./getDatabases"));
exports.getTiers = null;
exports.getTiersOutput = null;
utilities.lazyLoad(exports, ["getTiers", "getTiersOutput"], () => require("./getTiers"));
exports.ProvisionScript = null;
utilities.lazyLoad(exports, ["ProvisionScript"], () => require("./provisionScript"));
exports.SourceRepresentationInstance = null;
utilities.lazyLoad(exports, ["SourceRepresentationInstance"], () => require("./sourceRepresentationInstance"));
exports.SslCert = null;
utilities.lazyLoad(exports, ["SslCert"], () => require("./sslCert"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:sql/database:Database":
                return new exports.Database(name, undefined, { urn });
            case "gcp:sql/databaseInstance:DatabaseInstance":
                return new exports.DatabaseInstance(name, undefined, { urn });
            case "gcp:sql/provisionScript:ProvisionScript":
                return new exports.ProvisionScript(name, undefined, { urn });
            case "gcp:sql/sourceRepresentationInstance:SourceRepresentationInstance":
                return new exports.SourceRepresentationInstance(name, undefined, { urn });
            case "gcp:sql/sslCert:SslCert":
                return new exports.SslCert(name, undefined, { urn });
            case "gcp:sql/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "sql/database", _module);
pulumi.runtime.registerResourceModule("gcp", "sql/databaseInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "sql/provisionScript", _module);
pulumi.runtime.registerResourceModule("gcp", "sql/sourceRepresentationInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "sql/sslCert", _module);
pulumi.runtime.registerResourceModule("gcp", "sql/user", _module);
