// @pulumi/aws/cfg/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rule = exports.RetentionConfiguration = exports.RemediationConfiguration = exports.RecorderStatus = exports.Recorder = exports.OrganizationManagedRule = exports.OrganizationCustomRule = exports.OrganizationCustomPolicyRule = exports.OrganizationConformancePack = exports.DeliveryChannel = exports.ConformancePack = exports.ConfigurationAggregator = exports.AggregateAuthorization = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AggregateAuthorization = null;
utilities.lazyLoad(exports, ["AggregateAuthorization"], () => require("./aggregateAuthorization"));
exports.ConfigurationAggregator = null;
utilities.lazyLoad(exports, ["ConfigurationAggregator"], () => require("./configurationAggregator"));
exports.ConformancePack = null;
utilities.lazyLoad(exports, ["ConformancePack"], () => require("./conformancePack"));
exports.DeliveryChannel = null;
utilities.lazyLoad(exports, ["DeliveryChannel"], () => require("./deliveryChannel"));
exports.OrganizationConformancePack = null;
utilities.lazyLoad(exports, ["OrganizationConformancePack"], () => require("./organizationConformancePack"));
exports.OrganizationCustomPolicyRule = null;
utilities.lazyLoad(exports, ["OrganizationCustomPolicyRule"], () => require("./organizationCustomPolicyRule"));
exports.OrganizationCustomRule = null;
utilities.lazyLoad(exports, ["OrganizationCustomRule"], () => require("./organizationCustomRule"));
exports.OrganizationManagedRule = null;
utilities.lazyLoad(exports, ["OrganizationManagedRule"], () => require("./organizationManagedRule"));
exports.Recorder = null;
utilities.lazyLoad(exports, ["Recorder"], () => require("./recorder"));
exports.RecorderStatus = null;
utilities.lazyLoad(exports, ["RecorderStatus"], () => require("./recorderStatus"));
exports.RemediationConfiguration = null;
utilities.lazyLoad(exports, ["RemediationConfiguration"], () => require("./remediationConfiguration"));
exports.RetentionConfiguration = null;
utilities.lazyLoad(exports, ["RetentionConfiguration"], () => require("./retentionConfiguration"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:cfg/aggregateAuthorization:AggregateAuthorization":
                return new exports.AggregateAuthorization(name, undefined, { urn });
            case "aws:cfg/configurationAggregator:ConfigurationAggregator":
                return new exports.ConfigurationAggregator(name, undefined, { urn });
            case "aws:cfg/conformancePack:ConformancePack":
                return new exports.ConformancePack(name, undefined, { urn });
            case "aws:cfg/deliveryChannel:DeliveryChannel":
                return new exports.DeliveryChannel(name, undefined, { urn });
            case "aws:cfg/organizationConformancePack:OrganizationConformancePack":
                return new exports.OrganizationConformancePack(name, undefined, { urn });
            case "aws:cfg/organizationCustomPolicyRule:OrganizationCustomPolicyRule":
                return new exports.OrganizationCustomPolicyRule(name, undefined, { urn });
            case "aws:cfg/organizationCustomRule:OrganizationCustomRule":
                return new exports.OrganizationCustomRule(name, undefined, { urn });
            case "aws:cfg/organizationManagedRule:OrganizationManagedRule":
                return new exports.OrganizationManagedRule(name, undefined, { urn });
            case "aws:cfg/recorder:Recorder":
                return new exports.Recorder(name, undefined, { urn });
            case "aws:cfg/recorderStatus:RecorderStatus":
                return new exports.RecorderStatus(name, undefined, { urn });
            case "aws:cfg/remediationConfiguration:RemediationConfiguration":
                return new exports.RemediationConfiguration(name, undefined, { urn });
            case "aws:cfg/retentionConfiguration:RetentionConfiguration":
                return new exports.RetentionConfiguration(name, undefined, { urn });
            case "aws:cfg/rule:Rule":
                return new exports.Rule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "cfg/aggregateAuthorization", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/configurationAggregator", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/conformancePack", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/deliveryChannel", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationConformancePack", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationCustomPolicyRule", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationCustomRule", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/organizationManagedRule", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/recorder", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/recorderStatus", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/remediationConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/retentionConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "cfg/rule", _module);
