// @pulumi/gcp/essentialcontacts/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentAiWarehouseLocation = exports.DocumentAiWarehouseDocumentSchema = exports.DocumentAiProcessorDefaultVersion = exports.DocumentAiProcessor = exports.Contact = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Contact = null;
utilities.lazyLoad(exports, ["Contact"], () => require("./contact"));
exports.DocumentAiProcessor = null;
utilities.lazyLoad(exports, ["DocumentAiProcessor"], () => require("./documentAiProcessor"));
exports.DocumentAiProcessorDefaultVersion = null;
utilities.lazyLoad(exports, ["DocumentAiProcessorDefaultVersion"], () => require("./documentAiProcessorDefaultVersion"));
exports.DocumentAiWarehouseDocumentSchema = null;
utilities.lazyLoad(exports, ["DocumentAiWarehouseDocumentSchema"], () => require("./documentAiWarehouseDocumentSchema"));
exports.DocumentAiWarehouseLocation = null;
utilities.lazyLoad(exports, ["DocumentAiWarehouseLocation"], () => require("./documentAiWarehouseLocation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:essentialcontacts/contact:Contact":
                return new exports.Contact(name, undefined, { urn });
            case "gcp:essentialcontacts/documentAiProcessor:DocumentAiProcessor":
                return new exports.DocumentAiProcessor(name, undefined, { urn });
            case "gcp:essentialcontacts/documentAiProcessorDefaultVersion:DocumentAiProcessorDefaultVersion":
                return new exports.DocumentAiProcessorDefaultVersion(name, undefined, { urn });
            case "gcp:essentialcontacts/documentAiWarehouseDocumentSchema:DocumentAiWarehouseDocumentSchema":
                return new exports.DocumentAiWarehouseDocumentSchema(name, undefined, { urn });
            case "gcp:essentialcontacts/documentAiWarehouseLocation:DocumentAiWarehouseLocation":
                return new exports.DocumentAiWarehouseLocation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "essentialcontacts/contact", _module);
pulumi.runtime.registerResourceModule("gcp", "essentialcontacts/documentAiProcessor", _module);
pulumi.runtime.registerResourceModule("gcp", "essentialcontacts/documentAiProcessorDefaultVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "essentialcontacts/documentAiWarehouseDocumentSchema", _module);
pulumi.runtime.registerResourceModule("gcp", "essentialcontacts/documentAiWarehouseLocation", _module);
