// @pulumi/gcp/billing/accountIamBinding.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountIamBinding = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Three different resources help you manage IAM policies on billing accounts. Each of these resources serves a different use case:
 *
 * * `gcp.billing.AccountIamPolicy`: Authoritative. Sets the IAM policy for the billing accounts and replaces any existing policy already attached.
 * * `gcp.billing.AccountIamBinding`: Authoritative for a given role. Updates the IAM policy to grant a role to a list of members. Other roles within the IAM policy for the table are preserved.
 * * `gcp.billing.AccountIamMember`: Non-authoritative. Updates the IAM policy to grant a role to a new member. Other members for the role of the billing accounts are preserved.
 *
 * > **Note:** `gcp.billing.AccountIamPolicy` **cannot** be used in conjunction with `gcp.billing.AccountIamBinding` and `gcp.billing.AccountIamMember` or they will fight over what your policy should be. In addition, be careful not to accidentally unset ownership of the billing account as `gcp.billing.AccountIamPolicy` replaces the entire policy.
 *
 * > **Note:** `gcp.billing.AccountIamBinding` resources **can be** used in conjunction with `gcp.billing.AccountIamMember` resources **only if** they do not grant privilege to the same role.
 *
 * ## gcp.billing.AccountIamPolicy
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const admin = gcp.organizations.getIAMPolicy({
 *     bindings: [{
 *         role: "roles/billing.viewer",
 *         members: ["user:jane@example.com"],
 *     }],
 * });
 * const editor = new gcp.billing.AccountIamPolicy("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     policyData: admin.then(admin => admin.policyData),
 * });
 * ```
 *
 * ## gcp.billing.AccountIamBinding
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const editor = new gcp.billing.AccountIamBinding("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     role: "roles/billing.viewer",
 *     members: ["user:jane@example.com"],
 * });
 * ```
 *
 * ## gcp.billing.AccountIamMember
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const editor = new gcp.billing.AccountIamMember("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     role: "roles/billing.viewer",
 *     member: "user:jane@example.com",
 * });
 * ```
 *
 * ## gcp.billing.AccountIamPolicy
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const admin = gcp.organizations.getIAMPolicy({
 *     bindings: [{
 *         role: "roles/billing.viewer",
 *         members: ["user:jane@example.com"],
 *     }],
 * });
 * const editor = new gcp.billing.AccountIamPolicy("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     policyData: admin.then(admin => admin.policyData),
 * });
 * ```
 *
 * ## gcp.billing.AccountIamBinding
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const editor = new gcp.billing.AccountIamBinding("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     role: "roles/billing.viewer",
 *     members: ["user:jane@example.com"],
 * });
 * ```
 *
 * ## gcp.billing.AccountIamMember
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const editor = new gcp.billing.AccountIamMember("editor", {
 *     billingAccountId: "00AA00-000AAA-00AA0A",
 *     role: "roles/billing.viewer",
 *     member: "user:jane@example.com",
 * });
 * ```
 *
 * ## Import
 *
 * > **Custom Roles** If you're importing a IAM resource with a custom role, make sure to use the
 *  full name of the custom role, e.g. `organizations/my-org-id/roles/my-custom-role`.
 */
class AccountIamBinding extends pulumi.CustomResource {
    /**
     * Get an existing AccountIamBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccountIamBinding(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:billing/accountIamBinding:AccountIamBinding';
    /**
     * Returns true if the given object is an instance of AccountIamBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountIamBinding.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["billingAccountId"] = state?.billingAccountId;
            resourceInputs["condition"] = state?.condition;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["members"] = state?.members;
            resourceInputs["role"] = state?.role;
        }
        else {
            const args = argsOrState;
            if (args?.billingAccountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'billingAccountId'");
            }
            if (args?.members === undefined && !opts.urn) {
                throw new Error("Missing required property 'members'");
            }
            if (args?.role === undefined && !opts.urn) {
                throw new Error("Missing required property 'role'");
            }
            resourceInputs["billingAccountId"] = args?.billingAccountId;
            resourceInputs["condition"] = args?.condition;
            resourceInputs["members"] = args?.members;
            resourceInputs["role"] = args?.role;
            resourceInputs["etag"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccountIamBinding.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccountIamBinding = AccountIamBinding;
