// @pulumi/gcp/billing/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubAccount = exports.ProjectInfo = exports.getAccountIamPolicyOutput = exports.getAccountIamPolicy = exports.Budget = exports.AccountIamPolicy = exports.AccountIamMember = exports.AccountIamBinding = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccountIamBinding = null;
utilities.lazyLoad(exports, ["AccountIamBinding"], () => require("./accountIamBinding"));
exports.AccountIamMember = null;
utilities.lazyLoad(exports, ["AccountIamMember"], () => require("./accountIamMember"));
exports.AccountIamPolicy = null;
utilities.lazyLoad(exports, ["AccountIamPolicy"], () => require("./accountIamPolicy"));
exports.Budget = null;
utilities.lazyLoad(exports, ["Budget"], () => require("./budget"));
exports.getAccountIamPolicy = null;
exports.getAccountIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAccountIamPolicy", "getAccountIamPolicyOutput"], () => require("./getAccountIamPolicy"));
exports.ProjectInfo = null;
utilities.lazyLoad(exports, ["ProjectInfo"], () => require("./projectInfo"));
exports.SubAccount = null;
utilities.lazyLoad(exports, ["SubAccount"], () => require("./subAccount"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:billing/accountIamBinding:AccountIamBinding":
                return new exports.AccountIamBinding(name, undefined, { urn });
            case "gcp:billing/accountIamMember:AccountIamMember":
                return new exports.AccountIamMember(name, undefined, { urn });
            case "gcp:billing/accountIamPolicy:AccountIamPolicy":
                return new exports.AccountIamPolicy(name, undefined, { urn });
            case "gcp:billing/budget:Budget":
                return new exports.Budget(name, undefined, { urn });
            case "gcp:billing/projectInfo:ProjectInfo":
                return new exports.ProjectInfo(name, undefined, { urn });
            case "gcp:billing/subAccount:SubAccount":
                return new exports.SubAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/budget", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/projectInfo", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/subAccount", _module);
