// @pulumi/gcp/billing/subAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubAccount = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Allows creation and management of a Google Cloud Billing Subaccount.
 *
 * !> **WARNING:** Deleting this resource will not delete or close the billing subaccount.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const subaccount = new gcp.billing.SubAccount("subaccount", {
 *     displayName: "My Billing Account",
 *     masterBillingAccount: "012345-567890-ABCDEF",
 * });
 * ```
 *
 * ## Import
 *
 * Billing Subaccounts can be imported using any of these accepted formats:
 *
 * * `billingAccounts/{billing_account_id}`
 *
 * When using the `pulumi import` command, Billing Subaccounts can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:billing/subAccount:SubAccount default billingAccounts/{billing_account_id}
 * ```
 */
class SubAccount extends pulumi.CustomResource {
    /**
     * Get an existing SubAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SubAccount(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:billing/subAccount:SubAccount';
    /**
     * Returns true if the given object is an instance of SubAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SubAccount.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["billingAccountId"] = state?.billingAccountId;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["masterBillingAccount"] = state?.masterBillingAccount;
            resourceInputs["name"] = state?.name;
            resourceInputs["open"] = state?.open;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.masterBillingAccount === undefined && !opts.urn) {
                throw new Error("Missing required property 'masterBillingAccount'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["masterBillingAccount"] = args?.masterBillingAccount;
            resourceInputs["billingAccountId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["open"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SubAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SubAccount = SubAccount;
