// @pulumi/gcp/backupdisasterrecovery/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceConfig = exports.RestoreWorkload = exports.ManagementServer = exports.getManagementServerOutput = exports.getManagementServer = exports.getDataSourcesOutput = exports.getDataSources = exports.getDataSourceReferencesOutput = exports.getDataSourceReferences = exports.getDataSourceReferenceOutput = exports.getDataSourceReference = exports.getDataSourceOutput = exports.getDataSource = exports.getBackupVaultOutput = exports.getBackupVault = exports.getBackupPlanAssociationsOutput = exports.getBackupPlanAssociations = exports.getBackupPlanAssociationOutput = exports.getBackupPlanAssociation = exports.getBackupPlanOutput = exports.getBackupPlan = exports.getBackupOutput = exports.getBackup = exports.BackupVault = exports.BackupPlanAssociation = exports.BackupPlan = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.BackupPlan = null;
utilities.lazyLoad(exports, ["BackupPlan"], () => require("./backupPlan"));
exports.BackupPlanAssociation = null;
utilities.lazyLoad(exports, ["BackupPlanAssociation"], () => require("./backupPlanAssociation"));
exports.BackupVault = null;
utilities.lazyLoad(exports, ["BackupVault"], () => require("./backupVault"));
exports.getBackup = null;
exports.getBackupOutput = null;
utilities.lazyLoad(exports, ["getBackup", "getBackupOutput"], () => require("./getBackup"));
exports.getBackupPlan = null;
exports.getBackupPlanOutput = null;
utilities.lazyLoad(exports, ["getBackupPlan", "getBackupPlanOutput"], () => require("./getBackupPlan"));
exports.getBackupPlanAssociation = null;
exports.getBackupPlanAssociationOutput = null;
utilities.lazyLoad(exports, ["getBackupPlanAssociation", "getBackupPlanAssociationOutput"], () => require("./getBackupPlanAssociation"));
exports.getBackupPlanAssociations = null;
exports.getBackupPlanAssociationsOutput = null;
utilities.lazyLoad(exports, ["getBackupPlanAssociations", "getBackupPlanAssociationsOutput"], () => require("./getBackupPlanAssociations"));
exports.getBackupVault = null;
exports.getBackupVaultOutput = null;
utilities.lazyLoad(exports, ["getBackupVault", "getBackupVaultOutput"], () => require("./getBackupVault"));
exports.getDataSource = null;
exports.getDataSourceOutput = null;
utilities.lazyLoad(exports, ["getDataSource", "getDataSourceOutput"], () => require("./getDataSource"));
exports.getDataSourceReference = null;
exports.getDataSourceReferenceOutput = null;
utilities.lazyLoad(exports, ["getDataSourceReference", "getDataSourceReferenceOutput"], () => require("./getDataSourceReference"));
exports.getDataSourceReferences = null;
exports.getDataSourceReferencesOutput = null;
utilities.lazyLoad(exports, ["getDataSourceReferences", "getDataSourceReferencesOutput"], () => require("./getDataSourceReferences"));
exports.getDataSources = null;
exports.getDataSourcesOutput = null;
utilities.lazyLoad(exports, ["getDataSources", "getDataSourcesOutput"], () => require("./getDataSources"));
exports.getManagementServer = null;
exports.getManagementServerOutput = null;
utilities.lazyLoad(exports, ["getManagementServer", "getManagementServerOutput"], () => require("./getManagementServer"));
exports.ManagementServer = null;
utilities.lazyLoad(exports, ["ManagementServer"], () => require("./managementServer"));
exports.RestoreWorkload = null;
utilities.lazyLoad(exports, ["RestoreWorkload"], () => require("./restoreWorkload"));
exports.ServiceConfig = null;
utilities.lazyLoad(exports, ["ServiceConfig"], () => require("./serviceConfig"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:backupdisasterrecovery/backupPlan:BackupPlan":
                return new exports.BackupPlan(name, undefined, { urn });
            case "gcp:backupdisasterrecovery/backupPlanAssociation:BackupPlanAssociation":
                return new exports.BackupPlanAssociation(name, undefined, { urn });
            case "gcp:backupdisasterrecovery/backupVault:BackupVault":
                return new exports.BackupVault(name, undefined, { urn });
            case "gcp:backupdisasterrecovery/managementServer:ManagementServer":
                return new exports.ManagementServer(name, undefined, { urn });
            case "gcp:backupdisasterrecovery/restoreWorkload:RestoreWorkload":
                return new exports.RestoreWorkload(name, undefined, { urn });
            case "gcp:backupdisasterrecovery/serviceConfig:ServiceConfig":
                return new exports.ServiceConfig(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/backupPlan", _module);
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/backupPlanAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/backupVault", _module);
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/managementServer", _module);
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/restoreWorkload", _module);
pulumi.runtime.registerResourceModule("gcp", "backupdisasterrecovery/serviceConfig", _module);
