// @pulumi/gcp/securityposture/postureDeployment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostureDeployment = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a deployment of a security posture on a resource. A posture contains user curated policy sets. A posture can
 * be deployed on a project or on a folder or on an organization. To deploy a posture we need to populate the posture's name
 * and its revisionId in the posture deployment configuration. Every update to a deployed posture generates a new revision_id.
 * Thus, the updated revisionId should be used in the respective posture deployment's configuration to deploy that posture
 * on a resource.
 *
 * To get more information about PostureDeployment, see:
 *
 * * [API documentation](https://cloud.google.com/security-command-center/docs/reference/securityposture/rest/v1/organizations.locations.postureDeployments)
 * * How-to Guides
 *     * [Create and deploy a posture](https://cloud.google.com/security-command-center/docs/how-to-use-security-posture)
 *
 * ## Import
 *
 * PostureDeployment can be imported using any of these accepted formats:
 *
 * * `{{parent}}/locations/{{location}}/postureDeployments/{{posture_deployment_id}}`
 *
 * When using the `pulumi import` command, PostureDeployment can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:securityposture/postureDeployment:PostureDeployment default {{parent}}/locations/{{location}}/postureDeployments/{{posture_deployment_id}}
 * ```
 */
class PostureDeployment extends pulumi.CustomResource {
    /**
     * Get an existing PostureDeployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PostureDeployment(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:securityposture/postureDeployment:PostureDeployment';
    /**
     * Returns true if the given object is an instance of PostureDeployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PostureDeployment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["desiredPostureId"] = state?.desiredPostureId;
            resourceInputs["desiredPostureRevisionId"] = state?.desiredPostureRevisionId;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["failureMessage"] = state?.failureMessage;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["parent"] = state?.parent;
            resourceInputs["postureDeploymentId"] = state?.postureDeploymentId;
            resourceInputs["postureId"] = state?.postureId;
            resourceInputs["postureRevisionId"] = state?.postureRevisionId;
            resourceInputs["reconciling"] = state?.reconciling;
            resourceInputs["state"] = state?.state;
            resourceInputs["targetResource"] = state?.targetResource;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.parent === undefined && !opts.urn) {
                throw new Error("Missing required property 'parent'");
            }
            if (args?.postureDeploymentId === undefined && !opts.urn) {
                throw new Error("Missing required property 'postureDeploymentId'");
            }
            if (args?.postureId === undefined && !opts.urn) {
                throw new Error("Missing required property 'postureId'");
            }
            if (args?.postureRevisionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'postureRevisionId'");
            }
            if (args?.targetResource === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetResource'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["location"] = args?.location;
            resourceInputs["parent"] = args?.parent;
            resourceInputs["postureDeploymentId"] = args?.postureDeploymentId;
            resourceInputs["postureId"] = args?.postureId;
            resourceInputs["postureRevisionId"] = args?.postureRevisionId;
            resourceInputs["targetResource"] = args?.targetResource;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["desiredPostureId"] = undefined /*out*/;
            resourceInputs["desiredPostureRevisionId"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["failureMessage"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["reconciling"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PostureDeployment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PostureDeployment = PostureDeployment;
