// @pulumi/aws/efs/mountTarget.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MountTarget = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Elastic File System (EFS) mount target.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const foo = new aws.ec2.Vpc("foo", {cidrBlock: "10.0.0.0/16"});
 * const alphaSubnet = new aws.ec2.Subnet("alpha", {
 *     vpcId: foo.id,
 *     availabilityZone: "us-west-2a",
 *     cidrBlock: "10.0.1.0/24",
 * });
 * const alpha = new aws.efs.MountTarget("alpha", {
 *     fileSystemId: fooAwsEfsFileSystem.id,
 *     subnetId: alphaSubnet.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import the EFS mount targets using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:efs/mountTarget:MountTarget alpha fsmt-52a643fb
 * ```
 */
class MountTarget extends pulumi.CustomResource {
    /**
     * Get an existing MountTarget resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MountTarget(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:efs/mountTarget:MountTarget';
    /**
     * Returns true if the given object is an instance of MountTarget.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MountTarget.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["availabilityZoneId"] = state?.availabilityZoneId;
            resourceInputs["availabilityZoneName"] = state?.availabilityZoneName;
            resourceInputs["dnsName"] = state?.dnsName;
            resourceInputs["fileSystemArn"] = state?.fileSystemArn;
            resourceInputs["fileSystemId"] = state?.fileSystemId;
            resourceInputs["ipAddress"] = state?.ipAddress;
            resourceInputs["ipAddressType"] = state?.ipAddressType;
            resourceInputs["ipv6Address"] = state?.ipv6Address;
            resourceInputs["mountTargetDnsName"] = state?.mountTargetDnsName;
            resourceInputs["networkInterfaceId"] = state?.networkInterfaceId;
            resourceInputs["ownerId"] = state?.ownerId;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroups"] = state?.securityGroups;
            resourceInputs["subnetId"] = state?.subnetId;
        }
        else {
            const args = argsOrState;
            if (args?.fileSystemId === undefined && !opts.urn) {
                throw new Error("Missing required property 'fileSystemId'");
            }
            if (args?.subnetId === undefined && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            resourceInputs["fileSystemId"] = args?.fileSystemId;
            resourceInputs["ipAddress"] = args?.ipAddress;
            resourceInputs["ipAddressType"] = args?.ipAddressType;
            resourceInputs["ipv6Address"] = args?.ipv6Address;
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroups"] = args?.securityGroups;
            resourceInputs["subnetId"] = args?.subnetId;
            resourceInputs["availabilityZoneId"] = undefined /*out*/;
            resourceInputs["availabilityZoneName"] = undefined /*out*/;
            resourceInputs["dnsName"] = undefined /*out*/;
            resourceInputs["fileSystemArn"] = undefined /*out*/;
            resourceInputs["mountTargetDnsName"] = undefined /*out*/;
            resourceInputs["networkInterfaceId"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MountTarget.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MountTarget = MountTarget;
