// @pulumi/aws/account/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Region = exports.PrimaryContact = exports.getRegionsOutput = exports.getRegions = exports.getPrimaryContactOutput = exports.getPrimaryContact = exports.AlternativeContact = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AlternativeContact = null;
utilities.lazyLoad(exports, ["AlternativeContact"], () => require("./alternativeContact"));
exports.getPrimaryContact = null;
exports.getPrimaryContactOutput = null;
utilities.lazyLoad(exports, ["getPrimaryContact", "getPrimaryContactOutput"], () => require("./getPrimaryContact"));
exports.getRegions = null;
exports.getRegionsOutput = null;
utilities.lazyLoad(exports, ["getRegions", "getRegionsOutput"], () => require("./getRegions"));
exports.PrimaryContact = null;
utilities.lazyLoad(exports, ["PrimaryContact"], () => require("./primaryContact"));
exports.Region = null;
utilities.lazyLoad(exports, ["Region"], () => require("./region"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:account/alternativeContact:AlternativeContact":
                return new exports.AlternativeContact(name, undefined, { urn });
            case "aws:account/primaryContact:PrimaryContact":
                return new exports.PrimaryContact(name, undefined, { urn });
            case "aws:account/region:Region":
                return new exports.Region(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "account/alternativeContact", _module);
pulumi.runtime.registerResourceModule("aws", "account/primaryContact", _module);
pulumi.runtime.registerResourceModule("aws", "account/region", _module);
