// @pulumi/azure-native/types/enums/computeschedule/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.ResourceType = exports.OptimizationPreference = exports.NotificationType = exports.Month = exports.Language = exports.DeadlineType = exports.ActionType = void 0;
exports.ActionType = {
    /**
     * Perform a start action on the specified resources
     */
    Start: "Start",
    /**
     * Perform a deallocate action on the specified resources
     */
    Deallocate: "Deallocate",
    /**
     * Perform hibernate and deallocate on the specified resources
     */
    Hibernate: "Hibernate",
};
exports.DeadlineType = {
    /**
     * Default value of Unknown.
     */
    Unknown: "Unknown",
    /**
     * Initiate the operation at the given deadline.
     */
    InitiateAt: "InitiateAt",
    /**
     * Complete the operation by the given deadline.
     */
    CompleteBy: "CompleteBy",
};
exports.Language = {
    /**
     * American english language
     */
    EnUs: "en-us",
};
exports.Month = {
    /**
     * The January month.
     */
    January: "January",
    /**
     * The February month.
     */
    February: "February",
    /**
     * The March month.
     */
    March: "March",
    /**
     * The April month.
     */
    April: "April",
    /**
     * The May month.
     */
    May: "May",
    /**
     * The June month.
     */
    June: "June",
    /**
     * The July month.
     */
    July: "July",
    /**
     * The August month.
     */
    August: "August",
    /**
     * The September month.
     */
    September: "September",
    /**
     * The October month.
     */
    October: "October",
    /**
     * The November month.
     */
    November: "November",
    /**
     * The December month.
     */
    December: "December",
    /**
     * All months
     */
    All: "All",
};
exports.NotificationType = {
    /**
     * Notify through e-mail
     */
    Email: "Email",
};
exports.OptimizationPreference = {
    /**
     * Optimize while considering cost savings
     */
    Cost: "Cost",
    /**
     * Optimize while considering availability of resources
     */
    Availability: "Availability",
    /**
     * Optimize while considering a balance of cost and availability
     */
    CostAvailabilityBalanced: "CostAvailabilityBalanced",
};
exports.ResourceType = {
    /**
     * Resources defined are Virtual Machines
     */
    VirtualMachine: "VirtualMachine",
    /**
     * Resources defined are Virtual Machines Scale Sets
     */
    VirtualMachineScaleSet: "VirtualMachineScaleSet",
};
exports.WeekDay = {
    /**
     * Monday weekday.
     */
    Monday: "Monday",
    /**
     * Tuesday weekday.
     */
    Tuesday: "Tuesday",
    /**
     * Wednesday weekday.
     */
    Wednesday: "Wednesday",
    /**
     * Thursday weekday.
     */
    Thursday: "Thursday",
    /**
     * Friday weekday.
     */
    Friday: "Friday",
    /**
     * Saturday weekday.
     */
    Saturday: "Saturday",
    /**
     * Sunday weekday.
     */
    Sunday: "Sunday",
    /**
     * All week days
     */
    All: "All",
};
