// @pulumi/gcp/networksecurity/sacAttachment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SacAttachment = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a Secure Access Connect (SAC) attachment resource
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about SacAttachment, see:
 *
 * * [API documentation](https://cloud.google.com/secure-access-connect/docs/reference/network-security/rest/v1beta1/projects.locations.sacAttachments)
 * * How-to Guides
 *     * [QUICKSTART_TITLE](https://cloud.google.com/secure-access-connect/docs/overview)
 *
 * ## Example Usage
 *
 * ### Sac Attachment Prisma Access
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networksecurity.SacAttachment("default", {
 *     name: "sac-attachment-name",
 *     location: "us-central1",
 *     labels: {
 *         "label-one": "value-one",
 *     },
 *     nccGateway: "ncc_gateway_spoke_name",
 *     sacRealm: "sac_realm_name",
 * });
 * ```
 *
 * ## Import
 *
 * SacAttachment can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/sacAttachments/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, SacAttachment can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networksecurity/sacAttachment:SacAttachment default projects/{{project}}/locations/{{location}}/sacAttachments/{{name}}
 * $ pulumi import gcp:networksecurity/sacAttachment:SacAttachment default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:networksecurity/sacAttachment:SacAttachment default {{location}}/{{name}}
 * ```
 */
class SacAttachment extends pulumi.CustomResource {
    /**
     * Get an existing SacAttachment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SacAttachment(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networksecurity/sacAttachment:SacAttachment';
    /**
     * Returns true if the given object is an instance of SacAttachment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SacAttachment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["country"] = state?.country;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["nccGateway"] = state?.nccGateway;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["sacRealm"] = state?.sacRealm;
            resourceInputs["state"] = state?.state;
            resourceInputs["symantecOptions"] = state?.symantecOptions;
            resourceInputs["timeZone"] = state?.timeZone;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.nccGateway === undefined && !opts.urn) {
                throw new Error("Missing required property 'nccGateway'");
            }
            if (args?.sacRealm === undefined && !opts.urn) {
                throw new Error("Missing required property 'sacRealm'");
            }
            resourceInputs["country"] = args?.country;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["nccGateway"] = args?.nccGateway;
            resourceInputs["project"] = args?.project;
            resourceInputs["sacRealm"] = args?.sacRealm;
            resourceInputs["symantecOptions"] = args?.symantecOptions;
            resourceInputs["timeZone"] = args?.timeZone;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(SacAttachment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SacAttachment = SacAttachment;
