// @pulumi/azure-native/intune/androidMAMPolicyByName.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AndroidMAMPolicyByName = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Android Policy entity for Intune MAM.
 *
 * Uses Azure REST API version 2015-01-14-preview. In version 2.x of the Azure Native provider, it used API version 2015-01-14-preview.
 */
class AndroidMAMPolicyByName extends pulumi.CustomResource {
    /**
     * Get an existing AndroidMAMPolicyByName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AndroidMAMPolicyByName(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:intune:AndroidMAMPolicyByName';
    /**
     * Returns true if the given object is an instance of AndroidMAMPolicyByName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AndroidMAMPolicyByName.__pulumiType;
    }
    /**
     * Create a AndroidMAMPolicyByName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.friendlyName === undefined && !opts.urn) {
                throw new Error("Missing required property 'friendlyName'");
            }
            if (args?.hostName === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostName'");
            }
            resourceInputs["accessRecheckOfflineTimeout"] = args?.accessRecheckOfflineTimeout;
            resourceInputs["accessRecheckOnlineTimeout"] = args?.accessRecheckOnlineTimeout;
            resourceInputs["appSharingFromLevel"] = (args?.appSharingFromLevel) ?? "none";
            resourceInputs["appSharingToLevel"] = (args?.appSharingToLevel) ?? "none";
            resourceInputs["authentication"] = (args?.authentication) ?? "required";
            resourceInputs["clipboardSharingLevel"] = (args?.clipboardSharingLevel) ?? "blocked";
            resourceInputs["dataBackup"] = (args?.dataBackup) ?? "allow";
            resourceInputs["description"] = args?.description;
            resourceInputs["deviceCompliance"] = (args?.deviceCompliance) ?? "enable";
            resourceInputs["fileEncryption"] = (args?.fileEncryption) ?? "required";
            resourceInputs["fileSharingSaveAs"] = (args?.fileSharingSaveAs) ?? "allow";
            resourceInputs["friendlyName"] = args?.friendlyName;
            resourceInputs["hostName"] = args?.hostName;
            resourceInputs["location"] = args?.location;
            resourceInputs["managedBrowser"] = (args?.managedBrowser) ?? "required";
            resourceInputs["offlineWipeTimeout"] = args?.offlineWipeTimeout;
            resourceInputs["pin"] = (args?.pin) ?? "required";
            resourceInputs["pinNumRetry"] = args?.pinNumRetry;
            resourceInputs["policyName"] = args?.policyName;
            resourceInputs["screenCapture"] = (args?.screenCapture) ?? "allow";
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["groupStatus"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numOfApps"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessRecheckOfflineTimeout"] = undefined /*out*/;
            resourceInputs["accessRecheckOnlineTimeout"] = undefined /*out*/;
            resourceInputs["appSharingFromLevel"] = undefined /*out*/;
            resourceInputs["appSharingToLevel"] = undefined /*out*/;
            resourceInputs["authentication"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clipboardSharingLevel"] = undefined /*out*/;
            resourceInputs["dataBackup"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["deviceCompliance"] = undefined /*out*/;
            resourceInputs["fileEncryption"] = undefined /*out*/;
            resourceInputs["fileSharingSaveAs"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["groupStatus"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedBrowser"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numOfApps"] = undefined /*out*/;
            resourceInputs["offlineWipeTimeout"] = undefined /*out*/;
            resourceInputs["pin"] = undefined /*out*/;
            resourceInputs["pinNumRetry"] = undefined /*out*/;
            resourceInputs["screenCapture"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:intune/v20150114preview:AndroidMAMPolicyByName" }, { type: "azure-native:intune/v20150114privatepreview:AndroidMAMPolicyByName" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AndroidMAMPolicyByName.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AndroidMAMPolicyByName = AndroidMAMPolicyByName;
