// @pulumi/azure-native/management/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceGroup = exports.ManagementGroupSubscription = exports.ManagementGroup = exports.listServiceGroupAncestorsOutput = exports.listServiceGroupAncestors = exports.HierarchySetting = exports.getServiceGroupOutput = exports.getServiceGroup = exports.getManagementGroupSubscriptionOutput = exports.getManagementGroupSubscription = exports.getManagementGroupOutput = exports.getManagementGroup = exports.getHierarchySettingOutput = exports.getHierarchySetting = exports.getEntityOutput = exports.getEntity = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getEntity = null;
exports.getEntityOutput = null;
utilities.lazyLoad(exports, ["getEntity", "getEntityOutput"], () => require("./getEntity"));
exports.getHierarchySetting = null;
exports.getHierarchySettingOutput = null;
utilities.lazyLoad(exports, ["getHierarchySetting", "getHierarchySettingOutput"], () => require("./getHierarchySetting"));
exports.getManagementGroup = null;
exports.getManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getManagementGroup", "getManagementGroupOutput"], () => require("./getManagementGroup"));
exports.getManagementGroupSubscription = null;
exports.getManagementGroupSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getManagementGroupSubscription", "getManagementGroupSubscriptionOutput"], () => require("./getManagementGroupSubscription"));
exports.getServiceGroup = null;
exports.getServiceGroupOutput = null;
utilities.lazyLoad(exports, ["getServiceGroup", "getServiceGroupOutput"], () => require("./getServiceGroup"));
exports.HierarchySetting = null;
utilities.lazyLoad(exports, ["HierarchySetting"], () => require("./hierarchySetting"));
exports.listServiceGroupAncestors = null;
exports.listServiceGroupAncestorsOutput = null;
utilities.lazyLoad(exports, ["listServiceGroupAncestors", "listServiceGroupAncestorsOutput"], () => require("./listServiceGroupAncestors"));
exports.ManagementGroup = null;
utilities.lazyLoad(exports, ["ManagementGroup"], () => require("./managementGroup"));
exports.ManagementGroupSubscription = null;
utilities.lazyLoad(exports, ["ManagementGroupSubscription"], () => require("./managementGroupSubscription"));
exports.ServiceGroup = null;
utilities.lazyLoad(exports, ["ServiceGroup"], () => require("./serviceGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:management:HierarchySetting":
                return new exports.HierarchySetting(name, undefined, { urn });
            case "azure-native:management:ManagementGroup":
                return new exports.ManagementGroup(name, undefined, { urn });
            case "azure-native:management:ManagementGroupSubscription":
                return new exports.ManagementGroupSubscription(name, undefined, { urn });
            case "azure-native:management:ServiceGroup":
                return new exports.ServiceGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "management", _module);
