// @pulumi/azure-native/machinelearningservices/environmentSpecificationVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentSpecificationVersion = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2021-03-01-preview. In version 2.x of the Azure Native provider, it used API version 2021-03-01-preview.
 */
class EnvironmentSpecificationVersion extends pulumi.CustomResource {
    /**
     * Get an existing EnvironmentSpecificationVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnvironmentSpecificationVersion(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:machinelearningservices:EnvironmentSpecificationVersion';
    /**
     * Returns true if the given object is an instance of EnvironmentSpecificationVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentSpecificationVersion.__pulumiType;
    }
    /**
     * Create a EnvironmentSpecificationVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.properties === undefined && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["name"] = args?.name;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["version"] = args?.version;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20220201preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20220201preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20220501:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20220601preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20221001:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20221001preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20221201preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230201preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230401:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230401:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230401preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230401preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230601preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230601preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230801preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20230801preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20231001:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20231001:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240101preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20240101preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240401:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20240401:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240401preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20240401preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240701preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20240701preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20241001:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20241001:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20241001preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20241001preview:EnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20250101preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20250401:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20250401preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20250601:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20250701preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20250901:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20251001preview:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices/v20251201:EnvironmentSpecificationVersion" }, { type: "azure-native:machinelearningservices:EnvironmentVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EnvironmentSpecificationVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EnvironmentSpecificationVersion = EnvironmentSpecificationVersion;
