// @pulumi/azure-native/machinelearningservices/machineLearningDatastore.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MachineLearningDatastore = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Machine Learning datastore object wrapped into ARM resource envelope.
 *
 * Uses Azure REST API version 2020-05-01-preview. In version 2.x of the Azure Native provider, it used API version 2020-05-01-preview.
 */
class MachineLearningDatastore extends pulumi.CustomResource {
    /**
     * Get an existing MachineLearningDatastore resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MachineLearningDatastore(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:machinelearningservices:MachineLearningDatastore';
    /**
     * Returns true if the given object is an instance of MachineLearningDatastore.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MachineLearningDatastore.__pulumiType;
    }
    /**
     * Create a MachineLearningDatastore resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.dataStoreType === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataStoreType'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.workspaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["accountKey"] = args?.accountKey;
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["adlsResourceGroup"] = args?.adlsResourceGroup;
            resourceInputs["adlsSubscriptionId"] = args?.adlsSubscriptionId;
            resourceInputs["authorityUrl"] = args?.authorityUrl;
            resourceInputs["clientId"] = args?.clientId;
            resourceInputs["clientSecret"] = args?.clientSecret;
            resourceInputs["containerName"] = args?.containerName;
            resourceInputs["dataStoreType"] = args?.dataStoreType;
            resourceInputs["databaseName"] = args?.databaseName;
            resourceInputs["datastoreName"] = args?.datastoreName;
            resourceInputs["description"] = args?.description;
            resourceInputs["endpoint"] = args?.endpoint;
            resourceInputs["enforceSSL"] = (args?.enforceSSL) ?? true;
            resourceInputs["fileSystem"] = args?.fileSystem;
            resourceInputs["includeSecret"] = (args?.includeSecret) ?? true;
            resourceInputs["name"] = args?.name;
            resourceInputs["password"] = args?.password;
            resourceInputs["port"] = args?.port;
            resourceInputs["protocol"] = args?.protocol;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceUrl"] = args?.resourceUrl;
            resourceInputs["sasToken"] = args?.sasToken;
            resourceInputs["serverName"] = args?.serverName;
            resourceInputs["shareName"] = args?.shareName;
            resourceInputs["skipValidation"] = args?.skipValidation;
            resourceInputs["storageAccountResourceGroup"] = args?.storageAccountResourceGroup;
            resourceInputs["storageAccountSubscriptionId"] = args?.storageAccountSubscriptionId;
            resourceInputs["storeName"] = args?.storeName;
            resourceInputs["tenantId"] = args?.tenantId;
            resourceInputs["userId"] = args?.userId;
            resourceInputs["userName"] = args?.userName;
            resourceInputs["workspaceName"] = args?.workspaceName;
            resourceInputs["workspaceSystemAssignedIdentity"] = args?.workspaceSystemAssignedIdentity;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20200501preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20210301preview:Datastore" }, { type: "azure-native:machinelearningservices/v20210301preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20220201preview:Datastore" }, { type: "azure-native:machinelearningservices/v20220201preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20220501:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20220601preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20221001:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20221001preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20221201preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20230201preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20230401:Datastore" }, { type: "azure-native:machinelearningservices/v20230401:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20230401preview:Datastore" }, { type: "azure-native:machinelearningservices/v20230401preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20230601preview:Datastore" }, { type: "azure-native:machinelearningservices/v20230601preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20230801preview:Datastore" }, { type: "azure-native:machinelearningservices/v20230801preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20231001:Datastore" }, { type: "azure-native:machinelearningservices/v20231001:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20240101preview:Datastore" }, { type: "azure-native:machinelearningservices/v20240101preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20240401:Datastore" }, { type: "azure-native:machinelearningservices/v20240401:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20240401preview:Datastore" }, { type: "azure-native:machinelearningservices/v20240401preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20240701preview:Datastore" }, { type: "azure-native:machinelearningservices/v20240701preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20241001:Datastore" }, { type: "azure-native:machinelearningservices/v20241001:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20241001preview:Datastore" }, { type: "azure-native:machinelearningservices/v20241001preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250101preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250401:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250401preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250601:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250701preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20250901:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20251001preview:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices/v20251201:MachineLearningDatastore" }, { type: "azure-native:machinelearningservices:Datastore" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MachineLearningDatastore.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MachineLearningDatastore = MachineLearningDatastore;
