// @pulumi/azure-native/automanage/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getConfigurationProfilesVersionOutput = exports.getConfigurationProfilesVersion = exports.getConfigurationProfileHCRPAssignmentOutput = exports.getConfigurationProfileHCRPAssignment = exports.getConfigurationProfileHCIAssignmentOutput = exports.getConfigurationProfileHCIAssignment = exports.getConfigurationProfileAssignmentOutput = exports.getConfigurationProfileAssignment = exports.getConfigurationProfileOutput = exports.getConfigurationProfile = exports.ConfigurationProfilesVersion = exports.ConfigurationProfileHCRPAssignment = exports.ConfigurationProfileHCIAssignment = exports.ConfigurationProfileAssignment = exports.ConfigurationProfile = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ConfigurationProfile = null;
utilities.lazyLoad(exports, ["ConfigurationProfile"], () => require("./configurationProfile"));
exports.ConfigurationProfileAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileAssignment"], () => require("./configurationProfileAssignment"));
exports.ConfigurationProfileHCIAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileHCIAssignment"], () => require("./configurationProfileHCIAssignment"));
exports.ConfigurationProfileHCRPAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileHCRPAssignment"], () => require("./configurationProfileHCRPAssignment"));
exports.ConfigurationProfilesVersion = null;
utilities.lazyLoad(exports, ["ConfigurationProfilesVersion"], () => require("./configurationProfilesVersion"));
exports.getConfigurationProfile = null;
exports.getConfigurationProfileOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfile", "getConfigurationProfileOutput"], () => require("./getConfigurationProfile"));
exports.getConfigurationProfileAssignment = null;
exports.getConfigurationProfileAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileAssignment", "getConfigurationProfileAssignmentOutput"], () => require("./getConfigurationProfileAssignment"));
exports.getConfigurationProfileHCIAssignment = null;
exports.getConfigurationProfileHCIAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileHCIAssignment", "getConfigurationProfileHCIAssignmentOutput"], () => require("./getConfigurationProfileHCIAssignment"));
exports.getConfigurationProfileHCRPAssignment = null;
exports.getConfigurationProfileHCRPAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileHCRPAssignment", "getConfigurationProfileHCRPAssignmentOutput"], () => require("./getConfigurationProfileHCRPAssignment"));
exports.getConfigurationProfilesVersion = null;
exports.getConfigurationProfilesVersionOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfilesVersion", "getConfigurationProfilesVersionOutput"], () => require("./getConfigurationProfilesVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:automanage:ConfigurationProfile":
                return new exports.ConfigurationProfile(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileAssignment":
                return new exports.ConfigurationProfileAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileHCIAssignment":
                return new exports.ConfigurationProfileHCIAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileHCRPAssignment":
                return new exports.ConfigurationProfileHCRPAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfilesVersion":
                return new exports.ConfigurationProfilesVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "automanage", _module);
