// @pulumi/azure-native/trafficmanager/endpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Endpoint = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Class representing a Traffic Manager endpoint.
 *
 * Uses Azure REST API version 2022-04-01.
 *
 * Other available API versions: 2015-11-01, 2017-03-01, 2017-05-01, 2018-02-01, 2018-03-01, 2018-04-01, 2018-08-01, 2022-04-01-preview, 2024-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native trafficmanager [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Endpoint extends pulumi.CustomResource {
    /**
     * Get an existing Endpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Endpoint(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:trafficmanager:Endpoint';
    /**
     * Returns true if the given object is an instance of Endpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Endpoint.__pulumiType;
    }
    /**
     * Create a Endpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.endpointType === undefined && !opts.urn) {
                throw new Error("Missing required property 'endpointType'");
            }
            if (args?.profileName === undefined && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["alwaysServe"] = args?.alwaysServe;
            resourceInputs["customHeaders"] = args?.customHeaders;
            resourceInputs["endpointLocation"] = args?.endpointLocation;
            resourceInputs["endpointMonitorStatus"] = args?.endpointMonitorStatus;
            resourceInputs["endpointName"] = args?.endpointName;
            resourceInputs["endpointStatus"] = args?.endpointStatus;
            resourceInputs["endpointType"] = args?.endpointType;
            resourceInputs["geoMapping"] = args?.geoMapping;
            resourceInputs["id"] = args?.id;
            resourceInputs["minChildEndpoints"] = args?.minChildEndpoints;
            resourceInputs["minChildEndpointsIPv4"] = args?.minChildEndpointsIPv4;
            resourceInputs["minChildEndpointsIPv6"] = args?.minChildEndpointsIPv6;
            resourceInputs["name"] = args?.name;
            resourceInputs["priority"] = args?.priority;
            resourceInputs["profileName"] = args?.profileName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["subnets"] = args?.subnets;
            resourceInputs["target"] = args?.target;
            resourceInputs["targetResourceId"] = args?.targetResourceId;
            resourceInputs["type"] = args?.type;
            resourceInputs["weight"] = args?.weight;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
        }
        else {
            resourceInputs["alwaysServe"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["customHeaders"] = undefined /*out*/;
            resourceInputs["endpointLocation"] = undefined /*out*/;
            resourceInputs["endpointMonitorStatus"] = undefined /*out*/;
            resourceInputs["endpointStatus"] = undefined /*out*/;
            resourceInputs["geoMapping"] = undefined /*out*/;
            resourceInputs["minChildEndpoints"] = undefined /*out*/;
            resourceInputs["minChildEndpointsIPv4"] = undefined /*out*/;
            resourceInputs["minChildEndpointsIPv6"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["target"] = undefined /*out*/;
            resourceInputs["targetResourceId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["weight"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20220401:Endpoint" }, { type: "azure-native:network/v20220401preview:Endpoint" }, { type: "azure-native:network:Endpoint" }, { type: "azure-native:trafficmanager/v20151101:Endpoint" }, { type: "azure-native:trafficmanager/v20170301:Endpoint" }, { type: "azure-native:trafficmanager/v20170501:Endpoint" }, { type: "azure-native:trafficmanager/v20180201:Endpoint" }, { type: "azure-native:trafficmanager/v20180301:Endpoint" }, { type: "azure-native:trafficmanager/v20180401:Endpoint" }, { type: "azure-native:trafficmanager/v20180801:Endpoint" }, { type: "azure-native:trafficmanager/v20220401:Endpoint" }, { type: "azure-native:trafficmanager/v20220401preview:Endpoint" }, { type: "azure-native:trafficmanager/v20240401preview:Endpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Endpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Endpoint = Endpoint;
