// @pulumi/azure-native/trafficmanager/profile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Profile = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Class representing a Traffic Manager profile.
 *
 * Uses Azure REST API version 2022-04-01.
 *
 * Other available API versions: 2015-11-01, 2017-03-01, 2017-05-01, 2018-02-01, 2018-03-01, 2018-04-01, 2018-08-01, 2022-04-01-preview, 2024-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native trafficmanager [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Profile extends pulumi.CustomResource {
    /**
     * Get an existing Profile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Profile(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:trafficmanager:Profile';
    /**
     * Returns true if the given object is an instance of Profile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Profile.__pulumiType;
    }
    /**
     * Create a Profile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["allowedEndpointRecordTypes"] = args?.allowedEndpointRecordTypes;
            resourceInputs["dnsConfig"] = args?.dnsConfig;
            resourceInputs["endpoints"] = args?.endpoints;
            resourceInputs["id"] = args?.id;
            resourceInputs["location"] = args?.location;
            resourceInputs["maxReturn"] = args?.maxReturn;
            resourceInputs["monitorConfig"] = args?.monitorConfig;
            resourceInputs["name"] = args?.name;
            resourceInputs["profileName"] = args?.profileName;
            resourceInputs["profileStatus"] = args?.profileStatus;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["trafficRoutingMethod"] = args?.trafficRoutingMethod;
            resourceInputs["trafficViewEnrollmentStatus"] = args?.trafficViewEnrollmentStatus;
            resourceInputs["type"] = args?.type;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowedEndpointRecordTypes"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["dnsConfig"] = undefined /*out*/;
            resourceInputs["endpoints"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maxReturn"] = undefined /*out*/;
            resourceInputs["monitorConfig"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["profileStatus"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trafficRoutingMethod"] = undefined /*out*/;
            resourceInputs["trafficViewEnrollmentStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20220401:Profile" }, { type: "azure-native:network/v20220401preview:Profile" }, { type: "azure-native:network:Profile" }, { type: "azure-native:trafficmanager/v20151101:Profile" }, { type: "azure-native:trafficmanager/v20170301:Profile" }, { type: "azure-native:trafficmanager/v20170501:Profile" }, { type: "azure-native:trafficmanager/v20180201:Profile" }, { type: "azure-native:trafficmanager/v20180301:Profile" }, { type: "azure-native:trafficmanager/v20180401:Profile" }, { type: "azure-native:trafficmanager/v20180801:Profile" }, { type: "azure-native:trafficmanager/v20220401:Profile" }, { type: "azure-native:trafficmanager/v20220401preview:Profile" }, { type: "azure-native:trafficmanager/v20240401preview:Profile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Profile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Profile = Profile;
