// @pulumi/aws/apprunner/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcIngressConnection = exports.VpcConnector = exports.Service = exports.ObservabilityConfiguration = exports.getHostedZoneIdOutput = exports.getHostedZoneId = exports.Deployment = exports.DefaultAutoScalingConfigurationVersion = exports.CustomDomainAssociation = exports.Connection = exports.AutoScalingConfigurationVersion = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AutoScalingConfigurationVersion = null;
utilities.lazyLoad(exports, ["AutoScalingConfigurationVersion"], () => require("./autoScalingConfigurationVersion"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.CustomDomainAssociation = null;
utilities.lazyLoad(exports, ["CustomDomainAssociation"], () => require("./customDomainAssociation"));
exports.DefaultAutoScalingConfigurationVersion = null;
utilities.lazyLoad(exports, ["DefaultAutoScalingConfigurationVersion"], () => require("./defaultAutoScalingConfigurationVersion"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.getHostedZoneId = null;
exports.getHostedZoneIdOutput = null;
utilities.lazyLoad(exports, ["getHostedZoneId", "getHostedZoneIdOutput"], () => require("./getHostedZoneId"));
exports.ObservabilityConfiguration = null;
utilities.lazyLoad(exports, ["ObservabilityConfiguration"], () => require("./observabilityConfiguration"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.VpcConnector = null;
utilities.lazyLoad(exports, ["VpcConnector"], () => require("./vpcConnector"));
exports.VpcIngressConnection = null;
utilities.lazyLoad(exports, ["VpcIngressConnection"], () => require("./vpcIngressConnection"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:apprunner/autoScalingConfigurationVersion:AutoScalingConfigurationVersion":
                return new exports.AutoScalingConfigurationVersion(name, undefined, { urn });
            case "aws:apprunner/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "aws:apprunner/customDomainAssociation:CustomDomainAssociation":
                return new exports.CustomDomainAssociation(name, undefined, { urn });
            case "aws:apprunner/defaultAutoScalingConfigurationVersion:DefaultAutoScalingConfigurationVersion":
                return new exports.DefaultAutoScalingConfigurationVersion(name, undefined, { urn });
            case "aws:apprunner/deployment:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "aws:apprunner/observabilityConfiguration:ObservabilityConfiguration":
                return new exports.ObservabilityConfiguration(name, undefined, { urn });
            case "aws:apprunner/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "aws:apprunner/vpcConnector:VpcConnector":
                return new exports.VpcConnector(name, undefined, { urn });
            case "aws:apprunner/vpcIngressConnection:VpcIngressConnection":
                return new exports.VpcIngressConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "apprunner/autoScalingConfigurationVersion", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/connection", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/customDomainAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/defaultAutoScalingConfigurationVersion", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/deployment", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/observabilityConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/service", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/vpcConnector", _module);
pulumi.runtime.registerResourceModule("aws", "apprunner/vpcIngressConnection", _module);
