// @pulumi/aws/apprunner/vpcConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcConnector = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an App Runner VPC Connector.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const connector = new aws.apprunner.VpcConnector("connector", {
 *     vpcConnectorName: "name",
 *     subnets: [
 *         "subnet1",
 *         "subnet2",
 *     ],
 *     securityGroups: [
 *         "sg1",
 *         "sg2",
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `arn` (String) Amazon Resource Name (ARN) of the App Runner VPC connector.
 *
 * Using `pulumi import`, import App Runner vpc connector using the `arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:apprunner/vpcConnector:VpcConnector example arn:aws:apprunner:us-east-1:1234567890:vpcconnector/example/1/0a03292a89764e5882c41d8f991c82fe
 * ```
 */
class VpcConnector extends pulumi.CustomResource {
    /**
     * Get an existing VpcConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcConnector(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:apprunner/vpcConnector:VpcConnector';
    /**
     * Returns true if the given object is an instance of VpcConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcConnector.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroups"] = state?.securityGroups;
            resourceInputs["status"] = state?.status;
            resourceInputs["subnets"] = state?.subnets;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["vpcConnectorName"] = state?.vpcConnectorName;
            resourceInputs["vpcConnectorRevision"] = state?.vpcConnectorRevision;
        }
        else {
            const args = argsOrState;
            if (args?.securityGroups === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGroups'");
            }
            if (args?.subnets === undefined && !opts.urn) {
                throw new Error("Missing required property 'subnets'");
            }
            if (args?.vpcConnectorName === undefined && !opts.urn) {
                throw new Error("Missing required property 'vpcConnectorName'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroups"] = args?.securityGroups;
            resourceInputs["subnets"] = args?.subnets;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["vpcConnectorName"] = args?.vpcConnectorName;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["vpcConnectorRevision"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcConnector = VpcConnector;
