// @pulumi/gcp/dataloss/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreventionStoredInfoType = exports.PreventionJobTrigger = exports.PreventionInspectTemplate = exports.PreventionDiscoveryConfig = exports.PreventionDeidentifyTemplate = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.PreventionDeidentifyTemplate = null;
utilities.lazyLoad(exports, ["PreventionDeidentifyTemplate"], () => require("./preventionDeidentifyTemplate"));
exports.PreventionDiscoveryConfig = null;
utilities.lazyLoad(exports, ["PreventionDiscoveryConfig"], () => require("./preventionDiscoveryConfig"));
exports.PreventionInspectTemplate = null;
utilities.lazyLoad(exports, ["PreventionInspectTemplate"], () => require("./preventionInspectTemplate"));
exports.PreventionJobTrigger = null;
utilities.lazyLoad(exports, ["PreventionJobTrigger"], () => require("./preventionJobTrigger"));
exports.PreventionStoredInfoType = null;
utilities.lazyLoad(exports, ["PreventionStoredInfoType"], () => require("./preventionStoredInfoType"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:dataloss/preventionDeidentifyTemplate:PreventionDeidentifyTemplate":
                return new exports.PreventionDeidentifyTemplate(name, undefined, { urn });
            case "gcp:dataloss/preventionDiscoveryConfig:PreventionDiscoveryConfig":
                return new exports.PreventionDiscoveryConfig(name, undefined, { urn });
            case "gcp:dataloss/preventionInspectTemplate:PreventionInspectTemplate":
                return new exports.PreventionInspectTemplate(name, undefined, { urn });
            case "gcp:dataloss/preventionJobTrigger:PreventionJobTrigger":
                return new exports.PreventionJobTrigger(name, undefined, { urn });
            case "gcp:dataloss/preventionStoredInfoType:PreventionStoredInfoType":
                return new exports.PreventionStoredInfoType(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionDeidentifyTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionDiscoveryConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionInspectTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionJobTrigger", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionStoredInfoType", _module);
