// @pulumi/gcp/dataloss/preventionDiscoveryConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreventionDiscoveryConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Configuration for discovery to scan resources for profile generation. Only one discovery configuration may exist per organization, folder, or project.
 *
 * To get more information about DiscoveryConfig, see:
 *
 * * [API documentation](https://cloud.google.com/dlp/docs/reference/rest/v2/projects.locations.discoveryConfigs)
 * * How-to Guides
 *     * [Schedule inspection scan](https://cloud.google.com/dlp/docs/schedule-inspection-scan)
 *
 * ## Example Usage
 *
 * ## Import
 *
 * DiscoveryConfig can be imported using any of these accepted formats:
 *
 * * `{{parent}}/discoveryConfigs/{{name}}`
 * * `{{parent}}/{{name}}`
 *
 * When using the `pulumi import` command, DiscoveryConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataloss/preventionDiscoveryConfig:PreventionDiscoveryConfig default {{parent}}/discoveryConfigs/{{name}}
 * $ pulumi import gcp:dataloss/preventionDiscoveryConfig:PreventionDiscoveryConfig default {{parent}}/{{name}}
 * ```
 */
class PreventionDiscoveryConfig extends pulumi.CustomResource {
    /**
     * Get an existing PreventionDiscoveryConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PreventionDiscoveryConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:dataloss/preventionDiscoveryConfig:PreventionDiscoveryConfig';
    /**
     * Returns true if the given object is an instance of PreventionDiscoveryConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PreventionDiscoveryConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["actions"] = state?.actions;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["errors"] = state?.errors;
            resourceInputs["inspectTemplates"] = state?.inspectTemplates;
            resourceInputs["lastRunTime"] = state?.lastRunTime;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["orgConfig"] = state?.orgConfig;
            resourceInputs["otherCloudStartingLocation"] = state?.otherCloudStartingLocation;
            resourceInputs["parent"] = state?.parent;
            resourceInputs["status"] = state?.status;
            resourceInputs["targets"] = state?.targets;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.parent === undefined && !opts.urn) {
                throw new Error("Missing required property 'parent'");
            }
            resourceInputs["actions"] = args?.actions;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["inspectTemplates"] = args?.inspectTemplates;
            resourceInputs["location"] = args?.location;
            resourceInputs["orgConfig"] = args?.orgConfig;
            resourceInputs["otherCloudStartingLocation"] = args?.otherCloudStartingLocation;
            resourceInputs["parent"] = args?.parent;
            resourceInputs["status"] = args?.status;
            resourceInputs["targets"] = args?.targets;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["lastRunTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PreventionDiscoveryConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PreventionDiscoveryConfig = PreventionDiscoveryConfig;
