// @pulumi/aws/evidently/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Segment = exports.Project = exports.Launch = exports.Feature = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Feature = null;
utilities.lazyLoad(exports, ["Feature"], () => require("./feature"));
exports.Launch = null;
utilities.lazyLoad(exports, ["Launch"], () => require("./launch"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Segment = null;
utilities.lazyLoad(exports, ["Segment"], () => require("./segment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:evidently/feature:Feature":
                return new exports.Feature(name, undefined, { urn });
            case "aws:evidently/launch:Launch":
                return new exports.Launch(name, undefined, { urn });
            case "aws:evidently/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws:evidently/segment:Segment":
                return new exports.Segment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "evidently/feature", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/launch", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/project", _module);
pulumi.runtime.registerResourceModule("aws", "evidently/segment", _module);
