// @pulumi/aws/workmail/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Organization = exports.Group = exports.Domain = exports.DefaultDomain = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.DefaultDomain = null;
utilities.lazyLoad(exports, ["DefaultDomain"], () => require("./defaultDomain"));
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.Organization = null;
utilities.lazyLoad(exports, ["Organization"], () => require("./organization"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:workmail/defaultDomain:DefaultDomain":
                return new exports.DefaultDomain(name, undefined, { urn });
            case "aws:workmail/domain:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "aws:workmail/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws:workmail/organization:Organization":
                return new exports.Organization(name, undefined, { urn });
            case "aws:workmail/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "workmail/defaultDomain", _module);
pulumi.runtime.registerResourceModule("aws", "workmail/domain", _module);
pulumi.runtime.registerResourceModule("aws", "workmail/group", _module);
pulumi.runtime.registerResourceModule("aws", "workmail/organization", _module);
pulumi.runtime.registerResourceModule("aws", "workmail/user", _module);
