// @pulumi/aws/workmail/organization.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS WorkMail Organization.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.workmail.Organization("example", {organizationAlias: "example-org"});
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `organizationId` - (String) ID of the WorkMail Organization.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import WorkMail Organization using the `organizationId`. For example:
 *
 * ```sh
 * $ pulumi import aws:workmail/organization:Organization example m-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 * ```
 *
 * > **NOTE:** The `kmsKeyArn`, `deleteDirectory`, and `deleteIdentityCenterApplication` attributes are not returned by the AWS API and will not be set after import. Add them back to your configuration manually if needed.
 */
class Organization extends pulumi.CustomResource {
    /**
     * Get an existing Organization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Organization(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:workmail/organization:Organization';
    /**
     * Returns true if the given object is an instance of Organization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Organization.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["completedDate"] = state?.completedDate;
            resourceInputs["defaultMailDomain"] = state?.defaultMailDomain;
            resourceInputs["deleteDirectory"] = state?.deleteDirectory;
            resourceInputs["deleteIdentityCenterApplication"] = state?.deleteIdentityCenterApplication;
            resourceInputs["directoryId"] = state?.directoryId;
            resourceInputs["directoryType"] = state?.directoryType;
            resourceInputs["interoperabilityEnabled"] = state?.interoperabilityEnabled;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["migrationAdmin"] = state?.migrationAdmin;
            resourceInputs["organizationAlias"] = state?.organizationAlias;
            resourceInputs["organizationId"] = state?.organizationId;
            resourceInputs["region"] = state?.region;
            resourceInputs["state"] = state?.state;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.organizationAlias === undefined && !opts.urn) {
                throw new Error("Missing required property 'organizationAlias'");
            }
            resourceInputs["deleteDirectory"] = args?.deleteDirectory;
            resourceInputs["deleteIdentityCenterApplication"] = args?.deleteIdentityCenterApplication;
            resourceInputs["directoryId"] = args?.directoryId;
            resourceInputs["interoperabilityEnabled"] = args?.interoperabilityEnabled;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["organizationAlias"] = args?.organizationAlias;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["completedDate"] = undefined /*out*/;
            resourceInputs["defaultMailDomain"] = undefined /*out*/;
            resourceInputs["directoryType"] = undefined /*out*/;
            resourceInputs["migrationAdmin"] = undefined /*out*/;
            resourceInputs["organizationId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Organization.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Organization = Organization;
