// @pulumi/aws/workmail/user.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS WorkMail User.
 *
 * This resource registers the user with WorkMail on create so the mailbox is enabled and ready for use. This results in the accumulation of costs, for more details, see [pricing](https://aws.amazon.com/workmail/pricing/).
 * On destroy, it deregisters the user before deletion.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.workmail.Organization("example", {
 *     organizationAlias: "example-org",
 *     deleteDirectory: true,
 * });
 * const exampleUser = new aws.workmail.User("example", {
 *     organizationId: example.organizationId,
 *     name: "example-user",
 *     displayName: "Example User",
 *     email: "example-user@example.com",
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `organizationId` - Identifier of the WorkMail organization.
 * * `userId` - Identifier of the user.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import WorkMail User using `organization_id,user_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:workmail/user:User example m-1234567890abcdef1234567890abcdef,12345678-1234-1234-1234-123456789012
 * ```
 */
class User extends pulumi.CustomResource {
    /**
     * Get an existing User resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new User(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:workmail/user:User';
    /**
     * Returns true if the given object is an instance of User.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === User.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["city"] = state?.city;
            resourceInputs["company"] = state?.company;
            resourceInputs["country"] = state?.country;
            resourceInputs["department"] = state?.department;
            resourceInputs["disabledDate"] = state?.disabledDate;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["email"] = state?.email;
            resourceInputs["enabledDate"] = state?.enabledDate;
            resourceInputs["firstName"] = state?.firstName;
            resourceInputs["hiddenFromGlobalAddressList"] = state?.hiddenFromGlobalAddressList;
            resourceInputs["identityProviderIdentityStoreId"] = state?.identityProviderIdentityStoreId;
            resourceInputs["identityProviderUserId"] = state?.identityProviderUserId;
            resourceInputs["initials"] = state?.initials;
            resourceInputs["jobTitle"] = state?.jobTitle;
            resourceInputs["lastName"] = state?.lastName;
            resourceInputs["mailboxDeprovisionedDate"] = state?.mailboxDeprovisionedDate;
            resourceInputs["mailboxProvisionedDate"] = state?.mailboxProvisionedDate;
            resourceInputs["name"] = state?.name;
            resourceInputs["office"] = state?.office;
            resourceInputs["organizationId"] = state?.organizationId;
            resourceInputs["password"] = state?.password;
            resourceInputs["region"] = state?.region;
            resourceInputs["state"] = state?.state;
            resourceInputs["street"] = state?.street;
            resourceInputs["telephone"] = state?.telephone;
            resourceInputs["userId"] = state?.userId;
            resourceInputs["userRole"] = state?.userRole;
            resourceInputs["zipCode"] = state?.zipCode;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.email === undefined && !opts.urn) {
                throw new Error("Missing required property 'email'");
            }
            if (args?.organizationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'organizationId'");
            }
            resourceInputs["city"] = args?.city;
            resourceInputs["company"] = args?.company;
            resourceInputs["country"] = args?.country;
            resourceInputs["department"] = args?.department;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["email"] = args?.email;
            resourceInputs["firstName"] = args?.firstName;
            resourceInputs["hiddenFromGlobalAddressList"] = args?.hiddenFromGlobalAddressList;
            resourceInputs["identityProviderUserId"] = args?.identityProviderUserId;
            resourceInputs["initials"] = args?.initials;
            resourceInputs["jobTitle"] = args?.jobTitle;
            resourceInputs["lastName"] = args?.lastName;
            resourceInputs["name"] = args?.name;
            resourceInputs["office"] = args?.office;
            resourceInputs["organizationId"] = args?.organizationId;
            resourceInputs["password"] = args?.password ? pulumi.secret(args.password) : undefined;
            resourceInputs["region"] = args?.region;
            resourceInputs["street"] = args?.street;
            resourceInputs["telephone"] = args?.telephone;
            resourceInputs["userRole"] = args?.userRole;
            resourceInputs["zipCode"] = args?.zipCode;
            resourceInputs["disabledDate"] = undefined /*out*/;
            resourceInputs["enabledDate"] = undefined /*out*/;
            resourceInputs["identityProviderIdentityStoreId"] = undefined /*out*/;
            resourceInputs["mailboxDeprovisionedDate"] = undefined /*out*/;
            resourceInputs["mailboxProvisionedDate"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["userId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["password"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(User.__pulumiType, name, resourceInputs, opts);
    }
}
exports.User = User;
