// @pulumi/azure-native/webpubsub/webPubSubCustomCertificate.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebPubSubCustomCertificate = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A custom certificate.
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-08-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-08-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native webpubsub [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebPubSubCustomCertificate extends pulumi.CustomResource {
    /**
     * Get an existing WebPubSubCustomCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebPubSubCustomCertificate(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:webpubsub:WebPubSubCustomCertificate';
    /**
     * Returns true if the given object is an instance of WebPubSubCustomCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebPubSubCustomCertificate.__pulumiType;
    }
    /**
     * Create a WebPubSubCustomCertificate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.keyVaultBaseUri === undefined && !opts.urn) {
                throw new Error("Missing required property 'keyVaultBaseUri'");
            }
            if (args?.keyVaultSecretName === undefined && !opts.urn) {
                throw new Error("Missing required property 'keyVaultSecretName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.resourceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["certificateName"] = args?.certificateName;
            resourceInputs["keyVaultBaseUri"] = args?.keyVaultBaseUri;
            resourceInputs["keyVaultSecretName"] = args?.keyVaultSecretName;
            resourceInputs["keyVaultSecretVersion"] = args?.keyVaultSecretVersion;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceName"] = args?.resourceName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["keyVaultBaseUri"] = undefined /*out*/;
            resourceInputs["keyVaultSecretName"] = undefined /*out*/;
            resourceInputs["keyVaultSecretVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:webpubsub/v20220801preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20230201:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20230301preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20230601preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20230801preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20240101preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20240301:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20240401preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20240801preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20241001preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20250101preview:WebPubSubCustomCertificate" }, { type: "azure-native:webpubsub/v20250801preview:WebPubSubCustomCertificate" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebPubSubCustomCertificate.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebPubSubCustomCertificate = WebPubSubCustomCertificate;
