// @pulumi/azure-native/signalrservice/signalR.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalR = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * A class represent a resource.
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-08-01-preview, 2024-10-01-preview, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native signalrservice [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SignalR extends pulumi.CustomResource {
    /**
     * Get an existing SignalR resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SignalR(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:signalrservice:SignalR';
    /**
     * Returns true if the given object is an instance of SignalR.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SignalR.__pulumiType;
    }
    /**
     * Create a SignalR resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["cors"] = args?.cors;
            resourceInputs["disableAadAuth"] = (args?.disableAadAuth) ?? false;
            resourceInputs["disableLocalAuth"] = (args?.disableLocalAuth) ?? false;
            resourceInputs["features"] = args?.features;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["kind"] = args?.kind;
            resourceInputs["liveTraceConfiguration"] = args ? (args.liveTraceConfiguration ? pulumi.output(args.liveTraceConfiguration).apply(inputs.signalrservice.liveTraceConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["location"] = args?.location;
            resourceInputs["networkACLs"] = args?.networkACLs;
            resourceInputs["publicNetworkAccess"] = (args?.publicNetworkAccess) ?? "Enabled";
            resourceInputs["regionEndpointEnabled"] = (args?.regionEndpointEnabled) ?? "Enabled";
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceLogConfiguration"] = args?.resourceLogConfiguration;
            resourceInputs["resourceName"] = args?.resourceName;
            resourceInputs["resourceStopped"] = (args?.resourceStopped) ?? "false";
            resourceInputs["serverless"] = args ? (args.serverless ? pulumi.output(args.serverless).apply(inputs.signalrservice.serverlessSettingsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["sku"] = args?.sku;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tls"] = args ? (args.tls ? pulumi.output(args.tls).apply(inputs.signalrservice.signalRTlsSettingsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["upstream"] = args?.upstream;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["externalIP"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["hostNamePrefix"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicPort"] = undefined /*out*/;
            resourceInputs["serverPort"] = undefined /*out*/;
            resourceInputs["sharedPrivateLinkResources"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cors"] = undefined /*out*/;
            resourceInputs["disableAadAuth"] = undefined /*out*/;
            resourceInputs["disableLocalAuth"] = undefined /*out*/;
            resourceInputs["externalIP"] = undefined /*out*/;
            resourceInputs["features"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["hostNamePrefix"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["liveTraceConfiguration"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkACLs"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicNetworkAccess"] = undefined /*out*/;
            resourceInputs["publicPort"] = undefined /*out*/;
            resourceInputs["regionEndpointEnabled"] = undefined /*out*/;
            resourceInputs["resourceLogConfiguration"] = undefined /*out*/;
            resourceInputs["resourceStopped"] = undefined /*out*/;
            resourceInputs["serverPort"] = undefined /*out*/;
            resourceInputs["serverless"] = undefined /*out*/;
            resourceInputs["sharedPrivateLinkResources"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["tls"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upstream"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:signalrservice/v20180301preview:SignalR" }, { type: "azure-native:signalrservice/v20181001:SignalR" }, { type: "azure-native:signalrservice/v20200501:SignalR" }, { type: "azure-native:signalrservice/v20200701preview:SignalR" }, { type: "azure-native:signalrservice/v20210401preview:SignalR" }, { type: "azure-native:signalrservice/v20210601preview:SignalR" }, { type: "azure-native:signalrservice/v20210901preview:SignalR" }, { type: "azure-native:signalrservice/v20211001:SignalR" }, { type: "azure-native:signalrservice/v20220201:SignalR" }, { type: "azure-native:signalrservice/v20220801preview:SignalR" }, { type: "azure-native:signalrservice/v20230201:SignalR" }, { type: "azure-native:signalrservice/v20230301preview:SignalR" }, { type: "azure-native:signalrservice/v20230601preview:SignalR" }, { type: "azure-native:signalrservice/v20230801preview:SignalR" }, { type: "azure-native:signalrservice/v20240101preview:SignalR" }, { type: "azure-native:signalrservice/v20240301:SignalR" }, { type: "azure-native:signalrservice/v20240401preview:SignalR" }, { type: "azure-native:signalrservice/v20240801preview:SignalR" }, { type: "azure-native:signalrservice/v20241001preview:SignalR" }, { type: "azure-native:signalrservice/v20250101preview:SignalR" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SignalR.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SignalR = SignalR;
