// @pulumi/azure-native/marketplace/privateStoreCollection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateStoreCollection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The Collection data structure.
 *
 * Uses Azure REST API version 2023-01-01. In version 2.x of the Azure Native provider, it used API version 2023-01-01.
 *
 * Other available API versions: 2025-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native marketplace [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateStoreCollection extends pulumi.CustomResource {
    /**
     * Get an existing PrivateStoreCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateStoreCollection(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:marketplace:PrivateStoreCollection';
    /**
     * Returns true if the given object is an instance of PrivateStoreCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateStoreCollection.__pulumiType;
    }
    /**
     * Create a PrivateStoreCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.privateStoreId === undefined && !opts.urn) {
                throw new Error("Missing required property 'privateStoreId'");
            }
            resourceInputs["allSubscriptions"] = args?.allSubscriptions;
            resourceInputs["claim"] = args?.claim;
            resourceInputs["collectionId"] = args?.collectionId;
            resourceInputs["collectionName"] = args?.collectionName;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["privateStoreId"] = args?.privateStoreId;
            resourceInputs["subscriptionsList"] = args?.subscriptionsList;
            resourceInputs["appliedRules"] = undefined /*out*/;
            resourceInputs["approveAllItems"] = undefined /*out*/;
            resourceInputs["approveAllItemsModifiedAt"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfOffers"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allSubscriptions"] = undefined /*out*/;
            resourceInputs["appliedRules"] = undefined /*out*/;
            resourceInputs["approveAllItems"] = undefined /*out*/;
            resourceInputs["approveAllItemsModifiedAt"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["claim"] = undefined /*out*/;
            resourceInputs["collectionId"] = undefined /*out*/;
            resourceInputs["collectionName"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfOffers"] = undefined /*out*/;
            resourceInputs["subscriptionsList"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:marketplace/v20210601:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20211201:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20220301:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20220901:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20230101:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20250101:PrivateStoreCollection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateStoreCollection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateStoreCollection = PrivateStoreCollection;
