// @pulumi/azure-native/marketplace/privateStoreCollectionOffer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateStoreCollectionOffer = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The privateStore offer data structure.
 *
 * Uses Azure REST API version 2023-01-01. In version 2.x of the Azure Native provider, it used API version 2023-01-01.
 *
 * Other available API versions: 2025-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native marketplace [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateStoreCollectionOffer extends pulumi.CustomResource {
    /**
     * Get an existing PrivateStoreCollectionOffer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateStoreCollectionOffer(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:marketplace:PrivateStoreCollectionOffer';
    /**
     * Returns true if the given object is an instance of PrivateStoreCollectionOffer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateStoreCollectionOffer.__pulumiType;
    }
    /**
     * Create a PrivateStoreCollectionOffer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.collectionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'collectionId'");
            }
            if (args?.privateStoreId === undefined && !opts.urn) {
                throw new Error("Missing required property 'privateStoreId'");
            }
            resourceInputs["collectionId"] = args?.collectionId;
            resourceInputs["eTag"] = args?.eTag;
            resourceInputs["iconFileUris"] = args?.iconFileUris;
            resourceInputs["offerId"] = args?.offerId;
            resourceInputs["plans"] = args?.plans;
            resourceInputs["privateStoreId"] = args?.privateStoreId;
            resourceInputs["specificPlanIdsLimitation"] = args?.specificPlanIdsLimitation;
            resourceInputs["updateSuppressedDueIdempotence"] = args?.updateSuppressedDueIdempotence;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offerDisplayName"] = undefined /*out*/;
            resourceInputs["publisherDisplayName"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueOfferId"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["iconFileUris"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offerDisplayName"] = undefined /*out*/;
            resourceInputs["plans"] = undefined /*out*/;
            resourceInputs["privateStoreId"] = undefined /*out*/;
            resourceInputs["publisherDisplayName"] = undefined /*out*/;
            resourceInputs["specificPlanIdsLimitation"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueOfferId"] = undefined /*out*/;
            resourceInputs["updateSuppressedDueIdempotence"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:marketplace/v20210601:PrivateStoreCollectionOffer" }, { type: "azure-native:marketplace/v20211201:PrivateStoreCollectionOffer" }, { type: "azure-native:marketplace/v20220301:PrivateStoreCollectionOffer" }, { type: "azure-native:marketplace/v20220901:PrivateStoreCollectionOffer" }, { type: "azure-native:marketplace/v20230101:PrivateStoreCollectionOffer" }, { type: "azure-native:marketplace/v20250101:PrivateStoreCollectionOffer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateStoreCollectionOffer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateStoreCollectionOffer = PrivateStoreCollectionOffer;
