// @pulumi/azure-native/computelimit/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedLimit = exports.GuestSubscription = exports.getSharedLimitOutput = exports.getSharedLimit = exports.getGuestSubscriptionOutput = exports.getGuestSubscription = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getGuestSubscription = null;
exports.getGuestSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getGuestSubscription", "getGuestSubscriptionOutput"], () => require("./getGuestSubscription"));
exports.getSharedLimit = null;
exports.getSharedLimitOutput = null;
utilities.lazyLoad(exports, ["getSharedLimit", "getSharedLimitOutput"], () => require("./getSharedLimit"));
exports.GuestSubscription = null;
utilities.lazyLoad(exports, ["GuestSubscription"], () => require("./guestSubscription"));
exports.SharedLimit = null;
utilities.lazyLoad(exports, ["SharedLimit"], () => require("./sharedLimit"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:computelimit:GuestSubscription":
                return new exports.GuestSubscription(name, undefined, { urn });
            case "azure-native:computelimit:SharedLimit":
                return new exports.SharedLimit(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "computelimit", _module);
