// @pulumi/aws/grafana/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceServiceAccountToken = exports.WorkspaceServiceAccount = exports.WorkspaceSamlConfiguration = exports.WorkspaceApiKey = exports.Workspace = exports.RoleAssociation = exports.LicenseAssociation = exports.getWorkspaceOutput = exports.getWorkspace = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.LicenseAssociation = null;
utilities.lazyLoad(exports, ["LicenseAssociation"], () => require("./licenseAssociation"));
exports.RoleAssociation = null;
utilities.lazyLoad(exports, ["RoleAssociation"], () => require("./roleAssociation"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspaceApiKey = null;
utilities.lazyLoad(exports, ["WorkspaceApiKey"], () => require("./workspaceApiKey"));
exports.WorkspaceSamlConfiguration = null;
utilities.lazyLoad(exports, ["WorkspaceSamlConfiguration"], () => require("./workspaceSamlConfiguration"));
exports.WorkspaceServiceAccount = null;
utilities.lazyLoad(exports, ["WorkspaceServiceAccount"], () => require("./workspaceServiceAccount"));
exports.WorkspaceServiceAccountToken = null;
utilities.lazyLoad(exports, ["WorkspaceServiceAccountToken"], () => require("./workspaceServiceAccountToken"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:grafana/licenseAssociation:LicenseAssociation":
                return new exports.LicenseAssociation(name, undefined, { urn });
            case "aws:grafana/roleAssociation:RoleAssociation":
                return new exports.RoleAssociation(name, undefined, { urn });
            case "aws:grafana/workspace:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "aws:grafana/workspaceApiKey:WorkspaceApiKey":
                return new exports.WorkspaceApiKey(name, undefined, { urn });
            case "aws:grafana/workspaceSamlConfiguration:WorkspaceSamlConfiguration":
                return new exports.WorkspaceSamlConfiguration(name, undefined, { urn });
            case "aws:grafana/workspaceServiceAccount:WorkspaceServiceAccount":
                return new exports.WorkspaceServiceAccount(name, undefined, { urn });
            case "aws:grafana/workspaceServiceAccountToken:WorkspaceServiceAccountToken":
                return new exports.WorkspaceServiceAccountToken(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "grafana/licenseAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/roleAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/workspace", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/workspaceApiKey", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/workspaceSamlConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/workspaceServiceAccount", _module);
pulumi.runtime.registerResourceModule("aws", "grafana/workspaceServiceAccountToken", _module);
