// @pulumi/gcp/chronicle/dataTable.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataTable = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a Chronicle Data Table, a multicolumn structure used to ingest your own data into Google SecOps.
 *
 * To get more information about DataTable, see:
 *
 * * [API documentation](https://cloud.google.com/chronicle/docs/reference/rest/v1/projects.locations.instances.dataTables)
 * * How-to Guides
 *     * [Google SecOps Guides](https://cloud.google.com/chronicle/docs/secops/secops-overview)
 *
 * ## Example Usage
 *
 * ### Chronicle Data Table Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.chronicle.DataTable("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     dataTableId: "terraform_test",
 *     description: "sample desc",
 *     columnInfos: [
 *         {
 *             columnIndex: 0,
 *             originalColumn: "username",
 *             columnType: "STRING",
 *         },
 *         {
 *             columnIndex: 1,
 *             originalColumn: "ip_address",
 *             columnType: "CIDR",
 *         },
 *     ],
 * });
 * export const dataTableName = example.name;
 * export const dataTableId = example.dataTableId;
 * export const dataTableUuid = example.dataTableUuid;
 * export const dataTableDescription = example.description;
 * export const dataTableCreateTime = example.createTime;
 * ```
 * ### Chronicle Data Table With Optional Fields
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const testScopeAllowEveryone = new gcp.chronicle.DataAccessScope("test_scope_allow_everyone", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     dataAccessScopeId: "tf-scope-opt",
 *     description: "scope-description",
 *     allowedDataAccessLabels: [{
 *         logType: "GCP_CLOUDAUDIT",
 *     }],
 * });
 * const exampleDt = new gcp.chronicle.DataTable("example_dt", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     dataTableId: "tf_test_full",
 *     description: "Comprehensive test table with all teh fields",
 *     rowTimeToLive: "48h",
 *     columnInfos: [
 *         {
 *             columnIndex: 0,
 *             originalColumn: "username",
 *             keyColumn: true,
 *             mappedColumnPath: "entity.user.userid",
 *             repeatedValues: false,
 *         },
 *         {
 *             columnIndex: 1,
 *             originalColumn: "ip_address",
 *             columnType: "CIDR",
 *             keyColumn: false,
 *             repeatedValues: false,
 *         },
 *     ],
 *     scopeInfo: {
 *         dataAccessScopes: [testScopeAllowEveryone.name],
 *     },
 * }, {
 *     dependsOn: [testScopeAllowEveryone],
 * });
 * ```
 *
 * ## Import
 *
 * DataTable can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/instances/{{instance}}/dataTables/{{data_table_id}}`
 * * `{{project}}/{{location}}/{{instance}}/{{data_table_id}}`
 * * `{{location}}/{{instance}}/{{data_table_id}}`
 *
 * When using the `pulumi import` command, DataTable can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:chronicle/dataTable:DataTable default projects/{{project}}/locations/{{location}}/instances/{{instance}}/dataTables/{{data_table_id}}
 * $ pulumi import gcp:chronicle/dataTable:DataTable default {{project}}/{{location}}/{{instance}}/{{data_table_id}}
 * $ pulumi import gcp:chronicle/dataTable:DataTable default {{location}}/{{instance}}/{{data_table_id}}
 * ```
 */
class DataTable extends pulumi.CustomResource {
    /**
     * Get an existing DataTable resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DataTable(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:chronicle/dataTable:DataTable';
    /**
     * Returns true if the given object is an instance of DataTable.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataTable.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["approximateRowCount"] = state?.approximateRowCount;
            resourceInputs["columnInfos"] = state?.columnInfos;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["dataTableId"] = state?.dataTableId;
            resourceInputs["dataTableUuid"] = state?.dataTableUuid;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["instance"] = state?.instance;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["rowTimeToLive"] = state?.rowTimeToLive;
            resourceInputs["rowTimeToLiveUpdateTime"] = state?.rowTimeToLiveUpdateTime;
            resourceInputs["ruleAssociationsCount"] = state?.ruleAssociationsCount;
            resourceInputs["rules"] = state?.rules;
            resourceInputs["scopeInfo"] = state?.scopeInfo;
            resourceInputs["updateSource"] = state?.updateSource;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.dataTableId === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataTableId'");
            }
            if (args?.description === undefined && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if (args?.instance === undefined && !opts.urn) {
                throw new Error("Missing required property 'instance'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["columnInfos"] = args?.columnInfos;
            resourceInputs["dataTableId"] = args?.dataTableId;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["instance"] = args?.instance;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["rowTimeToLive"] = args?.rowTimeToLive;
            resourceInputs["scopeInfo"] = args?.scopeInfo;
            resourceInputs["approximateRowCount"] = undefined /*out*/;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["dataTableUuid"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rowTimeToLiveUpdateTime"] = undefined /*out*/;
            resourceInputs["ruleAssociationsCount"] = undefined /*out*/;
            resourceInputs["rules"] = undefined /*out*/;
            resourceInputs["updateSource"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataTable.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataTable = DataTable;
