// @pulumi/gcp/chronicle/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = exports.RuleDeployment = exports.Rule = exports.Retrohunt = exports.ReferenceList = exports.NativeDashboard = exports.Feed = exports.DataTableRow = exports.DataTable = exports.DataAccessScope = exports.DataAccessLabel = exports.DashboardChart = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.DashboardChart = null;
utilities.lazyLoad(exports, ["DashboardChart"], () => require("./dashboardChart"));
exports.DataAccessLabel = null;
utilities.lazyLoad(exports, ["DataAccessLabel"], () => require("./dataAccessLabel"));
exports.DataAccessScope = null;
utilities.lazyLoad(exports, ["DataAccessScope"], () => require("./dataAccessScope"));
exports.DataTable = null;
utilities.lazyLoad(exports, ["DataTable"], () => require("./dataTable"));
exports.DataTableRow = null;
utilities.lazyLoad(exports, ["DataTableRow"], () => require("./dataTableRow"));
exports.Feed = null;
utilities.lazyLoad(exports, ["Feed"], () => require("./feed"));
exports.NativeDashboard = null;
utilities.lazyLoad(exports, ["NativeDashboard"], () => require("./nativeDashboard"));
exports.ReferenceList = null;
utilities.lazyLoad(exports, ["ReferenceList"], () => require("./referenceList"));
exports.Retrohunt = null;
utilities.lazyLoad(exports, ["Retrohunt"], () => require("./retrohunt"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
exports.RuleDeployment = null;
utilities.lazyLoad(exports, ["RuleDeployment"], () => require("./ruleDeployment"));
exports.Watchlist = null;
utilities.lazyLoad(exports, ["Watchlist"], () => require("./watchlist"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:chronicle/dashboardChart:DashboardChart":
                return new exports.DashboardChart(name, undefined, { urn });
            case "gcp:chronicle/dataAccessLabel:DataAccessLabel":
                return new exports.DataAccessLabel(name, undefined, { urn });
            case "gcp:chronicle/dataAccessScope:DataAccessScope":
                return new exports.DataAccessScope(name, undefined, { urn });
            case "gcp:chronicle/dataTable:DataTable":
                return new exports.DataTable(name, undefined, { urn });
            case "gcp:chronicle/dataTableRow:DataTableRow":
                return new exports.DataTableRow(name, undefined, { urn });
            case "gcp:chronicle/feed:Feed":
                return new exports.Feed(name, undefined, { urn });
            case "gcp:chronicle/nativeDashboard:NativeDashboard":
                return new exports.NativeDashboard(name, undefined, { urn });
            case "gcp:chronicle/referenceList:ReferenceList":
                return new exports.ReferenceList(name, undefined, { urn });
            case "gcp:chronicle/retrohunt:Retrohunt":
                return new exports.Retrohunt(name, undefined, { urn });
            case "gcp:chronicle/rule:Rule":
                return new exports.Rule(name, undefined, { urn });
            case "gcp:chronicle/ruleDeployment:RuleDeployment":
                return new exports.RuleDeployment(name, undefined, { urn });
            case "gcp:chronicle/watchlist:Watchlist":
                return new exports.Watchlist(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "chronicle/dashboardChart", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/dataAccessLabel", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/dataAccessScope", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/dataTable", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/dataTableRow", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/feed", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/nativeDashboard", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/referenceList", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/retrohunt", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/rule", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/ruleDeployment", _module);
pulumi.runtime.registerResourceModule("gcp", "chronicle/watchlist", _module);
