// @pulumi/gcp/chronicle/ruleDeployment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleDeployment = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The RuleDeployment resource represents the deployment state of a Rule.
 *
 * To get more information about RuleDeployment, see:
 *
 * * [API documentation](https://cloud.google.com/chronicle/docs/reference/rest/v1/RuleDeployment)
 * * How-to Guides
 *     * [Google SecOps Guides](https://cloud.google.com/chronicle/docs/secops/secops-overview)
 *
 * ## Example Usage
 *
 * ### Chronicle Ruledeployment Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const my_rule = new gcp.chronicle.Rule("my-rule", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     text: "rule test_rule { meta: events:  $userid = $e.principal.user.userid  match: $userid over 10m condition: $e }\n",
 * });
 * const example = new gcp.chronicle.RuleDeployment("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     rule: pulumi.all([std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }), std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }).then(invoke => invoke.result).length]).apply(([invoke, length]) => invoke.result[length - 1]).apply(x =>String(x)),
 *     enabled: true,
 *     alerting: true,
 *     archived: false,
 *     runFrequency: "DAILY",
 * });
 * ```
 * ### Chronicle Ruledeployment Disabled
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const my_rule = new gcp.chronicle.Rule("my-rule", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     text: "rule test_rule { meta: events:  $userid = $e.principal.user.userid  match: $userid over 10m condition: $e }\n",
 * });
 * const example = new gcp.chronicle.RuleDeployment("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     rule: pulumi.all([std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }), std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }).then(invoke => invoke.result).length]).apply(([invoke, length]) => invoke.result[length - 1]).apply(x =>String(x)),
 *     enabled: false,
 *     runFrequency: "LIVE",
 * });
 * ```
 * ### Chronicle Ruledeployment Run Frequency Missing
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const my_rule = new gcp.chronicle.Rule("my-rule", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     text: "rule test_rule { meta: events:  $userid = $e.principal.user.userid  match: $userid over 10m condition: $e }\n",
 * });
 * const example = new gcp.chronicle.RuleDeployment("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     rule: pulumi.all([std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }), std.split({
 *         separator: "/",
 *         text: googleChronicleRule["my-rule"].name,
 *     }).then(invoke => invoke.result).length]).apply(([invoke, length]) => invoke.result[length - 1]).apply(x =>String(x)),
 *     enabled: true,
 *     alerting: true,
 *     archived: false,
 * });
 * ```
 *
 * ## Import
 *
 * RuleDeployment can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/instances/{{instance}}/rules/{{rule}}/deployment`
 * * `{{project}}/{{location}}/{{instance}}/{{rule}}`
 * * `{{location}}/{{instance}}/{{rule}}`
 *
 * When using the `pulumi import` command, RuleDeployment can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:chronicle/ruleDeployment:RuleDeployment default projects/{{project}}/locations/{{location}}/instances/{{instance}}/rules/{{rule}}/deployment
 * $ pulumi import gcp:chronicle/ruleDeployment:RuleDeployment default {{project}}/{{location}}/{{instance}}/{{rule}}
 * $ pulumi import gcp:chronicle/ruleDeployment:RuleDeployment default {{location}}/{{instance}}/{{rule}}
 * ```
 */
class RuleDeployment extends pulumi.CustomResource {
    /**
     * Get an existing RuleDeployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RuleDeployment(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:chronicle/ruleDeployment:RuleDeployment';
    /**
     * Returns true if the given object is an instance of RuleDeployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RuleDeployment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["alerting"] = state?.alerting;
            resourceInputs["archiveTime"] = state?.archiveTime;
            resourceInputs["archived"] = state?.archived;
            resourceInputs["consumerRules"] = state?.consumerRules;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["executionState"] = state?.executionState;
            resourceInputs["instance"] = state?.instance;
            resourceInputs["lastAlertStatusChangeTime"] = state?.lastAlertStatusChangeTime;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["producerRules"] = state?.producerRules;
            resourceInputs["project"] = state?.project;
            resourceInputs["rule"] = state?.rule;
            resourceInputs["runFrequency"] = state?.runFrequency;
        }
        else {
            const args = argsOrState;
            if (args?.instance === undefined && !opts.urn) {
                throw new Error("Missing required property 'instance'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.rule === undefined && !opts.urn) {
                throw new Error("Missing required property 'rule'");
            }
            resourceInputs["alerting"] = args?.alerting;
            resourceInputs["archived"] = args?.archived;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["instance"] = args?.instance;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["rule"] = args?.rule;
            resourceInputs["runFrequency"] = args?.runFrequency;
            resourceInputs["archiveTime"] = undefined /*out*/;
            resourceInputs["consumerRules"] = undefined /*out*/;
            resourceInputs["executionState"] = undefined /*out*/;
            resourceInputs["lastAlertStatusChangeTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["producerRules"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RuleDeployment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RuleDeployment = RuleDeployment;
