// @pulumi/gcp/chronicle/watchlist.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A watchlist is a list of entities that allows for bulk operations over the included entities.
 *
 * To get more information about Watchlist, see:
 *
 * * [API documentation](https://cloud.google.com/chronicle/docs/reference/rest/v1/projects.locations.instances.watchlists)
 * * How-to Guides
 *     * [Google SecOps Guides](https://cloud.google.com/chronicle/docs/secops/secops-overview)
 *
 * ## Example Usage
 *
 * ### Chronicle Watchlist Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.chronicle.Watchlist("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     watchlistId: "watchlist-id",
 *     description: "watchlist-description",
 *     displayName: "watchlist_name",
 *     multiplyingFactor: 1,
 *     entityPopulationMechanism: {
 *         manual: {},
 *     },
 *     watchlistUserPreferences: {
 *         pinned: true,
 *     },
 * });
 * ```
 * ### Chronicle Watchlist Without Id
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.chronicle.Watchlist("example", {
 *     location: "us",
 *     instance: "00000000-0000-0000-0000-000000000000",
 *     description: "watchlist-description",
 *     displayName: "watchlist-name",
 *     multiplyingFactor: 1,
 *     entityPopulationMechanism: {
 *         manual: {},
 *     },
 *     watchlistUserPreferences: {
 *         pinned: true,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Watchlist can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists/{{watchlist_id}}`
 * * `{{project}}/{{location}}/{{instance}}/{{watchlist_id}}`
 * * `{{location}}/{{instance}}/{{watchlist_id}}`
 *
 * When using the `pulumi import` command, Watchlist can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:chronicle/watchlist:Watchlist default projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists/{{watchlist_id}}
 * $ pulumi import gcp:chronicle/watchlist:Watchlist default {{project}}/{{location}}/{{instance}}/{{watchlist_id}}
 * $ pulumi import gcp:chronicle/watchlist:Watchlist default {{location}}/{{instance}}/{{watchlist_id}}
 * ```
 */
class Watchlist extends pulumi.CustomResource {
    /**
     * Get an existing Watchlist resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Watchlist(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:chronicle/watchlist:Watchlist';
    /**
     * Returns true if the given object is an instance of Watchlist.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Watchlist.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["entityCounts"] = state?.entityCounts;
            resourceInputs["entityPopulationMechanism"] = state?.entityPopulationMechanism;
            resourceInputs["instance"] = state?.instance;
            resourceInputs["location"] = state?.location;
            resourceInputs["multiplyingFactor"] = state?.multiplyingFactor;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["updateTime"] = state?.updateTime;
            resourceInputs["watchlistId"] = state?.watchlistId;
            resourceInputs["watchlistUserPreferences"] = state?.watchlistUserPreferences;
        }
        else {
            const args = argsOrState;
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.entityPopulationMechanism === undefined && !opts.urn) {
                throw new Error("Missing required property 'entityPopulationMechanism'");
            }
            if (args?.instance === undefined && !opts.urn) {
                throw new Error("Missing required property 'instance'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["entityPopulationMechanism"] = args?.entityPopulationMechanism;
            resourceInputs["instance"] = args?.instance;
            resourceInputs["location"] = args?.location;
            resourceInputs["multiplyingFactor"] = args?.multiplyingFactor;
            resourceInputs["project"] = args?.project;
            resourceInputs["watchlistId"] = args?.watchlistId;
            resourceInputs["watchlistUserPreferences"] = args?.watchlistUserPreferences;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["entityCounts"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Watchlist.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Watchlist = Watchlist;
