// @pulumi/aws/observabilityadmin/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TelemetryRule = exports.TelemetryPipeline = exports.TelemetryEvaluationForOrganization = exports.TelemetryEvaluation = exports.TelemetryEnrichment = exports.CentralizationRuleForOrganization = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CentralizationRuleForOrganization = null;
utilities.lazyLoad(exports, ["CentralizationRuleForOrganization"], () => require("./centralizationRuleForOrganization"));
exports.TelemetryEnrichment = null;
utilities.lazyLoad(exports, ["TelemetryEnrichment"], () => require("./telemetryEnrichment"));
exports.TelemetryEvaluation = null;
utilities.lazyLoad(exports, ["TelemetryEvaluation"], () => require("./telemetryEvaluation"));
exports.TelemetryEvaluationForOrganization = null;
utilities.lazyLoad(exports, ["TelemetryEvaluationForOrganization"], () => require("./telemetryEvaluationForOrganization"));
exports.TelemetryPipeline = null;
utilities.lazyLoad(exports, ["TelemetryPipeline"], () => require("./telemetryPipeline"));
exports.TelemetryRule = null;
utilities.lazyLoad(exports, ["TelemetryRule"], () => require("./telemetryRule"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:observabilityadmin/centralizationRuleForOrganization:CentralizationRuleForOrganization":
                return new exports.CentralizationRuleForOrganization(name, undefined, { urn });
            case "aws:observabilityadmin/telemetryEnrichment:TelemetryEnrichment":
                return new exports.TelemetryEnrichment(name, undefined, { urn });
            case "aws:observabilityadmin/telemetryEvaluation:TelemetryEvaluation":
                return new exports.TelemetryEvaluation(name, undefined, { urn });
            case "aws:observabilityadmin/telemetryEvaluationForOrganization:TelemetryEvaluationForOrganization":
                return new exports.TelemetryEvaluationForOrganization(name, undefined, { urn });
            case "aws:observabilityadmin/telemetryPipeline:TelemetryPipeline":
                return new exports.TelemetryPipeline(name, undefined, { urn });
            case "aws:observabilityadmin/telemetryRule:TelemetryRule":
                return new exports.TelemetryRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/centralizationRuleForOrganization", _module);
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/telemetryEnrichment", _module);
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/telemetryEvaluation", _module);
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/telemetryEvaluationForOrganization", _module);
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/telemetryPipeline", _module);
pulumi.runtime.registerResourceModule("aws", "observabilityadmin/telemetryRule", _module);
