// @pulumi/aws/observabilityadmin/telemetryEvaluationForOrganization.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TelemetryEvaluationForOrganization = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS CloudWatch Observability Admin Telemetry Evaluation For Organization.
 *
 * Telemetry evaluation for organization begins onboarding the organization and all member accounts to the telemetry config feature, enabling CloudWatch to evaluate and discover the state of telemetry configuration for AWS resources across the organization.
 *
 * For more information, see the [AWS CloudWatch Observability Admin documentation](https://docs.aws.amazon.com/cloudwatch/latest/observabilityadmin/what-is-observabilityadmin.html).
 *
 * > **NOTE:** This resource can only be used from the management account or a delegated admin account of an AWS Organization. Only one telemetry evaluation for organization can exist per organization. Creating this resource enables the feature; destroying it disables the feature.
 *
 * ## Example Usage
 *
 * ### Basic Usage (Current Region Only)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {});
 * ```
 *
 * ### All Regions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {allRegions: true});
 * ```
 *
 * ### Specific Regions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {regions: [
 *     "us-east-1",
 *     "eu-west-1",
 * ]});
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import CloudWatch Observability Admin Telemetry Evaluation For Organization using the region name. For example:
 *
 * ```sh
 * $ pulumi import aws:observabilityadmin/telemetryEvaluationForOrganization:TelemetryEvaluationForOrganization example us-west-2
 * ```
 */
class TelemetryEvaluationForOrganization extends pulumi.CustomResource {
    /**
     * Get an existing TelemetryEvaluationForOrganization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TelemetryEvaluationForOrganization(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:observabilityadmin/telemetryEvaluationForOrganization:TelemetryEvaluationForOrganization';
    /**
     * Returns true if the given object is an instance of TelemetryEvaluationForOrganization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TelemetryEvaluationForOrganization.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allRegions"] = state?.allRegions;
            resourceInputs["failureReason"] = state?.failureReason;
            resourceInputs["homeRegion"] = state?.homeRegion;
            resourceInputs["region"] = state?.region;
            resourceInputs["regions"] = state?.regions;
            resourceInputs["status"] = state?.status;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            resourceInputs["allRegions"] = args?.allRegions;
            resourceInputs["region"] = args?.region;
            resourceInputs["regions"] = args?.regions;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["homeRegion"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TelemetryEvaluationForOrganization.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TelemetryEvaluationForOrganization = TelemetryEvaluationForOrganization;
