// @pulumi/kubernetes/apps/v1beta1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatefulSetPatch = exports.StatefulSetList = exports.StatefulSet = exports.DeploymentPatch = exports.DeploymentList = exports.Deployment = exports.ControllerRevisionPatch = exports.ControllerRevisionList = exports.ControllerRevision = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.ControllerRevision = null;
utilities.lazyLoad(exports, ["ControllerRevision"], () => require("./controllerRevision"));
exports.ControllerRevisionList = null;
utilities.lazyLoad(exports, ["ControllerRevisionList"], () => require("./controllerRevisionList"));
exports.ControllerRevisionPatch = null;
utilities.lazyLoad(exports, ["ControllerRevisionPatch"], () => require("./controllerRevisionPatch"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.DeploymentList = null;
utilities.lazyLoad(exports, ["DeploymentList"], () => require("./deploymentList"));
exports.DeploymentPatch = null;
utilities.lazyLoad(exports, ["DeploymentPatch"], () => require("./deploymentPatch"));
exports.StatefulSet = null;
utilities.lazyLoad(exports, ["StatefulSet"], () => require("./statefulSet"));
exports.StatefulSetList = null;
utilities.lazyLoad(exports, ["StatefulSetList"], () => require("./statefulSetList"));
exports.StatefulSetPatch = null;
utilities.lazyLoad(exports, ["StatefulSetPatch"], () => require("./statefulSetPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:apps/v1beta1:ControllerRevision":
                return new exports.ControllerRevision(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:ControllerRevisionList":
                return new exports.ControllerRevisionList(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:ControllerRevisionPatch":
                return new exports.ControllerRevisionPatch(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:DeploymentList":
                return new exports.DeploymentList(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:DeploymentPatch":
                return new exports.DeploymentPatch(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:StatefulSet":
                return new exports.StatefulSet(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:StatefulSetList":
                return new exports.StatefulSetList(name, undefined, { urn });
            case "kubernetes:apps/v1beta1:StatefulSetPatch":
                return new exports.StatefulSetPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "apps/v1beta1", _module);
