// @pulumi/azure-native/deviceprovisioningservices/dpsCertificate.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DpsCertificate = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The X509 Certificate.
 *
 * Uses Azure REST API version 2023-03-01-preview.
 *
 * Other available API versions: 2017-08-21-preview, 2017-11-15, 2018-01-22, 2020-01-01, 2020-03-01, 2020-09-01-preview, 2021-10-15, 2022-02-05, 2022-12-12, 2025-02-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native deviceprovisioningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DpsCertificate extends pulumi.CustomResource {
    /**
     * Get an existing DpsCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DpsCertificate(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:deviceprovisioningservices:DpsCertificate';
    /**
     * Returns true if the given object is an instance of DpsCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DpsCertificate.__pulumiType;
    }
    /**
     * Create a DpsCertificate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.provisioningServiceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'provisioningServiceName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["certificateName"] = args?.certificateName;
            resourceInputs["properties"] = args?.properties;
            resourceInputs["provisioningServiceName"] = args?.provisioningServiceName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deviceprovisioningservices/v20170821preview:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20171115:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20180122:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20200101:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20200301:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20200901preview:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20211015:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20220205:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20221212:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20230301preview:DpsCertificate" }, { type: "azure-native:deviceprovisioningservices/v20250201preview:DpsCertificate" }, { type: "azure-native:devices/v20211015:DpsCertificate" }, { type: "azure-native:devices/v20221212:DpsCertificate" }, { type: "azure-native:devices/v20230301preview:DpsCertificate" }, { type: "azure-native:devices/v20250201preview:DpsCertificate" }, { type: "azure-native:devices:DpsCertificate" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DpsCertificate.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DpsCertificate = DpsCertificate;
