// @pulumi/gcp/networkservices/httpRoute.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpRoute = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * HttpRoute is the resource defining how HTTP traffic should be routed by a Mesh or Gateway resource.
 *
 * To get more information about HttpRoute, see:
 *
 * * [API documentation](https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.httpRoutes)
 * * How-to Guides
 *     * [Setup HTTP Services](https://cloud.google.com/traffic-director/docs/set-up-envoy-http-mesh)
 *
 * ## Example Usage
 *
 * ### Network Services Http Route Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     name: "my-http-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         matches: [{
 *             queryParameters: [{
 *                 queryParameter: "key",
 *                 exactMatch: "value",
 *             }],
 *             fullPathMatch: "example",
 *         }],
 *     }],
 * });
 * ```
 * ### Network Services Http Route Matches And Actions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     name: "my-http-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         matches: [
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     regexMatch: "header-value",
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     exactMatch: "value",
 *                 }],
 *                 prefixMatch: "example",
 *                 ignoreCase: false,
 *             },
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     presentMatch: true,
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     regexMatch: "value",
 *                 }],
 *                 regexMatch: "example",
 *                 ignoreCase: false,
 *             },
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     presentMatch: true,
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     presentMatch: true,
 *                 }],
 *                 fullPathMatch: "example",
 *                 ignoreCase: false,
 *             },
 *         ],
 *         action: {
 *             redirect: {
 *                 hostRedirect: "new-host",
 *                 pathRedirect: "new-path",
 *                 prefixRewrite: "new-prefix",
 *                 httpsRedirect: true,
 *                 stripQuery: true,
 *                 portRedirect: 8081,
 *             },
 *             urlRewrite: {
 *                 pathPrefixRewrite: "new-prefix",
 *                 hostRewrite: "new-host",
 *             },
 *             retryPolicy: {
 *                 retryConditions: ["server_error"],
 *                 numRetries: 1,
 *                 perTryTimeout: "1s",
 *             },
 *             requestMirrorPolicy: {
 *                 destination: {
 *                     serviceName: "new",
 *                     weight: 1,
 *                 },
 *             },
 *             corsPolicy: {
 *                 allowOrigins: ["example"],
 *                 allowMethods: [
 *                     "GET",
 *                     "PUT",
 *                 ],
 *                 allowHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 exposeHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 maxAge: "1s",
 *                 allowCredentials: true,
 *                 disabled: false,
 *             },
 *         },
 *     }],
 * });
 * ```
 * ### Network Services Http Route Actions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     name: "my-http-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         action: {
 *             faultInjectionPolicy: {
 *                 delay: {
 *                     fixedDelay: "1s",
 *                     percentage: 1,
 *                 },
 *                 abort: {
 *                     httpStatus: 500,
 *                     percentage: 1,
 *                 },
 *             },
 *             urlRewrite: {
 *                 pathPrefixRewrite: "new-prefix",
 *                 hostRewrite: "new-host",
 *             },
 *             retryPolicy: {
 *                 retryConditions: ["server_error"],
 *                 numRetries: 1,
 *                 perTryTimeout: "1s",
 *             },
 *             requestMirrorPolicy: {
 *                 destination: {
 *                     serviceName: "new",
 *                     weight: 1,
 *                 },
 *             },
 *             corsPolicy: {
 *                 allowOrigins: ["example"],
 *                 allowMethods: [
 *                     "GET",
 *                     "PUT",
 *                 ],
 *                 allowHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 exposeHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 maxAge: "1s",
 *                 allowCredentials: true,
 *                 disabled: false,
 *             },
 *             requestHeaderModifier: {
 *                 set: {
 *                     version: "1",
 *                     type: "json",
 *                 },
 *                 add: {
 *                     "minor-version": "1",
 *                 },
 *                 removes: ["arg"],
 *             },
 *             responseHeaderModifier: {
 *                 set: {
 *                     version: "1",
 *                     type: "json",
 *                 },
 *                 add: {
 *                     "minor-version": "1",
 *                 },
 *                 removes: ["removearg"],
 *             },
 *         },
 *     }],
 * });
 * ```
 * ### Network Services Http Route Mesh Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.Mesh("default", {
 *     name: "my-http-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 * });
 * const defaultHttpRoute = new gcp.networkservices.HttpRoute("default", {
 *     name: "my-http-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     meshes: [_default.id],
 *     rules: [{
 *         matches: [{
 *             queryParameters: [{
 *                 queryParameter: "key",
 *                 exactMatch: "value",
 *             }],
 *             fullPathMatch: "example",
 *         }],
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * HttpRoute can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/global/httpRoutes/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, HttpRoute can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/httpRoute:HttpRoute default projects/{{project}}/locations/global/httpRoutes/{{name}}
 * $ pulumi import gcp:networkservices/httpRoute:HttpRoute default {{project}}/{{name}}
 * $ pulumi import gcp:networkservices/httpRoute:HttpRoute default {{name}}
 * ```
 */
class HttpRoute extends pulumi.CustomResource {
    /**
     * Get an existing HttpRoute resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HttpRoute(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/httpRoute:HttpRoute';
    /**
     * Returns true if the given object is an instance of HttpRoute.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HttpRoute.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["gateways"] = state?.gateways;
            resourceInputs["hostnames"] = state?.hostnames;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["meshes"] = state?.meshes;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["rules"] = state?.rules;
            resourceInputs["selfLink"] = state?.selfLink;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.hostnames === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostnames'");
            }
            if (args?.rules === undefined && !opts.urn) {
                throw new Error("Missing required property 'rules'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["gateways"] = args?.gateways;
            resourceInputs["hostnames"] = args?.hostnames;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["meshes"] = args?.meshes;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["rules"] = args?.rules;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(HttpRoute.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HttpRoute = HttpRoute;
