// @pulumi/gcp/networkservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WasmPlugin = exports.TlsRoute = exports.TcpRoute = exports.ServiceLbPolicies = exports.ServiceBinding = exports.MulticastProducerAssociation = exports.MulticastGroupRangeActivation = exports.MulticastGroupRange = exports.MulticastGroupProducerActivation = exports.MulticastGroupConsumerActivation = exports.MulticastDomainGroup = exports.MulticastDomainActivation = exports.MulticastDomain = exports.MulticastConsumerAssociation = exports.Mesh = exports.LbTrafficExtension = exports.LbRouteExtension = exports.LbEdgeExtension = exports.HttpRoute = exports.GrpcRoute = exports.Gateway = exports.EndpointPolicy = exports.EdgeCacheService = exports.EdgeCacheOrigin = exports.EdgeCacheKeyset = exports.AuthzExtension = exports.AgentGateway = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AgentGateway = null;
utilities.lazyLoad(exports, ["AgentGateway"], () => require("./agentGateway"));
exports.AuthzExtension = null;
utilities.lazyLoad(exports, ["AuthzExtension"], () => require("./authzExtension"));
exports.EdgeCacheKeyset = null;
utilities.lazyLoad(exports, ["EdgeCacheKeyset"], () => require("./edgeCacheKeyset"));
exports.EdgeCacheOrigin = null;
utilities.lazyLoad(exports, ["EdgeCacheOrigin"], () => require("./edgeCacheOrigin"));
exports.EdgeCacheService = null;
utilities.lazyLoad(exports, ["EdgeCacheService"], () => require("./edgeCacheService"));
exports.EndpointPolicy = null;
utilities.lazyLoad(exports, ["EndpointPolicy"], () => require("./endpointPolicy"));
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.GrpcRoute = null;
utilities.lazyLoad(exports, ["GrpcRoute"], () => require("./grpcRoute"));
exports.HttpRoute = null;
utilities.lazyLoad(exports, ["HttpRoute"], () => require("./httpRoute"));
exports.LbEdgeExtension = null;
utilities.lazyLoad(exports, ["LbEdgeExtension"], () => require("./lbEdgeExtension"));
exports.LbRouteExtension = null;
utilities.lazyLoad(exports, ["LbRouteExtension"], () => require("./lbRouteExtension"));
exports.LbTrafficExtension = null;
utilities.lazyLoad(exports, ["LbTrafficExtension"], () => require("./lbTrafficExtension"));
exports.Mesh = null;
utilities.lazyLoad(exports, ["Mesh"], () => require("./mesh"));
exports.MulticastConsumerAssociation = null;
utilities.lazyLoad(exports, ["MulticastConsumerAssociation"], () => require("./multicastConsumerAssociation"));
exports.MulticastDomain = null;
utilities.lazyLoad(exports, ["MulticastDomain"], () => require("./multicastDomain"));
exports.MulticastDomainActivation = null;
utilities.lazyLoad(exports, ["MulticastDomainActivation"], () => require("./multicastDomainActivation"));
exports.MulticastDomainGroup = null;
utilities.lazyLoad(exports, ["MulticastDomainGroup"], () => require("./multicastDomainGroup"));
exports.MulticastGroupConsumerActivation = null;
utilities.lazyLoad(exports, ["MulticastGroupConsumerActivation"], () => require("./multicastGroupConsumerActivation"));
exports.MulticastGroupProducerActivation = null;
utilities.lazyLoad(exports, ["MulticastGroupProducerActivation"], () => require("./multicastGroupProducerActivation"));
exports.MulticastGroupRange = null;
utilities.lazyLoad(exports, ["MulticastGroupRange"], () => require("./multicastGroupRange"));
exports.MulticastGroupRangeActivation = null;
utilities.lazyLoad(exports, ["MulticastGroupRangeActivation"], () => require("./multicastGroupRangeActivation"));
exports.MulticastProducerAssociation = null;
utilities.lazyLoad(exports, ["MulticastProducerAssociation"], () => require("./multicastProducerAssociation"));
exports.ServiceBinding = null;
utilities.lazyLoad(exports, ["ServiceBinding"], () => require("./serviceBinding"));
exports.ServiceLbPolicies = null;
utilities.lazyLoad(exports, ["ServiceLbPolicies"], () => require("./serviceLbPolicies"));
exports.TcpRoute = null;
utilities.lazyLoad(exports, ["TcpRoute"], () => require("./tcpRoute"));
exports.TlsRoute = null;
utilities.lazyLoad(exports, ["TlsRoute"], () => require("./tlsRoute"));
exports.WasmPlugin = null;
utilities.lazyLoad(exports, ["WasmPlugin"], () => require("./wasmPlugin"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:networkservices/agentGateway:AgentGateway":
                return new exports.AgentGateway(name, undefined, { urn });
            case "gcp:networkservices/authzExtension:AuthzExtension":
                return new exports.AuthzExtension(name, undefined, { urn });
            case "gcp:networkservices/edgeCacheKeyset:EdgeCacheKeyset":
                return new exports.EdgeCacheKeyset(name, undefined, { urn });
            case "gcp:networkservices/edgeCacheOrigin:EdgeCacheOrigin":
                return new exports.EdgeCacheOrigin(name, undefined, { urn });
            case "gcp:networkservices/edgeCacheService:EdgeCacheService":
                return new exports.EdgeCacheService(name, undefined, { urn });
            case "gcp:networkservices/endpointPolicy:EndpointPolicy":
                return new exports.EndpointPolicy(name, undefined, { urn });
            case "gcp:networkservices/gateway:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "gcp:networkservices/grpcRoute:GrpcRoute":
                return new exports.GrpcRoute(name, undefined, { urn });
            case "gcp:networkservices/httpRoute:HttpRoute":
                return new exports.HttpRoute(name, undefined, { urn });
            case "gcp:networkservices/lbEdgeExtension:LbEdgeExtension":
                return new exports.LbEdgeExtension(name, undefined, { urn });
            case "gcp:networkservices/lbRouteExtension:LbRouteExtension":
                return new exports.LbRouteExtension(name, undefined, { urn });
            case "gcp:networkservices/lbTrafficExtension:LbTrafficExtension":
                return new exports.LbTrafficExtension(name, undefined, { urn });
            case "gcp:networkservices/mesh:Mesh":
                return new exports.Mesh(name, undefined, { urn });
            case "gcp:networkservices/multicastConsumerAssociation:MulticastConsumerAssociation":
                return new exports.MulticastConsumerAssociation(name, undefined, { urn });
            case "gcp:networkservices/multicastDomain:MulticastDomain":
                return new exports.MulticastDomain(name, undefined, { urn });
            case "gcp:networkservices/multicastDomainActivation:MulticastDomainActivation":
                return new exports.MulticastDomainActivation(name, undefined, { urn });
            case "gcp:networkservices/multicastDomainGroup:MulticastDomainGroup":
                return new exports.MulticastDomainGroup(name, undefined, { urn });
            case "gcp:networkservices/multicastGroupConsumerActivation:MulticastGroupConsumerActivation":
                return new exports.MulticastGroupConsumerActivation(name, undefined, { urn });
            case "gcp:networkservices/multicastGroupProducerActivation:MulticastGroupProducerActivation":
                return new exports.MulticastGroupProducerActivation(name, undefined, { urn });
            case "gcp:networkservices/multicastGroupRange:MulticastGroupRange":
                return new exports.MulticastGroupRange(name, undefined, { urn });
            case "gcp:networkservices/multicastGroupRangeActivation:MulticastGroupRangeActivation":
                return new exports.MulticastGroupRangeActivation(name, undefined, { urn });
            case "gcp:networkservices/multicastProducerAssociation:MulticastProducerAssociation":
                return new exports.MulticastProducerAssociation(name, undefined, { urn });
            case "gcp:networkservices/serviceBinding:ServiceBinding":
                return new exports.ServiceBinding(name, undefined, { urn });
            case "gcp:networkservices/serviceLbPolicies:ServiceLbPolicies":
                return new exports.ServiceLbPolicies(name, undefined, { urn });
            case "gcp:networkservices/tcpRoute:TcpRoute":
                return new exports.TcpRoute(name, undefined, { urn });
            case "gcp:networkservices/tlsRoute:TlsRoute":
                return new exports.TlsRoute(name, undefined, { urn });
            case "gcp:networkservices/wasmPlugin:WasmPlugin":
                return new exports.WasmPlugin(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "networkservices/agentGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/authzExtension", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/edgeCacheKeyset", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/edgeCacheOrigin", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/edgeCacheService", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/endpointPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/gateway", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/grpcRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/httpRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/lbEdgeExtension", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/lbRouteExtension", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/lbTrafficExtension", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/mesh", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastConsumerAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastDomain", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastDomainActivation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastDomainGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastGroupConsumerActivation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastGroupProducerActivation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastGroupRange", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastGroupRangeActivation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/multicastProducerAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/serviceBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/serviceLbPolicies", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/tcpRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/tlsRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkservices/wasmPlugin", _module);
