// @pulumi/gcp/networkservices/multicastGroupRangeActivation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MulticastGroupRangeActivation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Create a multicast group range activation in the specified location of the current project.
 *
 * To get more information about MulticastGroupRangeActivation, see:
 *
 * * [API documentation](https://docs.cloud.google.com/vpc/docs/multicast/reference/rest/v1/projects.locations.multicastGroupRangeActivations)
 * * How-to Guides
 *     * [Create Multicast Group Range Activation](https://docs.cloud.google.com/vpc/docs/multicast/create-group-ranges#activate_the_group_range)
 *
 * ## Example Usage
 *
 * ### Network Services Multicast Group Range Activation Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const network = new gcp.compute.Network("network", {
 *     name: "test-network-mgra",
 *     autoCreateSubnetworks: false,
 * });
 * const multicastDomain = new gcp.networkservices.MulticastDomain("multicast_domain", {
 *     multicastDomainId: "test-domain-mgra",
 *     location: "global",
 *     adminNetwork: network.id,
 *     connectionConfig: {
 *         connectionType: "SAME_VPC",
 *     },
 * }, {
 *     dependsOn: [network],
 * });
 * const internalRange = new gcp.networkconnectivity.InternalRange("internal_range", {
 *     name: "test-internal-range-mgra",
 *     network: network.selfLink,
 *     usage: "FOR_VPC",
 *     peering: "FOR_SELF",
 *     ipCidrRange: "224.2.0.2/32",
 * });
 * const groupRange = new gcp.networkservices.MulticastGroupRange("group_range", {
 *     multicastGroupRangeId: "test-group-range-mgra",
 *     location: "global",
 *     reservedInternalRange: internalRange.id,
 *     multicastDomain: multicastDomain.id,
 * });
 * const multicastDomainActivation = new gcp.networkservices.MulticastDomainActivation("multicast_domain_activation", {
 *     multicastDomainActivationId: "test-domain-activation-mgra",
 *     location: "us-central1-b",
 *     multicastDomain: multicastDomain.id,
 * });
 * const mgraTest = new gcp.networkservices.MulticastGroupRangeActivation("mgra_test", {
 *     multicastGroupRangeActivationId: "test-mgra-mgra",
 *     location: "us-central1-b",
 *     description: "my description",
 *     labels: {
 *         "test-label": "test-value",
 *     },
 *     multicastGroupRange: groupRange.id,
 *     multicastDomainActivation: multicastDomainActivation.id,
 * });
 * ```
 *
 * ## Import
 *
 * MulticastGroupRangeActivation can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/multicastGroupRangeActivations/{{multicast_group_range_activation_id}}`
 * * `{{project}}/{{location}}/{{multicast_group_range_activation_id}}`
 * * `{{location}}/{{multicast_group_range_activation_id}}`
 *
 * When using the `pulumi import` command, MulticastGroupRangeActivation can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/multicastGroupRangeActivation:MulticastGroupRangeActivation default projects/{{project}}/locations/{{location}}/multicastGroupRangeActivations/{{multicast_group_range_activation_id}}
 * $ pulumi import gcp:networkservices/multicastGroupRangeActivation:MulticastGroupRangeActivation default {{project}}/{{location}}/{{multicast_group_range_activation_id}}
 * $ pulumi import gcp:networkservices/multicastGroupRangeActivation:MulticastGroupRangeActivation default {{location}}/{{multicast_group_range_activation_id}}
 * ```
 */
class MulticastGroupRangeActivation extends pulumi.CustomResource {
    /**
     * Get an existing MulticastGroupRangeActivation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MulticastGroupRangeActivation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/multicastGroupRangeActivation:MulticastGroupRangeActivation';
    /**
     * Returns true if the given object is an instance of MulticastGroupRangeActivation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MulticastGroupRangeActivation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["ipCidrRange"] = state?.ipCidrRange;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["logConfig"] = state?.logConfig;
            resourceInputs["multicastDomainActivation"] = state?.multicastDomainActivation;
            resourceInputs["multicastGroupConsumerActivations"] = state?.multicastGroupConsumerActivations;
            resourceInputs["multicastGroupRange"] = state?.multicastGroupRange;
            resourceInputs["multicastGroupRangeActivationId"] = state?.multicastGroupRangeActivationId;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["states"] = state?.states;
            resourceInputs["uniqueId"] = state?.uniqueId;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.multicastDomainActivation === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastDomainActivation'");
            }
            if (args?.multicastGroupRange === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastGroupRange'");
            }
            if (args?.multicastGroupRangeActivationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastGroupRangeActivationId'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["logConfig"] = args?.logConfig;
            resourceInputs["multicastDomainActivation"] = args?.multicastDomainActivation;
            resourceInputs["multicastGroupRange"] = args?.multicastGroupRange;
            resourceInputs["multicastGroupRangeActivationId"] = args?.multicastGroupRangeActivationId;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["ipCidrRange"] = undefined /*out*/;
            resourceInputs["multicastGroupConsumerActivations"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["states"] = undefined /*out*/;
            resourceInputs["uniqueId"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(MulticastGroupRangeActivation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MulticastGroupRangeActivation = MulticastGroupRangeActivation;
