// @pulumi/gcp/networkservices/multicastDomainGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MulticastDomainGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Create a multicast domain group in the current project.
 *
 * To get more information about MulticastDomainGroup, see:
 *
 * * [API documentation](https://docs.cloud.google.com/vpc/docs/multicast/reference/rest/v1/projects.locations.multicastDomainGroups)
 * * How-to Guides
 *     * [Create Multicast Domain Group](https://docs.cloud.google.com/vpc/docs/multicast/create-domains#create-domain-group)
 *
 * ## Example Usage
 *
 * ### Network Services Multicast Domain Group Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const mdgTest = new gcp.networkservices.MulticastDomainGroup("mdg_test", {
 *     multicastDomainGroupId: "test-mdg-resource",
 *     location: "global",
 *     description: "my description",
 *     labels: {
 *         fake_label: "label123",
 *     },
 * });
 * const network = new gcp.compute.Network("network", {
 *     name: "test-mdg-network",
 *     autoCreateSubnetworks: false,
 * });
 * const multicastDomainA = new gcp.networkservices.MulticastDomain("multicast_domain_a", {
 *     multicastDomainId: "test-mdg-domain-a",
 *     location: "global",
 *     adminNetwork: network.id,
 *     connectionConfig: {
 *         connectionType: "SAME_VPC",
 *     },
 *     multicastDomainGroup: mdgTest.id,
 * }, {
 *     dependsOn: [network],
 * });
 * const multicastDomainB = new gcp.networkservices.MulticastDomain("multicast_domain_b", {
 *     multicastDomainId: "test-mdg-domain-b",
 *     location: "global",
 *     adminNetwork: network.id,
 *     connectionConfig: {
 *         connectionType: "SAME_VPC",
 *     },
 *     multicastDomainGroup: mdgTest.id,
 * }, {
 *     dependsOn: [network],
 * });
 * ```
 *
 * ## Import
 *
 * MulticastDomainGroup can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/multicastDomainGroups/{{multicast_domain_group_id}}`
 * * `{{project}}/{{location}}/{{multicast_domain_group_id}}`
 * * `{{location}}/{{multicast_domain_group_id}}`
 *
 * When using the `pulumi import` command, MulticastDomainGroup can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/multicastDomainGroup:MulticastDomainGroup default projects/{{project}}/locations/{{location}}/multicastDomainGroups/{{multicast_domain_group_id}}
 * $ pulumi import gcp:networkservices/multicastDomainGroup:MulticastDomainGroup default {{project}}/{{location}}/{{multicast_domain_group_id}}
 * $ pulumi import gcp:networkservices/multicastDomainGroup:MulticastDomainGroup default {{location}}/{{multicast_domain_group_id}}
 * ```
 */
class MulticastDomainGroup extends pulumi.CustomResource {
    /**
     * Get an existing MulticastDomainGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MulticastDomainGroup(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/multicastDomainGroup:MulticastDomainGroup';
    /**
     * Returns true if the given object is an instance of MulticastDomainGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MulticastDomainGroup.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["multicastDomainGroupId"] = state?.multicastDomainGroupId;
            resourceInputs["multicastDomains"] = state?.multicastDomains;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["states"] = state?.states;
            resourceInputs["uniqueId"] = state?.uniqueId;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.multicastDomainGroupId === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastDomainGroupId'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["multicastDomainGroupId"] = args?.multicastDomainGroupId;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["multicastDomains"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["states"] = undefined /*out*/;
            resourceInputs["uniqueId"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(MulticastDomainGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MulticastDomainGroup = MulticastDomainGroup;
