// @pulumi/gcp/networkservices/multicastConsumerAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MulticastConsumerAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Create a multicast consumer association in the specified location of the current project.
 *
 * To get more information about MulticastConsumerAssociation, see:
 *
 * * [API documentation](https://docs.cloud.google.com/vpc/docs/multicast/reference/rest/v1/projects.locations.multicastConsumerAssociations)
 * * How-to Guides
 *     * [Create Multicast Consumer Association](https://docs.cloud.google.com/vpc/docs/multicast/enable-consumer-network#add-consumer)
 *
 * ## Example Usage
 *
 * ### Network Services Multicast Consumer Association Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const network = new gcp.compute.Network("network", {
 *     name: "test-network-mca",
 *     autoCreateSubnetworks: false,
 * });
 * const multicastDomain = new gcp.networkservices.MulticastDomain("multicast_domain", {
 *     multicastDomainId: "test-domain-mca",
 *     location: "global",
 *     adminNetwork: network.id,
 *     connectionConfig: {
 *         connectionType: "SAME_VPC",
 *     },
 * }, {
 *     dependsOn: [network],
 * });
 * const multicastDomainActivation = new gcp.networkservices.MulticastDomainActivation("multicast_domain_activation", {
 *     multicastDomainActivationId: "test-domain-activation-mca",
 *     location: "us-central1-b",
 *     multicastDomain: multicastDomain.id,
 * });
 * const mcaTest = new gcp.networkservices.MulticastConsumerAssociation("mca_test", {
 *     multicastConsumerAssociationId: "test-consumer-association-mca",
 *     location: "us-central1-b",
 *     network: network.id,
 *     multicastDomainActivation: multicastDomainActivation.id,
 * }, {
 *     dependsOn: [network],
 * });
 * ```
 *
 * ## Import
 *
 * MulticastConsumerAssociation can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/multicastConsumerAssociations/{{multicast_consumer_association_id}}`
 * * `{{project}}/{{location}}/{{multicast_consumer_association_id}}`
 * * `{{location}}/{{multicast_consumer_association_id}}`
 *
 * When using the `pulumi import` command, MulticastConsumerAssociation can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/multicastConsumerAssociation:MulticastConsumerAssociation default projects/{{project}}/locations/{{location}}/multicastConsumerAssociations/{{multicast_consumer_association_id}}
 * $ pulumi import gcp:networkservices/multicastConsumerAssociation:MulticastConsumerAssociation default {{project}}/{{location}}/{{multicast_consumer_association_id}}
 * $ pulumi import gcp:networkservices/multicastConsumerAssociation:MulticastConsumerAssociation default {{location}}/{{multicast_consumer_association_id}}
 * ```
 */
class MulticastConsumerAssociation extends pulumi.CustomResource {
    /**
     * Get an existing MulticastConsumerAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MulticastConsumerAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/multicastConsumerAssociation:MulticastConsumerAssociation';
    /**
     * Returns true if the given object is an instance of MulticastConsumerAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MulticastConsumerAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["multicastConsumerAssociationId"] = state?.multicastConsumerAssociationId;
            resourceInputs["multicastDomainActivation"] = state?.multicastDomainActivation;
            resourceInputs["name"] = state?.name;
            resourceInputs["network"] = state?.network;
            resourceInputs["placementPolicy"] = state?.placementPolicy;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["states"] = state?.states;
            resourceInputs["uniqueId"] = state?.uniqueId;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if (args?.multicastConsumerAssociationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastConsumerAssociationId'");
            }
            if (args?.multicastDomainActivation === undefined && !opts.urn) {
                throw new Error("Missing required property 'multicastDomainActivation'");
            }
            if (args?.network === undefined && !opts.urn) {
                throw new Error("Missing required property 'network'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["multicastConsumerAssociationId"] = args?.multicastConsumerAssociationId;
            resourceInputs["multicastDomainActivation"] = args?.multicastDomainActivation;
            resourceInputs["network"] = args?.network;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["placementPolicy"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["states"] = undefined /*out*/;
            resourceInputs["uniqueId"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(MulticastConsumerAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MulticastConsumerAssociation = MulticastConsumerAssociation;
