// @pulumi/gcp/networkservices/serviceLbPolicies.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLbPolicies = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * ServiceLbPolicy holds global load balancing and traffic distribution configuration that can be applied to a BackendService.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about ServiceLbPolicies, see:
 *
 * * [API documentation](https://cloud.google.com/service-mesh/docs/reference/network-services/rest/v1/projects.locations.serviceLbPolicies)
 *
 * ## Example Usage
 *
 * ### Network Services Service Lb Policies Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.ServiceLbPolicies("default", {
 *     name: "my-lb-policy",
 *     location: "global",
 * });
 * ```
 * ### Network Services Service Lb Policies Advanced
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.ServiceLbPolicies("default", {
 *     name: "my-lb-policy",
 *     location: "global",
 *     description: "my description",
 *     loadBalancingAlgorithm: "SPRAY_TO_REGION",
 *     autoCapacityDrain: {
 *         enable: true,
 *     },
 *     failoverConfig: {
 *         failoverHealthThreshold: 70,
 *     },
 *     labels: {
 *         foo: "bar",
 *     },
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "my-lb-backend",
 *     description: "my description",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     protocol: "HTTP",
 *     serviceLbPolicy: pulumi.interpolate`//networkservices.googleapis.com/${_default.id}`,
 * });
 * ```
 * ### Network Services Service Lb Policies Beta
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.ServiceLbPolicies("default", {
 *     name: "my-lb-policy",
 *     location: "global",
 *     description: "my description",
 *     loadBalancingAlgorithm: "SPRAY_TO_REGION",
 *     autoCapacityDrain: {
 *         enable: true,
 *     },
 *     failoverConfig: {
 *         failoverHealthThreshold: 70,
 *     },
 *     isolationConfig: {
 *         isolationGranularity: "REGION",
 *         isolationMode: "NEAREST",
 *     },
 *     labels: {
 *         foo: "bar",
 *     },
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "my-lb-backend",
 *     description: "my description",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     protocol: "HTTP",
 *     serviceLbPolicy: pulumi.interpolate`//networkservices.googleapis.com/${_default.id}`,
 * });
 * ```
 *
 * ## Import
 *
 * ServiceLbPolicies can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/serviceLbPolicies/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, ServiceLbPolicies can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/serviceLbPolicies:ServiceLbPolicies default projects/{{project}}/locations/{{location}}/serviceLbPolicies/{{name}}
 * $ pulumi import gcp:networkservices/serviceLbPolicies:ServiceLbPolicies default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:networkservices/serviceLbPolicies:ServiceLbPolicies default {{location}}/{{name}}
 * ```
 */
class ServiceLbPolicies extends pulumi.CustomResource {
    /**
     * Get an existing ServiceLbPolicies resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceLbPolicies(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/serviceLbPolicies:ServiceLbPolicies';
    /**
     * Returns true if the given object is an instance of ServiceLbPolicies.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceLbPolicies.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["autoCapacityDrain"] = state?.autoCapacityDrain;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["failoverConfig"] = state?.failoverConfig;
            resourceInputs["isolationConfig"] = state?.isolationConfig;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["loadBalancingAlgorithm"] = state?.loadBalancingAlgorithm;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["autoCapacityDrain"] = args?.autoCapacityDrain;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["failoverConfig"] = args?.failoverConfig;
            resourceInputs["isolationConfig"] = args?.isolationConfig;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["loadBalancingAlgorithm"] = args?.loadBalancingAlgorithm;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(ServiceLbPolicies.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceLbPolicies = ServiceLbPolicies;
