// @pulumi/gcp/networkservices/tcpRoute.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TcpRoute = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * TcpRoute is the resource defining how TCP traffic should be routed by a Mesh/Gateway resource.
 *
 * To get more information about TcpRoute, see:
 *
 * * [API documentation](https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.tcpRoutes)
 * * How-to Guides
 *     * [Setup TCP Services](https://cloud.google.com/traffic-director/docs/set-up-tcp-route)
 *
 * ## Example Usage
 *
 * ### Network Services Tcp Route Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "backend-service-health-check",
 *     tcpHealthCheck: {
 *         port: 80,
 *     },
 * });
 * const _default = new gcp.compute.BackendService("default", {
 *     name: "my-backend-service",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const defaultTcpRoute = new gcp.networkservices.TcpRoute("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     rules: [{
 *         matches: [{
 *             address: "10.0.0.1/32",
 *             port: "8081",
 *         }],
 *         action: {
 *             destinations: [{
 *                 serviceName: _default.id,
 *                 weight: 1,
 *             }],
 *             originalDestination: false,
 *         },
 *     }],
 * });
 * ```
 * ### Network Services Tcp Route Actions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "backend-service-health-check",
 *     tcpHealthCheck: {
 *         port: 80,
 *     },
 * });
 * const _default = new gcp.compute.BackendService("default", {
 *     name: "my-backend-service",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const defaultTcpRoute = new gcp.networkservices.TcpRoute("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     rules: [{
 *         action: {
 *             destinations: [{
 *                 serviceName: _default.id,
 *                 weight: 1,
 *             }],
 *             originalDestination: false,
 *             idleTimeout: "60s",
 *         },
 *     }],
 * });
 * ```
 * ### Network Services Tcp Route Mesh Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "backend-service-health-check",
 *     tcpHealthCheck: {
 *         port: 80,
 *     },
 * });
 * const _default = new gcp.compute.BackendService("default", {
 *     name: "my-backend-service",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const defaultMesh = new gcp.networkservices.Mesh("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 * });
 * const defaultTcpRoute = new gcp.networkservices.TcpRoute("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     meshes: [defaultMesh.id],
 *     rules: [{
 *         matches: [{
 *             address: "10.0.0.1/32",
 *             port: "8081",
 *         }],
 *         action: {
 *             destinations: [{
 *                 serviceName: _default.id,
 *                 weight: 1,
 *             }],
 *             originalDestination: false,
 *         },
 *     }],
 * });
 * ```
 * ### Network Services Tcp Route Gateway Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "backend-service-health-check",
 *     tcpHealthCheck: {
 *         port: 80,
 *     },
 * });
 * const _default = new gcp.compute.BackendService("default", {
 *     name: "my-backend-service",
 *     loadBalancingScheme: "INTERNAL_SELF_MANAGED",
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const defaultGateway = new gcp.networkservices.Gateway("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     scope: "my-scope",
 *     type: "OPEN_MESH",
 *     ports: [443],
 * });
 * const defaultTcpRoute = new gcp.networkservices.TcpRoute("default", {
 *     name: "my-tcp-route",
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     gateways: [defaultGateway.id],
 *     rules: [{
 *         matches: [{
 *             address: "10.0.0.1/32",
 *             port: "8081",
 *         }],
 *         action: {
 *             destinations: [{
 *                 serviceName: _default.id,
 *                 weight: 1,
 *             }],
 *             originalDestination: false,
 *         },
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * TcpRoute can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/global/tcpRoutes/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, TcpRoute can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkservices/tcpRoute:TcpRoute default projects/{{project}}/locations/global/tcpRoutes/{{name}}
 * $ pulumi import gcp:networkservices/tcpRoute:TcpRoute default {{project}}/{{name}}
 * $ pulumi import gcp:networkservices/tcpRoute:TcpRoute default {{name}}
 * ```
 */
class TcpRoute extends pulumi.CustomResource {
    /**
     * Get an existing TcpRoute resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TcpRoute(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkservices/tcpRoute:TcpRoute';
    /**
     * Returns true if the given object is an instance of TcpRoute.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TcpRoute.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["gateways"] = state?.gateways;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["meshes"] = state?.meshes;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["rules"] = state?.rules;
            resourceInputs["selfLink"] = state?.selfLink;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.rules === undefined && !opts.urn) {
                throw new Error("Missing required property 'rules'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["gateways"] = args?.gateways;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["meshes"] = args?.meshes;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["rules"] = args?.rules;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(TcpRoute.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TcpRoute = TcpRoute;
