// @pulumi/gcp/apigee/keystoresAliasesSelfSignedCert.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeystoresAliasesSelfSignedCert = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An Environment Keystore Alias for Self Signed Certificate Format in Apigee
 *
 * To get more information about KeystoresAliasesSelfSignedCert, see:
 *
 * * [API documentation](https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.environments.keystores.aliases/create)
 * * How-to Guides
 *     * [Creating an environment](https://cloud.google.com/apigee/docs/api-platform/get-started/create-environment)
 *
 * ## Example Usage
 *
 * ### Apigee Env Keystore Alias Self Signed Cert
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumiverse/time";
 *
 * const project = new gcp.organizations.Project("project", {
 *     projectId: "my-project",
 *     name: "my-project",
 *     orgId: "123456789",
 *     billingAccount: "000000-0000000-0000000-000000",
 *     deletionPolicy: "DELETE",
 * });
 * const apigee = new gcp.projects.Service("apigee", {
 *     project: project.projectId,
 *     service: "apigee.googleapis.com",
 * });
 * const servicenetworking = new gcp.projects.Service("servicenetworking", {
 *     project: project.projectId,
 *     service: "servicenetworking.googleapis.com",
 * }, {
 *     dependsOn: [apigee],
 * });
 * const compute = new gcp.projects.Service("compute", {
 *     project: project.projectId,
 *     service: "compute.googleapis.com",
 * }, {
 *     dependsOn: [servicenetworking],
 * });
 * const wait300Seconds = new time.Sleep("wait_300_seconds", {createDuration: "300s"}, {
 *     dependsOn: [compute],
 * });
 * const apigeeNetwork = new gcp.compute.Network("apigee_network", {
 *     name: "apigee-network",
 *     project: project.projectId,
 * }, {
 *     dependsOn: [wait300Seconds],
 * });
 * const apigeeRange = new gcp.compute.GlobalAddress("apigee_range", {
 *     name: "apigee-range",
 *     purpose: "VPC_PEERING",
 *     addressType: "INTERNAL",
 *     prefixLength: 16,
 *     network: apigeeNetwork.id,
 *     project: project.projectId,
 * });
 * const apigeeVpcConnection = new gcp.servicenetworking.Connection("apigee_vpc_connection", {
 *     network: apigeeNetwork.id,
 *     service: "servicenetworking.googleapis.com",
 *     reservedPeeringRanges: [apigeeRange.name],
 * }, {
 *     dependsOn: [servicenetworking],
 * });
 * const apigeeOrg = new gcp.apigee.Organization("apigee_org", {
 *     analyticsRegion: "us-central1",
 *     projectId: project.projectId,
 *     authorizedNetwork: apigeeNetwork.id,
 * }, {
 *     dependsOn: [
 *         apigeeVpcConnection,
 *         apigee,
 *     ],
 * });
 * const apigeeEnvironmentKeystoreSsAlias = new gcp.apigee.Environment("apigee_environment_keystore_ss_alias", {
 *     orgId: apigeeOrg.id,
 *     name: "env-name",
 *     description: "Apigee Environment",
 *     displayName: "environment-1",
 * });
 * const apigeeEnvironmentKeystoreAlias = new gcp.apigee.EnvKeystore("apigee_environment_keystore_alias", {
 *     name: "env-keystore",
 *     envId: apigeeEnvironmentKeystoreSsAlias.id,
 * });
 * const apigeeEnvironmentKeystoreSsAliasKeystoresAliasesSelfSignedCert = new gcp.apigee.KeystoresAliasesSelfSignedCert("apigee_environment_keystore_ss_alias", {
 *     environment: apigeeEnvironmentKeystoreSsAlias.name,
 *     orgId: apigeeOrg.name,
 *     keystore: apigeeEnvironmentKeystoreAlias.name,
 *     alias: "alias",
 *     keySize: "1024",
 *     sigAlg: "SHA512withRSA",
 *     certValidityInDays: 4,
 *     subject: {
 *         commonName: "selfsigned_example",
 *         countryCode: "US",
 *         locality: "TX",
 *         org: "CCE",
 *         orgUnit: "PSO",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * KeystoresAliasesSelfSignedCert can be imported using any of these accepted formats:
 *
 * * `organizations/{{org_id}}/environments/{{environment}}/keystores/{{keystore}}/aliases/{{alias}}`
 * * `{{org_id}}/{{environment}}/{{keystore}}/{{alias}}`
 *
 * When using the `pulumi import` command, KeystoresAliasesSelfSignedCert can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:apigee/keystoresAliasesSelfSignedCert:KeystoresAliasesSelfSignedCert default organizations/{{org_id}}/environments/{{environment}}/keystores/{{keystore}}/aliases/{{alias}}
 * $ pulumi import gcp:apigee/keystoresAliasesSelfSignedCert:KeystoresAliasesSelfSignedCert default {{org_id}}/{{environment}}/{{keystore}}/{{alias}}
 * ```
 */
class KeystoresAliasesSelfSignedCert extends pulumi.CustomResource {
    /**
     * Get an existing KeystoresAliasesSelfSignedCert resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KeystoresAliasesSelfSignedCert(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:apigee/keystoresAliasesSelfSignedCert:KeystoresAliasesSelfSignedCert';
    /**
     * Returns true if the given object is an instance of KeystoresAliasesSelfSignedCert.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KeystoresAliasesSelfSignedCert.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["alias"] = state?.alias;
            resourceInputs["certValidityInDays"] = state?.certValidityInDays;
            resourceInputs["certsInfos"] = state?.certsInfos;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["environment"] = state?.environment;
            resourceInputs["keySize"] = state?.keySize;
            resourceInputs["keystore"] = state?.keystore;
            resourceInputs["orgId"] = state?.orgId;
            resourceInputs["sigAlg"] = state?.sigAlg;
            resourceInputs["subject"] = state?.subject;
            resourceInputs["subjectAlternativeDnsNames"] = state?.subjectAlternativeDnsNames;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.alias === undefined && !opts.urn) {
                throw new Error("Missing required property 'alias'");
            }
            if (args?.environment === undefined && !opts.urn) {
                throw new Error("Missing required property 'environment'");
            }
            if (args?.keystore === undefined && !opts.urn) {
                throw new Error("Missing required property 'keystore'");
            }
            if (args?.orgId === undefined && !opts.urn) {
                throw new Error("Missing required property 'orgId'");
            }
            if (args?.sigAlg === undefined && !opts.urn) {
                throw new Error("Missing required property 'sigAlg'");
            }
            if (args?.subject === undefined && !opts.urn) {
                throw new Error("Missing required property 'subject'");
            }
            resourceInputs["alias"] = args?.alias;
            resourceInputs["certValidityInDays"] = args?.certValidityInDays;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["environment"] = args?.environment;
            resourceInputs["keySize"] = args?.keySize;
            resourceInputs["keystore"] = args?.keystore;
            resourceInputs["orgId"] = args?.orgId;
            resourceInputs["sigAlg"] = args?.sigAlg;
            resourceInputs["subject"] = args?.subject;
            resourceInputs["subjectAlternativeDnsNames"] = args?.subjectAlternativeDnsNames;
            resourceInputs["certsInfos"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(KeystoresAliasesSelfSignedCert.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KeystoresAliasesSelfSignedCert = KeystoresAliasesSelfSignedCert;
