// @pulumi/kubernetes/certificates/v1beta1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PodCertificateRequestPatch = exports.PodCertificateRequestList = exports.PodCertificateRequest = exports.ClusterTrustBundlePatch = exports.ClusterTrustBundleList = exports.ClusterTrustBundle = exports.CertificateSigningRequestPatch = exports.CertificateSigningRequestList = exports.CertificateSigningRequest = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.CertificateSigningRequest = null;
utilities.lazyLoad(exports, ["CertificateSigningRequest"], () => require("./certificateSigningRequest"));
exports.CertificateSigningRequestList = null;
utilities.lazyLoad(exports, ["CertificateSigningRequestList"], () => require("./certificateSigningRequestList"));
exports.CertificateSigningRequestPatch = null;
utilities.lazyLoad(exports, ["CertificateSigningRequestPatch"], () => require("./certificateSigningRequestPatch"));
exports.ClusterTrustBundle = null;
utilities.lazyLoad(exports, ["ClusterTrustBundle"], () => require("./clusterTrustBundle"));
exports.ClusterTrustBundleList = null;
utilities.lazyLoad(exports, ["ClusterTrustBundleList"], () => require("./clusterTrustBundleList"));
exports.ClusterTrustBundlePatch = null;
utilities.lazyLoad(exports, ["ClusterTrustBundlePatch"], () => require("./clusterTrustBundlePatch"));
exports.PodCertificateRequest = null;
utilities.lazyLoad(exports, ["PodCertificateRequest"], () => require("./podCertificateRequest"));
exports.PodCertificateRequestList = null;
utilities.lazyLoad(exports, ["PodCertificateRequestList"], () => require("./podCertificateRequestList"));
exports.PodCertificateRequestPatch = null;
utilities.lazyLoad(exports, ["PodCertificateRequestPatch"], () => require("./podCertificateRequestPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:certificates.k8s.io/v1beta1:CertificateSigningRequest":
                return new exports.CertificateSigningRequest(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:CertificateSigningRequestList":
                return new exports.CertificateSigningRequestList(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:CertificateSigningRequestPatch":
                return new exports.CertificateSigningRequestPatch(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:ClusterTrustBundle":
                return new exports.ClusterTrustBundle(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:ClusterTrustBundleList":
                return new exports.ClusterTrustBundleList(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:ClusterTrustBundlePatch":
                return new exports.ClusterTrustBundlePatch(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:PodCertificateRequest":
                return new exports.PodCertificateRequest(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:PodCertificateRequestList":
                return new exports.PodCertificateRequestList(name, undefined, { urn });
            case "kubernetes:certificates.k8s.io/v1beta1:PodCertificateRequestPatch":
                return new exports.PodCertificateRequestPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "certificates.k8s.io/v1beta1", _module);
