// @pulumi/aws/costexplorer/costCategory.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostCategory = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a CE Cost Category.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.costexplorer.CostCategory("test", {
 *     name: "NAME",
 *     ruleVersion: "CostCategoryExpression.v1",
 *     rules: [
 *         {
 *             value: "production",
 *             rule: {
 *                 dimension: {
 *                     key: "LINKED_ACCOUNT_NAME",
 *                     values: ["-prod"],
 *                     matchOptions: ["ENDS_WITH"],
 *                 },
 *             },
 *         },
 *         {
 *             value: "staging",
 *             rule: {
 *                 dimension: {
 *                     key: "LINKED_ACCOUNT_NAME",
 *                     values: ["-stg"],
 *                     matchOptions: ["ENDS_WITH"],
 *                 },
 *             },
 *         },
 *         {
 *             value: "testing",
 *             rule: {
 *                 dimension: {
 *                     key: "LINKED_ACCOUNT_NAME",
 *                     values: ["-dev"],
 *                     matchOptions: ["ENDS_WITH"],
 *                 },
 *             },
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `arn` (String) Amazon Resource Name (ARN) of the Cost Explorer cost category.
 *
 * Using `pulumi import`, import `aws.costexplorer.CostCategory` using the id. For example:
 *
 * ```sh
 * $ pulumi import aws:costexplorer/costCategory:CostCategory example costCategoryARN
 * ```
 */
class CostCategory extends pulumi.CustomResource {
    /**
     * Get an existing CostCategory resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CostCategory(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:costexplorer/costCategory:CostCategory';
    /**
     * Returns true if the given object is an instance of CostCategory.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CostCategory.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["defaultValue"] = state?.defaultValue;
            resourceInputs["effectiveEnd"] = state?.effectiveEnd;
            resourceInputs["effectiveStart"] = state?.effectiveStart;
            resourceInputs["name"] = state?.name;
            resourceInputs["ruleVersion"] = state?.ruleVersion;
            resourceInputs["rules"] = state?.rules;
            resourceInputs["splitChargeRules"] = state?.splitChargeRules;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.ruleVersion === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleVersion'");
            }
            if (args?.rules === undefined && !opts.urn) {
                throw new Error("Missing required property 'rules'");
            }
            resourceInputs["defaultValue"] = args?.defaultValue;
            resourceInputs["effectiveStart"] = args?.effectiveStart;
            resourceInputs["name"] = args?.name;
            resourceInputs["ruleVersion"] = args?.ruleVersion;
            resourceInputs["rules"] = args?.rules;
            resourceInputs["splitChargeRules"] = args?.splitChargeRules;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["effectiveEnd"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CostCategory.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CostCategory = CostCategory;
