// @pulumi/gcp/alloydb/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Instance = exports.getSupportedDatabaseFlagsOutput = exports.getSupportedDatabaseFlags = exports.getLocationsOutput = exports.getLocations = exports.getInstanceOutput = exports.getInstance = exports.getClusterOutput = exports.getCluster = exports.Cluster = exports.Backup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Backup = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getLocations = null;
exports.getLocationsOutput = null;
utilities.lazyLoad(exports, ["getLocations", "getLocationsOutput"], () => require("./getLocations"));
exports.getSupportedDatabaseFlags = null;
exports.getSupportedDatabaseFlagsOutput = null;
utilities.lazyLoad(exports, ["getSupportedDatabaseFlags", "getSupportedDatabaseFlagsOutput"], () => require("./getSupportedDatabaseFlags"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:alloydb/backup:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "gcp:alloydb/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "gcp:alloydb/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "gcp:alloydb/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "alloydb/backup", _module);
pulumi.runtime.registerResourceModule("gcp", "alloydb/cluster", _module);
pulumi.runtime.registerResourceModule("gcp", "alloydb/instance", _module);
pulumi.runtime.registerResourceModule("gcp", "alloydb/user", _module);
