// @pulumi/azure-native/apimanagement/api.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Api = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * API details.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2022-08-01.
 *
 * Other available API versions: 2021-04-01-preview, 2021-08-01, 2021-12-01-preview, 2022-04-01-preview, 2022-08-01, 2022-09-01-preview, 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-06-01-preview, 2024-10-01-preview, 2025-03-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native apimanagement [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Api extends pulumi.CustomResource {
    /**
     * Get an existing Api resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Api(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:apimanagement:Api';
    /**
     * Returns true if the given object is an instance of Api.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Api.__pulumiType;
    }
    /**
     * Create a Api resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.path === undefined && !opts.urn) {
                throw new Error("Missing required property 'path'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.serviceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["apiId"] = args?.apiId;
            resourceInputs["apiRevision"] = args?.apiRevision;
            resourceInputs["apiRevisionDescription"] = args?.apiRevisionDescription;
            resourceInputs["apiType"] = args?.apiType;
            resourceInputs["apiVersion"] = args?.apiVersion;
            resourceInputs["apiVersionDescription"] = args?.apiVersionDescription;
            resourceInputs["apiVersionSet"] = args?.apiVersionSet;
            resourceInputs["apiVersionSetId"] = args?.apiVersionSetId;
            resourceInputs["authenticationSettings"] = args?.authenticationSettings;
            resourceInputs["contact"] = args?.contact;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["format"] = args?.format;
            resourceInputs["isCurrent"] = args?.isCurrent;
            resourceInputs["license"] = args?.license;
            resourceInputs["path"] = args?.path;
            resourceInputs["protocols"] = args?.protocols;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["serviceName"] = args?.serviceName;
            resourceInputs["serviceUrl"] = args?.serviceUrl;
            resourceInputs["soapApiType"] = args?.soapApiType;
            resourceInputs["sourceApiId"] = args?.sourceApiId;
            resourceInputs["subscriptionKeyParameterNames"] = args?.subscriptionKeyParameterNames;
            resourceInputs["subscriptionRequired"] = args?.subscriptionRequired;
            resourceInputs["termsOfServiceUrl"] = args?.termsOfServiceUrl;
            resourceInputs["translateRequiredQueryParametersConduct"] = args?.translateRequiredQueryParametersConduct;
            resourceInputs["value"] = args?.value;
            resourceInputs["wsdlSelector"] = args?.wsdlSelector;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["isOnline"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["apiRevision"] = undefined /*out*/;
            resourceInputs["apiRevisionDescription"] = undefined /*out*/;
            resourceInputs["apiType"] = undefined /*out*/;
            resourceInputs["apiVersion"] = undefined /*out*/;
            resourceInputs["apiVersionDescription"] = undefined /*out*/;
            resourceInputs["apiVersionSet"] = undefined /*out*/;
            resourceInputs["apiVersionSetId"] = undefined /*out*/;
            resourceInputs["authenticationSettings"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["contact"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["isCurrent"] = undefined /*out*/;
            resourceInputs["isOnline"] = undefined /*out*/;
            resourceInputs["license"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["path"] = undefined /*out*/;
            resourceInputs["protocols"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serviceUrl"] = undefined /*out*/;
            resourceInputs["sourceApiId"] = undefined /*out*/;
            resourceInputs["subscriptionKeyParameterNames"] = undefined /*out*/;
            resourceInputs["subscriptionRequired"] = undefined /*out*/;
            resourceInputs["termsOfServiceUrl"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20160707:Api" }, { type: "azure-native:apimanagement/v20161010:Api" }, { type: "azure-native:apimanagement/v20170301:Api" }, { type: "azure-native:apimanagement/v20180101:Api" }, { type: "azure-native:apimanagement/v20180601preview:Api" }, { type: "azure-native:apimanagement/v20190101:Api" }, { type: "azure-native:apimanagement/v20191201:Api" }, { type: "azure-native:apimanagement/v20191201preview:Api" }, { type: "azure-native:apimanagement/v20200601preview:Api" }, { type: "azure-native:apimanagement/v20201201:Api" }, { type: "azure-native:apimanagement/v20210101preview:Api" }, { type: "azure-native:apimanagement/v20210401preview:Api" }, { type: "azure-native:apimanagement/v20210801:Api" }, { type: "azure-native:apimanagement/v20211201preview:Api" }, { type: "azure-native:apimanagement/v20220401preview:Api" }, { type: "azure-native:apimanagement/v20220801:Api" }, { type: "azure-native:apimanagement/v20220901preview:Api" }, { type: "azure-native:apimanagement/v20230301preview:Api" }, { type: "azure-native:apimanagement/v20230501preview:Api" }, { type: "azure-native:apimanagement/v20230901preview:Api" }, { type: "azure-native:apimanagement/v20240501:Api" }, { type: "azure-native:apimanagement/v20240601preview:Api" }, { type: "azure-native:apimanagement/v20241001preview:Api" }, { type: "azure-native:apimanagement/v20250301preview:Api" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Api.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Api = Api;
