// @pulumi/azure-native/apimanagement/authorizationServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizationServer = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * External OAuth authorization server settings.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2022-08-01.
 *
 * Other available API versions: 2021-04-01-preview, 2021-08-01, 2021-12-01-preview, 2022-04-01-preview, 2022-08-01, 2022-09-01-preview, 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-06-01-preview, 2024-10-01-preview, 2025-03-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native apimanagement [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AuthorizationServer extends pulumi.CustomResource {
    /**
     * Get an existing AuthorizationServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AuthorizationServer(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:apimanagement:AuthorizationServer';
    /**
     * Returns true if the given object is an instance of AuthorizationServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AuthorizationServer.__pulumiType;
    }
    /**
     * Create a AuthorizationServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.authorizationEndpoint === undefined && !opts.urn) {
                throw new Error("Missing required property 'authorizationEndpoint'");
            }
            if (args?.clientId === undefined && !opts.urn) {
                throw new Error("Missing required property 'clientId'");
            }
            if (args?.clientRegistrationEndpoint === undefined && !opts.urn) {
                throw new Error("Missing required property 'clientRegistrationEndpoint'");
            }
            if (args?.displayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if (args?.grantTypes === undefined && !opts.urn) {
                throw new Error("Missing required property 'grantTypes'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.serviceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["authorizationEndpoint"] = args?.authorizationEndpoint;
            resourceInputs["authorizationMethods"] = args?.authorizationMethods;
            resourceInputs["authsid"] = args?.authsid;
            resourceInputs["bearerTokenSendingMethods"] = args?.bearerTokenSendingMethods;
            resourceInputs["clientAuthenticationMethod"] = args?.clientAuthenticationMethod;
            resourceInputs["clientId"] = args?.clientId;
            resourceInputs["clientRegistrationEndpoint"] = args?.clientRegistrationEndpoint;
            resourceInputs["clientSecret"] = args?.clientSecret;
            resourceInputs["defaultScope"] = args?.defaultScope;
            resourceInputs["description"] = args?.description;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["grantTypes"] = args?.grantTypes;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["resourceOwnerPassword"] = args?.resourceOwnerPassword;
            resourceInputs["resourceOwnerUsername"] = args?.resourceOwnerUsername;
            resourceInputs["serviceName"] = args?.serviceName;
            resourceInputs["supportState"] = args?.supportState;
            resourceInputs["tokenBodyParameters"] = args?.tokenBodyParameters;
            resourceInputs["tokenEndpoint"] = args?.tokenEndpoint;
            resourceInputs["useInApiDocumentation"] = args?.useInApiDocumentation;
            resourceInputs["useInTestConsole"] = args?.useInTestConsole;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["authorizationEndpoint"] = undefined /*out*/;
            resourceInputs["authorizationMethods"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bearerTokenSendingMethods"] = undefined /*out*/;
            resourceInputs["clientAuthenticationMethod"] = undefined /*out*/;
            resourceInputs["clientId"] = undefined /*out*/;
            resourceInputs["clientRegistrationEndpoint"] = undefined /*out*/;
            resourceInputs["clientSecret"] = undefined /*out*/;
            resourceInputs["defaultScope"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["grantTypes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resourceOwnerPassword"] = undefined /*out*/;
            resourceInputs["resourceOwnerUsername"] = undefined /*out*/;
            resourceInputs["supportState"] = undefined /*out*/;
            resourceInputs["tokenBodyParameters"] = undefined /*out*/;
            resourceInputs["tokenEndpoint"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useInApiDocumentation"] = undefined /*out*/;
            resourceInputs["useInTestConsole"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20160707:AuthorizationServer" }, { type: "azure-native:apimanagement/v20161010:AuthorizationServer" }, { type: "azure-native:apimanagement/v20170301:AuthorizationServer" }, { type: "azure-native:apimanagement/v20180101:AuthorizationServer" }, { type: "azure-native:apimanagement/v20180601preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20190101:AuthorizationServer" }, { type: "azure-native:apimanagement/v20191201:AuthorizationServer" }, { type: "azure-native:apimanagement/v20191201preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20200601preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20201201:AuthorizationServer" }, { type: "azure-native:apimanagement/v20210101preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20210401preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20210801:AuthorizationServer" }, { type: "azure-native:apimanagement/v20211201preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20220401preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20220801:AuthorizationServer" }, { type: "azure-native:apimanagement/v20220901preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20230301preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20230501preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20230901preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20240501:AuthorizationServer" }, { type: "azure-native:apimanagement/v20240601preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20241001preview:AuthorizationServer" }, { type: "azure-native:apimanagement/v20250301preview:AuthorizationServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AuthorizationServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AuthorizationServer = AuthorizationServer;
