// @pulumi/gcp/clouddeploy/customTargetType.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTargetType = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A Cloud Deploy `CustomTargetType` defines a type of custom target that can be referenced in a
 * Cloud Deploy `Target` in order to facilitate deploying to other systems besides the supported runtimes.
 *
 * To get more information about CustomTargetType, see:
 *
 * * [API documentation](https://cloud.google.com/deploy/docs/api/reference/rest/v1/projects.locations.customTargetTypes)
 * * How-to Guides
 *     * [Define and use a custom target type](https://cloud.google.com/deploy/docs/deploy-app-custom-target)
 *
 * ## Example Usage
 *
 * ### Clouddeploy Custom Target Type Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const custom_target_type = new gcp.clouddeploy.CustomTargetType("custom-target-type", {
 *     location: "us-central1",
 *     name: "my-custom-target-type",
 *     description: "My custom target type",
 *     annotations: {
 *         my_first_annotation: "example-annotation-1",
 *         my_second_annotation: "example-annotation-2",
 *     },
 *     labels: {
 *         my_first_label: "example-label-1",
 *         my_second_label: "example-label-2",
 *     },
 *     customActions: {
 *         renderAction: "renderAction",
 *         deployAction: "deployAction",
 *     },
 * });
 * ```
 * ### Clouddeploy Custom Target Type Git Skaffold Modules
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const custom_target_type = new gcp.clouddeploy.CustomTargetType("custom-target-type", {
 *     location: "us-central1",
 *     name: "my-custom-target-type",
 *     description: "My custom target type",
 *     customActions: {
 *         renderAction: "renderAction",
 *         deployAction: "deployAction",
 *         includeSkaffoldModules: [{
 *             configs: ["my-config"],
 *             git: {
 *                 repo: "http://github.com/example/example-repo.git",
 *                 path: "configs/skaffold.yaml",
 *                 ref: "main",
 *             },
 *         }],
 *     },
 * });
 * ```
 * ### Clouddeploy Custom Target Type Gcs Skaffold Modules
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const custom_target_type = new gcp.clouddeploy.CustomTargetType("custom-target-type", {
 *     location: "us-central1",
 *     name: "my-custom-target-type",
 *     description: "My custom target type",
 *     customActions: {
 *         renderAction: "renderAction",
 *         deployAction: "deployAction",
 *         includeSkaffoldModules: [{
 *             configs: ["my-config"],
 *             googleCloudStorage: {
 *                 source: "gs://example-bucket/dir/configs/*",
 *                 path: "skaffold.yaml",
 *             },
 *         }],
 *     },
 * });
 * ```
 * ### Clouddeploy Custom Target Type Gcb Repo Skaffold Modules
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const custom_target_type = new gcp.clouddeploy.CustomTargetType("custom-target-type", {
 *     location: "us-central1",
 *     name: "my-custom-target-type",
 *     description: "My custom target type",
 *     customActions: {
 *         renderAction: "renderAction",
 *         deployAction: "deployAction",
 *         includeSkaffoldModules: [{
 *             configs: ["my-config"],
 *             googleCloudBuildRepo: {
 *                 repository: "projects/example/locations/us-central1/connections/git/repositories/example-repo",
 *                 path: "configs/skaffold.yaml",
 *                 ref: "main",
 *             },
 *         }],
 *     },
 * });
 * ```
 * ### Clouddeploy Custom Target Type Tasks
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const custom_target_type = new gcp.clouddeploy.CustomTargetType("custom-target-type", {
 *     location: "us-central1",
 *     name: "my-custom-target-type",
 *     description: "My custom target type",
 *     tasks: {
 *         render: {
 *             container: {
 *                 image: "gcr.io/my-project/my-render-image",
 *             },
 *         },
 *         deploy: {
 *             container: {
 *                 image: "gcr.io/my-project/my-deploy-image",
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * CustomTargetType can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/customTargetTypes/{{name}}`
 * * `{{project}}/{{location}}/{{name}}`
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, CustomTargetType can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:clouddeploy/customTargetType:CustomTargetType default projects/{{project}}/locations/{{location}}/customTargetTypes/{{name}}
 * $ pulumi import gcp:clouddeploy/customTargetType:CustomTargetType default {{project}}/{{location}}/{{name}}
 * $ pulumi import gcp:clouddeploy/customTargetType:CustomTargetType default {{location}}/{{name}}
 * ```
 */
class CustomTargetType extends pulumi.CustomResource {
    /**
     * Get an existing CustomTargetType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomTargetType(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:clouddeploy/customTargetType:CustomTargetType';
    /**
     * Returns true if the given object is an instance of CustomTargetType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomTargetType.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["annotations"] = state?.annotations;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["customActions"] = state?.customActions;
            resourceInputs["customTargetTypeId"] = state?.customTargetTypeId;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveAnnotations"] = state?.effectiveAnnotations;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["tasks"] = state?.tasks;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["annotations"] = args?.annotations;
            resourceInputs["customActions"] = args?.customActions;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["tasks"] = args?.tasks;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["customTargetTypeId"] = undefined /*out*/;
            resourceInputs["effectiveAnnotations"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(CustomTargetType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomTargetType = CustomTargetType;
