// @pulumi/aws/amplify/domainAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Amplify Domain Association resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.amplify.App("example", {
 *     name: "app",
 *     customRules: [{
 *         source: "https://example.com",
 *         status: "302",
 *         target: "https://www.example.com",
 *     }],
 * });
 * const master = new aws.amplify.Branch("master", {
 *     appId: example.id,
 *     branchName: "master",
 * });
 * const exampleDomainAssociation = new aws.amplify.DomainAssociation("example", {
 *     appId: example.id,
 *     domainName: "example.com",
 *     subDomains: [
 *         {
 *             branchName: master.branchName,
 *             prefix: "",
 *         },
 *         {
 *             branchName: master.branchName,
 *             prefix: "www",
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Amplify domain association using `appId` and `domainName`. For example:
 *
 * ```sh
 * $ pulumi import aws:amplify/domainAssociation:DomainAssociation app d2ypk4k47z8u6/example.com
 * ```
 */
class DomainAssociation extends pulumi.CustomResource {
    /**
     * Get an existing DomainAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DomainAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:amplify/domainAssociation:DomainAssociation';
    /**
     * Returns true if the given object is an instance of DomainAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appId"] = state?.appId;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["certificateSettings"] = state?.certificateSettings;
            resourceInputs["certificateVerificationDnsRecord"] = state?.certificateVerificationDnsRecord;
            resourceInputs["domainName"] = state?.domainName;
            resourceInputs["enableAutoSubDomain"] = state?.enableAutoSubDomain;
            resourceInputs["region"] = state?.region;
            resourceInputs["subDomains"] = state?.subDomains;
            resourceInputs["waitForVerification"] = state?.waitForVerification;
        }
        else {
            const args = argsOrState;
            if (args?.appId === undefined && !opts.urn) {
                throw new Error("Missing required property 'appId'");
            }
            if (args?.domainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if (args?.subDomains === undefined && !opts.urn) {
                throw new Error("Missing required property 'subDomains'");
            }
            resourceInputs["appId"] = args?.appId;
            resourceInputs["certificateSettings"] = args?.certificateSettings;
            resourceInputs["domainName"] = args?.domainName;
            resourceInputs["enableAutoSubDomain"] = args?.enableAutoSubDomain;
            resourceInputs["region"] = args?.region;
            resourceInputs["subDomains"] = args?.subDomains;
            resourceInputs["waitForVerification"] = args?.waitForVerification;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["certificateVerificationDnsRecord"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DomainAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DomainAssociation = DomainAssociation;
