// @pulumi/aws/amplify/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.DomainAssociation = exports.Branch = exports.BackendEnvironment = exports.App = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.BackendEnvironment = null;
utilities.lazyLoad(exports, ["BackendEnvironment"], () => require("./backendEnvironment"));
exports.Branch = null;
utilities.lazyLoad(exports, ["Branch"], () => require("./branch"));
exports.DomainAssociation = null;
utilities.lazyLoad(exports, ["DomainAssociation"], () => require("./domainAssociation"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:amplify/app:App":
                return new exports.App(name, undefined, { urn });
            case "aws:amplify/backendEnvironment:BackendEnvironment":
                return new exports.BackendEnvironment(name, undefined, { urn });
            case "aws:amplify/branch:Branch":
                return new exports.Branch(name, undefined, { urn });
            case "aws:amplify/domainAssociation:DomainAssociation":
                return new exports.DomainAssociation(name, undefined, { urn });
            case "aws:amplify/webhook:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "amplify/app", _module);
pulumi.runtime.registerResourceModule("aws", "amplify/backendEnvironment", _module);
pulumi.runtime.registerResourceModule("aws", "amplify/branch", _module);
pulumi.runtime.registerResourceModule("aws", "amplify/domainAssociation", _module);
pulumi.runtime.registerResourceModule("aws", "amplify/webhook", _module);
