// @pulumi/gcp/securitycenter/v2folderMuteConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2FolderMuteConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Mute Findings is a volume management feature in Security Command Center
 * that lets you manually or programmatically hide irrelevant findings,
 * and create filters to automatically silence existing and future
 * findings based on criteria you specify.
 *
 * To get more information about FolderMuteConfig, see:
 *
 * * [API documentation](https://cloud.google.com/security-command-center/docs/reference/rest/v2/folders.muteConfigs)
 *
 * ## Example Usage
 *
 * ### Scc V2 Folder Mute Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const folder = new gcp.organizations.Folder("folder", {
 *     parent: "organizations/123456789",
 *     displayName: "folder-name",
 * });
 * const _default = new gcp.securitycenter.V2FolderMuteConfig("default", {
 *     muteConfigId: "my-config",
 *     folder: folder.folderId,
 *     location: "global",
 *     description: "My custom Cloud Security Command Center Finding Folder mute Configuration",
 *     filter: "severity = \"HIGH\"",
 *     type: "STATIC",
 * });
 * ```
 *
 * ## Import
 *
 * FolderMuteConfig can be imported using any of these accepted formats:
 *
 * * `folders/{{folder}}/locations/{{location}}/muteConfigs/{{mute_config_id}}`
 * * `{{folder}}/{{location}}/{{mute_config_id}}`
 *
 * When using the `pulumi import` command, FolderMuteConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:securitycenter/v2FolderMuteConfig:V2FolderMuteConfig default folders/{{folder}}/locations/{{location}}/muteConfigs/{{mute_config_id}}
 * $ pulumi import gcp:securitycenter/v2FolderMuteConfig:V2FolderMuteConfig default {{folder}}/{{location}}/{{mute_config_id}}
 * ```
 */
class V2FolderMuteConfig extends pulumi.CustomResource {
    /**
     * Get an existing V2FolderMuteConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new V2FolderMuteConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:securitycenter/v2FolderMuteConfig:V2FolderMuteConfig';
    /**
     * Returns true if the given object is an instance of V2FolderMuteConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === V2FolderMuteConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["filter"] = state?.filter;
            resourceInputs["folder"] = state?.folder;
            resourceInputs["location"] = state?.location;
            resourceInputs["mostRecentEditor"] = state?.mostRecentEditor;
            resourceInputs["muteConfigId"] = state?.muteConfigId;
            resourceInputs["name"] = state?.name;
            resourceInputs["type"] = state?.type;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.filter === undefined && !opts.urn) {
                throw new Error("Missing required property 'filter'");
            }
            if (args?.folder === undefined && !opts.urn) {
                throw new Error("Missing required property 'folder'");
            }
            if (args?.muteConfigId === undefined && !opts.urn) {
                throw new Error("Missing required property 'muteConfigId'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["filter"] = args?.filter;
            resourceInputs["folder"] = args?.folder;
            resourceInputs["location"] = args?.location;
            resourceInputs["muteConfigId"] = args?.muteConfigId;
            resourceInputs["type"] = args?.type;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["mostRecentEditor"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(V2FolderMuteConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.V2FolderMuteConfig = V2FolderMuteConfig;
