// @pulumi/gcp/securitycenter/v2projectSccBigQueryExport.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2ProjectSccBigQueryExport = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A Cloud Security Command Center (Cloud SCC) Big Query Export Config.
 * It represents exporting Security Command Center data, including assets, findings, and security marks
 * using gcloud scc bqexports
 * > **Note:** In order to use Cloud SCC resources, your organization must be enrolled
 * in [SCC Standard/Premium](https://cloud.google.com/security-command-center/docs/quickstart-security-command-center).
 * Without doing so, you may run into errors during resource creation.
 *
 * To get more information about ProjectSccBigQueryExport, see:
 *
 * * [API documentation](https://cloud.google.com/security-command-center/docs/reference/rest/v2/projects.locations.bigQueryExports)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/security-command-center/docs/how-to-analyze-findings-in-big-query)
 *
 * ## Example Usage
 *
 * ### Scc V2 Project Big Query Export Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.bigquery.Dataset("default", {
 *     datasetId: "my_dataset_id",
 *     friendlyName: "test",
 *     description: "This is a test description",
 *     location: "US",
 *     defaultTableExpirationMs: 3600000,
 *     defaultPartitionExpirationMs: null,
 *     labels: {
 *         env: "default",
 *     },
 * });
 * const customBigQueryExportConfig = new gcp.securitycenter.V2ProjectSccBigQueryExport("custom_big_query_export_config", {
 *     name: "my-export",
 *     bigQueryExportId: "my-export",
 *     project: "my-project-name",
 *     dataset: _default.id,
 *     location: "global",
 *     description: "Cloud Security Command Center Findings Big Query Export Config",
 *     filter: "state=\"ACTIVE\" AND NOT mute=\"MUTED\"",
 * });
 * ```
 *
 * ## Import
 *
 * ProjectSccBigQueryExport can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/bigQueryExports/{{big_query_export_id}}`
 * * `{{project}}/{{location}}/{{big_query_export_id}}`
 * * `{{location}}/{{big_query_export_id}}`
 *
 * When using the `pulumi import` command, ProjectSccBigQueryExport can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:securitycenter/v2ProjectSccBigQueryExport:V2ProjectSccBigQueryExport default projects/{{project}}/locations/{{location}}/bigQueryExports/{{big_query_export_id}}
 * $ pulumi import gcp:securitycenter/v2ProjectSccBigQueryExport:V2ProjectSccBigQueryExport default {{project}}/{{location}}/{{big_query_export_id}}
 * $ pulumi import gcp:securitycenter/v2ProjectSccBigQueryExport:V2ProjectSccBigQueryExport default {{location}}/{{big_query_export_id}}
 * ```
 */
class V2ProjectSccBigQueryExport extends pulumi.CustomResource {
    /**
     * Get an existing V2ProjectSccBigQueryExport resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new V2ProjectSccBigQueryExport(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:securitycenter/v2ProjectSccBigQueryExport:V2ProjectSccBigQueryExport';
    /**
     * Returns true if the given object is an instance of V2ProjectSccBigQueryExport.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === V2ProjectSccBigQueryExport.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bigQueryExportId"] = state?.bigQueryExportId;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["dataset"] = state?.dataset;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["filter"] = state?.filter;
            resourceInputs["location"] = state?.location;
            resourceInputs["mostRecentEditor"] = state?.mostRecentEditor;
            resourceInputs["name"] = state?.name;
            resourceInputs["principal"] = state?.principal;
            resourceInputs["project"] = state?.project;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.bigQueryExportId === undefined && !opts.urn) {
                throw new Error("Missing required property 'bigQueryExportId'");
            }
            resourceInputs["bigQueryExportId"] = args?.bigQueryExportId;
            resourceInputs["dataset"] = args?.dataset;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["filter"] = args?.filter;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["mostRecentEditor"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principal"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(V2ProjectSccBigQueryExport.__pulumiType, name, resourceInputs, opts);
    }
}
exports.V2ProjectSccBigQueryExport = V2ProjectSccBigQueryExport;
