// @pulumi/azure-native/types/enums/deviceregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicRetainType = exports.SystemAssignedServiceIdentityType = exports.SupportedKeyType = exports.StreamDestinationTarget = exports.SchemaType = exports.NamespaceDiscoveredManagementActionType = exports.MqttDestinationQos = exports.ManagementActionType = exports.Format = exports.EventObservabilityMode = exports.EventDestinationTarget = exports.DatasetDestinationTarget = exports.DataPointObservabilityMode = exports.AuthenticationMethod = void 0;
exports.AuthenticationMethod = {
    /**
     * The user authentication method is anonymous.
     */
    Anonymous: "Anonymous",
    /**
     * The user authentication method is an x509 certificate.
     */
    Certificate: "Certificate",
    /**
     * The user authentication method is a username and password.
     */
    UsernamePassword: "UsernamePassword",
};
exports.DataPointObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as counter to OpenTelemetry.
     */
    Counter: "Counter",
    /**
     * Map as gauge to OpenTelemetry.
     */
    Gauge: "Gauge",
    /**
     * Map as histogram to OpenTelemetry.
     */
    Histogram: "Histogram",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.DatasetDestinationTarget = {
    /**
     * MQTT target.
     */
    Mqtt: "Mqtt",
    /**
     * Broker State Store target.
     */
    BrokerStateStore: "BrokerStateStore",
    /**
     * Storage target.
     */
    Storage: "Storage",
};
exports.EventDestinationTarget = {
    /**
     * MQTT target.
     */
    Mqtt: "Mqtt",
    /**
     * Storage target.
     */
    Storage: "Storage",
};
exports.EventObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.Format = {
    /**
     * JSON Schema version draft 7 format
     */
    JsonSchema_draft7: "JsonSchema/draft-07",
    /**
     * Delta format
     */
    Delta_1_0: "Delta/1.0",
};
exports.ManagementActionType = {
    /**
     * Call action type.
     */
    Call: "Call",
    /**
     * Read action type.
     */
    Read: "Read",
    /**
     * Write action type.
     */
    Write: "Write",
};
exports.MqttDestinationQos = {
    /**
     * QoS 0.
     */
    Qos0: "Qos0",
    /**
     * QoS 1.
     */
    Qos1: "Qos1",
};
exports.NamespaceDiscoveredManagementActionType = {
    /**
     * Call action type.
     */
    Call: "Call",
    /**
     * Read action type.
     */
    Read: "Read",
    /**
     * Write action type.
     */
    Write: "Write",
};
exports.SchemaType = {
    /**
     * Message Schema schema type
     */
    MessageSchema: "MessageSchema",
};
exports.StreamDestinationTarget = {
    /**
     * MQTT target.
     */
    Mqtt: "Mqtt",
    /**
     * Storage target.
     */
    Storage: "Storage",
};
exports.SupportedKeyType = {
    /**
     * Indicates the ECC key type.
     */
    ECC: "ECC",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.TopicRetainType = {
    /**
     * Retain the messages.
     */
    Keep: "Keep",
    /**
     * Never retain messages.
     */
    Never: "Never",
};
