// @pulumi/gcp/dataproc/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowTemplate = exports.SessionTemplate = exports.MetastoreTableIamPolicy = exports.MetastoreTableIamMember = exports.MetastoreTableIamBinding = exports.MetastoreServiceIamPolicy = exports.MetastoreServiceIamMember = exports.MetastoreServiceIamBinding = exports.MetastoreService = exports.MetastoreFederationIamPolicy = exports.MetastoreFederationIamMember = exports.MetastoreFederationIamBinding = exports.MetastoreFederation = exports.MetastoreDatabaseIamPolicy = exports.MetastoreDatabaseIamMember = exports.MetastoreDatabaseIamBinding = exports.JobIAMPolicy = exports.JobIAMMember = exports.JobIAMBinding = exports.Job = exports.getMetastoreTableIamPolicyOutput = exports.getMetastoreTableIamPolicy = exports.getMetastoreServiceIamPolicyOutput = exports.getMetastoreServiceIamPolicy = exports.getMetastoreServiceOutput = exports.getMetastoreService = exports.getMetastoreFederationIamPolicyOutput = exports.getMetastoreFederationIamPolicy = exports.getMetastoreDatabaseIamPolicyOutput = exports.getMetastoreDatabaseIamPolicy = exports.getJobIamPolicyOutput = exports.getJobIamPolicy = exports.getClusterIamPolicyOutput = exports.getClusterIamPolicy = exports.getAutoscalingPolicyIamPolicyOutput = exports.getAutoscalingPolicyIamPolicy = exports.GdcSparkApplication = exports.GdcServiceInstance = exports.GdcApplicationEnvironment = exports.ClusterIAMPolicy = exports.ClusterIAMMember = exports.ClusterIAMBinding = exports.Cluster = exports.Batch = exports.AutoscalingPolicyIamPolicy = exports.AutoscalingPolicyIamMember = exports.AutoscalingPolicyIamBinding = exports.AutoscalingPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AutoscalingPolicy = null;
utilities.lazyLoad(exports, ["AutoscalingPolicy"], () => require("./autoscalingPolicy"));
exports.AutoscalingPolicyIamBinding = null;
utilities.lazyLoad(exports, ["AutoscalingPolicyIamBinding"], () => require("./autoscalingPolicyIamBinding"));
exports.AutoscalingPolicyIamMember = null;
utilities.lazyLoad(exports, ["AutoscalingPolicyIamMember"], () => require("./autoscalingPolicyIamMember"));
exports.AutoscalingPolicyIamPolicy = null;
utilities.lazyLoad(exports, ["AutoscalingPolicyIamPolicy"], () => require("./autoscalingPolicyIamPolicy"));
exports.Batch = null;
utilities.lazyLoad(exports, ["Batch"], () => require("./batch"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterIAMBinding = null;
utilities.lazyLoad(exports, ["ClusterIAMBinding"], () => require("./clusterIAMBinding"));
exports.ClusterIAMMember = null;
utilities.lazyLoad(exports, ["ClusterIAMMember"], () => require("./clusterIAMMember"));
exports.ClusterIAMPolicy = null;
utilities.lazyLoad(exports, ["ClusterIAMPolicy"], () => require("./clusterIAMPolicy"));
exports.GdcApplicationEnvironment = null;
utilities.lazyLoad(exports, ["GdcApplicationEnvironment"], () => require("./gdcApplicationEnvironment"));
exports.GdcServiceInstance = null;
utilities.lazyLoad(exports, ["GdcServiceInstance"], () => require("./gdcServiceInstance"));
exports.GdcSparkApplication = null;
utilities.lazyLoad(exports, ["GdcSparkApplication"], () => require("./gdcSparkApplication"));
exports.getAutoscalingPolicyIamPolicy = null;
exports.getAutoscalingPolicyIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAutoscalingPolicyIamPolicy", "getAutoscalingPolicyIamPolicyOutput"], () => require("./getAutoscalingPolicyIamPolicy"));
exports.getClusterIamPolicy = null;
exports.getClusterIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getClusterIamPolicy", "getClusterIamPolicyOutput"], () => require("./getClusterIamPolicy"));
exports.getJobIamPolicy = null;
exports.getJobIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getJobIamPolicy", "getJobIamPolicyOutput"], () => require("./getJobIamPolicy"));
exports.getMetastoreDatabaseIamPolicy = null;
exports.getMetastoreDatabaseIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getMetastoreDatabaseIamPolicy", "getMetastoreDatabaseIamPolicyOutput"], () => require("./getMetastoreDatabaseIamPolicy"));
exports.getMetastoreFederationIamPolicy = null;
exports.getMetastoreFederationIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getMetastoreFederationIamPolicy", "getMetastoreFederationIamPolicyOutput"], () => require("./getMetastoreFederationIamPolicy"));
exports.getMetastoreService = null;
exports.getMetastoreServiceOutput = null;
utilities.lazyLoad(exports, ["getMetastoreService", "getMetastoreServiceOutput"], () => require("./getMetastoreService"));
exports.getMetastoreServiceIamPolicy = null;
exports.getMetastoreServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getMetastoreServiceIamPolicy", "getMetastoreServiceIamPolicyOutput"], () => require("./getMetastoreServiceIamPolicy"));
exports.getMetastoreTableIamPolicy = null;
exports.getMetastoreTableIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getMetastoreTableIamPolicy", "getMetastoreTableIamPolicyOutput"], () => require("./getMetastoreTableIamPolicy"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.JobIAMBinding = null;
utilities.lazyLoad(exports, ["JobIAMBinding"], () => require("./jobIAMBinding"));
exports.JobIAMMember = null;
utilities.lazyLoad(exports, ["JobIAMMember"], () => require("./jobIAMMember"));
exports.JobIAMPolicy = null;
utilities.lazyLoad(exports, ["JobIAMPolicy"], () => require("./jobIAMPolicy"));
exports.MetastoreDatabaseIamBinding = null;
utilities.lazyLoad(exports, ["MetastoreDatabaseIamBinding"], () => require("./metastoreDatabaseIamBinding"));
exports.MetastoreDatabaseIamMember = null;
utilities.lazyLoad(exports, ["MetastoreDatabaseIamMember"], () => require("./metastoreDatabaseIamMember"));
exports.MetastoreDatabaseIamPolicy = null;
utilities.lazyLoad(exports, ["MetastoreDatabaseIamPolicy"], () => require("./metastoreDatabaseIamPolicy"));
exports.MetastoreFederation = null;
utilities.lazyLoad(exports, ["MetastoreFederation"], () => require("./metastoreFederation"));
exports.MetastoreFederationIamBinding = null;
utilities.lazyLoad(exports, ["MetastoreFederationIamBinding"], () => require("./metastoreFederationIamBinding"));
exports.MetastoreFederationIamMember = null;
utilities.lazyLoad(exports, ["MetastoreFederationIamMember"], () => require("./metastoreFederationIamMember"));
exports.MetastoreFederationIamPolicy = null;
utilities.lazyLoad(exports, ["MetastoreFederationIamPolicy"], () => require("./metastoreFederationIamPolicy"));
exports.MetastoreService = null;
utilities.lazyLoad(exports, ["MetastoreService"], () => require("./metastoreService"));
exports.MetastoreServiceIamBinding = null;
utilities.lazyLoad(exports, ["MetastoreServiceIamBinding"], () => require("./metastoreServiceIamBinding"));
exports.MetastoreServiceIamMember = null;
utilities.lazyLoad(exports, ["MetastoreServiceIamMember"], () => require("./metastoreServiceIamMember"));
exports.MetastoreServiceIamPolicy = null;
utilities.lazyLoad(exports, ["MetastoreServiceIamPolicy"], () => require("./metastoreServiceIamPolicy"));
exports.MetastoreTableIamBinding = null;
utilities.lazyLoad(exports, ["MetastoreTableIamBinding"], () => require("./metastoreTableIamBinding"));
exports.MetastoreTableIamMember = null;
utilities.lazyLoad(exports, ["MetastoreTableIamMember"], () => require("./metastoreTableIamMember"));
exports.MetastoreTableIamPolicy = null;
utilities.lazyLoad(exports, ["MetastoreTableIamPolicy"], () => require("./metastoreTableIamPolicy"));
exports.SessionTemplate = null;
utilities.lazyLoad(exports, ["SessionTemplate"], () => require("./sessionTemplate"));
exports.WorkflowTemplate = null;
utilities.lazyLoad(exports, ["WorkflowTemplate"], () => require("./workflowTemplate"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:dataproc/autoscalingPolicy:AutoscalingPolicy":
                return new exports.AutoscalingPolicy(name, undefined, { urn });
            case "gcp:dataproc/autoscalingPolicyIamBinding:AutoscalingPolicyIamBinding":
                return new exports.AutoscalingPolicyIamBinding(name, undefined, { urn });
            case "gcp:dataproc/autoscalingPolicyIamMember:AutoscalingPolicyIamMember":
                return new exports.AutoscalingPolicyIamMember(name, undefined, { urn });
            case "gcp:dataproc/autoscalingPolicyIamPolicy:AutoscalingPolicyIamPolicy":
                return new exports.AutoscalingPolicyIamPolicy(name, undefined, { urn });
            case "gcp:dataproc/batch:Batch":
                return new exports.Batch(name, undefined, { urn });
            case "gcp:dataproc/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "gcp:dataproc/clusterIAMBinding:ClusterIAMBinding":
                return new exports.ClusterIAMBinding(name, undefined, { urn });
            case "gcp:dataproc/clusterIAMMember:ClusterIAMMember":
                return new exports.ClusterIAMMember(name, undefined, { urn });
            case "gcp:dataproc/clusterIAMPolicy:ClusterIAMPolicy":
                return new exports.ClusterIAMPolicy(name, undefined, { urn });
            case "gcp:dataproc/gdcApplicationEnvironment:GdcApplicationEnvironment":
                return new exports.GdcApplicationEnvironment(name, undefined, { urn });
            case "gcp:dataproc/gdcServiceInstance:GdcServiceInstance":
                return new exports.GdcServiceInstance(name, undefined, { urn });
            case "gcp:dataproc/gdcSparkApplication:GdcSparkApplication":
                return new exports.GdcSparkApplication(name, undefined, { urn });
            case "gcp:dataproc/job:Job":
                return new exports.Job(name, undefined, { urn });
            case "gcp:dataproc/jobIAMBinding:JobIAMBinding":
                return new exports.JobIAMBinding(name, undefined, { urn });
            case "gcp:dataproc/jobIAMMember:JobIAMMember":
                return new exports.JobIAMMember(name, undefined, { urn });
            case "gcp:dataproc/jobIAMPolicy:JobIAMPolicy":
                return new exports.JobIAMPolicy(name, undefined, { urn });
            case "gcp:dataproc/metastoreDatabaseIamBinding:MetastoreDatabaseIamBinding":
                return new exports.MetastoreDatabaseIamBinding(name, undefined, { urn });
            case "gcp:dataproc/metastoreDatabaseIamMember:MetastoreDatabaseIamMember":
                return new exports.MetastoreDatabaseIamMember(name, undefined, { urn });
            case "gcp:dataproc/metastoreDatabaseIamPolicy:MetastoreDatabaseIamPolicy":
                return new exports.MetastoreDatabaseIamPolicy(name, undefined, { urn });
            case "gcp:dataproc/metastoreFederation:MetastoreFederation":
                return new exports.MetastoreFederation(name, undefined, { urn });
            case "gcp:dataproc/metastoreFederationIamBinding:MetastoreFederationIamBinding":
                return new exports.MetastoreFederationIamBinding(name, undefined, { urn });
            case "gcp:dataproc/metastoreFederationIamMember:MetastoreFederationIamMember":
                return new exports.MetastoreFederationIamMember(name, undefined, { urn });
            case "gcp:dataproc/metastoreFederationIamPolicy:MetastoreFederationIamPolicy":
                return new exports.MetastoreFederationIamPolicy(name, undefined, { urn });
            case "gcp:dataproc/metastoreService:MetastoreService":
                return new exports.MetastoreService(name, undefined, { urn });
            case "gcp:dataproc/metastoreServiceIamBinding:MetastoreServiceIamBinding":
                return new exports.MetastoreServiceIamBinding(name, undefined, { urn });
            case "gcp:dataproc/metastoreServiceIamMember:MetastoreServiceIamMember":
                return new exports.MetastoreServiceIamMember(name, undefined, { urn });
            case "gcp:dataproc/metastoreServiceIamPolicy:MetastoreServiceIamPolicy":
                return new exports.MetastoreServiceIamPolicy(name, undefined, { urn });
            case "gcp:dataproc/metastoreTableIamBinding:MetastoreTableIamBinding":
                return new exports.MetastoreTableIamBinding(name, undefined, { urn });
            case "gcp:dataproc/metastoreTableIamMember:MetastoreTableIamMember":
                return new exports.MetastoreTableIamMember(name, undefined, { urn });
            case "gcp:dataproc/metastoreTableIamPolicy:MetastoreTableIamPolicy":
                return new exports.MetastoreTableIamPolicy(name, undefined, { urn });
            case "gcp:dataproc/sessionTemplate:SessionTemplate":
                return new exports.SessionTemplate(name, undefined, { urn });
            case "gcp:dataproc/workflowTemplate:WorkflowTemplate":
                return new exports.WorkflowTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "dataproc/autoscalingPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/autoscalingPolicyIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/autoscalingPolicyIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/autoscalingPolicyIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/batch", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/cluster", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/clusterIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/clusterIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/clusterIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/gdcApplicationEnvironment", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/gdcServiceInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/gdcSparkApplication", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/job", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/jobIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/jobIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/jobIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreDatabaseIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreDatabaseIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreDatabaseIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreFederation", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreFederationIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreFederationIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreFederationIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreService", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreTableIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreTableIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/metastoreTableIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/sessionTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "dataproc/workflowTemplate", _module);
