// @pulumi/gcp/organizations/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceIdentity = exports.Project = exports.Policy = exports.IAMPolicy = exports.IAMMember = exports.IAMCustomRole = exports.IAMBinding = exports.IamAuditConfig = exports.getSOutput = exports.getS = exports.getProjectOutput = exports.getProject = exports.getOrganizationOutput = exports.getOrganization = exports.getIamCustomRolesOutput = exports.getIamCustomRoles = exports.getIamCustomRoleOutput = exports.getIamCustomRole = exports.getIAMPolicyOutput = exports.getIAMPolicy = exports.getFoldersOutput = exports.getFolders = exports.getFolderOutput = exports.getFolder = exports.getClientOpenIdUserInfoOutput = exports.getClientOpenIdUserInfo = exports.getClientConfigOutput = exports.getClientConfig = exports.getBillingAccountOutput = exports.getBillingAccount = exports.getActiveFolderOutput = exports.getActiveFolder = exports.Folder = exports.AccessApprovalSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AccessApprovalSettings = null;
utilities.lazyLoad(exports, ["AccessApprovalSettings"], () => require("./accessApprovalSettings"));
exports.Folder = null;
utilities.lazyLoad(exports, ["Folder"], () => require("./folder"));
exports.getActiveFolder = null;
exports.getActiveFolderOutput = null;
utilities.lazyLoad(exports, ["getActiveFolder", "getActiveFolderOutput"], () => require("./getActiveFolder"));
exports.getBillingAccount = null;
exports.getBillingAccountOutput = null;
utilities.lazyLoad(exports, ["getBillingAccount", "getBillingAccountOutput"], () => require("./getBillingAccount"));
exports.getClientConfig = null;
exports.getClientConfigOutput = null;
utilities.lazyLoad(exports, ["getClientConfig", "getClientConfigOutput"], () => require("./getClientConfig"));
exports.getClientOpenIdUserInfo = null;
exports.getClientOpenIdUserInfoOutput = null;
utilities.lazyLoad(exports, ["getClientOpenIdUserInfo", "getClientOpenIdUserInfoOutput"], () => require("./getClientOpenIdUserInfo"));
exports.getFolder = null;
exports.getFolderOutput = null;
utilities.lazyLoad(exports, ["getFolder", "getFolderOutput"], () => require("./getFolder"));
exports.getFolders = null;
exports.getFoldersOutput = null;
utilities.lazyLoad(exports, ["getFolders", "getFoldersOutput"], () => require("./getFolders"));
exports.getIAMPolicy = null;
exports.getIAMPolicyOutput = null;
utilities.lazyLoad(exports, ["getIAMPolicy", "getIAMPolicyOutput"], () => require("./getIAMPolicy"));
exports.getIamCustomRole = null;
exports.getIamCustomRoleOutput = null;
utilities.lazyLoad(exports, ["getIamCustomRole", "getIamCustomRoleOutput"], () => require("./getIamCustomRole"));
exports.getIamCustomRoles = null;
exports.getIamCustomRolesOutput = null;
utilities.lazyLoad(exports, ["getIamCustomRoles", "getIamCustomRolesOutput"], () => require("./getIamCustomRoles"));
exports.getOrganization = null;
exports.getOrganizationOutput = null;
utilities.lazyLoad(exports, ["getOrganization", "getOrganizationOutput"], () => require("./getOrganization"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getS = null;
exports.getSOutput = null;
utilities.lazyLoad(exports, ["getS", "getSOutput"], () => require("./getS"));
exports.IamAuditConfig = null;
utilities.lazyLoad(exports, ["IamAuditConfig"], () => require("./iamAuditConfig"));
exports.IAMBinding = null;
utilities.lazyLoad(exports, ["IAMBinding"], () => require("./iambinding"));
exports.IAMCustomRole = null;
utilities.lazyLoad(exports, ["IAMCustomRole"], () => require("./iamcustomRole"));
exports.IAMMember = null;
utilities.lazyLoad(exports, ["IAMMember"], () => require("./iammember"));
exports.IAMPolicy = null;
utilities.lazyLoad(exports, ["IAMPolicy"], () => require("./iampolicy"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ServiceIdentity = null;
utilities.lazyLoad(exports, ["ServiceIdentity"], () => require("./serviceIdentity"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:organizations/accessApprovalSettings:AccessApprovalSettings":
                return new exports.AccessApprovalSettings(name, undefined, { urn });
            case "gcp:organizations/folder:Folder":
                return new exports.Folder(name, undefined, { urn });
            case "gcp:organizations/iAMBinding:IAMBinding":
                return new exports.IAMBinding(name, undefined, { urn });
            case "gcp:organizations/iAMCustomRole:IAMCustomRole":
                return new exports.IAMCustomRole(name, undefined, { urn });
            case "gcp:organizations/iAMMember:IAMMember":
                return new exports.IAMMember(name, undefined, { urn });
            case "gcp:organizations/iAMPolicy:IAMPolicy":
                return new exports.IAMPolicy(name, undefined, { urn });
            case "gcp:organizations/iamAuditConfig:IamAuditConfig":
                return new exports.IamAuditConfig(name, undefined, { urn });
            case "gcp:organizations/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "gcp:organizations/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "gcp:organizations/serviceIdentity:ServiceIdentity":
                return new exports.ServiceIdentity(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "organizations/accessApprovalSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/folder", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/iAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/iAMCustomRole", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/iAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/iAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/iamAuditConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/policy", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/project", _module);
pulumi.runtime.registerResourceModule("gcp", "organizations/serviceIdentity", _module);
