// @pulumi/aws/verifiedpermissions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = exports.PolicyTemplate = exports.PolicyStore = exports.Policy = exports.IdentitySource = exports.getPolicyStoreOutput = exports.getPolicyStore = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getPolicyStore = null;
exports.getPolicyStoreOutput = null;
utilities.lazyLoad(exports, ["getPolicyStore", "getPolicyStoreOutput"], () => require("./getPolicyStore"));
exports.IdentitySource = null;
utilities.lazyLoad(exports, ["IdentitySource"], () => require("./identitySource"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.PolicyStore = null;
utilities.lazyLoad(exports, ["PolicyStore"], () => require("./policyStore"));
exports.PolicyTemplate = null;
utilities.lazyLoad(exports, ["PolicyTemplate"], () => require("./policyTemplate"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:verifiedpermissions/identitySource:IdentitySource":
                return new exports.IdentitySource(name, undefined, { urn });
            case "aws:verifiedpermissions/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "aws:verifiedpermissions/policyStore:PolicyStore":
                return new exports.PolicyStore(name, undefined, { urn });
            case "aws:verifiedpermissions/policyTemplate:PolicyTemplate":
                return new exports.PolicyTemplate(name, undefined, { urn });
            case "aws:verifiedpermissions/schema:Schema":
                return new exports.Schema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "verifiedpermissions/identitySource", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedpermissions/policy", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedpermissions/policyStore", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedpermissions/policyTemplate", _module);
pulumi.runtime.registerResourceModule("aws", "verifiedpermissions/schema", _module);
