// @pulumi/gcp/containeranalysis/occurence.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Occurence = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An occurrence is an instance of a Note, or type of analysis that
 * can be done for a resource.
 *
 * To get more information about Occurrence, see:
 *
 * * [API documentation](https://cloud.google.com/container-analysis/api/reference/rest/)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/container-analysis/)
 *
 * ## Example Usage
 *
 * ### Container Analysis Occurrence Kms
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as std from "@pulumi/std";
 *
 * const note = new gcp.containeranalysis.Note("note", {
 *     name: "attestation-note",
 *     attestationAuthority: {
 *         hint: {
 *             humanReadableName: "Attestor Note",
 *         },
 *     },
 * });
 * const keyring = gcp.kms.getKMSKeyRing({
 *     name: "my-key-ring",
 *     location: "global",
 * });
 * const crypto_key = keyring.then(keyring => gcp.kms.getKMSCryptoKey({
 *     name: "my-key",
 *     keyRing: keyring.id,
 * }));
 * const version = crypto_key.then(crypto_key => gcp.kms.getKMSCryptoKeyVersion({
 *     cryptoKey: crypto_key.id,
 * }));
 * const attestor = new gcp.binaryauthorization.Attestor("attestor", {
 *     name: "attestor",
 *     attestationAuthorityNote: {
 *         noteReference: note.name,
 *         publicKeys: [{
 *             id: version.then(version => version.id),
 *             pkixPublicKey: {
 *                 publicKeyPem: version.then(version => version.publicKeys?.[0]?.pem),
 *                 signatureAlgorithm: version.then(version => version.publicKeys?.[0]?.algorithm),
 *             },
 *         }],
 *     },
 * });
 * const occurrence = new gcp.containeranalysis.Occurence("occurrence", {
 *     resourceUri: "gcr.io/my-project/my-image",
 *     noteName: note.id,
 *     attestation: {
 *         serializedPayload: std.filebase64({
 *             input: "path/to/my/payload.json",
 *         }).then(invoke => invoke.result),
 *         signatures: [{
 *             publicKeyId: version.then(version => version.id),
 *             serializedPayload: std.filebase64({
 *                 input: "path/to/my/payload.json.sig",
 *             }).then(invoke => invoke.result),
 *         }],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Occurrence can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/occurrences/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, Occurrence can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:containeranalysis/occurence:Occurence default projects/{{project}}/occurrences/{{name}}
 * $ pulumi import gcp:containeranalysis/occurence:Occurence default {{project}}/{{name}}
 * $ pulumi import gcp:containeranalysis/occurence:Occurence default {{name}}
 * ```
 */
class Occurence extends pulumi.CustomResource {
    /**
     * Get an existing Occurence resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Occurence(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:containeranalysis/occurence:Occurence';
    /**
     * Returns true if the given object is an instance of Occurence.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Occurence.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["attestation"] = state?.attestation;
            resourceInputs["createTime"] = state?.createTime;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["kind"] = state?.kind;
            resourceInputs["name"] = state?.name;
            resourceInputs["noteName"] = state?.noteName;
            resourceInputs["project"] = state?.project;
            resourceInputs["remediation"] = state?.remediation;
            resourceInputs["resourceUri"] = state?.resourceUri;
            resourceInputs["updateTime"] = state?.updateTime;
        }
        else {
            const args = argsOrState;
            if (args?.attestation === undefined && !opts.urn) {
                throw new Error("Missing required property 'attestation'");
            }
            if (args?.noteName === undefined && !opts.urn) {
                throw new Error("Missing required property 'noteName'");
            }
            if (args?.resourceUri === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceUri'");
            }
            resourceInputs["attestation"] = args?.attestation;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["noteName"] = args?.noteName;
            resourceInputs["project"] = args?.project;
            resourceInputs["remediation"] = args?.remediation;
            resourceInputs["resourceUri"] = args?.resourceUri;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Occurence.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Occurence = Occurence;
