// @pulumi/aws/cloudfront/keyvaluestoreKeysExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyvaluestoreKeysExclusive = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Resource for maintaining exclusive management of resource key value pairs defined in an AWS CloudFront KeyValueStore.
 *
 * !> This resource takes exclusive ownership over key value pairs defined in a KeyValueStore. This includes removal of key value pairs which are not explicitly configured. To prevent persistent drift, ensure any `aws.cloudfront.KeyvaluestoreKey` resources managed alongside this resource have an equivalent `resourceKeyValuePair` argument.
 *
 * > Destruction of this resource means Terraform will no longer manage reconciliation of the configured key value pairs. It __will not__ delete the configured key value pairs from the KeyValueStore.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudfront.KeyValueStore("example", {
 *     name: "ExampleKeyValueStore",
 *     comment: "This is an example key value store",
 * });
 * const exampleKeyvaluestoreKeysExclusive = new aws.cloudfront.KeyvaluestoreKeysExclusive("example", {
 *     keyValueStoreArn: example.arn,
 *     resourceKeyValuePairs: [{
 *         key: "Test Key",
 *         value: "Test Value",
 *     }],
 * });
 * ```
 *
 * ### Disallow Key Value Pairs
 *
 * To automatically remove any configured key value pairs, omit a `resourceKeyValuePair` block.
 *
 * > This will not __prevent__ key value pairs from being defined in a KeyValueStore via Terraform (or any other interface). This resource enables bringing key value pairs into a configured state, however, this reconciliation happens only when `apply` is proactively run.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudfront.KeyvaluestoreKeysExclusive("example", {keyValueStoreArn: exampleAwsCloudfrontKeyValueStore.arn});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import AWS CloudFront KeyValueStore Key Value Pairs using the `keyValueStoreArn`. For example:
 *
 * ```sh
 * $ pulumi import aws:cloudfront/keyvaluestoreKeysExclusive:KeyvaluestoreKeysExclusive example arn:aws:cloudfront::111111111111:key-value-store/8562g61f-caba-2845-9d99-b97diwae5d3c
 * ```
 */
class KeyvaluestoreKeysExclusive extends pulumi.CustomResource {
    /**
     * Get an existing KeyvaluestoreKeysExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KeyvaluestoreKeysExclusive(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:cloudfront/keyvaluestoreKeysExclusive:KeyvaluestoreKeysExclusive';
    /**
     * Returns true if the given object is an instance of KeyvaluestoreKeysExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KeyvaluestoreKeysExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["keyValueStoreArn"] = state?.keyValueStoreArn;
            resourceInputs["maxBatchSize"] = state?.maxBatchSize;
            resourceInputs["resourceKeyValuePairs"] = state?.resourceKeyValuePairs;
            resourceInputs["totalSizeInBytes"] = state?.totalSizeInBytes;
        }
        else {
            const args = argsOrState;
            if (args?.keyValueStoreArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'keyValueStoreArn'");
            }
            resourceInputs["keyValueStoreArn"] = args?.keyValueStoreArn;
            resourceInputs["maxBatchSize"] = args?.maxBatchSize;
            resourceInputs["resourceKeyValuePairs"] = args?.resourceKeyValuePairs;
            resourceInputs["totalSizeInBytes"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(KeyvaluestoreKeysExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KeyvaluestoreKeysExclusive = KeyvaluestoreKeysExclusive;
