// @pulumi/aws/cloudfront/multitenantDistribution.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultitenantDistribution = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Creates an Amazon CloudFront multi-tenant distribution.
 *
 * Multi-tenant distributions are a specialized type of CloudFront distribution designed for multi-tenant applications. They have specific limitations and requirements compared to standard CloudFront distributions.
 *
 * For information about CloudFront multi-tenant distributions, see the [Amazon CloudFront Developer Guide](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/).
 *
 * > **NOTE:** CloudFront distributions take about 15 minutes to reach a deployed state after creation or modification. During this time, deletes to resources will be blocked. If you need to delete a distribution that is enabled and you do not want to wait, you need to use the `retainOnDelete` flag.
 *
 * ## Multi-tenant Distribution Limitations
 *
 * Multi-tenant distributions have the following limitations compared to standard CloudFront distributions:
 *
 * - **Connection Mode**: Automatically set to `tenant-only` and cannot be modified
 * - **Cache Policies**: Must use cache policies instead of legacy TTL settings
 * - **Trusted Key Groups**: Must use trusted key groups instead of trusted signers
 * - **WAF Integration**: Only supports WAF v2 web ACLs
 * - **Certificate Management**: Must use ACM certificates (IAM certificates not supported)
 *
 * ### Unsupported Attributes
 *
 * The following attributes that are available in standard CloudFront distributions are **not supported** for multi-tenant distributions:
 *
 * - `activeTrustedSigners` - Use `activeTrustedKeyGroups` instead
 * - `aliasIcpRecordals` - Managed by connection groups
 * - `aliases` - Managed by connection groups
 * - `anycastIpListId` - Use connection groups instead
 * - `continuousDeploymentPolicyId`
 * - `forwardedValues` in cache behaviors - Deprecated, use cache policies instead
 * - `isIpv6Enabled` - Managed by connection groups
 * - `priceClass` - Managed by connection groups
 * - `smoothStreaming` in cache behaviors
 * - `staging` mode
 * - `trustedSigners` in cache behaviors - Use `trustedKeyGroups` instead
 * - Cache behavior TTL settings (`defaultTtl`, `maxTtl`, `minTtl`) - Use cache policies instead
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudfront.MultitenantDistribution("example", {
 *     comment: "Multi-tenant distribution for my application",
 *     enabled: true,
 *     origins: [{
 *         domainName: "example.com",
 *         id: "example-origin",
 *         customOriginConfigs: [{
 *             httpPort: 80,
 *             httpsPort: 443,
 *             originProtocolPolicy: "https-only",
 *             originSslProtocols: ["TLSv1.2"],
 *         }],
 *     }],
 *     defaultCacheBehavior: {
 *         targetOriginId: "example-origin",
 *         viewerProtocolPolicy: "redirect-to-https",
 *         cachePolicyId: exampleAwsCloudfrontCachePolicy.id,
 *         allowedMethods: {
 *             items: [
 *                 "DELETE",
 *                 "GET",
 *                 "HEAD",
 *                 "OPTIONS",
 *                 "PATCH",
 *                 "POST",
 *                 "PUT",
 *             ],
 *             cachedMethods: [
 *                 "GET",
 *                 "HEAD",
 *             ],
 *         },
 *     },
 *     restrictions: {
 *         geoRestriction: {
 *             restrictionType: "none",
 *         },
 *     },
 *     viewerCertificate: {
 *         acmCertificateArn: exampleAwsAcmCertificate.arn,
 *         sslSupportMethod: "sni-only",
 *     },
 *     tenantConfig: {
 *         parameterDefinitions: [{
 *             name: "origin_domain",
 *             definitions: [{
 *                 stringSchemas: [{
 *                     required: true,
 *                     comment: "Origin domain parameter for tenants",
 *                 }],
 *             }],
 *         }],
 *     },
 *     tags: {
 *         Environment: "production",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import CloudFront Multi-tenant Distributions using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:cloudfront/multitenantDistribution:MultitenantDistribution distribution E74FTE3AEXAMPLE
 * ```
 */
class MultitenantDistribution extends pulumi.CustomResource {
    /**
     * Get an existing MultitenantDistribution resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MultitenantDistribution(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:cloudfront/multitenantDistribution:MultitenantDistribution';
    /**
     * Returns true if the given object is an instance of MultitenantDistribution.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MultitenantDistribution.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["activeTrustedKeyGroups"] = state?.activeTrustedKeyGroups;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["cacheBehaviors"] = state?.cacheBehaviors;
            resourceInputs["callerReference"] = state?.callerReference;
            resourceInputs["comment"] = state?.comment;
            resourceInputs["connectionMode"] = state?.connectionMode;
            resourceInputs["customErrorResponses"] = state?.customErrorResponses;
            resourceInputs["defaultCacheBehavior"] = state?.defaultCacheBehavior;
            resourceInputs["defaultRootObject"] = state?.defaultRootObject;
            resourceInputs["domainName"] = state?.domainName;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["httpVersion"] = state?.httpVersion;
            resourceInputs["inProgressInvalidationBatches"] = state?.inProgressInvalidationBatches;
            resourceInputs["lastModifiedTime"] = state?.lastModifiedTime;
            resourceInputs["originGroups"] = state?.originGroups;
            resourceInputs["origins"] = state?.origins;
            resourceInputs["restrictions"] = state?.restrictions;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["tenantConfig"] = state?.tenantConfig;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["viewerCertificate"] = state?.viewerCertificate;
            resourceInputs["webAclId"] = state?.webAclId;
        }
        else {
            const args = argsOrState;
            if (args?.comment === undefined && !opts.urn) {
                throw new Error("Missing required property 'comment'");
            }
            if (args?.defaultCacheBehavior === undefined && !opts.urn) {
                throw new Error("Missing required property 'defaultCacheBehavior'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.tenantConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'tenantConfig'");
            }
            if (args?.viewerCertificate === undefined && !opts.urn) {
                throw new Error("Missing required property 'viewerCertificate'");
            }
            resourceInputs["activeTrustedKeyGroups"] = args?.activeTrustedKeyGroups;
            resourceInputs["cacheBehaviors"] = args?.cacheBehaviors;
            resourceInputs["comment"] = args?.comment;
            resourceInputs["customErrorResponses"] = args?.customErrorResponses;
            resourceInputs["defaultCacheBehavior"] = args?.defaultCacheBehavior;
            resourceInputs["defaultRootObject"] = args?.defaultRootObject;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["httpVersion"] = args?.httpVersion;
            resourceInputs["originGroups"] = args?.originGroups;
            resourceInputs["origins"] = args?.origins;
            resourceInputs["restrictions"] = args?.restrictions;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tenantConfig"] = args?.tenantConfig;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["viewerCertificate"] = args?.viewerCertificate;
            resourceInputs["webAclId"] = args?.webAclId;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["callerReference"] = undefined /*out*/;
            resourceInputs["connectionMode"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["inProgressInvalidationBatches"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MultitenantDistribution.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MultitenantDistribution = MultitenantDistribution;
