// @pulumi/azure-native/orbital/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Spacecraft = exports.listSpacecraftAvailableContactsOutput = exports.listSpacecraftAvailableContacts = exports.listGroundStationL2ConnectionsOutput = exports.listGroundStationL2Connections = exports.listEdgeSiteL2ConnectionsOutput = exports.listEdgeSiteL2Connections = exports.L2Connection = exports.GroundStation = exports.getSpacecraftOutput = exports.getSpacecraft = exports.getL2ConnectionOutput = exports.getL2Connection = exports.getGroundStationOutput = exports.getGroundStation = exports.getGeoCatalogOutput = exports.getGeoCatalog = exports.getEdgeSiteOutput = exports.getEdgeSite = exports.getContactProfileOutput = exports.getContactProfile = exports.getContactOutput = exports.getContact = exports.GeoCatalog = exports.EdgeSite = exports.ContactProfile = exports.Contact = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Contact = null;
utilities.lazyLoad(exports, ["Contact"], () => require("./contact"));
exports.ContactProfile = null;
utilities.lazyLoad(exports, ["ContactProfile"], () => require("./contactProfile"));
exports.EdgeSite = null;
utilities.lazyLoad(exports, ["EdgeSite"], () => require("./edgeSite"));
exports.GeoCatalog = null;
utilities.lazyLoad(exports, ["GeoCatalog"], () => require("./geoCatalog"));
exports.getContact = null;
exports.getContactOutput = null;
utilities.lazyLoad(exports, ["getContact", "getContactOutput"], () => require("./getContact"));
exports.getContactProfile = null;
exports.getContactProfileOutput = null;
utilities.lazyLoad(exports, ["getContactProfile", "getContactProfileOutput"], () => require("./getContactProfile"));
exports.getEdgeSite = null;
exports.getEdgeSiteOutput = null;
utilities.lazyLoad(exports, ["getEdgeSite", "getEdgeSiteOutput"], () => require("./getEdgeSite"));
exports.getGeoCatalog = null;
exports.getGeoCatalogOutput = null;
utilities.lazyLoad(exports, ["getGeoCatalog", "getGeoCatalogOutput"], () => require("./getGeoCatalog"));
exports.getGroundStation = null;
exports.getGroundStationOutput = null;
utilities.lazyLoad(exports, ["getGroundStation", "getGroundStationOutput"], () => require("./getGroundStation"));
exports.getL2Connection = null;
exports.getL2ConnectionOutput = null;
utilities.lazyLoad(exports, ["getL2Connection", "getL2ConnectionOutput"], () => require("./getL2Connection"));
exports.getSpacecraft = null;
exports.getSpacecraftOutput = null;
utilities.lazyLoad(exports, ["getSpacecraft", "getSpacecraftOutput"], () => require("./getSpacecraft"));
exports.GroundStation = null;
utilities.lazyLoad(exports, ["GroundStation"], () => require("./groundStation"));
exports.L2Connection = null;
utilities.lazyLoad(exports, ["L2Connection"], () => require("./l2connection"));
exports.listEdgeSiteL2Connections = null;
exports.listEdgeSiteL2ConnectionsOutput = null;
utilities.lazyLoad(exports, ["listEdgeSiteL2Connections", "listEdgeSiteL2ConnectionsOutput"], () => require("./listEdgeSiteL2Connections"));
exports.listGroundStationL2Connections = null;
exports.listGroundStationL2ConnectionsOutput = null;
utilities.lazyLoad(exports, ["listGroundStationL2Connections", "listGroundStationL2ConnectionsOutput"], () => require("./listGroundStationL2Connections"));
exports.listSpacecraftAvailableContacts = null;
exports.listSpacecraftAvailableContactsOutput = null;
utilities.lazyLoad(exports, ["listSpacecraftAvailableContacts", "listSpacecraftAvailableContactsOutput"], () => require("./listSpacecraftAvailableContacts"));
exports.Spacecraft = null;
utilities.lazyLoad(exports, ["Spacecraft"], () => require("./spacecraft"));
// Export enums:
__exportStar(require("../types/enums/orbital"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:orbital:Contact":
                return new exports.Contact(name, undefined, { urn });
            case "azure-native:orbital:ContactProfile":
                return new exports.ContactProfile(name, undefined, { urn });
            case "azure-native:orbital:EdgeSite":
                return new exports.EdgeSite(name, undefined, { urn });
            case "azure-native:orbital:GeoCatalog":
                return new exports.GeoCatalog(name, undefined, { urn });
            case "azure-native:orbital:GroundStation":
                return new exports.GroundStation(name, undefined, { urn });
            case "azure-native:orbital:L2Connection":
                return new exports.L2Connection(name, undefined, { urn });
            case "azure-native:orbital:Spacecraft":
                return new exports.Spacecraft(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "orbital", _module);
