// @pulumi/aws/ssmincidents/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsePlan = exports.ReplicationSet = exports.getResponsePlanOutput = exports.getResponsePlan = exports.getReplicationSetOutput = exports.getReplicationSet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getReplicationSet = null;
exports.getReplicationSetOutput = null;
utilities.lazyLoad(exports, ["getReplicationSet", "getReplicationSetOutput"], () => require("./getReplicationSet"));
exports.getResponsePlan = null;
exports.getResponsePlanOutput = null;
utilities.lazyLoad(exports, ["getResponsePlan", "getResponsePlanOutput"], () => require("./getResponsePlan"));
exports.ReplicationSet = null;
utilities.lazyLoad(exports, ["ReplicationSet"], () => require("./replicationSet"));
exports.ResponsePlan = null;
utilities.lazyLoad(exports, ["ResponsePlan"], () => require("./responsePlan"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:ssmincidents/replicationSet:ReplicationSet":
                return new exports.ReplicationSet(name, undefined, { urn });
            case "aws:ssmincidents/responsePlan:ResponsePlan":
                return new exports.ResponsePlan(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "ssmincidents/replicationSet", _module);
pulumi.runtime.registerResourceModule("aws", "ssmincidents/responsePlan", _module);
