// @pulumi/aws/connect/phoneNumberContactFlowAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumberContactFlowAssociation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Associates a flow with a phone number claimed to an Amazon Connect instance.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.connect.PhoneNumberContactFlowAssociation("example", {
 *     phoneNumberId: exampleAwsConnectPhoneNumber.id,
 *     instanceId: exampleAwsConnectInstance.id,
 *     contactFlowId: exampleAwsConnectContactFlow.contactFlowId,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws.connect.PhoneNumberContactFlowAssociation` using the `phoneNumberId`, `instanceId` and `contactFlowId` separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:connect/phoneNumberContactFlowAssociation:PhoneNumberContactFlowAssociation example 36727a4c-4683-4e49-880c-3347c61110a4,fa6c1691-e2eb-4487-bdb9-1aaed6268ebd,c4acdc79-395e-4280-a294-9062f56b07bb
 * ```
 */
class PhoneNumberContactFlowAssociation extends pulumi.CustomResource {
    /**
     * Get an existing PhoneNumberContactFlowAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PhoneNumberContactFlowAssociation(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:connect/phoneNumberContactFlowAssociation:PhoneNumberContactFlowAssociation';
    /**
     * Returns true if the given object is an instance of PhoneNumberContactFlowAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PhoneNumberContactFlowAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["contactFlowId"] = state?.contactFlowId;
            resourceInputs["instanceId"] = state?.instanceId;
            resourceInputs["phoneNumberId"] = state?.phoneNumberId;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.contactFlowId === undefined && !opts.urn) {
                throw new Error("Missing required property 'contactFlowId'");
            }
            if (args?.instanceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'instanceId'");
            }
            if (args?.phoneNumberId === undefined && !opts.urn) {
                throw new Error("Missing required property 'phoneNumberId'");
            }
            resourceInputs["contactFlowId"] = args?.contactFlowId;
            resourceInputs["instanceId"] = args?.instanceId;
            resourceInputs["phoneNumberId"] = args?.phoneNumberId;
            resourceInputs["region"] = args?.region;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PhoneNumberContactFlowAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PhoneNumberContactFlowAssociation = PhoneNumberContactFlowAssociation;
