// @pulumi/azure-native/types/enums/computebulkactions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneDistributionStrategy = exports.WindowsVMGuestPatchMode = exports.WindowsVMGuestPatchAutomaticByPlatformRebootSetting = exports.WindowsPatchAssessmentMode = exports.VirtualMachineType = exports.VMCategory = exports.VMAttributeSupport = exports.StorageAccountTypes = exports.SettingNames = exports.SecurityTypes = exports.SecurityEncryptionTypes = exports.PublicIPAllocationMethod = exports.PublicIPAddressSkuTier = exports.PublicIPAddressSkuName = exports.ProtocolTypes = exports.PassName = exports.OperatingSystemTypes = exports.NetworkInterfaceAuxiliarySku = exports.NetworkInterfaceAuxiliaryMode = exports.NetworkApiVersion = exports.Modes = exports.Mode = exports.ManagedServiceIdentityType = exports.LocalStorageDiskType = exports.LinuxVMGuestPatchMode = exports.LinuxVMGuestPatchAutomaticByPlatformRebootSetting = exports.LinuxPatchAssessmentMode = exports.IPVersions = exports.HyperVGeneration = exports.EvictionPolicy = exports.DomainNameLabelScopeTypes = exports.DiskDetachOptionTypes = exports.DiskDeleteOptionTypes = exports.DiskCreateOptionTypes = exports.DiskControllerTypes = exports.DiffDiskPlacement = exports.DiffDiskOptions = exports.DeleteOptions = exports.CpuManufacturer = exports.ComponentName = exports.CapacityType = exports.CachingTypes = exports.ArchitectureType = exports.AllocationStrategy = exports.AcceleratorType = exports.AcceleratorManufacturer = void 0;
exports.AcceleratorManufacturer = {
    /**
     * AMD GpuType
     */
    AMD: "AMD",
    /**
     * Nvidia GpuType
     */
    Nvidia: "Nvidia",
    /**
     * Xilinx GpuType
     */
    Xilinx: "Xilinx",
};
exports.AcceleratorType = {
    /**
     * GPU Accelerator
     */
    GPU: "GPU",
    /**
     * FPGA Accelerator
     */
    FPGA: "FPGA",
};
exports.AllocationStrategy = {
    /**
     * Default. VM sizes distribution will be determined to optimize for price. Note: Capacity will still be considered here but will be given much less weight.
     */
    LowestPrice: "LowestPrice",
    /**
     * VM sizes distribution will be determined to optimize for capacity.
     */
    CapacityOptimized: "CapacityOptimized",
    /**
     * VM sizes distribution will be determined to optimize for the 'rank' specified for each vm size.
     */
    Prioritized: "Prioritized",
};
exports.ArchitectureType = {
    /**
     * ARM64 Architecture
     */
    ARM64: "ARM64",
    /**
     * X64 Architecture
     */
    X64: "X64",
};
exports.CachingTypes = {
    /**
     * Caching type:None
     */
    None: "None",
    /**
     * Caching type:ReadOnly
     */
    ReadOnly: "ReadOnly",
    /**
     * Caching type:ReadWrite
     */
    ReadWrite: "ReadWrite",
};
exports.CapacityType = {
    /**
     * Default. VM is the default capacity type for LaunchBulkInstancesOperation where capacity is provisioned in terms of VMs.
     */
    VM: "VM",
    /**
     * VCpu is the capacity type for LaunchBulkInstancesOperation where capacity is provisioned in terms of VCpus. If VCpu capacity is not exactly divisible by VCpu count in VMSizes, capacity in VCpus will be overprovisioned by default.
     */
    VCpu: "VCpu",
};
exports.ComponentName = {
    Microsoft_Windows_Shell_Setup: "Microsoft-Windows-Shell-Setup",
};
exports.CpuManufacturer = {
    /**
     * Intel CPU.
     */
    Intel: "Intel",
    /**
     * AMD CPU.
     */
    AMD: "AMD",
    /**
     * Microsoft CPU.
     */
    Microsoft: "Microsoft",
    /**
     * Ampere CPU.
     */
    Ampere: "Ampere",
};
exports.DeleteOptions = {
    /**
     * Delete network interface when the VM is deleted
     */
    Delete: "Delete",
    /**
     * Detach network interface when the VM is deleted
     */
    Detach: "Detach",
};
exports.DiffDiskOptions = {
    /**
     * Local Ephemeral disk option: Local
     */
    Local: "Local",
};
exports.DiffDiskPlacement = {
    /**
     * CacheDisk disk placement
     */
    CacheDisk: "CacheDisk",
    /**
     * ResourceDisk disk placement
     */
    ResourceDisk: "ResourceDisk",
    /**
     * NvmeDisk disk placement
     */
    NvmeDisk: "NvmeDisk",
};
exports.DiskControllerTypes = {
    /**
     * SCSI disk controller type
     */
    SCSI: "SCSI",
    /**
     * NVMe disk controller type
     */
    NVMe: "NVMe",
};
exports.DiskCreateOptionTypes = {
    /**
     * Create disk FromImage
     */
    FromImage: "FromImage",
    /**
     * Empty value
     */
    Empty: "Empty",
    /**
     * Create disk by Attach
     */
    Attach: "Attach",
    /**
     * Create disk by Copy
     */
    Copy: "Copy",
    /**
     * Create disk by Restore
     */
    Restore: "Restore",
};
exports.DiskDeleteOptionTypes = {
    /**
     * Delete the disk upon VM deletion
     */
    Delete: "Delete",
    /**
     * Detach the disk upon VM deletion
     */
    Detach: "Detach",
};
exports.DiskDetachOptionTypes = {
    /**
     * ForceDetach the disk
     */
    ForceDetach: "ForceDetach",
};
exports.DomainNameLabelScopeTypes = {
    /**
     * TenantReuse scope type
     */
    TenantReuse: "TenantReuse",
    /**
     * SubscriptionReuse scope type
     */
    SubscriptionReuse: "SubscriptionReuse",
    /**
     * ResourceGroupReuse scope type
     */
    ResourceGroupReuse: "ResourceGroupReuse",
    /**
     * NoReuse scope type
     */
    NoReuse: "NoReuse",
};
exports.EvictionPolicy = {
    /**
     * When evicted, the Spot VM will be deleted and the corresponding capacity will be updated to reflect this.
     */
    Delete: "Delete",
    /**
     * When evicted, the Spot VM will be deallocated/stopped
     */
    Deallocate: "Deallocate",
};
exports.HyperVGeneration = {
    /**
     * Gen1 hyperV.
     */
    Gen1: "Gen1",
    /**
     * Gen2 hyperV.
     */
    Gen2: "Gen2",
};
exports.IPVersions = {
    /**
     * IPv4 version
     */
    IPv4: "IPv4",
    /**
     * IPv6 version
     */
    IPv6: "IPv6",
};
exports.LinuxPatchAssessmentMode = {
    /**
     * ImageDefault mode
     */
    ImageDefault: "ImageDefault",
    /**
     * AutomaticByPlatform mode
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.LinuxVMGuestPatchAutomaticByPlatformRebootSetting = {
    /**
     * Unknown reboot setting
     */
    Unknown: "Unknown",
    /**
     * Reboot if required
     */
    IfRequired: "IfRequired",
    /**
     * Never reboot
     */
    Never: "Never",
    /**
     * Always reboot
     */
    Always: "Always",
};
exports.LinuxVMGuestPatchMode = {
    /**
     * ImageDefault linux VM guest patch mode
     */
    ImageDefault: "ImageDefault",
    /**
     * AutomaticByPlatform linux VM guest patch mode
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.LocalStorageDiskType = {
    /**
     * HDD DiskType.
     */
    HDD: "HDD",
    /**
     * SSD DiskType.
     */
    SSD: "SSD",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.Mode = {
    /**
     * Audit mode
     */
    Audit: "Audit",
    /**
     * Enforce mode
     */
    Enforce: "Enforce",
};
exports.Modes = {
    /**
     * Audit mode
     */
    Audit: "Audit",
    /**
     * Enforce mode
     */
    Enforce: "Enforce",
    /**
     * Disabled mode
     */
    Disabled: "Disabled",
};
exports.NetworkApiVersion = {
    /**
     * 2020-11-01 version
     */
    NetworkApiVersion_2020_11_01: "2020-11-01",
    /**
     * 2022-11-01 version
     */
    NetworkApiVersion_2022_11_01: "2022-11-01",
};
exports.NetworkInterfaceAuxiliaryMode = {
    /**
     * None mode
     */
    None: "None",
    /**
     * AcceleratedConnections mode
     */
    AcceleratedConnections: "AcceleratedConnections",
    /**
     * Floating mode
     */
    Floating: "Floating",
};
exports.NetworkInterfaceAuxiliarySku = {
    /**
     * None: None sku
     */
    None: "None",
    /**
     * A1 sku
     */
    A1: "A1",
    /**
     * A2 sku
     */
    A2: "A2",
    /**
     * A4 sku
     */
    A4: "A4",
    /**
     * A8 sku
     */
    A8: "A8",
};
exports.OperatingSystemTypes = {
    /**
     * Windows OS
     */
    Windows: "Windows",
    /**
     * Linux OS
     */
    Linux: "Linux",
};
exports.PassName = {
    OobeSystem: "OobeSystem",
};
exports.ProtocolTypes = {
    /**
     * Http protocol
     */
    Http: "Http",
    /**
     * Https protocol
     */
    Https: "Https",
};
exports.PublicIPAddressSkuName = {
    /**
     * Basic IP sku name
     */
    Basic: "Basic",
    /**
     * Standard IP sku name
     */
    Standard: "Standard",
};
exports.PublicIPAddressSkuTier = {
    /**
     * Regional IP address sku tier
     */
    Regional: "Regional",
    /**
     * Global IP address sku tier
     */
    Global: "Global",
};
exports.PublicIPAllocationMethod = {
    /**
     * Dynamic IP allocation
     */
    Dynamic: "Dynamic",
    /**
     * Static IP allocation
     */
    Static: "Static",
};
exports.SecurityEncryptionTypes = {
    /**
     * VMGuestStateOnly encryption
     */
    VMGuestStateOnly: "VMGuestStateOnly",
    /**
     * DiskWithVMGuestState encryption
     */
    DiskWithVMGuestState: "DiskWithVMGuestState",
    /**
     * NonPersistedTPM encryption
     */
    NonPersistedTPM: "NonPersistedTPM",
};
exports.SecurityTypes = {
    /**
     * TrustedLaunch security type
     */
    TrustedLaunch: "TrustedLaunch",
    /**
     * ConfidentialVM security type
     */
    ConfidentialVM: "ConfidentialVM",
};
exports.SettingNames = {
    /**
     * AutoLogon mode
     */
    AutoLogon: "AutoLogon",
    /**
     * FirstLogonCommands mode
     */
    FirstLogonCommands: "FirstLogonCommands",
};
exports.StorageAccountTypes = {
    /**
     * Standard_LRS storage account type
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Premium_LRS storage account type
     */
    Premium_LRS: "Premium_LRS",
    /**
     * StandardSSD_LRS storage account type
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * UltraSSD_LRS storage account type
     */
    UltraSSD_LRS: "UltraSSD_LRS",
    /**
     * Premium_ZRS storage account type
     */
    Premium_ZRS: "Premium_ZRS",
    /**
     * StandardSSD_ZRS storage account type
     */
    StandardSSD_ZRS: "StandardSSD_ZRS",
    /**
     * PremiumV2_LRS storage account type
     */
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.VMAttributeSupport = {
    /**
     * All VMSizes having the feature support will be excluded.
     */
    Excluded: "Excluded",
    /**
     *  VMSizes that have the feature support and that do not have the feature support will be used. Included is a union of Excluded and Required.
     */
    Included: "Included",
    /**
     * Only the VMSizes having the feature support will be used.
     */
    Required: "Required",
};
exports.VMCategory = {
    /**
     *     General purpose VM sizes provide balanced CPU-to-memory ratio. Ideal for testing and development, small to medium databases, and low to medium traffic web servers.
     */
    GeneralPurpose: "GeneralPurpose",
    /**
     *     Compute optimized VM sizes have a high CPU-to-memory ratio. These sizes are good for medium traffic web servers, network appliances, batch processes, and application servers.
     */
    ComputeOptimized: "ComputeOptimized",
    /**
     *     Memory optimized VM sizes offer a high memory-to-CPU ratio that is great for relational database servers, medium to large caches, and in-memory analytics.
     */
    MemoryOptimized: "MemoryOptimized",
    /**
     *     Storage optimized virtual machine (VM) sizes offer high disk throughput and IO, and are ideal for Big Data, SQL, NoSQL databases, data warehousing, and large transactional databases.
     *     Examples include Cassandra, MongoDB, Cloudera, and Redis.
     */
    StorageOptimized: "StorageOptimized",
    /**
     *     GPU optimized VM sizes are specialized virtual machines available with single, multiple, or fractional GPUs.
     *     These sizes are designed for compute-intensive, graphics-intensive, and visualization workloads.
     */
    GpuAccelerated: "GpuAccelerated",
    /**
     *     FPGA optimized VM sizes are specialized virtual machines available with single or multiple FPGA.
     *     These sizes are designed for compute-intensive workloads. This article provides information about the number and type of FPGA, vCPUs, data disks, and NICs.
     *     Storage throughput and network bandwidth are also included for each size in this grouping.
     */
    FpgaAccelerated: "FpgaAccelerated",
    /**
     *     Azure High Performance Compute VMs are optimized for various HPC workloads such as computational fluid dynamics, finite element analysis, frontend and backend EDA,
     *     rendering, molecular dynamics, computational geo science, weather simulation, and financial risk analysis.
     */
    HighPerformanceCompute: "HighPerformanceCompute",
};
exports.VirtualMachineType = {
    /**
     * Default. Regular/On-demand VMs will be launched
     */
    Regular: "Regular",
    /**
     * Spot VMs will be launched.
     */
    Spot: "Spot",
};
exports.WindowsPatchAssessmentMode = {
    /**
     * ImageDefault patch assessment mode
     */
    ImageDefault: "ImageDefault",
    /**
     * AutomaticByPlatform patch assessment mode
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.WindowsVMGuestPatchAutomaticByPlatformRebootSetting = {
    /**
     * Reboot setting for Unknown
     */
    Unknown: "Unknown",
    /**
     * Reboot setting for IfRequired
     */
    IfRequired: "IfRequired",
    /**
     * Reboot setting for Never
     */
    Never: "Never",
    /**
     * Reboot setting for Always
     */
    Always: "Always",
};
exports.WindowsVMGuestPatchMode = {
    /**
     * Manual VM guest patch mode
     */
    Manual: "Manual",
    /**
     * AutomaticByOS VM guest patch mode
     */
    AutomaticByOS: "AutomaticByOS",
    /**
     * AutomaticByPlatform VM guest patch mode
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.ZoneDistributionStrategy = {
    /**
     *     Default. Launch instances in a single zone based on best effort.
     *     If capacity is not available, LaunchBulkInstancesOperation can allocate capacity in different zones.
     */
    BestEffortSingleZone: "BestEffortSingleZone",
    /**
     *     Launch instances based on zone preferences.
     *     Higher priority zones are filled first before allocating to lower priority zones.
     */
    Prioritized: "Prioritized",
    /**
     *     Balance launching instances across zones specified based on best effort.
     *     If capacity is not available, LaunchBulkInstancesOperation can deviate balancing across all zones.
     */
    BestEffortBalanced: "BestEffortBalanced",
    /**
     * Launch instances across all provided zones, ensuring the difference between any two zones is no more than one instance.
     */
    StrictBalanced: "StrictBalanced",
};
