// @pulumi/gcp/discoveryengine/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidgetConfig = exports.UserStore = exports.TargetSite = exports.Sitemap = exports.ServingConfig = exports.SearchEngine = exports.Schema = exports.RecommendationEngine = exports.LicenseConfig = exports.getDataStoresOutput = exports.getDataStores = exports.getDataStoreOutput = exports.getDataStore = exports.DataStore = exports.DataConnector = exports.Control = exports.CmekConfig = exports.ChatEngine = exports.Assistant = exports.AclConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.AclConfig = null;
utilities.lazyLoad(exports, ["AclConfig"], () => require("./aclConfig"));
exports.Assistant = null;
utilities.lazyLoad(exports, ["Assistant"], () => require("./assistant"));
exports.ChatEngine = null;
utilities.lazyLoad(exports, ["ChatEngine"], () => require("./chatEngine"));
exports.CmekConfig = null;
utilities.lazyLoad(exports, ["CmekConfig"], () => require("./cmekConfig"));
exports.Control = null;
utilities.lazyLoad(exports, ["Control"], () => require("./control"));
exports.DataConnector = null;
utilities.lazyLoad(exports, ["DataConnector"], () => require("./dataConnector"));
exports.DataStore = null;
utilities.lazyLoad(exports, ["DataStore"], () => require("./dataStore"));
exports.getDataStore = null;
exports.getDataStoreOutput = null;
utilities.lazyLoad(exports, ["getDataStore", "getDataStoreOutput"], () => require("./getDataStore"));
exports.getDataStores = null;
exports.getDataStoresOutput = null;
utilities.lazyLoad(exports, ["getDataStores", "getDataStoresOutput"], () => require("./getDataStores"));
exports.LicenseConfig = null;
utilities.lazyLoad(exports, ["LicenseConfig"], () => require("./licenseConfig"));
exports.RecommendationEngine = null;
utilities.lazyLoad(exports, ["RecommendationEngine"], () => require("./recommendationEngine"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
exports.SearchEngine = null;
utilities.lazyLoad(exports, ["SearchEngine"], () => require("./searchEngine"));
exports.ServingConfig = null;
utilities.lazyLoad(exports, ["ServingConfig"], () => require("./servingConfig"));
exports.Sitemap = null;
utilities.lazyLoad(exports, ["Sitemap"], () => require("./sitemap"));
exports.TargetSite = null;
utilities.lazyLoad(exports, ["TargetSite"], () => require("./targetSite"));
exports.UserStore = null;
utilities.lazyLoad(exports, ["UserStore"], () => require("./userStore"));
exports.WidgetConfig = null;
utilities.lazyLoad(exports, ["WidgetConfig"], () => require("./widgetConfig"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:discoveryengine/aclConfig:AclConfig":
                return new exports.AclConfig(name, undefined, { urn });
            case "gcp:discoveryengine/assistant:Assistant":
                return new exports.Assistant(name, undefined, { urn });
            case "gcp:discoveryengine/chatEngine:ChatEngine":
                return new exports.ChatEngine(name, undefined, { urn });
            case "gcp:discoveryengine/cmekConfig:CmekConfig":
                return new exports.CmekConfig(name, undefined, { urn });
            case "gcp:discoveryengine/control:Control":
                return new exports.Control(name, undefined, { urn });
            case "gcp:discoveryengine/dataConnector:DataConnector":
                return new exports.DataConnector(name, undefined, { urn });
            case "gcp:discoveryengine/dataStore:DataStore":
                return new exports.DataStore(name, undefined, { urn });
            case "gcp:discoveryengine/licenseConfig:LicenseConfig":
                return new exports.LicenseConfig(name, undefined, { urn });
            case "gcp:discoveryengine/recommendationEngine:RecommendationEngine":
                return new exports.RecommendationEngine(name, undefined, { urn });
            case "gcp:discoveryengine/schema:Schema":
                return new exports.Schema(name, undefined, { urn });
            case "gcp:discoveryengine/searchEngine:SearchEngine":
                return new exports.SearchEngine(name, undefined, { urn });
            case "gcp:discoveryengine/servingConfig:ServingConfig":
                return new exports.ServingConfig(name, undefined, { urn });
            case "gcp:discoveryengine/sitemap:Sitemap":
                return new exports.Sitemap(name, undefined, { urn });
            case "gcp:discoveryengine/targetSite:TargetSite":
                return new exports.TargetSite(name, undefined, { urn });
            case "gcp:discoveryengine/userStore:UserStore":
                return new exports.UserStore(name, undefined, { urn });
            case "gcp:discoveryengine/widgetConfig:WidgetConfig":
                return new exports.WidgetConfig(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/aclConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/assistant", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/chatEngine", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/cmekConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/control", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/dataConnector", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/dataStore", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/licenseConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/recommendationEngine", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/schema", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/searchEngine", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/servingConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/sitemap", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/targetSite", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/userStore", _module);
pulumi.runtime.registerResourceModule("gcp", "discoveryengine/widgetConfig", _module);
