// @pulumi/gcp/discoveryengine/servingConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServingConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a serving config which is a singleton resource under engine. A default
 * serving config is automatically provisioned and deleted with its parent engine.
 *
 * To get more information about ServingConfig, see:
 *
 * * [API documentation](https://cloud.google.com/gemini/enterprise/docs/reference/rest/v1/projects.locations.collections.engines.servingConfigs)
 *
 * ## Import
 *
 * ServingConfig can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}/servingConfigs/{{serving_config_id}}`
 * * `{{project}}/{{location}}/{{collection_id}}/{{engine_id}}/{{serving_config_id}}`
 * * `{{location}}/{{collection_id}}/{{engine_id}}/{{serving_config_id}}`
 *
 * When using the `pulumi import` command, ServingConfig can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:discoveryengine/servingConfig:ServingConfig default projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}/servingConfigs/{{serving_config_id}}
 * $ pulumi import gcp:discoveryengine/servingConfig:ServingConfig default {{project}}/{{location}}/{{collection_id}}/{{engine_id}}/{{serving_config_id}}
 * $ pulumi import gcp:discoveryengine/servingConfig:ServingConfig default {{location}}/{{collection_id}}/{{engine_id}}/{{serving_config_id}}
 * ```
 */
class ServingConfig extends pulumi.CustomResource {
    /**
     * Get an existing ServingConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServingConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:discoveryengine/servingConfig:ServingConfig';
    /**
     * Returns true if the given object is an instance of ServingConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServingConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["boostControlIds"] = state?.boostControlIds;
            resourceInputs["collectionId"] = state?.collectionId;
            resourceInputs["engineId"] = state?.engineId;
            resourceInputs["filterControlIds"] = state?.filterControlIds;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["promoteControlIds"] = state?.promoteControlIds;
            resourceInputs["redirectControlIds"] = state?.redirectControlIds;
            resourceInputs["servingConfigId"] = state?.servingConfigId;
            resourceInputs["synonymsControlIds"] = state?.synonymsControlIds;
        }
        else {
            const args = argsOrState;
            if (args?.engineId === undefined && !opts.urn) {
                throw new Error("Missing required property 'engineId'");
            }
            if (args?.location === undefined && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["boostControlIds"] = args?.boostControlIds;
            resourceInputs["collectionId"] = args?.collectionId;
            resourceInputs["engineId"] = args?.engineId;
            resourceInputs["filterControlIds"] = args?.filterControlIds;
            resourceInputs["location"] = args?.location;
            resourceInputs["project"] = args?.project;
            resourceInputs["promoteControlIds"] = args?.promoteControlIds;
            resourceInputs["redirectControlIds"] = args?.redirectControlIds;
            resourceInputs["servingConfigId"] = args?.servingConfigId;
            resourceInputs["synonymsControlIds"] = args?.synonymsControlIds;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServingConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServingConfig = ServingConfig;
