// @pulumi/azure-native/sql/managedDatabase.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedDatabase = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A managed database resource.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2021-11-01.
 *
 * Other available API versions: 2017-03-01-preview, 2018-06-01-preview, 2019-06-01-preview, 2020-02-02-preview, 2020-08-01-preview, 2020-11-01-preview, 2021-02-01-preview, 2021-05-01-preview, 2021-08-01-preview, 2021-11-01, 2021-11-01-preview, 2022-02-01-preview, 2022-05-01-preview, 2022-08-01-preview, 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01-preview, 2024-05-01-preview, 2024-11-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ManagedDatabase extends pulumi.CustomResource {
    /**
     * Get an existing ManagedDatabase resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedDatabase(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:sql:ManagedDatabase';
    /**
     * Returns true if the given object is an instance of ManagedDatabase.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedDatabase.__pulumiType;
    }
    /**
     * Create a ManagedDatabase resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.managedInstanceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'managedInstanceName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["autoCompleteRestore"] = args?.autoCompleteRestore;
            resourceInputs["catalogCollation"] = args?.catalogCollation;
            resourceInputs["collation"] = args?.collation;
            resourceInputs["createMode"] = args?.createMode;
            resourceInputs["crossSubscriptionRestorableDroppedDatabaseId"] = args?.crossSubscriptionRestorableDroppedDatabaseId;
            resourceInputs["crossSubscriptionSourceDatabaseId"] = args?.crossSubscriptionSourceDatabaseId;
            resourceInputs["crossSubscriptionTargetManagedInstanceId"] = args?.crossSubscriptionTargetManagedInstanceId;
            resourceInputs["databaseName"] = args?.databaseName;
            resourceInputs["isLedgerOn"] = args?.isLedgerOn;
            resourceInputs["lastBackupName"] = args?.lastBackupName;
            resourceInputs["location"] = args?.location;
            resourceInputs["longTermRetentionBackupResourceId"] = args?.longTermRetentionBackupResourceId;
            resourceInputs["managedInstanceName"] = args?.managedInstanceName;
            resourceInputs["recoverableDatabaseId"] = args?.recoverableDatabaseId;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["restorableDroppedDatabaseId"] = args?.restorableDroppedDatabaseId;
            resourceInputs["restorePointInTime"] = args?.restorePointInTime;
            resourceInputs["sourceDatabaseId"] = args?.sourceDatabaseId;
            resourceInputs["storageContainerIdentity"] = args?.storageContainerIdentity;
            resourceInputs["storageContainerSasToken"] = args?.storageContainerSasToken;
            resourceInputs["storageContainerUri"] = args?.storageContainerUri;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["defaultSecondaryLocation"] = undefined /*out*/;
            resourceInputs["earliestRestorePoint"] = undefined /*out*/;
            resourceInputs["failoverGroupId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["catalogCollation"] = undefined /*out*/;
            resourceInputs["collation"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["defaultSecondaryLocation"] = undefined /*out*/;
            resourceInputs["earliestRestorePoint"] = undefined /*out*/;
            resourceInputs["failoverGroupId"] = undefined /*out*/;
            resourceInputs["isLedgerOn"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20170301preview:ManagedDatabase" }, { type: "azure-native:sql/v20180601preview:ManagedDatabase" }, { type: "azure-native:sql/v20190601preview:ManagedDatabase" }, { type: "azure-native:sql/v20200202preview:ManagedDatabase" }, { type: "azure-native:sql/v20200801preview:ManagedDatabase" }, { type: "azure-native:sql/v20201101preview:ManagedDatabase" }, { type: "azure-native:sql/v20210201preview:ManagedDatabase" }, { type: "azure-native:sql/v20210501preview:ManagedDatabase" }, { type: "azure-native:sql/v20210801preview:ManagedDatabase" }, { type: "azure-native:sql/v20211101:ManagedDatabase" }, { type: "azure-native:sql/v20211101preview:ManagedDatabase" }, { type: "azure-native:sql/v20220201preview:ManagedDatabase" }, { type: "azure-native:sql/v20220501preview:ManagedDatabase" }, { type: "azure-native:sql/v20220801preview:ManagedDatabase" }, { type: "azure-native:sql/v20221101preview:ManagedDatabase" }, { type: "azure-native:sql/v20230201preview:ManagedDatabase" }, { type: "azure-native:sql/v20230501preview:ManagedDatabase" }, { type: "azure-native:sql/v20230801:ManagedDatabase" }, { type: "azure-native:sql/v20230801preview:ManagedDatabase" }, { type: "azure-native:sql/v20240501preview:ManagedDatabase" }, { type: "azure-native:sql/v20241101preview:ManagedDatabase" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedDatabase.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagedDatabase = ManagedDatabase;
