// @pulumi/azure-native/sql/startStopManagedInstanceSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartStopManagedInstanceSchedule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Managed instance's Start/Stop schedule.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2022-11-01-preview.
 *
 * Other available API versions: 2022-08-01-preview, 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01-preview, 2024-05-01-preview, 2024-11-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class StartStopManagedInstanceSchedule extends pulumi.CustomResource {
    /**
     * Get an existing StartStopManagedInstanceSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StartStopManagedInstanceSchedule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:sql:StartStopManagedInstanceSchedule';
    /**
     * Returns true if the given object is an instance of StartStopManagedInstanceSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StartStopManagedInstanceSchedule.__pulumiType;
    }
    /**
     * Create a StartStopManagedInstanceSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.managedInstanceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'managedInstanceName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.scheduleList === undefined && !opts.urn) {
                throw new Error("Missing required property 'scheduleList'");
            }
            resourceInputs["description"] = (args?.description) ?? "";
            resourceInputs["managedInstanceName"] = args?.managedInstanceName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["scheduleList"] = args?.scheduleList;
            resourceInputs["startStopScheduleName"] = args?.startStopScheduleName;
            resourceInputs["timeZoneId"] = (args?.timeZoneId) ?? "UTC";
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nextExecutionTime"] = undefined /*out*/;
            resourceInputs["nextRunAction"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nextExecutionTime"] = undefined /*out*/;
            resourceInputs["nextRunAction"] = undefined /*out*/;
            resourceInputs["scheduleList"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeZoneId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20220801preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20221101preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20230201preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20230501preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20230801:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20230801preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20240501preview:StartStopManagedInstanceSchedule" }, { type: "azure-native:sql/v20241101preview:StartStopManagedInstanceSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StartStopManagedInstanceSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StartStopManagedInstanceSchedule = StartStopManagedInstanceSchedule;
