// @pulumi/azure-native/sql/syncMember.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncMember = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * An Azure SQL Database sync member.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2021-11-01.
 *
 * Other available API versions: 2015-05-01-preview, 2019-06-01-preview, 2020-02-02-preview, 2020-08-01-preview, 2020-11-01-preview, 2021-02-01-preview, 2021-05-01-preview, 2021-08-01-preview, 2021-11-01, 2021-11-01-preview, 2022-02-01-preview, 2022-05-01-preview, 2022-08-01-preview, 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01-preview, 2024-05-01-preview, 2024-11-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SyncMember extends pulumi.CustomResource {
    /**
     * Get an existing SyncMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SyncMember(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:sql:SyncMember';
    /**
     * Returns true if the given object is an instance of SyncMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SyncMember.__pulumiType;
    }
    /**
     * Create a SyncMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.databaseName === undefined && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.serverName === undefined && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if (args?.syncGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'syncGroupName'");
            }
            resourceInputs["databaseName"] = args?.databaseName;
            resourceInputs["databaseType"] = args?.databaseType;
            resourceInputs["password"] = args?.password;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["serverName"] = args?.serverName;
            resourceInputs["sqlServerDatabaseId"] = args?.sqlServerDatabaseId;
            resourceInputs["syncAgentId"] = args?.syncAgentId;
            resourceInputs["syncDirection"] = args?.syncDirection;
            resourceInputs["syncGroupName"] = args?.syncGroupName;
            resourceInputs["syncMemberAzureDatabaseResourceId"] = args?.syncMemberAzureDatabaseResourceId;
            resourceInputs["syncMemberName"] = args?.syncMemberName;
            resourceInputs["usePrivateLinkConnection"] = args?.usePrivateLinkConnection;
            resourceInputs["userName"] = args?.userName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointName"] = undefined /*out*/;
            resourceInputs["syncState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["databaseName"] = undefined /*out*/;
            resourceInputs["databaseType"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointName"] = undefined /*out*/;
            resourceInputs["serverName"] = undefined /*out*/;
            resourceInputs["sqlServerDatabaseId"] = undefined /*out*/;
            resourceInputs["syncAgentId"] = undefined /*out*/;
            resourceInputs["syncDirection"] = undefined /*out*/;
            resourceInputs["syncMemberAzureDatabaseResourceId"] = undefined /*out*/;
            resourceInputs["syncState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["usePrivateLinkConnection"] = undefined /*out*/;
            resourceInputs["userName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20150501preview:SyncMember" }, { type: "azure-native:sql/v20190601preview:SyncMember" }, { type: "azure-native:sql/v20200202preview:SyncMember" }, { type: "azure-native:sql/v20200801preview:SyncMember" }, { type: "azure-native:sql/v20201101preview:SyncMember" }, { type: "azure-native:sql/v20210201preview:SyncMember" }, { type: "azure-native:sql/v20210501preview:SyncMember" }, { type: "azure-native:sql/v20210801preview:SyncMember" }, { type: "azure-native:sql/v20211101:SyncMember" }, { type: "azure-native:sql/v20211101preview:SyncMember" }, { type: "azure-native:sql/v20220201preview:SyncMember" }, { type: "azure-native:sql/v20220501preview:SyncMember" }, { type: "azure-native:sql/v20220801preview:SyncMember" }, { type: "azure-native:sql/v20221101preview:SyncMember" }, { type: "azure-native:sql/v20230201preview:SyncMember" }, { type: "azure-native:sql/v20230501preview:SyncMember" }, { type: "azure-native:sql/v20230801:SyncMember" }, { type: "azure-native:sql/v20230801preview:SyncMember" }, { type: "azure-native:sql/v20240501preview:SyncMember" }, { type: "azure-native:sql/v20241101preview:SyncMember" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SyncMember.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SyncMember = SyncMember;
