// @pulumi/aws/dataexchange/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevisionAssets = exports.Revision = exports.EventAction = exports.DataSet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.DataSet = null;
utilities.lazyLoad(exports, ["DataSet"], () => require("./dataSet"));
exports.EventAction = null;
utilities.lazyLoad(exports, ["EventAction"], () => require("./eventAction"));
exports.Revision = null;
utilities.lazyLoad(exports, ["Revision"], () => require("./revision"));
exports.RevisionAssets = null;
utilities.lazyLoad(exports, ["RevisionAssets"], () => require("./revisionAssets"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:dataexchange/dataSet:DataSet":
                return new exports.DataSet(name, undefined, { urn });
            case "aws:dataexchange/eventAction:EventAction":
                return new exports.EventAction(name, undefined, { urn });
            case "aws:dataexchange/revision:Revision":
                return new exports.Revision(name, undefined, { urn });
            case "aws:dataexchange/revisionAssets:RevisionAssets":
                return new exports.RevisionAssets(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "dataexchange/dataSet", _module);
pulumi.runtime.registerResourceModule("aws", "dataexchange/eventAction", _module);
pulumi.runtime.registerResourceModule("aws", "dataexchange/revision", _module);
pulumi.runtime.registerResourceModule("aws", "dataexchange/revisionAssets", _module);
