// @pulumi/gcp/cloudquota/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SQuotaPreference = exports.SQuotaAdjusterSettings = exports.getSQuotaInfosOutput = exports.getSQuotaInfos = exports.getSQuotaInfoOutput = exports.getSQuotaInfo = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.getSQuotaInfo = null;
exports.getSQuotaInfoOutput = null;
utilities.lazyLoad(exports, ["getSQuotaInfo", "getSQuotaInfoOutput"], () => require("./getSQuotaInfo"));
exports.getSQuotaInfos = null;
exports.getSQuotaInfosOutput = null;
utilities.lazyLoad(exports, ["getSQuotaInfos", "getSQuotaInfosOutput"], () => require("./getSQuotaInfos"));
exports.SQuotaAdjusterSettings = null;
utilities.lazyLoad(exports, ["SQuotaAdjusterSettings"], () => require("./squotaAdjusterSettings"));
exports.SQuotaPreference = null;
utilities.lazyLoad(exports, ["SQuotaPreference"], () => require("./squotaPreference"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudquota/sQuotaAdjusterSettings:SQuotaAdjusterSettings":
                return new exports.SQuotaAdjusterSettings(name, undefined, { urn });
            case "gcp:cloudquota/sQuotaPreference:SQuotaPreference":
                return new exports.SQuotaPreference(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudquota/sQuotaAdjusterSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudquota/sQuotaPreference", _module);
