// @pulumi/gcp/activedirectory/domainTrust.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainTrust = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Adds a trust between Active Directory domains
 *
 * To get more information about DomainTrust, see:
 *
 * * [API documentation](https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains/attachTrust)
 * * How-to Guides
 *     * [Active Directory Trust](https://cloud.google.com/managed-microsoft-ad/docs/create-one-way-trust)
 *
 * ## Example Usage
 *
 * ### Active Directory Domain Trust Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const ad_domain_trust = new gcp.activedirectory.DomainTrust("ad-domain-trust", {
 *     domain: "test-managed-ad.com",
 *     targetDomainName: "example-gcp.com",
 *     targetDnsIpAddresses: ["10.1.0.100"],
 *     trustDirection: "OUTBOUND",
 *     trustType: "FOREST",
 *     trustHandshakeSecret: "Testing1!",
 *     deletionProtection: false,
 * });
 * ```
 *
 * ## Import
 *
 * DomainTrust can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/global/domains/{{domain}}/{{target_domain_name}}`
 * * `{{project}}/{{domain}}/{{target_domain_name}}`
 * * `{{domain}}/{{target_domain_name}}`
 *
 * When using the `pulumi import` command, DomainTrust can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:activedirectory/domainTrust:DomainTrust default projects/{{project}}/locations/global/domains/{{domain}}/{{target_domain_name}}
 * $ pulumi import gcp:activedirectory/domainTrust:DomainTrust default {{project}}/{{domain}}/{{target_domain_name}}
 * $ pulumi import gcp:activedirectory/domainTrust:DomainTrust default {{domain}}/{{target_domain_name}}
 * ```
 */
class DomainTrust extends pulumi.CustomResource {
    /**
     * Get an existing DomainTrust resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DomainTrust(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:activedirectory/domainTrust:DomainTrust';
    /**
     * Returns true if the given object is an instance of DomainTrust.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainTrust.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["domain"] = state?.domain;
            resourceInputs["project"] = state?.project;
            resourceInputs["selectiveAuthentication"] = state?.selectiveAuthentication;
            resourceInputs["targetDnsIpAddresses"] = state?.targetDnsIpAddresses;
            resourceInputs["targetDomainName"] = state?.targetDomainName;
            resourceInputs["trustDirection"] = state?.trustDirection;
            resourceInputs["trustHandshakeSecret"] = state?.trustHandshakeSecret;
            resourceInputs["trustType"] = state?.trustType;
        }
        else {
            const args = argsOrState;
            if (args?.domain === undefined && !opts.urn) {
                throw new Error("Missing required property 'domain'");
            }
            if (args?.targetDnsIpAddresses === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetDnsIpAddresses'");
            }
            if (args?.targetDomainName === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetDomainName'");
            }
            if (args?.trustDirection === undefined && !opts.urn) {
                throw new Error("Missing required property 'trustDirection'");
            }
            if (args?.trustHandshakeSecret === undefined && !opts.urn) {
                throw new Error("Missing required property 'trustHandshakeSecret'");
            }
            if (args?.trustType === undefined && !opts.urn) {
                throw new Error("Missing required property 'trustType'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["domain"] = args?.domain;
            resourceInputs["project"] = args?.project;
            resourceInputs["selectiveAuthentication"] = args?.selectiveAuthentication;
            resourceInputs["targetDnsIpAddresses"] = args?.targetDnsIpAddresses;
            resourceInputs["targetDomainName"] = args?.targetDomainName;
            resourceInputs["trustDirection"] = args?.trustDirection;
            resourceInputs["trustHandshakeSecret"] = args?.trustHandshakeSecret ? pulumi.secret(args.trustHandshakeSecret) : undefined;
            resourceInputs["trustType"] = args?.trustType;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["trustHandshakeSecret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(DomainTrust.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DomainTrust = DomainTrust;
