// @pulumi/azure-native/authorization/pimRoleEligibilitySchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PimRoleEligibilitySchedule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A PIM (Privileged Identity Management) Role Eligibility Schedule.
 *
 * Role Eligibility Schedules are used to limit standing administrator access to privileged roles in Azure PIM. See
 * [here](https://learn.microsoft.com/en-us/rest/api/authorization/privileged-role-eligibility-rest-sample) for details.
 *
 * A Role Eligibility Schedule is uniquely defined by scope, principal, and role. At present, only one instance of this
 * resource can exist for a given scope|principal|role tuple.
 *
 * Note that this resource cannot be updated. Each change leads to a recreation.
 *
 * Internally, this resource uses the
 * [Role Eligibility Schedule Requests](https://learn.microsoft.com/en-us/rest/api/authorization/role-eligibility-schedule-requests?view=rest-authorization-2020-10-01)
 * API to create and delete the schedules.
 */
class PimRoleEligibilitySchedule extends pulumi.CustomResource {
    /**
     * Get an existing PimRoleEligibilitySchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PimRoleEligibilitySchedule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:authorization:PimRoleEligibilitySchedule';
    /**
     * Returns true if the given object is an instance of PimRoleEligibilitySchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PimRoleEligibilitySchedule.__pulumiType;
    }
    /**
     * Create a PimRoleEligibilitySchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.principalId === undefined && !opts.urn) {
                throw new Error("Missing required property 'principalId'");
            }
            if (args?.roleDefinitionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleDefinitionId'");
            }
            if (args?.scope === undefined && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            resourceInputs["condition"] = args?.condition;
            resourceInputs["conditionVersion"] = args?.conditionVersion;
            resourceInputs["justification"] = args?.justification;
            resourceInputs["principalId"] = args?.principalId;
            resourceInputs["roleDefinitionId"] = args?.roleDefinitionId;
            resourceInputs["scheduleInfo"] = args?.scheduleInfo;
            resourceInputs["scope"] = args?.scope;
            resourceInputs["targetRoleEligibilityScheduleId"] = args?.targetRoleEligibilityScheduleId;
            resourceInputs["targetRoleEligibilityScheduleInstanceId"] = args?.targetRoleEligibilityScheduleInstanceId;
            resourceInputs["ticketInfo"] = args?.ticketInfo;
            resourceInputs["approvalId"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["expandedProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["requestType"] = undefined /*out*/;
            resourceInputs["requestorId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["approvalId"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["condition"] = undefined /*out*/;
            resourceInputs["conditionVersion"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["expandedProperties"] = undefined /*out*/;
            resourceInputs["justification"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["requestType"] = undefined /*out*/;
            resourceInputs["requestorId"] = undefined /*out*/;
            resourceInputs["roleDefinitionId"] = undefined /*out*/;
            resourceInputs["scheduleInfo"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["targetRoleEligibilityScheduleId"] = undefined /*out*/;
            resourceInputs["targetRoleEligibilityScheduleInstanceId"] = undefined /*out*/;
            resourceInputs["ticketInfo"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:authorization/v20201001:PimRoleEligibilitySchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PimRoleEligibilitySchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PimRoleEligibilitySchedule = PimRoleEligibilitySchedule;
