// @pulumi/aws/s3/bucketPublicAccessBlock.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BucketPublicAccessBlock = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages S3 bucket-level Public Access Block configuration. For more information about these settings, see the [AWS S3 Block Public Access documentation](https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html).
 *
 * > This resource cannot be used with S3 directory buckets.
 *
 * > Setting `skipDestroy` to `true` means that the AWS Provider will not destroy a public access block, even when running `terraform destroy`. The configuration is thus an intentional dangling resource that is not managed by Terraform and will remain in-place in your AWS account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.s3.Bucket("example", {bucket: "example"});
 * const exampleBucketPublicAccessBlock = new aws.s3.BucketPublicAccessBlock("example", {
 *     bucket: example.id,
 *     blockPublicAcls: true,
 *     blockPublicPolicy: true,
 *     ignorePublicAcls: true,
 *     restrictPublicBuckets: true,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws.s3.BucketPublicAccessBlock` using the bucket name. For example:
 *
 * ```sh
 * $ pulumi import aws:s3/bucketPublicAccessBlock:BucketPublicAccessBlock example my-bucket
 * ```
 */
class BucketPublicAccessBlock extends pulumi.CustomResource {
    /**
     * Get an existing BucketPublicAccessBlock resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BucketPublicAccessBlock(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:s3/bucketPublicAccessBlock:BucketPublicAccessBlock';
    /**
     * Returns true if the given object is an instance of BucketPublicAccessBlock.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BucketPublicAccessBlock.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["blockPublicAcls"] = state?.blockPublicAcls;
            resourceInputs["blockPublicPolicy"] = state?.blockPublicPolicy;
            resourceInputs["bucket"] = state?.bucket;
            resourceInputs["ignorePublicAcls"] = state?.ignorePublicAcls;
            resourceInputs["region"] = state?.region;
            resourceInputs["restrictPublicBuckets"] = state?.restrictPublicBuckets;
            resourceInputs["skipDestroy"] = state?.skipDestroy;
        }
        else {
            const args = argsOrState;
            if (args?.bucket === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucket'");
            }
            resourceInputs["blockPublicAcls"] = args?.blockPublicAcls;
            resourceInputs["blockPublicPolicy"] = args?.blockPublicPolicy;
            resourceInputs["bucket"] = args?.bucket;
            resourceInputs["ignorePublicAcls"] = args?.ignorePublicAcls;
            resourceInputs["region"] = args?.region;
            resourceInputs["restrictPublicBuckets"] = args?.restrictPublicBuckets;
            resourceInputs["skipDestroy"] = args?.skipDestroy;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BucketPublicAccessBlock.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BucketPublicAccessBlock = BucketPublicAccessBlock;
