// @pulumi/aws/networkflowmonitor/monitor.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Monitor = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a Network Flow Monitor Monitor.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ec2.Vpc("example", {
 *     cidrBlock: "10.0.0.0/16",
 *     tags: {
 *         Name: "example",
 *     },
 * });
 * const exampleMonitor = new aws.networkflowmonitor.Monitor("example", {
 *     monitorName: "example-monitor",
 *     scopeArn: exampleAwsNetworkflowmonitorScope.scopeArn,
 *     localResources: [{
 *         type: "AWS::EC2::VPC",
 *         identifier: example.arn,
 *     }],
 *     remoteResources: [{
 *         type: "AWS::EC2::VPC",
 *         identifier: example.arn,
 *     }],
 *     tags: {
 *         Name: "example",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Network Flow Monitor Monitor using the monitor name. For example:
 *
 * ```sh
 * $ pulumi import aws:networkflowmonitor/monitor:Monitor example example-monitor
 * ```
 */
class Monitor extends pulumi.CustomResource {
    /**
     * Get an existing Monitor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Monitor(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:networkflowmonitor/monitor:Monitor';
    /**
     * Returns true if the given object is an instance of Monitor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Monitor.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["localResources"] = state?.localResources;
            resourceInputs["monitorArn"] = state?.monitorArn;
            resourceInputs["monitorName"] = state?.monitorName;
            resourceInputs["region"] = state?.region;
            resourceInputs["remoteResources"] = state?.remoteResources;
            resourceInputs["scopeArn"] = state?.scopeArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.localResources === undefined && !opts.urn) {
                throw new Error("Missing required property 'localResources'");
            }
            if (args?.monitorName === undefined && !opts.urn) {
                throw new Error("Missing required property 'monitorName'");
            }
            if (args?.scopeArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'scopeArn'");
            }
            resourceInputs["localResources"] = args?.localResources;
            resourceInputs["monitorName"] = args?.monitorName;
            resourceInputs["region"] = args?.region;
            resourceInputs["remoteResources"] = args?.remoteResources;
            resourceInputs["scopeArn"] = args?.scopeArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["monitorArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Monitor.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Monitor = Monitor;
