// @pulumi/azure-native/migrate/avsAssessmentsOperation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvsAssessmentsOperation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * AVS assessment resource.
 *
 * Uses Azure REST API version 2024-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-03-15.
 *
 * Other available API versions: 2023-03-15, 2023-04-01-preview, 2023-05-01-preview, 2023-09-09-preview, 2024-01-15. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native migrate [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AvsAssessmentsOperation extends pulumi.CustomResource {
    /**
     * Get an existing AvsAssessmentsOperation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AvsAssessmentsOperation(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:migrate:AvsAssessmentsOperation';
    /**
     * Returns true if the given object is an instance of AvsAssessmentsOperation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AvsAssessmentsOperation.__pulumiType;
    }
    /**
     * Create a AvsAssessmentsOperation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.groupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if (args?.projectName === undefined && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assessmentName"] = args?.assessmentName;
            resourceInputs["avsAssessmentScenario"] = args?.avsAssessmentScenario;
            resourceInputs["azureLocation"] = args?.azureLocation;
            resourceInputs["azureOfferCode"] = args?.azureOfferCode;
            resourceInputs["cpuHeadroom"] = args?.cpuHeadroom;
            resourceInputs["currency"] = args?.currency;
            resourceInputs["dedupeCompression"] = args?.dedupeCompression;
            resourceInputs["discountPercentage"] = args?.discountPercentage;
            resourceInputs["externalStorageTypes"] = args?.externalStorageTypes;
            resourceInputs["failuresToTolerateAndRaidLevel"] = args?.failuresToTolerateAndRaidLevel;
            resourceInputs["failuresToTolerateAndRaidLevelList"] = args?.failuresToTolerateAndRaidLevelList;
            resourceInputs["groupName"] = args?.groupName;
            resourceInputs["isStretchClusterEnabled"] = args?.isStretchClusterEnabled;
            resourceInputs["isVcfByolEnabled"] = args?.isVcfByolEnabled;
            resourceInputs["memOvercommit"] = args?.memOvercommit;
            resourceInputs["nodeType"] = args?.nodeType;
            resourceInputs["nodeTypes"] = args?.nodeTypes;
            resourceInputs["percentile"] = args?.percentile;
            resourceInputs["perfDataEndTime"] = args?.perfDataEndTime;
            resourceInputs["perfDataStartTime"] = args?.perfDataStartTime;
            resourceInputs["projectName"] = args?.projectName;
            resourceInputs["provisioningState"] = args?.provisioningState;
            resourceInputs["reservedInstance"] = args?.reservedInstance;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["scalingFactor"] = args?.scalingFactor;
            resourceInputs["sizingCriterion"] = args?.sizingCriterion;
            resourceInputs["timeRange"] = args?.timeRange;
            resourceInputs["vcpuOversubscription"] = args?.vcpuOversubscription;
            resourceInputs["assessmentErrorSummary"] = undefined /*out*/;
            resourceInputs["assessmentType"] = undefined /*out*/;
            resourceInputs["avsEstimatedExternalStorages"] = undefined /*out*/;
            resourceInputs["avsEstimatedNetworks"] = undefined /*out*/;
            resourceInputs["avsEstimatedNodes"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["confidenceRatingInPercentage"] = undefined /*out*/;
            resourceInputs["costComponents"] = undefined /*out*/;
            resourceInputs["cpuUtilization"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["limitingFactor"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfMachines"] = undefined /*out*/;
            resourceInputs["numberOfNodes"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["ramUtilization"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["storageUtilization"] = undefined /*out*/;
            resourceInputs["suitability"] = undefined /*out*/;
            resourceInputs["suitabilityExplanation"] = undefined /*out*/;
            resourceInputs["suitabilitySummary"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["totalCpuCores"] = undefined /*out*/;
            resourceInputs["totalMonthlyCost"] = undefined /*out*/;
            resourceInputs["totalRamInGB"] = undefined /*out*/;
            resourceInputs["totalStorageInGB"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        else {
            resourceInputs["assessmentErrorSummary"] = undefined /*out*/;
            resourceInputs["assessmentType"] = undefined /*out*/;
            resourceInputs["avsAssessmentScenario"] = undefined /*out*/;
            resourceInputs["avsEstimatedExternalStorages"] = undefined /*out*/;
            resourceInputs["avsEstimatedNetworks"] = undefined /*out*/;
            resourceInputs["avsEstimatedNodes"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureLocation"] = undefined /*out*/;
            resourceInputs["azureOfferCode"] = undefined /*out*/;
            resourceInputs["confidenceRatingInPercentage"] = undefined /*out*/;
            resourceInputs["costComponents"] = undefined /*out*/;
            resourceInputs["cpuHeadroom"] = undefined /*out*/;
            resourceInputs["cpuUtilization"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["dedupeCompression"] = undefined /*out*/;
            resourceInputs["discountPercentage"] = undefined /*out*/;
            resourceInputs["externalStorageTypes"] = undefined /*out*/;
            resourceInputs["failuresToTolerateAndRaidLevel"] = undefined /*out*/;
            resourceInputs["failuresToTolerateAndRaidLevelList"] = undefined /*out*/;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["isStretchClusterEnabled"] = undefined /*out*/;
            resourceInputs["isVcfByolEnabled"] = undefined /*out*/;
            resourceInputs["limitingFactor"] = undefined /*out*/;
            resourceInputs["memOvercommit"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeType"] = undefined /*out*/;
            resourceInputs["nodeTypes"] = undefined /*out*/;
            resourceInputs["numberOfMachines"] = undefined /*out*/;
            resourceInputs["numberOfNodes"] = undefined /*out*/;
            resourceInputs["percentile"] = undefined /*out*/;
            resourceInputs["perfDataEndTime"] = undefined /*out*/;
            resourceInputs["perfDataStartTime"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["ramUtilization"] = undefined /*out*/;
            resourceInputs["reservedInstance"] = undefined /*out*/;
            resourceInputs["scalingFactor"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["sizingCriterion"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["storageUtilization"] = undefined /*out*/;
            resourceInputs["suitability"] = undefined /*out*/;
            resourceInputs["suitabilityExplanation"] = undefined /*out*/;
            resourceInputs["suitabilitySummary"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeRange"] = undefined /*out*/;
            resourceInputs["totalCpuCores"] = undefined /*out*/;
            resourceInputs["totalMonthlyCost"] = undefined /*out*/;
            resourceInputs["totalRamInGB"] = undefined /*out*/;
            resourceInputs["totalStorageInGB"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
            resourceInputs["vcpuOversubscription"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20230315:AvsAssessmentsOperation" }, { type: "azure-native:migrate/v20230401preview:AvsAssessmentsOperation" }, { type: "azure-native:migrate/v20230501preview:AvsAssessmentsOperation" }, { type: "azure-native:migrate/v20230909preview:AvsAssessmentsOperation" }, { type: "azure-native:migrate/v20240101preview:AvsAssessmentsOperation" }, { type: "azure-native:migrate/v20240115:AvsAssessmentsOperation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AvsAssessmentsOperation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AvsAssessmentsOperation = AvsAssessmentsOperation;
