// @pulumi/azure-native/migrate/sqlAssessmentV2Operation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlAssessmentV2Operation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * SQL Assessment REST resource.
 *
 * Uses Azure REST API version 2024-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-03-15.
 *
 * Other available API versions: 2023-03-15, 2023-04-01-preview, 2023-05-01-preview, 2023-09-09-preview, 2024-01-15. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native migrate [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SqlAssessmentV2Operation extends pulumi.CustomResource {
    /**
     * Get an existing SqlAssessmentV2Operation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlAssessmentV2Operation(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:migrate:SqlAssessmentV2Operation';
    /**
     * Returns true if the given object is an instance of SqlAssessmentV2Operation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlAssessmentV2Operation.__pulumiType;
    }
    /**
     * Create a SqlAssessmentV2Operation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.groupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if (args?.projectName === undefined && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assessmentName"] = args?.assessmentName;
            resourceInputs["assessmentType"] = args?.assessmentType;
            resourceInputs["asyncCommitModeIntent"] = args?.asyncCommitModeIntent;
            resourceInputs["azureLocation"] = args?.azureLocation;
            resourceInputs["azureOfferCode"] = args?.azureOfferCode;
            resourceInputs["azureOfferCodeForVm"] = args?.azureOfferCodeForVm;
            resourceInputs["azureSecurityOfferingType"] = args?.azureSecurityOfferingType;
            resourceInputs["azureSqlDatabaseSettings"] = args?.azureSqlDatabaseSettings;
            resourceInputs["azureSqlManagedInstanceSettings"] = args?.azureSqlManagedInstanceSettings;
            resourceInputs["azureSqlVmSettings"] = args?.azureSqlVmSettings;
            resourceInputs["confidenceRatingInPercentage"] = args?.confidenceRatingInPercentage;
            resourceInputs["currency"] = args?.currency;
            resourceInputs["disasterRecoveryLocation"] = args?.disasterRecoveryLocation;
            resourceInputs["discountPercentage"] = args?.discountPercentage;
            resourceInputs["eaSubscriptionId"] = args?.eaSubscriptionId;
            resourceInputs["enableHadrAssessment"] = args?.enableHadrAssessment;
            resourceInputs["entityUptime"] = args?.entityUptime;
            resourceInputs["environmentType"] = args?.environmentType;
            resourceInputs["groupName"] = args?.groupName;
            resourceInputs["groupType"] = args?.groupType;
            resourceInputs["isInternetAccessAvailable"] = args?.isInternetAccessAvailable;
            resourceInputs["multiSubnetIntent"] = args?.multiSubnetIntent;
            resourceInputs["optimizationLogic"] = args?.optimizationLogic;
            resourceInputs["osLicense"] = args?.osLicense;
            resourceInputs["percentile"] = args?.percentile;
            resourceInputs["perfDataEndTime"] = args?.perfDataEndTime;
            resourceInputs["perfDataStartTime"] = args?.perfDataStartTime;
            resourceInputs["projectName"] = args?.projectName;
            resourceInputs["provisioningState"] = args?.provisioningState;
            resourceInputs["reservedInstance"] = args?.reservedInstance;
            resourceInputs["reservedInstanceForVm"] = args?.reservedInstanceForVm;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["scalingFactor"] = args?.scalingFactor;
            resourceInputs["sizingCriterion"] = args?.sizingCriterion;
            resourceInputs["sqlServerLicense"] = args?.sqlServerLicense;
            resourceInputs["timeRange"] = args?.timeRange;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        else {
            resourceInputs["assessmentType"] = undefined /*out*/;
            resourceInputs["asyncCommitModeIntent"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureLocation"] = undefined /*out*/;
            resourceInputs["azureOfferCode"] = undefined /*out*/;
            resourceInputs["azureOfferCodeForVm"] = undefined /*out*/;
            resourceInputs["azureSecurityOfferingType"] = undefined /*out*/;
            resourceInputs["azureSqlDatabaseSettings"] = undefined /*out*/;
            resourceInputs["azureSqlManagedInstanceSettings"] = undefined /*out*/;
            resourceInputs["azureSqlVmSettings"] = undefined /*out*/;
            resourceInputs["confidenceRatingInPercentage"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["disasterRecoveryLocation"] = undefined /*out*/;
            resourceInputs["discountPercentage"] = undefined /*out*/;
            resourceInputs["eaSubscriptionId"] = undefined /*out*/;
            resourceInputs["enableHadrAssessment"] = undefined /*out*/;
            resourceInputs["entityUptime"] = undefined /*out*/;
            resourceInputs["environmentType"] = undefined /*out*/;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["isInternetAccessAvailable"] = undefined /*out*/;
            resourceInputs["multiSubnetIntent"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["optimizationLogic"] = undefined /*out*/;
            resourceInputs["osLicense"] = undefined /*out*/;
            resourceInputs["percentile"] = undefined /*out*/;
            resourceInputs["perfDataEndTime"] = undefined /*out*/;
            resourceInputs["perfDataStartTime"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["reservedInstance"] = undefined /*out*/;
            resourceInputs["reservedInstanceForVm"] = undefined /*out*/;
            resourceInputs["scalingFactor"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["sizingCriterion"] = undefined /*out*/;
            resourceInputs["sqlServerLicense"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeRange"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20230315:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230401preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230501preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230909preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20240101preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20240115:SqlAssessmentV2Operation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlAssessmentV2Operation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlAssessmentV2Operation = SqlAssessmentV2Operation;
