// @pulumi/azure-native/netapp/capacityPoolBackup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPoolBackup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Backup of a Volume
 *
 * Uses Azure REST API version 2022-11-01.
 */
class CapacityPoolBackup extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPoolBackup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPoolBackup(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:netapp:CapacityPoolBackup';
    /**
     * Returns true if the given object is an instance of CapacityPoolBackup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPoolBackup.__pulumiType;
    }
    /**
     * Create a CapacityPoolBackup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.poolName === undefined && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.volumeName === undefined && !opts.urn) {
                throw new Error("Missing required property 'volumeName'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["backupName"] = args?.backupName;
            resourceInputs["label"] = args?.label;
            resourceInputs["location"] = args?.location;
            resourceInputs["poolName"] = args?.poolName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["useExistingSnapshot"] = (args?.useExistingSnapshot) ?? false;
            resourceInputs["volumeName"] = args?.volumeName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupId"] = undefined /*out*/;
            resourceInputs["backupType"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupId"] = undefined /*out*/;
            resourceInputs["backupType"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["label"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useExistingSnapshot"] = undefined /*out*/;
            resourceInputs["volumeName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20200501:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200601:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200701:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200801:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200901:CapacityPoolBackup" }, { type: "azure-native:netapp/v20201101:CapacityPoolBackup" }, { type: "azure-native:netapp/v20201201:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210201:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210401:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210401preview:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210601:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210801:CapacityPoolBackup" }, { type: "azure-native:netapp/v20211001:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220101:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220301:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220501:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220901:CapacityPoolBackup" }, { type: "azure-native:netapp/v20221101:Backup" }, { type: "azure-native:netapp/v20221101:CapacityPoolBackup" }, { type: "azure-native:netapp:Backup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPoolBackup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPoolBackup = CapacityPoolBackup;
