// @pulumi/azure-native/netapp/capacityPoolSnapshot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPoolSnapshot = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Snapshot of a Volume
 *
 * Uses Azure REST API version 2024-09-01.
 *
 * Other available API versions: 2022-11-01, 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01-preview, 2025-01-01, 2025-01-01-preview, 2025-03-01, 2025-03-01-preview, 2025-06-01, 2025-07-01-preview, 2025-08-01, 2025-08-01-preview, 2025-09-01, 2025-09-01-preview, 2025-12-01, 2025-12-15-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native netapp [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CapacityPoolSnapshot extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPoolSnapshot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPoolSnapshot(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:netapp:CapacityPoolSnapshot';
    /**
     * Returns true if the given object is an instance of CapacityPoolSnapshot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPoolSnapshot.__pulumiType;
    }
    /**
     * Create a CapacityPoolSnapshot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.poolName === undefined && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.volumeName === undefined && !opts.urn) {
                throw new Error("Missing required property 'volumeName'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["location"] = args?.location;
            resourceInputs["poolName"] = args?.poolName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["snapshotName"] = args?.snapshotName;
            resourceInputs["volumeName"] = args?.volumeName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["snapshotId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["snapshotId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20170815:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20190501:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20190601:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20190701:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20190801:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20191001:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20191101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200201:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200301:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200501:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200601:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200701:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200801:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20200901:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20201101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20201201:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20210201:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20210401:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20210401preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20210601:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20210801:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20211001:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20220101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20220301:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20220501:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20220901:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20221101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20221101:Snapshot" }, { type: "azure-native:netapp/v20221101preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20221101preview:Snapshot" }, { type: "azure-native:netapp/v20230501:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20230501:Snapshot" }, { type: "azure-native:netapp/v20230501preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20230501preview:Snapshot" }, { type: "azure-native:netapp/v20230701:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20230701:Snapshot" }, { type: "azure-native:netapp/v20230701preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20230701preview:Snapshot" }, { type: "azure-native:netapp/v20231101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20231101:Snapshot" }, { type: "azure-native:netapp/v20231101preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20231101preview:Snapshot" }, { type: "azure-native:netapp/v20240101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240101:Snapshot" }, { type: "azure-native:netapp/v20240301:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240301:Snapshot" }, { type: "azure-native:netapp/v20240301preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240301preview:Snapshot" }, { type: "azure-native:netapp/v20240501:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240501:Snapshot" }, { type: "azure-native:netapp/v20240501preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240501preview:Snapshot" }, { type: "azure-native:netapp/v20240701:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240701:Snapshot" }, { type: "azure-native:netapp/v20240701preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240701preview:Snapshot" }, { type: "azure-native:netapp/v20240901:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20240901:Snapshot" }, { type: "azure-native:netapp/v20240901preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250101:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250101preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250301:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250301preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250601:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250701preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250801:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250801preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250901:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20250901preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20251201:CapacityPoolSnapshot" }, { type: "azure-native:netapp/v20251215preview:CapacityPoolSnapshot" }, { type: "azure-native:netapp:Snapshot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPoolSnapshot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPoolSnapshot = CapacityPoolSnapshot;
