// @pulumi/azure-native/databoxedge/ioTRole.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoTRole = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Compute role.
 *
 * Uses Azure REST API version 2023-07-01. In version 2.x of the Azure Native provider, it used API version 2022-03-01.
 */
class IoTRole extends pulumi.CustomResource {
    /**
     * Get an existing IoTRole resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IoTRole(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:databoxedge:IoTRole';
    /**
     * Returns true if the given object is an instance of IoTRole.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IoTRole.__pulumiType;
    }
    /**
     * Create a IoTRole resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.deviceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if (args?.hostPlatform === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostPlatform'");
            }
            if (args?.ioTDeviceDetails === undefined && !opts.urn) {
                throw new Error("Missing required property 'ioTDeviceDetails'");
            }
            if (args?.ioTEdgeDeviceDetails === undefined && !opts.urn) {
                throw new Error("Missing required property 'ioTEdgeDeviceDetails'");
            }
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.roleStatus === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleStatus'");
            }
            resourceInputs["computeResource"] = args?.computeResource;
            resourceInputs["deviceName"] = args?.deviceName;
            resourceInputs["hostPlatform"] = args?.hostPlatform;
            resourceInputs["ioTDeviceDetails"] = args?.ioTDeviceDetails;
            resourceInputs["ioTEdgeAgentInfo"] = args?.ioTEdgeAgentInfo;
            resourceInputs["ioTEdgeDeviceDetails"] = args?.ioTEdgeDeviceDetails;
            resourceInputs["kind"] = "IOT";
            resourceInputs["name"] = args?.name;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["roleStatus"] = args?.roleStatus;
            resourceInputs["shareMappings"] = args?.shareMappings;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["hostPlatformType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["computeResource"] = undefined /*out*/;
            resourceInputs["hostPlatform"] = undefined /*out*/;
            resourceInputs["hostPlatformType"] = undefined /*out*/;
            resourceInputs["ioTDeviceDetails"] = undefined /*out*/;
            resourceInputs["ioTEdgeAgentInfo"] = undefined /*out*/;
            resourceInputs["ioTEdgeDeviceDetails"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["roleStatus"] = undefined /*out*/;
            resourceInputs["shareMappings"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge/v20190301:IoTRole" }, { type: "azure-native:databoxedge/v20190701:IoTRole" }, { type: "azure-native:databoxedge/v20190801:IoTRole" }, { type: "azure-native:databoxedge/v20200501preview:IoTRole" }, { type: "azure-native:databoxedge/v20200901:IoTRole" }, { type: "azure-native:databoxedge/v20200901preview:IoTRole" }, { type: "azure-native:databoxedge/v20201201:IoTRole" }, { type: "azure-native:databoxedge/v20210201:IoTRole" }, { type: "azure-native:databoxedge/v20210201preview:IoTRole" }, { type: "azure-native:databoxedge/v20210601:IoTRole" }, { type: "azure-native:databoxedge/v20210601preview:IoTRole" }, { type: "azure-native:databoxedge/v20220301:IoTRole" }, { type: "azure-native:databoxedge/v20220401preview:IoTRole" }, { type: "azure-native:databoxedge/v20221201preview:IoTRole" }, { type: "azure-native:databoxedge/v20230101preview:CloudEdgeManagementRole" }, { type: "azure-native:databoxedge/v20230101preview:IoTRole" }, { type: "azure-native:databoxedge/v20230101preview:KubernetesRole" }, { type: "azure-native:databoxedge/v20230101preview:MECRole" }, { type: "azure-native:databoxedge/v20230701:CloudEdgeManagementRole" }, { type: "azure-native:databoxedge/v20230701:IoTRole" }, { type: "azure-native:databoxedge/v20230701:KubernetesRole" }, { type: "azure-native:databoxedge/v20230701:MECRole" }, { type: "azure-native:databoxedge/v20231201:CloudEdgeManagementRole" }, { type: "azure-native:databoxedge/v20231201:IoTRole" }, { type: "azure-native:databoxedge/v20231201:KubernetesRole" }, { type: "azure-native:databoxedge/v20231201:MECRole" }, { type: "azure-native:databoxedge:CloudEdgeManagementRole" }, { type: "azure-native:databoxedge:KubernetesRole" }, { type: "azure-native:databoxedge:MECRole" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IoTRole.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IoTRole = IoTRole;
