// @pulumi/azure-native/databoxedge/periodicTimerEventTrigger.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PeriodicTimerEventTrigger = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Trigger details.
 *
 * Uses Azure REST API version 2023-07-01. In version 2.x of the Azure Native provider, it used API version 2022-03-01.
 */
class PeriodicTimerEventTrigger extends pulumi.CustomResource {
    /**
     * Get an existing PeriodicTimerEventTrigger resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PeriodicTimerEventTrigger(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:databoxedge:PeriodicTimerEventTrigger';
    /**
     * Returns true if the given object is an instance of PeriodicTimerEventTrigger.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PeriodicTimerEventTrigger.__pulumiType;
    }
    /**
     * Create a PeriodicTimerEventTrigger resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.deviceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.sinkInfo === undefined && !opts.urn) {
                throw new Error("Missing required property 'sinkInfo'");
            }
            if (args?.sourceInfo === undefined && !opts.urn) {
                throw new Error("Missing required property 'sourceInfo'");
            }
            resourceInputs["customContextTag"] = args?.customContextTag;
            resourceInputs["deviceName"] = args?.deviceName;
            resourceInputs["kind"] = "PeriodicTimerEvent";
            resourceInputs["name"] = args?.name;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sinkInfo"] = args?.sinkInfo;
            resourceInputs["sourceInfo"] = args?.sourceInfo;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["customContextTag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["sinkInfo"] = undefined /*out*/;
            resourceInputs["sourceInfo"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge/v20190301:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20190701:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20190801:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20200501preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20200901:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20200901preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20201201:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20210201:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20210201preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20210601:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20210601preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20220301:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20220401preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20221201preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20230101preview:FileEventTrigger" }, { type: "azure-native:databoxedge/v20230101preview:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20230701:FileEventTrigger" }, { type: "azure-native:databoxedge/v20230701:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge/v20231201:FileEventTrigger" }, { type: "azure-native:databoxedge/v20231201:PeriodicTimerEventTrigger" }, { type: "azure-native:databoxedge:FileEventTrigger" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PeriodicTimerEventTrigger.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PeriodicTimerEventTrigger = PeriodicTimerEventTrigger;
