// @pulumi/aws/lightsail/lbStickinessPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LbStickinessPolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages session stickiness for a Lightsail Load Balancer.
 *
 * Use this resource to configure session stickiness to ensure that user sessions are consistently routed to the same backend instance. This helps maintain session state for applications that store session data locally on the server.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.lightsail.Lb("example", {
 *     name: "example-load-balancer",
 *     healthCheckPath: "/",
 *     instancePort: 80,
 *     tags: {
 *         foo: "bar",
 *     },
 * });
 * const exampleLbStickinessPolicy = new aws.lightsail.LbStickinessPolicy("example", {
 *     lbName: example.name,
 *     cookieDuration: 900,
 *     enabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws.lightsail.LbStickinessPolicy` using the `lbName` attribute. For example:
 *
 * ```sh
 * $ pulumi import aws:lightsail/lbStickinessPolicy:LbStickinessPolicy example example-load-balancer
 * ```
 */
class LbStickinessPolicy extends pulumi.CustomResource {
    /**
     * Get an existing LbStickinessPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LbStickinessPolicy(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:lightsail/lbStickinessPolicy:LbStickinessPolicy';
    /**
     * Returns true if the given object is an instance of LbStickinessPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LbStickinessPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cookieDuration"] = state?.cookieDuration;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["lbName"] = state?.lbName;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.cookieDuration === undefined && !opts.urn) {
                throw new Error("Missing required property 'cookieDuration'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.lbName === undefined && !opts.urn) {
                throw new Error("Missing required property 'lbName'");
            }
            resourceInputs["cookieDuration"] = args?.cookieDuration;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["lbName"] = args?.lbName;
            resourceInputs["region"] = args?.region;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LbStickinessPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LbStickinessPolicy = LbStickinessPolicy;
