// @pulumi/azure-native/azurestackhci/arcSetting.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArcSetting = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * ArcSetting details.
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2023-03-01.
 *
 * Other available API versions: 2022-12-15-preview, 2023-02-01, 2023-03-01, 2023-06-01, 2023-08-01, 2023-08-01-preview, 2023-11-01-preview, 2024-01-01, 2024-02-15-preview, 2024-09-01-preview, 2024-12-01-preview, 2025-02-01-preview, 2025-09-15-preview, 2025-10-01, 2025-11-01-preview, 2025-12-01-preview, 2026-02-01, 2026-02-15-preview, 2026-03-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native azurestackhci [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ArcSetting extends pulumi.CustomResource {
    /**
     * Get an existing ArcSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ArcSetting(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:azurestackhci:ArcSetting';
    /**
     * Returns true if the given object is an instance of ArcSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ArcSetting.__pulumiType;
    }
    /**
     * Create a ArcSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["arcApplicationClientId"] = args?.arcApplicationClientId;
            resourceInputs["arcApplicationObjectId"] = args?.arcApplicationObjectId;
            resourceInputs["arcApplicationTenantId"] = args?.arcApplicationTenantId;
            resourceInputs["arcInstanceResourceGroup"] = args?.arcInstanceResourceGroup;
            resourceInputs["arcServicePrincipalObjectId"] = args?.arcServicePrincipalObjectId;
            resourceInputs["arcSettingName"] = args?.arcSettingName;
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["connectivityProperties"] = args?.connectivityProperties;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["aggregateState"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["defaultExtensions"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["perNodeDetails"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["aggregateState"] = undefined /*out*/;
            resourceInputs["arcApplicationClientId"] = undefined /*out*/;
            resourceInputs["arcApplicationObjectId"] = undefined /*out*/;
            resourceInputs["arcApplicationTenantId"] = undefined /*out*/;
            resourceInputs["arcInstanceResourceGroup"] = undefined /*out*/;
            resourceInputs["arcServicePrincipalObjectId"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["connectivityProperties"] = undefined /*out*/;
            resourceInputs["defaultExtensions"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["perNodeDetails"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20210101preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20210901:ArcSetting" }, { type: "azure-native:azurestackhci/v20210901preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20220101:ArcSetting" }, { type: "azure-native:azurestackhci/v20220301:ArcSetting" }, { type: "azure-native:azurestackhci/v20220501:ArcSetting" }, { type: "azure-native:azurestackhci/v20220901:ArcSetting" }, { type: "azure-native:azurestackhci/v20221001:ArcSetting" }, { type: "azure-native:azurestackhci/v20221201:ArcSetting" }, { type: "azure-native:azurestackhci/v20221215preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20230201:ArcSetting" }, { type: "azure-native:azurestackhci/v20230301:ArcSetting" }, { type: "azure-native:azurestackhci/v20230601:ArcSetting" }, { type: "azure-native:azurestackhci/v20230801:ArcSetting" }, { type: "azure-native:azurestackhci/v20230801preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20231101preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20240101:ArcSetting" }, { type: "azure-native:azurestackhci/v20240215preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20240401:ArcSetting" }, { type: "azure-native:azurestackhci/v20240901preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20241201preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20250201preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20250915preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20251001:ArcSetting" }, { type: "azure-native:azurestackhci/v20251101preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20251201preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20260201:ArcSetting" }, { type: "azure-native:azurestackhci/v20260215preview:ArcSetting" }, { type: "azure-native:azurestackhci/v20260301preview:ArcSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ArcSetting.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ArcSetting = ArcSetting;
