// @pulumi/azure-native/azurestackhci/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPublicIPAddress = exports.getNetworkSecurityGroupOutput = exports.getNetworkSecurityGroup = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getNatGatewayOutput = exports.getNatGateway = exports.getMarketplaceGalleryImageOutput = exports.getMarketplaceGalleryImage = exports.getMachineExtensionOutput = exports.getMachineExtension = exports.getLogicalNetworkOutput = exports.getLogicalNetwork = exports.getLoadBalancerOutput = exports.getLoadBalancer = exports.getHybridIdentityMetadatumOutput = exports.getHybridIdentityMetadatum = exports.getHciEdgeDeviceJobOutput = exports.getHciEdgeDeviceJob = exports.getHciEdgeDeviceOutput = exports.getHciEdgeDevice = exports.getGuestAgentOutput = exports.getGuestAgent = exports.getGalleryImageOutput = exports.getGalleryImage = exports.getExtensionOutput = exports.getExtension = exports.getEdgeMachineJobOutput = exports.getEdgeMachineJob = exports.getEdgeMachineOutput = exports.getEdgeMachine = exports.getDevicePoolOutput = exports.getDevicePool = exports.getDeploymentSettingOutput = exports.getDeploymentSetting = exports.getClusterJobOutput = exports.getClusterJob = exports.getClusterOutput = exports.getCluster = exports.getArcSettingOutput = exports.getArcSetting = exports.GalleryImage = exports.Extension = exports.EdgeMachineJob = exports.EdgeMachine = exports.DevicePool = exports.DeploymentSetting = exports.ClusterJob = exports.Cluster = exports.ArcSetting = void 0;
exports.VirtualNetworkSubnet = exports.VirtualNetwork = exports.VirtualMachineInstance = exports.VirtualMachine = exports.VirtualHardDisk = exports.UpdateSummary = exports.UpdateRun = exports.Update = exports.StorageContainer = exports.Snapshot = exports.SecuritySetting = exports.SecurityRule = exports.PublicIPAddress = exports.NetworkSecurityGroup = exports.NetworkInterface = exports.NatGateway = exports.MarketplaceGalleryImage = exports.MachineExtension = exports.LogicalNetwork = exports.LoadBalancer = exports.HybridIdentityMetadatum = exports.HciEdgeDeviceJob = exports.HciEdgeDevice = exports.GuestAgent = exports.getVirtualNetworkSubnetOutput = exports.getVirtualNetworkSubnet = exports.getVirtualNetworkOutput = exports.getVirtualNetwork = exports.getVirtualMachineInstanceOutput = exports.getVirtualMachineInstance = exports.getVirtualMachineOutput = exports.getVirtualMachine = exports.getVirtualHardDiskOutput = exports.getVirtualHardDisk = exports.getUpdateSummaryOutput = exports.getUpdateSummary = exports.getUpdateRunOutput = exports.getUpdateRun = exports.getUpdateOutput = exports.getUpdate = exports.getStorageContainerOutput = exports.getStorageContainer = exports.getSnapshotOutput = exports.getSnapshot = exports.getSecuritySettingOutput = exports.getSecuritySetting = exports.getSecurityRuleOutput = exports.getSecurityRule = exports.getPublicIPAddressOutput = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.ArcSetting = null;
utilities.lazyLoad(exports, ["ArcSetting"], () => require("./arcSetting"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterJob = null;
utilities.lazyLoad(exports, ["ClusterJob"], () => require("./clusterJob"));
exports.DeploymentSetting = null;
utilities.lazyLoad(exports, ["DeploymentSetting"], () => require("./deploymentSetting"));
exports.DevicePool = null;
utilities.lazyLoad(exports, ["DevicePool"], () => require("./devicePool"));
exports.EdgeMachine = null;
utilities.lazyLoad(exports, ["EdgeMachine"], () => require("./edgeMachine"));
exports.EdgeMachineJob = null;
utilities.lazyLoad(exports, ["EdgeMachineJob"], () => require("./edgeMachineJob"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.getArcSetting = null;
exports.getArcSettingOutput = null;
utilities.lazyLoad(exports, ["getArcSetting", "getArcSettingOutput"], () => require("./getArcSetting"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterJob = null;
exports.getClusterJobOutput = null;
utilities.lazyLoad(exports, ["getClusterJob", "getClusterJobOutput"], () => require("./getClusterJob"));
exports.getDeploymentSetting = null;
exports.getDeploymentSettingOutput = null;
utilities.lazyLoad(exports, ["getDeploymentSetting", "getDeploymentSettingOutput"], () => require("./getDeploymentSetting"));
exports.getDevicePool = null;
exports.getDevicePoolOutput = null;
utilities.lazyLoad(exports, ["getDevicePool", "getDevicePoolOutput"], () => require("./getDevicePool"));
exports.getEdgeMachine = null;
exports.getEdgeMachineOutput = null;
utilities.lazyLoad(exports, ["getEdgeMachine", "getEdgeMachineOutput"], () => require("./getEdgeMachine"));
exports.getEdgeMachineJob = null;
exports.getEdgeMachineJobOutput = null;
utilities.lazyLoad(exports, ["getEdgeMachineJob", "getEdgeMachineJobOutput"], () => require("./getEdgeMachineJob"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGuestAgent = null;
exports.getGuestAgentOutput = null;
utilities.lazyLoad(exports, ["getGuestAgent", "getGuestAgentOutput"], () => require("./getGuestAgent"));
exports.getHciEdgeDevice = null;
exports.getHciEdgeDeviceOutput = null;
utilities.lazyLoad(exports, ["getHciEdgeDevice", "getHciEdgeDeviceOutput"], () => require("./getHciEdgeDevice"));
exports.getHciEdgeDeviceJob = null;
exports.getHciEdgeDeviceJobOutput = null;
utilities.lazyLoad(exports, ["getHciEdgeDeviceJob", "getHciEdgeDeviceJobOutput"], () => require("./getHciEdgeDeviceJob"));
exports.getHybridIdentityMetadatum = null;
exports.getHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getHybridIdentityMetadatum", "getHybridIdentityMetadatumOutput"], () => require("./getHybridIdentityMetadatum"));
exports.getLoadBalancer = null;
exports.getLoadBalancerOutput = null;
utilities.lazyLoad(exports, ["getLoadBalancer", "getLoadBalancerOutput"], () => require("./getLoadBalancer"));
exports.getLogicalNetwork = null;
exports.getLogicalNetworkOutput = null;
utilities.lazyLoad(exports, ["getLogicalNetwork", "getLogicalNetworkOutput"], () => require("./getLogicalNetwork"));
exports.getMachineExtension = null;
exports.getMachineExtensionOutput = null;
utilities.lazyLoad(exports, ["getMachineExtension", "getMachineExtensionOutput"], () => require("./getMachineExtension"));
exports.getMarketplaceGalleryImage = null;
exports.getMarketplaceGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getMarketplaceGalleryImage", "getMarketplaceGalleryImageOutput"], () => require("./getMarketplaceGalleryImage"));
exports.getNatGateway = null;
exports.getNatGatewayOutput = null;
utilities.lazyLoad(exports, ["getNatGateway", "getNatGatewayOutput"], () => require("./getNatGateway"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getNetworkSecurityGroup = null;
exports.getNetworkSecurityGroupOutput = null;
utilities.lazyLoad(exports, ["getNetworkSecurityGroup", "getNetworkSecurityGroupOutput"], () => require("./getNetworkSecurityGroup"));
exports.getPublicIPAddress = null;
exports.getPublicIPAddressOutput = null;
utilities.lazyLoad(exports, ["getPublicIPAddress", "getPublicIPAddressOutput"], () => require("./getPublicIPAddress"));
exports.getSecurityRule = null;
exports.getSecurityRuleOutput = null;
utilities.lazyLoad(exports, ["getSecurityRule", "getSecurityRuleOutput"], () => require("./getSecurityRule"));
exports.getSecuritySetting = null;
exports.getSecuritySettingOutput = null;
utilities.lazyLoad(exports, ["getSecuritySetting", "getSecuritySettingOutput"], () => require("./getSecuritySetting"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getStorageContainer = null;
exports.getStorageContainerOutput = null;
utilities.lazyLoad(exports, ["getStorageContainer", "getStorageContainerOutput"], () => require("./getStorageContainer"));
exports.getUpdate = null;
exports.getUpdateOutput = null;
utilities.lazyLoad(exports, ["getUpdate", "getUpdateOutput"], () => require("./getUpdate"));
exports.getUpdateRun = null;
exports.getUpdateRunOutput = null;
utilities.lazyLoad(exports, ["getUpdateRun", "getUpdateRunOutput"], () => require("./getUpdateRun"));
exports.getUpdateSummary = null;
exports.getUpdateSummaryOutput = null;
utilities.lazyLoad(exports, ["getUpdateSummary", "getUpdateSummaryOutput"], () => require("./getUpdateSummary"));
exports.getVirtualHardDisk = null;
exports.getVirtualHardDiskOutput = null;
utilities.lazyLoad(exports, ["getVirtualHardDisk", "getVirtualHardDiskOutput"], () => require("./getVirtualHardDisk"));
exports.getVirtualMachine = null;
exports.getVirtualMachineOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachine", "getVirtualMachineOutput"], () => require("./getVirtualMachine"));
exports.getVirtualMachineInstance = null;
exports.getVirtualMachineInstanceOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineInstance", "getVirtualMachineInstanceOutput"], () => require("./getVirtualMachineInstance"));
exports.getVirtualNetwork = null;
exports.getVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetwork", "getVirtualNetworkOutput"], () => require("./getVirtualNetwork"));
exports.getVirtualNetworkSubnet = null;
exports.getVirtualNetworkSubnetOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkSubnet", "getVirtualNetworkSubnetOutput"], () => require("./getVirtualNetworkSubnet"));
exports.GuestAgent = null;
utilities.lazyLoad(exports, ["GuestAgent"], () => require("./guestAgent"));
exports.HciEdgeDevice = null;
utilities.lazyLoad(exports, ["HciEdgeDevice"], () => require("./hciEdgeDevice"));
exports.HciEdgeDeviceJob = null;
utilities.lazyLoad(exports, ["HciEdgeDeviceJob"], () => require("./hciEdgeDeviceJob"));
exports.HybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["HybridIdentityMetadatum"], () => require("./hybridIdentityMetadatum"));
exports.LoadBalancer = null;
utilities.lazyLoad(exports, ["LoadBalancer"], () => require("./loadBalancer"));
exports.LogicalNetwork = null;
utilities.lazyLoad(exports, ["LogicalNetwork"], () => require("./logicalNetwork"));
exports.MachineExtension = null;
utilities.lazyLoad(exports, ["MachineExtension"], () => require("./machineExtension"));
exports.MarketplaceGalleryImage = null;
utilities.lazyLoad(exports, ["MarketplaceGalleryImage"], () => require("./marketplaceGalleryImage"));
exports.NatGateway = null;
utilities.lazyLoad(exports, ["NatGateway"], () => require("./natGateway"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.NetworkSecurityGroup = null;
utilities.lazyLoad(exports, ["NetworkSecurityGroup"], () => require("./networkSecurityGroup"));
exports.PublicIPAddress = null;
utilities.lazyLoad(exports, ["PublicIPAddress"], () => require("./publicIPAddress"));
exports.SecurityRule = null;
utilities.lazyLoad(exports, ["SecurityRule"], () => require("./securityRule"));
exports.SecuritySetting = null;
utilities.lazyLoad(exports, ["SecuritySetting"], () => require("./securitySetting"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.StorageContainer = null;
utilities.lazyLoad(exports, ["StorageContainer"], () => require("./storageContainer"));
exports.Update = null;
utilities.lazyLoad(exports, ["Update"], () => require("./update"));
exports.UpdateRun = null;
utilities.lazyLoad(exports, ["UpdateRun"], () => require("./updateRun"));
exports.UpdateSummary = null;
utilities.lazyLoad(exports, ["UpdateSummary"], () => require("./updateSummary"));
exports.VirtualHardDisk = null;
utilities.lazyLoad(exports, ["VirtualHardDisk"], () => require("./virtualHardDisk"));
exports.VirtualMachine = null;
utilities.lazyLoad(exports, ["VirtualMachine"], () => require("./virtualMachine"));
exports.VirtualMachineInstance = null;
utilities.lazyLoad(exports, ["VirtualMachineInstance"], () => require("./virtualMachineInstance"));
exports.VirtualNetwork = null;
utilities.lazyLoad(exports, ["VirtualNetwork"], () => require("./virtualNetwork"));
exports.VirtualNetworkSubnet = null;
utilities.lazyLoad(exports, ["VirtualNetworkSubnet"], () => require("./virtualNetworkSubnet"));
// Export enums:
__exportStar(require("../types/enums/azurestackhci"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestackhci:ArcSetting":
                return new exports.ArcSetting(name, undefined, { urn });
            case "azure-native:azurestackhci:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:azurestackhci:ClusterJob":
                return new exports.ClusterJob(name, undefined, { urn });
            case "azure-native:azurestackhci:DeploymentSetting":
                return new exports.DeploymentSetting(name, undefined, { urn });
            case "azure-native:azurestackhci:DevicePool":
                return new exports.DevicePool(name, undefined, { urn });
            case "azure-native:azurestackhci:EdgeMachine":
                return new exports.EdgeMachine(name, undefined, { urn });
            case "azure-native:azurestackhci:EdgeMachineJob":
                return new exports.EdgeMachineJob(name, undefined, { urn });
            case "azure-native:azurestackhci:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:azurestackhci:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci:GuestAgent":
                return new exports.GuestAgent(name, undefined, { urn });
            case "azure-native:azurestackhci:HciEdgeDevice":
                return new exports.HciEdgeDevice(name, undefined, { urn });
            case "azure-native:azurestackhci:HciEdgeDeviceJob":
                return new exports.HciEdgeDeviceJob(name, undefined, { urn });
            case "azure-native:azurestackhci:HybridIdentityMetadatum":
                return new exports.HybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:azurestackhci:LoadBalancer":
                return new exports.LoadBalancer(name, undefined, { urn });
            case "azure-native:azurestackhci:LogicalNetwork":
                return new exports.LogicalNetwork(name, undefined, { urn });
            case "azure-native:azurestackhci:MachineExtension":
                return new exports.MachineExtension(name, undefined, { urn });
            case "azure-native:azurestackhci:MarketplaceGalleryImage":
                return new exports.MarketplaceGalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci:NatGateway":
                return new exports.NatGateway(name, undefined, { urn });
            case "azure-native:azurestackhci:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:azurestackhci:NetworkSecurityGroup":
                return new exports.NetworkSecurityGroup(name, undefined, { urn });
            case "azure-native:azurestackhci:PublicIPAddress":
                return new exports.PublicIPAddress(name, undefined, { urn });
            case "azure-native:azurestackhci:SecurityRule":
                return new exports.SecurityRule(name, undefined, { urn });
            case "azure-native:azurestackhci:SecuritySetting":
                return new exports.SecuritySetting(name, undefined, { urn });
            case "azure-native:azurestackhci:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "azure-native:azurestackhci:StorageContainer":
                return new exports.StorageContainer(name, undefined, { urn });
            case "azure-native:azurestackhci:Update":
                return new exports.Update(name, undefined, { urn });
            case "azure-native:azurestackhci:UpdateRun":
                return new exports.UpdateRun(name, undefined, { urn });
            case "azure-native:azurestackhci:UpdateSummary":
                return new exports.UpdateSummary(name, undefined, { urn });
            case "azure-native:azurestackhci:VirtualHardDisk":
                return new exports.VirtualHardDisk(name, undefined, { urn });
            case "azure-native:azurestackhci:VirtualMachine":
                return new exports.VirtualMachine(name, undefined, { urn });
            case "azure-native:azurestackhci:VirtualMachineInstance":
                return new exports.VirtualMachineInstance(name, undefined, { urn });
            case "azure-native:azurestackhci:VirtualNetwork":
                return new exports.VirtualNetwork(name, undefined, { urn });
            case "azure-native:azurestackhci:VirtualNetworkSubnet":
                return new exports.VirtualNetworkSubnet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestackhci", _module);
