// @pulumi/azure-native/azurestackhci/updateRun.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRun = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Details of an Update run
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2023-03-01.
 *
 * Other available API versions: 2022-12-15-preview, 2023-02-01, 2023-03-01, 2023-06-01, 2023-08-01, 2023-08-01-preview, 2023-11-01-preview, 2024-01-01, 2024-02-15-preview, 2024-09-01-preview, 2024-12-01-preview, 2025-02-01-preview, 2025-09-15-preview, 2025-10-01, 2025-11-01-preview, 2025-12-01-preview, 2026-02-01, 2026-02-15-preview, 2026-03-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native azurestackhci [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class UpdateRun extends pulumi.CustomResource {
    /**
     * Get an existing UpdateRun resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new UpdateRun(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:azurestackhci:UpdateRun';
    /**
     * Returns true if the given object is an instance of UpdateRun.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UpdateRun.__pulumiType;
    }
    /**
     * Create a UpdateRun resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.updateName === undefined && !opts.urn) {
                throw new Error("Missing required property 'updateName'");
            }
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["description"] = args?.description;
            resourceInputs["duration"] = args?.duration;
            resourceInputs["endTimeUtc"] = args?.endTimeUtc;
            resourceInputs["errorMessage"] = args?.errorMessage;
            resourceInputs["expectedExecutionTime"] = args?.expectedExecutionTime;
            resourceInputs["lastUpdatedTime"] = args?.lastUpdatedTime;
            resourceInputs["lastUpdatedTimeUtc"] = args?.lastUpdatedTimeUtc;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["startTimeUtc"] = args?.startTimeUtc;
            resourceInputs["state"] = args?.state;
            resourceInputs["status"] = args?.status;
            resourceInputs["steps"] = args?.steps;
            resourceInputs["timeStarted"] = args?.timeStarted;
            resourceInputs["updateName"] = args?.updateName;
            resourceInputs["updateRunName"] = args?.updateRunName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["duration"] = undefined /*out*/;
            resourceInputs["endTimeUtc"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["expectedExecutionTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTimeUtc"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["startTimeUtc"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["steps"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeStarted"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20221201:UpdateRun" }, { type: "azure-native:azurestackhci/v20221215preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20230201:UpdateRun" }, { type: "azure-native:azurestackhci/v20230301:UpdateRun" }, { type: "azure-native:azurestackhci/v20230601:UpdateRun" }, { type: "azure-native:azurestackhci/v20230801:UpdateRun" }, { type: "azure-native:azurestackhci/v20230801preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20231101preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20240101:UpdateRun" }, { type: "azure-native:azurestackhci/v20240215preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20240401:UpdateRun" }, { type: "azure-native:azurestackhci/v20240901preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20241201preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20250201preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20250915preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20251001:UpdateRun" }, { type: "azure-native:azurestackhci/v20251101preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20251201preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20260201:UpdateRun" }, { type: "azure-native:azurestackhci/v20260215preview:UpdateRun" }, { type: "azure-native:azurestackhci/v20260301preview:UpdateRun" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(UpdateRun.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UpdateRun = UpdateRun;
