// @pulumi/aws/athena/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workgroup = exports.PreparedStatement = exports.NamedQuery = exports.getNamedQueryOutput = exports.getNamedQuery = exports.Database = exports.DataCatalog = exports.CapacityReservation = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.CapacityReservation = null;
utilities.lazyLoad(exports, ["CapacityReservation"], () => require("./capacityReservation"));
exports.DataCatalog = null;
utilities.lazyLoad(exports, ["DataCatalog"], () => require("./dataCatalog"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.getNamedQuery = null;
exports.getNamedQueryOutput = null;
utilities.lazyLoad(exports, ["getNamedQuery", "getNamedQueryOutput"], () => require("./getNamedQuery"));
exports.NamedQuery = null;
utilities.lazyLoad(exports, ["NamedQuery"], () => require("./namedQuery"));
exports.PreparedStatement = null;
utilities.lazyLoad(exports, ["PreparedStatement"], () => require("./preparedStatement"));
exports.Workgroup = null;
utilities.lazyLoad(exports, ["Workgroup"], () => require("./workgroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:athena/capacityReservation:CapacityReservation":
                return new exports.CapacityReservation(name, undefined, { urn });
            case "aws:athena/dataCatalog:DataCatalog":
                return new exports.DataCatalog(name, undefined, { urn });
            case "aws:athena/database:Database":
                return new exports.Database(name, undefined, { urn });
            case "aws:athena/namedQuery:NamedQuery":
                return new exports.NamedQuery(name, undefined, { urn });
            case "aws:athena/preparedStatement:PreparedStatement":
                return new exports.PreparedStatement(name, undefined, { urn });
            case "aws:athena/workgroup:Workgroup":
                return new exports.Workgroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "athena/capacityReservation", _module);
pulumi.runtime.registerResourceModule("aws", "athena/dataCatalog", _module);
pulumi.runtime.registerResourceModule("aws", "athena/database", _module);
pulumi.runtime.registerResourceModule("aws", "athena/namedQuery", _module);
pulumi.runtime.registerResourceModule("aws", "athena/preparedStatement", _module);
pulumi.runtime.registerResourceModule("aws", "athena/workgroup", _module);
