// @pulumi/aws/sagemaker/appImageConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppImageConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a SageMaker AI App Image Config resource.
 *
 * ## Example Usage
 *
 * ### Basic usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.sagemaker.AppImageConfig("test", {
 *     appImageConfigName: "example",
 *     kernelGatewayImageConfig: {
 *         kernelSpecs: [{
 *             name: "example",
 *         }],
 *     },
 * });
 * ```
 *
 * ### Using Code Editor with empty configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.sagemaker.AppImageConfig("test", {
 *     appImageConfigName: "example",
 *     codeEditorAppImageConfig: {},
 * });
 * ```
 *
 * ### Default File System Config
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.sagemaker.AppImageConfig("test", {
 *     appImageConfigName: "example",
 *     kernelGatewayImageConfig: {
 *         kernelSpecs: [{
 *             name: "example",
 *         }],
 *         fileSystemConfig: {},
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import SageMaker AI App Image Configs using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/appImageConfig:AppImageConfig example example
 * ```
 */
class AppImageConfig extends pulumi.CustomResource {
    /**
     * Get an existing AppImageConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AppImageConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/appImageConfig:AppImageConfig';
    /**
     * Returns true if the given object is an instance of AppImageConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AppImageConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appImageConfigName"] = state?.appImageConfigName;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["codeEditorAppImageConfig"] = state?.codeEditorAppImageConfig;
            resourceInputs["jupyterLabImageConfig"] = state?.jupyterLabImageConfig;
            resourceInputs["kernelGatewayImageConfig"] = state?.kernelGatewayImageConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.appImageConfigName === undefined && !opts.urn) {
                throw new Error("Missing required property 'appImageConfigName'");
            }
            resourceInputs["appImageConfigName"] = args?.appImageConfigName;
            resourceInputs["codeEditorAppImageConfig"] = args?.codeEditorAppImageConfig;
            resourceInputs["jupyterLabImageConfig"] = args?.jupyterLabImageConfig;
            resourceInputs["kernelGatewayImageConfig"] = args?.kernelGatewayImageConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AppImageConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AppImageConfig = AppImageConfig;
