// @pulumi/aws/sagemaker/featureGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a SageMaker AI Feature Group resource.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.FeatureGroup("example", {
 *     featureGroupName: "example",
 *     recordIdentifierFeatureName: "example",
 *     eventTimeFeatureName: "example",
 *     roleArn: test.arn,
 *     featureDefinitions: [{
 *         featureName: "example",
 *         featureType: "String",
 *     }],
 *     onlineStoreConfig: {
 *         enableOnlineStore: true,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Feature Groups using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/featureGroup:FeatureGroup test_feature_group feature_group-foo
 * ```
 */
class FeatureGroup extends pulumi.CustomResource {
    /**
     * Get an existing FeatureGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FeatureGroup(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/featureGroup:FeatureGroup';
    /**
     * Returns true if the given object is an instance of FeatureGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FeatureGroup.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["eventTimeFeatureName"] = state?.eventTimeFeatureName;
            resourceInputs["featureDefinitions"] = state?.featureDefinitions;
            resourceInputs["featureGroupName"] = state?.featureGroupName;
            resourceInputs["offlineStoreConfig"] = state?.offlineStoreConfig;
            resourceInputs["onlineStoreConfig"] = state?.onlineStoreConfig;
            resourceInputs["recordIdentifierFeatureName"] = state?.recordIdentifierFeatureName;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["throughputConfig"] = state?.throughputConfig;
        }
        else {
            const args = argsOrState;
            if (args?.eventTimeFeatureName === undefined && !opts.urn) {
                throw new Error("Missing required property 'eventTimeFeatureName'");
            }
            if (args?.featureDefinitions === undefined && !opts.urn) {
                throw new Error("Missing required property 'featureDefinitions'");
            }
            if (args?.featureGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'featureGroupName'");
            }
            if (args?.recordIdentifierFeatureName === undefined && !opts.urn) {
                throw new Error("Missing required property 'recordIdentifierFeatureName'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["eventTimeFeatureName"] = args?.eventTimeFeatureName;
            resourceInputs["featureDefinitions"] = args?.featureDefinitions;
            resourceInputs["featureGroupName"] = args?.featureGroupName;
            resourceInputs["offlineStoreConfig"] = args?.offlineStoreConfig;
            resourceInputs["onlineStoreConfig"] = args?.onlineStoreConfig;
            resourceInputs["recordIdentifierFeatureName"] = args?.recordIdentifierFeatureName;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["throughputConfig"] = args?.throughputConfig;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FeatureGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FeatureGroup = FeatureGroup;
