// @pulumi/aws/sagemaker/hyperParameterTuningJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HyperParameterTuningJob = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS SageMaker AI Hyper Parameter Tuning Job.
 *
 * > **NOTE:** This resource does not wait for the tuning job to complete before returning. Terraform may complete apply before the job reaches a terminal state.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.HyperParameterTuningJob("example", {
 *     name: "example",
 *     config: {
 *         strategy: "Bayesian",
 *         objective: {
 *             metricName: "test:msd",
 *             type: "Minimize",
 *         },
 *         parameterRanges: {
 *             categoricalParameterRanges: [{
 *                 name: "init_method",
 *                 values: [
 *                     "kmeans++",
 *                     "random",
 *                 ],
 *             }],
 *             integerParameterRanges: [
 *                 {
 *                     name: "epochs",
 *                     minValue: "1",
 *                     maxValue: "10",
 *                     scalingType: "Auto",
 *                 },
 *                 {
 *                     name: "extra_center_factor",
 *                     minValue: "4",
 *                     maxValue: "10",
 *                     scalingType: "Auto",
 *                 },
 *                 {
 *                     name: "mini_batch_size",
 *                     minValue: "3000",
 *                     maxValue: "15000",
 *                     scalingType: "Auto",
 *                 },
 *             ],
 *         },
 *         resourceLimits: {
 *             maxNumberOfTrainingJobs: 2,
 *             maxParallelTrainingJobs: 1,
 *         },
 *     },
 *     trainingJobDefinition: {
 *         roleArn: "arn:aws:iam::123456789012:role/example-sagemaker-execution-role",
 *         algorithmSpecification: {
 *             trainingImage: "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1",
 *             trainingInputMode: "File",
 *         },
 *         staticHyperParameters: {
 *             feature_dim: "3",
 *             k: "2",
 *         },
 *         inputDataConfigs: [
 *             {
 *                 channelName: "train",
 *                 contentType: "text/csv",
 *                 inputMode: "File",
 *                 dataSource: {
 *                     s3DataSource: {
 *                         s3DataType: "S3Prefix",
 *                         s3Uri: "s3://example-bucket/input/",
 *                     },
 *                 },
 *             },
 *             {
 *                 channelName: "test",
 *                 contentType: "text/csv",
 *                 inputMode: "File",
 *                 dataSource: {
 *                     s3DataSource: {
 *                         s3DataType: "S3Prefix",
 *                         s3Uri: "s3://example-bucket/input/",
 *                     },
 *                 },
 *             },
 *         ],
 *         outputDataConfig: {
 *             s3OutputPath: "s3://example-bucket/output/",
 *         },
 *         resourceConfig: {
 *             instanceCount: 1,
 *             instanceType: "ml.m5.large",
 *             volumeSizeInGb: 30,
 *         },
 *         stoppingCondition: {
 *             maxRuntimeInSeconds: 3600,
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `name` (String) Name of the Hyper Parameter Tuning Job.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import SageMaker AI Hyper Parameter Tuning Jobs using `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/hyperParameterTuningJob:HyperParameterTuningJob example example-hyper-parameter-tuning-job
 * ```
 */
class HyperParameterTuningJob extends pulumi.CustomResource {
    /**
     * Get an existing HyperParameterTuningJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HyperParameterTuningJob(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/hyperParameterTuningJob:HyperParameterTuningJob';
    /**
     * Returns true if the given object is an instance of HyperParameterTuningJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HyperParameterTuningJob.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["autotune"] = state?.autotune;
            resourceInputs["config"] = state?.config;
            resourceInputs["failureReason"] = state?.failureReason;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["trainingJobDefinition"] = state?.trainingJobDefinition;
            resourceInputs["trainingJobDefinitions"] = state?.trainingJobDefinitions;
            resourceInputs["warmStartConfig"] = state?.warmStartConfig;
        }
        else {
            const args = argsOrState;
            if (args?.config === undefined && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            resourceInputs["autotune"] = args?.autotune;
            resourceInputs["config"] = args?.config;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["trainingJobDefinition"] = args?.trainingJobDefinition;
            resourceInputs["trainingJobDefinitions"] = args?.trainingJobDefinitions;
            resourceInputs["warmStartConfig"] = args?.warmStartConfig;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HyperParameterTuningJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HyperParameterTuningJob = HyperParameterTuningJob;
