// @pulumi/aws/sagemaker/labelingJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelingJob = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manage an Amazon SageMaker labeling job.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * // https://docs.aws.amazon.com/sagemaker/latest/dg/sms-named-entity-recg.html#sms-creating-ner-api.
 * const test = new aws.sagemaker.LabelingJob("test", {
 *     labelAttributeName: "label1",
 *     labelingJobName: "my-labeling-job",
 *     roleArn: exampleAwsIamRole.arn,
 *     labelCategoryConfigS3Uri: `s3://${exampleAwsS3Bucket.bucket}/${exampleAwsS3Object.key}`,
 *     humanTaskConfig: {
 *         numberOfHumanWorkersPerDataObject: 1,
 *         taskDescription: "Apply the labels provided to specific words or phrases within the larger text block.",
 *         taskTitle: "Named entity Recognition task",
 *         taskTimeLimitInSeconds: 28800,
 *         workteamArn: example.arn,
 *         uiConfig: {
 *             humanTaskUiArn: "arn:aws:sagemaker:us-west-2:394669845002:human-task-ui/NamedEntityRecognition",
 *         },
 *         preHumanTaskLambdaArn: "arn:aws:lambda:us-west-2:081040173940:function:PRE-NamedEntityRecognition",
 *         annotationConsolidationConfig: {
 *             annotationConsolidationLambdaArn: "arn:aws:lambda:us-west-2:081040173940:function:ACS-NamedEntityRecognition",
 *         },
 *     },
 *     inputConfig: {
 *         dataSource: {
 *             snsDataSource: {
 *                 snsTopicArn: exampleAwsSnsTopic.arn,
 *             },
 *         },
 *     },
 *     outputConfig: {
 *         s3OutputPath: `s3://${exampleAwsS3Bucket.bucket}/`,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import labeling jobs using the `labelingJobName`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/labelingJob:LabelingJob example my-labeling-job
 * ```
 */
class LabelingJob extends pulumi.CustomResource {
    /**
     * Get an existing LabelingJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LabelingJob(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/labelingJob:LabelingJob';
    /**
     * Returns true if the given object is an instance of LabelingJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LabelingJob.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["failureReason"] = state?.failureReason;
            resourceInputs["humanTaskConfig"] = state?.humanTaskConfig;
            resourceInputs["inputConfig"] = state?.inputConfig;
            resourceInputs["jobReferenceCode"] = state?.jobReferenceCode;
            resourceInputs["labelAttributeName"] = state?.labelAttributeName;
            resourceInputs["labelCategoryConfigS3Uri"] = state?.labelCategoryConfigS3Uri;
            resourceInputs["labelCounters"] = state?.labelCounters;
            resourceInputs["labelingJobAlgorithmsConfig"] = state?.labelingJobAlgorithmsConfig;
            resourceInputs["labelingJobArn"] = state?.labelingJobArn;
            resourceInputs["labelingJobName"] = state?.labelingJobName;
            resourceInputs["labelingJobStatus"] = state?.labelingJobStatus;
            resourceInputs["outputConfig"] = state?.outputConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["stoppingConditions"] = state?.stoppingConditions;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.humanTaskConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'humanTaskConfig'");
            }
            if (args?.inputConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'inputConfig'");
            }
            if (args?.labelAttributeName === undefined && !opts.urn) {
                throw new Error("Missing required property 'labelAttributeName'");
            }
            if (args?.labelingJobName === undefined && !opts.urn) {
                throw new Error("Missing required property 'labelingJobName'");
            }
            if (args?.outputConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'outputConfig'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["humanTaskConfig"] = args?.humanTaskConfig;
            resourceInputs["inputConfig"] = args?.inputConfig;
            resourceInputs["labelAttributeName"] = args?.labelAttributeName;
            resourceInputs["labelCategoryConfigS3Uri"] = args?.labelCategoryConfigS3Uri;
            resourceInputs["labelingJobAlgorithmsConfig"] = args?.labelingJobAlgorithmsConfig;
            resourceInputs["labelingJobName"] = args?.labelingJobName;
            resourceInputs["outputConfig"] = args?.outputConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["stoppingConditions"] = args?.stoppingConditions;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["jobReferenceCode"] = undefined /*out*/;
            resourceInputs["labelCounters"] = undefined /*out*/;
            resourceInputs["labelingJobArn"] = undefined /*out*/;
            resourceInputs["labelingJobStatus"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LabelingJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LabelingJob = LabelingJob;
