// @pulumi/aws/sagemaker/mlflowApp.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MlflowApp = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a SageMaker AI MLflow App resource.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.MlflowApp("example", {
 *     name: "example",
 *     roleArn: exampleAwsIamRole.arn,
 *     artifactStoreUri: `s3://${exampleAwsS3Bucket.bucket}/path`,
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `arn` (String) ARN of the MLflow App.
 *
 * Using `pulumi import`, import SageMaker AI MLflow Apps using the `arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/mlflowApp:MlflowApp example arn:aws:sagemaker:us-east-1:123456789012:mlflow-app/app-ABCD1234
 * ```
 */
class MlflowApp extends pulumi.CustomResource {
    /**
     * Get an existing MlflowApp resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MlflowApp(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/mlflowApp:MlflowApp';
    /**
     * Returns true if the given object is an instance of MlflowApp.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MlflowApp.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountDefaultStatus"] = state?.accountDefaultStatus;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["artifactStoreUri"] = state?.artifactStoreUri;
            resourceInputs["defaultDomainIdLists"] = state?.defaultDomainIdLists;
            resourceInputs["modelRegistrationMode"] = state?.modelRegistrationMode;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["weeklyMaintenanceWindowStart"] = state?.weeklyMaintenanceWindowStart;
        }
        else {
            const args = argsOrState;
            if (args?.artifactStoreUri === undefined && !opts.urn) {
                throw new Error("Missing required property 'artifactStoreUri'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["accountDefaultStatus"] = args?.accountDefaultStatus;
            resourceInputs["artifactStoreUri"] = args?.artifactStoreUri;
            resourceInputs["defaultDomainIdLists"] = args?.defaultDomainIdLists;
            resourceInputs["modelRegistrationMode"] = args?.modelRegistrationMode;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["weeklyMaintenanceWindowStart"] = args?.weeklyMaintenanceWindowStart;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MlflowApp.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MlflowApp = MlflowApp;
