// @pulumi/aws/sagemaker/userProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfile = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a SageMaker AI User Profile resource.
 *
 * ## Example Usage
 *
 * ### Basic usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.UserProfile("example", {
 *     domainId: test.id,
 *     userProfileName: "example",
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `domainId` (String) SageMaker domain ID.
 * * `userProfileName` (String) Name of the user profile.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import SageMaker AI User Profiles using the `arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/userProfile:UserProfile example arn:aws:sagemaker:us-west-2:123456789012:user-profile/domain-id/profile-name
 * ```
 */
class UserProfile extends pulumi.CustomResource {
    /**
     * Get an existing UserProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new UserProfile(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:sagemaker/userProfile:UserProfile';
    /**
     * Returns true if the given object is an instance of UserProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserProfile.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["domainId"] = state?.domainId;
            resourceInputs["homeEfsFileSystemUid"] = state?.homeEfsFileSystemUid;
            resourceInputs["region"] = state?.region;
            resourceInputs["singleSignOnUserIdentifier"] = state?.singleSignOnUserIdentifier;
            resourceInputs["singleSignOnUserValue"] = state?.singleSignOnUserValue;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["userProfileName"] = state?.userProfileName;
            resourceInputs["userSettings"] = state?.userSettings;
        }
        else {
            const args = argsOrState;
            if (args?.domainId === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainId'");
            }
            if (args?.userProfileName === undefined && !opts.urn) {
                throw new Error("Missing required property 'userProfileName'");
            }
            resourceInputs["domainId"] = args?.domainId;
            resourceInputs["region"] = args?.region;
            resourceInputs["singleSignOnUserIdentifier"] = args?.singleSignOnUserIdentifier;
            resourceInputs["singleSignOnUserValue"] = args?.singleSignOnUserValue;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["userProfileName"] = args?.userProfileName;
            resourceInputs["userSettings"] = args?.userSettings;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["homeEfsFileSystemUid"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(UserProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UserProfile = UserProfile;
