// @pulumi/azure-native/types/enums/containerservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutboundType = exports.Operator = exports.OSType = exports.OSSKU = exports.OSDiskType = exports.NodeProvisioningMode = exports.NodeProvisioningDefaultNodePools = exports.NodeOSUpgradeChannel = exports.NodeImageSelectionType = exports.NginxIngressControllerType = exports.NetworkPolicy = exports.NetworkPluginMode = exports.NetworkPlugin = exports.NetworkMode = exports.NetworkDataplane = exports.ManagedServiceIdentityType = exports.ManagedClusterUpgradeType = exports.ManagedClusterSKUTier = exports.ManagedClusterSKUName = exports.LocalDNSServeStale = exports.LocalDNSQueryLogging = exports.LocalDNSProtocol = exports.LocalDNSMode = exports.LocalDNSForwardPolicy = exports.LocalDNSForwardDestination = exports.LoadBalancerSku = exports.LicenseType = exports.LabelSelectorOperator = exports.KubernetesSupportPlan = exports.KubeletDiskType = exports.KeyVaultNetworkAccessTypes = exports.IstioIngressGatewayMode = exports.IpFamily = exports.GPUInstanceProfile = exports.GPUDriver = exports.ExtendedLocationTypes = exports.Expander = exports.ExecutionPoint = exports.DeploymentSafeguardsLevel = exports.DeletePolicy = exports.ConnectionStatus = exports.Code = exports.BackendPoolType = exports.AutoUpgradeNodeImageSelectionType = exports.ArtifactSource = exports.AgentPoolType = exports.AgentPoolSSHAccess = exports.AgentPoolMode = exports.AdvancedNetworkPolicies = exports.AdoptionPolicy = void 0;
exports.WorkloadRuntime = exports.WeekDay = exports.UpgradeChannel = exports.UndrainableNodeBehavior = exports.Type = exports.TolerationOperator = exports.TaintEffect = exports.SnapshotType = exports.ServiceMeshMode = exports.ScriptType = exports.ScaleSetPriority = exports.ScaleSetEvictionPolicy = exports.ScaleDownMode = exports.RestrictionLevel = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.Protocol = exports.PropertySelectorOperator = exports.PropagationType = exports.PolicyRule = exports.PodIPAllocationMode = exports.PlacementType = void 0;
exports.AdoptionPolicy = {
    /**
     * If the namespace already exists in Kubernetes, attempts to create that same namespace in ARM will fail.
     */
    Never: "Never",
    /**
     * Take over the existing namespace to be managed by ARM, if there is no difference.
     */
    IfIdentical: "IfIdentical",
    /**
     * Always take over the existing namespace to be managed by ARM, some fields might be overwritten.
     */
    Always: "Always",
};
exports.AdvancedNetworkPolicies = {
    /**
     * Enable Layer7 network policies (FQDN, HTTP/S, Kafka). This option is a superset of the FQDN option.
     */
    L7: "L7",
    /**
     * Enable FQDN based network policies
     */
    FQDN: "FQDN",
    /**
     * Disable Layer 7 network policies (FQDN, HTTP/S, Kafka)
     */
    None: "None",
};
exports.AgentPoolMode = {
    /**
     * System agent pools are primarily for hosting critical system pods such as CoreDNS and metrics-server. System agent pools osType must be Linux. System agent pools VM SKU must have at least 2vCPUs and 4GB of memory.
     */
    System: "System",
    /**
     * User agent pools are primarily for hosting your application pods.
     */
    User: "User",
    /**
     * Gateway agent pools are dedicated to providing static egress IPs to pods. For more details, see https://aka.ms/aks/static-egress-gateway.
     */
    Gateway: "Gateway",
};
exports.AgentPoolSSHAccess = {
    /**
     * Can SSH onto the node as a local user using private key.
     */
    LocalUser: "LocalUser",
    /**
     * SSH service will be turned off on the node.
     */
    Disabled: "Disabled",
};
exports.AgentPoolType = {
    /**
     * Create an Agent Pool backed by a Virtual Machine Scale Set.
     */
    VirtualMachineScaleSets: "VirtualMachineScaleSets",
    /**
     * Use of this is strongly discouraged.
     */
    AvailabilitySet: "AvailabilitySet",
    /**
     * Create an Agent Pool backed by a Single Instance VM orchestration mode.
     */
    VirtualMachines: "VirtualMachines",
};
exports.ArtifactSource = {
    /**
     * pull images from Azure Container Registry with cache
     */
    Cache: "Cache",
    /**
     * pull images from Microsoft Artifact Registry
     */
    Direct: "Direct",
};
exports.AutoUpgradeNodeImageSelectionType = {
    /**
     * Use the latest image version when upgrading nodes. Clusters may use different image versions (e.g., 'AKSUbuntu-1804gen2containerd-2021.10.12' and 'AKSUbuntu-1804gen2containerd-2021.10.19') because, for example, the latest available version is different in different regions.
     */
    Latest: "Latest",
    /**
     * The image versions to upgrade nodes to are selected as described below: for each node pool in managed clusters affected by the update run, the system selects the latest image version such that it is available across all other node pools (in all other clusters) of the same image type. As a result, all node pools of the same image type will be upgraded to the same image version. For example, if the latest image version for image type 'AKSUbuntu-1804gen2containerd' is 'AKSUbuntu-1804gen2containerd-2021.10.12' for a node pool in cluster A in region X, and is 'AKSUbuntu-1804gen2containerd-2021.10.17' for a node pool in cluster B in region Y, the system will upgrade both node pools to image version 'AKSUbuntu-1804gen2containerd-2021.10.12'.
     */
    Consistent: "Consistent",
};
exports.BackendPoolType = {
    /**
     * The type of the managed inbound Load Balancer BackendPool. https://cloud-provider-azure.sigs.k8s.io/topics/loadbalancer/#configure-load-balancer-backend.
     */
    NodeIPConfiguration: "NodeIPConfiguration",
    /**
     * The type of the managed inbound Load Balancer BackendPool. https://cloud-provider-azure.sigs.k8s.io/topics/loadbalancer/#configure-load-balancer-backend.
     */
    NodeIP: "NodeIP",
};
exports.Code = {
    /**
     * The cluster is running.
     */
    Running: "Running",
    /**
     * The cluster is stopped.
     */
    Stopped: "Stopped",
};
exports.ConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.DeletePolicy = {
    /**
     * Only delete the ARM resource, keep the Kubernetes namespace. Also delete the ManagedByARM label.
     */
    Keep: "Keep",
    /**
     * Delete both the ARM resource and the Kubernetes namespace together.
     */
    Delete: "Delete",
};
exports.DeploymentSafeguardsLevel = {
    /**
     * Best practice violations will only show warnings
     */
    Warn: "Warn",
    /**
     * Best practice violations will be denied
     */
    Enforce: "Enforce",
};
exports.ExecutionPoint = {
    /**
     * Execute during node image build time.
     */
    NodeImageBuildTime: "NodeImageBuildTime",
    /**
     * Execute during node provisioning time.
     */
    NodeProvisionTime: "NodeProvisionTime",
};
exports.Expander = {
    /**
     * Selects the node group that will have the least idle CPU (if tied, unused memory) after scale-up. This is useful when you have different classes of nodes, for example, high CPU or high memory nodes, and only want to expand those when there are pending pods that need a lot of those resources.
     */
    Least_waste: "least-waste",
    /**
     * Selects the node group that would be able to schedule the most pods when scaling up. This is useful when you are using nodeSelector to make sure certain pods land on certain nodes. Note that this won't cause the autoscaler to select bigger nodes vs. smaller, as it can add multiple smaller nodes at once.
     */
    Most_pods: "most-pods",
    /**
     * Selects the node group that has the highest priority assigned by the user. It's configuration is described in more details [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/expander/priority/readme.md).
     */
    Priority: "priority",
    /**
     * Used when you don't have a particular need for the node groups to scale differently.
     */
    Random: "random",
};
exports.ExtendedLocationTypes = {
    EdgeZone: "EdgeZone",
};
exports.GPUDriver = {
    /**
     * Install driver.
     */
    Install: "Install",
    /**
     * Skip driver install.
     */
    None: "None",
};
exports.GPUInstanceProfile = {
    MIG1g: "MIG1g",
    MIG2g: "MIG2g",
    MIG3g: "MIG3g",
    MIG4g: "MIG4g",
    MIG7g: "MIG7g",
};
exports.IpFamily = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.IstioIngressGatewayMode = {
    /**
     * The ingress gateway is assigned a public IP address and is publicly accessible.
     */
    External: "External",
    /**
     * The ingress gateway is assigned an internal IP address and cannot is accessed publicly.
     */
    Internal: "Internal",
};
exports.KeyVaultNetworkAccessTypes = {
    Public: "Public",
    Private: "Private",
};
exports.KubeletDiskType = {
    /**
     * Kubelet will use the OS disk for its data.
     */
    OS: "OS",
    /**
     * Kubelet will use the temporary disk for its data.
     */
    Temporary: "Temporary",
};
exports.KubernetesSupportPlan = {
    /**
     * Support for the version is the same as for the open source Kubernetes offering. Official Kubernetes open source community support versions for 1 year after release.
     */
    KubernetesOfficial: "KubernetesOfficial",
    /**
     * Support for the version extended past the KubernetesOfficial support of 1 year. AKS continues to patch CVEs for another 1 year, for a total of 2 years of support.
     */
    AKSLongTermSupport: "AKSLongTermSupport",
};
exports.LabelSelectorOperator = {
    /**
     * Label Selector Operator In
     */
    In: "In",
    /**
     * Label Selector Operator NotIn
     */
    NotIn: "NotIn",
    /**
     * Label Selector Operator Exists
     */
    Exists: "Exists",
    /**
     * Label Selector Operator DoesNotExist
     */
    DoesNotExist: "DoesNotExist",
};
exports.LicenseType = {
    /**
     * No additional licensing is applied.
     */
    None: "None",
    /**
     * Enables Azure Hybrid User Benefits for Windows VMs.
     */
    Windows_Server: "Windows_Server",
};
exports.LoadBalancerSku = {
    /**
     * Use a a standard Load Balancer. This is the recommended Load Balancer SKU. For more information about on working with the load balancer in the managed cluster, see the [standard Load Balancer](https://docs.microsoft.com/azure/aks/load-balancer-standard) article.
     */
    Standard: "standard",
    /**
     * Use a basic Load Balancer with limited functionality.
     */
    Basic: "basic",
};
exports.LocalDNSForwardDestination = {
    /**
     * Forward DNS queries from localDNS to cluster CoreDNS.
     */
    ClusterCoreDNS: "ClusterCoreDNS",
    /**
     * Forward DNS queries from localDNS to DNS server configured in the VNET. A VNET can have multiple DNS servers configured.
     */
    VnetDNS: "VnetDNS",
};
exports.LocalDNSForwardPolicy = {
    /**
     * Implements sequential upstream DNS server selection. See [forward plugin](https://coredns.io/plugins/forward) for more information.
     */
    Sequential: "Sequential",
    /**
     * Implements round robin upstream DNS server selection. See [forward plugin](https://coredns.io/plugins/forward) for more information.
     */
    RoundRobin: "RoundRobin",
    /**
     * Implements random upstream DNS server selection. See [forward plugin](https://coredns.io/plugins/forward) for more information.
     */
    Random: "Random",
};
exports.LocalDNSMode = {
    /**
     * If the current orchestrator version supports this feature, prefer enabling localDNS.
     */
    Preferred: "Preferred",
    /**
     * Enable localDNS.
     */
    Required: "Required",
    /**
     * Disable localDNS.
     */
    Disabled: "Disabled",
};
exports.LocalDNSProtocol = {
    /**
     * Prefer UDP protocol for connections from localDNS to upstream DNS server.
     */
    PreferUDP: "PreferUDP",
    /**
     * Enforce TCP protocol for connections from localDNS to upstream DNS server.
     */
    ForceTCP: "ForceTCP",
};
exports.LocalDNSQueryLogging = {
    /**
     * Enables error logging in localDNS. See [errors plugin](https://coredns.io/plugins/errors) for more information.
     */
    Error: "Error",
    /**
     * Enables query logging in localDNS. See [log plugin](https://coredns.io/plugins/log) for more information.
     */
    Log: "Log",
};
exports.LocalDNSServeStale = {
    /**
     * Serve stale data with verification. First verify that an entry is still unavailable from the source before sending the expired entry to the client. See [cache plugin](https://coredns.io/plugins/cache) for more information.
     */
    Verify: "Verify",
    /**
     * Serve stale data immediately. Send the expired entry to the client before checking to see if the entry is available from the source. See [cache plugin](https://coredns.io/plugins/cache) for more information.
     */
    Immediate: "Immediate",
    /**
     * Disable serving stale data.
     */
    Disable: "Disable",
};
exports.ManagedClusterSKUName = {
    /**
     * Base option for the AKS control plane.
     */
    Base: "Base",
    /**
     * Automatic clusters are optimized to run most production workloads with configuration that follows AKS best practices and recommendations for cluster and workload setup, scalability, and security. For more details about Automatic clusters see aka.ms/aks/automatic.
     */
    Automatic: "Automatic",
};
exports.ManagedClusterSKUTier = {
    /**
     * Cluster has premium capabilities in addition to all of the capabilities included in 'Standard'. Premium enables selection of LongTermSupport (aka.ms/aks/lts) for certain Kubernetes versions.
     */
    Premium: "Premium",
    /**
     * Recommended for mission-critical and production workloads. Includes Kubernetes control plane autoscaling, workload-intensive testing, and up to 5,000 nodes per cluster. Guarantees 99.95% availability of the Kubernetes API server endpoint for clusters that use Availability Zones and 99.9% of availability for clusters that don't use Availability Zones.
     */
    Standard: "Standard",
    /**
     * The cluster management is free, but charged for VM, storage, and networking usage. Best for experimenting, learning, simple testing, or workloads with fewer than 10 nodes. Not recommended for production use cases.
     */
    Free: "Free",
};
exports.ManagedClusterUpgradeType = {
    /**
     * Full upgrades the control plane and all agent pools of the target ManagedClusters. Requires the ManagedClusterUpgradeSpec.KubernetesVersion property to be set.
     */
    Full: "Full",
    /**
     * NodeImageOnly upgrades only the node images of the target ManagedClusters. Requires the ManagedClusterUpgradeSpec.KubernetesVersion property to NOT be set.
     */
    NodeImageOnly: "NodeImageOnly",
    /**
     * ControlPlaneOnly upgrades only targets the KubernetesVersion of the ManagedClusters and will not be applied to the AgentPool. Requires the ManagedClusterUpgradeSpec.KubernetesVersion property to be set.
     */
    ControlPlaneOnly: "ControlPlaneOnly",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.NetworkDataplane = {
    /**
     * Use Azure network dataplane.
     */
    Azure: "azure",
    /**
     * Use Cilium network dataplane. See [Azure CNI Powered by Cilium](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium) for more information.
     */
    Cilium: "cilium",
};
exports.NetworkMode = {
    /**
     * No bridge is created. Intra-VM Pod to Pod communication is through IP routes created by Azure CNI. See [Transparent Mode](https://docs.microsoft.com/azure/aks/faq#transparent-mode) for more information.
     */
    Transparent: "transparent",
    /**
     * This is no longer supported
     */
    Bridge: "bridge",
};
exports.NetworkPlugin = {
    /**
     * Use the Azure CNI network plugin. See [Azure CNI (advanced) networking](https://docs.microsoft.com/azure/aks/concepts-network#azure-cni-advanced-networking) for more information.
     */
    Azure: "azure",
    /**
     * Use the Kubenet network plugin. See [Kubenet (basic) networking](https://docs.microsoft.com/azure/aks/concepts-network#kubenet-basic-networking) for more information.
     */
    Kubenet: "kubenet",
    /**
     * No CNI plugin is pre-installed. See [BYO CNI](https://docs.microsoft.com/en-us/azure/aks/use-byo-cni) for more information.
     */
    None: "none",
};
exports.NetworkPluginMode = {
    /**
     * Used with networkPlugin=azure, pods are given IPs from the PodCIDR address space but use Azure Routing Domains rather than Kubenet's method of route tables. For more information visit https://aka.ms/aks/azure-cni-overlay.
     */
    Overlay: "overlay",
};
exports.NetworkPolicy = {
    /**
     * Network policies will not be enforced. This is the default value when NetworkPolicy is not specified.
     */
    None: "none",
    /**
     * Use Calico network policies. See [differences between Azure and Calico policies](https://docs.microsoft.com/azure/aks/use-network-policies#differences-between-azure-and-calico-policies-and-their-capabilities) for more information.
     */
    Calico: "calico",
    /**
     * Use Azure network policies. See [differences between Azure and Calico policies](https://docs.microsoft.com/azure/aks/use-network-policies#differences-between-azure-and-calico-policies-and-their-capabilities) for more information.
     */
    Azure: "azure",
    /**
     * Use Cilium to enforce network policies. This requires networkDataplane to be 'cilium'.
     */
    Cilium: "cilium",
};
exports.NginxIngressControllerType = {
    /**
     * The default NginxIngressController will be created. Users can edit the default NginxIngressController Custom Resource to configure load balancer annotations.
     */
    AnnotationControlled: "AnnotationControlled",
    /**
     * The default NginxIngressController will be created and the operator will provision an external loadbalancer with it. Any annotation to make the default loadbalancer internal will be overwritten.
     */
    External: "External",
    /**
     * The default NginxIngressController will be created and the operator will provision an internal loadbalancer with it. Any annotation to make the default loadbalancer external will be overwritten.
     */
    Internal: "Internal",
    /**
     * The default Ingress Controller will not be created. It will not be deleted by the system if it exists. Users should delete the default NginxIngressController Custom Resource manually if desired.
     */
    None: "None",
};
exports.NodeImageSelectionType = {
    /**
     * Use the latest image version when upgrading nodes. Clusters may use different image versions (e.g., 'AKSUbuntu-1804gen2containerd-2021.10.12' and 'AKSUbuntu-1804gen2containerd-2021.10.19') because, for example, the latest available version is different in different regions.
     */
    Latest: "Latest",
    /**
     * The image versions to upgrade nodes to are selected as described below: for each node pool in managed clusters affected by the update run, the system selects the latest image version such that it is available across all other node pools (in all other clusters) of the same image type. As a result, all node pools of the same image type will be upgraded to the same image version. For example, if the latest image version for image type 'AKSUbuntu-1804gen2containerd' is 'AKSUbuntu-1804gen2containerd-2021.10.12' for a node pool in cluster A in region X, and is 'AKSUbuntu-1804gen2containerd-2021.10.17' for a node pool in cluster B in region Y, the system will upgrade both node pools to image version 'AKSUbuntu-1804gen2containerd-2021.10.12'.
     */
    Consistent: "Consistent",
    /**
     * Upgrade the nodes to the custom image versions. When set, update run will use node image versions provided in customNodeImageVersions to upgrade the nodes. If set, customNodeImageVersions must not be empty.
     */
    Custom: "Custom",
};
exports.NodeOSUpgradeChannel = {
    /**
     * No attempt to update your machines OS will be made either by OS or by rolling VHDs. This means you are responsible for your security updates
     */
    None: "None",
    /**
     * OS updates will be applied automatically through the OS built-in patching infrastructure. Newly scaled in machines will be unpatched initially and will be patched at some point by the OS's infrastructure. Behavior of this option depends on the OS in question. Ubuntu and Mariner apply security patches through unattended upgrade roughly once a day around 06:00 UTC. Windows does not apply security patches automatically and so for them this option is equivalent to None till further notice
     */
    Unmanaged: "Unmanaged",
    /**
     * AKS will update the nodes with a newly patched VHD containing security fixes and bugfixes on a weekly cadence. With the VHD update machines will be rolling reimaged to that VHD following maintenance windows and surge settings. No extra VHD cost is incurred when choosing this option as AKS hosts the images.
     */
    NodeImage: "NodeImage",
    /**
     * AKS downloads and updates the nodes with tested security updates. These updates honor the maintenance window settings and produce a new VHD that is used on new nodes. On some occasions it's not possible to apply the updates in place, in such cases the existing nodes will also be re-imaged to the newly produced VHD in order to apply the changes. This option incurs an extra cost of hosting the new Security Patch VHDs in your resource group for just in time consumption.
     */
    SecurityPatch: "SecurityPatch",
};
exports.NodeProvisioningDefaultNodePools = {
    /**
     * No Karpenter NodePools are provisioned automatically. Automatic scaling will not happen unless the user creates one or more NodePool CRD instances.
     */
    None: "None",
    /**
     * A standard set of Karpenter NodePools are provisioned
     */
    Auto: "Auto",
};
exports.NodeProvisioningMode = {
    /**
     * Nodes are provisioned manually by the user
     */
    Manual: "Manual",
    /**
     * Nodes are provisioned automatically by AKS using Karpenter (See aka.ms/aks/nap for more details). Fixed size Node Pools can still be created, but autoscaling Node Pools cannot be. (See aka.ms/aks/nap for more details).
     */
    Auto: "Auto",
};
exports.OSDiskType = {
    /**
     * Azure replicates the operating system disk for a virtual machine to Azure storage to avoid data loss should the VM need to be relocated to another host. Since containers aren't designed to have local state persisted, this behavior offers limited value while providing some drawbacks, including slower node provisioning and higher read/write latency.
     */
    Managed: "Managed",
    /**
     * Ephemeral OS disks are stored only on the host machine, just like a temporary disk. This provides lower read/write latency, along with faster node scaling and cluster upgrades.
     */
    Ephemeral: "Ephemeral",
};
exports.OSSKU = {
    /**
     * Use Ubuntu as the OS for node images.
     */
    Ubuntu: "Ubuntu",
    /**
     * Use AzureLinux as the OS for node images. Azure Linux is a container-optimized Linux distro built by Microsoft, visit https://aka.ms/azurelinux for more information.
     */
    AzureLinux: "AzureLinux",
    /**
     * Use AzureLinux3 as the OS for node images. Azure Linux is a container-optimized Linux distro built by Microsoft, visit https://aka.ms/azurelinux for more information. For limitations, visit https://aka.ms/aks/node-images. For OS migration guidance, see https://aka.ms/aks/upgrade-os-version.
     */
    AzureLinux3: "AzureLinux3",
    /**
     * Deprecated OSSKU. Microsoft recommends that new deployments choose 'AzureLinux' instead.
     */
    CBLMariner: "CBLMariner",
    /**
     * Use Windows2019 as the OS for node images. Unsupported for system node pools. Windows2019 only supports Windows2019 containers; it cannot run Windows2022 containers and vice versa.
     */
    Windows2019: "Windows2019",
    /**
     * Use Windows2022 as the OS for node images. Unsupported for system node pools. Windows2022 only supports Windows2022 containers; it cannot run Windows2019 containers and vice versa.
     */
    Windows2022: "Windows2022",
    /**
     * Use Ubuntu2204 as the OS for node images, however, Ubuntu 22.04 may not be supported for all nodepools. For limitations and supported kubernetes versions, see https://aka.ms/aks/supported-ubuntu-versions
     */
    Ubuntu2204: "Ubuntu2204",
};
exports.OSType = {
    /**
     * Use Linux.
     */
    Linux: "Linux",
    /**
     * Use Windows.
     */
    Windows: "Windows",
};
exports.Operator = {
    /**
     * The value of the key should be in the given list.
     */
    In: "In",
    /**
     * The value of the key should not be in the given list.
     */
    NotIn: "NotIn",
    /**
     * The value of the key should exist.
     */
    Exists: "Exists",
    /**
     * The value of the key should not exist.
     */
    DoesNotExist: "DoesNotExist",
};
exports.OutboundType = {
    /**
     * The load balancer is used for egress through an AKS assigned public IP. This supports Kubernetes services of type 'loadBalancer'. For more information see [outbound type loadbalancer](https://docs.microsoft.com/azure/aks/egress-outboundtype#outbound-type-of-loadbalancer).
     */
    LoadBalancer: "loadBalancer",
    /**
     * Egress paths must be defined by the user. This is an advanced scenario and requires proper network configuration. For more information see [outbound type userDefinedRouting](https://docs.microsoft.com/azure/aks/egress-outboundtype#outbound-type-of-userdefinedrouting).
     */
    UserDefinedRouting: "userDefinedRouting",
    /**
     * The AKS-managed NAT gateway is used for egress.
     */
    ManagedNATGateway: "managedNATGateway",
    /**
     * The user-assigned NAT gateway associated to the cluster subnet is used for egress. This is an advanced scenario and requires proper network configuration.
     */
    UserAssignedNATGateway: "userAssignedNATGateway",
    /**
     * The AKS cluster is not set with any outbound-type. All AKS nodes follows Azure VM default outbound behavior. Please refer to https://azure.microsoft.com/en-us/updates/default-outbound-access-for-vms-in-azure-will-be-retired-transition-to-a-new-method-of-internet-access/
     */
    None: "none",
};
exports.PlacementType = {
    /**
     * PickAll picks all clusters that satisfy the rules.
     */
    PickAll: "PickAll",
    /**
     * PickFixed picks a fixed set of clusters.
     */
    PickFixed: "PickFixed",
};
exports.PodIPAllocationMode = {
    /**
     * Each node gets allocated with a non-contiguous list of IP addresses assignable to pods. This is better for maximizing a small to medium subnet of size /16 or smaller. The Azure CNI cluster with dynamic IP allocation defaults to this mode if the customer does not explicitly specify a podIPAllocationMode
     */
    DynamicIndividual: "DynamicIndividual",
    /**
     * Each node is statically allocated CIDR block(s) of size /28 = 16 IPs per block to satisfy the maxPods per node. Number of CIDR blocks >= (maxPods / 16). The block, rather than a single IP, counts against the Azure Vnet Private IP limit of 65K. Therefore block mode is suitable for running larger workloads with more than the current limit of 65K pods in a cluster. This mode is better suited to scale with larger subnets of /15 or bigger
     */
    StaticBlock: "StaticBlock",
};
exports.PolicyRule = {
    /**
     * Deny all network traffic.
     */
    DenyAll: "DenyAll",
    /**
     * Allow all network traffic.
     */
    AllowAll: "AllowAll",
    /**
     * Allow traffic within the same namespace.
     */
    AllowSameNamespace: "AllowSameNamespace",
};
exports.PropagationType = {
    /**
     * Using ClusterResourcePlacement.
     */
    Placement: "Placement",
};
exports.PropertySelectorOperator = {
    /**
     * Gt dictates Fleet to select cluster if its observed value of a given property is greater than the value specified in the requirement.
     */
    Gt: "Gt",
    /**
     * Ge dictates Fleet to select cluster if its observed value of a given property is greater than or equal to the value specified in the requirement.
     */
    Ge: "Ge",
    /**
     * Eq dictates Fleet to select cluster if its observed value of a given property is equal to the values specified in the requirement.
     */
    Eq: "Eq",
    /**
     * Ne dictates Fleet to select cluster if its observed value of a given property is not equal to the values specified in the requirement.
     */
    Ne: "Ne",
    /**
     * Lt dictates Fleet to select cluster if its observed value of a given property is less than the value specified in the requirement.
     */
    Lt: "Lt",
    /**
     * Le dictates Fleet to select cluster if its observed value of a given property is less than or equal to the value specified in the requirement.
     */
    Le: "Le",
};
exports.Protocol = {
    /**
     * TCP protocol.
     */
    TCP: "TCP",
    /**
     * UDP protocol.
     */
    UDP: "UDP",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    /**
     * Use an implicitly created system assigned managed identity to manage cluster resources. Master components in the control plane such as kube-controller-manager will use the system assigned managed identity to manipulate Azure resources.
     */
    SystemAssigned: "SystemAssigned",
    /**
     * Use a user-specified identity to manage cluster resources. Master components in the control plane such as kube-controller-manager will use the specified user assigned managed identity to manipulate Azure resources.
     */
    UserAssigned: "UserAssigned",
    /**
     * Do not use a managed identity for the Managed Cluster, service principal will be used instead.
     */
    None: "None",
};
exports.RestrictionLevel = {
    /**
     * All RBAC permissions are allowed on the managed node resource group
     */
    Unrestricted: "Unrestricted",
    /**
     * Only *&#47;read RBAC permissions allowed on the managed node resource group
     */
    ReadOnly: "ReadOnly",
};
exports.ScaleDownMode = {
    /**
     * Create new instances during scale up and remove instances during scale down.
     */
    Delete: "Delete",
    /**
     * Attempt to start deallocated instances (if they exist) during scale up and deallocate instances during scale down.
     */
    Deallocate: "Deallocate",
};
exports.ScaleSetEvictionPolicy = {
    /**
     * Nodes in the underlying Scale Set of the node pool are deleted when they're evicted.
     */
    Delete: "Delete",
    /**
     * Nodes in the underlying Scale Set of the node pool are set to the stopped-deallocated state upon eviction. Nodes in the stopped-deallocated state count against your compute quota and can cause issues with cluster scaling or upgrading.
     */
    Deallocate: "Deallocate",
};
exports.ScaleSetPriority = {
    /**
     * Spot priority VMs will be used. There is no SLA for spot nodes. See [spot on AKS](https://docs.microsoft.com/azure/aks/spot-node-pool) for more information.
     */
    Spot: "Spot",
    /**
     * Regular VMs will be used.
     */
    Regular: "Regular",
};
exports.ScriptType = {
    /**
     * Bash script.
     */
    Bash: "Bash",
    /**
     * PowerShell script.
     */
    PowerShell: "PowerShell",
};
exports.ServiceMeshMode = {
    /**
     * Istio deployed as an AKS addon.
     */
    Istio: "Istio",
    /**
     * Mesh is disabled.
     */
    Disabled: "Disabled",
};
exports.SnapshotType = {
    /**
     * The snapshot is a snapshot of a node pool.
     */
    NodePool: "NodePool",
};
exports.TaintEffect = {
    /**
     * Do not allow new pods to schedule onto the node unless they tolerate the taint, but allow all pods submitted to Kubelet without going through the scheduler to start, and allow all already-running pods to continue running. Enforced by the scheduler.
     */
    NoSchedule: "NoSchedule",
};
exports.TolerationOperator = {
    /**
     * Toleration Operator Exists
     */
    Exists: "Exists",
    /**
     * Toleration Operator Equal
     */
    Equal: "Equal",
};
exports.Type = {
    /**
     * First week of the month.
     */
    First: "First",
    /**
     * Second week of the month.
     */
    Second: "Second",
    /**
     * Third week of the month.
     */
    Third: "Third",
    /**
     * Fourth week of the month.
     */
    Fourth: "Fourth",
    /**
     * Last week of the month.
     */
    Last: "Last",
};
exports.UndrainableNodeBehavior = {
    /**
     * AKS will cordon the blocked nodes and replace them with surge nodes during upgrade. The blocked nodes will be cordoned and replaced by surge nodes. The blocked nodes will have label 'kubernetes.azure.com/upgrade-status:Quarantined'. A surge node will be retained for each blocked node. A best-effort attempt will be made to delete all other surge nodes. If there are enough surge nodes to replace blocked nodes, then the upgrade operation and the managed cluster will be in failed state. Otherwise, the upgrade operation and the managed cluster will be in canceled state.
     */
    Cordon: "Cordon",
    /**
     * AKS will mark the blocked nodes schedulable, but the blocked nodes are not upgraded. A best-effort attempt will be made to delete all surge nodes. The upgrade operation and the managed cluster will be in failed state if there are any blocked nodes.
     */
    Schedule: "Schedule",
};
exports.UpgradeChannel = {
    /**
     * Automatically upgrade the cluster to the latest supported patch release on the latest supported minor version. In cases where the cluster is at a version of Kubernetes that is at an N-2 minor version where N is the latest supported minor version, the cluster first upgrades to the latest supported patch version on N-1 minor version. For example, if a cluster is running version 1.17.7 and versions 1.17.9, 1.18.4, 1.18.6, and 1.19.1 are available, your cluster first is upgraded to 1.18.6, then is upgraded to 1.19.1.
     */
    Rapid: "rapid",
    /**
     * Automatically upgrade the cluster to the latest supported patch release on minor version N-1, where N is the latest supported minor version. For example, if a cluster is running version 1.17.7 and versions 1.17.9, 1.18.4, 1.18.6, and 1.19.1 are available, your cluster is upgraded to 1.18.6.
     */
    Stable: "stable",
    /**
     * Automatically upgrade the cluster to the latest supported patch version when it becomes available while keeping the minor version the same. For example, if a cluster is running version 1.17.7 and versions 1.17.9, 1.18.4, 1.18.6, and 1.19.1 are available, your cluster is upgraded to 1.17.9.
     */
    Patch: "patch",
    /**
     * Automatically upgrade the node image to the latest version available. Consider using nodeOSUpgradeChannel instead as that allows you to configure node OS patching separate from Kubernetes version patching
     */
    Node_image: "node-image",
    /**
     * Disables auto-upgrades and keeps the cluster at its current version of Kubernetes.
     */
    None: "none",
};
exports.WeekDay = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.WorkloadRuntime = {
    /**
     * Nodes will use Kubelet to run standard OCI container workloads.
     */
    OCIContainer: "OCIContainer",
    /**
     * Nodes will use Krustlet to run WASM workloads using the WASI provider (Preview).
     */
    WasmWasi: "WasmWasi",
    /**
     * Nodes can use (Kata + Cloud Hypervisor + Hyper-V) to enable Nested VM-based pods. Due to the use Hyper-V, AKS node OS itself is a nested VM (the root OS) of Hyper-V. Thus it can only be used with VM series that support Nested Virtualization such as Dv3 series.
     */
    KataVmIsolation: "KataVmIsolation",
};
