// @pulumi/kubernetes/scheduling/v1alpha2/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadPatch = exports.WorkloadList = exports.Workload = exports.PodGroupPatch = exports.PodGroupList = exports.PodGroup = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.PodGroup = null;
utilities.lazyLoad(exports, ["PodGroup"], () => require("./podGroup"));
exports.PodGroupList = null;
utilities.lazyLoad(exports, ["PodGroupList"], () => require("./podGroupList"));
exports.PodGroupPatch = null;
utilities.lazyLoad(exports, ["PodGroupPatch"], () => require("./podGroupPatch"));
exports.Workload = null;
utilities.lazyLoad(exports, ["Workload"], () => require("./workload"));
exports.WorkloadList = null;
utilities.lazyLoad(exports, ["WorkloadList"], () => require("./workloadList"));
exports.WorkloadPatch = null;
utilities.lazyLoad(exports, ["WorkloadPatch"], () => require("./workloadPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:scheduling.k8s.io/v1alpha2:PodGroup":
                return new exports.PodGroup(name, undefined, { urn });
            case "kubernetes:scheduling.k8s.io/v1alpha2:PodGroupList":
                return new exports.PodGroupList(name, undefined, { urn });
            case "kubernetes:scheduling.k8s.io/v1alpha2:PodGroupPatch":
                return new exports.PodGroupPatch(name, undefined, { urn });
            case "kubernetes:scheduling.k8s.io/v1alpha2:Workload":
                return new exports.Workload(name, undefined, { urn });
            case "kubernetes:scheduling.k8s.io/v1alpha2:WorkloadList":
                return new exports.WorkloadList(name, undefined, { urn });
            case "kubernetes:scheduling.k8s.io/v1alpha2:WorkloadPatch":
                return new exports.WorkloadPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "scheduling.k8s.io/v1alpha2", _module);
