// @pulumi/azure-native/storage/storageAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccount = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * The storage account.
 *
 * Uses Azure REST API version 2024-01-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2022-09-01, 2023-01-01, 2023-04-01, 2023-05-01, 2025-01-01, 2025-06-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storage [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class StorageAccount extends pulumi.CustomResource {
    /**
     * Get an existing StorageAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageAccount(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:storage:StorageAccount';
    /**
     * Returns true if the given object is an instance of StorageAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageAccount.__pulumiType;
    }
    /**
     * Create a StorageAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.sku === undefined && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["accessTier"] = args?.accessTier;
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["allowBlobPublicAccess"] = args?.allowBlobPublicAccess;
            resourceInputs["allowCrossTenantReplication"] = args?.allowCrossTenantReplication;
            resourceInputs["allowSharedKeyAccess"] = args?.allowSharedKeyAccess;
            resourceInputs["allowedCopyScope"] = args?.allowedCopyScope;
            resourceInputs["azureFilesIdentityBasedAuthentication"] = args?.azureFilesIdentityBasedAuthentication;
            resourceInputs["customDomain"] = args?.customDomain;
            resourceInputs["defaultToOAuthAuthentication"] = args?.defaultToOAuthAuthentication;
            resourceInputs["dnsEndpointType"] = args?.dnsEndpointType;
            resourceInputs["enableExtendedGroups"] = args?.enableExtendedGroups;
            resourceInputs["enableHttpsTrafficOnly"] = args?.enableHttpsTrafficOnly;
            resourceInputs["enableNfsV3"] = args?.enableNfsV3;
            resourceInputs["encryption"] = args ? (args.encryption ? pulumi.output(args.encryption).apply(inputs.storage.encryptionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["extendedLocation"] = args?.extendedLocation;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["immutableStorageWithVersioning"] = args?.immutableStorageWithVersioning;
            resourceInputs["isHnsEnabled"] = args?.isHnsEnabled;
            resourceInputs["isLocalUserEnabled"] = args?.isLocalUserEnabled;
            resourceInputs["isSftpEnabled"] = args?.isSftpEnabled;
            resourceInputs["keyPolicy"] = args?.keyPolicy;
            resourceInputs["kind"] = args?.kind;
            resourceInputs["largeFileSharesState"] = args?.largeFileSharesState;
            resourceInputs["location"] = args?.location;
            resourceInputs["minimumTlsVersion"] = args?.minimumTlsVersion;
            resourceInputs["networkRuleSet"] = args ? (args.networkRuleSet ? pulumi.output(args.networkRuleSet).apply(inputs.storage.networkRuleSetArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["publicNetworkAccess"] = args?.publicNetworkAccess;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["routingPreference"] = args?.routingPreference;
            resourceInputs["sasPolicy"] = args ? (args.sasPolicy ? pulumi.output(args.sasPolicy).apply(inputs.storage.sasPolicyArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["sku"] = args?.sku;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["accountMigrationInProgress"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["blobRestoreStatus"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["failoverInProgress"] = undefined /*out*/;
            resourceInputs["geoReplicationStats"] = undefined /*out*/;
            resourceInputs["isSkuConversionBlocked"] = undefined /*out*/;
            resourceInputs["keyCreationTime"] = undefined /*out*/;
            resourceInputs["lastGeoFailoverTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["primaryEndpoints"] = undefined /*out*/;
            resourceInputs["primaryLocation"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["secondaryEndpoints"] = undefined /*out*/;
            resourceInputs["secondaryLocation"] = undefined /*out*/;
            resourceInputs["statusOfPrimary"] = undefined /*out*/;
            resourceInputs["statusOfSecondary"] = undefined /*out*/;
            resourceInputs["storageAccountSkuConversionStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessTier"] = undefined /*out*/;
            resourceInputs["accountMigrationInProgress"] = undefined /*out*/;
            resourceInputs["allowBlobPublicAccess"] = undefined /*out*/;
            resourceInputs["allowCrossTenantReplication"] = undefined /*out*/;
            resourceInputs["allowSharedKeyAccess"] = undefined /*out*/;
            resourceInputs["allowedCopyScope"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureFilesIdentityBasedAuthentication"] = undefined /*out*/;
            resourceInputs["blobRestoreStatus"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["customDomain"] = undefined /*out*/;
            resourceInputs["defaultToOAuthAuthentication"] = undefined /*out*/;
            resourceInputs["dnsEndpointType"] = undefined /*out*/;
            resourceInputs["enableExtendedGroups"] = undefined /*out*/;
            resourceInputs["enableHttpsTrafficOnly"] = undefined /*out*/;
            resourceInputs["enableNfsV3"] = undefined /*out*/;
            resourceInputs["encryption"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["failoverInProgress"] = undefined /*out*/;
            resourceInputs["geoReplicationStats"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["immutableStorageWithVersioning"] = undefined /*out*/;
            resourceInputs["isHnsEnabled"] = undefined /*out*/;
            resourceInputs["isLocalUserEnabled"] = undefined /*out*/;
            resourceInputs["isSftpEnabled"] = undefined /*out*/;
            resourceInputs["isSkuConversionBlocked"] = undefined /*out*/;
            resourceInputs["keyCreationTime"] = undefined /*out*/;
            resourceInputs["keyPolicy"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["largeFileSharesState"] = undefined /*out*/;
            resourceInputs["lastGeoFailoverTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["minimumTlsVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkRuleSet"] = undefined /*out*/;
            resourceInputs["primaryEndpoints"] = undefined /*out*/;
            resourceInputs["primaryLocation"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicNetworkAccess"] = undefined /*out*/;
            resourceInputs["routingPreference"] = undefined /*out*/;
            resourceInputs["sasPolicy"] = undefined /*out*/;
            resourceInputs["secondaryEndpoints"] = undefined /*out*/;
            resourceInputs["secondaryLocation"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["statusOfPrimary"] = undefined /*out*/;
            resourceInputs["statusOfSecondary"] = undefined /*out*/;
            resourceInputs["storageAccountSkuConversionStatus"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20150501preview:StorageAccount" }, { type: "azure-native:storage/v20150615:StorageAccount" }, { type: "azure-native:storage/v20160101:StorageAccount" }, { type: "azure-native:storage/v20160501:StorageAccount" }, { type: "azure-native:storage/v20161201:StorageAccount" }, { type: "azure-native:storage/v20170601:StorageAccount" }, { type: "azure-native:storage/v20171001:StorageAccount" }, { type: "azure-native:storage/v20180201:StorageAccount" }, { type: "azure-native:storage/v20180301preview:StorageAccount" }, { type: "azure-native:storage/v20180701:StorageAccount" }, { type: "azure-native:storage/v20181101:StorageAccount" }, { type: "azure-native:storage/v20190401:StorageAccount" }, { type: "azure-native:storage/v20190601:StorageAccount" }, { type: "azure-native:storage/v20200801preview:StorageAccount" }, { type: "azure-native:storage/v20210101:StorageAccount" }, { type: "azure-native:storage/v20210201:StorageAccount" }, { type: "azure-native:storage/v20210401:StorageAccount" }, { type: "azure-native:storage/v20210601:StorageAccount" }, { type: "azure-native:storage/v20210801:StorageAccount" }, { type: "azure-native:storage/v20210901:StorageAccount" }, { type: "azure-native:storage/v20220501:StorageAccount" }, { type: "azure-native:storage/v20220901:StorageAccount" }, { type: "azure-native:storage/v20230101:StorageAccount" }, { type: "azure-native:storage/v20230401:StorageAccount" }, { type: "azure-native:storage/v20230501:StorageAccount" }, { type: "azure-native:storage/v20240101:StorageAccount" }, { type: "azure-native:storage/v20250101:StorageAccount" }, { type: "azure-native:storage/v20250601:StorageAccount" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StorageAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StorageAccount = StorageAccount;
