// @pulumi/gcp/compute/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getForwardingRule = exports.getDiskIamPolicyOutput = exports.getDiskIamPolicy = exports.getDiskOutput = exports.getDisk = exports.getDefaultServiceAccountOutput = exports.getDefaultServiceAccount = exports.getCertificateOutput = exports.getCertificate = exports.getBackendServiceIamPolicyOutput = exports.getBackendServiceIamPolicy = exports.getBackendServiceOutput = exports.getBackendService = exports.getBackendBucketIamPolicyOutput = exports.getBackendBucketIamPolicy = exports.getBackendBucketOutput = exports.getBackendBucket = exports.getAddressesOutput = exports.getAddresses = exports.getAddressOutput = exports.getAddress = exports.FutureReservation = exports.ForwardingRule = exports.FirewallPolicyWithRules = exports.FirewallPolicyRule = exports.FirewallPolicyAssociation = exports.FirewallPolicy = exports.Firewall = exports.ExternalVpnGateway = exports.DiskResourcePolicyAttachment = exports.DiskIamPolicy = exports.DiskIamMember = exports.DiskIamBinding = exports.DiskAsyncReplication = exports.Disk = exports.CrossSiteNetwork = exports.CaExternalAccountKey = exports.BackendServiceSignedUrlKey = exports.BackendServiceIamPolicy = exports.BackendServiceIamMember = exports.BackendServiceIamBinding = exports.BackendService = exports.BackendBucketSignedUrlKey = exports.BackendBucketIamPolicy = exports.BackendBucketIamMember = exports.BackendBucketIamBinding = exports.BackendBucket = exports.Autoscaler = exports.AttachedDisk = exports.Address = void 0;
exports.getNetworkAttachment = exports.getNetworkOutput = exports.getNetwork = exports.getNetblockIPRangesOutput = exports.getNetblockIPRanges = exports.getMachineTypesOutput = exports.getMachineTypes = exports.getMachineImageIamPolicyOutput = exports.getMachineImageIamPolicy = exports.getLBIPRangesOutput = exports.getLBIPRanges = exports.getInterconnectLocationsOutput = exports.getInterconnectLocations = exports.getInterconnectLocationOutput = exports.getInterconnectLocation = exports.getInstantSnapshotIamPolicyOutput = exports.getInstantSnapshotIamPolicy = exports.getInstanceTemplateIamPolicyOutput = exports.getInstanceTemplateIamPolicy = exports.getInstanceTemplateOutput = exports.getInstanceTemplate = exports.getInstanceSerialPortOutput = exports.getInstanceSerialPort = exports.getInstanceIamPolicyOutput = exports.getInstanceIamPolicy = exports.getInstanceGuestAttributesOutput = exports.getInstanceGuestAttributes = exports.getInstanceGroupManagerOutput = exports.getInstanceGroupManager = exports.getInstanceGroupOutput = exports.getInstanceGroup = exports.getInstanceOutput = exports.getInstance = exports.getImagesOutput = exports.getImages = exports.getImageIamPolicyOutput = exports.getImageIamPolicy = exports.getImageOutput = exports.getImage = exports.getHealthCheckOutput = exports.getHealthCheck = exports.getHcVpnGatewayOutput = exports.getHcVpnGateway = exports.getGlobalForwardingRuleOutput = exports.getGlobalForwardingRule = exports.getGlobalAddressOutput = exports.getGlobalAddress = exports.getForwardingRulesOutput = exports.getForwardingRules = exports.getForwardingRuleOutput = void 0;
exports.getRouterNat = exports.getRouterOutput = exports.getRouter = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getReservationSubBlockOutput = exports.getReservationSubBlock = exports.getReservationBlockOutput = exports.getReservationBlock = exports.getReservationOutput = exports.getReservation = exports.getRegionsOutput = exports.getRegions = exports.getRegionSslCertificateOutput = exports.getRegionSslCertificate = exports.getRegionSecurityPolicyOutput = exports.getRegionSecurityPolicy = exports.getRegionSSLPolicyOutput = exports.getRegionSSLPolicy = exports.getRegionNetworkEndpointGroupOutput = exports.getRegionNetworkEndpointGroup = exports.getRegionInstantSnapshotIamPolicyOutput = exports.getRegionInstantSnapshotIamPolicy = exports.getRegionInstanceTemplateOutput = exports.getRegionInstanceTemplate = exports.getRegionInstanceGroupManagerOutput = exports.getRegionInstanceGroupManager = exports.getRegionInstanceGroupOutput = exports.getRegionInstanceGroup = exports.getRegionDiskIamPolicyOutput = exports.getRegionDiskIamPolicy = exports.getRegionDiskOutput = exports.getRegionDisk = exports.getRegionBackendServiceIamPolicyOutput = exports.getRegionBackendServiceIamPolicy = exports.getRegionBackendServiceOutput = exports.getRegionBackendService = exports.getRegionBackendBucketIamPolicyOutput = exports.getRegionBackendBucketIamPolicy = exports.getNodeTypesOutput = exports.getNodeTypes = exports.getNetworksOutput = exports.getNetworks = exports.getNetworkPeeringOutput = exports.getNetworkPeering = exports.getNetworkEndpointGroupsOutput = exports.getNetworkEndpointGroups = exports.getNetworkEndpointGroupOutput = exports.getNetworkEndpointGroup = exports.getNetworkAttachmentOutput = void 0;
exports.InstanceIAMMember = exports.InstanceIAMBinding = exports.InstanceGroupNamedPort = exports.InstanceGroupMembership = exports.InstanceGroupManager = exports.InstanceGroup = exports.InstanceFromTemplate = exports.InstanceFromMachineImage = exports.Instance = exports.ImageIamPolicy = exports.ImageIamMember = exports.ImageIamBinding = exports.Image = exports.HttpsHealthCheck = exports.HttpHealthCheck = exports.HealthCheck = exports.HaVpnGateway = exports.GlobalNetworkEndpointGroup = exports.GlobalNetworkEndpoint = exports.GlobalForwardingRule = exports.GlobalAddress = exports.getZonesOutput = exports.getZones = exports.getVPNGatewayOutput = exports.getVPNGateway = exports.getSubnetworksOutput = exports.getSubnetworks = exports.getSubnetworkIamPolicyOutput = exports.getSubnetworkIamPolicy = exports.getSubnetworkOutput = exports.getSubnetwork = exports.getStoragePoolTypesOutput = exports.getStoragePoolTypes = exports.getStoragePoolIamPolicyOutput = exports.getStoragePoolIamPolicy = exports.getStoragePoolOutput = exports.getStoragePool = exports.getSnapshotIamPolicyOutput = exports.getSnapshotIamPolicy = exports.getSnapshotOutput = exports.getSnapshot = exports.getSecurityPolicyOutput = exports.getSecurityPolicy = exports.getSSLPolicyOutput = exports.getSSLPolicy = exports.getRoutersOutput = exports.getRouters = exports.getRouterStatusOutput = exports.getRouterStatus = exports.getRouterNatOutput = void 0;
exports.RegionBackendBucketIamBinding = exports.RegionBackendBucket = exports.RegionAutoscaler = exports.PublicDelegatedPrefix = exports.PublicAdvertisedPrefix = exports.ProjectMetadataItem = exports.ProjectMetadata = exports.ProjectDefaultNetworkTier = exports.ProjectCloudArmorTier = exports.PreviewFeature = exports.PerInstanceConfig = exports.PacketMirroring = exports.OrganizationSecurityPolicyRule = exports.OrganizationSecurityPolicyAssociation = exports.OrganizationSecurityPolicy = exports.NodeTemplate = exports.NodeGroup = exports.NetworkPeeringRoutesConfig = exports.NetworkPeering = exports.NetworkFirewallPolicyWithRules = exports.NetworkFirewallPolicyRule = exports.NetworkFirewallPolicyPacketMirroringRule = exports.NetworkFirewallPolicyAssociation = exports.NetworkFirewallPolicy = exports.NetworkEndpointList = exports.NetworkEndpointGroup = exports.NetworkEndpoint = exports.NetworkEdgeSecurityService = exports.NetworkAttachment = exports.Network = exports.MangedSslCertificate = exports.ManagedSslCertificate = exports.MachineImageIamPolicy = exports.MachineImageIamMember = exports.MachineImageIamBinding = exports.MachineImage = exports.InterconnectGroup = exports.InterconnectAttachmentGroup = exports.InterconnectAttachment = exports.Interconnect = exports.InstantSnapshotIamPolicy = exports.InstantSnapshotIamMember = exports.InstantSnapshotIamBinding = exports.InstantSnapshot = exports.InstanceTemplateIamPolicy = exports.InstanceTemplateIamMember = exports.InstanceTemplateIamBinding = exports.InstanceTemplate = exports.InstanceSettings = exports.InstanceIAMPolicy = void 0;
exports.RouterPeer = exports.RouterNatAddress = exports.RouterNat = exports.RouterNamedSet = exports.RouterInterface = exports.Router = exports.Route = exports.RolloutPlan = exports.ResourcePolicyAttachment = exports.ResourcePolicy = exports.ResizeRequest = exports.Reservation = exports.RegionUrlMap = exports.RegionTargetTcpProxy = exports.RegionTargetHttpsProxy = exports.RegionTargetHttpProxy = exports.RegionSslPolicy = exports.RegionSslCertificate = exports.RegionSecurityPolicyRule = exports.RegionSecurityPolicy = exports.RegionResizeRequest = exports.RegionPerInstanceConfig = exports.RegionNetworkFirewallPolicyWithRules = exports.RegionNetworkFirewallPolicyRule = exports.RegionNetworkFirewallPolicyAssociation = exports.RegionNetworkFirewallPolicy = exports.RegionNetworkEndpointGroup = exports.RegionNetworkEndpoint = exports.RegionInstantSnapshotIamPolicy = exports.RegionInstantSnapshotIamMember = exports.RegionInstantSnapshotIamBinding = exports.RegionInstantSnapshot = exports.RegionInstanceTemplate = exports.RegionInstanceGroupManager = exports.RegionHealthSource = exports.RegionHealthCheck = exports.RegionHealthAggregationPolicy = exports.RegionDiskResourcePolicyAttachment = exports.RegionDiskIamPolicy = exports.RegionDiskIamMember = exports.RegionDiskIamBinding = exports.RegionDisk = exports.RegionCompositeHealthCheck = exports.RegionCommitment = exports.RegionBackendServiceIamPolicy = exports.RegionBackendServiceIamMember = exports.RegionBackendServiceIamBinding = exports.RegionBackendService = exports.RegionBackendBucketIamPolicy = exports.RegionBackendBucketIamMember = void 0;
exports.WireGroup = exports.VPNTunnel = exports.VPNGateway = exports.URLMap = exports.TargetTCPProxy = exports.TargetSSLProxy = exports.TargetPool = exports.TargetInstance = exports.TargetHttpsProxy = exports.TargetHttpProxy = exports.TargetGrpcProxy = exports.SubnetworkIAMPolicy = exports.SubnetworkIAMMember = exports.SubnetworkIAMBinding = exports.Subnetwork = exports.StoragePoolIamPolicy = exports.StoragePoolIamMember = exports.StoragePoolIamBinding = exports.StoragePool = exports.SSLPolicy = exports.SSLCertificate = exports.SnapshotSettings = exports.SnapshotIamPolicy = exports.SnapshotIamMember = exports.SnapshotIamBinding = exports.Snapshot = exports.SharedVPCServiceProject = exports.SharedVPCHostProject = exports.ServiceAttachment = exports.SecurityScanConfig = exports.SecurityPolicyRule = exports.SecurityPolicy = exports.routerStatusOutput = exports.routerStatus = exports.RouterRoutePolicy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Address = null;
utilities.lazyLoad(exports, ["Address"], () => require("./address"));
exports.AttachedDisk = null;
utilities.lazyLoad(exports, ["AttachedDisk"], () => require("./attachedDisk"));
exports.Autoscaler = null;
utilities.lazyLoad(exports, ["Autoscaler"], () => require("./autoscaler"));
exports.BackendBucket = null;
utilities.lazyLoad(exports, ["BackendBucket"], () => require("./backendBucket"));
exports.BackendBucketIamBinding = null;
utilities.lazyLoad(exports, ["BackendBucketIamBinding"], () => require("./backendBucketIamBinding"));
exports.BackendBucketIamMember = null;
utilities.lazyLoad(exports, ["BackendBucketIamMember"], () => require("./backendBucketIamMember"));
exports.BackendBucketIamPolicy = null;
utilities.lazyLoad(exports, ["BackendBucketIamPolicy"], () => require("./backendBucketIamPolicy"));
exports.BackendBucketSignedUrlKey = null;
utilities.lazyLoad(exports, ["BackendBucketSignedUrlKey"], () => require("./backendBucketSignedUrlKey"));
exports.BackendService = null;
utilities.lazyLoad(exports, ["BackendService"], () => require("./backendService"));
exports.BackendServiceIamBinding = null;
utilities.lazyLoad(exports, ["BackendServiceIamBinding"], () => require("./backendServiceIamBinding"));
exports.BackendServiceIamMember = null;
utilities.lazyLoad(exports, ["BackendServiceIamMember"], () => require("./backendServiceIamMember"));
exports.BackendServiceIamPolicy = null;
utilities.lazyLoad(exports, ["BackendServiceIamPolicy"], () => require("./backendServiceIamPolicy"));
exports.BackendServiceSignedUrlKey = null;
utilities.lazyLoad(exports, ["BackendServiceSignedUrlKey"], () => require("./backendServiceSignedUrlKey"));
exports.CaExternalAccountKey = null;
utilities.lazyLoad(exports, ["CaExternalAccountKey"], () => require("./caExternalAccountKey"));
exports.CrossSiteNetwork = null;
utilities.lazyLoad(exports, ["CrossSiteNetwork"], () => require("./crossSiteNetwork"));
exports.Disk = null;
utilities.lazyLoad(exports, ["Disk"], () => require("./disk"));
exports.DiskAsyncReplication = null;
utilities.lazyLoad(exports, ["DiskAsyncReplication"], () => require("./diskAsyncReplication"));
exports.DiskIamBinding = null;
utilities.lazyLoad(exports, ["DiskIamBinding"], () => require("./diskIamBinding"));
exports.DiskIamMember = null;
utilities.lazyLoad(exports, ["DiskIamMember"], () => require("./diskIamMember"));
exports.DiskIamPolicy = null;
utilities.lazyLoad(exports, ["DiskIamPolicy"], () => require("./diskIamPolicy"));
exports.DiskResourcePolicyAttachment = null;
utilities.lazyLoad(exports, ["DiskResourcePolicyAttachment"], () => require("./diskResourcePolicyAttachment"));
exports.ExternalVpnGateway = null;
utilities.lazyLoad(exports, ["ExternalVpnGateway"], () => require("./externalVpnGateway"));
exports.Firewall = null;
utilities.lazyLoad(exports, ["Firewall"], () => require("./firewall"));
exports.FirewallPolicy = null;
utilities.lazyLoad(exports, ["FirewallPolicy"], () => require("./firewallPolicy"));
exports.FirewallPolicyAssociation = null;
utilities.lazyLoad(exports, ["FirewallPolicyAssociation"], () => require("./firewallPolicyAssociation"));
exports.FirewallPolicyRule = null;
utilities.lazyLoad(exports, ["FirewallPolicyRule"], () => require("./firewallPolicyRule"));
exports.FirewallPolicyWithRules = null;
utilities.lazyLoad(exports, ["FirewallPolicyWithRules"], () => require("./firewallPolicyWithRules"));
exports.ForwardingRule = null;
utilities.lazyLoad(exports, ["ForwardingRule"], () => require("./forwardingRule"));
exports.FutureReservation = null;
utilities.lazyLoad(exports, ["FutureReservation"], () => require("./futureReservation"));
exports.getAddress = null;
exports.getAddressOutput = null;
utilities.lazyLoad(exports, ["getAddress", "getAddressOutput"], () => require("./getAddress"));
exports.getAddresses = null;
exports.getAddressesOutput = null;
utilities.lazyLoad(exports, ["getAddresses", "getAddressesOutput"], () => require("./getAddresses"));
exports.getBackendBucket = null;
exports.getBackendBucketOutput = null;
utilities.lazyLoad(exports, ["getBackendBucket", "getBackendBucketOutput"], () => require("./getBackendBucket"));
exports.getBackendBucketIamPolicy = null;
exports.getBackendBucketIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackendBucketIamPolicy", "getBackendBucketIamPolicyOutput"], () => require("./getBackendBucketIamPolicy"));
exports.getBackendService = null;
exports.getBackendServiceOutput = null;
utilities.lazyLoad(exports, ["getBackendService", "getBackendServiceOutput"], () => require("./getBackendService"));
exports.getBackendServiceIamPolicy = null;
exports.getBackendServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackendServiceIamPolicy", "getBackendServiceIamPolicyOutput"], () => require("./getBackendServiceIamPolicy"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getDefaultServiceAccount = null;
exports.getDefaultServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getDefaultServiceAccount", "getDefaultServiceAccountOutput"], () => require("./getDefaultServiceAccount"));
exports.getDisk = null;
exports.getDiskOutput = null;
utilities.lazyLoad(exports, ["getDisk", "getDiskOutput"], () => require("./getDisk"));
exports.getDiskIamPolicy = null;
exports.getDiskIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getDiskIamPolicy", "getDiskIamPolicyOutput"], () => require("./getDiskIamPolicy"));
exports.getForwardingRule = null;
exports.getForwardingRuleOutput = null;
utilities.lazyLoad(exports, ["getForwardingRule", "getForwardingRuleOutput"], () => require("./getForwardingRule"));
exports.getForwardingRules = null;
exports.getForwardingRulesOutput = null;
utilities.lazyLoad(exports, ["getForwardingRules", "getForwardingRulesOutput"], () => require("./getForwardingRules"));
exports.getGlobalAddress = null;
exports.getGlobalAddressOutput = null;
utilities.lazyLoad(exports, ["getGlobalAddress", "getGlobalAddressOutput"], () => require("./getGlobalAddress"));
exports.getGlobalForwardingRule = null;
exports.getGlobalForwardingRuleOutput = null;
utilities.lazyLoad(exports, ["getGlobalForwardingRule", "getGlobalForwardingRuleOutput"], () => require("./getGlobalForwardingRule"));
exports.getHcVpnGateway = null;
exports.getHcVpnGatewayOutput = null;
utilities.lazyLoad(exports, ["getHcVpnGateway", "getHcVpnGatewayOutput"], () => require("./getHcVpnGateway"));
exports.getHealthCheck = null;
exports.getHealthCheckOutput = null;
utilities.lazyLoad(exports, ["getHealthCheck", "getHealthCheckOutput"], () => require("./getHealthCheck"));
exports.getImage = null;
exports.getImageOutput = null;
utilities.lazyLoad(exports, ["getImage", "getImageOutput"], () => require("./getImage"));
exports.getImageIamPolicy = null;
exports.getImageIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getImageIamPolicy", "getImageIamPolicyOutput"], () => require("./getImageIamPolicy"));
exports.getImages = null;
exports.getImagesOutput = null;
utilities.lazyLoad(exports, ["getImages", "getImagesOutput"], () => require("./getImages"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getInstanceGroup = null;
exports.getInstanceGroupOutput = null;
utilities.lazyLoad(exports, ["getInstanceGroup", "getInstanceGroupOutput"], () => require("./getInstanceGroup"));
exports.getInstanceGroupManager = null;
exports.getInstanceGroupManagerOutput = null;
utilities.lazyLoad(exports, ["getInstanceGroupManager", "getInstanceGroupManagerOutput"], () => require("./getInstanceGroupManager"));
exports.getInstanceGuestAttributes = null;
exports.getInstanceGuestAttributesOutput = null;
utilities.lazyLoad(exports, ["getInstanceGuestAttributes", "getInstanceGuestAttributesOutput"], () => require("./getInstanceGuestAttributes"));
exports.getInstanceIamPolicy = null;
exports.getInstanceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getInstanceIamPolicy", "getInstanceIamPolicyOutput"], () => require("./getInstanceIamPolicy"));
exports.getInstanceSerialPort = null;
exports.getInstanceSerialPortOutput = null;
utilities.lazyLoad(exports, ["getInstanceSerialPort", "getInstanceSerialPortOutput"], () => require("./getInstanceSerialPort"));
exports.getInstanceTemplate = null;
exports.getInstanceTemplateOutput = null;
utilities.lazyLoad(exports, ["getInstanceTemplate", "getInstanceTemplateOutput"], () => require("./getInstanceTemplate"));
exports.getInstanceTemplateIamPolicy = null;
exports.getInstanceTemplateIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getInstanceTemplateIamPolicy", "getInstanceTemplateIamPolicyOutput"], () => require("./getInstanceTemplateIamPolicy"));
exports.getInstantSnapshotIamPolicy = null;
exports.getInstantSnapshotIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getInstantSnapshotIamPolicy", "getInstantSnapshotIamPolicyOutput"], () => require("./getInstantSnapshotIamPolicy"));
exports.getInterconnectLocation = null;
exports.getInterconnectLocationOutput = null;
utilities.lazyLoad(exports, ["getInterconnectLocation", "getInterconnectLocationOutput"], () => require("./getInterconnectLocation"));
exports.getInterconnectLocations = null;
exports.getInterconnectLocationsOutput = null;
utilities.lazyLoad(exports, ["getInterconnectLocations", "getInterconnectLocationsOutput"], () => require("./getInterconnectLocations"));
exports.getLBIPRanges = null;
exports.getLBIPRangesOutput = null;
utilities.lazyLoad(exports, ["getLBIPRanges", "getLBIPRangesOutput"], () => require("./getLBIPRanges"));
exports.getMachineImageIamPolicy = null;
exports.getMachineImageIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getMachineImageIamPolicy", "getMachineImageIamPolicyOutput"], () => require("./getMachineImageIamPolicy"));
exports.getMachineTypes = null;
exports.getMachineTypesOutput = null;
utilities.lazyLoad(exports, ["getMachineTypes", "getMachineTypesOutput"], () => require("./getMachineTypes"));
exports.getNetblockIPRanges = null;
exports.getNetblockIPRangesOutput = null;
utilities.lazyLoad(exports, ["getNetblockIPRanges", "getNetblockIPRangesOutput"], () => require("./getNetblockIPRanges"));
exports.getNetwork = null;
exports.getNetworkOutput = null;
utilities.lazyLoad(exports, ["getNetwork", "getNetworkOutput"], () => require("./getNetwork"));
exports.getNetworkAttachment = null;
exports.getNetworkAttachmentOutput = null;
utilities.lazyLoad(exports, ["getNetworkAttachment", "getNetworkAttachmentOutput"], () => require("./getNetworkAttachment"));
exports.getNetworkEndpointGroup = null;
exports.getNetworkEndpointGroupOutput = null;
utilities.lazyLoad(exports, ["getNetworkEndpointGroup", "getNetworkEndpointGroupOutput"], () => require("./getNetworkEndpointGroup"));
exports.getNetworkEndpointGroups = null;
exports.getNetworkEndpointGroupsOutput = null;
utilities.lazyLoad(exports, ["getNetworkEndpointGroups", "getNetworkEndpointGroupsOutput"], () => require("./getNetworkEndpointGroups"));
exports.getNetworkPeering = null;
exports.getNetworkPeeringOutput = null;
utilities.lazyLoad(exports, ["getNetworkPeering", "getNetworkPeeringOutput"], () => require("./getNetworkPeering"));
exports.getNetworks = null;
exports.getNetworksOutput = null;
utilities.lazyLoad(exports, ["getNetworks", "getNetworksOutput"], () => require("./getNetworks"));
exports.getNodeTypes = null;
exports.getNodeTypesOutput = null;
utilities.lazyLoad(exports, ["getNodeTypes", "getNodeTypesOutput"], () => require("./getNodeTypes"));
exports.getRegionBackendBucketIamPolicy = null;
exports.getRegionBackendBucketIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionBackendBucketIamPolicy", "getRegionBackendBucketIamPolicyOutput"], () => require("./getRegionBackendBucketIamPolicy"));
exports.getRegionBackendService = null;
exports.getRegionBackendServiceOutput = null;
utilities.lazyLoad(exports, ["getRegionBackendService", "getRegionBackendServiceOutput"], () => require("./getRegionBackendService"));
exports.getRegionBackendServiceIamPolicy = null;
exports.getRegionBackendServiceIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionBackendServiceIamPolicy", "getRegionBackendServiceIamPolicyOutput"], () => require("./getRegionBackendServiceIamPolicy"));
exports.getRegionDisk = null;
exports.getRegionDiskOutput = null;
utilities.lazyLoad(exports, ["getRegionDisk", "getRegionDiskOutput"], () => require("./getRegionDisk"));
exports.getRegionDiskIamPolicy = null;
exports.getRegionDiskIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionDiskIamPolicy", "getRegionDiskIamPolicyOutput"], () => require("./getRegionDiskIamPolicy"));
exports.getRegionInstanceGroup = null;
exports.getRegionInstanceGroupOutput = null;
utilities.lazyLoad(exports, ["getRegionInstanceGroup", "getRegionInstanceGroupOutput"], () => require("./getRegionInstanceGroup"));
exports.getRegionInstanceGroupManager = null;
exports.getRegionInstanceGroupManagerOutput = null;
utilities.lazyLoad(exports, ["getRegionInstanceGroupManager", "getRegionInstanceGroupManagerOutput"], () => require("./getRegionInstanceGroupManager"));
exports.getRegionInstanceTemplate = null;
exports.getRegionInstanceTemplateOutput = null;
utilities.lazyLoad(exports, ["getRegionInstanceTemplate", "getRegionInstanceTemplateOutput"], () => require("./getRegionInstanceTemplate"));
exports.getRegionInstantSnapshotIamPolicy = null;
exports.getRegionInstantSnapshotIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionInstantSnapshotIamPolicy", "getRegionInstantSnapshotIamPolicyOutput"], () => require("./getRegionInstantSnapshotIamPolicy"));
exports.getRegionNetworkEndpointGroup = null;
exports.getRegionNetworkEndpointGroupOutput = null;
utilities.lazyLoad(exports, ["getRegionNetworkEndpointGroup", "getRegionNetworkEndpointGroupOutput"], () => require("./getRegionNetworkEndpointGroup"));
exports.getRegionSSLPolicy = null;
exports.getRegionSSLPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionSSLPolicy", "getRegionSSLPolicyOutput"], () => require("./getRegionSSLPolicy"));
exports.getRegionSecurityPolicy = null;
exports.getRegionSecurityPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegionSecurityPolicy", "getRegionSecurityPolicyOutput"], () => require("./getRegionSecurityPolicy"));
exports.getRegionSslCertificate = null;
exports.getRegionSslCertificateOutput = null;
utilities.lazyLoad(exports, ["getRegionSslCertificate", "getRegionSslCertificateOutput"], () => require("./getRegionSslCertificate"));
exports.getRegions = null;
exports.getRegionsOutput = null;
utilities.lazyLoad(exports, ["getRegions", "getRegionsOutput"], () => require("./getRegions"));
exports.getReservation = null;
exports.getReservationOutput = null;
utilities.lazyLoad(exports, ["getReservation", "getReservationOutput"], () => require("./getReservation"));
exports.getReservationBlock = null;
exports.getReservationBlockOutput = null;
utilities.lazyLoad(exports, ["getReservationBlock", "getReservationBlockOutput"], () => require("./getReservationBlock"));
exports.getReservationSubBlock = null;
exports.getReservationSubBlockOutput = null;
utilities.lazyLoad(exports, ["getReservationSubBlock", "getReservationSubBlockOutput"], () => require("./getReservationSubBlock"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.getRouter = null;
exports.getRouterOutput = null;
utilities.lazyLoad(exports, ["getRouter", "getRouterOutput"], () => require("./getRouter"));
exports.getRouterNat = null;
exports.getRouterNatOutput = null;
utilities.lazyLoad(exports, ["getRouterNat", "getRouterNatOutput"], () => require("./getRouterNat"));
exports.getRouterStatus = null;
exports.getRouterStatusOutput = null;
utilities.lazyLoad(exports, ["getRouterStatus", "getRouterStatusOutput"], () => require("./getRouterStatus"));
exports.getRouters = null;
exports.getRoutersOutput = null;
utilities.lazyLoad(exports, ["getRouters", "getRoutersOutput"], () => require("./getRouters"));
exports.getSSLPolicy = null;
exports.getSSLPolicyOutput = null;
utilities.lazyLoad(exports, ["getSSLPolicy", "getSSLPolicyOutput"], () => require("./getSSLPolicy"));
exports.getSecurityPolicy = null;
exports.getSecurityPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecurityPolicy", "getSecurityPolicyOutput"], () => require("./getSecurityPolicy"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getSnapshotIamPolicy = null;
exports.getSnapshotIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getSnapshotIamPolicy", "getSnapshotIamPolicyOutput"], () => require("./getSnapshotIamPolicy"));
exports.getStoragePool = null;
exports.getStoragePoolOutput = null;
utilities.lazyLoad(exports, ["getStoragePool", "getStoragePoolOutput"], () => require("./getStoragePool"));
exports.getStoragePoolIamPolicy = null;
exports.getStoragePoolIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getStoragePoolIamPolicy", "getStoragePoolIamPolicyOutput"], () => require("./getStoragePoolIamPolicy"));
exports.getStoragePoolTypes = null;
exports.getStoragePoolTypesOutput = null;
utilities.lazyLoad(exports, ["getStoragePoolTypes", "getStoragePoolTypesOutput"], () => require("./getStoragePoolTypes"));
exports.getSubnetwork = null;
exports.getSubnetworkOutput = null;
utilities.lazyLoad(exports, ["getSubnetwork", "getSubnetworkOutput"], () => require("./getSubnetwork"));
exports.getSubnetworkIamPolicy = null;
exports.getSubnetworkIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getSubnetworkIamPolicy", "getSubnetworkIamPolicyOutput"], () => require("./getSubnetworkIamPolicy"));
exports.getSubnetworks = null;
exports.getSubnetworksOutput = null;
utilities.lazyLoad(exports, ["getSubnetworks", "getSubnetworksOutput"], () => require("./getSubnetworks"));
exports.getVPNGateway = null;
exports.getVPNGatewayOutput = null;
utilities.lazyLoad(exports, ["getVPNGateway", "getVPNGatewayOutput"], () => require("./getVPNGateway"));
exports.getZones = null;
exports.getZonesOutput = null;
utilities.lazyLoad(exports, ["getZones", "getZonesOutput"], () => require("./getZones"));
exports.GlobalAddress = null;
utilities.lazyLoad(exports, ["GlobalAddress"], () => require("./globalAddress"));
exports.GlobalForwardingRule = null;
utilities.lazyLoad(exports, ["GlobalForwardingRule"], () => require("./globalForwardingRule"));
exports.GlobalNetworkEndpoint = null;
utilities.lazyLoad(exports, ["GlobalNetworkEndpoint"], () => require("./globalNetworkEndpoint"));
exports.GlobalNetworkEndpointGroup = null;
utilities.lazyLoad(exports, ["GlobalNetworkEndpointGroup"], () => require("./globalNetworkEndpointGroup"));
exports.HaVpnGateway = null;
utilities.lazyLoad(exports, ["HaVpnGateway"], () => require("./haVpnGateway"));
exports.HealthCheck = null;
utilities.lazyLoad(exports, ["HealthCheck"], () => require("./healthCheck"));
exports.HttpHealthCheck = null;
utilities.lazyLoad(exports, ["HttpHealthCheck"], () => require("./httpHealthCheck"));
exports.HttpsHealthCheck = null;
utilities.lazyLoad(exports, ["HttpsHealthCheck"], () => require("./httpsHealthCheck"));
exports.Image = null;
utilities.lazyLoad(exports, ["Image"], () => require("./image"));
exports.ImageIamBinding = null;
utilities.lazyLoad(exports, ["ImageIamBinding"], () => require("./imageIamBinding"));
exports.ImageIamMember = null;
utilities.lazyLoad(exports, ["ImageIamMember"], () => require("./imageIamMember"));
exports.ImageIamPolicy = null;
utilities.lazyLoad(exports, ["ImageIamPolicy"], () => require("./imageIamPolicy"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceFromMachineImage = null;
utilities.lazyLoad(exports, ["InstanceFromMachineImage"], () => require("./instanceFromMachineImage"));
exports.InstanceFromTemplate = null;
utilities.lazyLoad(exports, ["InstanceFromTemplate"], () => require("./instanceFromTemplate"));
exports.InstanceGroup = null;
utilities.lazyLoad(exports, ["InstanceGroup"], () => require("./instanceGroup"));
exports.InstanceGroupManager = null;
utilities.lazyLoad(exports, ["InstanceGroupManager"], () => require("./instanceGroupManager"));
exports.InstanceGroupMembership = null;
utilities.lazyLoad(exports, ["InstanceGroupMembership"], () => require("./instanceGroupMembership"));
exports.InstanceGroupNamedPort = null;
utilities.lazyLoad(exports, ["InstanceGroupNamedPort"], () => require("./instanceGroupNamedPort"));
exports.InstanceIAMBinding = null;
utilities.lazyLoad(exports, ["InstanceIAMBinding"], () => require("./instanceIAMBinding"));
exports.InstanceIAMMember = null;
utilities.lazyLoad(exports, ["InstanceIAMMember"], () => require("./instanceIAMMember"));
exports.InstanceIAMPolicy = null;
utilities.lazyLoad(exports, ["InstanceIAMPolicy"], () => require("./instanceIAMPolicy"));
exports.InstanceSettings = null;
utilities.lazyLoad(exports, ["InstanceSettings"], () => require("./instanceSettings"));
exports.InstanceTemplate = null;
utilities.lazyLoad(exports, ["InstanceTemplate"], () => require("./instanceTemplate"));
exports.InstanceTemplateIamBinding = null;
utilities.lazyLoad(exports, ["InstanceTemplateIamBinding"], () => require("./instanceTemplateIamBinding"));
exports.InstanceTemplateIamMember = null;
utilities.lazyLoad(exports, ["InstanceTemplateIamMember"], () => require("./instanceTemplateIamMember"));
exports.InstanceTemplateIamPolicy = null;
utilities.lazyLoad(exports, ["InstanceTemplateIamPolicy"], () => require("./instanceTemplateIamPolicy"));
exports.InstantSnapshot = null;
utilities.lazyLoad(exports, ["InstantSnapshot"], () => require("./instantSnapshot"));
exports.InstantSnapshotIamBinding = null;
utilities.lazyLoad(exports, ["InstantSnapshotIamBinding"], () => require("./instantSnapshotIamBinding"));
exports.InstantSnapshotIamMember = null;
utilities.lazyLoad(exports, ["InstantSnapshotIamMember"], () => require("./instantSnapshotIamMember"));
exports.InstantSnapshotIamPolicy = null;
utilities.lazyLoad(exports, ["InstantSnapshotIamPolicy"], () => require("./instantSnapshotIamPolicy"));
exports.Interconnect = null;
utilities.lazyLoad(exports, ["Interconnect"], () => require("./interconnect"));
exports.InterconnectAttachment = null;
utilities.lazyLoad(exports, ["InterconnectAttachment"], () => require("./interconnectAttachment"));
exports.InterconnectAttachmentGroup = null;
utilities.lazyLoad(exports, ["InterconnectAttachmentGroup"], () => require("./interconnectAttachmentGroup"));
exports.InterconnectGroup = null;
utilities.lazyLoad(exports, ["InterconnectGroup"], () => require("./interconnectGroup"));
exports.MachineImage = null;
utilities.lazyLoad(exports, ["MachineImage"], () => require("./machineImage"));
exports.MachineImageIamBinding = null;
utilities.lazyLoad(exports, ["MachineImageIamBinding"], () => require("./machineImageIamBinding"));
exports.MachineImageIamMember = null;
utilities.lazyLoad(exports, ["MachineImageIamMember"], () => require("./machineImageIamMember"));
exports.MachineImageIamPolicy = null;
utilities.lazyLoad(exports, ["MachineImageIamPolicy"], () => require("./machineImageIamPolicy"));
exports.ManagedSslCertificate = null;
utilities.lazyLoad(exports, ["ManagedSslCertificate"], () => require("./managedSslCertificate"));
exports.MangedSslCertificate = null;
utilities.lazyLoad(exports, ["MangedSslCertificate"], () => require("./mangedSslCertificate"));
exports.Network = null;
utilities.lazyLoad(exports, ["Network"], () => require("./network"));
exports.NetworkAttachment = null;
utilities.lazyLoad(exports, ["NetworkAttachment"], () => require("./networkAttachment"));
exports.NetworkEdgeSecurityService = null;
utilities.lazyLoad(exports, ["NetworkEdgeSecurityService"], () => require("./networkEdgeSecurityService"));
exports.NetworkEndpoint = null;
utilities.lazyLoad(exports, ["NetworkEndpoint"], () => require("./networkEndpoint"));
exports.NetworkEndpointGroup = null;
utilities.lazyLoad(exports, ["NetworkEndpointGroup"], () => require("./networkEndpointGroup"));
exports.NetworkEndpointList = null;
utilities.lazyLoad(exports, ["NetworkEndpointList"], () => require("./networkEndpointList"));
exports.NetworkFirewallPolicy = null;
utilities.lazyLoad(exports, ["NetworkFirewallPolicy"], () => require("./networkFirewallPolicy"));
exports.NetworkFirewallPolicyAssociation = null;
utilities.lazyLoad(exports, ["NetworkFirewallPolicyAssociation"], () => require("./networkFirewallPolicyAssociation"));
exports.NetworkFirewallPolicyPacketMirroringRule = null;
utilities.lazyLoad(exports, ["NetworkFirewallPolicyPacketMirroringRule"], () => require("./networkFirewallPolicyPacketMirroringRule"));
exports.NetworkFirewallPolicyRule = null;
utilities.lazyLoad(exports, ["NetworkFirewallPolicyRule"], () => require("./networkFirewallPolicyRule"));
exports.NetworkFirewallPolicyWithRules = null;
utilities.lazyLoad(exports, ["NetworkFirewallPolicyWithRules"], () => require("./networkFirewallPolicyWithRules"));
exports.NetworkPeering = null;
utilities.lazyLoad(exports, ["NetworkPeering"], () => require("./networkPeering"));
exports.NetworkPeeringRoutesConfig = null;
utilities.lazyLoad(exports, ["NetworkPeeringRoutesConfig"], () => require("./networkPeeringRoutesConfig"));
exports.NodeGroup = null;
utilities.lazyLoad(exports, ["NodeGroup"], () => require("./nodeGroup"));
exports.NodeTemplate = null;
utilities.lazyLoad(exports, ["NodeTemplate"], () => require("./nodeTemplate"));
exports.OrganizationSecurityPolicy = null;
utilities.lazyLoad(exports, ["OrganizationSecurityPolicy"], () => require("./organizationSecurityPolicy"));
exports.OrganizationSecurityPolicyAssociation = null;
utilities.lazyLoad(exports, ["OrganizationSecurityPolicyAssociation"], () => require("./organizationSecurityPolicyAssociation"));
exports.OrganizationSecurityPolicyRule = null;
utilities.lazyLoad(exports, ["OrganizationSecurityPolicyRule"], () => require("./organizationSecurityPolicyRule"));
exports.PacketMirroring = null;
utilities.lazyLoad(exports, ["PacketMirroring"], () => require("./packetMirroring"));
exports.PerInstanceConfig = null;
utilities.lazyLoad(exports, ["PerInstanceConfig"], () => require("./perInstanceConfig"));
exports.PreviewFeature = null;
utilities.lazyLoad(exports, ["PreviewFeature"], () => require("./previewFeature"));
exports.ProjectCloudArmorTier = null;
utilities.lazyLoad(exports, ["ProjectCloudArmorTier"], () => require("./projectCloudArmorTier"));
exports.ProjectDefaultNetworkTier = null;
utilities.lazyLoad(exports, ["ProjectDefaultNetworkTier"], () => require("./projectDefaultNetworkTier"));
exports.ProjectMetadata = null;
utilities.lazyLoad(exports, ["ProjectMetadata"], () => require("./projectMetadata"));
exports.ProjectMetadataItem = null;
utilities.lazyLoad(exports, ["ProjectMetadataItem"], () => require("./projectMetadataItem"));
exports.PublicAdvertisedPrefix = null;
utilities.lazyLoad(exports, ["PublicAdvertisedPrefix"], () => require("./publicAdvertisedPrefix"));
exports.PublicDelegatedPrefix = null;
utilities.lazyLoad(exports, ["PublicDelegatedPrefix"], () => require("./publicDelegatedPrefix"));
exports.RegionAutoscaler = null;
utilities.lazyLoad(exports, ["RegionAutoscaler"], () => require("./regionAutoscaler"));
exports.RegionBackendBucket = null;
utilities.lazyLoad(exports, ["RegionBackendBucket"], () => require("./regionBackendBucket"));
exports.RegionBackendBucketIamBinding = null;
utilities.lazyLoad(exports, ["RegionBackendBucketIamBinding"], () => require("./regionBackendBucketIamBinding"));
exports.RegionBackendBucketIamMember = null;
utilities.lazyLoad(exports, ["RegionBackendBucketIamMember"], () => require("./regionBackendBucketIamMember"));
exports.RegionBackendBucketIamPolicy = null;
utilities.lazyLoad(exports, ["RegionBackendBucketIamPolicy"], () => require("./regionBackendBucketIamPolicy"));
exports.RegionBackendService = null;
utilities.lazyLoad(exports, ["RegionBackendService"], () => require("./regionBackendService"));
exports.RegionBackendServiceIamBinding = null;
utilities.lazyLoad(exports, ["RegionBackendServiceIamBinding"], () => require("./regionBackendServiceIamBinding"));
exports.RegionBackendServiceIamMember = null;
utilities.lazyLoad(exports, ["RegionBackendServiceIamMember"], () => require("./regionBackendServiceIamMember"));
exports.RegionBackendServiceIamPolicy = null;
utilities.lazyLoad(exports, ["RegionBackendServiceIamPolicy"], () => require("./regionBackendServiceIamPolicy"));
exports.RegionCommitment = null;
utilities.lazyLoad(exports, ["RegionCommitment"], () => require("./regionCommitment"));
exports.RegionCompositeHealthCheck = null;
utilities.lazyLoad(exports, ["RegionCompositeHealthCheck"], () => require("./regionCompositeHealthCheck"));
exports.RegionDisk = null;
utilities.lazyLoad(exports, ["RegionDisk"], () => require("./regionDisk"));
exports.RegionDiskIamBinding = null;
utilities.lazyLoad(exports, ["RegionDiskIamBinding"], () => require("./regionDiskIamBinding"));
exports.RegionDiskIamMember = null;
utilities.lazyLoad(exports, ["RegionDiskIamMember"], () => require("./regionDiskIamMember"));
exports.RegionDiskIamPolicy = null;
utilities.lazyLoad(exports, ["RegionDiskIamPolicy"], () => require("./regionDiskIamPolicy"));
exports.RegionDiskResourcePolicyAttachment = null;
utilities.lazyLoad(exports, ["RegionDiskResourcePolicyAttachment"], () => require("./regionDiskResourcePolicyAttachment"));
exports.RegionHealthAggregationPolicy = null;
utilities.lazyLoad(exports, ["RegionHealthAggregationPolicy"], () => require("./regionHealthAggregationPolicy"));
exports.RegionHealthCheck = null;
utilities.lazyLoad(exports, ["RegionHealthCheck"], () => require("./regionHealthCheck"));
exports.RegionHealthSource = null;
utilities.lazyLoad(exports, ["RegionHealthSource"], () => require("./regionHealthSource"));
exports.RegionInstanceGroupManager = null;
utilities.lazyLoad(exports, ["RegionInstanceGroupManager"], () => require("./regionInstanceGroupManager"));
exports.RegionInstanceTemplate = null;
utilities.lazyLoad(exports, ["RegionInstanceTemplate"], () => require("./regionInstanceTemplate"));
exports.RegionInstantSnapshot = null;
utilities.lazyLoad(exports, ["RegionInstantSnapshot"], () => require("./regionInstantSnapshot"));
exports.RegionInstantSnapshotIamBinding = null;
utilities.lazyLoad(exports, ["RegionInstantSnapshotIamBinding"], () => require("./regionInstantSnapshotIamBinding"));
exports.RegionInstantSnapshotIamMember = null;
utilities.lazyLoad(exports, ["RegionInstantSnapshotIamMember"], () => require("./regionInstantSnapshotIamMember"));
exports.RegionInstantSnapshotIamPolicy = null;
utilities.lazyLoad(exports, ["RegionInstantSnapshotIamPolicy"], () => require("./regionInstantSnapshotIamPolicy"));
exports.RegionNetworkEndpoint = null;
utilities.lazyLoad(exports, ["RegionNetworkEndpoint"], () => require("./regionNetworkEndpoint"));
exports.RegionNetworkEndpointGroup = null;
utilities.lazyLoad(exports, ["RegionNetworkEndpointGroup"], () => require("./regionNetworkEndpointGroup"));
exports.RegionNetworkFirewallPolicy = null;
utilities.lazyLoad(exports, ["RegionNetworkFirewallPolicy"], () => require("./regionNetworkFirewallPolicy"));
exports.RegionNetworkFirewallPolicyAssociation = null;
utilities.lazyLoad(exports, ["RegionNetworkFirewallPolicyAssociation"], () => require("./regionNetworkFirewallPolicyAssociation"));
exports.RegionNetworkFirewallPolicyRule = null;
utilities.lazyLoad(exports, ["RegionNetworkFirewallPolicyRule"], () => require("./regionNetworkFirewallPolicyRule"));
exports.RegionNetworkFirewallPolicyWithRules = null;
utilities.lazyLoad(exports, ["RegionNetworkFirewallPolicyWithRules"], () => require("./regionNetworkFirewallPolicyWithRules"));
exports.RegionPerInstanceConfig = null;
utilities.lazyLoad(exports, ["RegionPerInstanceConfig"], () => require("./regionPerInstanceConfig"));
exports.RegionResizeRequest = null;
utilities.lazyLoad(exports, ["RegionResizeRequest"], () => require("./regionResizeRequest"));
exports.RegionSecurityPolicy = null;
utilities.lazyLoad(exports, ["RegionSecurityPolicy"], () => require("./regionSecurityPolicy"));
exports.RegionSecurityPolicyRule = null;
utilities.lazyLoad(exports, ["RegionSecurityPolicyRule"], () => require("./regionSecurityPolicyRule"));
exports.RegionSslCertificate = null;
utilities.lazyLoad(exports, ["RegionSslCertificate"], () => require("./regionSslCertificate"));
exports.RegionSslPolicy = null;
utilities.lazyLoad(exports, ["RegionSslPolicy"], () => require("./regionSslPolicy"));
exports.RegionTargetHttpProxy = null;
utilities.lazyLoad(exports, ["RegionTargetHttpProxy"], () => require("./regionTargetHttpProxy"));
exports.RegionTargetHttpsProxy = null;
utilities.lazyLoad(exports, ["RegionTargetHttpsProxy"], () => require("./regionTargetHttpsProxy"));
exports.RegionTargetTcpProxy = null;
utilities.lazyLoad(exports, ["RegionTargetTcpProxy"], () => require("./regionTargetTcpProxy"));
exports.RegionUrlMap = null;
utilities.lazyLoad(exports, ["RegionUrlMap"], () => require("./regionUrlMap"));
exports.Reservation = null;
utilities.lazyLoad(exports, ["Reservation"], () => require("./reservation"));
exports.ResizeRequest = null;
utilities.lazyLoad(exports, ["ResizeRequest"], () => require("./resizeRequest"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.ResourcePolicyAttachment = null;
utilities.lazyLoad(exports, ["ResourcePolicyAttachment"], () => require("./resourcePolicyAttachment"));
exports.RolloutPlan = null;
utilities.lazyLoad(exports, ["RolloutPlan"], () => require("./rolloutPlan"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
exports.Router = null;
utilities.lazyLoad(exports, ["Router"], () => require("./router"));
exports.RouterInterface = null;
utilities.lazyLoad(exports, ["RouterInterface"], () => require("./routerInterface"));
exports.RouterNamedSet = null;
utilities.lazyLoad(exports, ["RouterNamedSet"], () => require("./routerNamedSet"));
exports.RouterNat = null;
utilities.lazyLoad(exports, ["RouterNat"], () => require("./routerNat"));
exports.RouterNatAddress = null;
utilities.lazyLoad(exports, ["RouterNatAddress"], () => require("./routerNatAddress"));
exports.RouterPeer = null;
utilities.lazyLoad(exports, ["RouterPeer"], () => require("./routerPeer"));
exports.RouterRoutePolicy = null;
utilities.lazyLoad(exports, ["RouterRoutePolicy"], () => require("./routerRoutePolicy"));
exports.routerStatus = null;
exports.routerStatusOutput = null;
utilities.lazyLoad(exports, ["routerStatus", "routerStatusOutput"], () => require("./routerStatus"));
exports.SecurityPolicy = null;
utilities.lazyLoad(exports, ["SecurityPolicy"], () => require("./securityPolicy"));
exports.SecurityPolicyRule = null;
utilities.lazyLoad(exports, ["SecurityPolicyRule"], () => require("./securityPolicyRule"));
exports.SecurityScanConfig = null;
utilities.lazyLoad(exports, ["SecurityScanConfig"], () => require("./securityScanConfig"));
exports.ServiceAttachment = null;
utilities.lazyLoad(exports, ["ServiceAttachment"], () => require("./serviceAttachment"));
exports.SharedVPCHostProject = null;
utilities.lazyLoad(exports, ["SharedVPCHostProject"], () => require("./sharedVPCHostProject"));
exports.SharedVPCServiceProject = null;
utilities.lazyLoad(exports, ["SharedVPCServiceProject"], () => require("./sharedVPCServiceProject"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.SnapshotIamBinding = null;
utilities.lazyLoad(exports, ["SnapshotIamBinding"], () => require("./snapshotIamBinding"));
exports.SnapshotIamMember = null;
utilities.lazyLoad(exports, ["SnapshotIamMember"], () => require("./snapshotIamMember"));
exports.SnapshotIamPolicy = null;
utilities.lazyLoad(exports, ["SnapshotIamPolicy"], () => require("./snapshotIamPolicy"));
exports.SnapshotSettings = null;
utilities.lazyLoad(exports, ["SnapshotSettings"], () => require("./snapshotSettings"));
exports.SSLCertificate = null;
utilities.lazyLoad(exports, ["SSLCertificate"], () => require("./sslcertificate"));
exports.SSLPolicy = null;
utilities.lazyLoad(exports, ["SSLPolicy"], () => require("./sslpolicy"));
exports.StoragePool = null;
utilities.lazyLoad(exports, ["StoragePool"], () => require("./storagePool"));
exports.StoragePoolIamBinding = null;
utilities.lazyLoad(exports, ["StoragePoolIamBinding"], () => require("./storagePoolIamBinding"));
exports.StoragePoolIamMember = null;
utilities.lazyLoad(exports, ["StoragePoolIamMember"], () => require("./storagePoolIamMember"));
exports.StoragePoolIamPolicy = null;
utilities.lazyLoad(exports, ["StoragePoolIamPolicy"], () => require("./storagePoolIamPolicy"));
exports.Subnetwork = null;
utilities.lazyLoad(exports, ["Subnetwork"], () => require("./subnetwork"));
exports.SubnetworkIAMBinding = null;
utilities.lazyLoad(exports, ["SubnetworkIAMBinding"], () => require("./subnetworkIAMBinding"));
exports.SubnetworkIAMMember = null;
utilities.lazyLoad(exports, ["SubnetworkIAMMember"], () => require("./subnetworkIAMMember"));
exports.SubnetworkIAMPolicy = null;
utilities.lazyLoad(exports, ["SubnetworkIAMPolicy"], () => require("./subnetworkIAMPolicy"));
exports.TargetGrpcProxy = null;
utilities.lazyLoad(exports, ["TargetGrpcProxy"], () => require("./targetGrpcProxy"));
exports.TargetHttpProxy = null;
utilities.lazyLoad(exports, ["TargetHttpProxy"], () => require("./targetHttpProxy"));
exports.TargetHttpsProxy = null;
utilities.lazyLoad(exports, ["TargetHttpsProxy"], () => require("./targetHttpsProxy"));
exports.TargetInstance = null;
utilities.lazyLoad(exports, ["TargetInstance"], () => require("./targetInstance"));
exports.TargetPool = null;
utilities.lazyLoad(exports, ["TargetPool"], () => require("./targetPool"));
exports.TargetSSLProxy = null;
utilities.lazyLoad(exports, ["TargetSSLProxy"], () => require("./targetSSLProxy"));
exports.TargetTCPProxy = null;
utilities.lazyLoad(exports, ["TargetTCPProxy"], () => require("./targetTCPProxy"));
exports.URLMap = null;
utilities.lazyLoad(exports, ["URLMap"], () => require("./urlmap"));
exports.VPNGateway = null;
utilities.lazyLoad(exports, ["VPNGateway"], () => require("./vpngateway"));
exports.VPNTunnel = null;
utilities.lazyLoad(exports, ["VPNTunnel"], () => require("./vpntunnel"));
exports.WireGroup = null;
utilities.lazyLoad(exports, ["WireGroup"], () => require("./wireGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:compute/address:Address":
                return new exports.Address(name, undefined, { urn });
            case "gcp:compute/attachedDisk:AttachedDisk":
                return new exports.AttachedDisk(name, undefined, { urn });
            case "gcp:compute/autoscaler:Autoscaler":
                return new exports.Autoscaler(name, undefined, { urn });
            case "gcp:compute/backendBucket:BackendBucket":
                return new exports.BackendBucket(name, undefined, { urn });
            case "gcp:compute/backendBucketIamBinding:BackendBucketIamBinding":
                return new exports.BackendBucketIamBinding(name, undefined, { urn });
            case "gcp:compute/backendBucketIamMember:BackendBucketIamMember":
                return new exports.BackendBucketIamMember(name, undefined, { urn });
            case "gcp:compute/backendBucketIamPolicy:BackendBucketIamPolicy":
                return new exports.BackendBucketIamPolicy(name, undefined, { urn });
            case "gcp:compute/backendBucketSignedUrlKey:BackendBucketSignedUrlKey":
                return new exports.BackendBucketSignedUrlKey(name, undefined, { urn });
            case "gcp:compute/backendService:BackendService":
                return new exports.BackendService(name, undefined, { urn });
            case "gcp:compute/backendServiceIamBinding:BackendServiceIamBinding":
                return new exports.BackendServiceIamBinding(name, undefined, { urn });
            case "gcp:compute/backendServiceIamMember:BackendServiceIamMember":
                return new exports.BackendServiceIamMember(name, undefined, { urn });
            case "gcp:compute/backendServiceIamPolicy:BackendServiceIamPolicy":
                return new exports.BackendServiceIamPolicy(name, undefined, { urn });
            case "gcp:compute/backendServiceSignedUrlKey:BackendServiceSignedUrlKey":
                return new exports.BackendServiceSignedUrlKey(name, undefined, { urn });
            case "gcp:compute/caExternalAccountKey:CaExternalAccountKey":
                return new exports.CaExternalAccountKey(name, undefined, { urn });
            case "gcp:compute/crossSiteNetwork:CrossSiteNetwork":
                return new exports.CrossSiteNetwork(name, undefined, { urn });
            case "gcp:compute/disk:Disk":
                return new exports.Disk(name, undefined, { urn });
            case "gcp:compute/diskAsyncReplication:DiskAsyncReplication":
                return new exports.DiskAsyncReplication(name, undefined, { urn });
            case "gcp:compute/diskIamBinding:DiskIamBinding":
                return new exports.DiskIamBinding(name, undefined, { urn });
            case "gcp:compute/diskIamMember:DiskIamMember":
                return new exports.DiskIamMember(name, undefined, { urn });
            case "gcp:compute/diskIamPolicy:DiskIamPolicy":
                return new exports.DiskIamPolicy(name, undefined, { urn });
            case "gcp:compute/diskResourcePolicyAttachment:DiskResourcePolicyAttachment":
                return new exports.DiskResourcePolicyAttachment(name, undefined, { urn });
            case "gcp:compute/externalVpnGateway:ExternalVpnGateway":
                return new exports.ExternalVpnGateway(name, undefined, { urn });
            case "gcp:compute/firewall:Firewall":
                return new exports.Firewall(name, undefined, { urn });
            case "gcp:compute/firewallPolicy:FirewallPolicy":
                return new exports.FirewallPolicy(name, undefined, { urn });
            case "gcp:compute/firewallPolicyAssociation:FirewallPolicyAssociation":
                return new exports.FirewallPolicyAssociation(name, undefined, { urn });
            case "gcp:compute/firewallPolicyRule:FirewallPolicyRule":
                return new exports.FirewallPolicyRule(name, undefined, { urn });
            case "gcp:compute/firewallPolicyWithRules:FirewallPolicyWithRules":
                return new exports.FirewallPolicyWithRules(name, undefined, { urn });
            case "gcp:compute/forwardingRule:ForwardingRule":
                return new exports.ForwardingRule(name, undefined, { urn });
            case "gcp:compute/futureReservation:FutureReservation":
                return new exports.FutureReservation(name, undefined, { urn });
            case "gcp:compute/globalAddress:GlobalAddress":
                return new exports.GlobalAddress(name, undefined, { urn });
            case "gcp:compute/globalForwardingRule:GlobalForwardingRule":
                return new exports.GlobalForwardingRule(name, undefined, { urn });
            case "gcp:compute/globalNetworkEndpoint:GlobalNetworkEndpoint":
                return new exports.GlobalNetworkEndpoint(name, undefined, { urn });
            case "gcp:compute/globalNetworkEndpointGroup:GlobalNetworkEndpointGroup":
                return new exports.GlobalNetworkEndpointGroup(name, undefined, { urn });
            case "gcp:compute/haVpnGateway:HaVpnGateway":
                return new exports.HaVpnGateway(name, undefined, { urn });
            case "gcp:compute/healthCheck:HealthCheck":
                return new exports.HealthCheck(name, undefined, { urn });
            case "gcp:compute/httpHealthCheck:HttpHealthCheck":
                return new exports.HttpHealthCheck(name, undefined, { urn });
            case "gcp:compute/httpsHealthCheck:HttpsHealthCheck":
                return new exports.HttpsHealthCheck(name, undefined, { urn });
            case "gcp:compute/image:Image":
                return new exports.Image(name, undefined, { urn });
            case "gcp:compute/imageIamBinding:ImageIamBinding":
                return new exports.ImageIamBinding(name, undefined, { urn });
            case "gcp:compute/imageIamMember:ImageIamMember":
                return new exports.ImageIamMember(name, undefined, { urn });
            case "gcp:compute/imageIamPolicy:ImageIamPolicy":
                return new exports.ImageIamPolicy(name, undefined, { urn });
            case "gcp:compute/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "gcp:compute/instanceFromMachineImage:InstanceFromMachineImage":
                return new exports.InstanceFromMachineImage(name, undefined, { urn });
            case "gcp:compute/instanceFromTemplate:InstanceFromTemplate":
                return new exports.InstanceFromTemplate(name, undefined, { urn });
            case "gcp:compute/instanceGroup:InstanceGroup":
                return new exports.InstanceGroup(name, undefined, { urn });
            case "gcp:compute/instanceGroupManager:InstanceGroupManager":
                return new exports.InstanceGroupManager(name, undefined, { urn });
            case "gcp:compute/instanceGroupMembership:InstanceGroupMembership":
                return new exports.InstanceGroupMembership(name, undefined, { urn });
            case "gcp:compute/instanceGroupNamedPort:InstanceGroupNamedPort":
                return new exports.InstanceGroupNamedPort(name, undefined, { urn });
            case "gcp:compute/instanceIAMBinding:InstanceIAMBinding":
                return new exports.InstanceIAMBinding(name, undefined, { urn });
            case "gcp:compute/instanceIAMMember:InstanceIAMMember":
                return new exports.InstanceIAMMember(name, undefined, { urn });
            case "gcp:compute/instanceIAMPolicy:InstanceIAMPolicy":
                return new exports.InstanceIAMPolicy(name, undefined, { urn });
            case "gcp:compute/instanceSettings:InstanceSettings":
                return new exports.InstanceSettings(name, undefined, { urn });
            case "gcp:compute/instanceTemplate:InstanceTemplate":
                return new exports.InstanceTemplate(name, undefined, { urn });
            case "gcp:compute/instanceTemplateIamBinding:InstanceTemplateIamBinding":
                return new exports.InstanceTemplateIamBinding(name, undefined, { urn });
            case "gcp:compute/instanceTemplateIamMember:InstanceTemplateIamMember":
                return new exports.InstanceTemplateIamMember(name, undefined, { urn });
            case "gcp:compute/instanceTemplateIamPolicy:InstanceTemplateIamPolicy":
                return new exports.InstanceTemplateIamPolicy(name, undefined, { urn });
            case "gcp:compute/instantSnapshot:InstantSnapshot":
                return new exports.InstantSnapshot(name, undefined, { urn });
            case "gcp:compute/instantSnapshotIamBinding:InstantSnapshotIamBinding":
                return new exports.InstantSnapshotIamBinding(name, undefined, { urn });
            case "gcp:compute/instantSnapshotIamMember:InstantSnapshotIamMember":
                return new exports.InstantSnapshotIamMember(name, undefined, { urn });
            case "gcp:compute/instantSnapshotIamPolicy:InstantSnapshotIamPolicy":
                return new exports.InstantSnapshotIamPolicy(name, undefined, { urn });
            case "gcp:compute/interconnect:Interconnect":
                return new exports.Interconnect(name, undefined, { urn });
            case "gcp:compute/interconnectAttachment:InterconnectAttachment":
                return new exports.InterconnectAttachment(name, undefined, { urn });
            case "gcp:compute/interconnectAttachmentGroup:InterconnectAttachmentGroup":
                return new exports.InterconnectAttachmentGroup(name, undefined, { urn });
            case "gcp:compute/interconnectGroup:InterconnectGroup":
                return new exports.InterconnectGroup(name, undefined, { urn });
            case "gcp:compute/machineImage:MachineImage":
                return new exports.MachineImage(name, undefined, { urn });
            case "gcp:compute/machineImageIamBinding:MachineImageIamBinding":
                return new exports.MachineImageIamBinding(name, undefined, { urn });
            case "gcp:compute/machineImageIamMember:MachineImageIamMember":
                return new exports.MachineImageIamMember(name, undefined, { urn });
            case "gcp:compute/machineImageIamPolicy:MachineImageIamPolicy":
                return new exports.MachineImageIamPolicy(name, undefined, { urn });
            case "gcp:compute/managedSslCertificate:ManagedSslCertificate":
                return new exports.ManagedSslCertificate(name, undefined, { urn });
            case "gcp:compute/mangedSslCertificate:MangedSslCertificate":
                return new exports.MangedSslCertificate(name, undefined, { urn });
            case "gcp:compute/network:Network":
                return new exports.Network(name, undefined, { urn });
            case "gcp:compute/networkAttachment:NetworkAttachment":
                return new exports.NetworkAttachment(name, undefined, { urn });
            case "gcp:compute/networkEdgeSecurityService:NetworkEdgeSecurityService":
                return new exports.NetworkEdgeSecurityService(name, undefined, { urn });
            case "gcp:compute/networkEndpoint:NetworkEndpoint":
                return new exports.NetworkEndpoint(name, undefined, { urn });
            case "gcp:compute/networkEndpointGroup:NetworkEndpointGroup":
                return new exports.NetworkEndpointGroup(name, undefined, { urn });
            case "gcp:compute/networkEndpointList:NetworkEndpointList":
                return new exports.NetworkEndpointList(name, undefined, { urn });
            case "gcp:compute/networkFirewallPolicy:NetworkFirewallPolicy":
                return new exports.NetworkFirewallPolicy(name, undefined, { urn });
            case "gcp:compute/networkFirewallPolicyAssociation:NetworkFirewallPolicyAssociation":
                return new exports.NetworkFirewallPolicyAssociation(name, undefined, { urn });
            case "gcp:compute/networkFirewallPolicyPacketMirroringRule:NetworkFirewallPolicyPacketMirroringRule":
                return new exports.NetworkFirewallPolicyPacketMirroringRule(name, undefined, { urn });
            case "gcp:compute/networkFirewallPolicyRule:NetworkFirewallPolicyRule":
                return new exports.NetworkFirewallPolicyRule(name, undefined, { urn });
            case "gcp:compute/networkFirewallPolicyWithRules:NetworkFirewallPolicyWithRules":
                return new exports.NetworkFirewallPolicyWithRules(name, undefined, { urn });
            case "gcp:compute/networkPeering:NetworkPeering":
                return new exports.NetworkPeering(name, undefined, { urn });
            case "gcp:compute/networkPeeringRoutesConfig:NetworkPeeringRoutesConfig":
                return new exports.NetworkPeeringRoutesConfig(name, undefined, { urn });
            case "gcp:compute/nodeGroup:NodeGroup":
                return new exports.NodeGroup(name, undefined, { urn });
            case "gcp:compute/nodeTemplate:NodeTemplate":
                return new exports.NodeTemplate(name, undefined, { urn });
            case "gcp:compute/organizationSecurityPolicy:OrganizationSecurityPolicy":
                return new exports.OrganizationSecurityPolicy(name, undefined, { urn });
            case "gcp:compute/organizationSecurityPolicyAssociation:OrganizationSecurityPolicyAssociation":
                return new exports.OrganizationSecurityPolicyAssociation(name, undefined, { urn });
            case "gcp:compute/organizationSecurityPolicyRule:OrganizationSecurityPolicyRule":
                return new exports.OrganizationSecurityPolicyRule(name, undefined, { urn });
            case "gcp:compute/packetMirroring:PacketMirroring":
                return new exports.PacketMirroring(name, undefined, { urn });
            case "gcp:compute/perInstanceConfig:PerInstanceConfig":
                return new exports.PerInstanceConfig(name, undefined, { urn });
            case "gcp:compute/previewFeature:PreviewFeature":
                return new exports.PreviewFeature(name, undefined, { urn });
            case "gcp:compute/projectCloudArmorTier:ProjectCloudArmorTier":
                return new exports.ProjectCloudArmorTier(name, undefined, { urn });
            case "gcp:compute/projectDefaultNetworkTier:ProjectDefaultNetworkTier":
                return new exports.ProjectDefaultNetworkTier(name, undefined, { urn });
            case "gcp:compute/projectMetadata:ProjectMetadata":
                return new exports.ProjectMetadata(name, undefined, { urn });
            case "gcp:compute/projectMetadataItem:ProjectMetadataItem":
                return new exports.ProjectMetadataItem(name, undefined, { urn });
            case "gcp:compute/publicAdvertisedPrefix:PublicAdvertisedPrefix":
                return new exports.PublicAdvertisedPrefix(name, undefined, { urn });
            case "gcp:compute/publicDelegatedPrefix:PublicDelegatedPrefix":
                return new exports.PublicDelegatedPrefix(name, undefined, { urn });
            case "gcp:compute/regionAutoscaler:RegionAutoscaler":
                return new exports.RegionAutoscaler(name, undefined, { urn });
            case "gcp:compute/regionBackendBucket:RegionBackendBucket":
                return new exports.RegionBackendBucket(name, undefined, { urn });
            case "gcp:compute/regionBackendBucketIamBinding:RegionBackendBucketIamBinding":
                return new exports.RegionBackendBucketIamBinding(name, undefined, { urn });
            case "gcp:compute/regionBackendBucketIamMember:RegionBackendBucketIamMember":
                return new exports.RegionBackendBucketIamMember(name, undefined, { urn });
            case "gcp:compute/regionBackendBucketIamPolicy:RegionBackendBucketIamPolicy":
                return new exports.RegionBackendBucketIamPolicy(name, undefined, { urn });
            case "gcp:compute/regionBackendService:RegionBackendService":
                return new exports.RegionBackendService(name, undefined, { urn });
            case "gcp:compute/regionBackendServiceIamBinding:RegionBackendServiceIamBinding":
                return new exports.RegionBackendServiceIamBinding(name, undefined, { urn });
            case "gcp:compute/regionBackendServiceIamMember:RegionBackendServiceIamMember":
                return new exports.RegionBackendServiceIamMember(name, undefined, { urn });
            case "gcp:compute/regionBackendServiceIamPolicy:RegionBackendServiceIamPolicy":
                return new exports.RegionBackendServiceIamPolicy(name, undefined, { urn });
            case "gcp:compute/regionCommitment:RegionCommitment":
                return new exports.RegionCommitment(name, undefined, { urn });
            case "gcp:compute/regionCompositeHealthCheck:RegionCompositeHealthCheck":
                return new exports.RegionCompositeHealthCheck(name, undefined, { urn });
            case "gcp:compute/regionDisk:RegionDisk":
                return new exports.RegionDisk(name, undefined, { urn });
            case "gcp:compute/regionDiskIamBinding:RegionDiskIamBinding":
                return new exports.RegionDiskIamBinding(name, undefined, { urn });
            case "gcp:compute/regionDiskIamMember:RegionDiskIamMember":
                return new exports.RegionDiskIamMember(name, undefined, { urn });
            case "gcp:compute/regionDiskIamPolicy:RegionDiskIamPolicy":
                return new exports.RegionDiskIamPolicy(name, undefined, { urn });
            case "gcp:compute/regionDiskResourcePolicyAttachment:RegionDiskResourcePolicyAttachment":
                return new exports.RegionDiskResourcePolicyAttachment(name, undefined, { urn });
            case "gcp:compute/regionHealthAggregationPolicy:RegionHealthAggregationPolicy":
                return new exports.RegionHealthAggregationPolicy(name, undefined, { urn });
            case "gcp:compute/regionHealthCheck:RegionHealthCheck":
                return new exports.RegionHealthCheck(name, undefined, { urn });
            case "gcp:compute/regionHealthSource:RegionHealthSource":
                return new exports.RegionHealthSource(name, undefined, { urn });
            case "gcp:compute/regionInstanceGroupManager:RegionInstanceGroupManager":
                return new exports.RegionInstanceGroupManager(name, undefined, { urn });
            case "gcp:compute/regionInstanceTemplate:RegionInstanceTemplate":
                return new exports.RegionInstanceTemplate(name, undefined, { urn });
            case "gcp:compute/regionInstantSnapshot:RegionInstantSnapshot":
                return new exports.RegionInstantSnapshot(name, undefined, { urn });
            case "gcp:compute/regionInstantSnapshotIamBinding:RegionInstantSnapshotIamBinding":
                return new exports.RegionInstantSnapshotIamBinding(name, undefined, { urn });
            case "gcp:compute/regionInstantSnapshotIamMember:RegionInstantSnapshotIamMember":
                return new exports.RegionInstantSnapshotIamMember(name, undefined, { urn });
            case "gcp:compute/regionInstantSnapshotIamPolicy:RegionInstantSnapshotIamPolicy":
                return new exports.RegionInstantSnapshotIamPolicy(name, undefined, { urn });
            case "gcp:compute/regionNetworkEndpoint:RegionNetworkEndpoint":
                return new exports.RegionNetworkEndpoint(name, undefined, { urn });
            case "gcp:compute/regionNetworkEndpointGroup:RegionNetworkEndpointGroup":
                return new exports.RegionNetworkEndpointGroup(name, undefined, { urn });
            case "gcp:compute/regionNetworkFirewallPolicy:RegionNetworkFirewallPolicy":
                return new exports.RegionNetworkFirewallPolicy(name, undefined, { urn });
            case "gcp:compute/regionNetworkFirewallPolicyAssociation:RegionNetworkFirewallPolicyAssociation":
                return new exports.RegionNetworkFirewallPolicyAssociation(name, undefined, { urn });
            case "gcp:compute/regionNetworkFirewallPolicyRule:RegionNetworkFirewallPolicyRule":
                return new exports.RegionNetworkFirewallPolicyRule(name, undefined, { urn });
            case "gcp:compute/regionNetworkFirewallPolicyWithRules:RegionNetworkFirewallPolicyWithRules":
                return new exports.RegionNetworkFirewallPolicyWithRules(name, undefined, { urn });
            case "gcp:compute/regionPerInstanceConfig:RegionPerInstanceConfig":
                return new exports.RegionPerInstanceConfig(name, undefined, { urn });
            case "gcp:compute/regionResizeRequest:RegionResizeRequest":
                return new exports.RegionResizeRequest(name, undefined, { urn });
            case "gcp:compute/regionSecurityPolicy:RegionSecurityPolicy":
                return new exports.RegionSecurityPolicy(name, undefined, { urn });
            case "gcp:compute/regionSecurityPolicyRule:RegionSecurityPolicyRule":
                return new exports.RegionSecurityPolicyRule(name, undefined, { urn });
            case "gcp:compute/regionSslCertificate:RegionSslCertificate":
                return new exports.RegionSslCertificate(name, undefined, { urn });
            case "gcp:compute/regionSslPolicy:RegionSslPolicy":
                return new exports.RegionSslPolicy(name, undefined, { urn });
            case "gcp:compute/regionTargetHttpProxy:RegionTargetHttpProxy":
                return new exports.RegionTargetHttpProxy(name, undefined, { urn });
            case "gcp:compute/regionTargetHttpsProxy:RegionTargetHttpsProxy":
                return new exports.RegionTargetHttpsProxy(name, undefined, { urn });
            case "gcp:compute/regionTargetTcpProxy:RegionTargetTcpProxy":
                return new exports.RegionTargetTcpProxy(name, undefined, { urn });
            case "gcp:compute/regionUrlMap:RegionUrlMap":
                return new exports.RegionUrlMap(name, undefined, { urn });
            case "gcp:compute/reservation:Reservation":
                return new exports.Reservation(name, undefined, { urn });
            case "gcp:compute/resizeRequest:ResizeRequest":
                return new exports.ResizeRequest(name, undefined, { urn });
            case "gcp:compute/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "gcp:compute/resourcePolicyAttachment:ResourcePolicyAttachment":
                return new exports.ResourcePolicyAttachment(name, undefined, { urn });
            case "gcp:compute/rolloutPlan:RolloutPlan":
                return new exports.RolloutPlan(name, undefined, { urn });
            case "gcp:compute/route:Route":
                return new exports.Route(name, undefined, { urn });
            case "gcp:compute/router:Router":
                return new exports.Router(name, undefined, { urn });
            case "gcp:compute/routerInterface:RouterInterface":
                return new exports.RouterInterface(name, undefined, { urn });
            case "gcp:compute/routerNamedSet:RouterNamedSet":
                return new exports.RouterNamedSet(name, undefined, { urn });
            case "gcp:compute/routerNat:RouterNat":
                return new exports.RouterNat(name, undefined, { urn });
            case "gcp:compute/routerNatAddress:RouterNatAddress":
                return new exports.RouterNatAddress(name, undefined, { urn });
            case "gcp:compute/routerPeer:RouterPeer":
                return new exports.RouterPeer(name, undefined, { urn });
            case "gcp:compute/routerRoutePolicy:RouterRoutePolicy":
                return new exports.RouterRoutePolicy(name, undefined, { urn });
            case "gcp:compute/sSLCertificate:SSLCertificate":
                return new exports.SSLCertificate(name, undefined, { urn });
            case "gcp:compute/sSLPolicy:SSLPolicy":
                return new exports.SSLPolicy(name, undefined, { urn });
            case "gcp:compute/securityPolicy:SecurityPolicy":
                return new exports.SecurityPolicy(name, undefined, { urn });
            case "gcp:compute/securityPolicyRule:SecurityPolicyRule":
                return new exports.SecurityPolicyRule(name, undefined, { urn });
            case "gcp:compute/securityScanConfig:SecurityScanConfig":
                return new exports.SecurityScanConfig(name, undefined, { urn });
            case "gcp:compute/serviceAttachment:ServiceAttachment":
                return new exports.ServiceAttachment(name, undefined, { urn });
            case "gcp:compute/sharedVPCHostProject:SharedVPCHostProject":
                return new exports.SharedVPCHostProject(name, undefined, { urn });
            case "gcp:compute/sharedVPCServiceProject:SharedVPCServiceProject":
                return new exports.SharedVPCServiceProject(name, undefined, { urn });
            case "gcp:compute/snapshot:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "gcp:compute/snapshotIamBinding:SnapshotIamBinding":
                return new exports.SnapshotIamBinding(name, undefined, { urn });
            case "gcp:compute/snapshotIamMember:SnapshotIamMember":
                return new exports.SnapshotIamMember(name, undefined, { urn });
            case "gcp:compute/snapshotIamPolicy:SnapshotIamPolicy":
                return new exports.SnapshotIamPolicy(name, undefined, { urn });
            case "gcp:compute/snapshotSettings:SnapshotSettings":
                return new exports.SnapshotSettings(name, undefined, { urn });
            case "gcp:compute/storagePool:StoragePool":
                return new exports.StoragePool(name, undefined, { urn });
            case "gcp:compute/storagePoolIamBinding:StoragePoolIamBinding":
                return new exports.StoragePoolIamBinding(name, undefined, { urn });
            case "gcp:compute/storagePoolIamMember:StoragePoolIamMember":
                return new exports.StoragePoolIamMember(name, undefined, { urn });
            case "gcp:compute/storagePoolIamPolicy:StoragePoolIamPolicy":
                return new exports.StoragePoolIamPolicy(name, undefined, { urn });
            case "gcp:compute/subnetwork:Subnetwork":
                return new exports.Subnetwork(name, undefined, { urn });
            case "gcp:compute/subnetworkIAMBinding:SubnetworkIAMBinding":
                return new exports.SubnetworkIAMBinding(name, undefined, { urn });
            case "gcp:compute/subnetworkIAMMember:SubnetworkIAMMember":
                return new exports.SubnetworkIAMMember(name, undefined, { urn });
            case "gcp:compute/subnetworkIAMPolicy:SubnetworkIAMPolicy":
                return new exports.SubnetworkIAMPolicy(name, undefined, { urn });
            case "gcp:compute/targetGrpcProxy:TargetGrpcProxy":
                return new exports.TargetGrpcProxy(name, undefined, { urn });
            case "gcp:compute/targetHttpProxy:TargetHttpProxy":
                return new exports.TargetHttpProxy(name, undefined, { urn });
            case "gcp:compute/targetHttpsProxy:TargetHttpsProxy":
                return new exports.TargetHttpsProxy(name, undefined, { urn });
            case "gcp:compute/targetInstance:TargetInstance":
                return new exports.TargetInstance(name, undefined, { urn });
            case "gcp:compute/targetPool:TargetPool":
                return new exports.TargetPool(name, undefined, { urn });
            case "gcp:compute/targetSSLProxy:TargetSSLProxy":
                return new exports.TargetSSLProxy(name, undefined, { urn });
            case "gcp:compute/targetTCPProxy:TargetTCPProxy":
                return new exports.TargetTCPProxy(name, undefined, { urn });
            case "gcp:compute/uRLMap:URLMap":
                return new exports.URLMap(name, undefined, { urn });
            case "gcp:compute/vPNGateway:VPNGateway":
                return new exports.VPNGateway(name, undefined, { urn });
            case "gcp:compute/vPNTunnel:VPNTunnel":
                return new exports.VPNTunnel(name, undefined, { urn });
            case "gcp:compute/wireGroup:WireGroup":
                return new exports.WireGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "compute/address", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/attachedDisk", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/autoscaler", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendBucket", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendBucketIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendBucketIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendBucketIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendBucketSignedUrlKey", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendService", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/backendServiceSignedUrlKey", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/caExternalAccountKey", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/crossSiteNetwork", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/disk", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/diskAsyncReplication", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/diskIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/diskIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/diskIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/diskResourcePolicyAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/externalVpnGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/firewall", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/firewallPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/firewallPolicyAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/firewallPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/firewallPolicyWithRules", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/forwardingRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/futureReservation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/globalAddress", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/globalForwardingRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/globalNetworkEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/globalNetworkEndpointGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/haVpnGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/healthCheck", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/httpHealthCheck", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/httpsHealthCheck", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/image", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/imageIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/imageIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/imageIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instance", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceFromMachineImage", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceFromTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceGroupManager", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceGroupMembership", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceGroupNamedPort", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceTemplateIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceTemplateIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instanceTemplateIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instantSnapshot", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instantSnapshotIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instantSnapshotIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/instantSnapshotIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/interconnect", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/interconnectAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/interconnectAttachmentGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/interconnectGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/machineImage", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/machineImageIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/machineImageIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/machineImageIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/managedSslCertificate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/mangedSslCertificate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/network", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkEdgeSecurityService", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkEndpointGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkEndpointList", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkFirewallPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkFirewallPolicyAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkFirewallPolicyPacketMirroringRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkFirewallPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkFirewallPolicyWithRules", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkPeering", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/networkPeeringRoutesConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/nodeGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/nodeTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/organizationSecurityPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/organizationSecurityPolicyAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/organizationSecurityPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/packetMirroring", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/perInstanceConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/previewFeature", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/projectCloudArmorTier", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/projectDefaultNetworkTier", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/projectMetadata", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/projectMetadataItem", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/publicAdvertisedPrefix", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/publicDelegatedPrefix", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionAutoscaler", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendBucket", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendBucketIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendBucketIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendBucketIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendService", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendServiceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendServiceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionBackendServiceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionCommitment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionCompositeHealthCheck", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionDisk", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionDiskIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionDiskIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionDiskIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionDiskResourcePolicyAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionHealthAggregationPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionHealthCheck", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionHealthSource", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstanceGroupManager", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstanceTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstantSnapshot", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstantSnapshotIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstantSnapshotIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionInstantSnapshotIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkEndpointGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkFirewallPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkFirewallPolicyAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkFirewallPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionNetworkFirewallPolicyWithRules", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionPerInstanceConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionResizeRequest", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionSecurityPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionSecurityPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionSslCertificate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionSslPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionTargetHttpProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionTargetHttpsProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionTargetTcpProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/regionUrlMap", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/reservation", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/resizeRequest", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/resourcePolicyAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/rolloutPlan", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/route", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/router", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerInterface", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerNamedSet", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerNat", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerNatAddress", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerPeer", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/routerRoutePolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/sSLCertificate", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/sSLPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/securityPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/securityPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/securityScanConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/serviceAttachment", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/sharedVPCHostProject", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/sharedVPCServiceProject", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/snapshot", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/snapshotIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/snapshotIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/snapshotIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/snapshotSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/storagePool", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/storagePoolIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/storagePoolIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/storagePoolIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/subnetwork", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/subnetworkIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/subnetworkIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/subnetworkIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetGrpcProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetHttpProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetHttpsProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetPool", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetSSLProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/targetTCPProxy", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/uRLMap", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/vPNGateway", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/vPNTunnel", _module);
pulumi.runtime.registerResourceModule("gcp", "compute/wireGroup", _module);
