// @pulumi/gcp/compute/previewFeature.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreviewFeature = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a single Google Compute Engine preview feature such as Alpha API access, which can be enabled or unspecified for a project.
 *
 * To get more information about PreviewFeature, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/rest/v1/previewFeatures)
 * * How-to Guides
 *     * [Use the Compute Engine alpha API](https://cloud.google.com/compute/docs/reference/rest/alpha)
 *
 * ## Example Usage
 *
 * ### Preview Feature Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const gcePreviewFeature = new gcp.compute.PreviewFeature("gce_preview_feature", {
 *     name: "alpha-api-access",
 *     activationStatus: "ACTIVATION_STATE_UNSPECIFIED",
 *     rolloutOperation: {
 *         rolloutInput: {
 *             predefinedRolloutPlan: "ROLLOUT_PLAN_FAST_ROLLOUT",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * PreviewFeature can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/global/previewFeatures/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, PreviewFeature can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/previewFeature:PreviewFeature default projects/{{project}}/global/previewFeatures/{{name}}
 * $ pulumi import gcp:compute/previewFeature:PreviewFeature default {{project}}/{{name}}
 * $ pulumi import gcp:compute/previewFeature:PreviewFeature default {{name}}
 * ```
 */
class PreviewFeature extends pulumi.CustomResource {
    /**
     * Get an existing PreviewFeature resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PreviewFeature(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:compute/previewFeature:PreviewFeature';
    /**
     * Returns true if the given object is an instance of PreviewFeature.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PreviewFeature.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["activationStatus"] = state?.activationStatus;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["rolloutOperation"] = state?.rolloutOperation;
        }
        else {
            const args = argsOrState;
            if (args?.activationStatus === undefined && !opts.urn) {
                throw new Error("Missing required property 'activationStatus'");
            }
            resourceInputs["activationStatus"] = args?.activationStatus;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["rolloutOperation"] = args?.rolloutOperation;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PreviewFeature.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PreviewFeature = PreviewFeature;
