// @pulumi/gcp/compute/regionBackendBucket.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionBackendBucket = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Regional backend buckets allow you to use Google Cloud Storage buckets with
 * regional HTTP(S) load balancing.
 *
 * A regional HTTP(S) load balancer can direct traffic to specified URLs to a
 * backend bucket rather than a backend service. It can send requests for
 * static content to a Cloud Storage bucket and requests for dynamic content
 * to a virtual machine instance.
 *
 * Regional backend buckets are used with:
 * - Regional internal Application Load Balancers
 * - Regional external Application Load Balancers
 *
 * > **Note:** Regional backend buckets have important limitations:
 * - Cloud CDN cannot be enabled
 * - Only public buckets are supported (private bucket access is not available)
 * - Only GET requests are supported
 * - The bucket must be in the same region as the load balancer
 * - Single-region buckets only (multi-region and dual-region buckets are not supported)
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about RegionBackendBucket, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/beta/regionBackendBuckets)
 * * How-to Guides
 *     * [Using a Cloud Storage bucket as a load balancer backend](https://cloud.google.com/compute/docs/load-balancing/http/backend-bucket)
 *
 * ## Example Usage
 *
 * ### Region Backend Bucket Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const imageBackendBucket = new gcp.storage.Bucket("image_backend", {
 *     name: "region-image-store-bucket",
 *     location: "US-CENTRAL1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 * });
 * const imageBackend = new gcp.compute.RegionBackendBucket("image_backend", {
 *     name: "region-image-backend-bucket",
 *     region: "us-central1",
 *     bucketName: imageBackendBucket.name,
 *     description: "Regional backend bucket example",
 *     loadBalancingScheme: "INTERNAL_MANAGED",
 * });
 * ```
 * ### Region Backend Bucket Internal Lb
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const internalBackendBucket = new gcp.storage.Bucket("internal_backend", {
 *     name: "regional-internal-bucket",
 *     location: "US-CENTRAL1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 *     website: {
 *         mainPageSuffix: "index.html",
 *         notFoundPage: "404.html",
 *     },
 * });
 * const internalBackend = new gcp.compute.RegionBackendBucket("internal_backend", {
 *     name: "regional-internal-backend",
 *     region: "us-central1",
 *     bucketName: internalBackendBucket.name,
 *     loadBalancingScheme: "INTERNAL_MANAGED",
 *     description: "Regional internal backend bucket for static content",
 * });
 * const index = new gcp.storage.BucketObject("index", {
 *     name: "index.html",
 *     bucket: internalBackendBucket.name,
 *     content: "<html><body><h1>Regional Internal LB Backend Bucket</h1></body></html>",
 * });
 * ```
 * ### Region Backend Bucket External Lb
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const externalBackendBucket = new gcp.storage.Bucket("external_backend", {
 *     name: "regional-external-bucket",
 *     location: "US-EAST1",
 *     forceDestroy: true,
 *     uniformBucketLevelAccess: true,
 *     website: {
 *         mainPageSuffix: "index.html",
 *         notFoundPage: "404.html",
 *     },
 * });
 * const externalBackend = new gcp.compute.RegionBackendBucket("external_backend", {
 *     name: "regional-external-backend",
 *     region: "us-east1",
 *     bucketName: externalBackendBucket.name,
 *     loadBalancingScheme: "EXTERNAL_MANAGED",
 *     description: "Regional external backend bucket for static content",
 * });
 * const index = new gcp.storage.BucketObject("index", {
 *     name: "index.html",
 *     bucket: externalBackendBucket.name,
 *     content: "<html><body><h1>Regional External LB Backend Bucket</h1></body></html>",
 * });
 * const staticAsset = new gcp.storage.BucketObject("static_asset", {
 *     name: "assets/style.css",
 *     bucket: externalBackendBucket.name,
 *     content: "body { font-family: Arial, sans-serif; }",
 * });
 * ```
 *
 * ## Import
 *
 * RegionBackendBucket can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/regions/{{region}}/backendBuckets/{{name}}`
 * * `{{project}}/{{region}}/{{name}}`
 * * `{{region}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, RegionBackendBucket can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/regionBackendBucket:RegionBackendBucket default projects/{{project}}/regions/{{region}}/backendBuckets/{{name}}
 * $ pulumi import gcp:compute/regionBackendBucket:RegionBackendBucket default {{project}}/{{region}}/{{name}}
 * $ pulumi import gcp:compute/regionBackendBucket:RegionBackendBucket default {{region}}/{{name}}
 * $ pulumi import gcp:compute/regionBackendBucket:RegionBackendBucket default {{name}}
 * ```
 */
class RegionBackendBucket extends pulumi.CustomResource {
    /**
     * Get an existing RegionBackendBucket resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegionBackendBucket(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:compute/regionBackendBucket:RegionBackendBucket';
    /**
     * Returns true if the given object is an instance of RegionBackendBucket.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegionBackendBucket.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bucketName"] = state?.bucketName;
            resourceInputs["creationTimestamp"] = state?.creationTimestamp;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["loadBalancingScheme"] = state?.loadBalancingScheme;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["region"] = state?.region;
            resourceInputs["selfLink"] = state?.selfLink;
        }
        else {
            const args = argsOrState;
            if (args?.bucketName === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucketName'");
            }
            if (args?.region === undefined && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            resourceInputs["bucketName"] = args?.bucketName;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["loadBalancingScheme"] = args?.loadBalancingScheme;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["region"] = args?.region;
            resourceInputs["creationTimestamp"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegionBackendBucket.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegionBackendBucket = RegionBackendBucket;
