// @pulumi/gcp/compute/rolloutPlan.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RolloutPlan = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A RolloutPlan is the customer-defined strategy to divide a large-scale change
 * into smaller increments, referred to as "waves". Each wave targets a specific
 * portion of the overall affected area and defines criteria that must be met
 * before progressing to the subsequent wave.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about RolloutPlan, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/rest/beta/rolloutPlans)
 *
 * ## Example Usage
 *
 * ### Rollout Plan Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.RolloutPlan("default", {
 *     name: "tf-test-rollout-plan-_75125",
 *     description: "A test rollout plan",
 *     locationScope: "ZONAL",
 *     waves: [{
 *         displayName: "wave-1",
 *         selectors: [{
 *             locationSelector: {
 *                 includedLocations: ["us-central1-a"],
 *             },
 *         }],
 *         validation: {
 *             type: "manual",
 *         },
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * RolloutPlan can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/global/rolloutPlans/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, RolloutPlan can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/rolloutPlan:RolloutPlan default projects/{{project}}/global/rolloutPlans/{{name}}
 * $ pulumi import gcp:compute/rolloutPlan:RolloutPlan default {{project}}/{{name}}
 * $ pulumi import gcp:compute/rolloutPlan:RolloutPlan default {{name}}
 * ```
 */
class RolloutPlan extends pulumi.CustomResource {
    /**
     * Get an existing RolloutPlan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RolloutPlan(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:compute/rolloutPlan:RolloutPlan';
    /**
     * Returns true if the given object is an instance of RolloutPlan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RolloutPlan.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["locationScope"] = state?.locationScope;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["selfLink"] = state?.selfLink;
            resourceInputs["waves"] = state?.waves;
        }
        else {
            const args = argsOrState;
            if (args?.waves === undefined && !opts.urn) {
                throw new Error("Missing required property 'waves'");
            }
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["locationScope"] = args?.locationScope;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["waves"] = args?.waves;
            resourceInputs["selfLink"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RolloutPlan.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RolloutPlan = RolloutPlan;
