// @pulumi/gcp/compute/securityPolicyRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPolicyRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A rule for the SecurityPolicy.
 *
 * To get more information about SecurityPolicyRule, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies/addRule)
 * * How-to Guides
 *     * [Creating global security policy rules](https://cloud.google.com/armor/docs/configure-security-policies)
 *
 * ## Example Usage
 *
 * ### Security Policy Rule Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.SecurityPolicy("default", {
 *     name: "policyruletest",
 *     description: "basic global security policy",
 *     type: "CLOUD_ARMOR",
 * });
 * const policyRule = new gcp.compute.SecurityPolicyRule("policy_rule", {
 *     securityPolicy: _default.name,
 *     description: "new rule",
 *     priority: 100,
 *     match: {
 *         versionedExpr: "SRC_IPS_V1",
 *         config: {
 *             srcIpRanges: ["10.10.0.0/16"],
 *         },
 *     },
 *     action: "allow",
 *     preview: true,
 * });
 * ```
 * ### Security Policy Rule Default Rule
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.SecurityPolicy("default", {
 *     name: "policyruletest",
 *     description: "basic global security policy",
 *     type: "CLOUD_ARMOR",
 * });
 * const defaultRule = new gcp.compute.SecurityPolicyRule("default_rule", {
 *     securityPolicy: _default.name,
 *     description: "default rule",
 *     action: "deny",
 *     priority: 2147483647,
 *     match: {
 *         versionedExpr: "SRC_IPS_V1",
 *         config: {
 *             srcIpRanges: ["*"],
 *         },
 *     },
 * });
 * const policyRule = new gcp.compute.SecurityPolicyRule("policy_rule", {
 *     securityPolicy: _default.name,
 *     description: "new rule",
 *     priority: 100,
 *     match: {
 *         versionedExpr: "SRC_IPS_V1",
 *         config: {
 *             srcIpRanges: ["10.10.0.0/16"],
 *         },
 *     },
 *     action: "allow",
 *     preview: true,
 * });
 * ```
 * ### Security Policy Rule Multiple Rules
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.SecurityPolicy("default", {
 *     name: "policywithmultiplerules",
 *     description: "basic global security policy",
 *     type: "CLOUD_ARMOR",
 * });
 * const policyRuleOne = new gcp.compute.SecurityPolicyRule("policy_rule_one", {
 *     securityPolicy: _default.name,
 *     description: "new rule one",
 *     priority: 100,
 *     match: {
 *         versionedExpr: "SRC_IPS_V1",
 *         config: {
 *             srcIpRanges: ["10.10.0.0/16"],
 *         },
 *     },
 *     action: "allow",
 *     preview: true,
 * });
 * const policyRuleTwo = new gcp.compute.SecurityPolicyRule("policy_rule_two", {
 *     securityPolicy: _default.name,
 *     description: "new rule two",
 *     priority: 101,
 *     match: {
 *         versionedExpr: "SRC_IPS_V1",
 *         config: {
 *             srcIpRanges: [
 *                 "192.168.0.0/16",
 *                 "10.0.0.0/8",
 *             ],
 *         },
 *     },
 *     action: "allow",
 *     preview: true,
 * });
 * ```
 * ### Security Policy Rule Advanced Features
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const policy = new gcp.compute.SecurityPolicy("policy", {
 *     name: "policyruletest",
 *     description: "Security policy with WAF exclusions, Headers, and Redirect",
 * });
 * const policySecurityPolicyRule = new gcp.compute.SecurityPolicyRule("policy", {
 *     securityPolicy: policy.name,
 *     description: "Complex rule using advanced features: WAF config, header actions, and redirect options",
 *     priority: 100,
 *     action: "allow",
 *     match: {
 *         expr: {
 *             expression: "request.path.matches('/api/v1/.*')",
 *         },
 *     },
 *     preconfiguredWafConfig: {
 *         exclusions: [{
 *             targetRuleSet: "sqli-v33-stable",
 *             targetRuleIds: ["owasp-crs-v030301-id942100-sqli"],
 *             requestHeaders: [{
 *                 operator: "EQUALS",
 *                 value: "internal-scan",
 *             }],
 *         }],
 *     },
 *     headerAction: {
 *         requestHeadersToAdds: [{
 *             headerName: "X-Added-By-Armor",
 *             headerValue: "Verified-Traffic",
 *         }],
 *     },
 * });
 * ```
 * ### Security Policy Rule With Body Exclude
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.Network("default", {
 *     name: "test-network",
 *     autoCreateSubnetworks: false,
 * });
 * const defaultSubnetwork = new gcp.compute.Subnetwork("default", {
 *     name: "test-subnet",
 *     region: "us-west2",
 *     network: _default.id,
 *     ipCidrRange: "10.10.0.0/24",
 * });
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "test-health-check",
 *     httpHealthCheck: {
 *         port: 80,
 *     },
 * });
 * const defaultSecurityPolicy = new gcp.compute.SecurityPolicy("default", {
 *     name: "policyruletest",
 *     description: "global security policy with body inspection",
 *     type: "CLOUD_ARMOR",
 *     advancedOptionsConfig: {
 *         jsonParsing: "STANDARD",
 *         logLevel: "VERBOSE",
 *     },
 * });
 * const defaultInstanceTemplate = new gcp.compute.InstanceTemplate("default", {
 *     networkInterfaces: [{
 *         accessConfigs: [{}],
 *         subnetwork: defaultSubnetwork.id,
 *     }],
 *     name: "backendpolicy",
 *     machineType: "e2-micro",
 *     disks: [{
 *         sourceImage: "projects/debian-cloud/global/images/family/debian-11",
 *         autoDelete: true,
 *         boot: true,
 *     }],
 * });
 * const defaultInstanceGroupManager = new gcp.compute.InstanceGroupManager("default", {
 *     name: "backendpolicy",
 *     baseInstanceName: "backend",
 *     zone: "us-west2-a",
 *     versions: [{
 *         instanceTemplate: defaultInstanceTemplate.id,
 *     }],
 *     targetSize: 1,
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "backendpolicy",
 *     protocol: "HTTP",
 *     loadBalancingScheme: "EXTERNAL_MANAGED",
 *     timeoutSec: 30,
 *     healthChecks: defaultHealthCheck.id,
 *     backends: [{
 *         group: defaultInstanceGroupManager.instanceGroup,
 *     }],
 *     securityPolicy: defaultSecurityPolicy.id,
 * });
 * const policyRuleOne = new gcp.compute.SecurityPolicyRule("policy_rule_one", {
 *     securityPolicy: defaultSecurityPolicy.name,
 *     description: "waf body rule",
 *     action: "deny(403)",
 *     priority: 100,
 *     preview: true,
 *     match: {
 *         expr: {
 *             expression: "evaluatePreconfiguredWaf('sqli-v33-stable')",
 *         },
 *     },
 *     preconfiguredWafConfig: {
 *         exclusions: [{
 *             targetRuleSet: "sqli-v33-stable",
 *             requestBodies: [{
 *                 operator: "EQUALS",
 *                 value: "safe-field",
 *             }],
 *         }],
 *     },
 * }, {
 *     dependsOn: [defaultBackendService],
 * });
 * ```
 *
 * ## Import
 *
 * SecurityPolicyRule can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/global/securityPolicies/{{security_policy}}/priority/{{priority}}`
 * * `{{project}}/{{security_policy}}/{{priority}}`
 * * `{{security_policy}}/{{priority}}`
 *
 * When using the `pulumi import` command, SecurityPolicyRule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/securityPolicyRule:SecurityPolicyRule default projects/{{project}}/global/securityPolicies/{{security_policy}}/priority/{{priority}}
 * $ pulumi import gcp:compute/securityPolicyRule:SecurityPolicyRule default {{project}}/{{security_policy}}/{{priority}}
 * $ pulumi import gcp:compute/securityPolicyRule:SecurityPolicyRule default {{security_policy}}/{{priority}}
 * ```
 */
class SecurityPolicyRule extends pulumi.CustomResource {
    /**
     * Get an existing SecurityPolicyRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SecurityPolicyRule(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:compute/securityPolicyRule:SecurityPolicyRule';
    /**
     * Returns true if the given object is an instance of SecurityPolicyRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityPolicyRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["action"] = state?.action;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["headerAction"] = state?.headerAction;
            resourceInputs["match"] = state?.match;
            resourceInputs["preconfiguredWafConfig"] = state?.preconfiguredWafConfig;
            resourceInputs["preview"] = state?.preview;
            resourceInputs["priority"] = state?.priority;
            resourceInputs["project"] = state?.project;
            resourceInputs["rateLimitOptions"] = state?.rateLimitOptions;
            resourceInputs["redirectOptions"] = state?.redirectOptions;
            resourceInputs["securityPolicy"] = state?.securityPolicy;
        }
        else {
            const args = argsOrState;
            if (args?.action === undefined && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if (args?.priority === undefined && !opts.urn) {
                throw new Error("Missing required property 'priority'");
            }
            if (args?.securityPolicy === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityPolicy'");
            }
            resourceInputs["action"] = args?.action;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["headerAction"] = args?.headerAction;
            resourceInputs["match"] = args?.match;
            resourceInputs["preconfiguredWafConfig"] = args?.preconfiguredWafConfig;
            resourceInputs["preview"] = args?.preview;
            resourceInputs["priority"] = args?.priority;
            resourceInputs["project"] = args?.project;
            resourceInputs["rateLimitOptions"] = args?.rateLimitOptions;
            resourceInputs["redirectOptions"] = args?.redirectOptions;
            resourceInputs["securityPolicy"] = args?.securityPolicy;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecurityPolicyRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecurityPolicyRule = SecurityPolicyRule;
