// @pulumi/gcp/compute/targetTCPProxy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetTCPProxy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Represents a TargetTcpProxy resource, which is used by one or more
 * global forwarding rule to route incoming TCP requests to a Backend
 * service.
 *
 * To get more information about TargetTcpProxy, see:
 *
 * * [API documentation](https://cloud.google.com/compute/docs/reference/v1/targetTcpProxies)
 * * How-to Guides
 *     * [Setting Up TCP proxy for Google Cloud Load Balancing](https://cloud.google.com/compute/docs/load-balancing/tcp-ssl/tcp-proxy)
 *
 * ## Example Usage
 *
 * ### Target Tcp Proxy Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "health-check",
 *     timeoutSec: 1,
 *     checkIntervalSec: 1,
 *     tcpHealthCheck: {
 *         port: 443,
 *     },
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "backend-service",
 *     protocol: "TCP",
 *     timeoutSec: 10,
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const _default = new gcp.compute.TargetTCPProxy("default", {
 *     name: "test-proxy",
 *     backendService: defaultBackendService.id,
 * });
 * ```
 * ### Target Tcp Proxy Basic Beta
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "health-check",
 *     timeoutSec: 1,
 *     checkIntervalSec: 1,
 *     tcpHealthCheck: {
 *         port: 443,
 *     },
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "backend-service",
 *     loadBalancingScheme: "EXTERNAL_MANAGED",
 *     protocol: "TCP",
 *     timeoutSec: 10,
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const _default = new gcp.compute.TargetTCPProxy("default", {
 *     name: "test-proxy",
 *     loadBalancingScheme: "EXTERNAL_MANAGED",
 *     backendService: defaultBackendService.id,
 * });
 * ```
 * ### Target Tcp Proxy Backendless
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.TargetTCPProxy("default", {
 *     name: "test-proxy",
 *     loadBalancingScheme: "INTERNAL_MANAGED",
 * });
 * ```
 * ### Target Tcp Proxy Tls Route
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.compute.TargetTCPProxy("default", {
 *     name: "test-proxy",
 *     loadBalancingScheme: "INTERNAL_MANAGED",
 * });
 * const defaultHealthCheck = new gcp.compute.HealthCheck("default", {
 *     name: "health-check",
 *     httpsHealthCheck: {
 *         port: 443,
 *     },
 * });
 * const defaultBackendService = new gcp.compute.BackendService("default", {
 *     name: "backend-service",
 *     loadBalancingScheme: "INTERNAL_MANAGED",
 *     protocol: "TCP",
 *     healthChecks: defaultHealthCheck.id,
 * });
 * const defaultTlsRoute = new gcp.networkservices.TlsRoute("default", {
 *     name: "tls-route-check",
 *     targetProxies: [_default.id],
 *     rules: [{
 *         matches: [{
 *             sniHosts: ["example.com"],
 *         }],
 *         action: {
 *             destinations: [{
 *                 serviceName: defaultBackendService.id,
 *             }],
 *         },
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * TargetTcpProxy can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/global/targetTcpProxies/{{name}}`
 * * `{{project}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, TargetTcpProxy can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:compute/targetTCPProxy:TargetTCPProxy default projects/{{project}}/global/targetTcpProxies/{{name}}
 * $ pulumi import gcp:compute/targetTCPProxy:TargetTCPProxy default {{project}}/{{name}}
 * $ pulumi import gcp:compute/targetTCPProxy:TargetTCPProxy default {{name}}
 * ```
 */
class TargetTCPProxy extends pulumi.CustomResource {
    /**
     * Get an existing TargetTCPProxy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TargetTCPProxy(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:compute/targetTCPProxy:TargetTCPProxy';
    /**
     * Returns true if the given object is an instance of TargetTCPProxy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TargetTCPProxy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["backendService"] = state?.backendService;
            resourceInputs["creationTimestamp"] = state?.creationTimestamp;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["loadBalancingScheme"] = state?.loadBalancingScheme;
            resourceInputs["name"] = state?.name;
            resourceInputs["project"] = state?.project;
            resourceInputs["proxyBind"] = state?.proxyBind;
            resourceInputs["proxyHeader"] = state?.proxyHeader;
            resourceInputs["proxyId"] = state?.proxyId;
            resourceInputs["selfLink"] = state?.selfLink;
        }
        else {
            const args = argsOrState;
            resourceInputs["backendService"] = args?.backendService;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["loadBalancingScheme"] = args?.loadBalancingScheme;
            resourceInputs["name"] = args?.name;
            resourceInputs["project"] = args?.project;
            resourceInputs["proxyBind"] = args?.proxyBind;
            resourceInputs["proxyHeader"] = args?.proxyHeader;
            resourceInputs["creationTimestamp"] = undefined /*out*/;
            resourceInputs["proxyId"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TargetTCPProxy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TargetTCPProxy = TargetTCPProxy;
