// @pulumi/kubernetes/resource/v1alpha1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceClassPatch = exports.ResourceClassList = exports.ResourceClass = exports.ResourceClaimTemplatePatch = exports.ResourceClaimTemplateList = exports.ResourceClaimTemplate = exports.ResourceClaimPatch = exports.ResourceClaimList = exports.ResourceClaim = exports.PodSchedulingPatch = exports.PodSchedulingList = exports.PodScheduling = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.PodScheduling = null;
utilities.lazyLoad(exports, ["PodScheduling"], () => require("./podScheduling"));
exports.PodSchedulingList = null;
utilities.lazyLoad(exports, ["PodSchedulingList"], () => require("./podSchedulingList"));
exports.PodSchedulingPatch = null;
utilities.lazyLoad(exports, ["PodSchedulingPatch"], () => require("./podSchedulingPatch"));
exports.ResourceClaim = null;
utilities.lazyLoad(exports, ["ResourceClaim"], () => require("./resourceClaim"));
exports.ResourceClaimList = null;
utilities.lazyLoad(exports, ["ResourceClaimList"], () => require("./resourceClaimList"));
exports.ResourceClaimPatch = null;
utilities.lazyLoad(exports, ["ResourceClaimPatch"], () => require("./resourceClaimPatch"));
exports.ResourceClaimTemplate = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplate"], () => require("./resourceClaimTemplate"));
exports.ResourceClaimTemplateList = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplateList"], () => require("./resourceClaimTemplateList"));
exports.ResourceClaimTemplatePatch = null;
utilities.lazyLoad(exports, ["ResourceClaimTemplatePatch"], () => require("./resourceClaimTemplatePatch"));
exports.ResourceClass = null;
utilities.lazyLoad(exports, ["ResourceClass"], () => require("./resourceClass"));
exports.ResourceClassList = null;
utilities.lazyLoad(exports, ["ResourceClassList"], () => require("./resourceClassList"));
exports.ResourceClassPatch = null;
utilities.lazyLoad(exports, ["ResourceClassPatch"], () => require("./resourceClassPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:resource.k8s.io/v1alpha1:PodScheduling":
                return new exports.PodScheduling(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:PodSchedulingList":
                return new exports.PodSchedulingList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:PodSchedulingPatch":
                return new exports.PodSchedulingPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaim":
                return new exports.ResourceClaim(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaimList":
                return new exports.ResourceClaimList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaimPatch":
                return new exports.ResourceClaimPatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaimTemplate":
                return new exports.ResourceClaimTemplate(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaimTemplateList":
                return new exports.ResourceClaimTemplateList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClaimTemplatePatch":
                return new exports.ResourceClaimTemplatePatch(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClass":
                return new exports.ResourceClass(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClassList":
                return new exports.ResourceClassList(name, undefined, { urn });
            case "kubernetes:resource.k8s.io/v1alpha1:ResourceClassPatch":
                return new exports.ResourceClassPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "resource.k8s.io/v1alpha1", _module);
