// @pulumi/azure-native/devcenter/pool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pool = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * A pool of Virtual Machines.
 *
 * Uses Azure REST API version 2024-02-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-04-01, 2023-08-01-preview, 2023-10-01-preview, 2024-05-01-preview, 2024-06-01-preview, 2024-07-01-preview, 2024-08-01-preview, 2024-10-01-preview, 2025-02-01, 2025-04-01-preview, 2025-07-01-preview, 2025-10-01-preview, 2026-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native devcenter [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Pool extends pulumi.CustomResource {
    /**
     * Get an existing Pool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Pool(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:devcenter:Pool';
    /**
     * Returns true if the given object is an instance of Pool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Pool.__pulumiType;
    }
    /**
     * Create a Pool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.devBoxDefinitionName === undefined && !opts.urn) {
                throw new Error("Missing required property 'devBoxDefinitionName'");
            }
            if (args?.licenseType === undefined && !opts.urn) {
                throw new Error("Missing required property 'licenseType'");
            }
            if (args?.localAdministrator === undefined && !opts.urn) {
                throw new Error("Missing required property 'localAdministrator'");
            }
            if (args?.networkConnectionName === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkConnectionName'");
            }
            if (args?.projectName === undefined && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["devBoxDefinitionName"] = args?.devBoxDefinitionName;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["licenseType"] = args?.licenseType;
            resourceInputs["localAdministrator"] = args?.localAdministrator;
            resourceInputs["location"] = args?.location;
            resourceInputs["managedVirtualNetworkRegions"] = args?.managedVirtualNetworkRegions;
            resourceInputs["networkConnectionName"] = args?.networkConnectionName;
            resourceInputs["poolName"] = args?.poolName;
            resourceInputs["projectName"] = args?.projectName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["singleSignOnStatus"] = args?.singleSignOnStatus;
            resourceInputs["stopOnDisconnect"] = args?.stopOnDisconnect;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["virtualNetworkType"] = args?.virtualNetworkType;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["devBoxCount"] = undefined /*out*/;
            resourceInputs["healthStatus"] = undefined /*out*/;
            resourceInputs["healthStatusDetails"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["devBoxCount"] = undefined /*out*/;
            resourceInputs["devBoxDefinitionName"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["healthStatus"] = undefined /*out*/;
            resourceInputs["healthStatusDetails"] = undefined /*out*/;
            resourceInputs["licenseType"] = undefined /*out*/;
            resourceInputs["localAdministrator"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedVirtualNetworkRegions"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkConnectionName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["singleSignOnStatus"] = undefined /*out*/;
            resourceInputs["stopOnDisconnect"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworkType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devcenter/v20220801preview:Pool" }, { type: "azure-native:devcenter/v20220901preview:Pool" }, { type: "azure-native:devcenter/v20221012preview:Pool" }, { type: "azure-native:devcenter/v20221111preview:Pool" }, { type: "azure-native:devcenter/v20230101preview:Pool" }, { type: "azure-native:devcenter/v20230401:Pool" }, { type: "azure-native:devcenter/v20230801preview:Pool" }, { type: "azure-native:devcenter/v20231001preview:Pool" }, { type: "azure-native:devcenter/v20240201:Pool" }, { type: "azure-native:devcenter/v20240501preview:Pool" }, { type: "azure-native:devcenter/v20240601preview:Pool" }, { type: "azure-native:devcenter/v20240701preview:Pool" }, { type: "azure-native:devcenter/v20240801preview:Pool" }, { type: "azure-native:devcenter/v20241001preview:Pool" }, { type: "azure-native:devcenter/v20250201:Pool" }, { type: "azure-native:devcenter/v20250401preview:Pool" }, { type: "azure-native:devcenter/v20250701preview:Pool" }, { type: "azure-native:devcenter/v20251001preview:Pool" }, { type: "azure-native:devcenter/v20260101preview:Pool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Pool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Pool = Pool;
