// @pulumi/aws/codebuild/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.SourceCredential = exports.ResourcePolicy = exports.ReportGroup = exports.Project = exports.getFleetOutput = exports.getFleet = exports.Fleet = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.getFleet = null;
exports.getFleetOutput = null;
utilities.lazyLoad(exports, ["getFleet", "getFleetOutput"], () => require("./getFleet"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ReportGroup = null;
utilities.lazyLoad(exports, ["ReportGroup"], () => require("./reportGroup"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.SourceCredential = null;
utilities.lazyLoad(exports, ["SourceCredential"], () => require("./sourceCredential"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:codebuild/fleet:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "aws:codebuild/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws:codebuild/reportGroup:ReportGroup":
                return new exports.ReportGroup(name, undefined, { urn });
            case "aws:codebuild/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:codebuild/sourceCredential:SourceCredential":
                return new exports.SourceCredential(name, undefined, { urn });
            case "aws:codebuild/webhook:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "codebuild/fleet", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/project", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/reportGroup", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/sourceCredential", _module);
pulumi.runtime.registerResourceModule("aws", "codebuild/webhook", _module);
