// @pulumi/gcp/binaryauthorization/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = exports.getAttestorIamPolicyOutput = exports.getAttestorIamPolicy = exports.AttestorIamPolicy = exports.AttestorIamMember = exports.AttestorIamBinding = exports.Attestor = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Attestor = null;
utilities.lazyLoad(exports, ["Attestor"], () => require("./attestor"));
exports.AttestorIamBinding = null;
utilities.lazyLoad(exports, ["AttestorIamBinding"], () => require("./attestorIamBinding"));
exports.AttestorIamMember = null;
utilities.lazyLoad(exports, ["AttestorIamMember"], () => require("./attestorIamMember"));
exports.AttestorIamPolicy = null;
utilities.lazyLoad(exports, ["AttestorIamPolicy"], () => require("./attestorIamPolicy"));
exports.getAttestorIamPolicy = null;
exports.getAttestorIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAttestorIamPolicy", "getAttestorIamPolicyOutput"], () => require("./getAttestorIamPolicy"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:binaryauthorization/attestor:Attestor":
                return new exports.Attestor(name, undefined, { urn });
            case "gcp:binaryauthorization/attestorIamBinding:AttestorIamBinding":
                return new exports.AttestorIamBinding(name, undefined, { urn });
            case "gcp:binaryauthorization/attestorIamMember:AttestorIamMember":
                return new exports.AttestorIamMember(name, undefined, { urn });
            case "gcp:binaryauthorization/attestorIamPolicy:AttestorIamPolicy":
                return new exports.AttestorIamPolicy(name, undefined, { urn });
            case "gcp:binaryauthorization/policy:Policy":
                return new exports.Policy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "binaryauthorization/attestor", _module);
pulumi.runtime.registerResourceModule("gcp", "binaryauthorization/attestorIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "binaryauthorization/attestorIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "binaryauthorization/attestorIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "binaryauthorization/policy", _module);
