// @pulumi/gcp/networkconnectivity/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transport = exports.Spoke = exports.ServiceConnectionPolicy = exports.RegionalEndpoint = exports.PolicyBasedRoute = exports.MulticloudDataTransferConfig = exports.InternalRange = exports.HubIamPolicy = exports.HubIamMember = exports.HubIamBinding = exports.Hub = exports.Group = exports.getHubIamPolicyOutput = exports.getHubIamPolicy = exports.GatewayAdvertisedRoute = exports.Destination = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
exports.Destination = null;
utilities.lazyLoad(exports, ["Destination"], () => require("./destination"));
exports.GatewayAdvertisedRoute = null;
utilities.lazyLoad(exports, ["GatewayAdvertisedRoute"], () => require("./gatewayAdvertisedRoute"));
exports.getHubIamPolicy = null;
exports.getHubIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getHubIamPolicy", "getHubIamPolicyOutput"], () => require("./getHubIamPolicy"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.Hub = null;
utilities.lazyLoad(exports, ["Hub"], () => require("./hub"));
exports.HubIamBinding = null;
utilities.lazyLoad(exports, ["HubIamBinding"], () => require("./hubIamBinding"));
exports.HubIamMember = null;
utilities.lazyLoad(exports, ["HubIamMember"], () => require("./hubIamMember"));
exports.HubIamPolicy = null;
utilities.lazyLoad(exports, ["HubIamPolicy"], () => require("./hubIamPolicy"));
exports.InternalRange = null;
utilities.lazyLoad(exports, ["InternalRange"], () => require("./internalRange"));
exports.MulticloudDataTransferConfig = null;
utilities.lazyLoad(exports, ["MulticloudDataTransferConfig"], () => require("./multicloudDataTransferConfig"));
exports.PolicyBasedRoute = null;
utilities.lazyLoad(exports, ["PolicyBasedRoute"], () => require("./policyBasedRoute"));
exports.RegionalEndpoint = null;
utilities.lazyLoad(exports, ["RegionalEndpoint"], () => require("./regionalEndpoint"));
exports.ServiceConnectionPolicy = null;
utilities.lazyLoad(exports, ["ServiceConnectionPolicy"], () => require("./serviceConnectionPolicy"));
exports.Spoke = null;
utilities.lazyLoad(exports, ["Spoke"], () => require("./spoke"));
exports.Transport = null;
utilities.lazyLoad(exports, ["Transport"], () => require("./transport"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:networkconnectivity/destination:Destination":
                return new exports.Destination(name, undefined, { urn });
            case "gcp:networkconnectivity/gatewayAdvertisedRoute:GatewayAdvertisedRoute":
                return new exports.GatewayAdvertisedRoute(name, undefined, { urn });
            case "gcp:networkconnectivity/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "gcp:networkconnectivity/hub:Hub":
                return new exports.Hub(name, undefined, { urn });
            case "gcp:networkconnectivity/hubIamBinding:HubIamBinding":
                return new exports.HubIamBinding(name, undefined, { urn });
            case "gcp:networkconnectivity/hubIamMember:HubIamMember":
                return new exports.HubIamMember(name, undefined, { urn });
            case "gcp:networkconnectivity/hubIamPolicy:HubIamPolicy":
                return new exports.HubIamPolicy(name, undefined, { urn });
            case "gcp:networkconnectivity/internalRange:InternalRange":
                return new exports.InternalRange(name, undefined, { urn });
            case "gcp:networkconnectivity/multicloudDataTransferConfig:MulticloudDataTransferConfig":
                return new exports.MulticloudDataTransferConfig(name, undefined, { urn });
            case "gcp:networkconnectivity/policyBasedRoute:PolicyBasedRoute":
                return new exports.PolicyBasedRoute(name, undefined, { urn });
            case "gcp:networkconnectivity/regionalEndpoint:RegionalEndpoint":
                return new exports.RegionalEndpoint(name, undefined, { urn });
            case "gcp:networkconnectivity/serviceConnectionPolicy:ServiceConnectionPolicy":
                return new exports.ServiceConnectionPolicy(name, undefined, { urn });
            case "gcp:networkconnectivity/spoke:Spoke":
                return new exports.Spoke(name, undefined, { urn });
            case "gcp:networkconnectivity/transport:Transport":
                return new exports.Transport(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/destination", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/gatewayAdvertisedRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/group", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/hub", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/hubIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/hubIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/hubIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/internalRange", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/multicloudDataTransferConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/policyBasedRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/regionalEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/serviceConnectionPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/spoke", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/transport", _module);
