// @pulumi/gcp/networkconnectivity/transport.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transport = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * The Transport resource is a top-level resource used by customers to control Partner Cross-Cloud Interconnect (CCI) connections.
 *
 * > **Warning:** This resource is in beta, and should be used with the terraform-provider-google-beta provider.
 * See Provider Versions for more details on beta resources.
 *
 * To get more information about Transport, see:
 *
 * * [API documentation](https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/v1beta/projects.locations.transport)
 * * How-to Guides
 *     * [Official Documentation](https://docs.cloud.google.com/sdk/gcloud/reference/beta/network-connectivity/transports)
 *
 * ## Example Usage
 *
 * ### Network Connectivity Transport Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const project = gcp.organizations.getProject({});
 * const primary_network = new gcp.compute.Network("primary-network", {
 *     name: "tf-test-my-vpc-network_44154",
 *     autoCreateSubnetworks: false,
 * });
 * const primary = new gcp.networkconnectivity.Transport("primary", {
 *     name: "tf-test-basic-transport_90221",
 *     region: "us-east4",
 *     description: "A sample transport",
 *     remoteProfile: project.then(project => `https://networkconnectivity.googleapis.com/v1beta/${project.id}/locations/us-east4/remoteTransportProfiles/aws-us-east-1`),
 *     network: primary_network.name,
 *     bandwidth: "BPS_1G",
 *     remoteAccountId: "123",
 *     labels: {
 *         "label-one": "value-one",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Transport can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{region}}/transports/{{name}}`
 * * `{{project}}/{{region}}/{{name}}`
 * * `{{region}}/{{name}}`
 * * `{{name}}`
 *
 * When using the `pulumi import` command, Transport can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:networkconnectivity/transport:Transport default projects/{{project}}/locations/{{region}}/transports/{{name}}
 * $ pulumi import gcp:networkconnectivity/transport:Transport default {{project}}/{{region}}/{{name}}
 * $ pulumi import gcp:networkconnectivity/transport:Transport default {{region}}/{{name}}
 * $ pulumi import gcp:networkconnectivity/transport:Transport default {{name}}
 * ```
 */
class Transport extends pulumi.CustomResource {
    /**
     * Get an existing Transport resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Transport(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'gcp:networkconnectivity/transport:Transport';
    /**
     * Returns true if the given object is an instance of Transport.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Transport.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["adminEnabled"] = state?.adminEnabled;
            resourceInputs["advertisedRoutes"] = state?.advertisedRoutes;
            resourceInputs["autoAccept"] = state?.autoAccept;
            resourceInputs["bandwidth"] = state?.bandwidth;
            resourceInputs["deletionPolicy"] = state?.deletionPolicy;
            resourceInputs["description"] = state?.description;
            resourceInputs["effectiveLabels"] = state?.effectiveLabels;
            resourceInputs["generatedActivationKey"] = state?.generatedActivationKey;
            resourceInputs["hub"] = state?.hub;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["mtuLimit"] = state?.mtuLimit;
            resourceInputs["name"] = state?.name;
            resourceInputs["network"] = state?.network;
            resourceInputs["peeringNetwork"] = state?.peeringNetwork;
            resourceInputs["project"] = state?.project;
            resourceInputs["providedActivationKey"] = state?.providedActivationKey;
            resourceInputs["pscRoutingEnabled"] = state?.pscRoutingEnabled;
            resourceInputs["pulumiLabels"] = state?.pulumiLabels;
            resourceInputs["region"] = state?.region;
            resourceInputs["remoteAccountId"] = state?.remoteAccountId;
            resourceInputs["remoteProfile"] = state?.remoteProfile;
            resourceInputs["stackType"] = state?.stackType;
            resourceInputs["state"] = state?.state;
        }
        else {
            const args = argsOrState;
            if (args?.network === undefined && !opts.urn) {
                throw new Error("Missing required property 'network'");
            }
            if (args?.region === undefined && !opts.urn) {
                throw new Error("Missing required property 'region'");
            }
            if (args?.remoteProfile === undefined && !opts.urn) {
                throw new Error("Missing required property 'remoteProfile'");
            }
            resourceInputs["adminEnabled"] = args?.adminEnabled;
            resourceInputs["advertisedRoutes"] = args?.advertisedRoutes;
            resourceInputs["autoAccept"] = args?.autoAccept;
            resourceInputs["bandwidth"] = args?.bandwidth;
            resourceInputs["deletionPolicy"] = args?.deletionPolicy;
            resourceInputs["description"] = args?.description;
            resourceInputs["hub"] = args?.hub;
            resourceInputs["labels"] = args?.labels;
            resourceInputs["mtuLimit"] = args?.mtuLimit;
            resourceInputs["name"] = args?.name;
            resourceInputs["network"] = args?.network;
            resourceInputs["project"] = args?.project;
            resourceInputs["providedActivationKey"] = args?.providedActivationKey;
            resourceInputs["pscRoutingEnabled"] = args?.pscRoutingEnabled;
            resourceInputs["region"] = args?.region;
            resourceInputs["remoteAccountId"] = args?.remoteAccountId;
            resourceInputs["remoteProfile"] = args?.remoteProfile;
            resourceInputs["stackType"] = args?.stackType;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["generatedActivationKey"] = undefined /*out*/;
            resourceInputs["peeringNetwork"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Transport.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Transport = Transport;
