// @pulumi/azure-native/network/defaultAdminRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultAdminRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Network default admin rule.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 */
class DefaultAdminRule extends pulumi.CustomResource {
    /**
     * Get an existing DefaultAdminRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DefaultAdminRule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:network:DefaultAdminRule';
    /**
     * Returns true if the given object is an instance of DefaultAdminRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DefaultAdminRule.__pulumiType;
    }
    /**
     * Create a DefaultAdminRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.configurationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'configurationName'");
            }
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.networkManagerName === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkManagerName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.ruleCollectionName === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleCollectionName'");
            }
            resourceInputs["configurationName"] = args?.configurationName;
            resourceInputs["flag"] = args?.flag;
            resourceInputs["kind"] = "Default";
            resourceInputs["networkManagerName"] = args?.networkManagerName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["ruleCollectionName"] = args?.ruleCollectionName;
            resourceInputs["ruleName"] = args?.ruleName;
            resourceInputs["access"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["destinationPortRanges"] = undefined /*out*/;
            resourceInputs["destinations"] = undefined /*out*/;
            resourceInputs["direction"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["protocol"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["sourcePortRanges"] = undefined /*out*/;
            resourceInputs["sources"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["access"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["destinationPortRanges"] = undefined /*out*/;
            resourceInputs["destinations"] = undefined /*out*/;
            resourceInputs["direction"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["flag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["protocol"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["sourcePortRanges"] = undefined /*out*/;
            resourceInputs["sources"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20210201preview:AdminRule" }, { type: "azure-native:network/v20210201preview:DefaultAdminRule" }, { type: "azure-native:network/v20210501preview:AdminRule" }, { type: "azure-native:network/v20210501preview:DefaultAdminRule" }, { type: "azure-native:network/v20220101:DefaultAdminRule" }, { type: "azure-native:network/v20220201preview:DefaultAdminRule" }, { type: "azure-native:network/v20220401preview:DefaultAdminRule" }, { type: "azure-native:network/v20220501:DefaultAdminRule" }, { type: "azure-native:network/v20220701:DefaultAdminRule" }, { type: "azure-native:network/v20220901:DefaultAdminRule" }, { type: "azure-native:network/v20221101:DefaultAdminRule" }, { type: "azure-native:network/v20230201:AdminRule" }, { type: "azure-native:network/v20230201:DefaultAdminRule" }, { type: "azure-native:network/v20230401:AdminRule" }, { type: "azure-native:network/v20230401:DefaultAdminRule" }, { type: "azure-native:network/v20230501:AdminRule" }, { type: "azure-native:network/v20230501:DefaultAdminRule" }, { type: "azure-native:network/v20230601:AdminRule" }, { type: "azure-native:network/v20230601:DefaultAdminRule" }, { type: "azure-native:network/v20230901:AdminRule" }, { type: "azure-native:network/v20230901:DefaultAdminRule" }, { type: "azure-native:network/v20231101:AdminRule" }, { type: "azure-native:network/v20231101:DefaultAdminRule" }, { type: "azure-native:network/v20240101:AdminRule" }, { type: "azure-native:network/v20240101:DefaultAdminRule" }, { type: "azure-native:network/v20240101preview:AdminRule" }, { type: "azure-native:network/v20240101preview:DefaultAdminRule" }, { type: "azure-native:network/v20240301:AdminRule" }, { type: "azure-native:network/v20240301:DefaultAdminRule" }, { type: "azure-native:network/v20240501:AdminRule" }, { type: "azure-native:network/v20240501:DefaultAdminRule" }, { type: "azure-native:network/v20240701:DefaultAdminRule" }, { type: "azure-native:network/v20241001:DefaultAdminRule" }, { type: "azure-native:network/v20250101:DefaultAdminRule" }, { type: "azure-native:network/v20250301:DefaultAdminRule" }, { type: "azure-native:network/v20250501:DefaultAdminRule" }, { type: "azure-native:network:AdminRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DefaultAdminRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DefaultAdminRule = DefaultAdminRule;
