// @pulumi/azure-native/network/flowLog.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowLog = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("../types/input"));
const utilities = __importStar(require("../utilities"));
/**
 * A flow log resource.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2019-11-01, 2019-12-01, 2020-03-01, 2020-04-01, 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01, 2024-10-01, 2025-01-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class FlowLog extends pulumi.CustomResource {
    /**
     * Get an existing FlowLog resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FlowLog(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:network:FlowLog';
    /**
     * Returns true if the given object is an instance of FlowLog.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FlowLog.__pulumiType;
    }
    /**
     * Create a FlowLog resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.networkWatcherName === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkWatcherName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.storageId === undefined && !opts.urn) {
                throw new Error("Missing required property 'storageId'");
            }
            if (args?.targetResourceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetResourceId'");
            }
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["enabledFilteringCriteria"] = args?.enabledFilteringCriteria;
            resourceInputs["flowAnalyticsConfiguration"] = args?.flowAnalyticsConfiguration;
            resourceInputs["flowLogName"] = args?.flowLogName;
            resourceInputs["format"] = args ? (args.format ? pulumi.output(args.format).apply(inputs.network.flowLogFormatParametersArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["id"] = args?.id;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["location"] = args?.location;
            resourceInputs["networkWatcherName"] = args?.networkWatcherName;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["retentionPolicy"] = args ? (args.retentionPolicy ? pulumi.output(args.retentionPolicy).apply(inputs.network.retentionPolicyParametersArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["storageId"] = args?.storageId;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["targetResourceId"] = args?.targetResourceId;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["targetResourceGuid"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["enabledFilteringCriteria"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["flowAnalyticsConfiguration"] = undefined /*out*/;
            resourceInputs["format"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["retentionPolicy"] = undefined /*out*/;
            resourceInputs["storageId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetResourceGuid"] = undefined /*out*/;
            resourceInputs["targetResourceId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20191101:FlowLog" }, { type: "azure-native:network/v20191201:FlowLog" }, { type: "azure-native:network/v20200301:FlowLog" }, { type: "azure-native:network/v20200401:FlowLog" }, { type: "azure-native:network/v20200501:FlowLog" }, { type: "azure-native:network/v20200601:FlowLog" }, { type: "azure-native:network/v20200701:FlowLog" }, { type: "azure-native:network/v20200801:FlowLog" }, { type: "azure-native:network/v20201101:FlowLog" }, { type: "azure-native:network/v20210201:FlowLog" }, { type: "azure-native:network/v20210301:FlowLog" }, { type: "azure-native:network/v20210501:FlowLog" }, { type: "azure-native:network/v20210801:FlowLog" }, { type: "azure-native:network/v20220101:FlowLog" }, { type: "azure-native:network/v20220501:FlowLog" }, { type: "azure-native:network/v20220701:FlowLog" }, { type: "azure-native:network/v20220901:FlowLog" }, { type: "azure-native:network/v20221101:FlowLog" }, { type: "azure-native:network/v20230201:FlowLog" }, { type: "azure-native:network/v20230401:FlowLog" }, { type: "azure-native:network/v20230501:FlowLog" }, { type: "azure-native:network/v20230601:FlowLog" }, { type: "azure-native:network/v20230901:FlowLog" }, { type: "azure-native:network/v20231101:FlowLog" }, { type: "azure-native:network/v20240101:FlowLog" }, { type: "azure-native:network/v20240301:FlowLog" }, { type: "azure-native:network/v20240501:FlowLog" }, { type: "azure-native:network/v20240701:FlowLog" }, { type: "azure-native:network/v20241001:FlowLog" }, { type: "azure-native:network/v20250101:FlowLog" }, { type: "azure-native:network/v20250301:FlowLog" }, { type: "azure-native:network/v20250501:FlowLog" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FlowLog.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FlowLog = FlowLog;
