// @pulumi/azure-native/network/hubVirtualNetworkConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HubVirtualNetworkConnection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * HubVirtualNetworkConnection Resource.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01, 2024-10-01, 2025-01-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class HubVirtualNetworkConnection extends pulumi.CustomResource {
    /**
     * Get an existing HubVirtualNetworkConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HubVirtualNetworkConnection(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:network:HubVirtualNetworkConnection';
    /**
     * Returns true if the given object is an instance of HubVirtualNetworkConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HubVirtualNetworkConnection.__pulumiType;
    }
    /**
     * Create a HubVirtualNetworkConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if (args?.virtualHubName === undefined && !opts.urn) {
                throw new Error("Missing required property 'virtualHubName'");
            }
            resourceInputs["allowHubToRemoteVnetTransit"] = args?.allowHubToRemoteVnetTransit;
            resourceInputs["allowRemoteVnetToUseHubVnetGateways"] = args?.allowRemoteVnetToUseHubVnetGateways;
            resourceInputs["connectionName"] = args?.connectionName;
            resourceInputs["enableInternetSecurity"] = args?.enableInternetSecurity;
            resourceInputs["id"] = args?.id;
            resourceInputs["name"] = args?.name;
            resourceInputs["remoteVirtualNetwork"] = args?.remoteVirtualNetwork;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["routingConfiguration"] = args?.routingConfiguration;
            resourceInputs["virtualHubName"] = args?.virtualHubName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowHubToRemoteVnetTransit"] = undefined /*out*/;
            resourceInputs["allowRemoteVnetToUseHubVnetGateways"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["enableInternetSecurity"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["remoteVirtualNetwork"] = undefined /*out*/;
            resourceInputs["routingConfiguration"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20200501:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20200601:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20200701:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20200801:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20201101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20210201:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20210301:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20210501:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20210801:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20220101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20220501:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20220701:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20220901:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20221101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20230201:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20230401:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20230501:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20230601:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20230901:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20231101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20240101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20240301:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20240501:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20240701:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20241001:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20250101:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20250301:HubVirtualNetworkConnection" }, { type: "azure-native:network/v20250501:HubVirtualNetworkConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HubVirtualNetworkConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HubVirtualNetworkConnection = HubVirtualNetworkConnection;
