// @pulumi/azure-native/network/inboundNatRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InboundNatRule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Inbound NAT rule of the load balancer.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2018-06-01, 2018-07-01, 2018-08-01, 2018-10-01, 2018-11-01, 2018-12-01, 2019-02-01, 2019-04-01, 2019-06-01, 2019-07-01, 2019-08-01, 2019-09-01, 2019-11-01, 2019-12-01, 2020-03-01, 2020-04-01, 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01, 2024-10-01, 2025-01-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class InboundNatRule extends pulumi.CustomResource {
    /**
     * Get an existing InboundNatRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InboundNatRule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:network:InboundNatRule';
    /**
     * Returns true if the given object is an instance of InboundNatRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InboundNatRule.__pulumiType;
    }
    /**
     * Create a InboundNatRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.loadBalancerName === undefined && !opts.urn) {
                throw new Error("Missing required property 'loadBalancerName'");
            }
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["backendAddressPool"] = args?.backendAddressPool;
            resourceInputs["backendPort"] = args?.backendPort;
            resourceInputs["enableFloatingIP"] = args?.enableFloatingIP;
            resourceInputs["enableTcpReset"] = args?.enableTcpReset;
            resourceInputs["frontendIPConfiguration"] = args?.frontendIPConfiguration;
            resourceInputs["frontendPort"] = args?.frontendPort;
            resourceInputs["frontendPortRangeEnd"] = args?.frontendPortRangeEnd;
            resourceInputs["frontendPortRangeStart"] = args?.frontendPortRangeStart;
            resourceInputs["id"] = args?.id;
            resourceInputs["idleTimeoutInMinutes"] = args?.idleTimeoutInMinutes;
            resourceInputs["inboundNatRuleName"] = args?.inboundNatRuleName;
            resourceInputs["loadBalancerName"] = args?.loadBalancerName;
            resourceInputs["name"] = args?.name;
            resourceInputs["protocol"] = args?.protocol;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backendIPConfiguration"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backendAddressPool"] = undefined /*out*/;
            resourceInputs["backendIPConfiguration"] = undefined /*out*/;
            resourceInputs["backendPort"] = undefined /*out*/;
            resourceInputs["enableFloatingIP"] = undefined /*out*/;
            resourceInputs["enableTcpReset"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["frontendIPConfiguration"] = undefined /*out*/;
            resourceInputs["frontendPort"] = undefined /*out*/;
            resourceInputs["frontendPortRangeEnd"] = undefined /*out*/;
            resourceInputs["frontendPortRangeStart"] = undefined /*out*/;
            resourceInputs["idleTimeoutInMinutes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["protocol"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20170601:InboundNatRule" }, { type: "azure-native:network/v20170801:InboundNatRule" }, { type: "azure-native:network/v20170901:InboundNatRule" }, { type: "azure-native:network/v20171001:InboundNatRule" }, { type: "azure-native:network/v20171101:InboundNatRule" }, { type: "azure-native:network/v20180101:InboundNatRule" }, { type: "azure-native:network/v20180201:InboundNatRule" }, { type: "azure-native:network/v20180401:InboundNatRule" }, { type: "azure-native:network/v20180601:InboundNatRule" }, { type: "azure-native:network/v20180701:InboundNatRule" }, { type: "azure-native:network/v20180801:InboundNatRule" }, { type: "azure-native:network/v20181001:InboundNatRule" }, { type: "azure-native:network/v20181101:InboundNatRule" }, { type: "azure-native:network/v20181201:InboundNatRule" }, { type: "azure-native:network/v20190201:InboundNatRule" }, { type: "azure-native:network/v20190401:InboundNatRule" }, { type: "azure-native:network/v20190601:InboundNatRule" }, { type: "azure-native:network/v20190701:InboundNatRule" }, { type: "azure-native:network/v20190801:InboundNatRule" }, { type: "azure-native:network/v20190901:InboundNatRule" }, { type: "azure-native:network/v20191101:InboundNatRule" }, { type: "azure-native:network/v20191201:InboundNatRule" }, { type: "azure-native:network/v20200301:InboundNatRule" }, { type: "azure-native:network/v20200401:InboundNatRule" }, { type: "azure-native:network/v20200501:InboundNatRule" }, { type: "azure-native:network/v20200601:InboundNatRule" }, { type: "azure-native:network/v20200701:InboundNatRule" }, { type: "azure-native:network/v20200801:InboundNatRule" }, { type: "azure-native:network/v20201101:InboundNatRule" }, { type: "azure-native:network/v20210201:InboundNatRule" }, { type: "azure-native:network/v20210301:InboundNatRule" }, { type: "azure-native:network/v20210501:InboundNatRule" }, { type: "azure-native:network/v20210801:InboundNatRule" }, { type: "azure-native:network/v20220101:InboundNatRule" }, { type: "azure-native:network/v20220501:InboundNatRule" }, { type: "azure-native:network/v20220701:InboundNatRule" }, { type: "azure-native:network/v20220901:InboundNatRule" }, { type: "azure-native:network/v20221101:InboundNatRule" }, { type: "azure-native:network/v20230201:InboundNatRule" }, { type: "azure-native:network/v20230401:InboundNatRule" }, { type: "azure-native:network/v20230501:InboundNatRule" }, { type: "azure-native:network/v20230601:InboundNatRule" }, { type: "azure-native:network/v20230901:InboundNatRule" }, { type: "azure-native:network/v20231101:InboundNatRule" }, { type: "azure-native:network/v20240101:InboundNatRule" }, { type: "azure-native:network/v20240301:InboundNatRule" }, { type: "azure-native:network/v20240501:InboundNatRule" }, { type: "azure-native:network/v20240701:InboundNatRule" }, { type: "azure-native:network/v20241001:InboundNatRule" }, { type: "azure-native:network/v20250101:InboundNatRule" }, { type: "azure-native:network/v20250301:InboundNatRule" }, { type: "azure-native:network/v20250501:InboundNatRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InboundNatRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InboundNatRule = InboundNatRule;
