// @pulumi/azure-native/network/networkVirtualAppliance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkVirtualAppliance = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * NetworkVirtualAppliance Resource.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2019-12-01, 2020-03-01, 2020-04-01, 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01, 2024-10-01, 2025-01-01, 2025-03-01, 2025-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class NetworkVirtualAppliance extends pulumi.CustomResource {
    /**
     * Get an existing NetworkVirtualAppliance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkVirtualAppliance(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'azure-native:network:NetworkVirtualAppliance';
    /**
     * Returns true if the given object is an instance of NetworkVirtualAppliance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkVirtualAppliance.__pulumiType;
    }
    /**
     * Create a NetworkVirtualAppliance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.resourceGroupName === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["additionalNics"] = args?.additionalNics;
            resourceInputs["bootStrapConfigurationBlobs"] = args?.bootStrapConfigurationBlobs;
            resourceInputs["cloudInitConfiguration"] = args?.cloudInitConfiguration;
            resourceInputs["cloudInitConfigurationBlobs"] = args?.cloudInitConfigurationBlobs;
            resourceInputs["delegation"] = args?.delegation;
            resourceInputs["id"] = args?.id;
            resourceInputs["identity"] = args?.identity;
            resourceInputs["internetIngressPublicIps"] = args?.internetIngressPublicIps;
            resourceInputs["location"] = args?.location;
            resourceInputs["networkProfile"] = args?.networkProfile;
            resourceInputs["networkVirtualApplianceName"] = args?.networkVirtualApplianceName;
            resourceInputs["nvaSku"] = args?.nvaSku;
            resourceInputs["resourceGroupName"] = args?.resourceGroupName;
            resourceInputs["sshPublicKey"] = args?.sshPublicKey;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["virtualApplianceAsn"] = args?.virtualApplianceAsn;
            resourceInputs["virtualHub"] = args?.virtualHub;
            resourceInputs["addressPrefix"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentType"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["inboundSecurityRules"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partnerManagedResource"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualApplianceConnections"] = undefined /*out*/;
            resourceInputs["virtualApplianceNics"] = undefined /*out*/;
            resourceInputs["virtualApplianceSites"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalNics"] = undefined /*out*/;
            resourceInputs["addressPrefix"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bootStrapConfigurationBlobs"] = undefined /*out*/;
            resourceInputs["cloudInitConfiguration"] = undefined /*out*/;
            resourceInputs["cloudInitConfigurationBlobs"] = undefined /*out*/;
            resourceInputs["delegation"] = undefined /*out*/;
            resourceInputs["deploymentType"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["inboundSecurityRules"] = undefined /*out*/;
            resourceInputs["internetIngressPublicIps"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkProfile"] = undefined /*out*/;
            resourceInputs["nvaSku"] = undefined /*out*/;
            resourceInputs["partnerManagedResource"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sshPublicKey"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualApplianceAsn"] = undefined /*out*/;
            resourceInputs["virtualApplianceConnections"] = undefined /*out*/;
            resourceInputs["virtualApplianceNics"] = undefined /*out*/;
            resourceInputs["virtualApplianceSites"] = undefined /*out*/;
            resourceInputs["virtualHub"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20191201:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200301:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200401:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200501:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200601:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200701:NetworkVirtualAppliance" }, { type: "azure-native:network/v20200801:NetworkVirtualAppliance" }, { type: "azure-native:network/v20201101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20210201:NetworkVirtualAppliance" }, { type: "azure-native:network/v20210301:NetworkVirtualAppliance" }, { type: "azure-native:network/v20210501:NetworkVirtualAppliance" }, { type: "azure-native:network/v20210801:NetworkVirtualAppliance" }, { type: "azure-native:network/v20220101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20220501:NetworkVirtualAppliance" }, { type: "azure-native:network/v20220701:NetworkVirtualAppliance" }, { type: "azure-native:network/v20220901:NetworkVirtualAppliance" }, { type: "azure-native:network/v20221101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20230201:NetworkVirtualAppliance" }, { type: "azure-native:network/v20230401:NetworkVirtualAppliance" }, { type: "azure-native:network/v20230501:NetworkVirtualAppliance" }, { type: "azure-native:network/v20230601:NetworkVirtualAppliance" }, { type: "azure-native:network/v20230901:NetworkVirtualAppliance" }, { type: "azure-native:network/v20231101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20240101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20240301:NetworkVirtualAppliance" }, { type: "azure-native:network/v20240501:NetworkVirtualAppliance" }, { type: "azure-native:network/v20240701:NetworkVirtualAppliance" }, { type: "azure-native:network/v20241001:NetworkVirtualAppliance" }, { type: "azure-native:network/v20250101:NetworkVirtualAppliance" }, { type: "azure-native:network/v20250301:NetworkVirtualAppliance" }, { type: "azure-native:network/v20250501:NetworkVirtualAppliance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkVirtualAppliance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkVirtualAppliance = NetworkVirtualAppliance;
